{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe3f3f5c",
   "metadata": {},
   "source": [
    "# Identication of putative contaminant taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "df5c0a42",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tempdir <- function() {return(\"~/.Rtmp\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "440f3650",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.3     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.0.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Loading required package: ANCOMBC\n",
      "\n",
      "Loading required package: phyloseq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(ANCOMBC)\n",
    "require(phyloseq)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "beebb9f2",
   "metadata": {},
   "source": [
    "### Create Phyloseq object"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e78f0ab1",
   "metadata": {},
   "source": [
    "#### Load abundance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d8cd3ad6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_data <- function(file_path) {\n",
    "    df <- fread(file_path) %>%\n",
    "        separate(sample, into = c(NA, \"npm_research_id\"), sep = \"\\\\.\")\n",
    "    return(df)\n",
    "}\n",
    "\n",
    "df_to_phy <- function(df) {\n",
    "    df <- as.matrix(df, rownames=1)\n",
    "    df <- otu_table(df, taxa_are_rows = F)\n",
    "    return(df)\n",
    "}\n",
    "\n",
    "df <- load_data(str_glue(\"data/temp_files_9999/07_abundance_matrix/abundance_matrix.subset_9999.S.tsv\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1659d251",
   "metadata": {},
   "source": [
    "#### Load metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4b5109a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_metadata <- function(file_path, df) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% df$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"\")\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "meta_to_phy <- function(meta) {\n",
    "    meta <- sample_data(meta_filt %>% column_to_rownames(\"npm_research_id\"))\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f99f62b",
   "metadata": {},
   "source": [
    "#### Get metadata subsets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9a00d0e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_metadata <- function(meta, n_subset) {\n",
    "    meta <- as.data.frame(meta)\n",
    "    cohorts <- unique(meta$site_supplying_sample)\n",
    "    subset_vec <- c()\n",
    "\n",
    "    for (i in cohorts) {\n",
    "        ids <- meta$npm_research_id[meta$site_supplying_sample == i]\n",
    "        \n",
    "        if (length(ids) > n_subset) {\n",
    "            subset_ids <- sample(ids, n_subset)\n",
    "            subset_vec <- c(subset_vec, subset_ids)\n",
    "        } else {\n",
    "            subset_vec <- c(subset_vec, ids)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    meta_sub <- meta %>%\n",
    "      filter(npm_research_id %in% subset_vec) \n",
    "    return(meta_sub)\n",
    "}\n",
    "\n",
    "meta_filt <- subset_metadata(meta, 10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6487980c",
   "metadata": {},
   "source": [
    "#### Retrieve rows in abundance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7d1b3c4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "retrieve_rows_from_meta <- function(df, meta) {\n",
    "    return(df %>% filter(npm_research_id %in% meta$npm_research_id)) \n",
    "}\n",
    "\n",
    "df_filt <- retrieve_rows_from_meta(df, meta_filt)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38190e36",
   "metadata": {},
   "source": [
    "#### Remove human and unclassified reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fe7b5779",
   "metadata": {},
   "outputs": [],
   "source": [
    "remove_cols <- function(df, col_to_exclude) {\n",
    "    return(df %>% select(-all_of(col_to_exclude)))\n",
    "}\n",
    "\n",
    "df_filt2 <- remove_cols(df_filt, c(\"Homo sapiens\", \"unclassified\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26bbe264",
   "metadata": {},
   "source": [
    "#### Remove taxa present in <~x% of samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6c6f4cff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1576 / 5200 taxa are present in 0.05 of samples\n"
     ]
    }
   ],
   "source": [
    "remove_low_freq_taxa <- function(df) {\n",
    "    n_original <- ncol(df)\n",
    "    frac_presence <- 0.05\n",
    "    PA_df <- apply(df[, 2:ncol(df)], 2, function(x) {ifelse(x > 0, T, F)})\n",
    "    frac_df <- apply(PA_df, 2, function(x) {sum(x) / nrow(PA_df)})\n",
    "    to_keep <- names(frac_df[frac_df > frac_presence])\n",
    "    to_keep <- c(\"npm_research_id\", to_keep)\n",
    "    print(str_glue(\"{length(to_keep)} / {n_original} taxa are present in {frac_presence} of samples\"))\n",
    "    return(df %>% select(all_of(to_keep)))\n",
    "}\n",
    "\n",
    "df_filt3 <- remove_low_freq_taxa(df_filt2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcbb9a9f",
   "metadata": {},
   "source": [
    "#### Plot metadata barcharts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c8f23c92",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_metadata_plots <- function(meta, meta_cols) {\n",
    "    meta <- as.matrix(meta)\n",
    "    meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "    plots <- list()\n",
    "    \n",
    "    for (column in meta_cols) {\n",
    "        plt <- meta %>%\n",
    "            mutate(across(everything(), as.character)) %>%\n",
    "            select(all_of(column)) %>%\n",
    "            group_by_at(column) %>%\n",
    "            summarise(n = n()) %>%\n",
    "            ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "                geom_bar(stat = \"identity\") +\n",
    "                theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "                geom_text(aes_string(label = \"n\"))\n",
    "        plots[[column]] <- plt\n",
    "    }\n",
    "    \n",
    "    return(plots)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f422731d",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_meta_cols <- function(meta) {\n",
    "    meta_cols <- colnames(meta)[grep(\"kit|flow_cell|instrument_id|site_supplying\", colnames(meta))]\n",
    "    to_exclude <- c(\"library_prep_kit\", \"hiseq_xtm_flow_cell_v2_5_id\")\n",
    "    meta_cols <- meta_cols[!(meta_cols %in% to_exclude)]\n",
    "    return(meta_cols)\n",
    "}\n",
    "\n",
    "meta_cols <- get_meta_cols(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6386ff0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"site_supplying_sample\"\n",
      "\u001b[90m# A tibble: 7 × 2\u001b[39m\n",
      "  `get(col)`     n\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m1\u001b[39m GUSTO         10\n",
      "\u001b[90m2\u001b[39m HELIOS        10\n",
      "\u001b[90m3\u001b[39m MEC           10\n",
      "\u001b[90m4\u001b[39m PRISM         10\n",
      "\u001b[90m5\u001b[39m SERI          10\n",
      "\u001b[90m6\u001b[39m SSMP          10\n",
      "\u001b[90m7\u001b[39m TTSH          10\n",
      "[1] \"extraction_kit\"\n",
      "\u001b[90m# A tibble: 6 × 2\u001b[39m\n",
      "  `get(col)`                                         n\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                          \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m1\u001b[39m abGENIX Whole Blood Genomic DNA Extraction Kit    10\n",
      "\u001b[90m2\u001b[39m Chemagic DNA Blood Kit (Perkin Elmer, MA)          8\n",
      "\u001b[90m3\u001b[39m Maxwell RSC Blood DNA Kit (AS1400)                10\n",
      "\u001b[90m4\u001b[39m QIAamp DNA Blood Mini Kit                         10\n",
      "\u001b[90m5\u001b[39m QIAsymphony DSP DNA Midi Kit                      10\n",
      "\u001b[90m6\u001b[39m Unknown                                           22\n",
      "[1] \"instrument_id\"\n",
      "\u001b[90m# A tibble: 6 × 2\u001b[39m\n",
      "  `get(col)`     n\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m1\u001b[39m NG001         10\n",
      "\u001b[90m2\u001b[39m NG002          6\n",
      "\u001b[90m3\u001b[39m NG003          6\n",
      "\u001b[90m4\u001b[39m NG004          8\n",
      "\u001b[90m5\u001b[39m NG005          7\n",
      "\u001b[90m6\u001b[39m Unknown       33\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "\u001b[90m# A tibble: 19 × 2\u001b[39m\n",
      "   `get(col)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20293078       1\n",
      "\u001b[90m 2\u001b[39m 20304523       2\n",
      "\u001b[90m 3\u001b[39m 20305982       2\n",
      "\u001b[90m 4\u001b[39m 20314080       1\n",
      "\u001b[90m 5\u001b[39m 20319700       2\n",
      "\u001b[90m 6\u001b[39m 20324969       2\n",
      "\u001b[90m 7\u001b[39m 20331242       1\n",
      "\u001b[90m 8\u001b[39m 20336598       3\n",
      "\u001b[90m 9\u001b[39m 20340298       5\n",
      "\u001b[90m10\u001b[39m 20340301       1\n",
      "\u001b[90m11\u001b[39m 20363558       6\n",
      "\u001b[90m12\u001b[39m 20364059       1\n",
      "\u001b[90m13\u001b[39m 20371255       2\n",
      "\u001b[90m14\u001b[39m 20374156       1\n",
      "\u001b[90m15\u001b[39m 20382082       1\n",
      "\u001b[90m16\u001b[39m 20392970       3\n",
      "\u001b[90m17\u001b[39m 20392972       1\n",
      "\u001b[90m18\u001b[39m 20396095       2\n",
      "\u001b[90m19\u001b[39m Unknown       33\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "\u001b[90m# A tibble: 16 × 2\u001b[39m\n",
      "   `get(col)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20298588       3\n",
      "\u001b[90m 2\u001b[39m 20314829       1\n",
      "\u001b[90m 3\u001b[39m 20314832       3\n",
      "\u001b[90m 4\u001b[39m 20316938       1\n",
      "\u001b[90m 5\u001b[39m 20331251       2\n",
      "\u001b[90m 6\u001b[39m 20336632       3\n",
      "\u001b[90m 7\u001b[39m 20340313       2\n",
      "\u001b[90m 8\u001b[39m 20345691       4\n",
      "\u001b[90m 9\u001b[39m 20354529       1\n",
      "\u001b[90m10\u001b[39m 20364085       6\n",
      "\u001b[90m11\u001b[39m 20374150       4\n",
      "\u001b[90m12\u001b[39m 20382958       1\n",
      "\u001b[90m13\u001b[39m 20392975       3\n",
      "\u001b[90m14\u001b[39m 20393199       2\n",
      "\u001b[90m15\u001b[39m 20397673       1\n",
      "\u001b[90m16\u001b[39m Unknown       33\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "\u001b[90m# A tibble: 18 × 2\u001b[39m\n",
      "   `get(col)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20285695       3\n",
      "\u001b[90m 2\u001b[39m 20291660       1\n",
      "\u001b[90m 3\u001b[39m 20303896       2\n",
      "\u001b[90m 4\u001b[39m 20309873       2\n",
      "\u001b[90m 5\u001b[39m 20320784       3\n",
      "\u001b[90m 6\u001b[39m 20324724       3\n",
      "\u001b[90m 7\u001b[39m 20337254       1\n",
      "\u001b[90m 8\u001b[39m 20344239       5\n",
      "\u001b[90m 9\u001b[39m 20348434       1\n",
      "\u001b[90m10\u001b[39m 20350208       3\n",
      "\u001b[90m11\u001b[39m 20356806       4\n",
      "\u001b[90m12\u001b[39m 20356807       1\n",
      "\u001b[90m13\u001b[39m 20357599       1\n",
      "\u001b[90m14\u001b[39m 20363360       3\n",
      "\u001b[90m15\u001b[39m 20365297       1\n",
      "\u001b[90m16\u001b[39m 20368191       2\n",
      "\u001b[90m17\u001b[39m 20377556       1\n",
      "\u001b[90m18\u001b[39m Unknown       33\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "\u001b[90m# A tibble: 18 × 2\u001b[39m\n",
      "   `get(col)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20300585       3\n",
      "\u001b[90m 2\u001b[39m 20305231       1\n",
      "\u001b[90m 3\u001b[39m 20315909       3\n",
      "\u001b[90m 4\u001b[39m 20320911       1\n",
      "\u001b[90m 5\u001b[39m 20331193       2\n",
      "\u001b[90m 6\u001b[39m 20331198       4\n",
      "\u001b[90m 7\u001b[39m 20340288       3\n",
      "\u001b[90m 8\u001b[39m 20345688       2\n",
      "\u001b[90m 9\u001b[39m 20355393       1\n",
      "\u001b[90m10\u001b[39m 20364080       1\n",
      "\u001b[90m11\u001b[39m 20364081       1\n",
      "\u001b[90m12\u001b[39m 20364082       1\n",
      "\u001b[90m13\u001b[39m 20371249       5\n",
      "\u001b[90m14\u001b[39m 20376900       2\n",
      "\u001b[90m15\u001b[39m 20382943       1\n",
      "\u001b[90m16\u001b[39m 20385639       3\n",
      "\u001b[90m17\u001b[39m 20392969       3\n",
      "\u001b[90m18\u001b[39m Unknown       33\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "\u001b[90m# A tibble: 20 × 2\u001b[39m\n",
      "   `get(col)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20285943       3\n",
      "\u001b[90m 2\u001b[39m 20294379       1\n",
      "\u001b[90m 3\u001b[39m 20302809       1\n",
      "\u001b[90m 4\u001b[39m 20302937       2\n",
      "\u001b[90m 5\u001b[39m 20308248       1\n",
      "\u001b[90m 6\u001b[39m 20316761       1\n",
      "\u001b[90m 7\u001b[39m 20316762       1\n",
      "\u001b[90m 8\u001b[39m 20328452       4\n",
      "\u001b[90m 9\u001b[39m 20335773       1\n",
      "\u001b[90m10\u001b[39m 20335774       3\n",
      "\u001b[90m11\u001b[39m 20347583       8\n",
      "\u001b[90m12\u001b[39m 20347589       1\n",
      "\u001b[90m13\u001b[39m 20350041       1\n",
      "\u001b[90m14\u001b[39m 20350042       2\n",
      "\u001b[90m15\u001b[39m 20358945       1\n",
      "\u001b[90m16\u001b[39m 20367079       2\n",
      "\u001b[90m17\u001b[39m 20367085       1\n",
      "\u001b[90m18\u001b[39m 20393560       2\n",
      "\u001b[90m19\u001b[39m 20394822       1\n",
      "\u001b[90m20\u001b[39m Unknown       33\n"
     ]
    }
   ],
   "source": [
    "print_group_freqs <- function(meta, meta_cols) {\n",
    "    for (col in meta_cols) {\n",
    "        tmp <- tibble(data.frame(meta)) %>%\n",
    "            group_by(get(col)) %>%\n",
    "            summarise(n = n())\n",
    "        print(col)\n",
    "        print(tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "print_group_freqs(meta_filt, meta_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1116f32",
   "metadata": {},
   "source": [
    "### Run ANCOM-BC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2393d8ca",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "run_ancom_bc <- function(phy_fn, meta_fn, meta_cols, n_samples, max_iter) {\n",
    "    result_df <- tibble()\n",
    "\n",
    "    for (column in meta_cols) {\n",
    "        print(column)\n",
    "#         column <- \"extraction_kit\"\n",
    "        # Remove levels with < x samples\n",
    "        tmp <- tibble(data.frame(meta_fn)) %>%\n",
    "            group_by(get(column)) %>%\n",
    "            summarise(n = n())\n",
    "        \n",
    "        print(tmp)\n",
    "\n",
    "        # Vector of group levels to keep\n",
    "        to_keep <- tmp[tmp$n >= n_samples, \"get(column)\"]$`get(column)`\n",
    "        to_keep <- to_keep[to_keep != \"Unknown\"]\n",
    "        \n",
    "        print(length(to_keep))\n",
    "\n",
    "        # Retrieve groupings\n",
    "        var_values <- sample_data(phy_fn)[[column]]\n",
    "\n",
    "        # Prune groups\n",
    "        phy_subset <- prune_samples(var_values %in% to_keep, phy_fn)\n",
    "        n_filt <- nrow(phy_subset)\n",
    "        print(str_glue(\"There are {n_filt} samples after pruning metadata levels\"))\n",
    "        \n",
    "        out <- ancombc(phyloseq = phy_subset, formula = column,\n",
    "                      p_adj_method = \"BH\", zero_cut = 1, lib_cut = 0,\n",
    "                      group = column, struc_zero = FALSE, neg_lb = FALSE,\n",
    "                      tol = 1e-5, max_iter = max_iter, conserve = FALSE,\n",
    "                      alpha = 0.05, global = TRUE)\n",
    "        \n",
    "        morsel <- out$res_global %>% mutate(meta_col = column)\n",
    "        \n",
    "        result_df <- result_df %>% bind_rows(morsel)\n",
    "    }\n",
    "    \n",
    "    return(result_df %>% rownames_to_column(\"taxa\"))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94d384b2",
   "metadata": {},
   "source": [
    "#### Decontamination procedure for each subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "60431428",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1747 / 5200 taxa are present in 0.05 of samples\n",
      "[1] \"site_supplying_sample\"\n",
      "\u001b[90m# A tibble: 7 × 2\u001b[39m\n",
      "  `get(column)`     n\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m1\u001b[39m GUSTO           968\n",
      "\u001b[90m2\u001b[39m HELIOS         \u001b[4m2\u001b[24m285\n",
      "\u001b[90m3\u001b[39m MEC            \u001b[4m2\u001b[24m787\n",
      "\u001b[90m4\u001b[39m PRISM          \u001b[4m1\u001b[24m248\n",
      "\u001b[90m5\u001b[39m SERI           \u001b[4m1\u001b[24m436\n",
      "\u001b[90m6\u001b[39m SSMP             63\n",
      "\u001b[90m7\u001b[39m TTSH            919\n",
      "[1] 7\n",
      "\n",
      "[1] \"extraction_kit\"\n",
      "\u001b[90m# A tibble: 7 × 2\u001b[39m\n",
      "  `get(column)`                                      n\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                          \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m1\u001b[39m abGENIX Whole Blood Genomic DNA Extraction Kit  \u001b[4m2\u001b[24m281\n",
      "\u001b[90m2\u001b[39m Chemagic DNA Blood Kit (Perkin Elmer, MA)       \u001b[4m1\u001b[24m040\n",
      "\u001b[90m3\u001b[39m Maxwell RSC Blood DNA Kit (AS1400)              \u001b[4m2\u001b[24m117\n",
      "\u001b[90m4\u001b[39m QIAamp DNA Blood Mini Kit                        919\n",
      "\u001b[90m5\u001b[39m QIAsymphony DSP DNA Midi Kit                     968\n",
      "\u001b[90m6\u001b[39m QIAsymphony DSP DNA Mini Kit (192)               670\n",
      "\u001b[90m7\u001b[39m Unknown                                         \u001b[4m1\u001b[24m711\n",
      "[1] 6\n",
      "\n",
      "[1] \"instrument_id\"\n",
      "\u001b[90m# A tibble: 6 × 2\u001b[39m\n",
      "  `get(column)`     n\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m1\u001b[39m NG001          \u001b[4m1\u001b[24m335\n",
      "\u001b[90m2\u001b[39m NG002          \u001b[4m1\u001b[24m221\n",
      "\u001b[90m3\u001b[39m NG003          \u001b[4m1\u001b[24m282\n",
      "\u001b[90m4\u001b[39m NG004          \u001b[4m1\u001b[24m516\n",
      "\u001b[90m5\u001b[39m NG005          \u001b[4m1\u001b[24m638\n",
      "\u001b[90m6\u001b[39m Unknown        \u001b[4m2\u001b[24m714\n",
      "[1] 5\n",
      "\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "\u001b[90m# A tibble: 34 × 2\u001b[39m\n",
      "   `get(column)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20235982         20\n",
      "\u001b[90m 2\u001b[39m 20293078         46\n",
      "\u001b[90m 3\u001b[39m 20304523        238\n",
      "\u001b[90m 4\u001b[39m 20305232         10\n",
      "\u001b[90m 5\u001b[39m 20305982        148\n",
      "\u001b[90m 6\u001b[39m 20314080        197\n",
      "\u001b[90m 7\u001b[39m 20314082        329\n",
      "\u001b[90m 8\u001b[39m 20319700        186\n",
      "\u001b[90m 9\u001b[39m 20322799         19\n",
      "\u001b[90m10\u001b[39m 20324969        342\n",
      "\u001b[90m# … with 24 more rows\u001b[39m\n",
      "[1] 33\n",
      "\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "\u001b[90m# A tibble: 34 × 2\u001b[39m\n",
      "   `get(column)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20277312          8\n",
      "\u001b[90m 2\u001b[39m 20295428         11\n",
      "\u001b[90m 3\u001b[39m 20298588        193\n",
      "\u001b[90m 4\u001b[39m 20305983         10\n",
      "\u001b[90m 5\u001b[39m 20314829        231\n",
      "\u001b[90m 6\u001b[39m 20314830        204\n",
      "\u001b[90m 7\u001b[39m 20314832        314\n",
      "\u001b[90m 8\u001b[39m 20316938         70\n",
      "\u001b[90m 9\u001b[39m 20316939        146\n",
      "\u001b[90m10\u001b[39m 20325219          5\n",
      "\u001b[90m# … with 24 more rows\u001b[39m\n",
      "[1] 33\n",
      "\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "\u001b[90m# A tibble: 29 × 2\u001b[39m\n",
      "   `get(column)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20265537         38\n",
      "\u001b[90m 2\u001b[39m 20285695        179\n",
      "\u001b[90m 3\u001b[39m 20286254         10\n",
      "\u001b[90m 4\u001b[39m 20290688         10\n",
      "\u001b[90m 5\u001b[39m 20291660        234\n",
      "\u001b[90m 6\u001b[39m 20303896        344\n",
      "\u001b[90m 7\u001b[39m 20309873        222\n",
      "\u001b[90m 8\u001b[39m 20315030        138\n",
      "\u001b[90m 9\u001b[39m 20315031         10\n",
      "\u001b[90m10\u001b[39m 20320784        331\n",
      "\u001b[90m# … with 19 more rows\u001b[39m\n",
      "[1] 28\n",
      "\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "\u001b[90m# A tibble: 33 × 2\u001b[39m\n",
      "   `get(column)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20295412         10\n",
      "\u001b[90m 2\u001b[39m 20300585        164\n",
      "\u001b[90m 3\u001b[39m 20305231        133\n",
      "\u001b[90m 4\u001b[39m 20305612          9\n",
      "\u001b[90m 5\u001b[39m 20307071         10\n",
      "\u001b[90m 6\u001b[39m 20308845        152\n",
      "\u001b[90m 7\u001b[39m 20311423        208\n",
      "\u001b[90m 8\u001b[39m 20315909        267\n",
      "\u001b[90m 9\u001b[39m 20320911        246\n",
      "\u001b[90m10\u001b[39m 20329385        136\n",
      "\u001b[90m# … with 23 more rows\u001b[39m\n",
      "[1] 32\n",
      "\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "\u001b[90m# A tibble: 36 × 2\u001b[39m\n",
      "   `get(column)`     n\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m         \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m 20285943        174\n",
      "\u001b[90m 2\u001b[39m 20294378         10\n",
      "\u001b[90m 3\u001b[39m 20294379        265\n",
      "\u001b[90m 4\u001b[39m 20302809        146\n",
      "\u001b[90m 5\u001b[39m 20302932         48\n",
      "\u001b[90m 6\u001b[39m 20302937        329\n",
      "\u001b[90m 7\u001b[39m 20308245         65\n",
      "\u001b[90m 8\u001b[39m 20308248        164\n",
      "\u001b[90m 9\u001b[39m 20316761        157\n",
      "\u001b[90m10\u001b[39m 20316762        153\n",
      "\u001b[90m# … with 26 more rows\u001b[39m\n",
      "[1] 35\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 13968 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>taxa</th><th scope=col>W</th><th scope=col>p_val</th><th scope=col>q_val</th><th scope=col>diff_abn</th><th scope=col>meta_col</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Paraburkholderia fungorum...1                </td><td> 2369.5949</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Paraburkholderia xenovorans...2              </td><td> 1431.8126</td><td>6.262869e-306</td><td>8.064137e-306</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Paraburkholderia hospita...3                 </td><td>  660.2795</td><td>4.593525e-139</td><td>4.828594e-139</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Ralstonia pickettii...4                      </td><td> 3803.0451</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Ralstonia insidiosa...5                      </td><td>11006.3824</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Ralstonia mannitolilytica...6                </td><td> 4609.9584</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Ralstonia solanacearum...7                   </td><td> 5302.4572</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia cepacia...8                     </td><td> 4335.3477</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia sp. LA-2-3-30-S1-D2...9         </td><td> 2308.2847</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia metallica...10                  </td><td> 1130.0050</td><td>1.343308e-240</td><td>1.571995e-240</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia contaminans...11                </td><td> 2846.8788</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia gladioli...12                   </td><td> 9847.3649</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia thailandensis...13              </td><td>  729.6536</td><td>4.833906e-154</td><td>5.162079e-154</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Burkholderia sp. OLGA172...14                </td><td> 1222.5535</td><td>1.258281e-260</td><td>1.513057e-260</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bordetella genomosp. 8...15                  </td><td> 1835.6105</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bordetella genomosp. 13...16                 </td><td>14540.9158</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Acidovorax sp. RAC01...17                    </td><td> 3324.5010</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Acidovorax sp. 1608163...18                  </td><td> 3620.3009</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Diaphorobacter polyhydroxybutyrativorans...19</td><td>  929.3545</td><td>3.383759e-197</td><td>3.775108e-197</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Massilia sp. WG5...20                        </td><td> 8254.7258</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Aromatoleum aromaticum...21                  </td><td> 4071.0045</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. SK17...22                 </td><td>58167.3505</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. BTAi1...23                </td><td>46329.7238</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium erythrophlei...24             </td><td>35114.9066</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium icense...25                   </td><td>36656.2486</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium diazoefficiens...26           </td><td>46012.8786</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. WSM471...27               </td><td>10420.0661</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Bradyrhizobium ottawaense...28               </td><td>26795.8880</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Rhodopseudomonas palustris...29              </td><td>97958.7439</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>Oligotropha carboxidovorans...30             </td><td>30202.0768</td><td> 0.000000e+00</td><td> 0.000000e+00</td><td>TRUE</td><td>site_supplying_sample</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Corynebacterium provencense...13939         </td><td>7367.846</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Corynebacterium glutamicum...13940          </td><td>8200.992</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Corynebacterium falsenii...13941            </td><td>6171.133</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Kitasatospora albolonga...13942             </td><td>4768.064</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Trueperella pyogenes...13943                </td><td>8182.139</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Gordonibacter pamelaeae...13944             </td><td>7645.546</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Deinococcus sp. S14-83...13945              </td><td>5823.598</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Chlorobaculum limnaeum...13946              </td><td>7838.732</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Gluconobacter albidus...13947               </td><td>5706.178</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Acetobacter ghanensis...13948               </td><td>6582.644</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Citrobacter koseri...13949                  </td><td>5088.732</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Adlercreutzia equolifaciens...13950         </td><td>5976.559</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Chlorobium phaeobacteroides...13951         </td><td>6001.430</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Aquaspirillum sp. LM1...13952               </td><td>5439.753</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Acinetobacter bereziniae...13953            </td><td>7144.056</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Pectobacterium parmentieri...13954          </td><td>7895.328</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Acidihalobacter ferrooxidans...13955        </td><td>4915.455</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Bifidobacterium choerinum...13956           </td><td>7063.347</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Stenotrophomonas sp. SAU14A_NAIMI4_8...13957</td><td>4801.020</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Desulfovibrio desulfuricans...13958         </td><td>6822.081</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Corynebacterium epidermidicanis...13959     </td><td>6587.287</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Sulfuricella denitrificans...13960          </td><td>7495.815</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Asaia bogorensis...13961                    </td><td>5794.385</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Caldilinea aerophila...13962                </td><td>5996.758</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Kosakonia cowanii...13963                   </td><td>6236.103</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Synechococcus sp. PCC 7336...13964          </td><td>6824.469</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Granulicella tundricola...13965             </td><td>5915.935</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Granulosicoccus antarcticus...13966         </td><td>6159.157</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Candidatus Nitrospira inopinata...13967     </td><td>6802.590</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "\t<tr><td>Lactococcus lactis...13968                  </td><td>5313.937</td><td>0</td><td>0</td><td>TRUE</td><td>hiseq_xtm_flow_cell_v2_5_lot</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 13968 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " taxa & W & p\\_val & q\\_val & diff\\_abn & meta\\_col\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t Paraburkholderia fungorum...1                 &  2369.5949 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Paraburkholderia xenovorans...2               &  1431.8126 & 6.262869e-306 & 8.064137e-306 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Paraburkholderia hospita...3                  &   660.2795 & 4.593525e-139 & 4.828594e-139 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Ralstonia pickettii...4                       &  3803.0451 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Ralstonia insidiosa...5                       & 11006.3824 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Ralstonia mannitolilytica...6                 &  4609.9584 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Ralstonia solanacearum...7                    &  5302.4572 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia cepacia...8                      &  4335.3477 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia sp. LA-2-3-30-S1-D2...9          &  2308.2847 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia metallica...10                   &  1130.0050 & 1.343308e-240 & 1.571995e-240 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia contaminans...11                 &  2846.8788 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia gladioli...12                    &  9847.3649 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia thailandensis...13               &   729.6536 & 4.833906e-154 & 5.162079e-154 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Burkholderia sp. OLGA172...14                 &  1222.5535 & 1.258281e-260 & 1.513057e-260 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bordetella genomosp. 8...15                   &  1835.6105 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bordetella genomosp. 13...16                  & 14540.9158 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Acidovorax sp. RAC01...17                     &  3324.5010 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Acidovorax sp. 1608163...18                   &  3620.3009 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Diaphorobacter polyhydroxybutyrativorans...19 &   929.3545 & 3.383759e-197 & 3.775108e-197 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Massilia sp. WG5...20                         &  8254.7258 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Aromatoleum aromaticum...21                   &  4071.0045 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium sp. SK17...22                  & 58167.3505 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium sp. BTAi1...23                 & 46329.7238 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium erythrophlei...24              & 35114.9066 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium icense...25                    & 36656.2486 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium diazoefficiens...26            & 46012.8786 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium sp. WSM471...27                & 10420.0661 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Bradyrhizobium ottawaense...28                & 26795.8880 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Rhodopseudomonas palustris...29               & 97958.7439 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t Oligotropha carboxidovorans...30              & 30202.0768 &  0.000000e+00 &  0.000000e+00 & TRUE & site\\_supplying\\_sample\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Corynebacterium provencense...13939          & 7367.846 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Corynebacterium glutamicum...13940           & 8200.992 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Corynebacterium falsenii...13941             & 6171.133 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Kitasatospora albolonga...13942              & 4768.064 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Trueperella pyogenes...13943                 & 8182.139 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Gordonibacter pamelaeae...13944              & 7645.546 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Deinococcus sp. S14-83...13945               & 5823.598 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Chlorobaculum limnaeum...13946               & 7838.732 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Gluconobacter albidus...13947                & 5706.178 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Acetobacter ghanensis...13948                & 6582.644 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Citrobacter koseri...13949                   & 5088.732 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Adlercreutzia equolifaciens...13950          & 5976.559 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Chlorobium phaeobacteroides...13951          & 6001.430 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Aquaspirillum sp. LM1...13952                & 5439.753 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Acinetobacter bereziniae...13953             & 7144.056 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Pectobacterium parmentieri...13954           & 7895.328 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Acidihalobacter ferrooxidans...13955         & 4915.455 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Bifidobacterium choerinum...13956            & 7063.347 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Stenotrophomonas sp. SAU14A\\_NAIMI4\\_8...13957 & 4801.020 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Desulfovibrio desulfuricans...13958          & 6822.081 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Corynebacterium epidermidicanis...13959      & 6587.287 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Sulfuricella denitrificans...13960           & 7495.815 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Asaia bogorensis...13961                     & 5794.385 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Caldilinea aerophila...13962                 & 5996.758 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Kosakonia cowanii...13963                    & 6236.103 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Synechococcus sp. PCC 7336...13964           & 6824.469 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Granulicella tundricola...13965              & 5915.935 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Granulosicoccus antarcticus...13966          & 6159.157 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Candidatus Nitrospira inopinata...13967      & 6802.590 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\t Lactococcus lactis...13968                   & 5313.937 & 0 & 0 & TRUE & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 13968 × 6\n",
       "\n",
       "| taxa &lt;chr&gt; | W &lt;dbl&gt; | p_val &lt;dbl&gt; | q_val &lt;dbl&gt; | diff_abn &lt;lgl&gt; | meta_col &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| Paraburkholderia fungorum...1                 |  2369.5949 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Paraburkholderia xenovorans...2               |  1431.8126 | 6.262869e-306 | 8.064137e-306 | TRUE | site_supplying_sample |\n",
       "| Paraburkholderia hospita...3                  |   660.2795 | 4.593525e-139 | 4.828594e-139 | TRUE | site_supplying_sample |\n",
       "| Ralstonia pickettii...4                       |  3803.0451 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Ralstonia insidiosa...5                       | 11006.3824 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Ralstonia mannitolilytica...6                 |  4609.9584 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Ralstonia solanacearum...7                    |  5302.4572 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Burkholderia cepacia...8                      |  4335.3477 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Burkholderia sp. LA-2-3-30-S1-D2...9          |  2308.2847 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Burkholderia metallica...10                   |  1130.0050 | 1.343308e-240 | 1.571995e-240 | TRUE | site_supplying_sample |\n",
       "| Burkholderia contaminans...11                 |  2846.8788 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Burkholderia gladioli...12                    |  9847.3649 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Burkholderia thailandensis...13               |   729.6536 | 4.833906e-154 | 5.162079e-154 | TRUE | site_supplying_sample |\n",
       "| Burkholderia sp. OLGA172...14                 |  1222.5535 | 1.258281e-260 | 1.513057e-260 | TRUE | site_supplying_sample |\n",
       "| Bordetella genomosp. 8...15                   |  1835.6105 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bordetella genomosp. 13...16                  | 14540.9158 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Acidovorax sp. RAC01...17                     |  3324.5010 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Acidovorax sp. 1608163...18                   |  3620.3009 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Diaphorobacter polyhydroxybutyrativorans...19 |   929.3545 | 3.383759e-197 | 3.775108e-197 | TRUE | site_supplying_sample |\n",
       "| Massilia sp. WG5...20                         |  8254.7258 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Aromatoleum aromaticum...21                   |  4071.0045 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium sp. SK17...22                  | 58167.3505 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium sp. BTAi1...23                 | 46329.7238 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium erythrophlei...24              | 35114.9066 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium icense...25                    | 36656.2486 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium diazoefficiens...26            | 46012.8786 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium sp. WSM471...27                | 10420.0661 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Bradyrhizobium ottawaense...28                | 26795.8880 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Rhodopseudomonas palustris...29               | 97958.7439 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| Oligotropha carboxidovorans...30              | 30202.0768 |  0.000000e+00 |  0.000000e+00 | TRUE | site_supplying_sample |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Corynebacterium provencense...13939          | 7367.846 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Corynebacterium glutamicum...13940           | 8200.992 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Corynebacterium falsenii...13941             | 6171.133 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Kitasatospora albolonga...13942              | 4768.064 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Trueperella pyogenes...13943                 | 8182.139 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Gordonibacter pamelaeae...13944              | 7645.546 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Deinococcus sp. S14-83...13945               | 5823.598 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Chlorobaculum limnaeum...13946               | 7838.732 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Gluconobacter albidus...13947                | 5706.178 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Acetobacter ghanensis...13948                | 6582.644 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Citrobacter koseri...13949                   | 5088.732 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Adlercreutzia equolifaciens...13950          | 5976.559 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Chlorobium phaeobacteroides...13951          | 6001.430 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Aquaspirillum sp. LM1...13952                | 5439.753 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Acinetobacter bereziniae...13953             | 7144.056 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Pectobacterium parmentieri...13954           | 7895.328 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Acidihalobacter ferrooxidans...13955         | 4915.455 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Bifidobacterium choerinum...13956            | 7063.347 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Stenotrophomonas sp. SAU14A_NAIMI4_8...13957 | 4801.020 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Desulfovibrio desulfuricans...13958          | 6822.081 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Corynebacterium epidermidicanis...13959      | 6587.287 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Sulfuricella denitrificans...13960           | 7495.815 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Asaia bogorensis...13961                     | 5794.385 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Caldilinea aerophila...13962                 | 5996.758 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Kosakonia cowanii...13963                    | 6236.103 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Synechococcus sp. PCC 7336...13964           | 6824.469 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Granulicella tundricola...13965              | 5915.935 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Granulosicoccus antarcticus...13966          | 6159.157 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Candidatus Nitrospira inopinata...13967      | 6802.590 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "| Lactococcus lactis...13968                   | 5313.937 | 0 | 0 | TRUE | hiseq_xtm_flow_cell_v2_5_lot |\n",
       "\n"
      ],
      "text/plain": [
       "      taxa                                          W          p_val        \n",
       "1     Paraburkholderia fungorum...1                  2369.5949  0.000000e+00\n",
       "2     Paraburkholderia xenovorans...2                1431.8126 6.262869e-306\n",
       "3     Paraburkholderia hospita...3                    660.2795 4.593525e-139\n",
       "4     Ralstonia pickettii...4                        3803.0451  0.000000e+00\n",
       "5     Ralstonia insidiosa...5                       11006.3824  0.000000e+00\n",
       "6     Ralstonia mannitolilytica...6                  4609.9584  0.000000e+00\n",
       "7     Ralstonia solanacearum...7                     5302.4572  0.000000e+00\n",
       "8     Burkholderia cepacia...8                       4335.3477  0.000000e+00\n",
       "9     Burkholderia sp. LA-2-3-30-S1-D2...9           2308.2847  0.000000e+00\n",
       "10    Burkholderia metallica...10                    1130.0050 1.343308e-240\n",
       "11    Burkholderia contaminans...11                  2846.8788  0.000000e+00\n",
       "12    Burkholderia gladioli...12                     9847.3649  0.000000e+00\n",
       "13    Burkholderia thailandensis...13                 729.6536 4.833906e-154\n",
       "14    Burkholderia sp. OLGA172...14                  1222.5535 1.258281e-260\n",
       "15    Bordetella genomosp. 8...15                    1835.6105  0.000000e+00\n",
       "16    Bordetella genomosp. 13...16                  14540.9158  0.000000e+00\n",
       "17    Acidovorax sp. RAC01...17                      3324.5010  0.000000e+00\n",
       "18    Acidovorax sp. 1608163...18                    3620.3009  0.000000e+00\n",
       "19    Diaphorobacter polyhydroxybutyrativorans...19   929.3545 3.383759e-197\n",
       "20    Massilia sp. WG5...20                          8254.7258  0.000000e+00\n",
       "21    Aromatoleum aromaticum...21                    4071.0045  0.000000e+00\n",
       "22    Bradyrhizobium sp. SK17...22                  58167.3505  0.000000e+00\n",
       "23    Bradyrhizobium sp. BTAi1...23                 46329.7238  0.000000e+00\n",
       "24    Bradyrhizobium erythrophlei...24              35114.9066  0.000000e+00\n",
       "25    Bradyrhizobium icense...25                    36656.2486  0.000000e+00\n",
       "26    Bradyrhizobium diazoefficiens...26            46012.8786  0.000000e+00\n",
       "27    Bradyrhizobium sp. WSM471...27                10420.0661  0.000000e+00\n",
       "28    Bradyrhizobium ottawaense...28                26795.8880  0.000000e+00\n",
       "29    Rhodopseudomonas palustris...29               97958.7439  0.000000e+00\n",
       "30    Oligotropha carboxidovorans...30              30202.0768  0.000000e+00\n",
       "⋮     ⋮                                             ⋮          ⋮            \n",
       "13939 Corynebacterium provencense...13939           7367.846   0            \n",
       "13940 Corynebacterium glutamicum...13940            8200.992   0            \n",
       "13941 Corynebacterium falsenii...13941              6171.133   0            \n",
       "13942 Kitasatospora albolonga...13942               4768.064   0            \n",
       "13943 Trueperella pyogenes...13943                  8182.139   0            \n",
       "13944 Gordonibacter pamelaeae...13944               7645.546   0            \n",
       "13945 Deinococcus sp. S14-83...13945                5823.598   0            \n",
       "13946 Chlorobaculum limnaeum...13946                7838.732   0            \n",
       "13947 Gluconobacter albidus...13947                 5706.178   0            \n",
       "13948 Acetobacter ghanensis...13948                 6582.644   0            \n",
       "13949 Citrobacter koseri...13949                    5088.732   0            \n",
       "13950 Adlercreutzia equolifaciens...13950           5976.559   0            \n",
       "13951 Chlorobium phaeobacteroides...13951           6001.430   0            \n",
       "13952 Aquaspirillum sp. LM1...13952                 5439.753   0            \n",
       "13953 Acinetobacter bereziniae...13953              7144.056   0            \n",
       "13954 Pectobacterium parmentieri...13954            7895.328   0            \n",
       "13955 Acidihalobacter ferrooxidans...13955          4915.455   0            \n",
       "13956 Bifidobacterium choerinum...13956             7063.347   0            \n",
       "13957 Stenotrophomonas sp. SAU14A_NAIMI4_8...13957  4801.020   0            \n",
       "13958 Desulfovibrio desulfuricans...13958           6822.081   0            \n",
       "13959 Corynebacterium epidermidicanis...13959       6587.287   0            \n",
       "13960 Sulfuricella denitrificans...13960            7495.815   0            \n",
       "13961 Asaia bogorensis...13961                      5794.385   0            \n",
       "13962 Caldilinea aerophila...13962                  5996.758   0            \n",
       "13963 Kosakonia cowanii...13963                     6236.103   0            \n",
       "13964 Synechococcus sp. PCC 7336...13964            6824.469   0            \n",
       "13965 Granulicella tundricola...13965               5915.935   0            \n",
       "13966 Granulosicoccus antarcticus...13966           6159.157   0            \n",
       "13967 Candidatus Nitrospira inopinata...13967       6802.590   0            \n",
       "13968 Lactococcus lactis...13968                    5313.937   0            \n",
       "      q_val         diff_abn meta_col                    \n",
       "1      0.000000e+00 TRUE     site_supplying_sample       \n",
       "2     8.064137e-306 TRUE     site_supplying_sample       \n",
       "3     4.828594e-139 TRUE     site_supplying_sample       \n",
       "4      0.000000e+00 TRUE     site_supplying_sample       \n",
       "5      0.000000e+00 TRUE     site_supplying_sample       \n",
       "6      0.000000e+00 TRUE     site_supplying_sample       \n",
       "7      0.000000e+00 TRUE     site_supplying_sample       \n",
       "8      0.000000e+00 TRUE     site_supplying_sample       \n",
       "9      0.000000e+00 TRUE     site_supplying_sample       \n",
       "10    1.571995e-240 TRUE     site_supplying_sample       \n",
       "11     0.000000e+00 TRUE     site_supplying_sample       \n",
       "12     0.000000e+00 TRUE     site_supplying_sample       \n",
       "13    5.162079e-154 TRUE     site_supplying_sample       \n",
       "14    1.513057e-260 TRUE     site_supplying_sample       \n",
       "15     0.000000e+00 TRUE     site_supplying_sample       \n",
       "16     0.000000e+00 TRUE     site_supplying_sample       \n",
       "17     0.000000e+00 TRUE     site_supplying_sample       \n",
       "18     0.000000e+00 TRUE     site_supplying_sample       \n",
       "19    3.775108e-197 TRUE     site_supplying_sample       \n",
       "20     0.000000e+00 TRUE     site_supplying_sample       \n",
       "21     0.000000e+00 TRUE     site_supplying_sample       \n",
       "22     0.000000e+00 TRUE     site_supplying_sample       \n",
       "23     0.000000e+00 TRUE     site_supplying_sample       \n",
       "24     0.000000e+00 TRUE     site_supplying_sample       \n",
       "25     0.000000e+00 TRUE     site_supplying_sample       \n",
       "26     0.000000e+00 TRUE     site_supplying_sample       \n",
       "27     0.000000e+00 TRUE     site_supplying_sample       \n",
       "28     0.000000e+00 TRUE     site_supplying_sample       \n",
       "29     0.000000e+00 TRUE     site_supplying_sample       \n",
       "30     0.000000e+00 TRUE     site_supplying_sample       \n",
       "⋮     ⋮             ⋮        ⋮                           \n",
       "13939 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13940 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13941 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13942 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13943 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13944 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13945 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13946 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13947 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13948 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13949 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13950 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13951 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13952 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13953 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13954 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13955 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13956 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13957 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13958 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13959 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13960 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13961 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13962 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13963 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13964 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13965 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13966 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13967 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot\n",
       "13968 0             TRUE     hiseq_xtm_flow_cell_v2_5_lot"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# decontaminate <- function(n) {\n",
    "    # Load data\n",
    "#     df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.S.tsv\"))\n",
    "    df <- load_data(str_glue(\"data/temp_files_9999/07_abundance_matrix/abundance_matrix.subset_9999.S.tsv\"))\n",
    "    meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "    \n",
    "    # Get metadata subset\n",
    "    meta_filt <- subset_metadata(meta, 9999)\n",
    "\n",
    "    # Filter data\n",
    "    df_filt <- retrieve_rows_from_meta(df, meta_filt)\n",
    "    df_filt2 <- remove_cols(df_filt, c(\"Homo sapiens\", \"unclassified\"))\n",
    "    df_filt3 <- remove_low_freq_taxa(df_filt2)\n",
    "\n",
    "    # Get metadata columns of interest\n",
    "    meta_cols <- get_meta_cols(meta_filt)\n",
    "    \n",
    "    # Create phyloseq object\n",
    "    phy <- phyloseq(df_to_phy(df_filt3), meta_to_phy(meta_filt))\n",
    "\n",
    "    # Print batch levels and frequencies\n",
    "#     print_group_freqs(meta_filt, meta_cols)\n",
    "\n",
    "    # Run ANCOM-BC\n",
    "    result_df <- run_ancom_bc(phy, meta_filt, meta_cols, n_samples = 3, max_iter = 100)\n",
    "    result_df\n",
    "    \n",
    "#     return(non_contaminants)\n",
    "#     }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1f9e6681",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Expected 2 pieces. Additional pieces discarded in 8 rows [143, 1889, 3635, 5381, 7127, 8873, 10619, 12365].”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1746 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>taxa</th><th scope=col>site_supplying_sample</th><th scope=col>extraction_kit</th><th scope=col>instrument_id</th><th scope=col>hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot</th><th scope=col>hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot</th><th scope=col>hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot</th><th scope=col>hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot</th><th scope=col>hiseq_xtm_flow_cell_v2_5_lot</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Paraburkholderia fungorum               </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Paraburkholderia xenovorans             </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Paraburkholderia hospita                </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Ralstonia pickettii                     </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Ralstonia insidiosa                     </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Ralstonia mannitolilytica               </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Ralstonia solanacearum                  </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia cepacia                    </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia sp. LA-2-3-30-S1-D2        </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia metallica                  </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia contaminans                </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia gladioli                   </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia thailandensis              </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Burkholderia sp. OLGA172                </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bordetella genomosp. 8                  </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bordetella genomosp. 13                 </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Acidovorax sp. RAC01                    </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Acidovorax sp. 1608163                  </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Diaphorobacter polyhydroxybutyrativorans</td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Massilia sp. WG5                        </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Aromatoleum aromaticum                  </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. SK17                 </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. BTAi1                </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium erythrophlei             </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium icense                   </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium diazoefficiens           </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. WSM471               </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bradyrhizobium ottawaense               </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Rhodopseudomonas palustris              </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Oligotropha carboxidovorans             </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Corynebacterium provencense         </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Corynebacterium glutamicum          </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Corynebacterium falsenii            </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Kitasatospora albolonga             </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Trueperella pyogenes                </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Gordonibacter pamelaeae             </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Deinococcus sp. S14-83              </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Chlorobaculum limnaeum              </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Gluconobacter albidus               </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Acetobacter ghanensis               </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Citrobacter koseri                  </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Adlercreutzia equolifaciens         </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Chlorobium phaeobacteroides         </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Aquaspirillum sp. LM1               </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Acinetobacter bereziniae            </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Pectobacterium parmentieri          </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Acidihalobacter ferrooxidans        </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Bifidobacterium choerinum           </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Stenotrophomonas sp. SAU14A_NAIMI4_8</td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Desulfovibrio desulfuricans         </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Corynebacterium epidermidicanis     </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Sulfuricella denitrificans          </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Asaia bogorensis                    </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Caldilinea aerophila                </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Kosakonia cowanii                   </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Synechococcus sp. PCC 7336          </td><td>TRUE</td><td>TRUE</td><td> TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Granulicella tundricola             </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Granulosicoccus antarcticus         </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Candidatus Nitrospira inopinata     </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><td>Lactococcus lactis                  </td><td>TRUE</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1746 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " taxa & site\\_supplying\\_sample & extraction\\_kit & instrument\\_id & hiseq\\_xtm\\_sbs\\_kit\\_300\\_cycles\\_v2\\_\\_box\\_1of\\_2\\_\\_lot & hiseq\\_xtm\\_sbs\\_kit\\_300\\_cycles\\_v2\\_\\_box\\_2\\_of\\_2\\_\\_lot & hiseq\\_xtm\\_pe\\_cluster\\_kit\\_cbottm\\_v2\\_\\_box\\_1\\_of\\_2\\_\\_lot & hiseq\\_xtm\\_pe\\_cluster\\_kit\\_cbottm\\_v2\\_\\_box\\_2\\_of\\_2\\_\\_lot & hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot\\\\\n",
       " <chr> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t Paraburkholderia fungorum                & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Paraburkholderia xenovorans              & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Paraburkholderia hospita                 & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Ralstonia pickettii                      & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Ralstonia insidiosa                      & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Ralstonia mannitolilytica                & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Ralstonia solanacearum                   & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia cepacia                     & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia sp. LA-2-3-30-S1-D2         & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia metallica                   & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia contaminans                 & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia gladioli                    & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia thailandensis               & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Burkholderia sp. OLGA172                 & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bordetella genomosp. 8                   & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bordetella genomosp. 13                  & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Acidovorax sp. RAC01                     & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Acidovorax sp. 1608163                   & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Diaphorobacter polyhydroxybutyrativorans & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Massilia sp. WG5                         & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Aromatoleum aromaticum                   & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium sp. SK17                  & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium sp. BTAi1                 & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium erythrophlei              & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium icense                    & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium diazoefficiens            & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium sp. WSM471                & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bradyrhizobium ottawaense                & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Rhodopseudomonas palustris               & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Oligotropha carboxidovorans              & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Corynebacterium provencense          & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Corynebacterium glutamicum           & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Corynebacterium falsenii             & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Kitasatospora albolonga              & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Trueperella pyogenes                 & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Gordonibacter pamelaeae              & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Deinococcus sp. S14-83               & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Chlorobaculum limnaeum               & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Gluconobacter albidus                & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Acetobacter ghanensis                & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Citrobacter koseri                   & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Adlercreutzia equolifaciens          & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Chlorobium phaeobacteroides          & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Aquaspirillum sp. LM1                & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Acinetobacter bereziniae             & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Pectobacterium parmentieri           & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Acidihalobacter ferrooxidans         & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Bifidobacterium choerinum            & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Stenotrophomonas sp. SAU14A\\_NAIMI4\\_8 & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Desulfovibrio desulfuricans          & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Corynebacterium epidermidicanis      & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Sulfuricella denitrificans           & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Asaia bogorensis                     & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Caldilinea aerophila                 & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Kosakonia cowanii                    & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Synechococcus sp. PCC 7336           & TRUE & TRUE &  TRUE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Granulicella tundricola              & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Granulosicoccus antarcticus          & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Candidatus Nitrospira inopinata      & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\t Lactococcus lactis                   & TRUE & TRUE & FALSE & TRUE & TRUE & TRUE & TRUE & TRUE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1746 × 9\n",
       "\n",
       "| taxa &lt;chr&gt; | site_supplying_sample &lt;lgl&gt; | extraction_kit &lt;lgl&gt; | instrument_id &lt;lgl&gt; | hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot &lt;lgl&gt; | hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot &lt;lgl&gt; | hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot &lt;lgl&gt; | hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot &lt;lgl&gt; | hiseq_xtm_flow_cell_v2_5_lot &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Paraburkholderia fungorum                | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Paraburkholderia xenovorans              | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Paraburkholderia hospita                 | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Ralstonia pickettii                      | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Ralstonia insidiosa                      | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Ralstonia mannitolilytica                | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Ralstonia solanacearum                   | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia cepacia                     | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia sp. LA-2-3-30-S1-D2         | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia metallica                   | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia contaminans                 | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia gladioli                    | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia thailandensis               | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Burkholderia sp. OLGA172                 | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bordetella genomosp. 8                   | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bordetella genomosp. 13                  | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Acidovorax sp. RAC01                     | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Acidovorax sp. 1608163                   | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Diaphorobacter polyhydroxybutyrativorans | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Massilia sp. WG5                         | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Aromatoleum aromaticum                   | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium sp. SK17                  | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium sp. BTAi1                 | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium erythrophlei              | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium icense                    | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium diazoefficiens            | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium sp. WSM471                | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bradyrhizobium ottawaense                | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Rhodopseudomonas palustris               | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Oligotropha carboxidovorans              | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Corynebacterium provencense          | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Corynebacterium glutamicum           | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Corynebacterium falsenii             | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Kitasatospora albolonga              | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Trueperella pyogenes                 | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Gordonibacter pamelaeae              | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Deinococcus sp. S14-83               | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Chlorobaculum limnaeum               | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Gluconobacter albidus                | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Acetobacter ghanensis                | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Citrobacter koseri                   | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Adlercreutzia equolifaciens          | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Chlorobium phaeobacteroides          | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Aquaspirillum sp. LM1                | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Acinetobacter bereziniae             | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Pectobacterium parmentieri           | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Acidihalobacter ferrooxidans         | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Bifidobacterium choerinum            | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Stenotrophomonas sp. SAU14A_NAIMI4_8 | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Desulfovibrio desulfuricans          | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Corynebacterium epidermidicanis      | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Sulfuricella denitrificans           | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Asaia bogorensis                     | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Caldilinea aerophila                 | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Kosakonia cowanii                    | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Synechococcus sp. PCC 7336           | TRUE | TRUE |  TRUE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Granulicella tundricola              | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Granulosicoccus antarcticus          | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Candidatus Nitrospira inopinata      | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "| Lactococcus lactis                   | TRUE | TRUE | FALSE | TRUE | TRUE | TRUE | TRUE | TRUE |\n",
       "\n"
      ],
      "text/plain": [
       "     taxa                                     site_supplying_sample\n",
       "1    Paraburkholderia fungorum                TRUE                 \n",
       "2    Paraburkholderia xenovorans              TRUE                 \n",
       "3    Paraburkholderia hospita                 TRUE                 \n",
       "4    Ralstonia pickettii                      TRUE                 \n",
       "5    Ralstonia insidiosa                      TRUE                 \n",
       "6    Ralstonia mannitolilytica                TRUE                 \n",
       "7    Ralstonia solanacearum                   TRUE                 \n",
       "8    Burkholderia cepacia                     TRUE                 \n",
       "9    Burkholderia sp. LA-2-3-30-S1-D2         TRUE                 \n",
       "10   Burkholderia metallica                   TRUE                 \n",
       "11   Burkholderia contaminans                 TRUE                 \n",
       "12   Burkholderia gladioli                    TRUE                 \n",
       "13   Burkholderia thailandensis               TRUE                 \n",
       "14   Burkholderia sp. OLGA172                 TRUE                 \n",
       "15   Bordetella genomosp. 8                   TRUE                 \n",
       "16   Bordetella genomosp. 13                  TRUE                 \n",
       "17   Acidovorax sp. RAC01                     TRUE                 \n",
       "18   Acidovorax sp. 1608163                   TRUE                 \n",
       "19   Diaphorobacter polyhydroxybutyrativorans TRUE                 \n",
       "20   Massilia sp. WG5                         TRUE                 \n",
       "21   Aromatoleum aromaticum                   TRUE                 \n",
       "22   Bradyrhizobium sp. SK17                  TRUE                 \n",
       "23   Bradyrhizobium sp. BTAi1                 TRUE                 \n",
       "24   Bradyrhizobium erythrophlei              TRUE                 \n",
       "25   Bradyrhizobium icense                    TRUE                 \n",
       "26   Bradyrhizobium diazoefficiens            TRUE                 \n",
       "27   Bradyrhizobium sp. WSM471                TRUE                 \n",
       "28   Bradyrhizobium ottawaense                TRUE                 \n",
       "29   Rhodopseudomonas palustris               TRUE                 \n",
       "30   Oligotropha carboxidovorans              TRUE                 \n",
       "⋮    ⋮                                        ⋮                    \n",
       "1717 Corynebacterium provencense              TRUE                 \n",
       "1718 Corynebacterium glutamicum               TRUE                 \n",
       "1719 Corynebacterium falsenii                 TRUE                 \n",
       "1720 Kitasatospora albolonga                  TRUE                 \n",
       "1721 Trueperella pyogenes                     TRUE                 \n",
       "1722 Gordonibacter pamelaeae                  TRUE                 \n",
       "1723 Deinococcus sp. S14-83                   TRUE                 \n",
       "1724 Chlorobaculum limnaeum                   TRUE                 \n",
       "1725 Gluconobacter albidus                    TRUE                 \n",
       "1726 Acetobacter ghanensis                    TRUE                 \n",
       "1727 Citrobacter koseri                       TRUE                 \n",
       "1728 Adlercreutzia equolifaciens              TRUE                 \n",
       "1729 Chlorobium phaeobacteroides              TRUE                 \n",
       "1730 Aquaspirillum sp. LM1                    TRUE                 \n",
       "1731 Acinetobacter bereziniae                 TRUE                 \n",
       "1732 Pectobacterium parmentieri               TRUE                 \n",
       "1733 Acidihalobacter ferrooxidans             TRUE                 \n",
       "1734 Bifidobacterium choerinum                TRUE                 \n",
       "1735 Stenotrophomonas sp. SAU14A_NAIMI4_8     TRUE                 \n",
       "1736 Desulfovibrio desulfuricans              TRUE                 \n",
       "1737 Corynebacterium epidermidicanis          TRUE                 \n",
       "1738 Sulfuricella denitrificans               TRUE                 \n",
       "1739 Asaia bogorensis                         TRUE                 \n",
       "1740 Caldilinea aerophila                     TRUE                 \n",
       "1741 Kosakonia cowanii                        TRUE                 \n",
       "1742 Synechococcus sp. PCC 7336               TRUE                 \n",
       "1743 Granulicella tundricola                  TRUE                 \n",
       "1744 Granulosicoccus antarcticus              TRUE                 \n",
       "1745 Candidatus Nitrospira inopinata          TRUE                 \n",
       "1746 Lactococcus lactis                       TRUE                 \n",
       "     extraction_kit instrument_id\n",
       "1    TRUE            TRUE        \n",
       "2    TRUE           FALSE        \n",
       "3    TRUE           FALSE        \n",
       "4    TRUE            TRUE        \n",
       "5    TRUE            TRUE        \n",
       "6    TRUE            TRUE        \n",
       "7    TRUE            TRUE        \n",
       "8    TRUE           FALSE        \n",
       "9    TRUE           FALSE        \n",
       "10   TRUE           FALSE        \n",
       "11   TRUE           FALSE        \n",
       "12   TRUE           FALSE        \n",
       "13   TRUE           FALSE        \n",
       "14   TRUE           FALSE        \n",
       "15   TRUE            TRUE        \n",
       "16   TRUE           FALSE        \n",
       "17   TRUE            TRUE        \n",
       "18   TRUE            TRUE        \n",
       "19   TRUE           FALSE        \n",
       "20   TRUE           FALSE        \n",
       "21   TRUE           FALSE        \n",
       "22   TRUE           FALSE        \n",
       "23   TRUE            TRUE        \n",
       "24   TRUE            TRUE        \n",
       "25   TRUE            TRUE        \n",
       "26   TRUE            TRUE        \n",
       "27   TRUE            TRUE        \n",
       "28   TRUE            TRUE        \n",
       "29   TRUE            TRUE        \n",
       "30   TRUE            TRUE        \n",
       "⋮    ⋮              ⋮            \n",
       "1717 TRUE           FALSE        \n",
       "1718 TRUE           FALSE        \n",
       "1719 TRUE           FALSE        \n",
       "1720 TRUE           FALSE        \n",
       "1721 TRUE           FALSE        \n",
       "1722 TRUE           FALSE        \n",
       "1723 TRUE           FALSE        \n",
       "1724 TRUE           FALSE        \n",
       "1725 TRUE           FALSE        \n",
       "1726 TRUE           FALSE        \n",
       "1727 TRUE           FALSE        \n",
       "1728 TRUE           FALSE        \n",
       "1729 TRUE           FALSE        \n",
       "1730 TRUE           FALSE        \n",
       "1731 TRUE           FALSE        \n",
       "1732 TRUE           FALSE        \n",
       "1733 TRUE           FALSE        \n",
       "1734 TRUE           FALSE        \n",
       "1735 TRUE           FALSE        \n",
       "1736 TRUE           FALSE        \n",
       "1737 TRUE           FALSE        \n",
       "1738 TRUE            TRUE        \n",
       "1739 TRUE           FALSE        \n",
       "1740 TRUE           FALSE        \n",
       "1741 TRUE           FALSE        \n",
       "1742 TRUE            TRUE        \n",
       "1743 TRUE           FALSE        \n",
       "1744 TRUE           FALSE        \n",
       "1745 TRUE           FALSE        \n",
       "1746 TRUE           FALSE        \n",
       "     hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\n",
       "1    TRUE                                           \n",
       "2    TRUE                                           \n",
       "3    TRUE                                           \n",
       "4    TRUE                                           \n",
       "5    TRUE                                           \n",
       "6    TRUE                                           \n",
       "7    TRUE                                           \n",
       "8    TRUE                                           \n",
       "9    TRUE                                           \n",
       "10   TRUE                                           \n",
       "11   TRUE                                           \n",
       "12   TRUE                                           \n",
       "13   TRUE                                           \n",
       "14   TRUE                                           \n",
       "15   TRUE                                           \n",
       "16   TRUE                                           \n",
       "17   TRUE                                           \n",
       "18   TRUE                                           \n",
       "19   TRUE                                           \n",
       "20   TRUE                                           \n",
       "21   TRUE                                           \n",
       "22   TRUE                                           \n",
       "23   TRUE                                           \n",
       "24   TRUE                                           \n",
       "25   TRUE                                           \n",
       "26   TRUE                                           \n",
       "27   TRUE                                           \n",
       "28   TRUE                                           \n",
       "29   TRUE                                           \n",
       "30   TRUE                                           \n",
       "⋮    ⋮                                              \n",
       "1717 TRUE                                           \n",
       "1718 TRUE                                           \n",
       "1719 TRUE                                           \n",
       "1720 TRUE                                           \n",
       "1721 TRUE                                           \n",
       "1722 TRUE                                           \n",
       "1723 TRUE                                           \n",
       "1724 TRUE                                           \n",
       "1725 TRUE                                           \n",
       "1726 TRUE                                           \n",
       "1727 TRUE                                           \n",
       "1728 TRUE                                           \n",
       "1729 TRUE                                           \n",
       "1730 TRUE                                           \n",
       "1731 TRUE                                           \n",
       "1732 TRUE                                           \n",
       "1733 TRUE                                           \n",
       "1734 TRUE                                           \n",
       "1735 TRUE                                           \n",
       "1736 TRUE                                           \n",
       "1737 TRUE                                           \n",
       "1738 TRUE                                           \n",
       "1739 TRUE                                           \n",
       "1740 TRUE                                           \n",
       "1741 TRUE                                           \n",
       "1742 TRUE                                           \n",
       "1743 TRUE                                           \n",
       "1744 TRUE                                           \n",
       "1745 TRUE                                           \n",
       "1746 TRUE                                           \n",
       "     hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\n",
       "1    TRUE                                            \n",
       "2    TRUE                                            \n",
       "3    TRUE                                            \n",
       "4    TRUE                                            \n",
       "5    TRUE                                            \n",
       "6    TRUE                                            \n",
       "7    TRUE                                            \n",
       "8    TRUE                                            \n",
       "9    TRUE                                            \n",
       "10   TRUE                                            \n",
       "11   TRUE                                            \n",
       "12   TRUE                                            \n",
       "13   TRUE                                            \n",
       "14   TRUE                                            \n",
       "15   TRUE                                            \n",
       "16   TRUE                                            \n",
       "17   TRUE                                            \n",
       "18   TRUE                                            \n",
       "19   TRUE                                            \n",
       "20   TRUE                                            \n",
       "21   TRUE                                            \n",
       "22   TRUE                                            \n",
       "23   TRUE                                            \n",
       "24   TRUE                                            \n",
       "25   TRUE                                            \n",
       "26   TRUE                                            \n",
       "27   TRUE                                            \n",
       "28   TRUE                                            \n",
       "29   TRUE                                            \n",
       "30   TRUE                                            \n",
       "⋮    ⋮                                               \n",
       "1717 TRUE                                            \n",
       "1718 TRUE                                            \n",
       "1719 TRUE                                            \n",
       "1720 TRUE                                            \n",
       "1721 TRUE                                            \n",
       "1722 TRUE                                            \n",
       "1723 TRUE                                            \n",
       "1724 TRUE                                            \n",
       "1725 TRUE                                            \n",
       "1726 TRUE                                            \n",
       "1727 TRUE                                            \n",
       "1728 TRUE                                            \n",
       "1729 TRUE                                            \n",
       "1730 TRUE                                            \n",
       "1731 TRUE                                            \n",
       "1732 TRUE                                            \n",
       "1733 TRUE                                            \n",
       "1734 TRUE                                            \n",
       "1735 TRUE                                            \n",
       "1736 TRUE                                            \n",
       "1737 TRUE                                            \n",
       "1738 TRUE                                            \n",
       "1739 TRUE                                            \n",
       "1740 TRUE                                            \n",
       "1741 TRUE                                            \n",
       "1742 TRUE                                            \n",
       "1743 TRUE                                            \n",
       "1744 TRUE                                            \n",
       "1745 TRUE                                            \n",
       "1746 TRUE                                            \n",
       "     hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\n",
       "1    TRUE                                               \n",
       "2    TRUE                                               \n",
       "3    TRUE                                               \n",
       "4    TRUE                                               \n",
       "5    TRUE                                               \n",
       "6    TRUE                                               \n",
       "7    TRUE                                               \n",
       "8    TRUE                                               \n",
       "9    TRUE                                               \n",
       "10   TRUE                                               \n",
       "11   TRUE                                               \n",
       "12   TRUE                                               \n",
       "13   TRUE                                               \n",
       "14   TRUE                                               \n",
       "15   TRUE                                               \n",
       "16   TRUE                                               \n",
       "17   TRUE                                               \n",
       "18   TRUE                                               \n",
       "19   TRUE                                               \n",
       "20   TRUE                                               \n",
       "21   TRUE                                               \n",
       "22   TRUE                                               \n",
       "23   TRUE                                               \n",
       "24   TRUE                                               \n",
       "25   TRUE                                               \n",
       "26   TRUE                                               \n",
       "27   TRUE                                               \n",
       "28   TRUE                                               \n",
       "29   TRUE                                               \n",
       "30   TRUE                                               \n",
       "⋮    ⋮                                                  \n",
       "1717 TRUE                                               \n",
       "1718 TRUE                                               \n",
       "1719 TRUE                                               \n",
       "1720 TRUE                                               \n",
       "1721 TRUE                                               \n",
       "1722 TRUE                                               \n",
       "1723 TRUE                                               \n",
       "1724 TRUE                                               \n",
       "1725 TRUE                                               \n",
       "1726 TRUE                                               \n",
       "1727 TRUE                                               \n",
       "1728 TRUE                                               \n",
       "1729 TRUE                                               \n",
       "1730 TRUE                                               \n",
       "1731 TRUE                                               \n",
       "1732 TRUE                                               \n",
       "1733 TRUE                                               \n",
       "1734 TRUE                                               \n",
       "1735 TRUE                                               \n",
       "1736 TRUE                                               \n",
       "1737 TRUE                                               \n",
       "1738 TRUE                                               \n",
       "1739 TRUE                                               \n",
       "1740 TRUE                                               \n",
       "1741 TRUE                                               \n",
       "1742 TRUE                                               \n",
       "1743 TRUE                                               \n",
       "1744 TRUE                                               \n",
       "1745 TRUE                                               \n",
       "1746 TRUE                                               \n",
       "     hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\n",
       "1    TRUE                                               \n",
       "2    TRUE                                               \n",
       "3    TRUE                                               \n",
       "4    TRUE                                               \n",
       "5    TRUE                                               \n",
       "6    TRUE                                               \n",
       "7    TRUE                                               \n",
       "8    TRUE                                               \n",
       "9    TRUE                                               \n",
       "10   TRUE                                               \n",
       "11   TRUE                                               \n",
       "12   TRUE                                               \n",
       "13   TRUE                                               \n",
       "14   TRUE                                               \n",
       "15   TRUE                                               \n",
       "16   TRUE                                               \n",
       "17   TRUE                                               \n",
       "18   TRUE                                               \n",
       "19   TRUE                                               \n",
       "20   TRUE                                               \n",
       "21   TRUE                                               \n",
       "22   TRUE                                               \n",
       "23   TRUE                                               \n",
       "24   TRUE                                               \n",
       "25   TRUE                                               \n",
       "26   TRUE                                               \n",
       "27   TRUE                                               \n",
       "28   TRUE                                               \n",
       "29   TRUE                                               \n",
       "30   TRUE                                               \n",
       "⋮    ⋮                                                  \n",
       "1717 TRUE                                               \n",
       "1718 TRUE                                               \n",
       "1719 TRUE                                               \n",
       "1720 TRUE                                               \n",
       "1721 TRUE                                               \n",
       "1722 TRUE                                               \n",
       "1723 TRUE                                               \n",
       "1724 TRUE                                               \n",
       "1725 TRUE                                               \n",
       "1726 TRUE                                               \n",
       "1727 TRUE                                               \n",
       "1728 TRUE                                               \n",
       "1729 TRUE                                               \n",
       "1730 TRUE                                               \n",
       "1731 TRUE                                               \n",
       "1732 TRUE                                               \n",
       "1733 TRUE                                               \n",
       "1734 TRUE                                               \n",
       "1735 TRUE                                               \n",
       "1736 TRUE                                               \n",
       "1737 TRUE                                               \n",
       "1738 TRUE                                               \n",
       "1739 TRUE                                               \n",
       "1740 TRUE                                               \n",
       "1741 TRUE                                               \n",
       "1742 TRUE                                               \n",
       "1743 TRUE                                               \n",
       "1744 TRUE                                               \n",
       "1745 TRUE                                               \n",
       "1746 TRUE                                               \n",
       "     hiseq_xtm_flow_cell_v2_5_lot\n",
       "1    TRUE                        \n",
       "2    TRUE                        \n",
       "3    TRUE                        \n",
       "4    TRUE                        \n",
       "5    TRUE                        \n",
       "6    TRUE                        \n",
       "7    TRUE                        \n",
       "8    TRUE                        \n",
       "9    TRUE                        \n",
       "10   TRUE                        \n",
       "11   TRUE                        \n",
       "12   TRUE                        \n",
       "13   TRUE                        \n",
       "14   TRUE                        \n",
       "15   TRUE                        \n",
       "16   TRUE                        \n",
       "17   TRUE                        \n",
       "18   TRUE                        \n",
       "19   TRUE                        \n",
       "20   TRUE                        \n",
       "21   TRUE                        \n",
       "22   TRUE                        \n",
       "23   TRUE                        \n",
       "24   TRUE                        \n",
       "25   TRUE                        \n",
       "26   TRUE                        \n",
       "27   TRUE                        \n",
       "28   TRUE                        \n",
       "29   TRUE                        \n",
       "30   TRUE                        \n",
       "⋮    ⋮                           \n",
       "1717 TRUE                        \n",
       "1718 TRUE                        \n",
       "1719 TRUE                        \n",
       "1720 TRUE                        \n",
       "1721 TRUE                        \n",
       "1722 TRUE                        \n",
       "1723 TRUE                        \n",
       "1724 TRUE                        \n",
       "1725 TRUE                        \n",
       "1726 TRUE                        \n",
       "1727 TRUE                        \n",
       "1728 TRUE                        \n",
       "1729 TRUE                        \n",
       "1730 TRUE                        \n",
       "1731 TRUE                        \n",
       "1732 TRUE                        \n",
       "1733 TRUE                        \n",
       "1734 TRUE                        \n",
       "1735 TRUE                        \n",
       "1736 TRUE                        \n",
       "1737 TRUE                        \n",
       "1738 TRUE                        \n",
       "1739 TRUE                        \n",
       "1740 TRUE                        \n",
       "1741 TRUE                        \n",
       "1742 TRUE                        \n",
       "1743 TRUE                        \n",
       "1744 TRUE                        \n",
       "1745 TRUE                        \n",
       "1746 TRUE                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Expected 2 pieces. Additional pieces discarded in 8 rows [143, 1889, 3635, 5381, 7127, 8873, 10619, 12365].”\n",
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    }
   ],
   "source": [
    "result_df <- fread(\"results/decontamination/ANCOMBC_decontamination_results.n9706.100821.csv\")\n",
    "result_df %>%\n",
    "    separate(taxa, into = c(\"taxa\", NA), remove = T, sep = \"\\\\.\\\\.\") %>%\n",
    "    pivot_wider(names_from = meta_col, values_from = diff_abn, id_cols = taxa)\n",
    "\n",
    "hm <- result_df %>%\n",
    "    separate(taxa, into = c(\"taxa\", NA), remove = T, sep = \"\\\\.\\\\.\") %>%\n",
    "    ggplot(aes(x = meta_col, y = taxa, fill = diff_abn)) +\n",
    "    geom_tile() +\n",
    "    theme(axis.text.y = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "    labs(x = \"Batch Info\", y = \"Species\", fill = \"Contaminant?\")\n",
    "\n",
    "ggsave(\"results/decontamination/ANCOMBC_decontamination_heatmap.n9606.100821.png\", dpi = 300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "53b36f15",
   "metadata": {},
   "outputs": [],
   "source": [
    "# fwrite(result_df, \"results/decontamination/ANCOMBC_decontamination_results.n9706.100821.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "277c2ac3",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'dawdawdawdwadwa' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'dawdawdawdwadwa' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "dawdawdawdwadwa####################################################"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eafeab9d",
   "metadata": {},
   "source": [
    "### Explore metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96b8e232",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_subset <- 100\n",
    "\n",
    "# Load data\n",
    "df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.S.tsv\"))\n",
    "\n",
    "# Retain relevant metadata\n",
    "meta <- parse_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "\n",
    "# Get metadata columns of interest\n",
    "meta_cols <- get_meta_cols(meta)\n",
    "\n",
    "meta <- as.matrix(meta)\n",
    "meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "plots <- list()\n",
    "\n",
    "for (column in meta_cols) {\n",
    "    plt <- meta %>%\n",
    "        mutate(across(everything(), as.character)) %>%\n",
    "        select(all_of(column)) %>%\n",
    "        group_by_at(column) %>%\n",
    "        summarise(n = n()) %>%\n",
    "        ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "            geom_bar(stat = \"identity\") +\n",
    "            theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "            geom_text(aes_string(label = \"n\"))\n",
    "    plots[[column]] <- plt\n",
    "}\n",
    "\n",
    "plots\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00b1961d",
   "metadata": {},
   "source": [
    "### MAIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e4a03ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nc_list <- list()\n",
    "\n",
    "# for (i in c(100)) {\n",
    "#     nc_list[[str_glue(\"subset_{i}\")]] <- decontaminate(i)\n",
    "# }\n",
    "\n",
    "# nc_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4bf03c2",
   "metadata": {},
   "source": [
    "### Sanity check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86d69fb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reduce(intersect, nc_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f32bf183",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_sanity_plot <- function(X, Y, tax) {\n",
    "    print(tax)\n",
    "    temp_plt <- cbind(X, Y) %>%\n",
    "        mutate(log_abundance = log(get(tax) + 0.0001)) %>%\n",
    "        ggplot(aes(x = site_supplying_sample, y = get(tax), fill = site_supplying_sample)) + \n",
    "        geom_point() +\n",
    "        geom_boxplot(alpha = 0.5) +\n",
    "        labs(x = \"Source cohort\", y = paste(tax, \"% abundance\")) +\n",
    "        theme(legend.position = \"none\",\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "          text = element_text(size = 15))\n",
    "    return(temp_plt)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e663c7a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load data\n",
    "taxon <- \"Cutibacterium acnes\"\n",
    "df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.S.tsv\"))\n",
    "\n",
    "# Remove homo and unclassified\n",
    "col_to_exclude <- colnames(df)[grepl(\"Homo|unclassified\", colnames(df))]\n",
    "df <- remove_cols(df, col_to_exclude)\n",
    "\n",
    "# Retain relevant metadata\n",
    "meta <- parse_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "\n",
    "# Normalise to RA\n",
    "normalise <- function(df) {\n",
    "    RA_df <- t(apply(df, 1, function(x) {x / sum(x) * 100}))\n",
    "    return(as_tibble(RA_df))\n",
    "}\n",
    "\n",
    "species_df_RA <- normalise(df)\n",
    "\n",
    "plot_sanity_plot(df, meta, taxon)\n",
    "print(str_glue(\"Max no. of reads = {max(df[, taxon])}\"))\n",
    "print(str_glue(\"Mean no. of reads = {mean(df[, taxon])}\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d60d68d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "l1 <- tibble(l1)\n",
    "\n",
    "fwrite(l1, str_glue(\"/home/projects/14001280/PROJECTS/blood_microbiome/results/decontamination/decontaminated_list.n{n_subset}.csv\"))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
