#!/bin/bash
#PBS -q normal
#PBS -l select=1:ncpus=24:mem=48G
#PBS -l walltime=90:00:00
#PBS -P personal-tancsc
#PBS -N run_kraken2

## Extract unmapped read and mates from CRAM files ##
# Input: CRAM file
# Output: fastq file containing read and mates

source ~/miniconda3/etc/profile.d/conda.sh
conda activate metameta

WKDIR=/scratch/users/astar/gis/tancsc/blood_microbiome_files
SCRIPTS=/home/projects/14001280/PROJECTS/blood_microbiome/pipeline
N_THREADS=24

# Create directories 
mkdir ${WKDIR}
mkdir ${WKDIR}/01_bam ${WKDIR}/01_fastq/ ${WKDIR}/02_fastq/ ${WKDIR}/03_fastq/ ${WKDIR}/04_fastq/ ${WKDIR}/05_fastq/ ${WKDIR}/06_reports/ ${WKDIR}/06_unclassified_fastq/ ${WKDIR}/07_abundance_matrix/

while read line
do
        COHORT=$(echo $line | awk '{split($0, a, ","); print a[1]}')
        MUX=$(echo $line | awk '{split($0, a, ","); print a[2]}')
        SAMPLE=$(echo $line | awk '{split($0, a, ","); print a[3]}')

	echo $COHORT $MUX $SAMPLE
        
	# Step 1: Extract unmapped read and mates from CRAM files
        sh ${SCRIPTS}/01_extract_unmapped_reads_and_mates.sh $COHORT $MUX $SAMPLE $WKDIR $N_THREADS
        
        # Step 2: Trim 5' and 3' ends of reads below Qxx quality and Ns
        sh ${SCRIPTS}/02_trim.sh $COHORT $MUX $SAMPLE $WKDIR $N_THREADS
        
        # Step 3: Filter reads below Qxx
        sh ${SCRIPTS}/03_filter.sh $COHORT $MUX $SAMPLE $WKDIR $N_THREADS
        
        # Step 4: Remove low complexity reads
        sh ${SCRIPTS}/04_remove_low_complexity.sh $COHORT $MUX $SAMPLE $WKDIR $N_THREADS
        
        # Step 5: Host decontamination
        sh ${SCRIPTS}/05_host_decontamination.sh $COHORT $MUX $SAMPLE $WKDIR $N_THREADS
        
        # Step 6: Kraken2 classification
        sh ${SCRIPTS}/06_kraken2_assignment.sh $COHORT $MUX $SAMPLE $WKDIR $N_THREADS

done < /home/projects/14001280/PROJECTS/blood_microbiome/data/subset_list_9999.csv

echo -------SCRIPT COMPLETE------- 
