#!/bin/bash
#PBS -q normal
#PBS -l select=1:ncpus=12:mem=48G
#PBS -l walltime=03:00:00
#PBS -P 14001280
#PBS -N poore_map_to_human

source ~/miniconda3/etc/profile.d/conda.sh
conda activate metagenomics

wkdir=/home/projects/14001280/PROJECTS/blood_microbiome/data/poore_et_al
fastq_dir=$wkdir/raw_fastqs2
out_dir=$wkdir/cram_files
ref_dir=/home/projects/14001280/PROJECTS/blood_microbiome/data/simulation_references

N_THREADS=12

for input1 in $fastq_dir/*.1.fastq.gz
do
        input2=$(echo ${input1}| sed "s|.1.fastq.gz|.2.fastq.gz|g")
        output=$(echo ${input1}| sed "s|.1.fastq.gz|.cram|g"| sed "s|$fastq_dir|$out_dir|g")
        echo $input1
        echo $input2
        echo $output

	bowtie2 -x ${ref_dir}/human/ref \
       		-p ${N_THREADS} \
	        -1 $input1 \
	        -2 $input2 | samtools view -C -T $ref_dir/human/GRCh38_latest_genomic.fna > $output

done
