{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe3f3f5c",
   "metadata": {},
   "source": [
    "# Identication of putative contaminant taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "440f3650",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32maccumulate()\u001b[39m masks \u001b[34mforeach\u001b[39m::accumulate()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mwhen()\u001b[39m       masks \u001b[34mforeach\u001b[39m::when()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Loading required package: compositions\n",
      "\n",
      "Welcome to compositions, a package for compositional data analysis.\n",
      "Find an intro with \"? compositions\"\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘compositions’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    anova, cor, cov, dist, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    %*%, norm, scale, scale.default\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "Loading required package: VennDiagram\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "Loading required package: futile.logger\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(foreach)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(compositions)\n",
    "require(doParallel)\n",
    "require(VennDiagram)\n",
    "registerDoParallel(cores=8)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "215972d1",
   "metadata": {},
   "source": [
    "### Data pre-processing functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b5109a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_data <- function(file_path) {\n",
    "    dat <- as.data.frame(fread(file_path)) %>%\n",
    "        separate(sample, into = c(NA, \"npm_research_id\"), sep = \"\\\\.\")\n",
    "    return(dat)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, dat) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% dat$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "\n",
    "subset_metadata <- function(meta, n_subset) {\n",
    "    meta <- as.data.frame(meta)\n",
    "    cohorts <- unique(meta$site_supplying_sample)\n",
    "    subset_vec <- c()\n",
    "\n",
    "    for (i in cohorts) {\n",
    "        ids <- meta$npm_research_id[meta$site_supplying_sample == i]\n",
    "        \n",
    "        if (length(ids) > n_subset) {\n",
    "            subset_ids <- sample(ids, n_subset)\n",
    "            subset_vec <- c(subset_vec, subset_ids)\n",
    "        } else {\n",
    "            subset_vec <- c(subset_vec, ids)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    meta_sub <- meta %>%\n",
    "      filter(npm_research_id %in% subset_vec) \n",
    "    return(meta_sub)\n",
    "}\n",
    "\n",
    "\n",
    "retrieve_rows_from_meta <- function(dat, meta) {\n",
    "    return(dat %>% filter(npm_research_id %in% meta$npm_research_id)) \n",
    "}\n",
    "\n",
    "\n",
    "remove_cols <- function(dat, col_to_exclude) {\n",
    "    return(dat %>% select(-all_of(col_to_exclude)))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_prev_taxa <- function(dat, frac_presence, read_threshold) {\n",
    "    n_original <- ncol(dat[, colnames(dat) != \"npm_research_id\"])\n",
    "    PA_df <- apply(dat[, 2:ncol(dat)], 2, function(x) {ifelse(x > read_threshold, T, F)})\n",
    "    frac_df <- apply(PA_df, 2, function(x) {sum(x) / nrow(PA_df)})\n",
    "    to_keep <- names(frac_df[frac_df > frac_presence])\n",
    "    to_keep <- c(\"npm_research_id\", to_keep)\n",
    "    n_new <- length(to_keep) - 1\n",
    "    print(str_glue(\"{n_new} / {n_original} taxa are present in {frac_presence} of samples\"))\n",
    "    return(dat %>% select(all_of(to_keep)))\n",
    "}\n",
    "\n",
    "\n",
    "otu_to_RA <- function(dat) {\n",
    "    mat <- as.matrix(dat[, colnames(dat) != \"npm_research_id\"])\n",
    "    RA_df <- as.data.frame(mat / rowSums(mat))\n",
    "    RA_df <- add_column(RA_df, dat$npm_research_id, .before = 1)\n",
    "    colnames(RA_df)[1] <- \"npm_research_id\"\n",
    "    \n",
    "    return(RA_df)\n",
    "}\n",
    "\n",
    "\n",
    "RA_to_clr <- function(dat) {\n",
    "    mat <- dat[, colnames(dat) != \"npm_research_id\"]\n",
    "    clr_df <- clr(mat)\n",
    "    return(cbind(data.frame(npm_research_id = dat$npm_research_id), as.data.frame(clr_df, check.names = F)))\n",
    "}\n",
    "\n",
    "\n",
    "get_metadata_plots <- function(meta, meta_cols) {\n",
    "    meta <- as.matrix(meta)\n",
    "    meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "    plots <- list()\n",
    "    \n",
    "    for (column in meta_cols) {\n",
    "        plt <- meta %>%\n",
    "            mutate(across(everything(), as.character)) %>%\n",
    "            select(all_of(column)) %>%\n",
    "            group_by_at(column) %>%\n",
    "            summarise(n = n()) %>%\n",
    "            ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "                geom_bar(stat = \"identity\") +\n",
    "                theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "                geom_text(aes_string(label = \"n\"))\n",
    "        plots[[column]] <- plt\n",
    "    }\n",
    "    \n",
    "    return(plots)\n",
    "}\n",
    "\n",
    "\n",
    "get_meta_cols <- function(meta) {\n",
    "    meta_cols <- colnames(meta)[grep(\"kit|flow_cell|instrument_id|site_supplying\", colnames(meta))]\n",
    "    to_exclude <- c(\"hiseq_xtm_flow_cell_v2_5_id\")\n",
    "    meta_cols <- meta_cols[!(meta_cols %in% to_exclude)]\n",
    "    return(meta_cols)\n",
    "}\n",
    "\n",
    "\n",
    "print_group_freqs <- function(meta, meta_cols) {\n",
    "    for (col in meta_cols) {\n",
    "        tmp <- tibble(data.frame(meta)) %>%\n",
    "            group_by(get(col)) %>%\n",
    "            summarise(n = n())\n",
    "        print(col)\n",
    "        print(tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "get_batch_prevalence <- function(x) {sum(x) / length(x)}\n",
    "\n",
    "\n",
    "remove_low_sample_levels <- function(dat, metadat, column, min_samples) {\n",
    "    # Remove levels with < x samples\n",
    "    tmp <- tibble(data.frame(metadat)) %>%\n",
    "            group_by(get(column)) %>%\n",
    "            summarise(n = n())\n",
    "\n",
    "    # Vector of group levels to keep\n",
    "    to_keep <- tmp[tmp$n >= min_samples, \"get(column)\"]$`get(column)`\n",
    "    to_keep <- to_keep[to_keep != \"Unknown\"]\n",
    "    n_levels <- length(to_keep)\n",
    "\n",
    "    if (n_levels < 2) {\n",
    "        print(str_glue(\"After pruning, {column} has < 2 levels\"))\n",
    "    } else {\n",
    "        print(str_glue(\"After pruning, {column} has {n_levels} levels\"))\n",
    "    }\n",
    "\n",
    "    # Remove rows in metadata\n",
    "    metadat_filt <- metadat %>% filter(get(column) %in% to_keep)\n",
    "\n",
    "    # Retrieve rows\n",
    "    dat_filt <- retrieve_rows_from_meta(dat, metadat_filt)\n",
    "    metadat_filt <- metadat_filt %>% filter(npm_research_id %in% dat_filt$npm_research_id)\n",
    "\n",
    "    return(list(dat_filt = dat_filt, metadat_filt = metadat_filt))\n",
    "}\n",
    "\n",
    "\n",
    "remove_empty_rows <- function(dat) {\n",
    "    mat <- as.matrix(dat[, colnames(dat) != \"npm_research_id\"])\n",
    "    row_sums <- rowSums(mat)\n",
    "    df_filt <- dat[row_sums != 0, ]\n",
    "    nrow(df_filt)\n",
    "\n",
    "    n_original <- nrow(dat)\n",
    "    n_removed <- n_original - nrow(df_filt)\n",
    "    \n",
    "    print(str_glue(\"{n_removed}/{n_original} samples removed due to having no reads of interest\"))\n",
    "    \n",
    "    return(df_filt)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, dat) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% dat$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a2662ae",
   "metadata": {},
   "source": [
    "#### Params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1cc062e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "human <- \"Homo sapiens\"\n",
    "rank <- \"S\"\n",
    "max_prev_t <- 0.25\n",
    "fold_diff_t <- 2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33dff39c",
   "metadata": {},
   "source": [
    "#### Load and parse data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6fd9cdfa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8828</li><li>871</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8828\n",
       "\\item 871\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8828\n",
       "2. 871\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8828  871"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "to_retain <- fread(\"data/samples_above_100_microbial_reads.txt\")$npm_research_id\n",
    "\n",
    "prev_df <- fread(\"results/decontamination/prevalence_RA0.005_read10.csv\")\n",
    "\n",
    "species_df <- load_data(str_glue(\"data/taxonomic_profiles/07_abundance_matrix/abundance_matrix.S.pipeline2_210322.tsv\")) %>% \n",
    "    filter(npm_research_id %in% to_retain) %>%\n",
    "    select(all_of(colnames(prev_df)))\n",
    "\n",
    "all(colnames(species_df) == colnames(prev_df))\n",
    "\n",
    "meta_filt <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", species_df)\n",
    "\n",
    "# Convert to CLR abundances\n",
    "df_filt <- RA_to_clr(otu_to_RA(species_df))\n",
    "dim(species_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be3d58a4",
   "metadata": {},
   "source": [
    "### Correlation filter within batches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0a968228",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_decontam <- function(dat, non_contaminants, contaminants, meta_col, level) {\n",
    "    nc_list <- foreach (non_contaminant_taxon = non_contaminants) %do% {\n",
    "        morsel <- tibble()\n",
    "        \n",
    "        for (contaminant_taxon in contaminants) {\n",
    "            spearman_test <- cor.test(dat[ , contaminant_taxon], dat[, non_contaminant_taxon])\n",
    "            rho <- spearman_test$estimate\n",
    "            print(str_glue(\"Computing correlations for {meta_col}: {level}\"))\n",
    "            morsel <- morsel %>% bind_rows(tibble(meta_col = meta_col, \n",
    "                                                  level = level,\n",
    "                                                  non_contaminant_taxon = non_contaminant_taxon, \n",
    "                                                  contaminant_taxon = contaminant_taxon,\n",
    "                                                  rho = rho))\n",
    "        }\n",
    "\n",
    "        return(morsel)\n",
    "    }\n",
    "\n",
    "    corr_df <- bind_rows(nc_list)  \n",
    "    return(corr_df)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b8aa3bc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get non-contaminants/contaminants from diff prev. filter\n",
    "diff_prev_contams_nc <- read.csv(str_glue(\"results/decontamination/diff_prev_V3_no_split/noncontaminants.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.txt\"))$taxa\n",
    "diff_prev_contams <- read.csv(str_glue(\"results/decontamination/diff_prev_V3_no_split/contaminants.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.txt\"))$taxa\n",
    "\n",
    "diff_prev <- fread(\"results/decontamination/diff_prev_V3_no_split/decon_V3_raw.RA0.005.read_threshold10.csv\") %>%\n",
    "    as_tibble()\n",
    "\n",
    "diff_filt <- diff_prev %>%\n",
    "    filter(fold_diff > 2, max_prev > 0.25)\n",
    "\n",
    "iter_list <- diff_filt %>% \n",
    "    distinct(meta_col, max_level)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b614c935",
   "metadata": {},
   "outputs": [],
   "source": [
    "morsels <- foreach(i = seq(nrow(iter_list))) %dopar% {\n",
    "    var_name <- iter_list[i, ]$meta_col\n",
    "    level_name <- iter_list[i, ]$max_level\n",
    "\n",
    "    contams <- (diff_filt %>% filter(meta_col == var_name, max_level == level_name))$taxa\n",
    "\n",
    "    noncontams <- colnames(df_filt)\n",
    "    noncontams <- noncontams[!(noncontams %in% c(\"npm_research_id\", contams))]\n",
    "\n",
    "    batch_dat <- df_filt %>%\n",
    "        left_join(meta_filt %>% select(all_of(c(\"npm_research_id\", var_name)))) %>%\n",
    "        filter(get(var_name) == level_name)\n",
    "\n",
    "    corr_decontam(batch_dat, noncontams, contams, var_name, level_name)\n",
    "}\n",
    "\n",
    "corr_res <- bind_rows(morsels)\n",
    "# corr_res <- fread(\"results/decontamination/correlation_decontamination_no_split/corr_raw.diff_prev_V3.RA0.005.read_threshold10.max_prev0.25.fold_diff2.corr_t0.7.within_batch.S.n9999.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "665b1e0e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 817281 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>meta_col</th><th scope=col>level</th><th scope=col>non_contaminant_taxon</th><th scope=col>contaminant_taxon</th><th scope=col>rho</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingobium sp. YG1                </td><td>0.9289072</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingobium yanoikuyae             </td><td>0.9367592</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingobium baderi                 </td><td>0.9574772</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. LM7               </td><td>0.8393325</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. Cra20             </td><td>0.8395500</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. CL5.1             </td><td>0.9118260</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. IC081             </td><td>0.9681173</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. XS-10             </td><td>0.8675847</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. FARSPH            </td><td>0.8638212</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. LK11              </td><td>0.8976870</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. DH-S5             </td><td>0.8707972</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. NBWT7             </td><td>0.8946001</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. MM-1              </td><td>0.8874182</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. AP4-R1            </td><td>0.8986929</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. AAP5              </td><td>0.8814412</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas hengshuiensis         </td><td>0.8438169</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas koreensis             </td><td>0.8585692</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sanxanigenens         </td><td>0.9215114</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas alpina                </td><td>0.8594948</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas paucimobilis          </td><td>0.8996606</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas taxi                  </td><td>0.8676601</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas panacisoli            </td><td>0.8759012</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas panacis               </td><td>0.8885410</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Novosphingobium sp. ABRDHK2        </td><td>0.9118342</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingosinithalassobacter sp. zrk23</td><td>0.8786016</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingosinithalassobacter sp. CS137</td><td>0.8696049</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. BTAi1           </td><td>0.2632180</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. CCBAU 051011    </td><td>0.2203451</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. CCBAU 51753     </td><td>0.2678297</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium erythrophlei        </td><td>0.2707322</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Primate erythroparvovirus 1                </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Primate erythroparvovirus 1                </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia felis                           </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia felis                           </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Paracoccus haeundaensis                    </td><td>Microbacterium chocolatum</td><td>-0.06738632</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Paracoccus haeundaensis                    </td><td>Bosea sp. RAC05          </td><td>-0.05779772</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia helvetica                       </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia helvetica                       </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia rhipicephali                    </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia rhipicephali                    </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia endosymbiont of Ixodes pacificus</td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia endosymbiont of Ixodes pacificus</td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Woolly monkey hepatitis B virus            </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Woolly monkey hepatitis B virus            </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Human betaherpesvirus 5                    </td><td>Microbacterium chocolatum</td><td>-0.06099780</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Human betaherpesvirus 5                    </td><td>Bosea sp. RAC05          </td><td>-0.06480933</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Human alphaherpesvirus 2                   </td><td>Microbacterium chocolatum</td><td>-0.04234247</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Human alphaherpesvirus 2                   </td><td>Bosea sp. RAC05          </td><td>-0.05230832</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Streptococcus sp. FDAARGOS_522             </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Streptococcus sp. FDAARGOS_522             </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Torque teno virus 2                        </td><td>Microbacterium chocolatum</td><td> 0.19967735</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Torque teno virus 2                        </td><td>Bosea sp. RAC05          </td><td> 0.17061056</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Bacillus sp. TK-2                          </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Bacillus sp. TK-2                          </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia massiliae                       </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Rickettsia massiliae                       </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Lactobacillus phage Lv-1                   </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Lactobacillus phage Lv-1                   </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Capuchin monkey hepatitis B virus          </td><td>Microbacterium chocolatum</td><td>         NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20302809</td><td>Capuchin monkey hepatitis B virus          </td><td>Bosea sp. RAC05          </td><td>         NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 817281 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " meta\\_col & level & non\\_contaminant\\_taxon & contaminant\\_taxon & rho\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingobium sp. YG1                 & 0.9289072\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingobium yanoikuyae              & 0.9367592\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingobium baderi                  & 0.9574772\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. LM7                & 0.8393325\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. Cra20              & 0.8395500\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. CL5.1              & 0.9118260\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. IC081              & 0.9681173\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. XS-10              & 0.8675847\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. FARSPH             & 0.8638212\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. LK11               & 0.8976870\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. DH-S5              & 0.8707972\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. NBWT7              & 0.8946001\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. MM-1               & 0.8874182\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. AP4-R1             & 0.8986929\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. AAP5               & 0.8814412\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas hengshuiensis          & 0.8438169\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas koreensis              & 0.8585692\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sanxanigenens          & 0.9215114\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas alpina                 & 0.8594948\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas paucimobilis           & 0.8996606\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas taxi                   & 0.8676601\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas panacisoli             & 0.8759012\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas panacis                & 0.8885410\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Novosphingobium sp. ABRDHK2         & 0.9118342\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingosinithalassobacter sp. zrk23 & 0.8786016\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingosinithalassobacter sp. CS137 & 0.8696049\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. BTAi1            & 0.2632180\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. CCBAU 051011     & 0.2203451\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. CCBAU 51753      & 0.2678297\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium erythrophlei         & 0.2707322\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Primate erythroparvovirus 1                 & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Primate erythroparvovirus 1                 & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia felis                            & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia felis                            & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Paracoccus haeundaensis                     & Microbacterium chocolatum & -0.06738632\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Paracoccus haeundaensis                     & Bosea sp. RAC05           & -0.05779772\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia helvetica                        & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia helvetica                        & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia rhipicephali                     & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia rhipicephali                     & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia endosymbiont of Ixodes pacificus & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia endosymbiont of Ixodes pacificus & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Woolly monkey hepatitis B virus             & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Woolly monkey hepatitis B virus             & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Human betaherpesvirus 5                     & Microbacterium chocolatum & -0.06099780\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Human betaherpesvirus 5                     & Bosea sp. RAC05           & -0.06480933\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Human alphaherpesvirus 2                    & Microbacterium chocolatum & -0.04234247\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Human alphaherpesvirus 2                    & Bosea sp. RAC05           & -0.05230832\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Streptococcus sp. FDAARGOS\\_522              & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Streptococcus sp. FDAARGOS\\_522              & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Torque teno virus 2                         & Microbacterium chocolatum &  0.19967735\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Torque teno virus 2                         & Bosea sp. RAC05           &  0.17061056\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Bacillus sp. TK-2                           & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Bacillus sp. TK-2                           & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia massiliae                        & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Rickettsia massiliae                        & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Lactobacillus phage Lv-1                    & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Lactobacillus phage Lv-1                    & Bosea sp. RAC05           &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Capuchin monkey hepatitis B virus           & Microbacterium chocolatum &          NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20302809 & Capuchin monkey hepatitis B virus           & Bosea sp. RAC05           &          NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 817281 × 5\n",
       "\n",
       "| meta_col &lt;chr&gt; | level &lt;chr&gt; | non_contaminant_taxon &lt;chr&gt; | contaminant_taxon &lt;chr&gt; | rho &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingobium sp. YG1                 | 0.9289072 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingobium yanoikuyae              | 0.9367592 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingobium baderi                  | 0.9574772 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. LM7                | 0.8393325 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. Cra20              | 0.8395500 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. CL5.1              | 0.9118260 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. IC081              | 0.9681173 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. XS-10              | 0.8675847 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. FARSPH             | 0.8638212 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. LK11               | 0.8976870 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. DH-S5              | 0.8707972 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. NBWT7              | 0.8946001 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. MM-1               | 0.8874182 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. AP4-R1             | 0.8986929 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. AAP5               | 0.8814412 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas hengshuiensis          | 0.8438169 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas koreensis              | 0.8585692 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sanxanigenens          | 0.9215114 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas alpina                 | 0.8594948 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas paucimobilis           | 0.8996606 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas taxi                   | 0.8676601 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas panacisoli             | 0.8759012 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas panacis                | 0.8885410 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Novosphingobium sp. ABRDHK2         | 0.9118342 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingosinithalassobacter sp. zrk23 | 0.8786016 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingosinithalassobacter sp. CS137 | 0.8696049 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. BTAi1            | 0.2632180 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. CCBAU 051011     | 0.2203451 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. CCBAU 51753      | 0.2678297 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium erythrophlei         | 0.2707322 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Primate erythroparvovirus 1                 | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Primate erythroparvovirus 1                 | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia felis                            | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia felis                            | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Paracoccus haeundaensis                     | Microbacterium chocolatum | -0.06738632 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Paracoccus haeundaensis                     | Bosea sp. RAC05           | -0.05779772 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia helvetica                        | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia helvetica                        | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia rhipicephali                     | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia rhipicephali                     | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia endosymbiont of Ixodes pacificus | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia endosymbiont of Ixodes pacificus | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Woolly monkey hepatitis B virus             | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Woolly monkey hepatitis B virus             | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Human betaherpesvirus 5                     | Microbacterium chocolatum | -0.06099780 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Human betaherpesvirus 5                     | Bosea sp. RAC05           | -0.06480933 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Human alphaherpesvirus 2                    | Microbacterium chocolatum | -0.04234247 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Human alphaherpesvirus 2                    | Bosea sp. RAC05           | -0.05230832 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Streptococcus sp. FDAARGOS_522              | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Streptococcus sp. FDAARGOS_522              | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Torque teno virus 2                         | Microbacterium chocolatum |  0.19967735 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Torque teno virus 2                         | Bosea sp. RAC05           |  0.17061056 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Bacillus sp. TK-2                           | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Bacillus sp. TK-2                           | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia massiliae                        | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Rickettsia massiliae                        | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Lactobacillus phage Lv-1                    | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Lactobacillus phage Lv-1                    | Bosea sp. RAC05           |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Capuchin monkey hepatitis B virus           | Microbacterium chocolatum |          NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20302809 | Capuchin monkey hepatitis B virus           | Bosea sp. RAC05           |          NA |\n",
       "\n"
      ],
      "text/plain": [
       "       meta_col                     level   \n",
       "1      site_supplying_sample        MEC     \n",
       "2      site_supplying_sample        MEC     \n",
       "3      site_supplying_sample        MEC     \n",
       "4      site_supplying_sample        MEC     \n",
       "5      site_supplying_sample        MEC     \n",
       "6      site_supplying_sample        MEC     \n",
       "7      site_supplying_sample        MEC     \n",
       "8      site_supplying_sample        MEC     \n",
       "9      site_supplying_sample        MEC     \n",
       "10     site_supplying_sample        MEC     \n",
       "11     site_supplying_sample        MEC     \n",
       "12     site_supplying_sample        MEC     \n",
       "13     site_supplying_sample        MEC     \n",
       "14     site_supplying_sample        MEC     \n",
       "15     site_supplying_sample        MEC     \n",
       "16     site_supplying_sample        MEC     \n",
       "17     site_supplying_sample        MEC     \n",
       "18     site_supplying_sample        MEC     \n",
       "19     site_supplying_sample        MEC     \n",
       "20     site_supplying_sample        MEC     \n",
       "21     site_supplying_sample        MEC     \n",
       "22     site_supplying_sample        MEC     \n",
       "23     site_supplying_sample        MEC     \n",
       "24     site_supplying_sample        MEC     \n",
       "25     site_supplying_sample        MEC     \n",
       "26     site_supplying_sample        MEC     \n",
       "27     site_supplying_sample        MEC     \n",
       "28     site_supplying_sample        MEC     \n",
       "29     site_supplying_sample        MEC     \n",
       "30     site_supplying_sample        MEC     \n",
       "⋮      ⋮                            ⋮       \n",
       "817252 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817253 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817254 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817255 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817256 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817257 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817258 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817259 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817260 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817261 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817262 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817263 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817264 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817265 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817266 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817267 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817268 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817269 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817270 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817271 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817272 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817273 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817274 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817275 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817276 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817277 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817278 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817279 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817280 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "817281 hiseq_xtm_flow_cell_v2_5_lot 20302809\n",
       "       non_contaminant_taxon                      \n",
       "1      Sphingobium sp. RAC03                      \n",
       "2      Sphingobium sp. RAC03                      \n",
       "3      Sphingobium sp. RAC03                      \n",
       "4      Sphingobium sp. RAC03                      \n",
       "5      Sphingobium sp. RAC03                      \n",
       "6      Sphingobium sp. RAC03                      \n",
       "7      Sphingobium sp. RAC03                      \n",
       "8      Sphingobium sp. RAC03                      \n",
       "9      Sphingobium sp. RAC03                      \n",
       "10     Sphingobium sp. RAC03                      \n",
       "11     Sphingobium sp. RAC03                      \n",
       "12     Sphingobium sp. RAC03                      \n",
       "13     Sphingobium sp. RAC03                      \n",
       "14     Sphingobium sp. RAC03                      \n",
       "15     Sphingobium sp. RAC03                      \n",
       "16     Sphingobium sp. RAC03                      \n",
       "17     Sphingobium sp. RAC03                      \n",
       "18     Sphingobium sp. RAC03                      \n",
       "19     Sphingobium sp. RAC03                      \n",
       "20     Sphingobium sp. RAC03                      \n",
       "21     Sphingobium sp. RAC03                      \n",
       "22     Sphingobium sp. RAC03                      \n",
       "23     Sphingobium sp. RAC03                      \n",
       "24     Sphingobium sp. RAC03                      \n",
       "25     Sphingobium sp. RAC03                      \n",
       "26     Sphingobium sp. RAC03                      \n",
       "27     Sphingobium sp. RAC03                      \n",
       "28     Sphingobium sp. RAC03                      \n",
       "29     Sphingobium sp. RAC03                      \n",
       "30     Sphingobium sp. RAC03                      \n",
       "⋮      ⋮                                          \n",
       "817252 Primate erythroparvovirus 1                \n",
       "817253 Primate erythroparvovirus 1                \n",
       "817254 Rickettsia felis                           \n",
       "817255 Rickettsia felis                           \n",
       "817256 Paracoccus haeundaensis                    \n",
       "817257 Paracoccus haeundaensis                    \n",
       "817258 Rickettsia helvetica                       \n",
       "817259 Rickettsia helvetica                       \n",
       "817260 Rickettsia rhipicephali                    \n",
       "817261 Rickettsia rhipicephali                    \n",
       "817262 Rickettsia endosymbiont of Ixodes pacificus\n",
       "817263 Rickettsia endosymbiont of Ixodes pacificus\n",
       "817264 Woolly monkey hepatitis B virus            \n",
       "817265 Woolly monkey hepatitis B virus            \n",
       "817266 Human betaherpesvirus 5                    \n",
       "817267 Human betaherpesvirus 5                    \n",
       "817268 Human alphaherpesvirus 2                   \n",
       "817269 Human alphaherpesvirus 2                   \n",
       "817270 Streptococcus sp. FDAARGOS_522             \n",
       "817271 Streptococcus sp. FDAARGOS_522             \n",
       "817272 Torque teno virus 2                        \n",
       "817273 Torque teno virus 2                        \n",
       "817274 Bacillus sp. TK-2                          \n",
       "817275 Bacillus sp. TK-2                          \n",
       "817276 Rickettsia massiliae                       \n",
       "817277 Rickettsia massiliae                       \n",
       "817278 Lactobacillus phage Lv-1                   \n",
       "817279 Lactobacillus phage Lv-1                   \n",
       "817280 Capuchin monkey hepatitis B virus          \n",
       "817281 Capuchin monkey hepatitis B virus          \n",
       "       contaminant_taxon                   rho        \n",
       "1      Sphingobium sp. YG1                 0.9289072  \n",
       "2      Sphingobium yanoikuyae              0.9367592  \n",
       "3      Sphingobium baderi                  0.9574772  \n",
       "4      Sphingomonas sp. LM7                0.8393325  \n",
       "5      Sphingomonas sp. Cra20              0.8395500  \n",
       "6      Sphingomonas sp. CL5.1              0.9118260  \n",
       "7      Sphingomonas sp. IC081              0.9681173  \n",
       "8      Sphingomonas sp. XS-10              0.8675847  \n",
       "9      Sphingomonas sp. FARSPH             0.8638212  \n",
       "10     Sphingomonas sp. LK11               0.8976870  \n",
       "11     Sphingomonas sp. DH-S5              0.8707972  \n",
       "12     Sphingomonas sp. NBWT7              0.8946001  \n",
       "13     Sphingomonas sp. MM-1               0.8874182  \n",
       "14     Sphingomonas sp. AP4-R1             0.8986929  \n",
       "15     Sphingomonas sp. AAP5               0.8814412  \n",
       "16     Sphingomonas hengshuiensis          0.8438169  \n",
       "17     Sphingomonas koreensis              0.8585692  \n",
       "18     Sphingomonas sanxanigenens          0.9215114  \n",
       "19     Sphingomonas alpina                 0.8594948  \n",
       "20     Sphingomonas paucimobilis           0.8996606  \n",
       "21     Sphingomonas taxi                   0.8676601  \n",
       "22     Sphingomonas panacisoli             0.8759012  \n",
       "23     Sphingomonas panacis                0.8885410  \n",
       "24     Novosphingobium sp. ABRDHK2         0.9118342  \n",
       "25     Sphingosinithalassobacter sp. zrk23 0.8786016  \n",
       "26     Sphingosinithalassobacter sp. CS137 0.8696049  \n",
       "27     Bradyrhizobium sp. BTAi1            0.2632180  \n",
       "28     Bradyrhizobium sp. CCBAU 051011     0.2203451  \n",
       "29     Bradyrhizobium sp. CCBAU 51753      0.2678297  \n",
       "30     Bradyrhizobium erythrophlei         0.2707322  \n",
       "⋮      ⋮                                   ⋮          \n",
       "817252 Microbacterium chocolatum                    NA\n",
       "817253 Bosea sp. RAC05                              NA\n",
       "817254 Microbacterium chocolatum                    NA\n",
       "817255 Bosea sp. RAC05                              NA\n",
       "817256 Microbacterium chocolatum           -0.06738632\n",
       "817257 Bosea sp. RAC05                     -0.05779772\n",
       "817258 Microbacterium chocolatum                    NA\n",
       "817259 Bosea sp. RAC05                              NA\n",
       "817260 Microbacterium chocolatum                    NA\n",
       "817261 Bosea sp. RAC05                              NA\n",
       "817262 Microbacterium chocolatum                    NA\n",
       "817263 Bosea sp. RAC05                              NA\n",
       "817264 Microbacterium chocolatum                    NA\n",
       "817265 Bosea sp. RAC05                              NA\n",
       "817266 Microbacterium chocolatum           -0.06099780\n",
       "817267 Bosea sp. RAC05                     -0.06480933\n",
       "817268 Microbacterium chocolatum           -0.04234247\n",
       "817269 Bosea sp. RAC05                     -0.05230832\n",
       "817270 Microbacterium chocolatum                    NA\n",
       "817271 Bosea sp. RAC05                              NA\n",
       "817272 Microbacterium chocolatum            0.19967735\n",
       "817273 Bosea sp. RAC05                      0.17061056\n",
       "817274 Microbacterium chocolatum                    NA\n",
       "817275 Bosea sp. RAC05                              NA\n",
       "817276 Microbacterium chocolatum                    NA\n",
       "817277 Bosea sp. RAC05                              NA\n",
       "817278 Microbacterium chocolatum                    NA\n",
       "817279 Bosea sp. RAC05                              NA\n",
       "817280 Microbacterium chocolatum                    NA\n",
       "817281 Bosea sp. RAC05                              NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "corr_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ad84e36b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "corr_c <- corr_res %>% \n",
    "    filter(rho > 0.7) %>%\n",
    "    distinct(non_contaminant_taxon) %>%\n",
    "    rename(contaminants = non_contaminant_taxon)\n",
    "\n",
    "corr_nc <- tibble(non_contaminant_taxon = colnames(df_filt)[!(colnames(df_filt) %in% c(\"npm_research_id\", \n",
    "                                                                                       corr_c$contaminants, \n",
    "                                                                                       diff_prev_contams))])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b34eb0e5",
   "metadata": {},
   "source": [
    "### Parse and save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c5ee1861",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 340\n",
      "[1] 526\n"
     ]
    }
   ],
   "source": [
    "corr_t <- 0.7\n",
    "\n",
    "fwrite(corr_res,  \n",
    "       str_glue(\"results/decontamination/correlation_decontamination_no_split/corr_raw.diff_prev_V3.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.corr_t{corr_t}.within_batch.S.n9999.txt\"))\n",
    "fwrite(corr_c, \n",
    "       str_glue(\"results/decontamination/correlation_decontamination_no_split/contam.diff_prev_V3.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.corr_t{corr_t}.within_batch.S.n9999.txt\"))\n",
    "fwrite(corr_nc, \n",
    "       str_glue(\"results/decontamination/correlation_decontamination_no_split/nc.diff_prev_V3.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.corr_t{corr_t}.within_batch.S.n9999.txt\"))\n",
    "\n",
    "# print(length(corr_nc$non_contaminant_taxon))\n",
    "print(length(corr_c$contaminants))\n",
    "print(length(corr_nc$non_contaminant_taxon))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "763028c0",
   "metadata": {},
   "source": [
    "## Plot correlation heatmaps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "93bc8abf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot_heatmap <- function(dat) {\n",
    "#     as.data.frame(cor(dat)) %>%\n",
    "#         rownames_to_column(\"taxa1\") %>%\n",
    "#         pivot_longer(!taxa1, names_to = \"taxa2\", values_to = \"rho\") %>%\n",
    "#                 mutate(fill_color = case_when(rho <= 0 ~ \"<0\",\n",
    "#                          rho > 0 & rho <= 0.1 ~ \"0-0.1\",\n",
    "#                          rho > 0.1 & rho <= 0.2 ~ \"0.1-0.2\",\n",
    "#                          rho > 0.2 & rho <= 0.3 ~ \"0.2-0.3\",\n",
    "#                          rho > 0.3 & rho <= 0.5 ~  \"0.3-0.5\",\n",
    "#                          rho > 0.5 & rho <= 0.7 ~ \"0.5-0.7\",\n",
    "#                          rho > 0.7 ~ \">0.7\")) %>%\n",
    "#         mutate(fill_color = factor(fill_color, levels = c(\"<0\", \"0-0.1\", \"0.1-0.2\", \"0.2-0.3\", \"0.3-0.5\", \"0.5-0.7\", \">0.7\"))) %>%\n",
    "#         ggplot(aes(x = taxa1, y = taxa2, fill = fill_color)) + \n",
    "#         geom_tile() +\n",
    "#         scale_fill_manual(values = c(\"#6699cc\", \"#ffffcc\", \"#f39b7fb2\", \"#e64b35b2\", \"#d11141\", \"darkred\", \"black\"), drop = F) +\n",
    "#         theme(axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "#               axis.title = element_blank()) +\n",
    "#     labs(fill = \"Rho\", title = str_glue(\"n = {nrow(dat)}\"))\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d20f2e0b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# plot_iter <- list(c(\"site_supplying_sample\", \"MEC\"), \n",
    "#                   c(\"extraction_kit\", \"Chemagic DNA Blood Kit (Perkin Elmer, MA)\"),\n",
    "#                   c(\"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\", \"20392972\"),\n",
    "#                   c(\"hiseq_xtm_flow_cell_v2_5_lot\", \"20367079\"))\n",
    "\n",
    "# for (item in plot_iter) {\n",
    "#     var_name <- item[1]\n",
    "#     level_name <- item[2]\n",
    "#     mec_contam <- corr_res %>% \n",
    "#         filter(level == level_name) %>%\n",
    "#         filter(rho > 0.7)\n",
    "#     mec_contam <- c(mec_contam$contaminant_taxon, mec_contam$non_contaminant_taxon)\n",
    "\n",
    "#     mec_nc <- corr_res %>% \n",
    "#         filter(level == level_name) %>%\n",
    "#         filter(rho <= 0.7)\n",
    "#     mec_nc <- mec_nc$non_contaminant_taxon\n",
    "\n",
    "#     contam_df <- df_filt %>% \n",
    "#         left_join(meta_filt) %>%\n",
    "#         filter(get(var_name) == level_name) %>%\n",
    "#         select(all_of(mec_contam))\n",
    "#     nc_df <- df_filt %>% \n",
    "#         left_join(meta_filt) %>%\n",
    "#         filter(get(var_name) == level_name) %>%\n",
    "#         select(all_of(mec_nc))\n",
    "\n",
    "#     plot_heatmap(contam_df)\n",
    "#     ggsave(str_glue(\"results/decontamination/correlation_decontamination/correlation_contaminants.{var_name}.{level_name}.png\"), \n",
    "#            dpi = 300, \n",
    "#            height = 6.67, width = 7.5)\n",
    "#     plot_heatmap(nc_df)\n",
    "#         ggsave(str_glue(\"results/decontamination/correlation_decontamination/correlation_non_contaminants.{var_name}.{level_name}.png\"), \n",
    "#            dpi = 300, \n",
    "#            height = 6.67, width = 7.5)\n",
    "# }\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
