{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe3f3f5c",
   "metadata": {},
   "source": [
    "# Identication of putative contaminant taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "440f3650",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32maccumulate()\u001b[39m masks \u001b[34mforeach\u001b[39m::accumulate()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mwhen()\u001b[39m       masks \u001b[34mforeach\u001b[39m::when()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Loading required package: compositions\n",
      "\n",
      "Welcome to compositions, a package for compositional data analysis.\n",
      "Find an intro with \"? compositions\"\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘compositions’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    anova, cor, cov, dist, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    %*%, norm, scale, scale.default\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "Loading required package: VennDiagram\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "Loading required package: futile.logger\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(foreach)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(compositions)\n",
    "require(doParallel)\n",
    "require(VennDiagram)\n",
    "registerDoParallel(cores=8)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "215972d1",
   "metadata": {},
   "source": [
    "### Data pre-processing functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b5109a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_data <- function(file_path) {\n",
    "    dat <- as.data.frame(fread(file_path)) %>%\n",
    "        separate(sample, into = c(NA, \"npm_research_id\"), sep = \"\\\\.\")\n",
    "    return(dat)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, dat) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% dat$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "\n",
    "subset_metadata <- function(meta, n_subset) {\n",
    "    meta <- as.data.frame(meta)\n",
    "    cohorts <- unique(meta$site_supplying_sample)\n",
    "    subset_vec <- c()\n",
    "\n",
    "    for (i in cohorts) {\n",
    "        ids <- meta$npm_research_id[meta$site_supplying_sample == i]\n",
    "        \n",
    "        if (length(ids) > n_subset) {\n",
    "            subset_ids <- sample(ids, n_subset)\n",
    "            subset_vec <- c(subset_vec, subset_ids)\n",
    "        } else {\n",
    "            subset_vec <- c(subset_vec, ids)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    meta_sub <- meta %>%\n",
    "      filter(npm_research_id %in% subset_vec) \n",
    "    return(meta_sub)\n",
    "}\n",
    "\n",
    "\n",
    "retrieve_rows_from_meta <- function(dat, meta) {\n",
    "    return(dat %>% filter(npm_research_id %in% meta$npm_research_id)) \n",
    "}\n",
    "\n",
    "\n",
    "remove_cols <- function(dat, col_to_exclude) {\n",
    "    return(dat %>% select(-all_of(col_to_exclude)))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_prev_taxa <- function(dat, frac_presence, read_threshold) {\n",
    "    n_original <- ncol(dat[, colnames(dat) != \"npm_research_id\"])\n",
    "    PA_df <- apply(dat[, 2:ncol(dat)], 2, function(x) {ifelse(x > read_threshold, T, F)})\n",
    "    frac_df <- apply(PA_df, 2, function(x) {sum(x) / nrow(PA_df)})\n",
    "    to_keep <- names(frac_df[frac_df > frac_presence])\n",
    "    to_keep <- c(\"npm_research_id\", to_keep)\n",
    "    n_new <- length(to_keep) - 1\n",
    "    print(str_glue(\"{n_new} / {n_original} taxa are present in {frac_presence} of samples\"))\n",
    "    return(dat %>% select(all_of(to_keep)))\n",
    "}\n",
    "\n",
    "\n",
    "otu_to_RA <- function(dat) {\n",
    "    mat <- as.matrix(dat[, colnames(dat) != \"npm_research_id\"])\n",
    "    RA_df <- as.data.frame(mat / rowSums(mat))\n",
    "    RA_df <- add_column(RA_df, dat$npm_research_id, .before = 1)\n",
    "    colnames(RA_df)[1] <- \"npm_research_id\"\n",
    "    \n",
    "    return(RA_df)\n",
    "}\n",
    "\n",
    "\n",
    "RA_to_clr <- function(dat) {\n",
    "    mat <- dat[, colnames(dat) != \"npm_research_id\"]\n",
    "    clr_df <- clr(mat)\n",
    "    return(cbind(data.frame(npm_research_id = dat$npm_research_id), as.data.frame(clr_df, check.names = F)))\n",
    "}\n",
    "\n",
    "\n",
    "get_metadata_plots <- function(meta, meta_cols) {\n",
    "    meta <- as.matrix(meta)\n",
    "    meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "    plots <- list()\n",
    "    \n",
    "    for (column in meta_cols) {\n",
    "        plt <- meta %>%\n",
    "            mutate(across(everything(), as.character)) %>%\n",
    "            select(all_of(column)) %>%\n",
    "            group_by_at(column) %>%\n",
    "            summarise(n = n()) %>%\n",
    "            ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "                geom_bar(stat = \"identity\") +\n",
    "                theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "                geom_text(aes_string(label = \"n\"))\n",
    "        plots[[column]] <- plt\n",
    "    }\n",
    "    \n",
    "    return(plots)\n",
    "}\n",
    "\n",
    "\n",
    "get_meta_cols <- function(meta) {\n",
    "    meta_cols <- colnames(meta)[grep(\"kit|flow_cell|instrument_id|site_supplying\", colnames(meta))]\n",
    "    to_exclude <- c(\"library_prep_kit\", \"hiseq_xtm_flow_cell_v2_5_id\")\n",
    "    meta_cols <- meta_cols[!(meta_cols %in% to_exclude)]\n",
    "    return(meta_cols)\n",
    "}\n",
    "\n",
    "\n",
    "print_group_freqs <- function(meta, meta_cols) {\n",
    "    for (col in meta_cols) {\n",
    "        tmp <- tibble(data.frame(meta)) %>%\n",
    "            group_by(get(col)) %>%\n",
    "            summarise(n = n())\n",
    "        print(col)\n",
    "        print(tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "get_batch_prevalence <- function(x) {sum(x) / length(x)}\n",
    "\n",
    "\n",
    "remove_low_sample_levels <- function(dat, metadat, column, min_samples) {\n",
    "    # Remove levels with < x samples\n",
    "    tmp <- tibble(data.frame(metadat)) %>%\n",
    "            group_by(get(column)) %>%\n",
    "            summarise(n = n())\n",
    "\n",
    "    # Vector of group levels to keep\n",
    "    to_keep <- tmp[tmp$n >= min_samples, \"get(column)\"]$`get(column)`\n",
    "    to_keep <- to_keep[to_keep != \"Unknown\"]\n",
    "    n_levels <- length(to_keep)\n",
    "\n",
    "    if (n_levels < 2) {\n",
    "        print(str_glue(\"After pruning, {column} has < 2 levels\"))\n",
    "    } else {\n",
    "        print(str_glue(\"After pruning, {column} has {n_levels} levels\"))\n",
    "    }\n",
    "\n",
    "    # Remove rows in metadata\n",
    "    metadat_filt <- metadat %>% filter(get(column) %in% to_keep)\n",
    "\n",
    "    # Retrieve rows\n",
    "    dat_filt <- retrieve_rows_from_meta(dat, metadat_filt)\n",
    "    metadat_filt <- metadat_filt %>% filter(npm_research_id %in% dat_filt$npm_research_id)\n",
    "\n",
    "    return(list(dat_filt = dat_filt, metadat_filt = metadat_filt))\n",
    "}\n",
    "\n",
    "\n",
    "remove_empty_rows <- function(dat) {\n",
    "    mat <- as.matrix(dat[, colnames(dat) != \"npm_research_id\"])\n",
    "    row_sums <- rowSums(mat)\n",
    "    df_filt <- dat[row_sums != 0, ]\n",
    "    nrow(df_filt)\n",
    "\n",
    "    n_original <- nrow(dat)\n",
    "    n_removed <- n_original - nrow(df_filt)\n",
    "    \n",
    "    print(str_glue(\"{n_removed}/{n_original} samples removed due to having no reads of interest\"))\n",
    "    \n",
    "    return(df_filt)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, dat) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% dat$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a2662ae",
   "metadata": {},
   "source": [
    "#### Params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1cc062e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "human <- \"Homo sapiens\"\n",
    "rank <- \"S\"\n",
    "max_prev_t <- 0.25\n",
    "fold_diff_t <- 2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33dff39c",
   "metadata": {},
   "source": [
    "#### Load and parse data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6fd9cdfa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8828</li><li>871</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8828\n",
       "\\item 871\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8828\n",
       "2. 871\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8828  871"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "to_retain <- fread(\"data/samples_above_100_microbial_reads.txt\")$npm_research_id\n",
    "\n",
    "prev_df <- fread(\"results/decontamination/prevalence_RA0.005_read10.csv\")\n",
    "\n",
    "species_df <- load_data(str_glue(\"data/taxonomic_profiles/07_abundance_matrix/abundance_matrix.S.pipeline2_210322.tsv\")) %>% \n",
    "    filter(npm_research_id %in% to_retain) %>%\n",
    "    select(all_of(colnames(prev_df)))\n",
    "\n",
    "all(colnames(species_df) == colnames(prev_df))\n",
    "\n",
    "meta_filt <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", species_df)\n",
    "\n",
    "# Convert to CLR abundances\n",
    "df_filt <- RA_to_clr(otu_to_RA(species_df))\n",
    "dim(species_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be3d58a4",
   "metadata": {},
   "source": [
    "### Correlation filter within batches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0a968228",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_decontam <- function(dat, non_contaminants, contaminants, meta_col, level) {\n",
    "    nc_list <- foreach (non_contaminant_taxon = non_contaminants) %dopar% {\n",
    "        morsel <- tibble()\n",
    "        \n",
    "        for (contaminant_taxon in contaminants) {\n",
    "            spearman_test <- cor.test(dat[ , contaminant_taxon], dat[, non_contaminant_taxon])\n",
    "            rho <- spearman_test$estimate\n",
    "            print(str_glue(\"Computing correlations for {meta_col}: {level}\"))\n",
    "            morsel <- morsel %>% bind_rows(tibble(meta_col = meta_col, \n",
    "                                                  level = level,\n",
    "                                                  non_contaminant_taxon = non_contaminant_taxon, \n",
    "                                                  contaminant_taxon = contaminant_taxon,\n",
    "                                                  rho = rho))\n",
    "        }\n",
    "\n",
    "        return(morsel)\n",
    "    }\n",
    "\n",
    "    corr_df <- bind_rows(nc_list)  \n",
    "    return(corr_df)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b8aa3bc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get non-contaminants/contaminants from diff prev. filter\n",
    "diff_prev_contams_nc <- read.csv(str_glue(\"results/decontamination/diff_prev_V3/noncontaminants.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.txt\"))$taxa\n",
    "diff_prev_contams <- read.csv(str_glue(\"results/decontamination/diff_prev_V3/contaminants.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.txt\"))$taxa\n",
    "\n",
    "diff_prev <- fread(\"results/decontamination/diff_prev_V3/decon_V3_raw.RA0.005.read_threshold10.csv\") %>%\n",
    "    as_tibble()\n",
    "\n",
    "diff_filt <- diff_prev %>%\n",
    "    filter(fold_diff > 2, max_prev > 0.25)\n",
    "\n",
    "iter_list <- diff_filt %>% \n",
    "    distinct(meta_col, max_level)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b614c935",
   "metadata": {},
   "outputs": [],
   "source": [
    "# morsels <- foreach(i = seq(nrow(iter_list))) %dopar% {\n",
    "#     var_name <- iter_list[i, ]$meta_col\n",
    "#     level_name <- iter_list[i, ]$max_level\n",
    "\n",
    "#     contams <- (diff_filt %>% filter(meta_col == var_name, max_level == level_name))$taxa\n",
    "\n",
    "#     noncontams <- colnames(df_filt)\n",
    "#     noncontams <- noncontams[!(noncontams %in% c(\"npm_research_id\", contams))]\n",
    "\n",
    "#     batch_dat <- df_filt %>%\n",
    "#         left_join(meta_filt %>% select(all_of(c(\"npm_research_id\", var_name)))) %>%\n",
    "#         filter(get(var_name) == level_name)\n",
    "\n",
    "#     corr_decontam(batch_dat, noncontams, contams, var_name, level_name)\n",
    "# }\n",
    "\n",
    "# corr_res <- bind_rows(morsels)\n",
    "corr_res <- fread(\"results/decontamination/correlation_decontamination/corr_raw.diff_prev_V3.RA0.005.read_threshold10.max_prev0.25.fold_diff2.corr_t0.7.within_batch.S.n9999.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "665b1e0e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 726972 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>meta_col</th><th scope=col>level</th><th scope=col>non_contaminant_taxon</th><th scope=col>contaminant_taxon</th><th scope=col>rho</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingobium sp. YG1                </td><td>0.9289072</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingobium yanoikuyae             </td><td>0.9367592</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingobium baderi                 </td><td>0.9574772</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. LM7               </td><td>0.8393325</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. Cra20             </td><td>0.8395500</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. CL5.1             </td><td>0.9118260</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. IC081             </td><td>0.9681173</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. XS-10             </td><td>0.8675847</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. FARSPH            </td><td>0.8638212</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. LK11              </td><td>0.8976870</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. DH-S5             </td><td>0.8707972</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. NBWT7             </td><td>0.8946001</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. MM-1              </td><td>0.8874182</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. AP4-R1            </td><td>0.8986929</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sp. AAP5              </td><td>0.8814412</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas hengshuiensis         </td><td>0.8438169</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas koreensis             </td><td>0.8585692</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas sanxanigenens         </td><td>0.9215114</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas alpina                </td><td>0.8594948</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas paucimobilis          </td><td>0.8996606</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas taxi                  </td><td>0.8676601</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas panacisoli            </td><td>0.8759012</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingomonas panacis               </td><td>0.8885410</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Novosphingobium sp. ABRDHK2        </td><td>0.9118342</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingosinithalassobacter sp. zrk23</td><td>0.8786016</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Sphingosinithalassobacter sp. CS137</td><td>0.8696049</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. BTAi1           </td><td>0.2632180</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. PSBB068         </td><td>0.4630786</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. CCBAU 051011    </td><td>0.2203451</td></tr>\n",
       "\t<tr><td>site_supplying_sample</td><td>MEC</td><td>Sphingobium sp. RAC03</td><td>Bradyrhizobium sp. CCBAU 51753     </td><td>0.2678297</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Rickettsia endosymbiont of Ixodes pacificus</td><td>Achromobacter denitrificans</td><td>-0.003038504</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Rickettsia endosymbiont of Ixodes pacificus</td><td>Achromobacter spanius      </td><td> 0.012366081</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Rickettsia endosymbiont of Ixodes pacificus</td><td>Achromobacter sp. AONIH1   </td><td>-0.016858115</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Woolly monkey hepatitis B virus            </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Woolly monkey hepatitis B virus            </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Woolly monkey hepatitis B virus            </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Human betaherpesvirus 5                    </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Human betaherpesvirus 5                    </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Human betaherpesvirus 5                    </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Human alphaherpesvirus 2                   </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Human alphaherpesvirus 2                   </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Human alphaherpesvirus 2                   </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Streptococcus sp. FDAARGOS_522             </td><td>Achromobacter denitrificans</td><td> 0.037425043</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Streptococcus sp. FDAARGOS_522             </td><td>Achromobacter spanius      </td><td> 0.089340519</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Streptococcus sp. FDAARGOS_522             </td><td>Achromobacter sp. AONIH1   </td><td> 0.041405038</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Torque teno virus 2                        </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Torque teno virus 2                        </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Torque teno virus 2                        </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Bacillus sp. TK-2                          </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Bacillus sp. TK-2                          </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Bacillus sp. TK-2                          </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Rickettsia massiliae                       </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Rickettsia massiliae                       </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Rickettsia massiliae                       </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Lactobacillus phage Lv-1                   </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Lactobacillus phage Lv-1                   </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Lactobacillus phage Lv-1                   </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Capuchin monkey hepatitis B virus          </td><td>Achromobacter denitrificans</td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Capuchin monkey hepatitis B virus          </td><td>Achromobacter spanius      </td><td>          NA</td></tr>\n",
       "\t<tr><td>hiseq_xtm_flow_cell_v2_5_lot</td><td>20347589</td><td>Capuchin monkey hepatitis B virus          </td><td>Achromobacter sp. AONIH1   </td><td>          NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 726972 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " meta\\_col & level & non\\_contaminant\\_taxon & contaminant\\_taxon & rho\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingobium sp. YG1                 & 0.9289072\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingobium yanoikuyae              & 0.9367592\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingobium baderi                  & 0.9574772\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. LM7                & 0.8393325\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. Cra20              & 0.8395500\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. CL5.1              & 0.9118260\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. IC081              & 0.9681173\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. XS-10              & 0.8675847\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. FARSPH             & 0.8638212\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. LK11               & 0.8976870\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. DH-S5              & 0.8707972\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. NBWT7              & 0.8946001\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. MM-1               & 0.8874182\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. AP4-R1             & 0.8986929\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sp. AAP5               & 0.8814412\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas hengshuiensis          & 0.8438169\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas koreensis              & 0.8585692\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas sanxanigenens          & 0.9215114\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas alpina                 & 0.8594948\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas paucimobilis           & 0.8996606\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas taxi                   & 0.8676601\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas panacisoli             & 0.8759012\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingomonas panacis                & 0.8885410\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Novosphingobium sp. ABRDHK2         & 0.9118342\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingosinithalassobacter sp. zrk23 & 0.8786016\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Sphingosinithalassobacter sp. CS137 & 0.8696049\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. BTAi1            & 0.2632180\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. PSBB068          & 0.4630786\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. CCBAU 051011     & 0.2203451\\\\\n",
       "\t site\\_supplying\\_sample & MEC & Sphingobium sp. RAC03 & Bradyrhizobium sp. CCBAU 51753      & 0.2678297\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Rickettsia endosymbiont of Ixodes pacificus & Achromobacter denitrificans & -0.003038504\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Rickettsia endosymbiont of Ixodes pacificus & Achromobacter spanius       &  0.012366081\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Rickettsia endosymbiont of Ixodes pacificus & Achromobacter sp. AONIH1    & -0.016858115\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Woolly monkey hepatitis B virus             & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Woolly monkey hepatitis B virus             & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Woolly monkey hepatitis B virus             & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Human betaherpesvirus 5                     & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Human betaherpesvirus 5                     & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Human betaherpesvirus 5                     & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Human alphaherpesvirus 2                    & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Human alphaherpesvirus 2                    & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Human alphaherpesvirus 2                    & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Streptococcus sp. FDAARGOS\\_522              & Achromobacter denitrificans &  0.037425043\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Streptococcus sp. FDAARGOS\\_522              & Achromobacter spanius       &  0.089340519\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Streptococcus sp. FDAARGOS\\_522              & Achromobacter sp. AONIH1    &  0.041405038\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Torque teno virus 2                         & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Torque teno virus 2                         & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Torque teno virus 2                         & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Bacillus sp. TK-2                           & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Bacillus sp. TK-2                           & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Bacillus sp. TK-2                           & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Rickettsia massiliae                        & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Rickettsia massiliae                        & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Rickettsia massiliae                        & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Lactobacillus phage Lv-1                    & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Lactobacillus phage Lv-1                    & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Lactobacillus phage Lv-1                    & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Capuchin monkey hepatitis B virus           & Achromobacter denitrificans &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Capuchin monkey hepatitis B virus           & Achromobacter spanius       &           NA\\\\\n",
       "\t hiseq\\_xtm\\_flow\\_cell\\_v2\\_5\\_lot & 20347589 & Capuchin monkey hepatitis B virus           & Achromobacter sp. AONIH1    &           NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 726972 × 5\n",
       "\n",
       "| meta_col &lt;chr&gt; | level &lt;chr&gt; | non_contaminant_taxon &lt;chr&gt; | contaminant_taxon &lt;chr&gt; | rho &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingobium sp. YG1                 | 0.9289072 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingobium yanoikuyae              | 0.9367592 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingobium baderi                  | 0.9574772 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. LM7                | 0.8393325 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. Cra20              | 0.8395500 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. CL5.1              | 0.9118260 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. IC081              | 0.9681173 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. XS-10              | 0.8675847 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. FARSPH             | 0.8638212 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. LK11               | 0.8976870 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. DH-S5              | 0.8707972 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. NBWT7              | 0.8946001 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. MM-1               | 0.8874182 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. AP4-R1             | 0.8986929 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sp. AAP5               | 0.8814412 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas hengshuiensis          | 0.8438169 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas koreensis              | 0.8585692 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas sanxanigenens          | 0.9215114 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas alpina                 | 0.8594948 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas paucimobilis           | 0.8996606 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas taxi                   | 0.8676601 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas panacisoli             | 0.8759012 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingomonas panacis                | 0.8885410 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Novosphingobium sp. ABRDHK2         | 0.9118342 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingosinithalassobacter sp. zrk23 | 0.8786016 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Sphingosinithalassobacter sp. CS137 | 0.8696049 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. BTAi1            | 0.2632180 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. PSBB068          | 0.4630786 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. CCBAU 051011     | 0.2203451 |\n",
       "| site_supplying_sample | MEC | Sphingobium sp. RAC03 | Bradyrhizobium sp. CCBAU 51753      | 0.2678297 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Rickettsia endosymbiont of Ixodes pacificus | Achromobacter denitrificans | -0.003038504 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Rickettsia endosymbiont of Ixodes pacificus | Achromobacter spanius       |  0.012366081 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Rickettsia endosymbiont of Ixodes pacificus | Achromobacter sp. AONIH1    | -0.016858115 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Woolly monkey hepatitis B virus             | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Woolly monkey hepatitis B virus             | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Woolly monkey hepatitis B virus             | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Human betaherpesvirus 5                     | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Human betaherpesvirus 5                     | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Human betaherpesvirus 5                     | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Human alphaherpesvirus 2                    | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Human alphaherpesvirus 2                    | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Human alphaherpesvirus 2                    | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Streptococcus sp. FDAARGOS_522              | Achromobacter denitrificans |  0.037425043 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Streptococcus sp. FDAARGOS_522              | Achromobacter spanius       |  0.089340519 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Streptococcus sp. FDAARGOS_522              | Achromobacter sp. AONIH1    |  0.041405038 |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Torque teno virus 2                         | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Torque teno virus 2                         | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Torque teno virus 2                         | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Bacillus sp. TK-2                           | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Bacillus sp. TK-2                           | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Bacillus sp. TK-2                           | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Rickettsia massiliae                        | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Rickettsia massiliae                        | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Rickettsia massiliae                        | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Lactobacillus phage Lv-1                    | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Lactobacillus phage Lv-1                    | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Lactobacillus phage Lv-1                    | Achromobacter sp. AONIH1    |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Capuchin monkey hepatitis B virus           | Achromobacter denitrificans |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Capuchin monkey hepatitis B virus           | Achromobacter spanius       |           NA |\n",
       "| hiseq_xtm_flow_cell_v2_5_lot | 20347589 | Capuchin monkey hepatitis B virus           | Achromobacter sp. AONIH1    |           NA |\n",
       "\n"
      ],
      "text/plain": [
       "       meta_col                     level   \n",
       "1      site_supplying_sample        MEC     \n",
       "2      site_supplying_sample        MEC     \n",
       "3      site_supplying_sample        MEC     \n",
       "4      site_supplying_sample        MEC     \n",
       "5      site_supplying_sample        MEC     \n",
       "6      site_supplying_sample        MEC     \n",
       "7      site_supplying_sample        MEC     \n",
       "8      site_supplying_sample        MEC     \n",
       "9      site_supplying_sample        MEC     \n",
       "10     site_supplying_sample        MEC     \n",
       "11     site_supplying_sample        MEC     \n",
       "12     site_supplying_sample        MEC     \n",
       "13     site_supplying_sample        MEC     \n",
       "14     site_supplying_sample        MEC     \n",
       "15     site_supplying_sample        MEC     \n",
       "16     site_supplying_sample        MEC     \n",
       "17     site_supplying_sample        MEC     \n",
       "18     site_supplying_sample        MEC     \n",
       "19     site_supplying_sample        MEC     \n",
       "20     site_supplying_sample        MEC     \n",
       "21     site_supplying_sample        MEC     \n",
       "22     site_supplying_sample        MEC     \n",
       "23     site_supplying_sample        MEC     \n",
       "24     site_supplying_sample        MEC     \n",
       "25     site_supplying_sample        MEC     \n",
       "26     site_supplying_sample        MEC     \n",
       "27     site_supplying_sample        MEC     \n",
       "28     site_supplying_sample        MEC     \n",
       "29     site_supplying_sample        MEC     \n",
       "30     site_supplying_sample        MEC     \n",
       "⋮      ⋮                            ⋮       \n",
       "726943 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726944 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726945 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726946 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726947 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726948 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726949 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726950 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726951 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726952 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726953 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726954 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726955 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726956 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726957 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726958 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726959 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726960 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726961 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726962 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726963 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726964 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726965 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726966 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726967 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726968 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726969 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726970 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726971 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "726972 hiseq_xtm_flow_cell_v2_5_lot 20347589\n",
       "       non_contaminant_taxon                      \n",
       "1      Sphingobium sp. RAC03                      \n",
       "2      Sphingobium sp. RAC03                      \n",
       "3      Sphingobium sp. RAC03                      \n",
       "4      Sphingobium sp. RAC03                      \n",
       "5      Sphingobium sp. RAC03                      \n",
       "6      Sphingobium sp. RAC03                      \n",
       "7      Sphingobium sp. RAC03                      \n",
       "8      Sphingobium sp. RAC03                      \n",
       "9      Sphingobium sp. RAC03                      \n",
       "10     Sphingobium sp. RAC03                      \n",
       "11     Sphingobium sp. RAC03                      \n",
       "12     Sphingobium sp. RAC03                      \n",
       "13     Sphingobium sp. RAC03                      \n",
       "14     Sphingobium sp. RAC03                      \n",
       "15     Sphingobium sp. RAC03                      \n",
       "16     Sphingobium sp. RAC03                      \n",
       "17     Sphingobium sp. RAC03                      \n",
       "18     Sphingobium sp. RAC03                      \n",
       "19     Sphingobium sp. RAC03                      \n",
       "20     Sphingobium sp. RAC03                      \n",
       "21     Sphingobium sp. RAC03                      \n",
       "22     Sphingobium sp. RAC03                      \n",
       "23     Sphingobium sp. RAC03                      \n",
       "24     Sphingobium sp. RAC03                      \n",
       "25     Sphingobium sp. RAC03                      \n",
       "26     Sphingobium sp. RAC03                      \n",
       "27     Sphingobium sp. RAC03                      \n",
       "28     Sphingobium sp. RAC03                      \n",
       "29     Sphingobium sp. RAC03                      \n",
       "30     Sphingobium sp. RAC03                      \n",
       "⋮      ⋮                                          \n",
       "726943 Rickettsia endosymbiont of Ixodes pacificus\n",
       "726944 Rickettsia endosymbiont of Ixodes pacificus\n",
       "726945 Rickettsia endosymbiont of Ixodes pacificus\n",
       "726946 Woolly monkey hepatitis B virus            \n",
       "726947 Woolly monkey hepatitis B virus            \n",
       "726948 Woolly monkey hepatitis B virus            \n",
       "726949 Human betaherpesvirus 5                    \n",
       "726950 Human betaherpesvirus 5                    \n",
       "726951 Human betaherpesvirus 5                    \n",
       "726952 Human alphaherpesvirus 2                   \n",
       "726953 Human alphaherpesvirus 2                   \n",
       "726954 Human alphaherpesvirus 2                   \n",
       "726955 Streptococcus sp. FDAARGOS_522             \n",
       "726956 Streptococcus sp. FDAARGOS_522             \n",
       "726957 Streptococcus sp. FDAARGOS_522             \n",
       "726958 Torque teno virus 2                        \n",
       "726959 Torque teno virus 2                        \n",
       "726960 Torque teno virus 2                        \n",
       "726961 Bacillus sp. TK-2                          \n",
       "726962 Bacillus sp. TK-2                          \n",
       "726963 Bacillus sp. TK-2                          \n",
       "726964 Rickettsia massiliae                       \n",
       "726965 Rickettsia massiliae                       \n",
       "726966 Rickettsia massiliae                       \n",
       "726967 Lactobacillus phage Lv-1                   \n",
       "726968 Lactobacillus phage Lv-1                   \n",
       "726969 Lactobacillus phage Lv-1                   \n",
       "726970 Capuchin monkey hepatitis B virus          \n",
       "726971 Capuchin monkey hepatitis B virus          \n",
       "726972 Capuchin monkey hepatitis B virus          \n",
       "       contaminant_taxon                   rho         \n",
       "1      Sphingobium sp. YG1                 0.9289072   \n",
       "2      Sphingobium yanoikuyae              0.9367592   \n",
       "3      Sphingobium baderi                  0.9574772   \n",
       "4      Sphingomonas sp. LM7                0.8393325   \n",
       "5      Sphingomonas sp. Cra20              0.8395500   \n",
       "6      Sphingomonas sp. CL5.1              0.9118260   \n",
       "7      Sphingomonas sp. IC081              0.9681173   \n",
       "8      Sphingomonas sp. XS-10              0.8675847   \n",
       "9      Sphingomonas sp. FARSPH             0.8638212   \n",
       "10     Sphingomonas sp. LK11               0.8976870   \n",
       "11     Sphingomonas sp. DH-S5              0.8707972   \n",
       "12     Sphingomonas sp. NBWT7              0.8946001   \n",
       "13     Sphingomonas sp. MM-1               0.8874182   \n",
       "14     Sphingomonas sp. AP4-R1             0.8986929   \n",
       "15     Sphingomonas sp. AAP5               0.8814412   \n",
       "16     Sphingomonas hengshuiensis          0.8438169   \n",
       "17     Sphingomonas koreensis              0.8585692   \n",
       "18     Sphingomonas sanxanigenens          0.9215114   \n",
       "19     Sphingomonas alpina                 0.8594948   \n",
       "20     Sphingomonas paucimobilis           0.8996606   \n",
       "21     Sphingomonas taxi                   0.8676601   \n",
       "22     Sphingomonas panacisoli             0.8759012   \n",
       "23     Sphingomonas panacis                0.8885410   \n",
       "24     Novosphingobium sp. ABRDHK2         0.9118342   \n",
       "25     Sphingosinithalassobacter sp. zrk23 0.8786016   \n",
       "26     Sphingosinithalassobacter sp. CS137 0.8696049   \n",
       "27     Bradyrhizobium sp. BTAi1            0.2632180   \n",
       "28     Bradyrhizobium sp. PSBB068          0.4630786   \n",
       "29     Bradyrhizobium sp. CCBAU 051011     0.2203451   \n",
       "30     Bradyrhizobium sp. CCBAU 51753      0.2678297   \n",
       "⋮      ⋮                                   ⋮           \n",
       "726943 Achromobacter denitrificans         -0.003038504\n",
       "726944 Achromobacter spanius                0.012366081\n",
       "726945 Achromobacter sp. AONIH1            -0.016858115\n",
       "726946 Achromobacter denitrificans                   NA\n",
       "726947 Achromobacter spanius                         NA\n",
       "726948 Achromobacter sp. AONIH1                      NA\n",
       "726949 Achromobacter denitrificans                   NA\n",
       "726950 Achromobacter spanius                         NA\n",
       "726951 Achromobacter sp. AONIH1                      NA\n",
       "726952 Achromobacter denitrificans                   NA\n",
       "726953 Achromobacter spanius                         NA\n",
       "726954 Achromobacter sp. AONIH1                      NA\n",
       "726955 Achromobacter denitrificans          0.037425043\n",
       "726956 Achromobacter spanius                0.089340519\n",
       "726957 Achromobacter sp. AONIH1             0.041405038\n",
       "726958 Achromobacter denitrificans                   NA\n",
       "726959 Achromobacter spanius                         NA\n",
       "726960 Achromobacter sp. AONIH1                      NA\n",
       "726961 Achromobacter denitrificans                   NA\n",
       "726962 Achromobacter spanius                         NA\n",
       "726963 Achromobacter sp. AONIH1                      NA\n",
       "726964 Achromobacter denitrificans                   NA\n",
       "726965 Achromobacter spanius                         NA\n",
       "726966 Achromobacter sp. AONIH1                      NA\n",
       "726967 Achromobacter denitrificans                   NA\n",
       "726968 Achromobacter spanius                         NA\n",
       "726969 Achromobacter sp. AONIH1                      NA\n",
       "726970 Achromobacter denitrificans                   NA\n",
       "726971 Achromobacter spanius                         NA\n",
       "726972 Achromobacter sp. AONIH1                      NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "corr_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ad84e36b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Sphingobium sp. YG1'</li><li>'Sphingobium yanoikuyae'</li><li>'Sphingobium baderi'</li><li>'Sphingomonas sp. LM7'</li><li>'Sphingomonas sp. Cra20'</li><li>'Sphingomonas sp. CL5.1'</li><li>'Sphingomonas sp. IC081'</li><li>'Sphingomonas sp. XS-10'</li><li>'Sphingomonas sp. FARSPH'</li><li>'Sphingomonas sp. LK11'</li><li>'Sphingomonas sp. DH-S5'</li><li>'Sphingomonas sp. NBWT7'</li><li>'Sphingomonas sp. MM-1'</li><li>'Sphingomonas sp. AP4-R1'</li><li>'Sphingomonas sp. AAP5'</li><li>'Sphingomonas hengshuiensis'</li><li>'Sphingomonas koreensis'</li><li>'Sphingomonas sanxanigenens'</li><li>'Sphingomonas alpina'</li><li>'Sphingomonas melonis'</li><li>'Sphingomonas paucimobilis'</li><li>'Sphingomonas taxi'</li><li>'Sphingomonas panacisoli'</li><li>'Sphingomonas panacis'</li><li>'Novosphingobium sp. ABRDHK2'</li><li>'Sphingosinithalassobacter sp. zrk23'</li><li>'Sphingosinithalassobacter sp. CS137'</li><li>'Bradyrhizobium sp. BTAi1'</li><li>'Bradyrhizobium sp. PSBB068'</li><li>'Bradyrhizobium sp. CCBAU 051011'</li><li>'Bradyrhizobium sp. SK17'</li><li>'Bradyrhizobium sp. CCBAU 51753'</li><li>'Bradyrhizobium erythrophlei'</li><li>'Bradyrhizobium diazoefficiens'</li><li>'Bradyrhizobium lablabi'</li><li>'Bradyrhizobium betae'</li><li>'Bradyrhizobium canariense'</li><li>'Bradyrhizobium icense'</li><li>'Rhodopseudomonas palustris'</li><li>'Tardiphaga robiniae'</li><li>'Afipia carboxidovorans'</li><li>'Methylobacterium currus'</li><li>'Methylobacterium sp. 17Sr1-43'</li><li>'Methylobacterium aquaticum'</li><li>'Methylobacterium durans'</li><li>'Caulobacter flavus'</li><li>'Caulobacter mirabilis'</li><li>'Caulobacter sp. K31'</li><li>'Caulobacter soli'</li><li>'Caulobacter vibrioides'</li><li>'Caulobacter rhizosphaerae'</li><li>'Burkholderia contaminans'</li><li>'Ralstonia solanacearum'</li><li>'Ralstonia insidiosa'</li><li>'Ralstonia pickettii'</li><li>'Kinneretia sp. DAIF2'</li><li>'Achromobacter xylosoxidans'</li><li>'Pseudomonas alcaligenes'</li><li>'Pseudomonas aeruginosa'</li><li>'Pseudomonas stutzeri'</li><li>'Caulobacter segnis'</li><li>'Caulobacter sp. FWC26'</li><li>'Caulobacter henricii'</li><li>'Methylobacterium brachiatum'</li><li>'Alcaligenes sp. SORT26'</li><li>'Alcaligenes faecalis'</li><li>'Paraburkholderia fungorum'</li><li>'Herbaspirillum huttiense'</li><li>'Acinetobacter johnsonii'</li><li>'Pseudomonas mendocina'</li><li>'Pseudomonas sp. phDV1'</li><li>'Pseudomonas sp. LPH1'</li><li>'Pseudomonas toyotomiensis'</li><li>'Pseudomonas sihuiensis'</li><li>'Pseudomonas sediminis'</li><li>'Microbacterium sp. BH-3-3-3'</li><li>'Microbacterium testaceum'</li><li>'Microbacterium chocolatum'</li><li>'Brevibacterium casei'</li><li>'Rhodococcus sp. P-2'</li><li>'Ralstonia mannitolilytica'</li><li>'Pseudomonas sp. B11D7D'</li><li>'Pseudomonas oleovorans'</li><li>'Pseudomonas sp. MSPm1'</li><li>'Pseudomonas alcaliphila'</li><li>'Bordetella genomosp. 13'</li><li>'Achromobacter deleyi'</li><li>'Microcella alkaliphila'</li><li>'Sphingobium sp. RAC03'</li><li>'Bradyrhizobium sp. ORS 285'</li><li>'Bradyrhizobium sp. CCBAU 53421'</li><li>'Bradyrhizobium sp. 6(2017)'</li><li>'Bradyrhizobium paxllaeri'</li><li>'Nitrobacter hamburgensis'</li><li>'Methylorubrum populi'</li><li>'Brevundimonas sp. CS1'</li><li>'Sphingomonas sp. LMO-1'</li><li>'Sphingomonas sp. NIC1'</li><li>'Sphingomonas sp. HMP6'</li><li>'Sphingomonas sp. HMP9'</li><li>'Sphingomonas wittichii'</li><li>'Sphingomonas insulae'</li><li>'Bradyrhizobium sp. ORS 278'</li><li>'Bradyrhizobium sp. CCGE-LA001'</li><li>'Bradyrhizobium sp. 1(2017)'</li><li>'Bradyrhizobium sp. CCBAU 53338'</li><li>'Bradyrhizobium sp. CCBAU 51765'</li><li>'Bradyrhizobium sp. CCBAU 53351'</li><li>'Bradyrhizobium sp. CCBAU 53340'</li><li>'Bradyrhizobium japonicum'</li><li>'Bradyrhizobium genosp. L'</li><li>'Bradyrhizobium guangzhouense'</li><li>'Bradyrhizobium zhanjiangense'</li><li>'Bradyrhizobium oligotrophicum'</li><li>'Bradyrhizobium vignae'</li><li>'Bradyrhizobium guangdongense'</li><li>'Bradyrhizobium symbiodeficiens'</li><li>'Bradyrhizobium genosp. B'</li><li>'Bradyrhizobium cosmicum'</li><li>'Afipia sp. GAS231'</li><li>'Bosea sp. Tri-49'</li><li>'Nitrobacter winogradskyi'</li><li>'Methylobacterium sp. WL1'</li><li>'Methylobacterium terrae'</li><li>'Methylobacterium indicum'</li><li>'Methylobacterium mesophilicum'</li><li>'Methylorubrum extorquens'</li><li>'Brevundimonas sp. SGAir0440'</li><li>'Microbacterium hominis'</li><li>'Brevundimonas sp. GW460-12-10-14-LB2'</li><li>'Methylobacterium phyllosphaerae'</li><li>'Methylobacterium sp. 4-46'</li><li>'Methylobacterium nodulans'</li><li>'Rhizobium oryzihabitans'</li><li>'Achromobacter denitrificans'</li><li>'Delftia acidovorans'</li><li>'Delftia tsuruhatensis'</li><li>'Massilia oculi'</li><li>'Microbacterium sp. 10M-3C3'</li><li>'Methylobacterium sp. 17Sr1-1'</li><li>'Methylobacterium sp. AMS5'</li><li>'Bosea sp. RAC05'</li><li>'Phreatobacter cathodiphilus'</li><li>'Achromobacter spanius'</li><li>'Achromobacter sp. AONIH1'</li><li>'Leifsonia shinshuensis'</li><li>'Methylobacterium sp. DM1'</li><li>'Massilia sp. LPB0304'</li><li>'Microbacterium lemovicicum'</li><li>'Bradyrhizobium sp.'</li><li>'Phreatobacter stygius'</li><li>'Microbacterium oleivorans'</li><li>'Methylobacterium sp. C1'</li><li>'Methylobacterium radiotolerans'</li><li>'Phyllobacterium sp. 628'</li><li>'Stenotrophomonas maltophilia'</li><li>'Methylobacterium sp. XJLW'</li><li>'Methylorubrum zatmanii'</li><li>'Toxoplasma gondii'</li><li>'Prevotella copri'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Sphingobium sp. YG1'\n",
       "\\item 'Sphingobium yanoikuyae'\n",
       "\\item 'Sphingobium baderi'\n",
       "\\item 'Sphingomonas sp. LM7'\n",
       "\\item 'Sphingomonas sp. Cra20'\n",
       "\\item 'Sphingomonas sp. CL5.1'\n",
       "\\item 'Sphingomonas sp. IC081'\n",
       "\\item 'Sphingomonas sp. XS-10'\n",
       "\\item 'Sphingomonas sp. FARSPH'\n",
       "\\item 'Sphingomonas sp. LK11'\n",
       "\\item 'Sphingomonas sp. DH-S5'\n",
       "\\item 'Sphingomonas sp. NBWT7'\n",
       "\\item 'Sphingomonas sp. MM-1'\n",
       "\\item 'Sphingomonas sp. AP4-R1'\n",
       "\\item 'Sphingomonas sp. AAP5'\n",
       "\\item 'Sphingomonas hengshuiensis'\n",
       "\\item 'Sphingomonas koreensis'\n",
       "\\item 'Sphingomonas sanxanigenens'\n",
       "\\item 'Sphingomonas alpina'\n",
       "\\item 'Sphingomonas melonis'\n",
       "\\item 'Sphingomonas paucimobilis'\n",
       "\\item 'Sphingomonas taxi'\n",
       "\\item 'Sphingomonas panacisoli'\n",
       "\\item 'Sphingomonas panacis'\n",
       "\\item 'Novosphingobium sp. ABRDHK2'\n",
       "\\item 'Sphingosinithalassobacter sp. zrk23'\n",
       "\\item 'Sphingosinithalassobacter sp. CS137'\n",
       "\\item 'Bradyrhizobium sp. BTAi1'\n",
       "\\item 'Bradyrhizobium sp. PSBB068'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 051011'\n",
       "\\item 'Bradyrhizobium sp. SK17'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 51753'\n",
       "\\item 'Bradyrhizobium erythrophlei'\n",
       "\\item 'Bradyrhizobium diazoefficiens'\n",
       "\\item 'Bradyrhizobium lablabi'\n",
       "\\item 'Bradyrhizobium betae'\n",
       "\\item 'Bradyrhizobium canariense'\n",
       "\\item 'Bradyrhizobium icense'\n",
       "\\item 'Rhodopseudomonas palustris'\n",
       "\\item 'Tardiphaga robiniae'\n",
       "\\item 'Afipia carboxidovorans'\n",
       "\\item 'Methylobacterium currus'\n",
       "\\item 'Methylobacterium sp. 17Sr1-43'\n",
       "\\item 'Methylobacterium aquaticum'\n",
       "\\item 'Methylobacterium durans'\n",
       "\\item 'Caulobacter flavus'\n",
       "\\item 'Caulobacter mirabilis'\n",
       "\\item 'Caulobacter sp. K31'\n",
       "\\item 'Caulobacter soli'\n",
       "\\item 'Caulobacter vibrioides'\n",
       "\\item 'Caulobacter rhizosphaerae'\n",
       "\\item 'Burkholderia contaminans'\n",
       "\\item 'Ralstonia solanacearum'\n",
       "\\item 'Ralstonia insidiosa'\n",
       "\\item 'Ralstonia pickettii'\n",
       "\\item 'Kinneretia sp. DAIF2'\n",
       "\\item 'Achromobacter xylosoxidans'\n",
       "\\item 'Pseudomonas alcaligenes'\n",
       "\\item 'Pseudomonas aeruginosa'\n",
       "\\item 'Pseudomonas stutzeri'\n",
       "\\item 'Caulobacter segnis'\n",
       "\\item 'Caulobacter sp. FWC26'\n",
       "\\item 'Caulobacter henricii'\n",
       "\\item 'Methylobacterium brachiatum'\n",
       "\\item 'Alcaligenes sp. SORT26'\n",
       "\\item 'Alcaligenes faecalis'\n",
       "\\item 'Paraburkholderia fungorum'\n",
       "\\item 'Herbaspirillum huttiense'\n",
       "\\item 'Acinetobacter johnsonii'\n",
       "\\item 'Pseudomonas mendocina'\n",
       "\\item 'Pseudomonas sp. phDV1'\n",
       "\\item 'Pseudomonas sp. LPH1'\n",
       "\\item 'Pseudomonas toyotomiensis'\n",
       "\\item 'Pseudomonas sihuiensis'\n",
       "\\item 'Pseudomonas sediminis'\n",
       "\\item 'Microbacterium sp. BH-3-3-3'\n",
       "\\item 'Microbacterium testaceum'\n",
       "\\item 'Microbacterium chocolatum'\n",
       "\\item 'Brevibacterium casei'\n",
       "\\item 'Rhodococcus sp. P-2'\n",
       "\\item 'Ralstonia mannitolilytica'\n",
       "\\item 'Pseudomonas sp. B11D7D'\n",
       "\\item 'Pseudomonas oleovorans'\n",
       "\\item 'Pseudomonas sp. MSPm1'\n",
       "\\item 'Pseudomonas alcaliphila'\n",
       "\\item 'Bordetella genomosp. 13'\n",
       "\\item 'Achromobacter deleyi'\n",
       "\\item 'Microcella alkaliphila'\n",
       "\\item 'Sphingobium sp. RAC03'\n",
       "\\item 'Bradyrhizobium sp. ORS 285'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 53421'\n",
       "\\item 'Bradyrhizobium sp. 6(2017)'\n",
       "\\item 'Bradyrhizobium paxllaeri'\n",
       "\\item 'Nitrobacter hamburgensis'\n",
       "\\item 'Methylorubrum populi'\n",
       "\\item 'Brevundimonas sp. CS1'\n",
       "\\item 'Sphingomonas sp. LMO-1'\n",
       "\\item 'Sphingomonas sp. NIC1'\n",
       "\\item 'Sphingomonas sp. HMP6'\n",
       "\\item 'Sphingomonas sp. HMP9'\n",
       "\\item 'Sphingomonas wittichii'\n",
       "\\item 'Sphingomonas insulae'\n",
       "\\item 'Bradyrhizobium sp. ORS 278'\n",
       "\\item 'Bradyrhizobium sp. CCGE-LA001'\n",
       "\\item 'Bradyrhizobium sp. 1(2017)'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 53338'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 51765'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 53351'\n",
       "\\item 'Bradyrhizobium sp. CCBAU 53340'\n",
       "\\item 'Bradyrhizobium japonicum'\n",
       "\\item 'Bradyrhizobium genosp. L'\n",
       "\\item 'Bradyrhizobium guangzhouense'\n",
       "\\item 'Bradyrhizobium zhanjiangense'\n",
       "\\item 'Bradyrhizobium oligotrophicum'\n",
       "\\item 'Bradyrhizobium vignae'\n",
       "\\item 'Bradyrhizobium guangdongense'\n",
       "\\item 'Bradyrhizobium symbiodeficiens'\n",
       "\\item 'Bradyrhizobium genosp. B'\n",
       "\\item 'Bradyrhizobium cosmicum'\n",
       "\\item 'Afipia sp. GAS231'\n",
       "\\item 'Bosea sp. Tri-49'\n",
       "\\item 'Nitrobacter winogradskyi'\n",
       "\\item 'Methylobacterium sp. WL1'\n",
       "\\item 'Methylobacterium terrae'\n",
       "\\item 'Methylobacterium indicum'\n",
       "\\item 'Methylobacterium mesophilicum'\n",
       "\\item 'Methylorubrum extorquens'\n",
       "\\item 'Brevundimonas sp. SGAir0440'\n",
       "\\item 'Microbacterium hominis'\n",
       "\\item 'Brevundimonas sp. GW460-12-10-14-LB2'\n",
       "\\item 'Methylobacterium phyllosphaerae'\n",
       "\\item 'Methylobacterium sp. 4-46'\n",
       "\\item 'Methylobacterium nodulans'\n",
       "\\item 'Rhizobium oryzihabitans'\n",
       "\\item 'Achromobacter denitrificans'\n",
       "\\item 'Delftia acidovorans'\n",
       "\\item 'Delftia tsuruhatensis'\n",
       "\\item 'Massilia oculi'\n",
       "\\item 'Microbacterium sp. 10M-3C3'\n",
       "\\item 'Methylobacterium sp. 17Sr1-1'\n",
       "\\item 'Methylobacterium sp. AMS5'\n",
       "\\item 'Bosea sp. RAC05'\n",
       "\\item 'Phreatobacter cathodiphilus'\n",
       "\\item 'Achromobacter spanius'\n",
       "\\item 'Achromobacter sp. AONIH1'\n",
       "\\item 'Leifsonia shinshuensis'\n",
       "\\item 'Methylobacterium sp. DM1'\n",
       "\\item 'Massilia sp. LPB0304'\n",
       "\\item 'Microbacterium lemovicicum'\n",
       "\\item 'Bradyrhizobium sp.'\n",
       "\\item 'Phreatobacter stygius'\n",
       "\\item 'Microbacterium oleivorans'\n",
       "\\item 'Methylobacterium sp. C1'\n",
       "\\item 'Methylobacterium radiotolerans'\n",
       "\\item 'Phyllobacterium sp. 628'\n",
       "\\item 'Stenotrophomonas maltophilia'\n",
       "\\item 'Methylobacterium sp. XJLW'\n",
       "\\item 'Methylorubrum zatmanii'\n",
       "\\item 'Toxoplasma gondii'\n",
       "\\item 'Prevotella copri'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Sphingobium sp. YG1'\n",
       "2. 'Sphingobium yanoikuyae'\n",
       "3. 'Sphingobium baderi'\n",
       "4. 'Sphingomonas sp. LM7'\n",
       "5. 'Sphingomonas sp. Cra20'\n",
       "6. 'Sphingomonas sp. CL5.1'\n",
       "7. 'Sphingomonas sp. IC081'\n",
       "8. 'Sphingomonas sp. XS-10'\n",
       "9. 'Sphingomonas sp. FARSPH'\n",
       "10. 'Sphingomonas sp. LK11'\n",
       "11. 'Sphingomonas sp. DH-S5'\n",
       "12. 'Sphingomonas sp. NBWT7'\n",
       "13. 'Sphingomonas sp. MM-1'\n",
       "14. 'Sphingomonas sp. AP4-R1'\n",
       "15. 'Sphingomonas sp. AAP5'\n",
       "16. 'Sphingomonas hengshuiensis'\n",
       "17. 'Sphingomonas koreensis'\n",
       "18. 'Sphingomonas sanxanigenens'\n",
       "19. 'Sphingomonas alpina'\n",
       "20. 'Sphingomonas melonis'\n",
       "21. 'Sphingomonas paucimobilis'\n",
       "22. 'Sphingomonas taxi'\n",
       "23. 'Sphingomonas panacisoli'\n",
       "24. 'Sphingomonas panacis'\n",
       "25. 'Novosphingobium sp. ABRDHK2'\n",
       "26. 'Sphingosinithalassobacter sp. zrk23'\n",
       "27. 'Sphingosinithalassobacter sp. CS137'\n",
       "28. 'Bradyrhizobium sp. BTAi1'\n",
       "29. 'Bradyrhizobium sp. PSBB068'\n",
       "30. 'Bradyrhizobium sp. CCBAU 051011'\n",
       "31. 'Bradyrhizobium sp. SK17'\n",
       "32. 'Bradyrhizobium sp. CCBAU 51753'\n",
       "33. 'Bradyrhizobium erythrophlei'\n",
       "34. 'Bradyrhizobium diazoefficiens'\n",
       "35. 'Bradyrhizobium lablabi'\n",
       "36. 'Bradyrhizobium betae'\n",
       "37. 'Bradyrhizobium canariense'\n",
       "38. 'Bradyrhizobium icense'\n",
       "39. 'Rhodopseudomonas palustris'\n",
       "40. 'Tardiphaga robiniae'\n",
       "41. 'Afipia carboxidovorans'\n",
       "42. 'Methylobacterium currus'\n",
       "43. 'Methylobacterium sp. 17Sr1-43'\n",
       "44. 'Methylobacterium aquaticum'\n",
       "45. 'Methylobacterium durans'\n",
       "46. 'Caulobacter flavus'\n",
       "47. 'Caulobacter mirabilis'\n",
       "48. 'Caulobacter sp. K31'\n",
       "49. 'Caulobacter soli'\n",
       "50. 'Caulobacter vibrioides'\n",
       "51. 'Caulobacter rhizosphaerae'\n",
       "52. 'Burkholderia contaminans'\n",
       "53. 'Ralstonia solanacearum'\n",
       "54. 'Ralstonia insidiosa'\n",
       "55. 'Ralstonia pickettii'\n",
       "56. 'Kinneretia sp. DAIF2'\n",
       "57. 'Achromobacter xylosoxidans'\n",
       "58. 'Pseudomonas alcaligenes'\n",
       "59. 'Pseudomonas aeruginosa'\n",
       "60. 'Pseudomonas stutzeri'\n",
       "61. 'Caulobacter segnis'\n",
       "62. 'Caulobacter sp. FWC26'\n",
       "63. 'Caulobacter henricii'\n",
       "64. 'Methylobacterium brachiatum'\n",
       "65. 'Alcaligenes sp. SORT26'\n",
       "66. 'Alcaligenes faecalis'\n",
       "67. 'Paraburkholderia fungorum'\n",
       "68. 'Herbaspirillum huttiense'\n",
       "69. 'Acinetobacter johnsonii'\n",
       "70. 'Pseudomonas mendocina'\n",
       "71. 'Pseudomonas sp. phDV1'\n",
       "72. 'Pseudomonas sp. LPH1'\n",
       "73. 'Pseudomonas toyotomiensis'\n",
       "74. 'Pseudomonas sihuiensis'\n",
       "75. 'Pseudomonas sediminis'\n",
       "76. 'Microbacterium sp. BH-3-3-3'\n",
       "77. 'Microbacterium testaceum'\n",
       "78. 'Microbacterium chocolatum'\n",
       "79. 'Brevibacterium casei'\n",
       "80. 'Rhodococcus sp. P-2'\n",
       "81. 'Ralstonia mannitolilytica'\n",
       "82. 'Pseudomonas sp. B11D7D'\n",
       "83. 'Pseudomonas oleovorans'\n",
       "84. 'Pseudomonas sp. MSPm1'\n",
       "85. 'Pseudomonas alcaliphila'\n",
       "86. 'Bordetella genomosp. 13'\n",
       "87. 'Achromobacter deleyi'\n",
       "88. 'Microcella alkaliphila'\n",
       "89. 'Sphingobium sp. RAC03'\n",
       "90. 'Bradyrhizobium sp. ORS 285'\n",
       "91. 'Bradyrhizobium sp. CCBAU 53421'\n",
       "92. 'Bradyrhizobium sp. 6(2017)'\n",
       "93. 'Bradyrhizobium paxllaeri'\n",
       "94. 'Nitrobacter hamburgensis'\n",
       "95. 'Methylorubrum populi'\n",
       "96. 'Brevundimonas sp. CS1'\n",
       "97. 'Sphingomonas sp. LMO-1'\n",
       "98. 'Sphingomonas sp. NIC1'\n",
       "99. 'Sphingomonas sp. HMP6'\n",
       "100. 'Sphingomonas sp. HMP9'\n",
       "101. 'Sphingomonas wittichii'\n",
       "102. 'Sphingomonas insulae'\n",
       "103. 'Bradyrhizobium sp. ORS 278'\n",
       "104. 'Bradyrhizobium sp. CCGE-LA001'\n",
       "105. 'Bradyrhizobium sp. 1(2017)'\n",
       "106. 'Bradyrhizobium sp. CCBAU 53338'\n",
       "107. 'Bradyrhizobium sp. CCBAU 51765'\n",
       "108. 'Bradyrhizobium sp. CCBAU 53351'\n",
       "109. 'Bradyrhizobium sp. CCBAU 53340'\n",
       "110. 'Bradyrhizobium japonicum'\n",
       "111. 'Bradyrhizobium genosp. L'\n",
       "112. 'Bradyrhizobium guangzhouense'\n",
       "113. 'Bradyrhizobium zhanjiangense'\n",
       "114. 'Bradyrhizobium oligotrophicum'\n",
       "115. 'Bradyrhizobium vignae'\n",
       "116. 'Bradyrhizobium guangdongense'\n",
       "117. 'Bradyrhizobium symbiodeficiens'\n",
       "118. 'Bradyrhizobium genosp. B'\n",
       "119. 'Bradyrhizobium cosmicum'\n",
       "120. 'Afipia sp. GAS231'\n",
       "121. 'Bosea sp. Tri-49'\n",
       "122. 'Nitrobacter winogradskyi'\n",
       "123. 'Methylobacterium sp. WL1'\n",
       "124. 'Methylobacterium terrae'\n",
       "125. 'Methylobacterium indicum'\n",
       "126. 'Methylobacterium mesophilicum'\n",
       "127. 'Methylorubrum extorquens'\n",
       "128. 'Brevundimonas sp. SGAir0440'\n",
       "129. 'Microbacterium hominis'\n",
       "130. 'Brevundimonas sp. GW460-12-10-14-LB2'\n",
       "131. 'Methylobacterium phyllosphaerae'\n",
       "132. 'Methylobacterium sp. 4-46'\n",
       "133. 'Methylobacterium nodulans'\n",
       "134. 'Rhizobium oryzihabitans'\n",
       "135. 'Achromobacter denitrificans'\n",
       "136. 'Delftia acidovorans'\n",
       "137. 'Delftia tsuruhatensis'\n",
       "138. 'Massilia oculi'\n",
       "139. 'Microbacterium sp. 10M-3C3'\n",
       "140. 'Methylobacterium sp. 17Sr1-1'\n",
       "141. 'Methylobacterium sp. AMS5'\n",
       "142. 'Bosea sp. RAC05'\n",
       "143. 'Phreatobacter cathodiphilus'\n",
       "144. 'Achromobacter spanius'\n",
       "145. 'Achromobacter sp. AONIH1'\n",
       "146. 'Leifsonia shinshuensis'\n",
       "147. 'Methylobacterium sp. DM1'\n",
       "148. 'Massilia sp. LPB0304'\n",
       "149. 'Microbacterium lemovicicum'\n",
       "150. 'Bradyrhizobium sp.'\n",
       "151. 'Phreatobacter stygius'\n",
       "152. 'Microbacterium oleivorans'\n",
       "153. 'Methylobacterium sp. C1'\n",
       "154. 'Methylobacterium radiotolerans'\n",
       "155. 'Phyllobacterium sp. 628'\n",
       "156. 'Stenotrophomonas maltophilia'\n",
       "157. 'Methylobacterium sp. XJLW'\n",
       "158. 'Methylorubrum zatmanii'\n",
       "159. 'Toxoplasma gondii'\n",
       "160. 'Prevotella copri'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"Sphingobium sp. YG1\"                 \n",
       "  [2] \"Sphingobium yanoikuyae\"              \n",
       "  [3] \"Sphingobium baderi\"                  \n",
       "  [4] \"Sphingomonas sp. LM7\"                \n",
       "  [5] \"Sphingomonas sp. Cra20\"              \n",
       "  [6] \"Sphingomonas sp. CL5.1\"              \n",
       "  [7] \"Sphingomonas sp. IC081\"              \n",
       "  [8] \"Sphingomonas sp. XS-10\"              \n",
       "  [9] \"Sphingomonas sp. FARSPH\"             \n",
       " [10] \"Sphingomonas sp. LK11\"               \n",
       " [11] \"Sphingomonas sp. DH-S5\"              \n",
       " [12] \"Sphingomonas sp. NBWT7\"              \n",
       " [13] \"Sphingomonas sp. MM-1\"               \n",
       " [14] \"Sphingomonas sp. AP4-R1\"             \n",
       " [15] \"Sphingomonas sp. AAP5\"               \n",
       " [16] \"Sphingomonas hengshuiensis\"          \n",
       " [17] \"Sphingomonas koreensis\"              \n",
       " [18] \"Sphingomonas sanxanigenens\"          \n",
       " [19] \"Sphingomonas alpina\"                 \n",
       " [20] \"Sphingomonas melonis\"                \n",
       " [21] \"Sphingomonas paucimobilis\"           \n",
       " [22] \"Sphingomonas taxi\"                   \n",
       " [23] \"Sphingomonas panacisoli\"             \n",
       " [24] \"Sphingomonas panacis\"                \n",
       " [25] \"Novosphingobium sp. ABRDHK2\"         \n",
       " [26] \"Sphingosinithalassobacter sp. zrk23\" \n",
       " [27] \"Sphingosinithalassobacter sp. CS137\" \n",
       " [28] \"Bradyrhizobium sp. BTAi1\"            \n",
       " [29] \"Bradyrhizobium sp. PSBB068\"          \n",
       " [30] \"Bradyrhizobium sp. CCBAU 051011\"     \n",
       " [31] \"Bradyrhizobium sp. SK17\"             \n",
       " [32] \"Bradyrhizobium sp. CCBAU 51753\"      \n",
       " [33] \"Bradyrhizobium erythrophlei\"         \n",
       " [34] \"Bradyrhizobium diazoefficiens\"       \n",
       " [35] \"Bradyrhizobium lablabi\"              \n",
       " [36] \"Bradyrhizobium betae\"                \n",
       " [37] \"Bradyrhizobium canariense\"           \n",
       " [38] \"Bradyrhizobium icense\"               \n",
       " [39] \"Rhodopseudomonas palustris\"          \n",
       " [40] \"Tardiphaga robiniae\"                 \n",
       " [41] \"Afipia carboxidovorans\"              \n",
       " [42] \"Methylobacterium currus\"             \n",
       " [43] \"Methylobacterium sp. 17Sr1-43\"       \n",
       " [44] \"Methylobacterium aquaticum\"          \n",
       " [45] \"Methylobacterium durans\"             \n",
       " [46] \"Caulobacter flavus\"                  \n",
       " [47] \"Caulobacter mirabilis\"               \n",
       " [48] \"Caulobacter sp. K31\"                 \n",
       " [49] \"Caulobacter soli\"                    \n",
       " [50] \"Caulobacter vibrioides\"              \n",
       " [51] \"Caulobacter rhizosphaerae\"           \n",
       " [52] \"Burkholderia contaminans\"            \n",
       " [53] \"Ralstonia solanacearum\"              \n",
       " [54] \"Ralstonia insidiosa\"                 \n",
       " [55] \"Ralstonia pickettii\"                 \n",
       " [56] \"Kinneretia sp. DAIF2\"                \n",
       " [57] \"Achromobacter xylosoxidans\"          \n",
       " [58] \"Pseudomonas alcaligenes\"             \n",
       " [59] \"Pseudomonas aeruginosa\"              \n",
       " [60] \"Pseudomonas stutzeri\"                \n",
       " [61] \"Caulobacter segnis\"                  \n",
       " [62] \"Caulobacter sp. FWC26\"               \n",
       " [63] \"Caulobacter henricii\"                \n",
       " [64] \"Methylobacterium brachiatum\"         \n",
       " [65] \"Alcaligenes sp. SORT26\"              \n",
       " [66] \"Alcaligenes faecalis\"                \n",
       " [67] \"Paraburkholderia fungorum\"           \n",
       " [68] \"Herbaspirillum huttiense\"            \n",
       " [69] \"Acinetobacter johnsonii\"             \n",
       " [70] \"Pseudomonas mendocina\"               \n",
       " [71] \"Pseudomonas sp. phDV1\"               \n",
       " [72] \"Pseudomonas sp. LPH1\"                \n",
       " [73] \"Pseudomonas toyotomiensis\"           \n",
       " [74] \"Pseudomonas sihuiensis\"              \n",
       " [75] \"Pseudomonas sediminis\"               \n",
       " [76] \"Microbacterium sp. BH-3-3-3\"         \n",
       " [77] \"Microbacterium testaceum\"            \n",
       " [78] \"Microbacterium chocolatum\"           \n",
       " [79] \"Brevibacterium casei\"                \n",
       " [80] \"Rhodococcus sp. P-2\"                 \n",
       " [81] \"Ralstonia mannitolilytica\"           \n",
       " [82] \"Pseudomonas sp. B11D7D\"              \n",
       " [83] \"Pseudomonas oleovorans\"              \n",
       " [84] \"Pseudomonas sp. MSPm1\"               \n",
       " [85] \"Pseudomonas alcaliphila\"             \n",
       " [86] \"Bordetella genomosp. 13\"             \n",
       " [87] \"Achromobacter deleyi\"                \n",
       " [88] \"Microcella alkaliphila\"              \n",
       " [89] \"Sphingobium sp. RAC03\"               \n",
       " [90] \"Bradyrhizobium sp. ORS 285\"          \n",
       " [91] \"Bradyrhizobium sp. CCBAU 53421\"      \n",
       " [92] \"Bradyrhizobium sp. 6(2017)\"          \n",
       " [93] \"Bradyrhizobium paxllaeri\"            \n",
       " [94] \"Nitrobacter hamburgensis\"            \n",
       " [95] \"Methylorubrum populi\"                \n",
       " [96] \"Brevundimonas sp. CS1\"               \n",
       " [97] \"Sphingomonas sp. LMO-1\"              \n",
       " [98] \"Sphingomonas sp. NIC1\"               \n",
       " [99] \"Sphingomonas sp. HMP6\"               \n",
       "[100] \"Sphingomonas sp. HMP9\"               \n",
       "[101] \"Sphingomonas wittichii\"              \n",
       "[102] \"Sphingomonas insulae\"                \n",
       "[103] \"Bradyrhizobium sp. ORS 278\"          \n",
       "[104] \"Bradyrhizobium sp. CCGE-LA001\"       \n",
       "[105] \"Bradyrhizobium sp. 1(2017)\"          \n",
       "[106] \"Bradyrhizobium sp. CCBAU 53338\"      \n",
       "[107] \"Bradyrhizobium sp. CCBAU 51765\"      \n",
       "[108] \"Bradyrhizobium sp. CCBAU 53351\"      \n",
       "[109] \"Bradyrhizobium sp. CCBAU 53340\"      \n",
       "[110] \"Bradyrhizobium japonicum\"            \n",
       "[111] \"Bradyrhizobium genosp. L\"            \n",
       "[112] \"Bradyrhizobium guangzhouense\"        \n",
       "[113] \"Bradyrhizobium zhanjiangense\"        \n",
       "[114] \"Bradyrhizobium oligotrophicum\"       \n",
       "[115] \"Bradyrhizobium vignae\"               \n",
       "[116] \"Bradyrhizobium guangdongense\"        \n",
       "[117] \"Bradyrhizobium symbiodeficiens\"      \n",
       "[118] \"Bradyrhizobium genosp. B\"            \n",
       "[119] \"Bradyrhizobium cosmicum\"             \n",
       "[120] \"Afipia sp. GAS231\"                   \n",
       "[121] \"Bosea sp. Tri-49\"                    \n",
       "[122] \"Nitrobacter winogradskyi\"            \n",
       "[123] \"Methylobacterium sp. WL1\"            \n",
       "[124] \"Methylobacterium terrae\"             \n",
       "[125] \"Methylobacterium indicum\"            \n",
       "[126] \"Methylobacterium mesophilicum\"       \n",
       "[127] \"Methylorubrum extorquens\"            \n",
       "[128] \"Brevundimonas sp. SGAir0440\"         \n",
       "[129] \"Microbacterium hominis\"              \n",
       "[130] \"Brevundimonas sp. GW460-12-10-14-LB2\"\n",
       "[131] \"Methylobacterium phyllosphaerae\"     \n",
       "[132] \"Methylobacterium sp. 4-46\"           \n",
       "[133] \"Methylobacterium nodulans\"           \n",
       "[134] \"Rhizobium oryzihabitans\"             \n",
       "[135] \"Achromobacter denitrificans\"         \n",
       "[136] \"Delftia acidovorans\"                 \n",
       "[137] \"Delftia tsuruhatensis\"               \n",
       "[138] \"Massilia oculi\"                      \n",
       "[139] \"Microbacterium sp. 10M-3C3\"          \n",
       "[140] \"Methylobacterium sp. 17Sr1-1\"        \n",
       "[141] \"Methylobacterium sp. AMS5\"           \n",
       "[142] \"Bosea sp. RAC05\"                     \n",
       "[143] \"Phreatobacter cathodiphilus\"         \n",
       "[144] \"Achromobacter spanius\"               \n",
       "[145] \"Achromobacter sp. AONIH1\"            \n",
       "[146] \"Leifsonia shinshuensis\"              \n",
       "[147] \"Methylobacterium sp. DM1\"            \n",
       "[148] \"Massilia sp. LPB0304\"                \n",
       "[149] \"Microbacterium lemovicicum\"          \n",
       "[150] \"Bradyrhizobium sp.\"                  \n",
       "[151] \"Phreatobacter stygius\"               \n",
       "[152] \"Microbacterium oleivorans\"           \n",
       "[153] \"Methylobacterium sp. C1\"             \n",
       "[154] \"Methylobacterium radiotolerans\"      \n",
       "[155] \"Phyllobacterium sp. 628\"             \n",
       "[156] \"Stenotrophomonas maltophilia\"        \n",
       "[157] \"Methylobacterium sp. XJLW\"           \n",
       "[158] \"Methylorubrum zatmanii\"              \n",
       "[159] \"Toxoplasma gondii\"                   \n",
       "[160] \"Prevotella copri\"                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "corr_c <- corr_res %>% \n",
    "    filter(rho > 0.7) %>%\n",
    "    distinct(non_contaminant_taxon) %>%\n",
    "    rename(contaminants = non_contaminant_taxon)\n",
    "\n",
    "corr_nc <- tibble(non_contaminant_taxon = colnames(df_filt)[!(colnames(df_filt) %in% c(\"npm_research_id\", \n",
    "                                                                                       corr_c$contaminants, \n",
    "                                                                                       diff_prev_contams))])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b34eb0e5",
   "metadata": {},
   "source": [
    "### Parse and save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c5ee1861",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 330\n",
      "[1] 535\n"
     ]
    }
   ],
   "source": [
    "corr_t <- 0.7\n",
    "\n",
    "fwrite(corr_res,  \n",
    "       str_glue(\"results/decontamination/correlation_decontamination/corr_raw.diff_prev_V3.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.corr_t{corr_t}.within_batch.S.n9999.txt\"))\n",
    "fwrite(corr_c, \n",
    "       str_glue(\"results/decontamination/correlation_decontamination/contam.diff_prev_V3.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.corr_t{corr_t}.within_batch.S.n9999.txt\"))\n",
    "fwrite(corr_nc, \n",
    "       str_glue(\"results/decontamination/correlation_decontamination/nc.diff_prev_V3.RA0.005.read_threshold10.max_prev{max_prev_t}.fold_diff{fold_diff_t}.corr_t{corr_t}.within_batch.S.n9999.txt\"))\n",
    "\n",
    "# print(length(corr_nc$non_contaminant_taxon))\n",
    "print(length(corr_c$contaminants))\n",
    "print(length(corr_nc$non_contaminant_taxon))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "763028c0",
   "metadata": {},
   "source": [
    "## Plot correlation heatmaps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "93bc8abf",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_heatmap <- function(dat) {\n",
    "    as.data.frame(cor(dat)) %>%\n",
    "        rownames_to_column(\"taxa1\") %>%\n",
    "        pivot_longer(!taxa1, names_to = \"taxa2\", values_to = \"rho\") %>%\n",
    "                mutate(fill_color = case_when(rho <= 0 ~ \"<0\",\n",
    "                         rho > 0 & rho <= 0.1 ~ \"0-0.1\",\n",
    "                         rho > 0.1 & rho <= 0.2 ~ \"0.1-0.2\",\n",
    "                         rho > 0.2 & rho <= 0.3 ~ \"0.2-0.3\",\n",
    "                         rho > 0.3 & rho <= 0.5 ~  \"0.3-0.5\",\n",
    "                         rho > 0.5 & rho <= 0.7 ~ \"0.5-0.7\",\n",
    "                         rho > 0.7 ~ \">0.7\")) %>%\n",
    "        mutate(fill_color = factor(fill_color, levels = c(\"<0\", \"0-0.1\", \"0.1-0.2\", \"0.2-0.3\", \"0.3-0.5\", \"0.5-0.7\", \">0.7\"))) %>%\n",
    "        ggplot(aes(x = taxa1, y = taxa2, fill = fill_color)) + \n",
    "        geom_tile() +\n",
    "        scale_fill_manual(values = c(\"#6699cc\", \"#ffffcc\", \"#f39b7fb2\", \"#e64b35b2\", \"#d11141\", \"darkred\", \"black\"), drop = F) +\n",
    "        theme(axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "              axis.title = element_blank()) +\n",
    "    labs(fill = \"Rho\", title = str_glue(\"n = {nrow(dat)}\"))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "d20f2e0b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = \"npm_research_id\"\n"
     ]
    }
   ],
   "source": [
    "plot_iter <- list(c(\"site_supplying_sample\", \"MEC\"), \n",
    "                  c(\"extraction_kit\", \"Chemagic DNA Blood Kit (Perkin Elmer, MA)\"),\n",
    "                  c(\"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\", \"20392972\"),\n",
    "                  c(\"hiseq_xtm_flow_cell_v2_5_lot\", \"20367079\"))\n",
    "\n",
    "for (item in plot_iter) {\n",
    "    var_name <- item[1]\n",
    "    level_name <- item[2]\n",
    "    mec_contam <- corr_res %>% \n",
    "        filter(level == level_name) %>%\n",
    "        filter(rho > 0.7)\n",
    "    mec_contam <- c(mec_contam$contaminant_taxon, mec_contam$non_contaminant_taxon)\n",
    "\n",
    "    mec_nc <- corr_res %>% \n",
    "        filter(level == level_name) %>%\n",
    "        filter(rho <= 0.7)\n",
    "    mec_nc <- mec_nc$non_contaminant_taxon\n",
    "\n",
    "    contam_df <- df_filt %>% \n",
    "        left_join(meta_filt) %>%\n",
    "        filter(get(var_name) == level_name) %>%\n",
    "        select(all_of(mec_contam))\n",
    "    nc_df <- df_filt %>% \n",
    "        left_join(meta_filt) %>%\n",
    "        filter(get(var_name) == level_name) %>%\n",
    "        select(all_of(mec_nc))\n",
    "\n",
    "    plot_heatmap(contam_df)\n",
    "    ggsave(str_glue(\"results/decontamination/correlation_decontamination/correlation_contaminants.{var_name}.{level_name}.png\"), \n",
    "           dpi = 300, \n",
    "           height = 6.67, width = 7.5)\n",
    "    plot_heatmap(nc_df)\n",
    "        ggsave(str_glue(\"results/decontamination/correlation_decontamination/correlation_non_contaminants.{var_name}.{level_name}.png\"), \n",
    "           dpi = 300, \n",
    "           height = 6.67, width = 7.5)\n",
    "}\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
