{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe3f3f5c",
   "metadata": {},
   "source": [
    "# Identication of putative contaminant taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "440f3650",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.3     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.0.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32maccumulate()\u001b[39m masks \u001b[34mforeach\u001b[39m::accumulate()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mwhen()\u001b[39m       masks \u001b[34mforeach\u001b[39m::when()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Loading required package: compositions\n",
      "\n",
      "Welcome to compositions, a package for compositional data analysis.\n",
      "Find an intro with \"? compositions\"\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘compositions’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    anova, cor, cov, dist, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    %*%, norm, scale, scale.default\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(foreach)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(compositions)\n",
    "require(doParallel)\n",
    "registerDoParallel(cores=8)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "215972d1",
   "metadata": {},
   "source": [
    "### Data pre-processing functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b5109a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_data <- function(file_path) {\n",
    "    df <- as.data.frame(fread(file_path)) %>%\n",
    "        separate(sample, into = c(NA, \"npm_research_id\"), sep = \"\\\\.\")\n",
    "    return(df)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, df) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% df$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "\n",
    "subset_metadata <- function(meta, n_subset) {\n",
    "    meta <- as.data.frame(meta)\n",
    "    cohorts <- unique(meta$site_supplying_sample)\n",
    "    subset_vec <- c()\n",
    "\n",
    "    for (i in cohorts) {\n",
    "        ids <- meta$npm_research_id[meta$site_supplying_sample == i]\n",
    "        \n",
    "        if (length(ids) > n_subset) {\n",
    "            subset_ids <- sample(ids, n_subset)\n",
    "            subset_vec <- c(subset_vec, subset_ids)\n",
    "        } else {\n",
    "            subset_vec <- c(subset_vec, ids)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    meta_sub <- meta %>%\n",
    "      filter(npm_research_id %in% subset_vec) \n",
    "    return(meta_sub)\n",
    "}\n",
    "\n",
    "\n",
    "retrieve_rows_from_meta <- function(df, meta) {\n",
    "    return(df %>% filter(npm_research_id %in% meta$npm_research_id)) \n",
    "}\n",
    "\n",
    "\n",
    "remove_cols <- function(df, col_to_exclude) {\n",
    "    return(df %>% select(-all_of(col_to_exclude)))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_prev_taxa <- function(df, frac_presence) {\n",
    "    n_original <- ncol(df[, colnames(df) != \"npm_research_id\"])\n",
    "    PA_df <- apply(df[, 2:ncol(df)], 2, function(x) {ifelse(x > 0, T, F)})\n",
    "    frac_df <- apply(PA_df, 2, function(x) {sum(x) / nrow(PA_df)})\n",
    "    to_keep <- names(frac_df[frac_df > frac_presence])\n",
    "    to_keep <- c(\"npm_research_id\", to_keep)\n",
    "    n_new <- length(to_keep) - 1\n",
    "    print(str_glue(\"{n_new} / {n_original} taxa are present in {frac_presence} of samples\"))\n",
    "    return(df %>% select(all_of(to_keep)))\n",
    "}\n",
    "\n",
    "\n",
    "otu_to_RA <- function(df) {\n",
    "    mat <- as.matrix(df[, colnames(df) != \"npm_research_id\"])\n",
    "    RA_df <- as.data.frame(mat / rowSums(mat))\n",
    "    RA_df <- add_column(RA_df, df$npm_research_id, .before = 1)\n",
    "    colnames(RA_df)[1] <- \"npm_research_id\"\n",
    "    \n",
    "    return(RA_df)\n",
    "}\n",
    "\n",
    "\n",
    "RA_to_clr <- function(df) {\n",
    "    mat <- df_filt[, colnames(df_filt) != \"npm_research_id\"]\n",
    "    clr_df <- clr(mat)\n",
    "    return(cbind(data.frame(npm_research_id = df$npm_research_id), as.data.frame(clr_df, check.names = F)))\n",
    "}\n",
    "\n",
    "\n",
    "get_metadata_plots <- function(meta, meta_cols) {\n",
    "    meta <- as.matrix(meta)\n",
    "    meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "    plots <- list()\n",
    "    \n",
    "    for (column in meta_cols) {\n",
    "        plt <- meta %>%\n",
    "            mutate(across(everything(), as.character)) %>%\n",
    "            select(all_of(column)) %>%\n",
    "            group_by_at(column) %>%\n",
    "            summarise(n = n()) %>%\n",
    "            ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "                geom_bar(stat = \"identity\") +\n",
    "                theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "                geom_text(aes_string(label = \"n\"))\n",
    "        plots[[column]] <- plt\n",
    "    }\n",
    "    \n",
    "    return(plots)\n",
    "}\n",
    "\n",
    "\n",
    "get_meta_cols <- function(meta) {\n",
    "    meta_cols <- colnames(meta)[grep(\"kit|flow_cell|instrument_id|site_supplying\", colnames(meta))]\n",
    "    to_exclude <- c(\"library_prep_kit\", \"hiseq_xtm_flow_cell_v2_5_id\")\n",
    "    meta_cols <- meta_cols[!(meta_cols %in% to_exclude)]\n",
    "    return(meta_cols)\n",
    "}\n",
    "\n",
    "\n",
    "print_group_freqs <- function(meta, meta_cols) {\n",
    "    for (col in meta_cols) {\n",
    "        tmp <- tibble(data.frame(meta)) %>%\n",
    "            group_by(get(col)) %>%\n",
    "            summarise(n = n())\n",
    "        print(col)\n",
    "        print(tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "get_batch_prevalence <- function(x) {sum(x) / length(x)}\n",
    "\n",
    "\n",
    "remove_low_read_samples <- function(df) {\n",
    "    to_retain <- fread(\"data/samples_above_10_reads.txt\")$npm_research_id\n",
    "    return(df %>% filter(npm_research_id %in% to_retain))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_sample_levels <- function(dat, metadat, column, min_samples) {\n",
    "    # Remove levels with < x samples\n",
    "    tmp <- tibble(data.frame(metadat)) %>%\n",
    "            group_by(get(column)) %>%\n",
    "            summarise(n = n())\n",
    "\n",
    "    # Vector of group levels to keep\n",
    "    to_keep <- tmp[tmp$n >= min_samples, \"get(column)\"]$`get(column)`\n",
    "    to_keep <- to_keep[to_keep != \"Unknown\"]\n",
    "    n_levels <- length(to_keep)\n",
    "\n",
    "    if (n_levels < 2) {\n",
    "        print(str_glue(\"After pruning, {column} has < 2 levels\"))\n",
    "    } else {\n",
    "        print(str_glue(\"After pruning, {column} has {n_levels} levels\"))\n",
    "    }\n",
    "\n",
    "    # Remove rows in metadata\n",
    "    metadat_filt <- metadat %>% filter(get(column) %in% to_keep)\n",
    "\n",
    "    # Retrieve rows\n",
    "    dat_filt <- retrieve_rows_from_meta(dat, metadat_filt)\n",
    "    metadat_filt <- metadat_filt %>% filter(npm_research_id %in% dat_filt$npm_research_id)\n",
    "\n",
    "    return(list(dat_filt = dat_filt, metadat_filt = metadat_filt))\n",
    "}\n",
    "\n",
    "\n",
    "remove_empty_rows <- function(df) {\n",
    "    mat <- as.matrix(df[, colnames(df) != \"npm_research_id\"])\n",
    "    row_sums <- rowSums(mat)\n",
    "    df_filt <- df[row_sums != 0, ]\n",
    "    nrow(df_filt)\n",
    "\n",
    "    n_original <- nrow(df)\n",
    "    n_removed <- n_original - nrow(df_filt)\n",
    "    \n",
    "    print(str_glue(\"{n_removed}/{n_original} samples removed due to having no reads of interest\"))\n",
    "    \n",
    "    return(df_filt)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "363b9d21",
   "metadata": {},
   "source": [
    "#### Functions for testing differential abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c0e346fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_diff_abn <- function(dat_filt, metadat_filt, taxon, column, lq_p = 0.25, hq_p = 0.75, fold_threshold = 0) {\n",
    "    # Remove zeros in CLR values\n",
    "    q_df <- dat_filt %>%\n",
    "        left_join(metadat_filt, by = \"npm_research_id\") %>%\n",
    "        select(all_of(c(taxon, column))) %>%\n",
    "        filter(get(taxon) != 0) %>%\n",
    "        group_by(get(column)) %>%\n",
    "        summarise(lq = quantile(get(taxon), probs = lq_p),\n",
    "                  hq = quantile(get(taxon), probs = hq_p))\n",
    "\n",
    "    # Do pairwise comparisons between levels\n",
    "    pairwise_results <- foreach(idx = seq(nrow(q_df))) %do% {\n",
    "        level1 <- q_df[idx, ]$`get(column)`\n",
    "        lowest_lq <- q_df[idx, ]$lq\n",
    "        lowest_lq\n",
    "        diff_df <- q_df %>% \n",
    "            mutate(diff_abn = hq + fold_threshold < lowest_lq) %>%\n",
    "            rename(\"level2\" = \"get(column)\")\n",
    "\n",
    "        if(any(diff_df$diff_abn)) {\n",
    "            res <- diff_df %>%\n",
    "                filter(diff_abn == T) %>%\n",
    "                mutate(taxa = taxon, meta_col = column, level1 = level1, .before = 1) %>%\n",
    "                select(-diff_abn) %>% \n",
    "                rename(level1_lq = lq, level2_hq = hq)\n",
    "\n",
    "            return(res)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    return(bind_rows(pairwise_results))\n",
    "}\n",
    "\n",
    "\n",
    "decontaminate_this_col <- function(dat, metadat, column, min_samples, lq_p, hq_p, fold_threshold) {\n",
    "    data_list <- remove_low_sample_levels(dat, metadat, column, min_samples)\n",
    "    metadat_filt <- data_list[[\"metadat_filt\"]]\n",
    "    dat_filt <- data_list[[\"dat_filt\"]]\n",
    "    taxa <- colnames(dat_filt %>% select(-npm_research_id))\n",
    "    \n",
    "    # Get differential abundance results for all taxa\n",
    "    col_results <- foreach(taxon = taxa) %dopar% {\n",
    "        taxon_results <- get_diff_abn(dat_filt, metadat_filt, taxon, column, lq_p = lq_p, hq_p = hq_p, fold_threshold = fold_threshold)\n",
    "    }\n",
    "\n",
    "    return(bind_rows(col_results))\n",
    "}\n",
    "\n",
    "decontaminate_all_cols <- function(df_filt, meta_filt, meta_cols, min_samples, lq_p, hq_p, fold_threshold) {\n",
    "    results <- foreach(column = meta_cols) %do% {\n",
    "        col_res <- decontaminate_this_col(df_filt, meta_filt, column, \n",
    "                                          min_samples = min_samples, \n",
    "                                          lq_p = lq_p, \n",
    "                                          hq_p = hq_p, \n",
    "                                          fold_threshold = fold_threshold)\n",
    "        return(col_res)\n",
    "    }\n",
    "    \n",
    "    return(bind_rows(results))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bc78537",
   "metadata": {},
   "source": [
    "## Main"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3abe6ac6",
   "metadata": {},
   "source": [
    "#### Params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3c25c40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "min_samples <- 100\n",
    "n_subset <- 9999\n",
    "human <- \"Homo sapiens\"\n",
    "rank <- \"S\"\n",
    "n <- 9999\n",
    "frac_presence <- 0.05\n",
    "lq_p <- 0.10\n",
    "hq_p <- 0.90\n",
    "fold_threshold <- 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06652f5d",
   "metadata": {},
   "source": [
    "#### Load and parse data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6fd9cdfa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1792 / 5199 taxa are present in 0.05 of samples\n"
     ]
    }
   ],
   "source": [
    "to_retain <- fread(\"data/samples_above_95_reads.txt\")$npm_research_id\n",
    "df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.{rank}.tsv\")) %>% \n",
    "    filter(npm_research_id %in% to_retain)\n",
    "meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "\n",
    "# Get metadata subset\n",
    "meta_filt <- subset_metadata(meta, n)\n",
    "# Filter data\n",
    "df_filt <- retrieve_rows_from_meta(df, meta_filt)\n",
    "df_filt <- remove_cols(df_filt, c(human, \"unclassified\"))\n",
    "df_filt <- remove_low_prev_taxa(df_filt, frac_presence = frac_presence)\n",
    "df_filt <- otu_to_RA(df_filt)\n",
    "df_filt <- RA_to_clr(df_filt)\n",
    "\n",
    "# Get metadata columns of interest\n",
    "meta_cols <- get_meta_cols(meta_filt)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bc45bfe",
   "metadata": {},
   "source": [
    "#### Run decontamination"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "60df7a12",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Run decontamination\n",
    "# result_df <- system.time(decontaminate_all_cols(df_filt, meta_filt, meta_cols, min_samples, lq_p, hq_p, fold_threshold))\n",
    "# head(result_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76601803",
   "metadata": {},
   "source": [
    "#### Parse results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3f1d7d0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# n_contam <- length(unique(result_df$taxa))\n",
    "\n",
    "# all_species <- colnames(df_filt)[colnames(df_filt) != \"npm_research_id\"]\n",
    "# non_contaminants <- all_species[!(all_species %in% unique(result_df$taxa))]\n",
    "\n",
    "# nc_df <- tibble(non_contaminants = non_contaminants)\n",
    "# fwrite(nc_df, str_glue(\"results/decontamination/clr_decontamination/NC.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"))\n",
    "# fwrite(result_df, str_glue(\"results/decontamination/clr_decontamination/results.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.csv\"))\n",
    "# length(non_contaminants)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70042918",
   "metadata": {},
   "source": [
    "### Run decontamination for multiple thresholds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "84d0a6e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "params <- foreach (m = c(50, 100), .combine = 'c') %do% {\n",
    "    foreach (hq = seq(0.70, 1, 0.01)) %do% {\n",
    "        lq <- 1 - hq\n",
    "        return(c(m, lq, hq))\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c3221148",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants\n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m           \n",
      "\u001b[90m1\u001b[39m          50   0.3   0.7           0 \u001b[90m\"\u001b[39m\u001b[90m\"\u001b[39m              \n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants\n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m           \n",
      "\u001b[90m1\u001b[39m          50  0.29  0.71           0 \u001b[90m\"\u001b[39m\u001b[90m\"\u001b[39m              \n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants\n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m           \n",
      "\u001b[90m1\u001b[39m          50  0.28  0.72           0 \u001b[90m\"\u001b[39m\u001b[90m\"\u001b[39m              \n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants     \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                \n",
      "\u001b[90m1\u001b[39m          50  0.27  0.73           1 Rhizobium sp. 11515TR\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                        \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                   \n",
      "\u001b[90m1\u001b[39m          50  0.26  0.74           2 Rhizobium sp. 11515TR;Rhizobium phaseoli\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                        \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                   \n",
      "\u001b[90m1\u001b[39m          50  0.25  0.75           2 Rhizobium sp. 11515TR;Rhizobium phaseoli\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.24  0.76           3 Rhizobium sp. 11515TR;Rhizobium phaseoli;…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.23  0.77           3 Rhizobium sp. 11515TR;Rhizobium phaseoli;…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.22  0.78           3 Rhizobium sp. 11515TR;Rhizobium phaseoli;…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.21  0.79           3 Rhizobium sp. 11515TR;Rhizobium phaseoli;…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50   0.2   0.8           5 Rhizobium sp. 11515TR;Methylocella silves…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.19  0.81           8 Rhizobium sp. 11515TR;Methylocella silves…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.18  0.82          12 Rhizobium sp. 11515TR;Agrobacterium vitis…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.17  0.83          18 Ralstonia solanacearum;Rhizobium sp. 1151…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.16  0.84          24 Ralstonia solanacearum;Rhizobium sp. 1151…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.15  0.85          33 Ralstonia solanacearum;Bordetella genomos…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.14  0.86          43 Ralstonia solanacearum;Bordetella genomos…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50  0.13  0.87          55 Ralstonia solanacearum;Bordetella genomos…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50 0.120  0.88          64 Paraburkholderia fungorum;Ralstonia solan…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50 0.110  0.89          84 Paraburkholderia fungorum;Ralstonia solan…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50 0.100   0.9         103 Paraburkholderia fungorum;Ralstonia solan…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.090\u001b[4m0\u001b[24m  0.91         137 Paraburkholderia fungorum;Ralstonia sola…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.080\u001b[4m0\u001b[24m  0.92         161 Paraburkholderia fungorum;Ralstonia sola…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.070\u001b[4m0\u001b[24m  0.93         180 Paraburkholderia fungorum;Ralstonia sola…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.060\u001b[4m0\u001b[24m  0.94         209 Paraburkholderia fungorum;Ralstonia sola…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.050\u001b[4m0\u001b[24m  0.95         254 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.040\u001b[4m0\u001b[24m  0.96         287 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.030\u001b[4m0\u001b[24m  0.97         332 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.020\u001b[4m0\u001b[24m  0.98         368 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m          50 0.010\u001b[4m0\u001b[24m  0.99         401 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 22 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 25 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 26 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m          50     0     1         449 Paraburkholderia fungorum;Paraburkholderi…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants\n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m           \n",
      "\u001b[90m1\u001b[39m         100   0.3   0.7           0 \u001b[90m\"\u001b[39m\u001b[90m\"\u001b[39m              \n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants\n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m           \n",
      "\u001b[90m1\u001b[39m         100  0.29  0.71           0 \u001b[90m\"\u001b[39m\u001b[90m\"\u001b[39m              \n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants     \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                \n",
      "\u001b[90m1\u001b[39m         100  0.28  0.72           1 Rhizobium sp. 11515TR\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants     \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                \n",
      "\u001b[90m1\u001b[39m         100  0.27  0.73           1 Rhizobium sp. 11515TR\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                        \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                   \n",
      "\u001b[90m1\u001b[39m         100  0.26  0.74           2 Rhizobium sp. 11515TR;Rhizobium phaseoli\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.25  0.75           4 Rhizobium sp. 11515TR;Dokdonella koreensi…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.24  0.76           5 Rhizobium sp. 11515TR;Dokdonella koreensi…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.23  0.77           6 Rhizobium sp. 11515TR;Dokdonella koreensi…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.22  0.78          11 Rhizobium sp. 11515TR;Mesorhizobium sp. M…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.21  0.79          18 Rhizobium sp. 11515TR;Sinorhizobium ameri…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100   0.2   0.8          22 Rhizobium sp. 11515TR;Sinorhizobium ameri…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.19  0.81          32 Rhizobium tropici;Rhizobium sp. 11515TR;S…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.18  0.82          44 Rhizobium tropici;Rhizobium sp. 11515TR;R…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.17  0.83          58 Ralstonia solanacearum;Rhizobium tropici;…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.16  0.84          66 Ralstonia solanacearum;Rhizobium tropici;…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.15  0.85          83 Ralstonia solanacearum;Burkholderia cepac…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.14  0.86         107 Ralstonia solanacearum;Burkholderia cepac…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100  0.13  0.87         126 Ralstonia solanacearum;Burkholderia cepac…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100 0.120  0.88         154 Paraburkholderia fungorum;Ralstonia solan…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100 0.110  0.89         189 Paraburkholderia fungorum;Ralstonia solan…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100 0.100   0.9         216 Paraburkholderia fungorum;Ralstonia insid…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.090\u001b[4m0\u001b[24m  0.91         246 Paraburkholderia fungorum;Ralstonia insi…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.080\u001b[4m0\u001b[24m  0.92         283 Paraburkholderia fungorum;Ralstonia insi…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.070\u001b[4m0\u001b[24m  0.93         314 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.060\u001b[4m0\u001b[24m  0.94         362 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.050\u001b[4m0\u001b[24m  0.95         401 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.040\u001b[4m0\u001b[24m  0.96         440 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.030\u001b[4m0\u001b[24m  0.97         491 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.020\u001b[4m0\u001b[24m  0.98         545 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples   lq_p  hq_p n_noncontam non_contaminants                         \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m  \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                    \n",
      "\u001b[90m1\u001b[39m         100 0.010\u001b[4m0\u001b[24m  0.99         601 Paraburkholderia fungorum;Paraburkholder…\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "\u001b[90m# A tibble: 1 × 5\u001b[39m\n",
      "  min_samples  lq_p  hq_p n_noncontam non_contaminants                          \n",
      "        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \n",
      "\u001b[90m1\u001b[39m         100     0     1         653 Paraburkholderia fungorum;Paraburkholderi…\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in fwrite(\"results/decontamination/clr_decontamination/multiple_threshold_results.csv\", : is.list(x) is not TRUE\n",
     "output_type": "error",
     "traceback": [
      "Error in fwrite(\"results/decontamination/clr_decontamination/multiple_threshold_results.csv\", : is.list(x) is not TRUE\nTraceback:\n",
      "1. fwrite(\"results/decontamination/clr_decontamination/multiple_threshold_results.csv\", \n .     threshold_df)",
      "2. stopifnot(is.list(x), identical(quote, \"auto\") || isTRUEorFALSE(quote), \n .     is.character(sep) && length(sep) == 1L && nchar(sep) == 1L, \n .     is.character(sep2) && length(sep2) == 3L && nchar(sep2[2L]) == \n .         1L, is.character(dec) && length(dec) == 1L && nchar(dec) == \n .         1L, dec != sep, is.character(eol) && length(eol) == 1L, \n .     length(qmethod) == 1L && qmethod %chin% c(\"double\", \"escape\"), \n .     length(compress) == 1L && compress %chin% c(\"auto\", \"none\", \n .         \"gzip\"), isTRUEorFALSE(col.names), isTRUEorFALSE(append), \n .     isTRUEorFALSE(row.names), isTRUEorFALSE(verbose), isTRUEorFALSE(showProgress), \n .     isTRUEorFALSE(logical01), isTRUEorFALSE(bom), length(na) == \n .         1L, is.character(file) && length(file) == 1L && !is.na(file), \n .     length(buffMB) == 1L && !is.na(buffMB) && 1L <= buffMB && \n .         buffMB <= 1024L, length(nThread) == 1L && !is.na(nThread) && \n .         nThread >= 1L)"
     ]
    }
   ],
   "source": [
    "all_species <- colnames(df_filt)[colnames(df_filt) != \"npm_research_id\"]\n",
    "\n",
    "threshold_results <- foreach (param = params) %do% {\n",
    "    min_samples <- param[1]\n",
    "    lq_p <- param[2]\n",
    "    hq_p <- param[3]\n",
    "    \n",
    "    # Run decontamination\n",
    "    result_df <- decontaminate_all_cols(df_filt, meta_filt, meta_cols, min_samples, lq_p, hq_p, fold_threshold)\n",
    "    \n",
    "    # Parse results\n",
    "    non_contaminants <- all_species[!(all_species %in% unique(result_df$taxa))]\n",
    "    n_noncontam <- length(non_contaminants)\n",
    "    nc_text <- paste0(non_contaminants, collapse = \";\")\n",
    "    nc_df <- tibble(non_contaminants = non_contaminants)\n",
    "    \n",
    "    fwrite(nc_df, str_glue(\"results/decontamination/clr_decontamination/NC.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"), quote = F)\n",
    "    fwrite(result_df, str_glue(\"results/decontamination/clr_decontamination/results.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.csv\"))\n",
    "\n",
    "    # Save summary\n",
    "    morsel <- tibble(min_samples = min_samples, \n",
    "                  lq_p = lq_p, \n",
    "                  hq_p = hq_p, \n",
    "                  n_noncontam = n_noncontam, \n",
    "                  non_contaminants = nc_text)\n",
    "    print(morsel)\n",
    "    \n",
    "    return(morsel)\n",
    "}\n",
    "\n",
    "threshold_df <- bind_rows(threshold_results)\n",
    "fwrite(\"results/decontamination/clr_decontamination/multiple_threshold_results.csv\", threshold_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b374698d",
   "metadata": {},
   "outputs": [],
   "source": [
    "dawdawdawdawdd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82b61826",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat_filt %>% \n",
    "    left_join(metadat_filt) %>%\n",
    "    filter(get(tax) != 0) %>%\n",
    "    group_by(site_supplying_sample) %>%\n",
    "    summarise(lq = quantile(get(tax), 0.25), hq = quantile(get(tax), 0.75))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06040606",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat_filt %>% \n",
    "    pivot_longer(!npm_research_id, names_to = \"taxa\", values_to = \"clr\") %>%\n",
    "    filter(clr != 0) %>%\n",
    "    ggplot(aes(x = clr)) +\n",
    "        geom_density()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e084553",
   "metadata": {},
   "outputs": [],
   "source": [
    "tax <- \"Burkholderia stabilis\"\n",
    "df_filt %>% \n",
    "    left_join(meta_filt) %>%\n",
    "    filter(get(tax) != 0) %>%\n",
    "    ggplot(aes(y = get(tax), x = site_supplying_sample)) +\n",
    "        geom_boxplot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db45209a",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df <- bind_rows(results)\n",
    "result_df %>% distinct(taxa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58e3b427",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadat <- meta_filt\n",
    "dat <- df_filt\n",
    "taxon <- \"Streptococcus oralis\"\n",
    "lq_p <- 0.25\n",
    "hq_p <- 0.75\n",
    "fold_threshold <- 0"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
