{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe3f3f5c",
   "metadata": {},
   "source": [
    "# Identication of putative contaminant taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "440f3650",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(foreach)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(doParallel)\n",
    "registerDoParallel(cores=6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "4b5109a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_data <- function(file_path) {\n",
    "    df <- as.data.frame(fread(file_path)) %>%\n",
    "        separate(sample, into = c(NA, \"npm_research_id\"), sep = \"\\\\.\")\n",
    "    return(df)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, df) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% df$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "\n",
    "subset_metadata <- function(meta, n_subset) {\n",
    "    meta <- as.data.frame(meta)\n",
    "    cohorts <- unique(meta$site_supplying_sample)\n",
    "    subset_vec <- c()\n",
    "\n",
    "    for (i in cohorts) {\n",
    "        ids <- meta$npm_research_id[meta$site_supplying_sample == i]\n",
    "        \n",
    "        if (length(ids) > n_subset) {\n",
    "            subset_ids <- sample(ids, n_subset)\n",
    "            subset_vec <- c(subset_vec, subset_ids)\n",
    "        } else {\n",
    "            subset_vec <- c(subset_vec, ids)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    meta_sub <- meta %>%\n",
    "      filter(npm_research_id %in% subset_vec) \n",
    "    return(meta_sub)\n",
    "}\n",
    "\n",
    "\n",
    "retrieve_rows_from_meta <- function(df, meta) {\n",
    "    return(df %>% filter(npm_research_id %in% meta$npm_research_id)) \n",
    "}\n",
    "\n",
    "\n",
    "remove_cols <- function(df, col_to_exclude) {\n",
    "    return(df %>% select(-all_of(col_to_exclude)))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_freq_taxa <- function(df, frac_presence) {\n",
    "    n_original <- ncol(df[, colnames(df) != \"npm_research_id\"])\n",
    "    PA_df <- apply(df[, 2:ncol(df)], 2, function(x) {ifelse(x > 0, T, F)})\n",
    "    frac_df <- apply(PA_df, 2, function(x) {sum(x) / nrow(PA_df)})\n",
    "    to_keep <- names(frac_df[frac_df > frac_presence])\n",
    "    to_keep <- c(\"npm_research_id\", to_keep)\n",
    "    n_new <- length(to_keep) - 1\n",
    "    print(str_glue(\"{n_new} / {n_original} taxa are present in {frac_presence} of samples\"))\n",
    "    return(df %>% select(all_of(to_keep)))\n",
    "}\n",
    "\n",
    "\n",
    "otu_to_RA <- function(df) {\n",
    "    mat <- as.matrix(df[, colnames(df) != \"npm_research_id\"])\n",
    "    RA_df <- as.data.frame(mat / rowSums(mat))\n",
    "    RA_df <- add_column(RA_df, df$npm_research_id, .before = 1)\n",
    "    colnames(RA_df)[1] <- \"npm_research_id\"\n",
    "    \n",
    "    return(RA_df)\n",
    "}\n",
    "\n",
    "\n",
    "get_metadata_plots <- function(meta, meta_cols) {\n",
    "    meta <- as.matrix(meta)\n",
    "    meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "    plots <- list()\n",
    "    \n",
    "    for (column in meta_cols) {\n",
    "        plt <- meta %>%\n",
    "            mutate(across(everything(), as.character)) %>%\n",
    "            select(all_of(column)) %>%\n",
    "            group_by_at(column) %>%\n",
    "            summarise(n = n()) %>%\n",
    "            ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "                geom_bar(stat = \"identity\") +\n",
    "                theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "                geom_text(aes_string(label = \"n\"))\n",
    "        plots[[column]] <- plt\n",
    "    }\n",
    "    \n",
    "    return(plots)\n",
    "}\n",
    "\n",
    "\n",
    "get_meta_cols <- function(meta) {\n",
    "    meta_cols <- colnames(meta)[grep(\"kit|flow_cell|instrument_id\", colnames(meta))]\n",
    "    to_exclude <- c(\"library_prep_kit\", \"hiseq_xtm_flow_cell_v2_5_id\")\n",
    "    meta_cols <- meta_cols[!(meta_cols %in% to_exclude)]\n",
    "    return(meta_cols)\n",
    "}\n",
    "\n",
    "\n",
    "print_group_freqs <- function(meta, meta_cols) {\n",
    "    for (col in meta_cols) {\n",
    "        tmp <- tibble(data.frame(meta)) %>%\n",
    "            group_by(get(col)) %>%\n",
    "            summarise(n = n())\n",
    "        print(col)\n",
    "        print(tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "get_batch_prevalence <- function(x) {sum(x) / length(x)}\n",
    "\n",
    "\n",
    "remove_low_read_samples <- function(df) {\n",
    "    to_retain <- fread(\"data/samples_above_10_reads.txt\")$npm_research_id\n",
    "    return(df %>% filter(npm_research_id %in% to_retain))\n",
    "}\n",
    "\n",
    "\n",
    "remove_empty_rows <- function(df) {\n",
    "    df_filt <- df %>% \n",
    "        rowwise(npm_research_id) %>% \n",
    "        mutate(total = sum(across(everything()))) %>%\n",
    "        filter(total != 0) %>%\n",
    "        select(-total)\n",
    "    \n",
    "    n_original <- nrow(df)\n",
    "    n_removed <- n_original - nrow(df_filt)\n",
    "    \n",
    "    print(str_glue(\"{n_removed}/{n_original} samples removed due to having no reads of interest\"))\n",
    "    \n",
    "    return(as.data.frame(df_filt))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "363b9d21",
   "metadata": {},
   "source": [
    "#### Test for differential prevalence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "2393d8ca",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "run_diff_prev <- function(dat, metadat, meta_cols, read_threshold, prev_threshold, min_samples) {\n",
    "    morsels <- foreach (column = meta_cols) %do% {\n",
    "        print(column)\n",
    "        # Remove levels with < x samples\n",
    "        tmp <- tibble(data.frame(metadat)) %>%\n",
    "                group_by(get(column)) %>%\n",
    "                summarise(n = n())\n",
    "\n",
    "        # Vector of group levels to keep\n",
    "        to_keep <- tmp[tmp$n >= min_samples, \"get(column)\"]$`get(column)`\n",
    "        to_keep <- to_keep[to_keep != \"Unknown\"]\n",
    "        n_levels <- length(to_keep)\n",
    "\n",
    "        if (n_levels < 2) {\n",
    "            print(str_glue(\"After pruning, {column} has < 2 levels\"))\n",
    "        } else {\n",
    "            print(str_glue(\"After pruning, {column} has {n_levels} levels\"))\n",
    "        }\n",
    "\n",
    "        # Remove rows in metadata\n",
    "        metadat_filt <- metadat %>% filter(get(column) %in% to_keep)\n",
    "\n",
    "        # Retrieve rows\n",
    "        dat_filt <- retrieve_rows_from_meta(dat, metadat_filt)\n",
    "        metadat_filt <- metadat_filt %>% filter(npm_research_id %in% dat_filt$npm_research_id)\n",
    "\n",
    "        # Convert abundance table to presence absence table\n",
    "        prev_filt <- dat_filt[, colnames(dat_filt) != \"npm_research_id\"]\n",
    "        prev_filt[prev_filt <= read_threshold] <- 0\n",
    "        prev_filt[prev_filt > read_threshold] <- 1\n",
    "        prev_filt$npm_research_id <- dat_filt$npm_research_id\n",
    "        \n",
    "        taxa_vec <- colnames(prev_filt)\n",
    "        taxa_vec <- taxa_vec[taxa_vec != \"npm_research_id\"]\n",
    "\n",
    "        crumbs <- foreach (taxon = taxa_vec) %dopar% {\n",
    "            prev_stats <- prev_filt %>% \n",
    "                left_join(metadat_filt, by = \"npm_research_id\") %>%\n",
    "                group_by_at(c(column)) %>%\n",
    "                summarise(sum = sum(get(taxon))) %>%\n",
    "                mutate(batch_n = as.vector(table(metadat_filt[, column])),\n",
    "                       prevalence = sum / as.vector(table(metadat_filt[, column])))\n",
    "\n",
    "            max_prev <- max(prev_stats$prevalence)\n",
    "            min_prev <- min(prev_stats$prevalence)\n",
    "            fold_diff <- max_prev / min_prev\n",
    "            max_level <- pull(prev_stats, column)[which(prev_stats$prevalence == max(prev_stats$prevalence))][1]\n",
    "            min_level <- pull(prev_stats, column)[which(prev_stats$prevalence == min(prev_stats$prevalence))][1]\n",
    "            diff_prev <- ifelse(fold_diff > prev_threshold, T, F)\n",
    "\n",
    "            crumb <- tibble(taxa = taxon, meta_col = column,\n",
    "                   max_level = max_level, min_level = min_level, \n",
    "                   max_prev = max_prev, min_prev = min_prev,\n",
    "                   fold_diff = fold_diff,\n",
    "                   diff_prev = diff_prev)\n",
    "            \n",
    "            crumb <- distinct(crumb, taxa, .keep_all = T)\n",
    "\n",
    "            return(crumb)\n",
    "        }\n",
    "        \n",
    "        stopImplicitCluster()\n",
    "        \n",
    "        morsel <- bind_rows(crumbs)\n",
    "        return(morsel)\n",
    "    }\n",
    "\n",
    "    result_df <- bind_rows(morsels)\n",
    "    return(result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94d384b2",
   "metadata": {},
   "source": [
    "#### Diff. prevalence decontamination procedure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "60431428",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "decontaminate <- function(rank, n_subset, n, human, read_threshold, min_samples) {\n",
    "    # Load data and remove low microbial read samples\n",
    "    to_retain <- fread(\"data/samples_above_95_reads.txt\")$npm_research_id\n",
    "    df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.{rank}.tsv\")) %>% \n",
    "        filter(npm_research_id %in% to_retain)\n",
    "    meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "    \n",
    "    # Get metadata subset\n",
    "    meta_filt <- subset_metadata(meta, n)\n",
    "\n",
    "    # Filter data\n",
    "    df_filt <- retrieve_rows_from_meta(df, meta_filt)\n",
    "    df_filt2 <- remove_cols(df_filt, c(human, \"unclassified\"))\n",
    "    df_filt3 <- remove_low_freq_taxa(df_filt2, frac_presence = 0.05)\n",
    "    df_filt4 <- remove_empty_rows(df_filt3)\n",
    "#     df_filt4 <- otu_to_RA(df_filt4)\n",
    "\n",
    "    # Get metadata columns of interest\n",
    "    meta_cols <- get_meta_cols(meta_filt)\n",
    "    \n",
    "#     # Print group freqs\n",
    "#     print_group_freqs(meta_filt, meta_cols)\n",
    "    \n",
    "    # Differential prevalence\n",
    "    res <- run_diff_prev(df_filt4, meta_filt, meta_cols, read_threshold = read_threshold, prev_threshold = 2 , min_samples = min_samples)\n",
    "    \n",
    "    return(res)\n",
    "}\n",
    "\n",
    "\n",
    "parse_decontamination_results <- function(decontam_res) {\n",
    "    non_contaminants <- decontam_res %>%     \n",
    "        group_by(taxa) %>%\n",
    "        summarise(n_diff = sum(diff_prev)) %>%\n",
    "        filter(n_diff == 0)\n",
    "    non_contaminants <- non_contaminants$taxa\n",
    "\n",
    "    contaminants <- decontam_res %>%\n",
    "        group_by(taxa) %>%\n",
    "        summarise(n_diff = sum(diff_prev)) %>%\n",
    "        filter(n_diff > 0)\n",
    "    contaminants <- contaminants$taxa\n",
    "    \n",
    "    return(list(non_contaminants = non_contaminants, contaminants = contaminants))\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d833846c",
   "metadata": {},
   "source": [
    "#### Rarefy decontamination procedure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "566ca4e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# rarefy_decontaminate <- function(rank, human, breaks) {\n",
    "#     morsels <- foreach (n = breaks) %do% {\n",
    "#         res <- decontaminate(rank, 9999, n, human, 0.001)\n",
    "#         res_list <- parse_decontamination_results(res)\n",
    "#         morsel <- tibble(non_contaminants = res_list[[\"non_contaminants\"]], n = n)\n",
    "#         return(morsel)\n",
    "#     }\n",
    "\n",
    "#     nc_df <- bind_rows(morsels)\n",
    "#     return(nc_df)\n",
    "# }"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0744c30",
   "metadata": {},
   "source": [
    "#### Visualise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "7f3288ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "# diff_prev_heatmap <- function(nc) {\n",
    "#     plt <- nc %>% \n",
    "#         mutate(n = as.factor(n)) %>%\n",
    "#         ggplot(aes(x = n, y = non_contaminants)) +\n",
    "#         geom_tile(color = \"black\") +\n",
    "#         scale_x_discrete(breaks = breaks) +\n",
    "#         theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "#         labs(x = \"No. of samples\", y = \"Non-contaminant taxa\")\n",
    "\n",
    "#         return(plt)\n",
    "# }"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b68fe14d",
   "metadata": {},
   "source": [
    "### Main for diff. prevalence decontamination"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "6d7934a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# breaks <- c(100, 200, 400, 800, seq(1000, 9000, 1000), 9999)\n",
    "# # breaks <- c(100, 200, 400)\n",
    "# # breaks <- c(9999)\n",
    "# breaks\n",
    "\n",
    "# species_nc_df <- rarefy_decontaminate(\"S\", \"Homo sapiens\", breaks)\n",
    "# genus_nc_df <- rarefy_decontaminate(\"G\", \"Homo\", breaks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "875c26a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# diff_prev_heatmap(species_nc_df)\n",
    "# ggsave(\"results/decontamination/non_contaminant_species_heatmap.png\", dpi = 300, width = 5, height = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "a4e9e693",
   "metadata": {},
   "outputs": [],
   "source": [
    "# diff_prev_heatmap(genus_nc_df)\n",
    "# ggsave(\"results/decontamination/non_contaminant_genus_heatmap.png\", dpi = 300, width = 5, height = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "72df1525",
   "metadata": {},
   "outputs": [],
   "source": [
    "# min_samples <- 50\n",
    "# # full_genus_df <- decontaminate(\"G\", 9999, 9999, \"Homo\", 0.001, min_samples)\n",
    "# # full_genus_res <- parse_decontamination_results(full_genus_df)\n",
    "# # fwrite(full_genus_df, str_glue(\"results/decontamination/diff_prev_raw/diff_prev_results.9999.min_samples{min_samples}.G.RA.csv\"))\n",
    "\n",
    "# full_species_df <- decontaminate(\"S\", 9999, 9999, \"Homo sapiens\", 0.001, min_samples)\n",
    "# full_species_res <- parse_decontamination_results(full_species_df)\n",
    "# fwrite(full_species_df, str_glue(\"results/decontamination/diff_prev_raw/diff_prev_results.9999.min_samples{min_samples}.S.RA.csv\"))\n",
    "\n",
    "# # fwrite(tibble(non_contaminants = full_genus_res[[\"non_contaminants\"]]), str_glue(\"results/decontamination/prevalence.noncontam.min_samples{min_samples}.G.n9999.RA.txt\"))\n",
    "# # fwrite(tibble(contaminants = full_genus_res[[\"contaminants\"]]), str_glue(\"results/decontamination/prevalence.contam.G.n9999.RA.txt\"))\n",
    "\n",
    "# fwrite(tibble(non_contaminants = full_species_res[[\"non_contaminants\"]]), str_glue(\"results/decontamination/prevalence.noncontam.min_samples{min_samples}.S.n9999.RA.txt\"))\n",
    "# fwrite(tibble(contaminants = full_species_res[[\"contaminants\"]]), str_glue(\"results/decontamination/prevalence.contam.min_samples{min_samples}.S.n9999.RA.txt\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "f41ff121",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n"
     ]
    }
   ],
   "source": [
    "min_samples <- 100\n",
    "full_species_df <- decontaminate(\"S\", 9999, 9999, \"Homo sapiens\", 0, min_samples)\n",
    "full_species_res <- parse_decontamination_results(full_species_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "22cedc32",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$non_contaminants</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Altererythrobacter sp. B11'</li><li>'Bradyrhizobium erythrophlei'</li><li>'Bradyrhizobium sp. SK17'</li><li>'Cutibacterium acnes'</li><li>'Rhodopseudomonas palustris'</li></ol>\n",
       "</dd>\n",
       "\t<dt>$contaminants</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Acetobacter aceti'</li><li>'Acetobacter ghanensis'</li><li>'Acetobacter pasteurianus'</li><li>'Acetobacter persici'</li><li>'Achromobacter denitrificans'</li><li>'Achromobacter insolitus'</li><li>'Achromobacter ruhlandii'</li><li>'Achromobacter sp. AONIH1'</li><li>'Achromobacter sp. B7'</li><li>'Achromobacter sp. MFA1 R4'</li><li>'Achromobacter spanius'</li><li>'Achromobacter xylosoxidans'</li><li>'Acidiferrobacter sp. SPIII_3'</li><li>'Acidihalobacter ferrooxidans'</li><li>'Acidihalobacter prosperus'</li><li>'Acidiphilium cryptum'</li><li>'Acidiphilium multivorum'</li><li>'Acidipropionibacterium acidipropionici'</li><li>'Acidipropionibacterium jensenii'</li><li>'Acidipropionibacterium virtanenii'</li><li>'Acidisphaera sp. G45-3'</li><li>'Acidithiobacillus caldus'</li><li>'Acidithiobacillus ferridurans'</li><li>'Acidobacteriaceae bacterium SBC82'</li><li>'Acidobacterium capsulatum'</li><li>'Acidovorax avenae'</li><li>'Acidovorax carolinensis'</li><li>'Acidovorax cattleyae'</li><li>'Acidovorax citrulli'</li><li>'Acidovorax ebreus'</li><li>'Acidovorax sp. 1608163'</li><li>'Acidovorax sp. JS42'</li><li>'Acidovorax sp. KKS102'</li><li>'Acidovorax sp. RAC01'</li><li>'Acidovorax sp. T1'</li><li>'Acinetobacter baumannii'</li><li>'Acinetobacter bereziniae'</li><li>'Acinetobacter guillouiae'</li><li>'Acinetobacter haemolyticus'</li><li>'Acinetobacter indicus'</li><li>'Acinetobacter johnsonii'</li><li>'Acinetobacter junii'</li><li>'Acinetobacter lwoffii'</li><li>'Acinetobacter pittii'</li><li>'Acinetobacter radioresistens'</li><li>'Acinetobacter schindleri'</li><li>'Acinetobacter soli'</li><li>'Acinetobacter sp. ACNIH1'</li><li>'Acinetobacter sp. ACNIH2'</li><li>'Acinetobacter sp. LoGeW2-3'</li><li>'Acinetobacter sp. NCu2D-2'</li><li>'Acinetobacter sp. SWBY1'</li><li>'Acinetobacter sp. WCHA45'</li><li>'Acinetobacter sp. WCHA55'</li><li>'Acinetobacter sp. WCHAc010034'</li><li>'Acinetobacter ursingii'</li><li>'Acinetobacter venetianus'</li><li>'Acinetobacter wuhouensis'</li><li>'Actinoalloteichus hoggarensis'</li><li>'Actinoalloteichus hymeniacidonis'</li><li>'Actinoalloteichus sp. AHMU CJ021'</li><li>'Actinobacteria bacterium YIM 96077'</li><li>'Actinomadura amylolytica'</li><li>'Actinomyces hongkongensis'</li><li>'Actinomyces howellii'</li><li>'Actinomyces israelii'</li><li>'Actinomyces oris'</li><li>'Actinomyces radicidentis'</li><li>'Actinomyces slackii'</li><li>'Actinomyces sp. Chiba101'</li><li>'Actinomyces sp. oral taxon 414'</li><li>'Actinomyces sp. oral taxon 897'</li><li>'Actinomyces sp. Z16'</li><li>'Actinomyces viscosus'</li><li>'Actinoplanes derwentensis'</li><li>'Actinoplanes friuliensis'</li><li>'Actinoplanes missouriensis'</li><li>'Actinoplanes sp. N902-109'</li><li>'Actinoplanes teichomyceticus'</li><li>'Actinopolymorpha singaporensis'</li><li>'Actinopolyspora erythraea'</li><li>'Actinosynnema pretiosum'</li><li>'Adlercreutzia equolifaciens'</li><li>'Advenella kashmirensis'</li><li>'Advenella mimigardefordensis'</li><li>'Aeromicrobium choanae'</li><li>'Aeromicrobium erythreum'</li><li>'Aeromicrobium marinum'</li><li>'Aeromicrobium sp. 592'</li><li>'Aeromicrobium sp. A1-2'</li><li>'Aeromonas caviae'</li><li>'Aeromonas encheleia'</li><li>'Aeromonas hydrophila'</li><li>'Aeromonas media'</li><li>'Aeromonas salmonicida'</li><li>'Aeromonas schubertii'</li><li>'Aeromonas sp. ASNIH1'</li><li>'Aeromonas sp. ASNIH3'</li><li>'Aeromonas sp. CA23'</li><li>'Aeromonas veronii'</li><li>'Afipia sp. GAS231'</li><li>'Agrobacterium fabrum'</li><li>'Agrobacterium rhizogenes'</li><li>'Agrobacterium sp. H13-3'</li><li>'Agrobacterium sp. RAC06'</li><li>'Agrobacterium tumefaciens'</li><li>'Agrobacterium vitis'</li><li>'Agrococcus carbonis'</li><li>'Agrococcus jejuensis'</li><li>'Agromyces aureus'</li><li>'Agromyces flavus'</li><li>'Agromyces sp. 30A'</li><li>'Ahniella affigens'</li><li>'Alcaligenes aquatilis'</li><li>'Alcaligenes faecalis'</li><li>'Alcanivorax dieselolei'</li><li>'Alcanivorax pacificus'</li><li>'Alcanivorax sp. N3-2A'</li><li>'Alcanivorax xenomutans'</li><li>'Alicycliphilus denitrificans'</li><li>'Alicyclobacillus acidocaldarius'</li><li>'Alistipes shahii'</li><li>'Alistipes sp. Marseille-P5997'</li><li>'Alkalilimnicola ehrlichii'</li><li>'Alloactinosynnema sp. L-07'</li><li>'Allochromatium vinosum'</li><li>'Allokutzneria albata'</li><li>'Alphaproteobacteria bacterium WS11'</li><li>'Altererythrobacter atlanticus'</li><li>'Altererythrobacter dongtanensis'</li><li>'Altererythrobacter epoxidivorans'</li><li>'Altererythrobacter ishigakiensis'</li><li>'Altererythrobacter mangrovi'</li><li>'Altererythrobacter marensis'</li><li>'Altererythrobacter namhicola'</li><li>'Altererythrobacter sp. NS1'</li><li>'Altererythrobacter sp. ZODW24'</li><li>'Alteromonas macleodii'</li><li>'Alteromonas mediterranea'</li><li>'Alteromonas sp. MB-3u-76'</li><li>'Aminobacter aminovorans'</li><li>'Aminobacter sp. MSH1'</li><li>'Amycolatopsis albispora'</li><li>'Amycolatopsis japonica'</li><li>'Amycolatopsis keratiniphila'</li><li>'Amycolatopsis mediterranei'</li><li>'Amycolatopsis methanolica'</li><li>'Amycolatopsis orientalis'</li><li>'Amycolatopsis sp. AA4'</li><li>'Amycolatopsis sp. BJA-103'</li><li>'Anaeromyxobacter dehalogenans'</li><li>'Anaeromyxobacter sp. Fw109-5'</li><li>'Anaeromyxobacter sp. K'</li><li>'Anderseniella sp. Alg231-50'</li><li>'Antarctobacter heliothermus'</li><li>'Aquabacterium olei'</li><li>'Aquaspirillum sp. LM1'</li><li>'Aquitalea magnusonii'</li><li>'Archangium gephyra'</li><li>'Aromatoleum aromaticum'</li><li>'Arsenicicoccus sp. oral taxon 190'</li><li>'Arthrobacter alpinus'</li><li>'Arthrobacter crystallopoietes'</li><li>'Arthrobacter sp. DCT-5'</li><li>'Arthrobacter sp. ERGS1:01'</li><li>'Arthrobacter sp. FB24'</li><li>'Arthrobacter sp. PAMC 25486'</li><li>'Arthrobacter sp. PGP41'</li><li>'Arthrobacter sp. QXT-31'</li><li>'Arthrobacter sp. U41'</li><li>'Arthrobacter sp. YC-RL1'</li><li>'Arthrobacter sp. YN'</li><li>'Arthrobacter sp. ZXY-2'</li><li>'Asaia bogorensis'</li><li>'Asticcacaulis excentricus'</li><li>'Atlantibacter hermannii'</li><li>'Aurantimicrobium sp. MWH-Mo1'</li><li>'Auraticoccus monumenti'</li><li>'Aureimonas sp. AU20'</li><li>'Auritidibacter sp. NML130574'</li><li>'Austwickia chelonae'</li><li>'Azoarcus communis'</li><li>'Azoarcus olearius'</li><li>'Azoarcus sp. BH72'</li><li>'Azoarcus sp. CIB'</li><li>'Azoarcus sp. DN11'</li><li>'Azoarcus sp. KH32C'</li><li>'Azoarcus sp. SY39'</li><li>'Azorhizobium caulinodans'</li><li>'Azospira oryzae'</li><li>'Azospirillum brasilense'</li><li>'Azospirillum humicireducens'</li><li>'Azospirillum lipoferum'</li><li>'Azospirillum sp. CFH 70021'</li><li>'Azospirillum sp. M2T2B2'</li><li>'Azospirillum sp. TSH58'</li><li>'Azospirillum thiophilum'</li><li>'Azotobacter chroococcum'</li><li>'Azotobacter vinelandii'</li><li>'Bacillus cereus'</li><li>⋯</li><li>'Streptomyces griseochromogenes'</li><li>'Streptomyces griseorubiginosus'</li><li>'Streptomyces griseus'</li><li>'Streptomyces hundungensis'</li><li>'Streptomyces hygroscopicus'</li><li>'Streptomyces katrae'</li><li>'Streptomyces koyangensis'</li><li>'Streptomyces lavendulae'</li><li>'Streptomyces leeuwenhoekii'</li><li>'Streptomyces lincolnensis'</li><li>'Streptomyces lunaelactis'</li><li>'Streptomyces luteoverticillatus'</li><li>'Streptomyces lydicus'</li><li>'Streptomyces malaysiensis'</li><li>'Streptomyces nigra'</li><li>'Streptomyces niveus'</li><li>'Streptomyces nodosus'</li><li>'Streptomyces noursei'</li><li>'Streptomyces olivaceus'</li><li>'Streptomyces olivoreticuli'</li><li>'Streptomyces pactum'</li><li>'Streptomyces parvulus'</li><li>'Streptomyces peucetius'</li><li>'Streptomyces pluripotens'</li><li>'Streptomyces pratensis'</li><li>'Streptomyces pristinaespiralis'</li><li>'Streptomyces puniciscabiei'</li><li>'Streptomyces qaidamensis'</li><li>'Streptomyces reticuli'</li><li>'Streptomyces roseochromogenus'</li><li>'Streptomyces rubrolavendulae'</li><li>'Streptomyces scabiei'</li><li>'Streptomyces sp. 11-1-2'</li><li>'Streptomyces sp. 2323.1'</li><li>'Streptomyces sp. 3211'</li><li>'Streptomyces sp. 3214.6'</li><li>'Streptomyces sp. 4F'</li><li>'Streptomyces sp. 769'</li><li>'Streptomyces sp. ADI95-16'</li><li>'Streptomyces sp. CB09001'</li><li>'Streptomyces sp. CCM_MD2014'</li><li>'Streptomyces sp. CdTB01'</li><li>'Streptomyces sp. CFMR 7'</li><li>'Streptomyces sp. CMB-StM0423'</li><li>'Streptomyces sp. CNQ-509'</li><li>'Streptomyces sp. ETH9427'</li><li>'Streptomyces sp. fd1-xmd'</li><li>'Streptomyces sp. FR-008'</li><li>'Streptomyces sp. Go-475'</li><li>'Streptomyces sp. GSSD-12'</li><li>'Streptomyces sp. HNM0039'</li><li>'Streptomyces sp. KPB2'</li><li>'Streptomyces sp. M56'</li><li>'Streptomyces sp. Mg1'</li><li>'Streptomyces sp. MK45'</li><li>'Streptomyces sp. NEAU-S7GS2'</li><li>'Streptomyces sp. P3'</li><li>'Streptomyces sp. RTd22'</li><li>'Streptomyces sp. S8'</li><li>'Streptomyces sp. SAT1'</li><li>'Streptomyces sp. SCSIO 03032'</li><li>'Streptomyces sp. Sge12'</li><li>'Streptomyces sp. SirexAA-E'</li><li>'Streptomyces sp. SM18'</li><li>'Streptomyces sp. TLI_053'</li><li>'Streptomyces sp. TN58'</li><li>'Streptomyces sp. Tue 6075'</li><li>'Streptomyces sp. W1SF4'</li><li>'Streptomyces sp. WAC 01438'</li><li>'Streptomyces sp. WAC 01529'</li><li>'Streptomyces sp. WAC 06738'</li><li>'Streptomyces sp. WAC00288'</li><li>'Streptomyces sp. Z022'</li><li>'Streptomyces sp. ZFG47'</li><li>'Streptomyces spongiicola'</li><li>'Streptomyces venezuelae'</li><li>'Streptomyces vietnamensis'</li><li>'Streptomyces violaceoruber'</li><li>'Streptomyces violaceusniger'</li><li>'Streptomyces xiamenensis'</li><li>'Streptomyces xinghaiensis'</li><li>'Streptosporangium roseum'</li><li>'Streptosporangium sp. \\'caverna\\''</li><li>'Sulfitobacter pseudonitzschiae'</li><li>'Sulfitobacter sp. AM1-D1'</li><li>'Sulfitobacter sp. D7'</li><li>'Sulfitobacter sp. JL08'</li><li>'Sulfitobacter sp. SK011'</li><li>'Sulfitobacter sp. SK012'</li><li>'Sulfitobacter sp. SK025'</li><li>'Sulfuricaulis limicola'</li><li>'Sulfuricella denitrificans'</li><li>'Sulfuriferula sp. AH1'</li><li>'Sulfurifustis variabilis'</li><li>'Sulfuritalea hydrogenivorans'</li><li>'Sulfuritortus calidifontis'</li><li>'Sulfurospirillum multivorans'</li><li>'Sutterella megalosphaeroides'</li><li>'Symbiobacterium thermophilum'</li><li>'Synechococcus sp. KORDI-49'</li><li>'Synechococcus sp. PCC 7336'</li><li>'Synechococcus sp. RCC307'</li><li>'Synechococcus sp. SynAce01'</li><li>'Syntrophobotulus glycolicus'</li><li>'Tabrizicola sp. K13M18'</li><li>'Tateyamaria omphalii'</li><li>'Terriglobus roseus'</li><li>'Tessaracoccus aquimaris'</li><li>'Tessaracoccus flavescens'</li><li>'Tessaracoccus flavus'</li><li>'Tessaracoccus sp. Marseille-P5995'</li><li>'Tessaracoccus sp. T2.5-30'</li><li>'Thalassococcus sp. SH-1'</li><li>'Thalassospira marina'</li><li>'Thalassospira xiamenensis'</li><li>'Thauera aromatica'</li><li>'Thauera chlorobenzoica'</li><li>'Thauera humireducens'</li><li>'Thauera hydrothermalis'</li><li>'Thauera sp. K11'</li><li>'Thauera sp. MZ1T'</li><li>'Thermaerobacter marianensis'</li><li>'Thermoanaerobacterales bacterium SK-G1'</li><li>'Thermobacillus composti'</li><li>'Thermobifida fusca'</li><li>'Thermobispora bispora'</li><li>'Thermomonospora curvata'</li><li>'Thermus thermophilus'</li><li>'Thioalkalivibrio nitratireducens'</li><li>'Thioalkalivibrio paradoxus'</li><li>'Thioalkalivibrio sp. K90mix'</li><li>'Thioalkalivibrio sulfidiphilus'</li><li>'Thioalkalivibrio versutus'</li><li>'Thiobacillus denitrificans'</li><li>'Thioclava nitratireducens'</li><li>'Thiocystis violascens'</li><li>'Thioflavicoccus mobilis'</li><li>'Thiohalobacter thiocyanaticus'</li><li>'Thiolapillus brandeum'</li><li>'Thiomonas arsenitoxydans'</li><li>'Thiomonas intermedia'</li><li>'Thiomonas sp. X19'</li><li>'Tistrella mobilis'</li><li>'Tolumonas auensis'</li><li>'Torque teno virus 13'</li><li>'Torque teno virus 15'</li><li>'Torque teno virus 19'</li><li>'Torque teno virus 24'</li><li>'Torque teno virus 29'</li><li>'Truepera radiovictrix'</li><li>'Trueperella pyogenes'</li><li>'Tsukamurella paurometabola'</li><li>'Tsukamurella tyrosinosolvens'</li><li>'Undibacterium parvum'</li><li>'Variibacter gotjawalensis'</li><li>'Variovorax boronicumulans'</li><li>'Variovorax paradoxus'</li><li>'Variovorax sp. HW608'</li><li>'Variovorax sp. PAMC 28711'</li><li>'Variovorax sp. PMC12'</li><li>'Verminephrobacter eiseniae'</li><li>'Verrucomicrobia bacterium'</li><li>'Verrucomicrobia bacterium IMCC26134'</li><li>'Verrucomicrobium sp. GAS474'</li><li>'Verrucomicrobium spinosum'</li><li>'Verrucosispora maris'</li><li>'Vibrio fluvialis'</li><li>'Vibrio furnissii'</li><li>'Vibrio vulnificus'</li><li>'Victivallales bacterium CCUG 44730'</li><li>'Vitreoscilla filiformis'</li><li>'Vogesella sp. LIG4'</li><li>'Vulgatibacter incomptus'</li><li>'Wenzhouxiangella marina'</li><li>'Xanthobacter autotrophicus'</li><li>'Xanthomonas albilineans'</li><li>'Xanthomonas arboricola'</li><li>'Xanthomonas campestris'</li><li>'Xanthomonas cassavae'</li><li>'Xanthomonas citri'</li><li>'Xanthomonas euvesicatoria'</li><li>'Xanthomonas fragariae'</li><li>'Xanthomonas gardneri'</li><li>'Xanthomonas hortorum'</li><li>'Xanthomonas oryzae'</li><li>'Xanthomonas phaseoli'</li><li>'Xanthomonas sacchari'</li><li>'Xanthomonas translucens'</li><li>'Xanthomonas vesicatoria'</li><li>'Xylanimonas cellulosilytica'</li><li>'Xylella fastidiosa'</li><li>'Yangia pacifica'</li><li>'Yangia sp. CCB-MM3'</li><li>'Yersinia enterocolitica'</li><li>'Yoonia vestfoldensis'</li><li>'Zhihengliuella sp. ISTPL4'</li><li>'Zobellella denitrificans'</li><li>'Zoogloeaceae bacteirum Par-f-2'</li><li>'Zymobacter palmae'</li><li>'Zymomonas mobilis'</li></ol>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$non\\_contaminants] \\begin{enumerate*}\n",
       "\\item 'Altererythrobacter sp. B11'\n",
       "\\item 'Bradyrhizobium erythrophlei'\n",
       "\\item 'Bradyrhizobium sp. SK17'\n",
       "\\item 'Cutibacterium acnes'\n",
       "\\item 'Rhodopseudomonas palustris'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$contaminants] \\begin{enumerate*}\n",
       "\\item 'Acetobacter aceti'\n",
       "\\item 'Acetobacter ghanensis'\n",
       "\\item 'Acetobacter pasteurianus'\n",
       "\\item 'Acetobacter persici'\n",
       "\\item 'Achromobacter denitrificans'\n",
       "\\item 'Achromobacter insolitus'\n",
       "\\item 'Achromobacter ruhlandii'\n",
       "\\item 'Achromobacter sp. AONIH1'\n",
       "\\item 'Achromobacter sp. B7'\n",
       "\\item 'Achromobacter sp. MFA1 R4'\n",
       "\\item 'Achromobacter spanius'\n",
       "\\item 'Achromobacter xylosoxidans'\n",
       "\\item 'Acidiferrobacter sp. SPIII\\_3'\n",
       "\\item 'Acidihalobacter ferrooxidans'\n",
       "\\item 'Acidihalobacter prosperus'\n",
       "\\item 'Acidiphilium cryptum'\n",
       "\\item 'Acidiphilium multivorum'\n",
       "\\item 'Acidipropionibacterium acidipropionici'\n",
       "\\item 'Acidipropionibacterium jensenii'\n",
       "\\item 'Acidipropionibacterium virtanenii'\n",
       "\\item 'Acidisphaera sp. G45-3'\n",
       "\\item 'Acidithiobacillus caldus'\n",
       "\\item 'Acidithiobacillus ferridurans'\n",
       "\\item 'Acidobacteriaceae bacterium SBC82'\n",
       "\\item 'Acidobacterium capsulatum'\n",
       "\\item 'Acidovorax avenae'\n",
       "\\item 'Acidovorax carolinensis'\n",
       "\\item 'Acidovorax cattleyae'\n",
       "\\item 'Acidovorax citrulli'\n",
       "\\item 'Acidovorax ebreus'\n",
       "\\item 'Acidovorax sp. 1608163'\n",
       "\\item 'Acidovorax sp. JS42'\n",
       "\\item 'Acidovorax sp. KKS102'\n",
       "\\item 'Acidovorax sp. RAC01'\n",
       "\\item 'Acidovorax sp. T1'\n",
       "\\item 'Acinetobacter baumannii'\n",
       "\\item 'Acinetobacter bereziniae'\n",
       "\\item 'Acinetobacter guillouiae'\n",
       "\\item 'Acinetobacter haemolyticus'\n",
       "\\item 'Acinetobacter indicus'\n",
       "\\item 'Acinetobacter johnsonii'\n",
       "\\item 'Acinetobacter junii'\n",
       "\\item 'Acinetobacter lwoffii'\n",
       "\\item 'Acinetobacter pittii'\n",
       "\\item 'Acinetobacter radioresistens'\n",
       "\\item 'Acinetobacter schindleri'\n",
       "\\item 'Acinetobacter soli'\n",
       "\\item 'Acinetobacter sp. ACNIH1'\n",
       "\\item 'Acinetobacter sp. ACNIH2'\n",
       "\\item 'Acinetobacter sp. LoGeW2-3'\n",
       "\\item 'Acinetobacter sp. NCu2D-2'\n",
       "\\item 'Acinetobacter sp. SWBY1'\n",
       "\\item 'Acinetobacter sp. WCHA45'\n",
       "\\item 'Acinetobacter sp. WCHA55'\n",
       "\\item 'Acinetobacter sp. WCHAc010034'\n",
       "\\item 'Acinetobacter ursingii'\n",
       "\\item 'Acinetobacter venetianus'\n",
       "\\item 'Acinetobacter wuhouensis'\n",
       "\\item 'Actinoalloteichus hoggarensis'\n",
       "\\item 'Actinoalloteichus hymeniacidonis'\n",
       "\\item 'Actinoalloteichus sp. AHMU CJ021'\n",
       "\\item 'Actinobacteria bacterium YIM 96077'\n",
       "\\item 'Actinomadura amylolytica'\n",
       "\\item 'Actinomyces hongkongensis'\n",
       "\\item 'Actinomyces howellii'\n",
       "\\item 'Actinomyces israelii'\n",
       "\\item 'Actinomyces oris'\n",
       "\\item 'Actinomyces radicidentis'\n",
       "\\item 'Actinomyces slackii'\n",
       "\\item 'Actinomyces sp. Chiba101'\n",
       "\\item 'Actinomyces sp. oral taxon 414'\n",
       "\\item 'Actinomyces sp. oral taxon 897'\n",
       "\\item 'Actinomyces sp. Z16'\n",
       "\\item 'Actinomyces viscosus'\n",
       "\\item 'Actinoplanes derwentensis'\n",
       "\\item 'Actinoplanes friuliensis'\n",
       "\\item 'Actinoplanes missouriensis'\n",
       "\\item 'Actinoplanes sp. N902-109'\n",
       "\\item 'Actinoplanes teichomyceticus'\n",
       "\\item 'Actinopolymorpha singaporensis'\n",
       "\\item 'Actinopolyspora erythraea'\n",
       "\\item 'Actinosynnema pretiosum'\n",
       "\\item 'Adlercreutzia equolifaciens'\n",
       "\\item 'Advenella kashmirensis'\n",
       "\\item 'Advenella mimigardefordensis'\n",
       "\\item 'Aeromicrobium choanae'\n",
       "\\item 'Aeromicrobium erythreum'\n",
       "\\item 'Aeromicrobium marinum'\n",
       "\\item 'Aeromicrobium sp. 592'\n",
       "\\item 'Aeromicrobium sp. A1-2'\n",
       "\\item 'Aeromonas caviae'\n",
       "\\item 'Aeromonas encheleia'\n",
       "\\item 'Aeromonas hydrophila'\n",
       "\\item 'Aeromonas media'\n",
       "\\item 'Aeromonas salmonicida'\n",
       "\\item 'Aeromonas schubertii'\n",
       "\\item 'Aeromonas sp. ASNIH1'\n",
       "\\item 'Aeromonas sp. ASNIH3'\n",
       "\\item 'Aeromonas sp. CA23'\n",
       "\\item 'Aeromonas veronii'\n",
       "\\item 'Afipia sp. GAS231'\n",
       "\\item 'Agrobacterium fabrum'\n",
       "\\item 'Agrobacterium rhizogenes'\n",
       "\\item 'Agrobacterium sp. H13-3'\n",
       "\\item 'Agrobacterium sp. RAC06'\n",
       "\\item 'Agrobacterium tumefaciens'\n",
       "\\item 'Agrobacterium vitis'\n",
       "\\item 'Agrococcus carbonis'\n",
       "\\item 'Agrococcus jejuensis'\n",
       "\\item 'Agromyces aureus'\n",
       "\\item 'Agromyces flavus'\n",
       "\\item 'Agromyces sp. 30A'\n",
       "\\item 'Ahniella affigens'\n",
       "\\item 'Alcaligenes aquatilis'\n",
       "\\item 'Alcaligenes faecalis'\n",
       "\\item 'Alcanivorax dieselolei'\n",
       "\\item 'Alcanivorax pacificus'\n",
       "\\item 'Alcanivorax sp. N3-2A'\n",
       "\\item 'Alcanivorax xenomutans'\n",
       "\\item 'Alicycliphilus denitrificans'\n",
       "\\item 'Alicyclobacillus acidocaldarius'\n",
       "\\item 'Alistipes shahii'\n",
       "\\item 'Alistipes sp. Marseille-P5997'\n",
       "\\item 'Alkalilimnicola ehrlichii'\n",
       "\\item 'Alloactinosynnema sp. L-07'\n",
       "\\item 'Allochromatium vinosum'\n",
       "\\item 'Allokutzneria albata'\n",
       "\\item 'Alphaproteobacteria bacterium WS11'\n",
       "\\item 'Altererythrobacter atlanticus'\n",
       "\\item 'Altererythrobacter dongtanensis'\n",
       "\\item 'Altererythrobacter epoxidivorans'\n",
       "\\item 'Altererythrobacter ishigakiensis'\n",
       "\\item 'Altererythrobacter mangrovi'\n",
       "\\item 'Altererythrobacter marensis'\n",
       "\\item 'Altererythrobacter namhicola'\n",
       "\\item 'Altererythrobacter sp. NS1'\n",
       "\\item 'Altererythrobacter sp. ZODW24'\n",
       "\\item 'Alteromonas macleodii'\n",
       "\\item 'Alteromonas mediterranea'\n",
       "\\item 'Alteromonas sp. MB-3u-76'\n",
       "\\item 'Aminobacter aminovorans'\n",
       "\\item 'Aminobacter sp. MSH1'\n",
       "\\item 'Amycolatopsis albispora'\n",
       "\\item 'Amycolatopsis japonica'\n",
       "\\item 'Amycolatopsis keratiniphila'\n",
       "\\item 'Amycolatopsis mediterranei'\n",
       "\\item 'Amycolatopsis methanolica'\n",
       "\\item 'Amycolatopsis orientalis'\n",
       "\\item 'Amycolatopsis sp. AA4'\n",
       "\\item 'Amycolatopsis sp. BJA-103'\n",
       "\\item 'Anaeromyxobacter dehalogenans'\n",
       "\\item 'Anaeromyxobacter sp. Fw109-5'\n",
       "\\item 'Anaeromyxobacter sp. K'\n",
       "\\item 'Anderseniella sp. Alg231-50'\n",
       "\\item 'Antarctobacter heliothermus'\n",
       "\\item 'Aquabacterium olei'\n",
       "\\item 'Aquaspirillum sp. LM1'\n",
       "\\item 'Aquitalea magnusonii'\n",
       "\\item 'Archangium gephyra'\n",
       "\\item 'Aromatoleum aromaticum'\n",
       "\\item 'Arsenicicoccus sp. oral taxon 190'\n",
       "\\item 'Arthrobacter alpinus'\n",
       "\\item 'Arthrobacter crystallopoietes'\n",
       "\\item 'Arthrobacter sp. DCT-5'\n",
       "\\item 'Arthrobacter sp. ERGS1:01'\n",
       "\\item 'Arthrobacter sp. FB24'\n",
       "\\item 'Arthrobacter sp. PAMC 25486'\n",
       "\\item 'Arthrobacter sp. PGP41'\n",
       "\\item 'Arthrobacter sp. QXT-31'\n",
       "\\item 'Arthrobacter sp. U41'\n",
       "\\item 'Arthrobacter sp. YC-RL1'\n",
       "\\item 'Arthrobacter sp. YN'\n",
       "\\item 'Arthrobacter sp. ZXY-2'\n",
       "\\item 'Asaia bogorensis'\n",
       "\\item 'Asticcacaulis excentricus'\n",
       "\\item 'Atlantibacter hermannii'\n",
       "\\item 'Aurantimicrobium sp. MWH-Mo1'\n",
       "\\item 'Auraticoccus monumenti'\n",
       "\\item 'Aureimonas sp. AU20'\n",
       "\\item 'Auritidibacter sp. NML130574'\n",
       "\\item 'Austwickia chelonae'\n",
       "\\item 'Azoarcus communis'\n",
       "\\item 'Azoarcus olearius'\n",
       "\\item 'Azoarcus sp. BH72'\n",
       "\\item 'Azoarcus sp. CIB'\n",
       "\\item 'Azoarcus sp. DN11'\n",
       "\\item 'Azoarcus sp. KH32C'\n",
       "\\item 'Azoarcus sp. SY39'\n",
       "\\item 'Azorhizobium caulinodans'\n",
       "\\item 'Azospira oryzae'\n",
       "\\item 'Azospirillum brasilense'\n",
       "\\item 'Azospirillum humicireducens'\n",
       "\\item 'Azospirillum lipoferum'\n",
       "\\item 'Azospirillum sp. CFH 70021'\n",
       "\\item 'Azospirillum sp. M2T2B2'\n",
       "\\item 'Azospirillum sp. TSH58'\n",
       "\\item 'Azospirillum thiophilum'\n",
       "\\item 'Azotobacter chroococcum'\n",
       "\\item 'Azotobacter vinelandii'\n",
       "\\item 'Bacillus cereus'\n",
       "\\item ⋯\n",
       "\\item 'Streptomyces griseochromogenes'\n",
       "\\item 'Streptomyces griseorubiginosus'\n",
       "\\item 'Streptomyces griseus'\n",
       "\\item 'Streptomyces hundungensis'\n",
       "\\item 'Streptomyces hygroscopicus'\n",
       "\\item 'Streptomyces katrae'\n",
       "\\item 'Streptomyces koyangensis'\n",
       "\\item 'Streptomyces lavendulae'\n",
       "\\item 'Streptomyces leeuwenhoekii'\n",
       "\\item 'Streptomyces lincolnensis'\n",
       "\\item 'Streptomyces lunaelactis'\n",
       "\\item 'Streptomyces luteoverticillatus'\n",
       "\\item 'Streptomyces lydicus'\n",
       "\\item 'Streptomyces malaysiensis'\n",
       "\\item 'Streptomyces nigra'\n",
       "\\item 'Streptomyces niveus'\n",
       "\\item 'Streptomyces nodosus'\n",
       "\\item 'Streptomyces noursei'\n",
       "\\item 'Streptomyces olivaceus'\n",
       "\\item 'Streptomyces olivoreticuli'\n",
       "\\item 'Streptomyces pactum'\n",
       "\\item 'Streptomyces parvulus'\n",
       "\\item 'Streptomyces peucetius'\n",
       "\\item 'Streptomyces pluripotens'\n",
       "\\item 'Streptomyces pratensis'\n",
       "\\item 'Streptomyces pristinaespiralis'\n",
       "\\item 'Streptomyces puniciscabiei'\n",
       "\\item 'Streptomyces qaidamensis'\n",
       "\\item 'Streptomyces reticuli'\n",
       "\\item 'Streptomyces roseochromogenus'\n",
       "\\item 'Streptomyces rubrolavendulae'\n",
       "\\item 'Streptomyces scabiei'\n",
       "\\item 'Streptomyces sp. 11-1-2'\n",
       "\\item 'Streptomyces sp. 2323.1'\n",
       "\\item 'Streptomyces sp. 3211'\n",
       "\\item 'Streptomyces sp. 3214.6'\n",
       "\\item 'Streptomyces sp. 4F'\n",
       "\\item 'Streptomyces sp. 769'\n",
       "\\item 'Streptomyces sp. ADI95-16'\n",
       "\\item 'Streptomyces sp. CB09001'\n",
       "\\item 'Streptomyces sp. CCM\\_MD2014'\n",
       "\\item 'Streptomyces sp. CdTB01'\n",
       "\\item 'Streptomyces sp. CFMR 7'\n",
       "\\item 'Streptomyces sp. CMB-StM0423'\n",
       "\\item 'Streptomyces sp. CNQ-509'\n",
       "\\item 'Streptomyces sp. ETH9427'\n",
       "\\item 'Streptomyces sp. fd1-xmd'\n",
       "\\item 'Streptomyces sp. FR-008'\n",
       "\\item 'Streptomyces sp. Go-475'\n",
       "\\item 'Streptomyces sp. GSSD-12'\n",
       "\\item 'Streptomyces sp. HNM0039'\n",
       "\\item 'Streptomyces sp. KPB2'\n",
       "\\item 'Streptomyces sp. M56'\n",
       "\\item 'Streptomyces sp. Mg1'\n",
       "\\item 'Streptomyces sp. MK45'\n",
       "\\item 'Streptomyces sp. NEAU-S7GS2'\n",
       "\\item 'Streptomyces sp. P3'\n",
       "\\item 'Streptomyces sp. RTd22'\n",
       "\\item 'Streptomyces sp. S8'\n",
       "\\item 'Streptomyces sp. SAT1'\n",
       "\\item 'Streptomyces sp. SCSIO 03032'\n",
       "\\item 'Streptomyces sp. Sge12'\n",
       "\\item 'Streptomyces sp. SirexAA-E'\n",
       "\\item 'Streptomyces sp. SM18'\n",
       "\\item 'Streptomyces sp. TLI\\_053'\n",
       "\\item 'Streptomyces sp. TN58'\n",
       "\\item 'Streptomyces sp. Tue 6075'\n",
       "\\item 'Streptomyces sp. W1SF4'\n",
       "\\item 'Streptomyces sp. WAC 01438'\n",
       "\\item 'Streptomyces sp. WAC 01529'\n",
       "\\item 'Streptomyces sp. WAC 06738'\n",
       "\\item 'Streptomyces sp. WAC00288'\n",
       "\\item 'Streptomyces sp. Z022'\n",
       "\\item 'Streptomyces sp. ZFG47'\n",
       "\\item 'Streptomyces spongiicola'\n",
       "\\item 'Streptomyces venezuelae'\n",
       "\\item 'Streptomyces vietnamensis'\n",
       "\\item 'Streptomyces violaceoruber'\n",
       "\\item 'Streptomyces violaceusniger'\n",
       "\\item 'Streptomyces xiamenensis'\n",
       "\\item 'Streptomyces xinghaiensis'\n",
       "\\item 'Streptosporangium roseum'\n",
       "\\item 'Streptosporangium sp. \\textbackslash{}'caverna\\textbackslash{}''\n",
       "\\item 'Sulfitobacter pseudonitzschiae'\n",
       "\\item 'Sulfitobacter sp. AM1-D1'\n",
       "\\item 'Sulfitobacter sp. D7'\n",
       "\\item 'Sulfitobacter sp. JL08'\n",
       "\\item 'Sulfitobacter sp. SK011'\n",
       "\\item 'Sulfitobacter sp. SK012'\n",
       "\\item 'Sulfitobacter sp. SK025'\n",
       "\\item 'Sulfuricaulis limicola'\n",
       "\\item 'Sulfuricella denitrificans'\n",
       "\\item 'Sulfuriferula sp. AH1'\n",
       "\\item 'Sulfurifustis variabilis'\n",
       "\\item 'Sulfuritalea hydrogenivorans'\n",
       "\\item 'Sulfuritortus calidifontis'\n",
       "\\item 'Sulfurospirillum multivorans'\n",
       "\\item 'Sutterella megalosphaeroides'\n",
       "\\item 'Symbiobacterium thermophilum'\n",
       "\\item 'Synechococcus sp. KORDI-49'\n",
       "\\item 'Synechococcus sp. PCC 7336'\n",
       "\\item 'Synechococcus sp. RCC307'\n",
       "\\item 'Synechococcus sp. SynAce01'\n",
       "\\item 'Syntrophobotulus glycolicus'\n",
       "\\item 'Tabrizicola sp. K13M18'\n",
       "\\item 'Tateyamaria omphalii'\n",
       "\\item 'Terriglobus roseus'\n",
       "\\item 'Tessaracoccus aquimaris'\n",
       "\\item 'Tessaracoccus flavescens'\n",
       "\\item 'Tessaracoccus flavus'\n",
       "\\item 'Tessaracoccus sp. Marseille-P5995'\n",
       "\\item 'Tessaracoccus sp. T2.5-30'\n",
       "\\item 'Thalassococcus sp. SH-1'\n",
       "\\item 'Thalassospira marina'\n",
       "\\item 'Thalassospira xiamenensis'\n",
       "\\item 'Thauera aromatica'\n",
       "\\item 'Thauera chlorobenzoica'\n",
       "\\item 'Thauera humireducens'\n",
       "\\item 'Thauera hydrothermalis'\n",
       "\\item 'Thauera sp. K11'\n",
       "\\item 'Thauera sp. MZ1T'\n",
       "\\item 'Thermaerobacter marianensis'\n",
       "\\item 'Thermoanaerobacterales bacterium SK-G1'\n",
       "\\item 'Thermobacillus composti'\n",
       "\\item 'Thermobifida fusca'\n",
       "\\item 'Thermobispora bispora'\n",
       "\\item 'Thermomonospora curvata'\n",
       "\\item 'Thermus thermophilus'\n",
       "\\item 'Thioalkalivibrio nitratireducens'\n",
       "\\item 'Thioalkalivibrio paradoxus'\n",
       "\\item 'Thioalkalivibrio sp. K90mix'\n",
       "\\item 'Thioalkalivibrio sulfidiphilus'\n",
       "\\item 'Thioalkalivibrio versutus'\n",
       "\\item 'Thiobacillus denitrificans'\n",
       "\\item 'Thioclava nitratireducens'\n",
       "\\item 'Thiocystis violascens'\n",
       "\\item 'Thioflavicoccus mobilis'\n",
       "\\item 'Thiohalobacter thiocyanaticus'\n",
       "\\item 'Thiolapillus brandeum'\n",
       "\\item 'Thiomonas arsenitoxydans'\n",
       "\\item 'Thiomonas intermedia'\n",
       "\\item 'Thiomonas sp. X19'\n",
       "\\item 'Tistrella mobilis'\n",
       "\\item 'Tolumonas auensis'\n",
       "\\item 'Torque teno virus 13'\n",
       "\\item 'Torque teno virus 15'\n",
       "\\item 'Torque teno virus 19'\n",
       "\\item 'Torque teno virus 24'\n",
       "\\item 'Torque teno virus 29'\n",
       "\\item 'Truepera radiovictrix'\n",
       "\\item 'Trueperella pyogenes'\n",
       "\\item 'Tsukamurella paurometabola'\n",
       "\\item 'Tsukamurella tyrosinosolvens'\n",
       "\\item 'Undibacterium parvum'\n",
       "\\item 'Variibacter gotjawalensis'\n",
       "\\item 'Variovorax boronicumulans'\n",
       "\\item 'Variovorax paradoxus'\n",
       "\\item 'Variovorax sp. HW608'\n",
       "\\item 'Variovorax sp. PAMC 28711'\n",
       "\\item 'Variovorax sp. PMC12'\n",
       "\\item 'Verminephrobacter eiseniae'\n",
       "\\item 'Verrucomicrobia bacterium'\n",
       "\\item 'Verrucomicrobia bacterium IMCC26134'\n",
       "\\item 'Verrucomicrobium sp. GAS474'\n",
       "\\item 'Verrucomicrobium spinosum'\n",
       "\\item 'Verrucosispora maris'\n",
       "\\item 'Vibrio fluvialis'\n",
       "\\item 'Vibrio furnissii'\n",
       "\\item 'Vibrio vulnificus'\n",
       "\\item 'Victivallales bacterium CCUG 44730'\n",
       "\\item 'Vitreoscilla filiformis'\n",
       "\\item 'Vogesella sp. LIG4'\n",
       "\\item 'Vulgatibacter incomptus'\n",
       "\\item 'Wenzhouxiangella marina'\n",
       "\\item 'Xanthobacter autotrophicus'\n",
       "\\item 'Xanthomonas albilineans'\n",
       "\\item 'Xanthomonas arboricola'\n",
       "\\item 'Xanthomonas campestris'\n",
       "\\item 'Xanthomonas cassavae'\n",
       "\\item 'Xanthomonas citri'\n",
       "\\item 'Xanthomonas euvesicatoria'\n",
       "\\item 'Xanthomonas fragariae'\n",
       "\\item 'Xanthomonas gardneri'\n",
       "\\item 'Xanthomonas hortorum'\n",
       "\\item 'Xanthomonas oryzae'\n",
       "\\item 'Xanthomonas phaseoli'\n",
       "\\item 'Xanthomonas sacchari'\n",
       "\\item 'Xanthomonas translucens'\n",
       "\\item 'Xanthomonas vesicatoria'\n",
       "\\item 'Xylanimonas cellulosilytica'\n",
       "\\item 'Xylella fastidiosa'\n",
       "\\item 'Yangia pacifica'\n",
       "\\item 'Yangia sp. CCB-MM3'\n",
       "\\item 'Yersinia enterocolitica'\n",
       "\\item 'Yoonia vestfoldensis'\n",
       "\\item 'Zhihengliuella sp. ISTPL4'\n",
       "\\item 'Zobellella denitrificans'\n",
       "\\item 'Zoogloeaceae bacteirum Par-f-2'\n",
       "\\item 'Zymobacter palmae'\n",
       "\\item 'Zymomonas mobilis'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$non_contaminants\n",
       ":   1. 'Altererythrobacter sp. B11'\n",
       "2. 'Bradyrhizobium erythrophlei'\n",
       "3. 'Bradyrhizobium sp. SK17'\n",
       "4. 'Cutibacterium acnes'\n",
       "5. 'Rhodopseudomonas palustris'\n",
       "\n",
       "\n",
       "\n",
       "$contaminants\n",
       ":   1. 'Acetobacter aceti'\n",
       "2. 'Acetobacter ghanensis'\n",
       "3. 'Acetobacter pasteurianus'\n",
       "4. 'Acetobacter persici'\n",
       "5. 'Achromobacter denitrificans'\n",
       "6. 'Achromobacter insolitus'\n",
       "7. 'Achromobacter ruhlandii'\n",
       "8. 'Achromobacter sp. AONIH1'\n",
       "9. 'Achromobacter sp. B7'\n",
       "10. 'Achromobacter sp. MFA1 R4'\n",
       "11. 'Achromobacter spanius'\n",
       "12. 'Achromobacter xylosoxidans'\n",
       "13. 'Acidiferrobacter sp. SPIII_3'\n",
       "14. 'Acidihalobacter ferrooxidans'\n",
       "15. 'Acidihalobacter prosperus'\n",
       "16. 'Acidiphilium cryptum'\n",
       "17. 'Acidiphilium multivorum'\n",
       "18. 'Acidipropionibacterium acidipropionici'\n",
       "19. 'Acidipropionibacterium jensenii'\n",
       "20. 'Acidipropionibacterium virtanenii'\n",
       "21. 'Acidisphaera sp. G45-3'\n",
       "22. 'Acidithiobacillus caldus'\n",
       "23. 'Acidithiobacillus ferridurans'\n",
       "24. 'Acidobacteriaceae bacterium SBC82'\n",
       "25. 'Acidobacterium capsulatum'\n",
       "26. 'Acidovorax avenae'\n",
       "27. 'Acidovorax carolinensis'\n",
       "28. 'Acidovorax cattleyae'\n",
       "29. 'Acidovorax citrulli'\n",
       "30. 'Acidovorax ebreus'\n",
       "31. 'Acidovorax sp. 1608163'\n",
       "32. 'Acidovorax sp. JS42'\n",
       "33. 'Acidovorax sp. KKS102'\n",
       "34. 'Acidovorax sp. RAC01'\n",
       "35. 'Acidovorax sp. T1'\n",
       "36. 'Acinetobacter baumannii'\n",
       "37. 'Acinetobacter bereziniae'\n",
       "38. 'Acinetobacter guillouiae'\n",
       "39. 'Acinetobacter haemolyticus'\n",
       "40. 'Acinetobacter indicus'\n",
       "41. 'Acinetobacter johnsonii'\n",
       "42. 'Acinetobacter junii'\n",
       "43. 'Acinetobacter lwoffii'\n",
       "44. 'Acinetobacter pittii'\n",
       "45. 'Acinetobacter radioresistens'\n",
       "46. 'Acinetobacter schindleri'\n",
       "47. 'Acinetobacter soli'\n",
       "48. 'Acinetobacter sp. ACNIH1'\n",
       "49. 'Acinetobacter sp. ACNIH2'\n",
       "50. 'Acinetobacter sp. LoGeW2-3'\n",
       "51. 'Acinetobacter sp. NCu2D-2'\n",
       "52. 'Acinetobacter sp. SWBY1'\n",
       "53. 'Acinetobacter sp. WCHA45'\n",
       "54. 'Acinetobacter sp. WCHA55'\n",
       "55. 'Acinetobacter sp. WCHAc010034'\n",
       "56. 'Acinetobacter ursingii'\n",
       "57. 'Acinetobacter venetianus'\n",
       "58. 'Acinetobacter wuhouensis'\n",
       "59. 'Actinoalloteichus hoggarensis'\n",
       "60. 'Actinoalloteichus hymeniacidonis'\n",
       "61. 'Actinoalloteichus sp. AHMU CJ021'\n",
       "62. 'Actinobacteria bacterium YIM 96077'\n",
       "63. 'Actinomadura amylolytica'\n",
       "64. 'Actinomyces hongkongensis'\n",
       "65. 'Actinomyces howellii'\n",
       "66. 'Actinomyces israelii'\n",
       "67. 'Actinomyces oris'\n",
       "68. 'Actinomyces radicidentis'\n",
       "69. 'Actinomyces slackii'\n",
       "70. 'Actinomyces sp. Chiba101'\n",
       "71. 'Actinomyces sp. oral taxon 414'\n",
       "72. 'Actinomyces sp. oral taxon 897'\n",
       "73. 'Actinomyces sp. Z16'\n",
       "74. 'Actinomyces viscosus'\n",
       "75. 'Actinoplanes derwentensis'\n",
       "76. 'Actinoplanes friuliensis'\n",
       "77. 'Actinoplanes missouriensis'\n",
       "78. 'Actinoplanes sp. N902-109'\n",
       "79. 'Actinoplanes teichomyceticus'\n",
       "80. 'Actinopolymorpha singaporensis'\n",
       "81. 'Actinopolyspora erythraea'\n",
       "82. 'Actinosynnema pretiosum'\n",
       "83. 'Adlercreutzia equolifaciens'\n",
       "84. 'Advenella kashmirensis'\n",
       "85. 'Advenella mimigardefordensis'\n",
       "86. 'Aeromicrobium choanae'\n",
       "87. 'Aeromicrobium erythreum'\n",
       "88. 'Aeromicrobium marinum'\n",
       "89. 'Aeromicrobium sp. 592'\n",
       "90. 'Aeromicrobium sp. A1-2'\n",
       "91. 'Aeromonas caviae'\n",
       "92. 'Aeromonas encheleia'\n",
       "93. 'Aeromonas hydrophila'\n",
       "94. 'Aeromonas media'\n",
       "95. 'Aeromonas salmonicida'\n",
       "96. 'Aeromonas schubertii'\n",
       "97. 'Aeromonas sp. ASNIH1'\n",
       "98. 'Aeromonas sp. ASNIH3'\n",
       "99. 'Aeromonas sp. CA23'\n",
       "100. 'Aeromonas veronii'\n",
       "101. 'Afipia sp. GAS231'\n",
       "102. 'Agrobacterium fabrum'\n",
       "103. 'Agrobacterium rhizogenes'\n",
       "104. 'Agrobacterium sp. H13-3'\n",
       "105. 'Agrobacterium sp. RAC06'\n",
       "106. 'Agrobacterium tumefaciens'\n",
       "107. 'Agrobacterium vitis'\n",
       "108. 'Agrococcus carbonis'\n",
       "109. 'Agrococcus jejuensis'\n",
       "110. 'Agromyces aureus'\n",
       "111. 'Agromyces flavus'\n",
       "112. 'Agromyces sp. 30A'\n",
       "113. 'Ahniella affigens'\n",
       "114. 'Alcaligenes aquatilis'\n",
       "115. 'Alcaligenes faecalis'\n",
       "116. 'Alcanivorax dieselolei'\n",
       "117. 'Alcanivorax pacificus'\n",
       "118. 'Alcanivorax sp. N3-2A'\n",
       "119. 'Alcanivorax xenomutans'\n",
       "120. 'Alicycliphilus denitrificans'\n",
       "121. 'Alicyclobacillus acidocaldarius'\n",
       "122. 'Alistipes shahii'\n",
       "123. 'Alistipes sp. Marseille-P5997'\n",
       "124. 'Alkalilimnicola ehrlichii'\n",
       "125. 'Alloactinosynnema sp. L-07'\n",
       "126. 'Allochromatium vinosum'\n",
       "127. 'Allokutzneria albata'\n",
       "128. 'Alphaproteobacteria bacterium WS11'\n",
       "129. 'Altererythrobacter atlanticus'\n",
       "130. 'Altererythrobacter dongtanensis'\n",
       "131. 'Altererythrobacter epoxidivorans'\n",
       "132. 'Altererythrobacter ishigakiensis'\n",
       "133. 'Altererythrobacter mangrovi'\n",
       "134. 'Altererythrobacter marensis'\n",
       "135. 'Altererythrobacter namhicola'\n",
       "136. 'Altererythrobacter sp. NS1'\n",
       "137. 'Altererythrobacter sp. ZODW24'\n",
       "138. 'Alteromonas macleodii'\n",
       "139. 'Alteromonas mediterranea'\n",
       "140. 'Alteromonas sp. MB-3u-76'\n",
       "141. 'Aminobacter aminovorans'\n",
       "142. 'Aminobacter sp. MSH1'\n",
       "143. 'Amycolatopsis albispora'\n",
       "144. 'Amycolatopsis japonica'\n",
       "145. 'Amycolatopsis keratiniphila'\n",
       "146. 'Amycolatopsis mediterranei'\n",
       "147. 'Amycolatopsis methanolica'\n",
       "148. 'Amycolatopsis orientalis'\n",
       "149. 'Amycolatopsis sp. AA4'\n",
       "150. 'Amycolatopsis sp. BJA-103'\n",
       "151. 'Anaeromyxobacter dehalogenans'\n",
       "152. 'Anaeromyxobacter sp. Fw109-5'\n",
       "153. 'Anaeromyxobacter sp. K'\n",
       "154. 'Anderseniella sp. Alg231-50'\n",
       "155. 'Antarctobacter heliothermus'\n",
       "156. 'Aquabacterium olei'\n",
       "157. 'Aquaspirillum sp. LM1'\n",
       "158. 'Aquitalea magnusonii'\n",
       "159. 'Archangium gephyra'\n",
       "160. 'Aromatoleum aromaticum'\n",
       "161. 'Arsenicicoccus sp. oral taxon 190'\n",
       "162. 'Arthrobacter alpinus'\n",
       "163. 'Arthrobacter crystallopoietes'\n",
       "164. 'Arthrobacter sp. DCT-5'\n",
       "165. 'Arthrobacter sp. ERGS1:01'\n",
       "166. 'Arthrobacter sp. FB24'\n",
       "167. 'Arthrobacter sp. PAMC 25486'\n",
       "168. 'Arthrobacter sp. PGP41'\n",
       "169. 'Arthrobacter sp. QXT-31'\n",
       "170. 'Arthrobacter sp. U41'\n",
       "171. 'Arthrobacter sp. YC-RL1'\n",
       "172. 'Arthrobacter sp. YN'\n",
       "173. 'Arthrobacter sp. ZXY-2'\n",
       "174. 'Asaia bogorensis'\n",
       "175. 'Asticcacaulis excentricus'\n",
       "176. 'Atlantibacter hermannii'\n",
       "177. 'Aurantimicrobium sp. MWH-Mo1'\n",
       "178. 'Auraticoccus monumenti'\n",
       "179. 'Aureimonas sp. AU20'\n",
       "180. 'Auritidibacter sp. NML130574'\n",
       "181. 'Austwickia chelonae'\n",
       "182. 'Azoarcus communis'\n",
       "183. 'Azoarcus olearius'\n",
       "184. 'Azoarcus sp. BH72'\n",
       "185. 'Azoarcus sp. CIB'\n",
       "186. 'Azoarcus sp. DN11'\n",
       "187. 'Azoarcus sp. KH32C'\n",
       "188. 'Azoarcus sp. SY39'\n",
       "189. 'Azorhizobium caulinodans'\n",
       "190. 'Azospira oryzae'\n",
       "191. 'Azospirillum brasilense'\n",
       "192. 'Azospirillum humicireducens'\n",
       "193. 'Azospirillum lipoferum'\n",
       "194. 'Azospirillum sp. CFH 70021'\n",
       "195. 'Azospirillum sp. M2T2B2'\n",
       "196. 'Azospirillum sp. TSH58'\n",
       "197. 'Azospirillum thiophilum'\n",
       "198. 'Azotobacter chroococcum'\n",
       "199. 'Azotobacter vinelandii'\n",
       "200. 'Bacillus cereus'\n",
       "201. ⋯\n",
       "202. 'Streptomyces griseochromogenes'\n",
       "203. 'Streptomyces griseorubiginosus'\n",
       "204. 'Streptomyces griseus'\n",
       "205. 'Streptomyces hundungensis'\n",
       "206. 'Streptomyces hygroscopicus'\n",
       "207. 'Streptomyces katrae'\n",
       "208. 'Streptomyces koyangensis'\n",
       "209. 'Streptomyces lavendulae'\n",
       "210. 'Streptomyces leeuwenhoekii'\n",
       "211. 'Streptomyces lincolnensis'\n",
       "212. 'Streptomyces lunaelactis'\n",
       "213. 'Streptomyces luteoverticillatus'\n",
       "214. 'Streptomyces lydicus'\n",
       "215. 'Streptomyces malaysiensis'\n",
       "216. 'Streptomyces nigra'\n",
       "217. 'Streptomyces niveus'\n",
       "218. 'Streptomyces nodosus'\n",
       "219. 'Streptomyces noursei'\n",
       "220. 'Streptomyces olivaceus'\n",
       "221. 'Streptomyces olivoreticuli'\n",
       "222. 'Streptomyces pactum'\n",
       "223. 'Streptomyces parvulus'\n",
       "224. 'Streptomyces peucetius'\n",
       "225. 'Streptomyces pluripotens'\n",
       "226. 'Streptomyces pratensis'\n",
       "227. 'Streptomyces pristinaespiralis'\n",
       "228. 'Streptomyces puniciscabiei'\n",
       "229. 'Streptomyces qaidamensis'\n",
       "230. 'Streptomyces reticuli'\n",
       "231. 'Streptomyces roseochromogenus'\n",
       "232. 'Streptomyces rubrolavendulae'\n",
       "233. 'Streptomyces scabiei'\n",
       "234. 'Streptomyces sp. 11-1-2'\n",
       "235. 'Streptomyces sp. 2323.1'\n",
       "236. 'Streptomyces sp. 3211'\n",
       "237. 'Streptomyces sp. 3214.6'\n",
       "238. 'Streptomyces sp. 4F'\n",
       "239. 'Streptomyces sp. 769'\n",
       "240. 'Streptomyces sp. ADI95-16'\n",
       "241. 'Streptomyces sp. CB09001'\n",
       "242. 'Streptomyces sp. CCM_MD2014'\n",
       "243. 'Streptomyces sp. CdTB01'\n",
       "244. 'Streptomyces sp. CFMR 7'\n",
       "245. 'Streptomyces sp. CMB-StM0423'\n",
       "246. 'Streptomyces sp. CNQ-509'\n",
       "247. 'Streptomyces sp. ETH9427'\n",
       "248. 'Streptomyces sp. fd1-xmd'\n",
       "249. 'Streptomyces sp. FR-008'\n",
       "250. 'Streptomyces sp. Go-475'\n",
       "251. 'Streptomyces sp. GSSD-12'\n",
       "252. 'Streptomyces sp. HNM0039'\n",
       "253. 'Streptomyces sp. KPB2'\n",
       "254. 'Streptomyces sp. M56'\n",
       "255. 'Streptomyces sp. Mg1'\n",
       "256. 'Streptomyces sp. MK45'\n",
       "257. 'Streptomyces sp. NEAU-S7GS2'\n",
       "258. 'Streptomyces sp. P3'\n",
       "259. 'Streptomyces sp. RTd22'\n",
       "260. 'Streptomyces sp. S8'\n",
       "261. 'Streptomyces sp. SAT1'\n",
       "262. 'Streptomyces sp. SCSIO 03032'\n",
       "263. 'Streptomyces sp. Sge12'\n",
       "264. 'Streptomyces sp. SirexAA-E'\n",
       "265. 'Streptomyces sp. SM18'\n",
       "266. 'Streptomyces sp. TLI_053'\n",
       "267. 'Streptomyces sp. TN58'\n",
       "268. 'Streptomyces sp. Tue 6075'\n",
       "269. 'Streptomyces sp. W1SF4'\n",
       "270. 'Streptomyces sp. WAC 01438'\n",
       "271. 'Streptomyces sp. WAC 01529'\n",
       "272. 'Streptomyces sp. WAC 06738'\n",
       "273. 'Streptomyces sp. WAC00288'\n",
       "274. 'Streptomyces sp. Z022'\n",
       "275. 'Streptomyces sp. ZFG47'\n",
       "276. 'Streptomyces spongiicola'\n",
       "277. 'Streptomyces venezuelae'\n",
       "278. 'Streptomyces vietnamensis'\n",
       "279. 'Streptomyces violaceoruber'\n",
       "280. 'Streptomyces violaceusniger'\n",
       "281. 'Streptomyces xiamenensis'\n",
       "282. 'Streptomyces xinghaiensis'\n",
       "283. 'Streptosporangium roseum'\n",
       "284. 'Streptosporangium sp. \\'caverna\\''\n",
       "285. 'Sulfitobacter pseudonitzschiae'\n",
       "286. 'Sulfitobacter sp. AM1-D1'\n",
       "287. 'Sulfitobacter sp. D7'\n",
       "288. 'Sulfitobacter sp. JL08'\n",
       "289. 'Sulfitobacter sp. SK011'\n",
       "290. 'Sulfitobacter sp. SK012'\n",
       "291. 'Sulfitobacter sp. SK025'\n",
       "292. 'Sulfuricaulis limicola'\n",
       "293. 'Sulfuricella denitrificans'\n",
       "294. 'Sulfuriferula sp. AH1'\n",
       "295. 'Sulfurifustis variabilis'\n",
       "296. 'Sulfuritalea hydrogenivorans'\n",
       "297. 'Sulfuritortus calidifontis'\n",
       "298. 'Sulfurospirillum multivorans'\n",
       "299. 'Sutterella megalosphaeroides'\n",
       "300. 'Symbiobacterium thermophilum'\n",
       "301. 'Synechococcus sp. KORDI-49'\n",
       "302. 'Synechococcus sp. PCC 7336'\n",
       "303. 'Synechococcus sp. RCC307'\n",
       "304. 'Synechococcus sp. SynAce01'\n",
       "305. 'Syntrophobotulus glycolicus'\n",
       "306. 'Tabrizicola sp. K13M18'\n",
       "307. 'Tateyamaria omphalii'\n",
       "308. 'Terriglobus roseus'\n",
       "309. 'Tessaracoccus aquimaris'\n",
       "310. 'Tessaracoccus flavescens'\n",
       "311. 'Tessaracoccus flavus'\n",
       "312. 'Tessaracoccus sp. Marseille-P5995'\n",
       "313. 'Tessaracoccus sp. T2.5-30'\n",
       "314. 'Thalassococcus sp. SH-1'\n",
       "315. 'Thalassospira marina'\n",
       "316. 'Thalassospira xiamenensis'\n",
       "317. 'Thauera aromatica'\n",
       "318. 'Thauera chlorobenzoica'\n",
       "319. 'Thauera humireducens'\n",
       "320. 'Thauera hydrothermalis'\n",
       "321. 'Thauera sp. K11'\n",
       "322. 'Thauera sp. MZ1T'\n",
       "323. 'Thermaerobacter marianensis'\n",
       "324. 'Thermoanaerobacterales bacterium SK-G1'\n",
       "325. 'Thermobacillus composti'\n",
       "326. 'Thermobifida fusca'\n",
       "327. 'Thermobispora bispora'\n",
       "328. 'Thermomonospora curvata'\n",
       "329. 'Thermus thermophilus'\n",
       "330. 'Thioalkalivibrio nitratireducens'\n",
       "331. 'Thioalkalivibrio paradoxus'\n",
       "332. 'Thioalkalivibrio sp. K90mix'\n",
       "333. 'Thioalkalivibrio sulfidiphilus'\n",
       "334. 'Thioalkalivibrio versutus'\n",
       "335. 'Thiobacillus denitrificans'\n",
       "336. 'Thioclava nitratireducens'\n",
       "337. 'Thiocystis violascens'\n",
       "338. 'Thioflavicoccus mobilis'\n",
       "339. 'Thiohalobacter thiocyanaticus'\n",
       "340. 'Thiolapillus brandeum'\n",
       "341. 'Thiomonas arsenitoxydans'\n",
       "342. 'Thiomonas intermedia'\n",
       "343. 'Thiomonas sp. X19'\n",
       "344. 'Tistrella mobilis'\n",
       "345. 'Tolumonas auensis'\n",
       "346. 'Torque teno virus 13'\n",
       "347. 'Torque teno virus 15'\n",
       "348. 'Torque teno virus 19'\n",
       "349. 'Torque teno virus 24'\n",
       "350. 'Torque teno virus 29'\n",
       "351. 'Truepera radiovictrix'\n",
       "352. 'Trueperella pyogenes'\n",
       "353. 'Tsukamurella paurometabola'\n",
       "354. 'Tsukamurella tyrosinosolvens'\n",
       "355. 'Undibacterium parvum'\n",
       "356. 'Variibacter gotjawalensis'\n",
       "357. 'Variovorax boronicumulans'\n",
       "358. 'Variovorax paradoxus'\n",
       "359. 'Variovorax sp. HW608'\n",
       "360. 'Variovorax sp. PAMC 28711'\n",
       "361. 'Variovorax sp. PMC12'\n",
       "362. 'Verminephrobacter eiseniae'\n",
       "363. 'Verrucomicrobia bacterium'\n",
       "364. 'Verrucomicrobia bacterium IMCC26134'\n",
       "365. 'Verrucomicrobium sp. GAS474'\n",
       "366. 'Verrucomicrobium spinosum'\n",
       "367. 'Verrucosispora maris'\n",
       "368. 'Vibrio fluvialis'\n",
       "369. 'Vibrio furnissii'\n",
       "370. 'Vibrio vulnificus'\n",
       "371. 'Victivallales bacterium CCUG 44730'\n",
       "372. 'Vitreoscilla filiformis'\n",
       "373. 'Vogesella sp. LIG4'\n",
       "374. 'Vulgatibacter incomptus'\n",
       "375. 'Wenzhouxiangella marina'\n",
       "376. 'Xanthobacter autotrophicus'\n",
       "377. 'Xanthomonas albilineans'\n",
       "378. 'Xanthomonas arboricola'\n",
       "379. 'Xanthomonas campestris'\n",
       "380. 'Xanthomonas cassavae'\n",
       "381. 'Xanthomonas citri'\n",
       "382. 'Xanthomonas euvesicatoria'\n",
       "383. 'Xanthomonas fragariae'\n",
       "384. 'Xanthomonas gardneri'\n",
       "385. 'Xanthomonas hortorum'\n",
       "386. 'Xanthomonas oryzae'\n",
       "387. 'Xanthomonas phaseoli'\n",
       "388. 'Xanthomonas sacchari'\n",
       "389. 'Xanthomonas translucens'\n",
       "390. 'Xanthomonas vesicatoria'\n",
       "391. 'Xylanimonas cellulosilytica'\n",
       "392. 'Xylella fastidiosa'\n",
       "393. 'Yangia pacifica'\n",
       "394. 'Yangia sp. CCB-MM3'\n",
       "395. 'Yersinia enterocolitica'\n",
       "396. 'Yoonia vestfoldensis'\n",
       "397. 'Zhihengliuella sp. ISTPL4'\n",
       "398. 'Zobellella denitrificans'\n",
       "399. 'Zoogloeaceae bacteirum Par-f-2'\n",
       "400. 'Zymobacter palmae'\n",
       "401. 'Zymomonas mobilis'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$non_contaminants\n",
       "[1] \"Altererythrobacter sp. B11\"  \"Bradyrhizobium erythrophlei\"\n",
       "[3] \"Bradyrhizobium sp. SK17\"     \"Cutibacterium acnes\"        \n",
       "[5] \"Rhodopseudomonas palustris\" \n",
       "\n",
       "$contaminants\n",
       "   [1] \"Acetobacter aceti\"                                     \n",
       "   [2] \"Acetobacter ghanensis\"                                 \n",
       "   [3] \"Acetobacter pasteurianus\"                              \n",
       "   [4] \"Acetobacter persici\"                                   \n",
       "   [5] \"Achromobacter denitrificans\"                           \n",
       "   [6] \"Achromobacter insolitus\"                               \n",
       "   [7] \"Achromobacter ruhlandii\"                               \n",
       "   [8] \"Achromobacter sp. AONIH1\"                              \n",
       "   [9] \"Achromobacter sp. B7\"                                  \n",
       "  [10] \"Achromobacter sp. MFA1 R4\"                             \n",
       "  [11] \"Achromobacter spanius\"                                 \n",
       "  [12] \"Achromobacter xylosoxidans\"                            \n",
       "  [13] \"Acidiferrobacter sp. SPIII_3\"                          \n",
       "  [14] \"Acidihalobacter ferrooxidans\"                          \n",
       "  [15] \"Acidihalobacter prosperus\"                             \n",
       "  [16] \"Acidiphilium cryptum\"                                  \n",
       "  [17] \"Acidiphilium multivorum\"                               \n",
       "  [18] \"Acidipropionibacterium acidipropionici\"                \n",
       "  [19] \"Acidipropionibacterium jensenii\"                       \n",
       "  [20] \"Acidipropionibacterium virtanenii\"                     \n",
       "  [21] \"Acidisphaera sp. G45-3\"                                \n",
       "  [22] \"Acidithiobacillus caldus\"                              \n",
       "  [23] \"Acidithiobacillus ferridurans\"                         \n",
       "  [24] \"Acidobacteriaceae bacterium SBC82\"                     \n",
       "  [25] \"Acidobacterium capsulatum\"                             \n",
       "  [26] \"Acidovorax avenae\"                                     \n",
       "  [27] \"Acidovorax carolinensis\"                               \n",
       "  [28] \"Acidovorax cattleyae\"                                  \n",
       "  [29] \"Acidovorax citrulli\"                                   \n",
       "  [30] \"Acidovorax ebreus\"                                     \n",
       "  [31] \"Acidovorax sp. 1608163\"                                \n",
       "  [32] \"Acidovorax sp. JS42\"                                   \n",
       "  [33] \"Acidovorax sp. KKS102\"                                 \n",
       "  [34] \"Acidovorax sp. RAC01\"                                  \n",
       "  [35] \"Acidovorax sp. T1\"                                     \n",
       "  [36] \"Acinetobacter baumannii\"                               \n",
       "  [37] \"Acinetobacter bereziniae\"                              \n",
       "  [38] \"Acinetobacter guillouiae\"                              \n",
       "  [39] \"Acinetobacter haemolyticus\"                            \n",
       "  [40] \"Acinetobacter indicus\"                                 \n",
       "  [41] \"Acinetobacter johnsonii\"                               \n",
       "  [42] \"Acinetobacter junii\"                                   \n",
       "  [43] \"Acinetobacter lwoffii\"                                 \n",
       "  [44] \"Acinetobacter pittii\"                                  \n",
       "  [45] \"Acinetobacter radioresistens\"                          \n",
       "  [46] \"Acinetobacter schindleri\"                              \n",
       "  [47] \"Acinetobacter soli\"                                    \n",
       "  [48] \"Acinetobacter sp. ACNIH1\"                              \n",
       "  [49] \"Acinetobacter sp. ACNIH2\"                              \n",
       "  [50] \"Acinetobacter sp. LoGeW2-3\"                            \n",
       "  [51] \"Acinetobacter sp. NCu2D-2\"                             \n",
       "  [52] \"Acinetobacter sp. SWBY1\"                               \n",
       "  [53] \"Acinetobacter sp. WCHA45\"                              \n",
       "  [54] \"Acinetobacter sp. WCHA55\"                              \n",
       "  [55] \"Acinetobacter sp. WCHAc010034\"                         \n",
       "  [56] \"Acinetobacter ursingii\"                                \n",
       "  [57] \"Acinetobacter venetianus\"                              \n",
       "  [58] \"Acinetobacter wuhouensis\"                              \n",
       "  [59] \"Actinoalloteichus hoggarensis\"                         \n",
       "  [60] \"Actinoalloteichus hymeniacidonis\"                      \n",
       "  [61] \"Actinoalloteichus sp. AHMU CJ021\"                      \n",
       "  [62] \"Actinobacteria bacterium YIM 96077\"                    \n",
       "  [63] \"Actinomadura amylolytica\"                              \n",
       "  [64] \"Actinomyces hongkongensis\"                             \n",
       "  [65] \"Actinomyces howellii\"                                  \n",
       "  [66] \"Actinomyces israelii\"                                  \n",
       "  [67] \"Actinomyces oris\"                                      \n",
       "  [68] \"Actinomyces radicidentis\"                              \n",
       "  [69] \"Actinomyces slackii\"                                   \n",
       "  [70] \"Actinomyces sp. Chiba101\"                              \n",
       "  [71] \"Actinomyces sp. oral taxon 414\"                        \n",
       "  [72] \"Actinomyces sp. oral taxon 897\"                        \n",
       "  [73] \"Actinomyces sp. Z16\"                                   \n",
       "  [74] \"Actinomyces viscosus\"                                  \n",
       "  [75] \"Actinoplanes derwentensis\"                             \n",
       "  [76] \"Actinoplanes friuliensis\"                              \n",
       "  [77] \"Actinoplanes missouriensis\"                            \n",
       "  [78] \"Actinoplanes sp. N902-109\"                             \n",
       "  [79] \"Actinoplanes teichomyceticus\"                          \n",
       "  [80] \"Actinopolymorpha singaporensis\"                        \n",
       "  [81] \"Actinopolyspora erythraea\"                             \n",
       "  [82] \"Actinosynnema pretiosum\"                               \n",
       "  [83] \"Adlercreutzia equolifaciens\"                           \n",
       "  [84] \"Advenella kashmirensis\"                                \n",
       "  [85] \"Advenella mimigardefordensis\"                          \n",
       "  [86] \"Aeromicrobium choanae\"                                 \n",
       "  [87] \"Aeromicrobium erythreum\"                               \n",
       "  [88] \"Aeromicrobium marinum\"                                 \n",
       "  [89] \"Aeromicrobium sp. 592\"                                 \n",
       "  [90] \"Aeromicrobium sp. A1-2\"                                \n",
       "  [91] \"Aeromonas caviae\"                                      \n",
       "  [92] \"Aeromonas encheleia\"                                   \n",
       "  [93] \"Aeromonas hydrophila\"                                  \n",
       "  [94] \"Aeromonas media\"                                       \n",
       "  [95] \"Aeromonas salmonicida\"                                 \n",
       "  [96] \"Aeromonas schubertii\"                                  \n",
       "  [97] \"Aeromonas sp. ASNIH1\"                                  \n",
       "  [98] \"Aeromonas sp. ASNIH3\"                                  \n",
       "  [99] \"Aeromonas sp. CA23\"                                    \n",
       " [100] \"Aeromonas veronii\"                                     \n",
       " [101] \"Afipia sp. GAS231\"                                     \n",
       " [102] \"Agrobacterium fabrum\"                                  \n",
       " [103] \"Agrobacterium rhizogenes\"                              \n",
       " [104] \"Agrobacterium sp. H13-3\"                               \n",
       " [105] \"Agrobacterium sp. RAC06\"                               \n",
       " [106] \"Agrobacterium tumefaciens\"                             \n",
       " [107] \"Agrobacterium vitis\"                                   \n",
       " [108] \"Agrococcus carbonis\"                                   \n",
       " [109] \"Agrococcus jejuensis\"                                  \n",
       " [110] \"Agromyces aureus\"                                      \n",
       " [111] \"Agromyces flavus\"                                      \n",
       " [112] \"Agromyces sp. 30A\"                                     \n",
       " [113] \"Ahniella affigens\"                                     \n",
       " [114] \"Alcaligenes aquatilis\"                                 \n",
       " [115] \"Alcaligenes faecalis\"                                  \n",
       " [116] \"Alcanivorax dieselolei\"                                \n",
       " [117] \"Alcanivorax pacificus\"                                 \n",
       " [118] \"Alcanivorax sp. N3-2A\"                                 \n",
       " [119] \"Alcanivorax xenomutans\"                                \n",
       " [120] \"Alicycliphilus denitrificans\"                          \n",
       " [121] \"Alicyclobacillus acidocaldarius\"                       \n",
       " [122] \"Alistipes shahii\"                                      \n",
       " [123] \"Alistipes sp. Marseille-P5997\"                         \n",
       " [124] \"Alkalilimnicola ehrlichii\"                             \n",
       " [125] \"Alloactinosynnema sp. L-07\"                            \n",
       " [126] \"Allochromatium vinosum\"                                \n",
       " [127] \"Allokutzneria albata\"                                  \n",
       " [128] \"Alphaproteobacteria bacterium WS11\"                    \n",
       " [129] \"Altererythrobacter atlanticus\"                         \n",
       " [130] \"Altererythrobacter dongtanensis\"                       \n",
       " [131] \"Altererythrobacter epoxidivorans\"                      \n",
       " [132] \"Altererythrobacter ishigakiensis\"                      \n",
       " [133] \"Altererythrobacter mangrovi\"                           \n",
       " [134] \"Altererythrobacter marensis\"                           \n",
       " [135] \"Altererythrobacter namhicola\"                          \n",
       " [136] \"Altererythrobacter sp. NS1\"                            \n",
       " [137] \"Altererythrobacter sp. ZODW24\"                         \n",
       " [138] \"Alteromonas macleodii\"                                 \n",
       " [139] \"Alteromonas mediterranea\"                              \n",
       " [140] \"Alteromonas sp. MB-3u-76\"                              \n",
       " [141] \"Aminobacter aminovorans\"                               \n",
       " [142] \"Aminobacter sp. MSH1\"                                  \n",
       " [143] \"Amycolatopsis albispora\"                               \n",
       " [144] \"Amycolatopsis japonica\"                                \n",
       " [145] \"Amycolatopsis keratiniphila\"                           \n",
       " [146] \"Amycolatopsis mediterranei\"                            \n",
       " [147] \"Amycolatopsis methanolica\"                             \n",
       " [148] \"Amycolatopsis orientalis\"                              \n",
       " [149] \"Amycolatopsis sp. AA4\"                                 \n",
       " [150] \"Amycolatopsis sp. BJA-103\"                             \n",
       " [151] \"Anaeromyxobacter dehalogenans\"                         \n",
       " [152] \"Anaeromyxobacter sp. Fw109-5\"                          \n",
       " [153] \"Anaeromyxobacter sp. K\"                                \n",
       " [154] \"Anderseniella sp. Alg231-50\"                           \n",
       " [155] \"Antarctobacter heliothermus\"                           \n",
       " [156] \"Aquabacterium olei\"                                    \n",
       " [157] \"Aquaspirillum sp. LM1\"                                 \n",
       " [158] \"Aquitalea magnusonii\"                                  \n",
       " [159] \"Archangium gephyra\"                                    \n",
       " [160] \"Aromatoleum aromaticum\"                                \n",
       " [161] \"Arsenicicoccus sp. oral taxon 190\"                     \n",
       " [162] \"Arthrobacter alpinus\"                                  \n",
       " [163] \"Arthrobacter crystallopoietes\"                         \n",
       " [164] \"Arthrobacter sp. DCT-5\"                                \n",
       " [165] \"Arthrobacter sp. ERGS1:01\"                             \n",
       " [166] \"Arthrobacter sp. FB24\"                                 \n",
       " [167] \"Arthrobacter sp. PAMC 25486\"                           \n",
       " [168] \"Arthrobacter sp. PGP41\"                                \n",
       " [169] \"Arthrobacter sp. QXT-31\"                               \n",
       " [170] \"Arthrobacter sp. U41\"                                  \n",
       " [171] \"Arthrobacter sp. YC-RL1\"                               \n",
       " [172] \"Arthrobacter sp. YN\"                                   \n",
       " [173] \"Arthrobacter sp. ZXY-2\"                                \n",
       " [174] \"Asaia bogorensis\"                                      \n",
       " [175] \"Asticcacaulis excentricus\"                             \n",
       " [176] \"Atlantibacter hermannii\"                               \n",
       " [177] \"Aurantimicrobium sp. MWH-Mo1\"                          \n",
       " [178] \"Auraticoccus monumenti\"                                \n",
       " [179] \"Aureimonas sp. AU20\"                                   \n",
       " [180] \"Auritidibacter sp. NML130574\"                          \n",
       " [181] \"Austwickia chelonae\"                                   \n",
       " [182] \"Azoarcus communis\"                                     \n",
       " [183] \"Azoarcus olearius\"                                     \n",
       " [184] \"Azoarcus sp. BH72\"                                     \n",
       " [185] \"Azoarcus sp. CIB\"                                      \n",
       " [186] \"Azoarcus sp. DN11\"                                     \n",
       " [187] \"Azoarcus sp. KH32C\"                                    \n",
       " [188] \"Azoarcus sp. SY39\"                                     \n",
       " [189] \"Azorhizobium caulinodans\"                              \n",
       " [190] \"Azospira oryzae\"                                       \n",
       " [191] \"Azospirillum brasilense\"                               \n",
       " [192] \"Azospirillum humicireducens\"                           \n",
       " [193] \"Azospirillum lipoferum\"                                \n",
       " [194] \"Azospirillum sp. CFH 70021\"                            \n",
       " [195] \"Azospirillum sp. M2T2B2\"                               \n",
       " [196] \"Azospirillum sp. TSH58\"                                \n",
       " [197] \"Azospirillum thiophilum\"                               \n",
       " [198] \"Azotobacter chroococcum\"                               \n",
       " [199] \"Azotobacter vinelandii\"                                \n",
       " [200] \"Bacillus cereus\"                                       \n",
       " [201] \"Bacillus coagulans\"                                    \n",
       " [202] \"Bacillus thuringiensis\"                                \n",
       " [203] \"Bacteroides fragilis\"                                  \n",
       " [204] \"Bacteroides zoogleoformans\"                            \n",
       " [205] \"Bartonella apis\"                                       \n",
       " [206] \"Beijerinckia indica\"                                   \n",
       " [207] \"Betaproteobacteria bacterium GR16-43\"                  \n",
       " [208] \"Beutenbergia cavernae\"                                 \n",
       " [209] \"Bifidobacterium adolescentis\"                          \n",
       " [210] \"Bifidobacterium animalis\"                              \n",
       " [211] \"Bifidobacterium bifidum\"                               \n",
       " [212] \"Bifidobacterium breve\"                                 \n",
       " [213] \"Bifidobacterium choerinum\"                             \n",
       " [214] \"Bifidobacterium dentium\"                               \n",
       " [215] \"Bifidobacterium longum\"                                \n",
       " [216] \"Bifidobacterium pseudolongum\"                          \n",
       " [217] \"Bifidobacterium scardovii\"                             \n",
       " [218] \"Blastochloris sp. GI\"                                  \n",
       " [219] \"Blastochloris viridis\"                                 \n",
       " [220] \"Blastococcus saxobsidens\"                              \n",
       " [221] \"Blastomonas fulva\"                                     \n",
       " [222] \"Blastomonas sp. RAC04\"                                 \n",
       " [223] \"blood disease bacterium A2-HR MARDI\"                   \n",
       " [224] \"Bordetella avium\"                                      \n",
       " [225] \"Bordetella bronchialis\"                                \n",
       " [226] \"Bordetella bronchiseptica\"                             \n",
       " [227] \"Bordetella flabilis\"                                   \n",
       " [228] \"Bordetella genomosp. 13\"                               \n",
       " [229] \"Bordetella genomosp. 6\"                                \n",
       " [230] \"Bordetella genomosp. 8\"                                \n",
       " [231] \"Bordetella genomosp. 9\"                                \n",
       " [232] \"Bordetella hinzii\"                                     \n",
       " [233] \"Bordetella holmesii\"                                   \n",
       " [234] \"Bordetella parapertussis\"                              \n",
       " [235] \"Bordetella petrii\"                                     \n",
       " [236] \"Bordetella pseudohinzii\"                               \n",
       " [237] \"Bordetella sp. H567\"                                   \n",
       " [238] \"Bordetella sp. HZ20\"                                   \n",
       " [239] \"Bordetella sp. N\"                                      \n",
       " [240] \"Bordetella trematum\"                                   \n",
       " [241] \"Bosea sp. AS-1\"                                        \n",
       " [242] \"Bosea sp. PAMC 26642\"                                  \n",
       " [243] \"Bosea sp. RAC05\"                                       \n",
       " [244] \"Bosea sp. Tri-49\"                                      \n",
       " [245] \"Bosea vaviloviae\"                                      \n",
       " [246] \"Brachybacterium faecium\"                               \n",
       " [247] \"Brachybacterium ginsengisoli\"                          \n",
       " [248] \"Brachybacterium saurashtrense\"                         \n",
       " [249] \"Brachybacterium sp. P6-10-X1\"                          \n",
       " [250] \"Brachybacterium sp. VM2412\"                            \n",
       " [251] \"Brachybacterium sp. VR2415\"                            \n",
       " [252] \"Bradymonas sediminis\"                                  \n",
       " [253] \"Bradyrhizobiaceae bacterium SG-6C\"                     \n",
       " [254] \"Bradyrhizobium diazoefficiens\"                         \n",
       " [255] \"Bradyrhizobium icense\"                                 \n",
       " [256] \"Bradyrhizobium japonicum\"                              \n",
       " [257] \"Bradyrhizobium lablabi\"                                \n",
       " [258] \"Bradyrhizobium oligotrophicum\"                         \n",
       " [259] \"Bradyrhizobium ottawaense\"                             \n",
       " [260] \"Bradyrhizobium sp.\"                                    \n",
       " [261] \"Bradyrhizobium sp. 2 39S1MB\"                           \n",
       " [262] \"Bradyrhizobium sp. 3 85S1MB\"                           \n",
       " [263] \"Bradyrhizobium sp. BTAi1\"                              \n",
       " [264] \"Bradyrhizobium sp. CCGE-LA001\"                         \n",
       " [265] \"Bradyrhizobium sp. ORS 278\"                            \n",
       " [266] \"Bradyrhizobium sp. ORS 285\"                            \n",
       " [267] \"Bradyrhizobium sp. ORS 3257\"                           \n",
       " [268] \"Bradyrhizobium sp. S23321\"                             \n",
       " [269] \"Bradyrhizobium sp. WSM471\"                             \n",
       " [270] \"Brenneria goodwinii\"                                   \n",
       " [271] \"Brenneria sp. EniD312\"                                 \n",
       " [272] \"Breoghania sp. L-A4\"                                   \n",
       " [273] \"Brevibacillus sp. SCSIO 07484\"                         \n",
       " [274] \"Brevibacterium aurantiacum\"                            \n",
       " [275] \"Brevibacterium linens\"                                 \n",
       " [276] \"Brevibacterium sandarakinum\"                           \n",
       " [277] \"Brevibacterium siliguriense\"                           \n",
       " [278] \"Brevirhabdus pacifica\"                                 \n",
       " [279] \"Brevundimonas diminuta\"                                \n",
       " [280] \"Brevundimonas naejangsanensis\"                         \n",
       " [281] \"Brevundimonas sp. DS20\"                                \n",
       " [282] \"Brevundimonas sp. GW460-12-10-14-LB2\"                  \n",
       " [283] \"Brevundimonas sp. LM2\"                                 \n",
       " [284] \"Brevundimonas subvibrioides\"                           \n",
       " [285] \"Brevundimonas vesicularis\"                             \n",
       " [286] \"Burkholderia ambifaria\"                                \n",
       " [287] \"Burkholderia anthina\"                                  \n",
       " [288] \"Burkholderia cenocepacia\"                              \n",
       " [289] \"Burkholderia cepacia\"                                  \n",
       " [290] \"Burkholderia contaminans\"                              \n",
       " [291] \"Burkholderia diffusa\"                                  \n",
       " [292] \"Burkholderia dolosa\"                                   \n",
       " [293] \"Burkholderia gladioli\"                                 \n",
       " [294] \"Burkholderia glumae\"                                   \n",
       " [295] \"Burkholderia insecticola\"                              \n",
       " [296] \"Burkholderia lata\"                                     \n",
       " [297] \"Burkholderia latens\"                                   \n",
       " [298] \"Burkholderia metallica\"                                \n",
       " [299] \"Burkholderia multivorans\"                              \n",
       " [300] \"Burkholderia oklahomensis\"                             \n",
       " [301] \"Burkholderia plantarii\"                                \n",
       " [302] \"Burkholderia pseudomallei\"                             \n",
       " [303] \"Burkholderia pseudomultivorans\"                        \n",
       " [304] \"Burkholderia pyrrocinia\"                               \n",
       " [305] \"Burkholderia seminalis\"                                \n",
       " [306] \"Burkholderia sp. BDU6\"                                 \n",
       " [307] \"Burkholderia sp. BDU8\"                                 \n",
       " [308] \"Burkholderia sp. Bp7605\"                               \n",
       " [309] \"Burkholderia sp. CCGE1001\"                             \n",
       " [310] \"Burkholderia sp. CCGE1002\"                             \n",
       " [311] \"Burkholderia sp. CCGE1003\"                             \n",
       " [312] \"Burkholderia sp. HB1\"                                  \n",
       " [313] \"Burkholderia sp. IDO3\"                                 \n",
       " [314] \"Burkholderia sp. JP2-270\"                              \n",
       " [315] \"Burkholderia sp. KK1\"                                  \n",
       " [316] \"Burkholderia sp. LA-2-3-30-S1-D2\"                      \n",
       " [317] \"Burkholderia sp. MSMB0856\"                             \n",
       " [318] \"Burkholderia sp. NRF60-BP8\"                            \n",
       " [319] \"Burkholderia sp. OLGA172\"                              \n",
       " [320] \"Burkholderia sp. PAMC 26561\"                           \n",
       " [321] \"Burkholderia sp. PAMC 28687\"                           \n",
       " [322] \"Burkholderia sp. RPE67\"                                \n",
       " [323] \"Burkholderia sp. YI23\"                                 \n",
       " [324] \"Burkholderia stabilis\"                                 \n",
       " [325] \"Burkholderia stagnalis\"                                \n",
       " [326] \"Burkholderia territorii\"                               \n",
       " [327] \"Burkholderia thailandensis\"                            \n",
       " [328] \"Burkholderia ubonensis\"                                \n",
       " [329] \"Burkholderia vietnamiensis\"                            \n",
       " [330] \"Burkholderiales bacterium GJ-E10\"                      \n",
       " [331] \"Burkholderiales bacterium JOSHI_001\"                   \n",
       " [332] \"Caldilinea aerophila\"                                  \n",
       " [333] \"Candidatus Accumulibacter phosphatis\"                  \n",
       " [334] \"Candidatus Desulforudis audaxviator\"                   \n",
       " [335] \"Candidatus Filomicrobium marinum\"                      \n",
       " [336] \"Candidatus Nitrospira inopinata\"                       \n",
       " [337] \"Candidatus Nucleicultrix amoebiphila\"                  \n",
       " [338] \"Candidatus Phaeomarinobacter ectocarpi\"                \n",
       " [339] \"Candidatus Promineofilum breve\"                        \n",
       " [340] \"Candidatus Solibacter usitatus\"                        \n",
       " [341] \"Candidatus Symbiobacter mobilis\"                       \n",
       " [342] \"Candidatus Thiodictyon syntrophicum\"                   \n",
       " [343] \"Candidatus Xiphinematobacter sp. Idaho Grape\"          \n",
       " [344] \"Cardiobacterium hominis\"                               \n",
       " [345] \"Castellaniella defragrans\"                             \n",
       " [346] \"Catenulispora acidiphila\"                              \n",
       " [347] \"Caulobacter flavus\"                                    \n",
       " [348] \"Caulobacter henricii\"                                  \n",
       " [349] \"Caulobacter mirabilis\"                                 \n",
       " [350] \"Caulobacter segnis\"                                    \n",
       " [351] \"Caulobacter sp. K31\"                                   \n",
       " [352] \"Caulobacter vibrioides\"                                \n",
       " [353] \"Caulobacteraceae bacterium OTSz_A_272\"                 \n",
       " [354] \"Cedecea neteri\"                                        \n",
       " [355] \"Celeribacter baekdonensis\"                             \n",
       " [356] \"Celeribacter ethanolicus\"                              \n",
       " [357] \"Celeribacter indicus\"                                  \n",
       " [358] \"Celeribacter manganoxidans\"                            \n",
       " [359] \"Celeribacter marinus\"                                  \n",
       " [360] \"Cellulomonas fimi\"                                     \n",
       " [361] \"Cellulomonas flavigena\"                                \n",
       " [362] \"Cellulomonas gilvus\"                                   \n",
       " [363] \"Cellulomonas sp. PSBB021\"                              \n",
       " [364] \"Cellulosimicrobium cellulans\"                          \n",
       " [365] \"Cellulosimicrobium sp. TH-20\"                          \n",
       " [366] \"Cellvibrio japonicus\"                                  \n",
       " [367] \"Cellvibrio sp. PSBB006\"                                \n",
       " [368] \"Chelativorans sp. BNC1\"                                \n",
       " [369] \"Chelatococcus daeguensis\"                              \n",
       " [370] \"Chelatococcus sp. CO-6\"                                \n",
       " [371] \"Chloracidobacterium thermophilum\"                      \n",
       " [372] \"Chlorobaculum limnaeum\"                                \n",
       " [373] \"Chlorobium phaeobacteroides\"                           \n",
       " [374] \"Chloroflexus aggregans\"                                \n",
       " [375] \"Chondromyces crocatus\"                                 \n",
       " [376] \"Chromatiaceae bacterium 2141T.STBD.0c.01a\"             \n",
       " [377] \"Chromobacterium rhizoryzae\"                            \n",
       " [378] \"Chromobacterium sp. ATCC 53434\"                        \n",
       " [379] \"Chromobacterium vaccinii\"                              \n",
       " [380] \"Chromobacterium violaceum\"                             \n",
       " [381] \"Chromohalobacter salexigens\"                           \n",
       " [382] \"Chryseobacterium shandongense\"                         \n",
       " [383] \"Chryseobacterium sp. F5649\"                            \n",
       " [384] \"Chryseobacterium sp. H6466\"                            \n",
       " [385] \"Chryseobacterium taklimakanense\"                       \n",
       " [386] \"Citrobacter amalonaticus\"                              \n",
       " [387] \"Citrobacter freundii\"                                  \n",
       " [388] \"Citrobacter koseri\"                                    \n",
       " [389] \"Citrobacter rodentium\"                                 \n",
       " [390] \"Citrobacter werkmanii\"                                 \n",
       " [391] \"Citromicrobium sp. JL477\"                              \n",
       " [392] \"Clavibacter michiganensis\"                             \n",
       " [393] \"Cloacibacterium normanense\"                            \n",
       " [394] \"Clostridium botulinum\"                                 \n",
       " [395] \"Cnuibacter physcomitrellae\"                            \n",
       " [396] \"Cobetia marina\"                                        \n",
       " [397] \"Cohaesibacter sp. ES.047\"                              \n",
       " [398] \"Collimonas arenae\"                                     \n",
       " [399] \"Collimonas fungivorans\"                                \n",
       " [400] \"Collimonas pratensis\"                                  \n",
       " [401] \"Collinsella aerofaciens\"                               \n",
       " [402] \"Comamonas aquatica\"                                    \n",
       " [403] \"Comamonas kerstersii\"                                  \n",
       " [404] \"Comamonas serinivorans\"                                \n",
       " [405] \"Comamonas testosteroni\"                                \n",
       " [406] \"Comamonas thiooxydans\"                                 \n",
       " [407] \"Conexibacter woesei\"                                   \n",
       " [408] \"Confluentimicrobium sp. EMB200-NS6\"                    \n",
       " [409] \"Corallococcus coralloides\"                             \n",
       " [410] \"Coriobacterium glomerans\"                              \n",
       " [411] \"Corynebacterium ammoniagenes\"                          \n",
       " [412] \"Corynebacterium atypicum\"                              \n",
       " [413] \"Corynebacterium aurimucosum\"                           \n",
       " [414] \"Corynebacterium cystitidis\"                            \n",
       " [415] \"Corynebacterium doosanense\"                            \n",
       " [416] \"Corynebacterium efficiens\"                             \n",
       " [417] \"Corynebacterium epidermidicanis\"                       \n",
       " [418] \"Corynebacterium falsenii\"                              \n",
       " [419] \"Corynebacterium frankenforstense\"                      \n",
       " [420] \"Corynebacterium genitalium\"                            \n",
       " [421] \"Corynebacterium glaucum\"                               \n",
       " [422] \"Corynebacterium glutamicum\"                            \n",
       " [423] \"Corynebacterium glyciniphilum\"                         \n",
       " [424] \"Corynebacterium halotolerans\"                          \n",
       " [425] \"Corynebacterium humireducens\"                          \n",
       " [426] \"Corynebacterium imitans\"                               \n",
       " [427] \"Corynebacterium jeikeium\"                              \n",
       " [428] \"Corynebacterium marinum\"                               \n",
       " [429] \"Corynebacterium maris\"                                 \n",
       " [430] \"Corynebacterium matruchotii\"                           \n",
       " [431] \"Corynebacterium mycetoides\"                            \n",
       " [432] \"Corynebacterium provencense\"                           \n",
       " [433] \"Corynebacterium renale\"                                \n",
       " [434] \"Corynebacterium resistens\"                             \n",
       " [435] \"Corynebacterium segmentosum\"                           \n",
       " [436] \"Corynebacterium simulans\"                              \n",
       " [437] \"Corynebacterium singulare\"                             \n",
       " [438] \"Corynebacterium sp. 2183\"                              \n",
       " [439] \"Corynebacterium sp. 2184\"                              \n",
       " [440] \"Corynebacterium sphenisci\"                             \n",
       " [441] \"Corynebacterium striatum\"                              \n",
       " [442] \"Corynebacterium terpenotabidum\"                        \n",
       " [443] \"Corynebacterium testudinoris\"                          \n",
       " [444] \"Corynebacterium timonense\"                             \n",
       " [445] \"Corynebacterium urealyticum\"                           \n",
       " [446] \"Corynebacterium ureicelerivorans\"                      \n",
       " [447] \"Corynebacterium uterequi\"                              \n",
       " [448] \"Corynebacterium variabile\"                             \n",
       " [449] \"Corynebacterium vitaeruminis\"                          \n",
       " [450] \"Corynebacterium xerosis\"                               \n",
       " [451] \"Coxiella burnetii\"                                     \n",
       " [452] \"Crenobacter sp. K1W11S-77\"                             \n",
       " [453] \"Croceicoccus marinus\"                                  \n",
       " [454] \"Croceicoccus naphthovorans\"                            \n",
       " [455] \"Cronobacter dublinensis\"                               \n",
       " [456] \"Cronobacter turicensis\"                                \n",
       " [457] \"Cryobacterium arcticum\"                                \n",
       " [458] \"Cryobacterium sp. GCJ02\"                               \n",
       " [459] \"Cryobacterium sp. LW097\"                               \n",
       " [460] \"Cupriavidus basilensis\"                                \n",
       " [461] \"Cupriavidus gilardii\"                                  \n",
       " [462] \"Cupriavidus metallidurans\"                             \n",
       " [463] \"Cupriavidus nantongensis\"                              \n",
       " [464] \"Cupriavidus necator\"                                   \n",
       " [465] \"Cupriavidus oxalaticus\"                                \n",
       " [466] \"Cupriavidus pauculus\"                                  \n",
       " [467] \"Cupriavidus pinatubonensis\"                            \n",
       " [468] \"Cupriavidus sp. USMAA2-4\"                              \n",
       " [469] \"Cupriavidus sp. USMAHM13\"                              \n",
       " [470] \"Cupriavidus taiwanensis\"                               \n",
       " [471] \"Curtobacterium pusillum\"                               \n",
       " [472] \"Curtobacterium sp. BH-2-1-1\"                           \n",
       " [473] \"Curtobacterium sp. MR_MD2014\"                          \n",
       " [474] \"Curvibacter sp. AEP1-3\"                                \n",
       " [475] \"Cutibacterium avidum\"                                  \n",
       " [476] \"Cutibacterium granulosum\"                              \n",
       " [477] \"Cyanobium gracile\"                                     \n",
       " [478] \"Cyanobium sp. NIES-981\"                                \n",
       " [479] \"Cystobacter fuscus\"                                    \n",
       " [480] \"Dechloromonas aromatica\"                               \n",
       " [481] \"Dechloromonas sp. HYN0024\"                             \n",
       " [482] \"Defluviimonas alba\"                                    \n",
       " [483] \"Deinococcus actinosclerus\"                             \n",
       " [484] \"Deinococcus ficus\"                                     \n",
       " [485] \"Deinococcus geothermalis\"                              \n",
       " [486] \"Deinococcus gobiensis\"                                 \n",
       " [487] \"Deinococcus irradiatisoli\"                             \n",
       " [488] \"Deinococcus maricopensis\"                              \n",
       " [489] \"Deinococcus peraridilitoris\"                           \n",
       " [490] \"Deinococcus proteolyticus\"                             \n",
       " [491] \"Deinococcus radiodurans\"                               \n",
       " [492] \"Deinococcus soli Cha et al. 2016\"                      \n",
       " [493] \"Deinococcus sp. NW-56\"                                 \n",
       " [494] \"Deinococcus sp. S14-83\"                                \n",
       " [495] \"Deinococcus swuensis\"                                  \n",
       " [496] \"Deinococcus wulumuqiensis\"                             \n",
       " [497] \"Delftia acidovorans\"                                   \n",
       " [498] \"Delftia sp. Cs1-4\"                                     \n",
       " [499] \"Delftia sp. HK171\"                                     \n",
       " [500] \"Delftia tsuruhatensis\"                                 \n",
       " [501] \"Dermabacter vaginalis\"                                 \n",
       " [502] \"Dermacoccus nishinomiyaensis\"                          \n",
       " [503] \"Desulfarculus baarsii\"                                 \n",
       " [504] \"Desulfomicrobium baculatum\"                            \n",
       " [505] \"Desulfovibrio africanus\"                               \n",
       " [506] \"Desulfovibrio desulfuricans\"                           \n",
       " [507] \"Desulfovibrio fairfieldensis\"                          \n",
       " [508] \"Desulfovibrio ferrophilus\"                             \n",
       " [509] \"Desulfovibrio gigas\"                                   \n",
       " [510] \"Desulfovibrio magneticus\"                              \n",
       " [511] \"Desulfovibrio piger\"                                   \n",
       " [512] \"Desulfovibrio sp. FW1012B\"                             \n",
       " [513] \"Desulfovibrio vulgaris\"                                \n",
       " [514] \"Desulfurivibrio alkaliphilus\"                          \n",
       " [515] \"Desulfuromonas soudanensis\"                            \n",
       " [516] \"Desulfuromonas sp. DDH964\"                             \n",
       " [517] \"Devosia sp. A16\"                                       \n",
       " [518] \"Devosia sp. H5989\"                                     \n",
       " [519] \"Devosia sp. I507\"                                      \n",
       " [520] \"Diaphorobacter polyhydroxybutyrativorans\"              \n",
       " [521] \"Dickeya chrysanthemi\"                                  \n",
       " [522] \"Dickeya dadantii\"                                      \n",
       " [523] \"Dickeya dianthicola\"                                   \n",
       " [524] \"Dickeya sp. Secpp 1600\"                                \n",
       " [525] \"Dickeya zeae\"                                          \n",
       " [526] \"Dietzia lutea\"                                         \n",
       " [527] \"Dietzia psychralcaliphila\"                             \n",
       " [528] \"Dietzia sp. JS16-p6b\"                                  \n",
       " [529] \"Dietzia sp. oral taxon 368\"                            \n",
       " [530] \"Dietzia timorensis\"                                    \n",
       " [531] \"Dinoroseobacter shibae\"                                \n",
       " [532] \"Dokdonella koreensis\"                                  \n",
       " [533] \"Dyella japonica\"                                       \n",
       " [534] \"Dyella jiangningensis\"                                 \n",
       " [535] \"Dyella thiooxydans\"                                    \n",
       " [536] \"Ectothiorhodospira sp. BSL-9\"                          \n",
       " [537] \"Edwardsiella hoshinae\"                                 \n",
       " [538] \"Eggerthella lenta\"                                     \n",
       " [539] \"endosymbiont of unidentified scaly snail isolate Monju\"\n",
       " [540] \"Ensifer adhaerens\"                                     \n",
       " [541] \"Ensifer sojae\"                                         \n",
       " [542] \"Enterobacter asburiae\"                                 \n",
       " [543] \"Enterobacter cloacae\"                                  \n",
       " [544] \"Enterobacter hormaechei\"                               \n",
       " [545] \"Enterobacter lignolyticus\"                             \n",
       " [546] \"Enterobacter sp. FY-07\"                                \n",
       " [547] \"Enterobacter sp. SA187\"                                \n",
       " [548] \"Epibacterium mobile\"                                   \n",
       " [549] \"Erwinia amylovora\"                                     \n",
       " [550] \"Erwinia billingiae\"                                    \n",
       " [551] \"Erwinia persicina\"                                     \n",
       " [552] \"Erythrobacter atlanticus\"                              \n",
       " [553] \"Erythrobacter flavus\"                                  \n",
       " [554] \"Erythrobacter gangjinensis\"                            \n",
       " [555] \"Erythrobacter litoralis\"                               \n",
       " [556] \"Erythrobacter seohaensis\"                              \n",
       " [557] \"Erythrobacter sp. Alg231-14\"                           \n",
       " [558] \"Erythrobacter sp. HL-111\"                              \n",
       " [559] \"Erythrobacter sp. KY5\"                                 \n",
       " [560] \"Erythrobacter sp. YH-07\"                               \n",
       " [561] \"Escherichia coli\"                                      \n",
       " [562] \"Euzebya sp. DY32-46\"                                   \n",
       " [563] \"Faecalibacterium prausnitzii\"                          \n",
       " [564] \"Ferrimonas balearica\"                                  \n",
       " [565] \"Ferriphaselus amnicola\"                                \n",
       " [566] \"Fibrella aestuarina\"                                   \n",
       " [567] \"Filimonas lacunae\"                                     \n",
       " [568] \"Fimbriimonas ginsengisoli\"                             \n",
       " [569] \"Finegoldia magna\"                                      \n",
       " [570] \"Flaviflexus salsibiostraticola\"                        \n",
       " [571] \"Flavisolibacter tropicus\"                              \n",
       " [572] \"Flavonifractor plautii\"                                \n",
       " [573] \"Frankia alni\"                                          \n",
       " [574] \"Frankia casuarinae\"                                    \n",
       " [575] \"Frankia inefficax\"                                     \n",
       " [576] \"Frankia sp. EAN1pec\"                                   \n",
       " [577] \"Frankia sp. QA3\"                                       \n",
       " [578] \"Frankia symbiont of Datisca glomerata\"                 \n",
       " [579] \"Frankineae bacterium MT45\"                             \n",
       " [580] \"Frateuria aurantia\"                                    \n",
       " [581] \"Friedmanniella luteola\"                                \n",
       " [582] \"Friedmanniella sagamiharensis\"                         \n",
       " [583] \"Frondihabitans sp. 762G35\"                             \n",
       " [584] \"Frondihabitans sp. PAMC 28766\"                         \n",
       " [585] \"Gallaecimonas sp. HK-28\"                               \n",
       " [586] \"gamma proteobacterium HdN1\"                            \n",
       " [587] \"Gardnerella vaginalis\"                                 \n",
       " [588] \"Gemmata obscuriglobus\"                                 \n",
       " [589] \"Gemmata sp. SH-PL17\"                                   \n",
       " [590] \"Gemmatimonas aurantiaca\"                               \n",
       " [591] \"Gemmatimonas phototrophica\"                            \n",
       " [592] \"Gemmatirosa kalamazoonesis\"                            \n",
       " [593] \"Gemmobacter sp. HYN0069\"                               \n",
       " [594] \"Geoalkalibacter subterraneus\"                          \n",
       " [595] \"Geobacter anodireducens\"                               \n",
       " [596] \"Geobacter pickeringii\"                                 \n",
       " [597] \"Geobacter sp. M18\"                                     \n",
       " [598] \"Geobacter sp. M21\"                                     \n",
       " [599] \"Geobacter sulfurreducens\"                              \n",
       " [600] \"Geodermatophilus obscurus\"                             \n",
       " [601] \"Georgenia sp. ZLJ0423\"                                 \n",
       " [602] \"Gibbsiella quercinecans\"                               \n",
       " [603] \"Gloeobacter kilaueensis\"                               \n",
       " [604] \"Gluconacetobacter diazotrophicus\"                      \n",
       " [605] \"Gluconobacter albidus\"                                 \n",
       " [606] \"Gluconobacter oxydans\"                                 \n",
       " [607] \"Glutamicibacter halophytocola\"                         \n",
       " [608] \"Glutamicibacter nicotianae\"                            \n",
       " [609] \"Gordonia bronchialis\"                                  \n",
       " [610] \"Gordonia iterans\"                                      \n",
       " [611] \"Gordonia phthalatica\"                                  \n",
       " [612] \"Gordonia polyisoprenivorans\"                           \n",
       " [613] \"Gordonia rubripertincta\"                               \n",
       " [614] \"Gordonia sp. 1D\"                                       \n",
       " [615] \"Gordonia sp. KTR9\"                                     \n",
       " [616] \"Gordonia sp. MMS17-SY073\"                              \n",
       " [617] \"Gordonia sp. YC-JH1\"                                   \n",
       " [618] \"Gordonia terrae\"                                       \n",
       " [619] \"Gordonibacter massiliensis\"                            \n",
       " [620] \"Gordonibacter pamelaeae\"                               \n",
       " [621] \"Granulibacter bethesdensis\"                            \n",
       " [622] \"Granulicella tundricola\"                               \n",
       " [623] \"Granulosicoccus antarcticus\"                           \n",
       " [624] \"Gryllotalpicola sp. 2DFW10M-5\"                         \n",
       " [625] \"Haemophilus parainfluenzae\"                            \n",
       " [626] \"Hahella chejuensis\"                                    \n",
       " [627] \"Haliangium ochraceum\"                                  \n",
       " [628] \"Haliscomenobacter hydrossis\"                           \n",
       " [629] \"Haloarculaceae archaeon HArcel1\"                       \n",
       " [630] \"Halobiforma lacisalsi\"                                 \n",
       " [631] \"Halocynthiibacter arcticus\"                            \n",
       " [632] \"Halodesulfurarchaeum formicicum\"                       \n",
       " [633] \"Halomonas aestuarii\"                                   \n",
       " [634] \"Halomonas beimenensis\"                                 \n",
       " [635] \"Halomonas chromatireducens\"                            \n",
       " [636] \"Halomonas elongata\"                                    \n",
       " [637] \"Halomonas huangheensis\"                                \n",
       " [638] \"Halomonas hydrothermalis\"                              \n",
       " [639] \"Halomonas sp. 1513\"                                    \n",
       " [640] \"Halomonas sp. A3H3\"                                    \n",
       " [641] \"Halomonas sp. JS92-SW72\"                               \n",
       " [642] \"Halomonas sp. KO116\"                                   \n",
       " [643] \"Halomonas sp. SF2003\"                                  \n",
       " [644] \"Halomonas subglaciescola\"                              \n",
       " [645] \"Halopenitus persicus\"                                  \n",
       " [646] \"Halopiger xanaduensis\"                                 \n",
       " [647] \"Halorhabdus utahensis\"                                 \n",
       " [648] \"Halorhodospira halophila\"                              \n",
       " [649] \"Halorientalis sp. IM1011\"                              \n",
       " [650] \"Halorubrum lacusprofundi\"                              \n",
       " [651] \"Halorubrum trapanicum\"                                 \n",
       " [652] \"Halotalea alkalilenta\"                                 \n",
       " [653] \"Haloterrigena turkmenica\"                              \n",
       " [654] \"Halothiobacillus neapolitanus\"                         \n",
       " [655] \"Hartmannibacter diazotrophicus\"                        \n",
       " [656] \"Herbaspirillum hiltneri\"                               \n",
       " [657] \"Herbaspirillum robiniae\"                               \n",
       " [658] \"Herbaspirillum rubrisubalbicans\"                       \n",
       " [659] \"Herbaspirillum seropedicae\"                            \n",
       " [660] \"Herbaspirillum sp. meg3\"                               \n",
       " [661] \"Herminiimonas arsenicoxydans\"                          \n",
       " [662] \"Herminiimonas arsenitoxidans\"                          \n",
       " [663] \"Hoeflea phototrophica\"                                 \n",
       " [664] \"Hoeflea sp. IMCC20628\"                                 \n",
       " [665] \"Hoyosella subflava\"                                    \n",
       " [666] \"Human betaherpesvirus 7\"                               \n",
       " [667] \"Human mastadenovirus C\"                                \n",
       " [668] \"Humibacter sp. BT305\"                                  \n",
       " [669] \"Hydrogenophaga crassostreae\"                           \n",
       " [670] \"Hydrogenophaga sp. PBC\"                                \n",
       " [671] \"Hydrogenophaga sp. RAC07\"                              \n",
       " [672] \"Hydrogenophilus thermoluteolus\"                        \n",
       " [673] \"Hymenobacter nivis\"                                    \n",
       " [674] \"Hymenobacter sedentarius\"                              \n",
       " [675] \"Hymenobacter sp. APR13\"                                \n",
       " [676] \"Hymenobacter sp. PAMC 26554\"                           \n",
       " [677] \"Hymenobacter sp. PAMC 26628\"                           \n",
       " [678] \"Hymenobacter sp. sh-6\"                                 \n",
       " [679] \"Hyphomicrobium denitrificans\"                          \n",
       " [680] \"Hyphomicrobium nitrativorans\"                          \n",
       " [681] \"Hyphomicrobium sp. MC1\"                                \n",
       " [682] \"Hyphomonas neptunium\"                                  \n",
       " [683] \"Hyphomonas sp. CACIAM 19H1\"                            \n",
       " [684] \"Hyphomonas sp. Mor2\"                                   \n",
       " [685] \"Immundisolibacter cernigliae\"                          \n",
       " [686] \"Indioceanicola profundi\"                               \n",
       " [687] \"Intrasporangium calvum\"                                \n",
       " [688] \"Isoptericola dokdonensis\"                              \n",
       " [689] \"Isoptericola variabilis\"                               \n",
       " [690] \"Isosphaera pallida\"                                    \n",
       " [691] \"Janibacter indicus\"                                    \n",
       " [692] \"Jannaschia sp. CCS1\"                                   \n",
       " [693] \"Janthinobacterium agaricidamnosum\"                     \n",
       " [694] \"Janthinobacterium sp. 1_2014MBL_MicDiv\"                \n",
       " [695] \"Janthinobacterium sp. B9-8\"                            \n",
       " [696] \"Janthinobacterium sp. LM6\"                             \n",
       " [697] \"Janthinobacterium sp. Marseille\"                       \n",
       " [698] \"Janthinobacterium svalbardensis\"                       \n",
       " [699] \"Jatrophihabitans sp. GAS493\"                           \n",
       " [700] \"Jeongeupia sp. USM3\"                                   \n",
       " [701] \"Jiangella alkaliphila\"                                 \n",
       " [702] \"Jiangella sp. DSM 45060\"                               \n",
       " [703] \"Ketogulonicigenium robustum\"                           \n",
       " [704] \"Ketogulonicigenium vulgare\"                            \n",
       " [705] \"Kibdelosporangium phytohabitans\"                       \n",
       " [706] \"Kineococcus radiotolerans\"                             \n",
       " [707] \"Kiritimatiella glycovorans\"                            \n",
       " [708] \"Kitasatospora albolonga\"                               \n",
       " [709] \"Kitasatospora aureofaciens\"                            \n",
       " [710] \"Kitasatospora setae\"                                   \n",
       " [711] \"Kitasatospora sp. MMS16-BH015\"                         \n",
       " [712] \"Klebsiella aerogenes\"                                  \n",
       " [713] \"Klebsiella michiganensis\"                              \n",
       " [714] \"Klebsiella oxytoca\"                                    \n",
       " [715] \"Klebsiella pneumoniae\"                                 \n",
       " [716] \"Kluyvera intermedia\"                                   \n",
       " [717] \"Kocuria flava\"                                         \n",
       " [718] \"Kocuria palustris\"                                     \n",
       " [719] \"Kocuria rosea\"                                         \n",
       " [720] \"Kocuria turfanensis\"                                   \n",
       " [721] \"Komagataeibacter europaeus\"                            \n",
       " [722] \"Komagataeibacter hansenii\"                             \n",
       " [723] \"Komagataeibacter medellinensis\"                        \n",
       " [724] \"Komagataeibacter nataicola\"                            \n",
       " [725] \"Komagataeibacter saccharivorans\"                       \n",
       " [726] \"Komagataeibacter xylinus\"                              \n",
       " [727] \"Kosakonia cowanii\"                                     \n",
       " [728] \"Kosakonia radicincitans\"                               \n",
       " [729] \"Kosakonia sp. CCTCC M2018092\"                          \n",
       " [730] \"Kozakia baliensis\"                                     \n",
       " [731] \"Kribbella flavida\"                                     \n",
       " [732] \"Kushneria konosiri\"                                    \n",
       " [733] \"Kushneria marisflavi\"                                  \n",
       " [734] \"Kutzneria albida\"                                      \n",
       " [735] \"Kytococcus sedentarius\"                                \n",
       " [736] \"Labrenzia aggregata\"                                   \n",
       " [737] \"Labrenzia sp. CP4\"                                     \n",
       " [738] \"Labrenzia sp. VG12\"                                    \n",
       " [739] \"Lactobacillus crispatus\"                               \n",
       " [740] \"Lactococcus lactis\"                                    \n",
       " [741] \"Lacunisphaera limnophila\"                              \n",
       " [742] \"Laribacter hongkongensis\"                              \n",
       " [743] \"Lautropia mirabilis\"                                   \n",
       " [744] \"Lawsonella clevelandensis\"                             \n",
       " [745] \"Legionella hackeliae\"                                  \n",
       " [746] \"Legionella pneumophila\"                                \n",
       " [747] \"Legionella spiritensis\"                                \n",
       " [748] \"Leifsonia sp. 21MFCrub1.1\"                             \n",
       " [749] \"Leifsonia xyli\"                                        \n",
       " [750] \"Leisingera methylohalidivorans\"                        \n",
       " [751] \"Lelliottia amnigena\"                                   \n",
       " [752] \"Leminorella richardii\"                                 \n",
       " [753] \"Lentzea guizhouensis\"                                  \n",
       " [754] \"Leptolyngbya sp. O-77\"                                 \n",
       " [755] \"Leptothrix cholodnii\"                                  \n",
       " [756] \"Limnohabitans sp. 103DPR2\"                             \n",
       " [757] \"Limnohabitans sp. 63ED37-2\"                            \n",
       " [758] \"Lonsdalea britannica\"                                  \n",
       " [759] \"Luteibacter rhizovicinus\"                              \n",
       " [760] \"Luteimonas sp. 100111\"                                 \n",
       " [761] \"Luteimonas sp. 83-4\"                                   \n",
       " [762] \"Luteimonas sp. JM171\"                                  \n",
       " [763] \"Luteipulveratus mongoliensis\"                          \n",
       " [764] \"Luteitalea pratensis\"                                  \n",
       " [765] \"Lysinimonas sp. 2DFWR-13\"                              \n",
       " [766] \"Lysobacter antibioticus\"                               \n",
       " [767] \"Lysobacter capsici\"                                    \n",
       " [768] \"Lysobacter enzymogenes\"                                \n",
       " [769] \"Lysobacter gummosus\"                                   \n",
       " [770] \"Lysobacter maris\"                                      \n",
       " [771] \"Lysobacter sp. TY2-98\"                                 \n",
       " [772] \"Magnetospira sp. QH-2\"                                 \n",
       " [773] \"Magnetospirillum gryphiswaldense\"                      \n",
       " [774] \"Magnetospirillum magneticum\"                           \n",
       " [775] \"Magnetospirillum sp. ME-1\"                             \n",
       " [776] \"Magnetospirillum sp. XM-1\"                             \n",
       " [777] \"Maricaulis maris\"                                      \n",
       " [778] \"Marichromatium purpuratum\"                             \n",
       " [779] \"Marinobacter hydrocarbonoclasticus\"                    \n",
       " [780] \"Marinobacter salarius\"                                 \n",
       " [781] \"Marinobacter similis\"                                  \n",
       " [782] \"Marinobacter sp. Arc7-DN-1\"                            \n",
       " [783] \"Marinobacter sp. es.042\"                               \n",
       " [784] \"Marinobacter sp. LQ44\"                                 \n",
       " [785] \"Marinobacter sp. LV10R510-11A\"                         \n",
       " [786] \"Marinobacterium aestuarii\"                             \n",
       " [787] \"Marinovum algicola\"                                    \n",
       " [788] \"Mariprofundus aestuarium\"                              \n",
       " [789] \"Maritalea myrionectae\"                                 \n",
       " [790] \"Marivivens sp. JLT3646\"                                \n",
       " [791] \"Marmoricola scoriae\"                                   \n",
       " [792] \"Martelella endophytica\"                                \n",
       " [793] \"Martelella mediterranea\"                               \n",
       " [794] \"Martelella sp. AD-3\"                                   \n",
       " [795] \"Massilia armeniaca\"                                    \n",
       " [796] \"Massilia oculi\"                                        \n",
       " [797] \"Massilia putida\"                                       \n",
       " [798] \"Massilia sp. NR 4-1\"                                   \n",
       " [799] \"Massilia sp. WG5\"                                      \n",
       " [800] \"Massilia sp. YMA4\"                                     \n",
       " [801] \"Massilia violaceinigra\"                                \n",
       " [802] \"Melaminivora sp. SC2-7\"                                \n",
       " [803] \"Melaminivora sp. SC2-9\"                                \n",
       " [804] \"Melittangium boletus\"                                  \n",
       " [805] \"Mesorhizobium amorphae\"                                \n",
       " [806] \"Mesorhizobium australicum\"                             \n",
       " [807] \"Mesorhizobium ciceri\"                                  \n",
       " [808] \"Mesorhizobium huakuii\"                                 \n",
       " [809] \"Mesorhizobium japonicum\"                               \n",
       " [810] \"Mesorhizobium loti\"                                    \n",
       " [811] \"Mesorhizobium oceanicum\"                               \n",
       " [812] \"Mesorhizobium opportunistum\"                           \n",
       " [813] \"Mesorhizobium sp. DCY119\"                              \n",
       " [814] \"Mesorhizobium sp. M1B.F.Ca.ET.045.04.1.1\"              \n",
       " [815] \"Mesorhizobium sp. M1D.F.Ca.ET.043.01.1.1\"              \n",
       " [816] \"Mesorhizobium sp. M1E.F.Ca.ET.045.02.1.1\"              \n",
       " [817] \"Mesorhizobium sp. M2A.F.Ca.ET.043.05.1.1\"              \n",
       " [818] \"Mesorhizobium sp. M2A.F.Ca.ET.046.03.2.1\"              \n",
       " [819] \"Mesorhizobium sp. M4B.F.Ca.ET.058.02.1.1\"              \n",
       " [820] \"Mesorhizobium sp. M6A.T.Cr.TU.016.01.1.1\"              \n",
       " [821] \"Mesorhizobium sp. M7D.F.Ca.US.005.01.1.1\"              \n",
       " [822] \"Mesorhizobium sp. M9A.F.Ca.ET.002.03.1.2\"              \n",
       " [823] \"Mesorhizobium sp. WSM1497\"                             \n",
       " [824] \"Methylacidiphilum infernorum\"                          \n",
       " [825] \"Methylibium petroleiphilum\"                            \n",
       " [826] \"Methylobacterium aquaticum\"                            \n",
       " [827] \"Methylobacterium brachiatum\"                           \n",
       " [828] \"Methylobacterium currus\"                               \n",
       " [829] \"Methylobacterium nodulans\"                             \n",
       " [830] \"Methylobacterium oryzae\"                               \n",
       " [831] \"Methylobacterium phyllosphaerae\"                       \n",
       " [832] \"Methylobacterium radiotolerans\"                        \n",
       " [833] \"Methylobacterium sp. 17SD2-17\"                         \n",
       " [834] \"Methylobacterium sp. 17Sr1-1\"                          \n",
       " [835] \"Methylobacterium sp. 17Sr1-28\"                         \n",
       " [836] \"Methylobacterium sp. 17Sr1-43\"                         \n",
       " [837] \"Methylobacterium sp. 4-46\"                             \n",
       " [838] \"Methylobacterium sp. AMS5\"                             \n",
       " [839] \"Methylobacterium sp. C1\"                               \n",
       " [840] \"Methylobacterium sp. DM1\"                              \n",
       " [841] \"Methylobacterium sp. XJLW\"                             \n",
       " [842] \"Methylocaldum marinum\"                                 \n",
       " [843] \"Methyloceanibacter caenitepidi\"                        \n",
       " [844] \"Methyloceanibacter sp. wino2\"                          \n",
       " [845] \"Methylocella silvestris\"                               \n",
       " [846] \"Methylococcus capsulatus\"                              \n",
       " [847] \"Methylocystis bryophila\"                               \n",
       " [848] \"Methylocystis rosea\"                                   \n",
       " [849] \"Methylocystis sp. SC2\"                                 \n",
       " [850] \"Methylomicrobium album\"                                \n",
       " [851] \"Methylomicrobium sp. wino1\"                            \n",
       " [852] \"Methylomonas denitrificans\"                            \n",
       " [853] \"Methylomonas koyamae\"                                  \n",
       " [854] \"Methylomonas methanica\"                                \n",
       " [855] \"Methylomonas sp. DH-1\"                                 \n",
       " [856] \"Methylophilus sp. TWE2\"                                \n",
       " [857] \"Methylorubrum extorquens\"                              \n",
       " [858] \"Methylorubrum populi\"                                  \n",
       " [859] \"Methylorubrum zatmanii\"                                \n",
       " [860] \"Methylosinus trichosporium\"                            \n",
       " [861] \"Methyloversatilis sp. RAC08\"                           \n",
       " [862] \"Methylovulum psychrotolerans\"                          \n",
       " [863] \"Micavibrio aeruginosavorus\"                            \n",
       " [864] \"Microbacterium aurum\"                                  \n",
       " [865] \"Microbacterium chocolatum\"                             \n",
       " [866] \"Microbacterium foliorum\"                               \n",
       " [867] \"Microbacterium hominis\"                                \n",
       " [868] \"Microbacterium paludicola\"                             \n",
       " [869] \"Microbacterium paraoxydans\"                            \n",
       " [870] \"Microbacterium pygmaeum\"                               \n",
       " [871] \"Microbacterium sp. 1.5R\"                               \n",
       " [872] \"Microbacterium sp. 10M-3C3\"                            \n",
       " [873] \"Microbacterium sp. ABRD_28\"                            \n",
       " [874] \"Microbacterium sp. BH-3-3-3\"                           \n",
       " [875] \"Microbacterium sp. CGR1\"                               \n",
       " [876] \"Microbacterium sp. LKL04\"                              \n",
       " [877] \"Microbacterium sp. No. 7\"                              \n",
       " [878] \"Microbacterium sp. PAMC 28756\"                         \n",
       " [879] \"Microbacterium sp. PM5\"                                \n",
       " [880] \"Microbacterium sp. str. 'China'\"                       \n",
       " [881] \"Microbacterium sp. TPU 3598\"                           \n",
       " [882] \"Microbacterium sp. XT11\"                               \n",
       " [883] \"Microbacterium sp. Y-01\"                               \n",
       " [884] \"Microbacterium testaceum\"                              \n",
       " [885] \"Microbulbifer agarilyticus\"                            \n",
       " [886] \"Microbulbifer aggregans\"                               \n",
       " [887] \"Microbulbifer thermotolerans\"                          \n",
       " [888] \"Microcella alkaliphila\"                                \n",
       " [889] \"Micrococcus luteus\"                                    \n",
       " [890] \"Microlunatus phosphovorus\"                             \n",
       " [891] \"Microlunatus soli\"                                     \n",
       " [892] \"Micromonospora aurantiaca\"                             \n",
       " [893] \"Micromonospora auratinigra\"                            \n",
       " [894] \"Micromonospora chokoriensis\"                           \n",
       " [895] \"Micromonospora coriariae\"                              \n",
       " [896] \"Micromonospora coxensis\"                               \n",
       " [897] \"Micromonospora echinaurantiaca\"                        \n",
       " [898] \"Micromonospora echinofusca\"                            \n",
       " [899] \"Micromonospora echinospora\"                            \n",
       " [900] \"Micromonospora inositola\"                              \n",
       " [901] \"Micromonospora krabiensis\"                             \n",
       " [902] \"Micromonospora narathiwatensis\"                        \n",
       " [903] \"Micromonospora purpureochromogenes\"                    \n",
       " [904] \"Micromonospora rifamycinica\"                           \n",
       " [905] \"Micromonospora siamensis\"                              \n",
       " [906] \"Micromonospora sp. B006\"                               \n",
       " [907] \"Micromonospora sp. WMMA2032\"                           \n",
       " [908] \"Micromonospora tulbaghiae\"                             \n",
       " [909] \"Micromonospora viridifaciens\"                          \n",
       " [910] \"Micromonospora zamorensis\"                             \n",
       " [911] \"Micropruina glycogenica\"                               \n",
       " [912] \"Microterricola viridarii\"                              \n",
       " [913] \"Microvirga ossetica\"                                   \n",
       " [914] \"Microvirga sp. 17 mud 1-3\"                             \n",
       " [915] \"Microvirgula aerodenitrificans\"                        \n",
       " [916] \"Miniimonas sp. S16\"                                    \n",
       " [917] \"Mitsuaria sp. 7\"                                       \n",
       " [918] \"Mixta gaviniae\"                                        \n",
       " [919] \"Modestobacter marinus\"                                 \n",
       " [920] \"Moorella thermoacetica\"                                \n",
       " [921] \"Moraxella osloensis\"                                   \n",
       " [922] \"Morganella morganii\"                                   \n",
       " [923] \"Mucilaginibacter mallensis\"                            \n",
       " [924] \"Mycetocola sp. 449\"                                    \n",
       " [925] \"Mycobacterium avium\"                                   \n",
       " [926] \"Mycobacterium canettii\"                                \n",
       " [927] \"Mycobacterium chimaera\"                                \n",
       " [928] \"Mycobacterium colombiense\"                             \n",
       " [929] \"Mycobacterium dioxanotrophicus\"                        \n",
       " [930] \"Mycobacterium haemophilum\"                             \n",
       " [931] \"Mycobacterium intracellulare\"                          \n",
       " [932] \"Mycobacterium kansasii\"                                \n",
       " [933] \"Mycobacterium leprae\"                                  \n",
       " [934] \"Mycobacterium marinum\"                                 \n",
       " [935] \"Mycobacterium marseillense\"                            \n",
       " [936] \"Mycobacterium paragordonae\"                            \n",
       " [937] \"Mycobacterium shigaense\"                               \n",
       " [938] \"Mycobacterium sp. djl-10\"                              \n",
       " [939] \"Mycobacterium sp. DL90\"                                \n",
       " [940] \"Mycobacterium sp. EPa45\"                               \n",
       " [941] \"Mycobacterium sp. JS623\"                               \n",
       " [942] \"Mycobacterium sp. MS1601\"                              \n",
       " [943] \"Mycobacterium sp. PYR15\"                               \n",
       " [944] \"Mycobacterium sp. QIA-37\"                              \n",
       " [945] \"Mycobacterium sp. VKM Ac-1817D\"                        \n",
       " [946] \"Mycobacterium sp. WY10\"                                \n",
       " [947] \"Mycobacterium sp. YC-RL4\"                              \n",
       " [948] \"Mycobacterium stephanolepidis\"                         \n",
       " [949] \"Mycobacterium tuberculosis\"                            \n",
       " [950] \"Mycobacteroides abscessus\"                             \n",
       " [951] \"Mycobacteroides chelonae\"                              \n",
       " [952] \"Mycobacteroides immunogenum\"                           \n",
       " [953] \"Mycobacteroides saopaulense\"                           \n",
       " [954] \"Mycolicibacter sinensis\"                               \n",
       " [955] \"Mycolicibacter terrae\"                                 \n",
       " [956] \"Mycolicibacterium aurum\"                               \n",
       " [957] \"Mycolicibacterium chitae\"                              \n",
       " [958] \"Mycolicibacterium chubuense\"                           \n",
       " [959] \"Mycolicibacterium flavescens\"                          \n",
       " [960] \"Mycolicibacterium gilvum\"                              \n",
       " [961] \"Mycolicibacterium goodii\"                              \n",
       " [962] \"Mycolicibacterium hassiacum\"                           \n",
       " [963] \"Mycolicibacterium litorale\"                            \n",
       " [964] \"Mycolicibacterium rhodesiae\"                           \n",
       " [965] \"Mycolicibacterium rutilum\"                             \n",
       " [966] \"Mycolicibacterium smegmatis\"                           \n",
       " [967] \"Mycolicibacterium thermoresistibile\"                   \n",
       " [968] \"Mycolicibacterium vaccae\"                              \n",
       " [969] \"Mycolicibacterium vanbaalenii\"                         \n",
       " [970] \"Myxococcus fulvus\"                                     \n",
       " [971] \"Myxococcus hansupus\"                                   \n",
       " [972] \"Myxococcus macrosporus\"                                \n",
       " [973] \"Myxococcus stipitatus\"                                 \n",
       " [974] \"Myxococcus xanthus\"                                    \n",
       " [975] \"Nakamurella multipartita\"                              \n",
       " [976] \"Nakamurella panacisegetis\"                             \n",
       " [977] \"Nakamurella sp. s14-144\"                               \n",
       " [978] \"Natrinema pellirubrum\"                                 \n",
       " [979] \"Natronococcus occultus\"                                \n",
       " [980] \"Neisseria elongata\"                                    \n",
       " [981] \"Neisseria meningitidis\"                                \n",
       " [982] \"Neisseria sp. 10023\"                                   \n",
       " [983] \"Neisseria sp. KEM232\"                                  \n",
       " [984] \"Neisseria subflava\"                                    \n",
       " [985] \"Neoasaia chiangmaiensis\"                               \n",
       " [986] \"Neomicrococcus aestuarii\"                              \n",
       " [987] \"Neorhizobium galegae\"                                  \n",
       " [988] \"Neorhizobium sp. NCHU2750\"                             \n",
       " [989] \"Neorhizobium sp. SOG26\"                                \n",
       " [990] \"Niastella koreensis\"                                   \n",
       " [991] \"Nitratireductor basaltis\"                              \n",
       " [992] \"Nitratireductor sp. OM-1\"                              \n",
       " [993] \"Nitrobacter hamburgensis\"                              \n",
       " [994] \"Nitrobacter winogradskyi\"                              \n",
       " [995] \"Nitrosomonas europaea\"                                 \n",
       " [996] \"Nitrosomonas eutropha\"                                 \n",
       " [997] \"Nitrosospira briensis\"                                 \n",
       " [998] \"Nitrosospira lacus\"                                    \n",
       " [999] \"Nitrospira defluvii\"                                   \n",
       "[1000] \"Nitrospira moscoviensis\"                               \n",
       "[1001] \"Nitrospirillum amazonense\"                             \n",
       "[1002] \"Niveispirillum cyanobacteriorum\"                       \n",
       "[1003] \"Nocardia asteroides\"                                   \n",
       "[1004] \"Nocardia brasiliensis\"                                 \n",
       "[1005] \"Nocardia cyriacigeorgica\"                              \n",
       "[1006] \"Nocardia farcinica\"                                    \n",
       "[1007] \"Nocardia nova\"                                         \n",
       "[1008] \"Nocardia seriolae\"                                     \n",
       "[1009] \"Nocardia sp. CFHS0054\"                                 \n",
       "[1010] \"Nocardia sp. Y48\"                                      \n",
       "[1011] \"Nocardia terpenica\"                                    \n",
       "[1012] \"Nocardioides baekrokdamisoli\"                          \n",
       "[1013] \"Nocardioides dokdonensis\"                              \n",
       "[1014] \"Nocardioides sp. 603\"                                  \n",
       "[1015] \"Nocardioides sp. 78\"                                   \n",
       "[1016] \"Nocardioides sp. CF8\"                                  \n",
       "[1017] \"Nocardioides sp. JS614\"                                \n",
       "[1018] \"Nocardiopsis alba\"                                     \n",
       "[1019] \"Nocardiopsis dassonvillei\"                             \n",
       "[1020] \"Nocardiopsis gilva\"                                    \n",
       "[1021] \"Nonomuraea sp. ATCC 55076\"                             \n",
       "[1022] \"Novosphingobium aromaticivorans\"                       \n",
       "[1023] \"Novosphingobium pentaromativorans\"                     \n",
       "[1024] \"Novosphingobium resinovorum\"                           \n",
       "[1025] \"Novosphingobium sp. P6W\"                               \n",
       "[1026] \"Novosphingobium sp. PP1Y\"                              \n",
       "[1027] \"Novosphingobium sp. THN1\"                              \n",
       "[1028] \"Novosphingobium tardaugens\"                            \n",
       "[1029] \"Oblitimonas alkaliphila\"                               \n",
       "[1030] \"Oceanimonas sp. GK1\"                                   \n",
       "[1031] \"Oceanithermus profundus\"                               \n",
       "[1032] \"Ochrobactrum anthropi\"                                 \n",
       "[1033] \"Ochrobactrum pituitosum\"                               \n",
       "[1034] \"Ochrobactrum pseudogrignonense\"                        \n",
       "[1035] \"Ochrobactrum sp. A44\"                                  \n",
       "[1036] \"Octadecabacter antarcticus\"                            \n",
       "[1037] \"Octadecabacter arcticus\"                               \n",
       "[1038] \"Oligella urethralis\"                                   \n",
       "[1039] \"Oligotropha carboxidovorans\"                           \n",
       "[1040] \"Olsenella uli\"                                         \n",
       "[1041] \"Opitutaceae bacterium TAV5\"                            \n",
       "[1042] \"Opitutus sp. GAS368\"                                   \n",
       "[1043] \"Opitutus terrae\"                                       \n",
       "[1044] \"Ornithinimicrobium sp. AMA3305\"                        \n",
       "[1045] \"Orrella dioscoreae\"                                    \n",
       "[1046] \"Oscillibacter sp. PEA192\"                              \n",
       "[1047] \"Ottowia oryzae\"                                        \n",
       "[1048] \"Ottowia sp. oral taxon 894\"                            \n",
       "[1049] \"Paenalcaligenes hominis\"                               \n",
       "[1050] \"Paenarthrobacter aurescens\"                            \n",
       "[1051] \"Paenibacillus beijingensis\"                            \n",
       "[1052] \"Paenibacillus ihbetae\"                                 \n",
       "[1053] \"Paenibacillus lautus\"                                  \n",
       "[1054] \"Paenibacillus mucilaginosus\"                           \n",
       "[1055] \"Paenibacillus sp. FSL R7-0331\"                         \n",
       "[1056] \"Paenibacillus sp. RUD330\"                              \n",
       "[1057] \"Paludisphaera borealis\"                                \n",
       "[1058] \"Pandoraea apista\"                                      \n",
       "[1059] \"Pandoraea faecigallinarum\"                             \n",
       "[1060] \"Pandoraea norimbergensis\"                              \n",
       "[1061] \"Pandoraea oxalativorans\"                               \n",
       "[1062] \"Pandoraea pnomenusa\"                                   \n",
       "[1063] \"Pandoraea pulmonicola\"                                 \n",
       "[1064] \"Pandoraea sputorum\"                                    \n",
       "[1065] \"Pandoraea thiooxydans\"                                 \n",
       "[1066] \"Pandoraea vervacti\"                                    \n",
       "[1067] \"Pannonibacter phragmitetus\"                            \n",
       "[1068] \"Pantoea agglomerans\"                                   \n",
       "[1069] \"Pantoea ananatis\"                                      \n",
       "[1070] \"Pantoea sp. At-9b\"                                     \n",
       "[1071] \"Pantoea stewartii\"                                     \n",
       "[1072] \"Pantoea vagans\"                                        \n",
       "[1073] \"Paraburkholderia aromaticivorans\"                      \n",
       "[1074] \"Paraburkholderia caffeinilytica\"                       \n",
       "[1075] \"Paraburkholderia caledonica\"                           \n",
       "[1076] \"Paraburkholderia caribensis\"                           \n",
       "[1077] \"Paraburkholderia fungorum\"                             \n",
       "[1078] \"Paraburkholderia graminis\"                             \n",
       "[1079] \"Paraburkholderia hospita\"                              \n",
       "[1080] \"Paraburkholderia phenoliruptrix\"                       \n",
       "[1081] \"Paraburkholderia phymatum\"                             \n",
       "[1082] \"Paraburkholderia phytofirmans\"                         \n",
       "[1083] \"Paraburkholderia rhizoxinica\"                          \n",
       "[1084] \"Paraburkholderia sp. SOS3\"                             \n",
       "[1085] \"Paraburkholderia sprentiae\"                            \n",
       "[1086] \"Paraburkholderia terrae\"                               \n",
       "[1087] \"Paraburkholderia terricola\"                            \n",
       "[1088] \"Paraburkholderia xenovorans\"                           \n",
       "[1089] \"Paracoccus aminophilus\"                                \n",
       "[1090] \"Paracoccus aminovorans\"                                \n",
       "[1091] \"Paracoccus contaminans\"                                \n",
       "[1092] \"Paracoccus denitrificans\"                              \n",
       "[1093] \"Paracoccus mutanolyticus\"                              \n",
       "[1094] \"Paracoccus sp. BM15\"                                   \n",
       "[1095] \"Paracoccus sp. CBA4604\"                                \n",
       "[1096] \"Paracoccus sp. SC2-6\"                                  \n",
       "[1097] \"Paracoccus yeei\"                                       \n",
       "[1098] \"Paracoccus zhejiangensis\"                              \n",
       "[1099] \"Paraoerskovia marina\"                                  \n",
       "[1100] \"Pararhodospirillum photometricum\"                      \n",
       "[1101] \"Parasaccharibacter apium\"                              \n",
       "[1102] \"Parvibaculum lavamentivorans\"                          \n",
       "[1103] \"Parvularcula bermudensis\"                              \n",
       "[1104] \"Paucibacter sp. KCTC 42545\"                            \n",
       "[1105] \"Paucimonas lemoignei\"                                  \n",
       "[1106] \"Pectobacterium carotovorum\"                            \n",
       "[1107] \"Pectobacterium parmentieri\"                            \n",
       "[1108] \"Pelagibaca abyssi\"                                     \n",
       "[1109] \"Pelagibacterium halotolerans\"                          \n",
       "[1110] \"Pelobacter carbinolicus\"                               \n",
       "[1111] \"Pelobacter propionicus\"                                \n",
       "[1112] \"Phaeobacter gallaeciensis\"                             \n",
       "[1113] \"Phaeobacter inhibens\"                                  \n",
       "[1114] \"Phaeobacter piscinae\"                                  \n",
       "[1115] \"Phaeobacter porticola\"                                 \n",
       "[1116] \"Phenylobacterium sp. HYN0004\"                          \n",
       "[1117] \"Phenylobacterium zucineum\"                             \n",
       "[1118] \"Phreatobacter cathodiphilus\"                           \n",
       "[1119] \"Phycicoccus dokdonensis\"                               \n",
       "[1120] \"Phycisphaera mikurensis\"                               \n",
       "[1121] \"Phyllobacterium zundukense\"                            \n",
       "[1122] \"Pigmentiphaga sp. H8\"                                  \n",
       "[1123] \"Pimelobacter simplex\"                                  \n",
       "[1124] \"Planctomyces sp. SH-PL14\"                              \n",
       "[1125] \"Planctomyces sp. SH-PL62\"                              \n",
       "[1126] \"Plantactinospora sp. BB1\"                              \n",
       "[1127] \"Plantactinospora sp. KBS50\"                            \n",
       "[1128] \"Plantibacter flavus\"                                   \n",
       "[1129] \"Plantibacter sp. PA-3-X8\"                              \n",
       "[1130] \"Pleomorphomonas sp. SM30\"                              \n",
       "[1131] \"Pluralibacter gergoviae\"                               \n",
       "[1132] \"Polaromonas naphthalenivorans\"                         \n",
       "[1133] \"Polaromonas sp. JS666\"                                 \n",
       "[1134] \"Polaromonas sp. SP1\"                                   \n",
       "[1135] \"Polyangium brachysporum\"                               \n",
       "[1136] \"Polymorphum gilvum\"                                    \n",
       "[1137] \"Polynucleobacter necessarius\"                          \n",
       "[1138] \"Pontibacter akesuensis\"                                \n",
       "[1139] \"Pontibacter korlensis\"                                 \n",
       "[1140] \"Pontimonas salivibrio\"                                 \n",
       "[1141] \"Porphyrobacter HT-58-2\"                                \n",
       "[1142] \"Porphyrobacter neustonensis\"                           \n",
       "[1143] \"Porphyrobacter sp. CACIAM 03H1\"                        \n",
       "[1144] \"Porphyrobacter sp. LM 6\"                               \n",
       "[1145] \"Prauserella marina\"                                    \n",
       "[1146] \"Prevotella intermedia\"                                 \n",
       "[1147] \"Propionibacterium acidifaciens\"                        \n",
       "[1148] \"Propionibacterium australiense\"                        \n",
       "[1149] \"Propionibacterium freudenreichii\"                      \n",
       "[1150] \"Propionibacterium sp. oral taxon 193\"                  \n",
       "[1151] \"Pseudarthrobacter chlorophenolicus\"                    \n",
       "[1152] \"Pseudarthrobacter equi\"                                \n",
       "[1153] \"Pseudarthrobacter phenanthrenivorans\"                  \n",
       "[1154] \"Pseudarthrobacter sulfonivorans\"                       \n",
       "[1155] \"Pseudodesulfovibrio aespoeensis\"                       \n",
       "[1156] \"Pseudodesulfovibrio indicus\"                           \n",
       "[1157] \"Pseudoflavitalea sp. 5GH32-13\"                         \n",
       "[1158] \"Pseudogulbenkiania sp. NH8B\"                           \n",
       "[1159] \"Pseudolabrys taiwanensis\"                              \n",
       "[1160] \"Pseudomonas aeruginosa\"                                \n",
       "[1161] \"Pseudomonas agarici\"                                   \n",
       "[1162] \"Pseudomonas alcaligenes\"                               \n",
       "[1163] \"Pseudomonas alcaliphila\"                               \n",
       "[1164] \"Pseudomonas alkylphenolica\"                            \n",
       "[1165] \"Pseudomonas amygdali\"                                  \n",
       "[1166] \"Pseudomonas antarctica\"                                \n",
       "[1167] \"Pseudomonas arsenicoxydans\"                            \n",
       "[1168] \"Pseudomonas asplenii\"                                  \n",
       "[1169] \"Pseudomonas azotoformans\"                              \n",
       "[1170] \"Pseudomonas balearica\"                                 \n",
       "[1171] \"Pseudomonas brassicacearum\"                            \n",
       "[1172] \"Pseudomonas brenneri\"                                  \n",
       "[1173] \"Pseudomonas cedrina\"                                   \n",
       "[1174] \"Pseudomonas cerasi\"                                    \n",
       "[1175] \"Pseudomonas chlororaphis\"                              \n",
       "[1176] \"Pseudomonas cichorii\"                                  \n",
       "[1177] \"Pseudomonas citronellolis\"                             \n",
       "[1178] \"Pseudomonas corrugata\"                                 \n",
       "[1179] \"Pseudomonas cremoricolorata\"                           \n",
       "[1180] \"Pseudomonas entomophila\"                               \n",
       "[1181] \"Pseudomonas extremaustralis\"                           \n",
       "[1182] \"Pseudomonas extremorientalis\"                          \n",
       "[1183] \"Pseudomonas fluorescens\"                               \n",
       "[1184] \"Pseudomonas fragi\"                                     \n",
       "[1185] \"Pseudomonas frederiksbergensis\"                        \n",
       "[1186] \"Pseudomonas fulva\"                                     \n",
       "[1187] \"Pseudomonas furukawaii\"                                \n",
       "[1188] \"Pseudomonas fuscovaginae\"                              \n",
       "[1189] \"Pseudomonas granadensis\"                               \n",
       "[1190] \"Pseudomonas guangdongensis\"                            \n",
       "[1191] \"Pseudomonas knackmussii\"                               \n",
       "[1192] \"Pseudomonas koreensis\"                                 \n",
       "[1193] \"Pseudomonas kribbensis\"                                \n",
       "[1194] \"Pseudomonas libanensis\"                                \n",
       "[1195] \"Pseudomonas lini\"                                      \n",
       "[1196] \"Pseudomonas litoralis\"                                 \n",
       "[1197] \"Pseudomonas lurida\"                                    \n",
       "[1198] \"Pseudomonas mandelii\"                                  \n",
       "[1199] \"Pseudomonas mediterranea\"                              \n",
       "[1200] \"Pseudomonas mendocina\"                                 \n",
       "[1201] \"Pseudomonas mesoacidophila\"                            \n",
       "[1202] \"Pseudomonas monteilii\"                                 \n",
       "[1203] \"Pseudomonas moraviensis\"                               \n",
       "[1204] \"Pseudomonas mosselii\"                                  \n",
       "[1205] \"Pseudomonas mucidolens\"                                \n",
       "[1206] \"Pseudomonas orientalis\"                                \n",
       "[1207] \"Pseudomonas oryzae\"                                    \n",
       "[1208] \"Pseudomonas oryzihabitans\"                             \n",
       "[1209] \"Pseudomonas palleroniana\"                              \n",
       "[1210] \"Pseudomonas parafulva\"                                 \n",
       "[1211] \"Pseudomonas plecoglossicida\"                           \n",
       "[1212] \"Pseudomonas poae\"                                      \n",
       "[1213] \"Pseudomonas pohangensis\"                               \n",
       "[1214] \"Pseudomonas prosekii\"                                  \n",
       "[1215] \"Pseudomonas protegens\"                                 \n",
       "[1216] \"Pseudomonas pseudoalcaligenes\"                         \n",
       "[1217] \"Pseudomonas psychrophila\"                              \n",
       "[1218] \"Pseudomonas psychrotolerans\"                           \n",
       "[1219] \"Pseudomonas putida\"                                    \n",
       "[1220] \"Pseudomonas reinekei\"                                  \n",
       "[1221] \"Pseudomonas resinovorans\"                              \n",
       "[1222] \"Pseudomonas rhizosphaerae\"                             \n",
       "[1223] \"Pseudomonas rhodesiae\"                                 \n",
       "[1224] \"Pseudomonas sabulinigri\"                               \n",
       "[1225] \"Pseudomonas salegens\"                                  \n",
       "[1226] \"Pseudomonas saudiphocaensis\"                           \n",
       "[1227] \"Pseudomonas savastanoi\"                                \n",
       "[1228] \"Pseudomonas sihuiensis\"                                \n",
       "[1229] \"Pseudomonas silesiensis\"                               \n",
       "[1230] \"Pseudomonas simiae\"                                    \n",
       "[1231] \"Pseudomonas soli\"                                      \n",
       "[1232] \"Pseudomonas sp. 02C 26\"                                \n",
       "[1233] \"Pseudomonas sp. 09C 129\"                               \n",
       "[1234] \"Pseudomonas sp. 31-12\"                                 \n",
       "[1235] \"Pseudomonas sp. 7SR1\"                                  \n",
       "[1236] \"Pseudomonas sp. A214\"                                  \n",
       "[1237] \"Pseudomonas sp. ATCC 13867\"                            \n",
       "[1238] \"Pseudomonas sp. B10\"                                   \n",
       "[1239] \"Pseudomonas sp. bs2935\"                                \n",
       "[1240] \"Pseudomonas sp. CC6-YY-74\"                             \n",
       "[1241] \"Pseudomonas sp. CCOS 191\"                              \n",
       "[1242] \"Pseudomonas sp. CMR12a\"                                \n",
       "[1243] \"Pseudomonas sp. CMR5c\"                                 \n",
       "[1244] \"Pseudomonas sp. DR 5-09\"                               \n",
       "[1245] \"Pseudomonas sp. DY-1\"                                  \n",
       "[1246] \"Pseudomonas sp. FDAARGOS_380\"                          \n",
       "[1247] \"Pseudomonas sp. FGI182\"                                \n",
       "[1248] \"Pseudomonas sp. GR 6-02\"                               \n",
       "[1249] \"Pseudomonas sp. HLS-6\"                                 \n",
       "[1250] \"Pseudomonas sp. JY-Q\"                                  \n",
       "[1251] \"Pseudomonas sp. K2W31S-8\"                              \n",
       "[1252] \"Pseudomonas sp. LAB-08\"                                \n",
       "[1253] \"Pseudomonas sp. LBUM920\"                               \n",
       "[1254] \"Pseudomonas sp. Leaf58\"                                \n",
       "[1255] \"Pseudomonas sp. LG1D9\"                                 \n",
       "[1256] \"Pseudomonas sp. LG1E9\"                                 \n",
       "[1257] \"Pseudomonas sp. LH1G9\"                                 \n",
       "[1258] \"Pseudomonas sp. LPH1\"                                  \n",
       "[1259] \"Pseudomonas sp. LTJR-52\"                               \n",
       "[1260] \"Pseudomonas sp. Lz4W\"                                  \n",
       "[1261] \"Pseudomonas sp. M30-35\"                                \n",
       "[1262] \"Pseudomonas sp. MRSN12121\"                             \n",
       "[1263] \"Pseudomonas sp. MT-1\"                                  \n",
       "[1264] \"Pseudomonas sp. MYb193\"                                \n",
       "[1265] \"Pseudomonas sp. NC02\"                                  \n",
       "[1266] \"Pseudomonas sp. NS1(2017)\"                             \n",
       "[1267] \"Pseudomonas sp. Os17\"                                  \n",
       "[1268] \"Pseudomonas sp. phDV1\"                                 \n",
       "[1269] \"Pseudomonas sp. PONIH3\"                                \n",
       "[1270] \"Pseudomonas sp. R1-43-08\"                              \n",
       "[1271] \"Pseudomonas sp. R2-37-08W\"                             \n",
       "[1272] \"Pseudomonas sp. R2-60-08W\"                             \n",
       "[1273] \"Pseudomonas sp. R2A2\"                                  \n",
       "[1274] \"Pseudomonas sp. R3-18-08\"                              \n",
       "[1275] \"Pseudomonas sp. R4-34-07\"                              \n",
       "[1276] \"Pseudomonas sp. R4-35-07\"                              \n",
       "[1277] \"Pseudomonas sp. R4-39-08\"                              \n",
       "[1278] \"Pseudomonas sp. R5-89-07\"                              \n",
       "[1279] \"Pseudomonas sp. S-6-2\"                                 \n",
       "[1280] \"Pseudomonas sp. S09G 359\"                              \n",
       "[1281] \"Pseudomonas sp. s211(2017)\"                            \n",
       "[1282] \"Pseudomonas sp. St29\"                                  \n",
       "[1283] \"Pseudomonas sp. StFLB209\"                              \n",
       "[1284] \"Pseudomonas sp. SWI36\"                                 \n",
       "[1285] \"Pseudomonas sp. SWI44\"                                 \n",
       "[1286] \"Pseudomonas sp. TCU-HL1\"                               \n",
       "[1287] \"Pseudomonas sp. TKP\"                                   \n",
       "[1288] \"Pseudomonas sp. TMW 2.1634\"                            \n",
       "[1289] \"Pseudomonas sp. URMO17WK12:I11\"                        \n",
       "[1290] \"Pseudomonas sp. UW4\"                                   \n",
       "[1291] \"Pseudomonas sp. XWY-1\"                                 \n",
       "[1292] \"Pseudomonas sp. Z003-0.4C(8344-21)\"                    \n",
       "[1293] \"Pseudomonas stutzeri\"                                  \n",
       "[1294] \"Pseudomonas synxantha\"                                 \n",
       "[1295] \"Pseudomonas syringae\"                                  \n",
       "[1296] \"Pseudomonas syringae group genomosp. 3\"                \n",
       "[1297] \"Pseudomonas taetrolens\"                                \n",
       "[1298] \"Pseudomonas thivervalensis\"                            \n",
       "[1299] \"Pseudomonas tolaasii\"                                  \n",
       "[1300] \"Pseudomonas trivialis\"                                 \n",
       "[1301] \"Pseudomonas umsongensis\"                               \n",
       "[1302] \"Pseudomonas vancouverensis\"                            \n",
       "[1303] \"Pseudomonas veronii\"                                   \n",
       "[1304] \"Pseudomonas versuta\"                                   \n",
       "[1305] \"Pseudomonas viridiflava\"                               \n",
       "[1306] \"Pseudomonas xanthomarina\"                              \n",
       "[1307] \"Pseudomonas xinjiangensis\"                             \n",
       "[1308] \"Pseudomonas yamanorum\"                                 \n",
       "[1309] \"Pseudonocardia autotrophica\"                           \n",
       "[1310] \"Pseudonocardia dioxanivorans\"                          \n",
       "[1311] \"Pseudonocardia sp. AL041005-10\"                        \n",
       "[1312] \"Pseudonocardia sp. HH130629-09\"                        \n",
       "[1313] \"Pseudonocardia sp. HH130630-07\"                        \n",
       "[1314] \"Pseudopropionibacterium propionicum\"                   \n",
       "[1315] \"Pseudorhodoplanes sinuspersici\"                        \n",
       "[1316] \"Pseudovibrio sp. FO-BEG1\"                              \n",
       "[1317] \"Pseudoxanthomonas spadix\"                              \n",
       "[1318] \"Pseudoxanthomonas suwonensis\"                          \n",
       "[1319] \"Pusillimonas sp. T7-7\"                                 \n",
       "[1320] \"Rahnella aquatilis\"                                    \n",
       "[1321] \"Ralstonia insidiosa\"                                   \n",
       "[1322] \"Ralstonia mannitolilytica\"                             \n",
       "[1323] \"Ralstonia pickettii\"                                   \n",
       "[1324] \"Ralstonia pseudosolanacearum\"                          \n",
       "[1325] \"Ralstonia solanacearum\"                                \n",
       "[1326] \"Ramlibacter tataouinensis\"                             \n",
       "[1327] \"Raoultella ornithinolytica\"                            \n",
       "[1328] \"Raoultella planticola\"                                 \n",
       "[1329] \"Rathayibacter toxicus\"                                 \n",
       "[1330] \"Rathayibacter tritici\"                                 \n",
       "[1331] \"Rhizobacter gummiphilus\"                               \n",
       "[1332] \"Rhizobium etli\"                                        \n",
       "[1333] \"Rhizobium favelukesii\"                                 \n",
       "[1334] \"Rhizobium gallicum\"                                    \n",
       "[1335] \"Rhizobium jaguaris\"                                    \n",
       "[1336] \"Rhizobium leguminosarum\"                               \n",
       "[1337] \"Rhizobium phaseoli\"                                    \n",
       "[1338] \"Rhizobium sp. 11515TR\"                                 \n",
       "[1339] \"Rhizobium sp. ACO-34A\"                                 \n",
       "[1340] \"Rhizobium sp. CCGE531\"                                 \n",
       "[1341] \"Rhizobium sp. CIAT894\"                                 \n",
       "[1342] \"Rhizobium sp. IE4771\"                                  \n",
       "[1343] \"Rhizobium sp. IRBG74\"                                  \n",
       "[1344] \"Rhizobium sp. Kim5\"                                    \n",
       "[1345] \"Rhizobium sp. NT-26\"                                   \n",
       "[1346] \"Rhizobium sp. NXC14\"                                   \n",
       "[1347] \"Rhizobium sp. NXC24\"                                   \n",
       "[1348] \"Rhizobium sp. S41\"                                     \n",
       "[1349] \"Rhizobium sp. Y9\"                                      \n",
       "[1350] \"Rhizobium tropici\"                                     \n",
       "[1351] \"Rhizorhabdus dicambivorans\"                            \n",
       "[1352] \"Rhodanobacter denitrificans\"                           \n",
       "[1353] \"Rhodanobacteraceae bacterium Dysh456\"                  \n",
       "[1354] \"Rhodobaca barguzinensis\"                               \n",
       "[1355] \"Rhodobacter blasticus\"                                 \n",
       "[1356] \"Rhodobacter capsulatus\"                                \n",
       "[1357] \"Rhodobacter sp. CZR27\"                                 \n",
       "[1358] \"Rhodobacter sp. LPB0142\"                               \n",
       "[1359] \"Rhodobacter sphaeroides\"                               \n",
       "[1360] \"Rhodobacteraceae bacterium\"                            \n",
       "[1361] \"Rhodobacteraceae bacterium BAR1\"                       \n",
       "[1362] \"Rhodobacteraceae bacterium QY30\"                       \n",
       "[1363] \"Rhodobiaceae bacterium\"                                \n",
       "[1364] \"Rhodococcus aetherivorans\"                             \n",
       "[1365] \"Rhodococcus biphenylivorans\"                           \n",
       "[1366] \"Rhodococcus coprophilus\"                               \n",
       "[1367] \"Rhodococcus erythropolis\"                              \n",
       "[1368] \"Rhodococcus fascians\"                                  \n",
       "[1369] \"Rhodococcus hoagii\"                                    \n",
       "[1370] \"Rhodococcus jostii\"                                    \n",
       "[1371] \"Rhodococcus opacus\"                                    \n",
       "[1372] \"Rhodococcus pyridinivorans\"                            \n",
       "[1373] \"Rhodococcus qingshengii\"                               \n",
       "[1374] \"Rhodococcus rhodochrous\"                               \n",
       "[1375] \"Rhodococcus ruber\"                                     \n",
       "[1376] \"Rhodococcus sp. 008\"                                   \n",
       "[1377] \"Rhodococcus sp. 2G\"                                    \n",
       "[1378] \"Rhodococcus sp. B7740\"                                 \n",
       "[1379] \"Rhodococcus sp. BH4\"                                   \n",
       "[1380] \"Rhodococcus sp. H-CA8f\"                                \n",
       "[1381] \"Rhodococcus sp. MTM3W5.2\"                              \n",
       "[1382] \"Rhodococcus sp. NJ-530\"                                \n",
       "[1383] \"Rhodococcus sp. P1Y\"                                   \n",
       "[1384] \"Rhodococcus sp. PBTS 1\"                                \n",
       "[1385] \"Rhodococcus sp. PBTS 2\"                                \n",
       "[1386] \"Rhodococcus sp. S2-17\"                                 \n",
       "[1387] \"Rhodococcus sp. WMMA185\"                               \n",
       "[1388] \"Rhodococcus sp. YL-1\"                                  \n",
       "[1389] \"Rhodocyclaceae bacterium\"                              \n",
       "[1390] \"Rhodoferax antarcticus\"                                \n",
       "[1391] \"Rhodoferax ferrireducens\"                              \n",
       "[1392] \"Rhodoferax koreense\"                                   \n",
       "[1393] \"Rhodoferax saidenbachensis\"                            \n",
       "[1394] \"Rhodomicrobium vannielii\"                              \n",
       "[1395] \"Rhodoplanes sp. Z2-YC6860\"                             \n",
       "[1396] \"Rhodospirillum centenum\"                               \n",
       "[1397] \"Rhodospirillum rubrum\"                                 \n",
       "[1398] \"Rhodothermaceae bacterium\"                             \n",
       "[1399] \"Rhodothermus marinus\"                                  \n",
       "[1400] \"Rhodovulum sp. MB263\"                                  \n",
       "[1401] \"Rhodovulum sp. P5\"                                     \n",
       "[1402] \"Rhodovulum sulfidophilum\"                              \n",
       "[1403] \"Roseateles depolymerans\"                               \n",
       "[1404] \"Roseibacterium elongatum\"                              \n",
       "[1405] \"Roseiflexus sp. RS-1\"                                  \n",
       "[1406] \"Roseobacter denitrificans\"                             \n",
       "[1407] \"Roseobacter litoralis\"                                 \n",
       "[1408] \"Roseomonas gilardii\"                                   \n",
       "[1409] \"Roseomonas sp. FDAARGOS_362\"                           \n",
       "[1410] \"Roseovarius mucosus\"                                   \n",
       "[1411] \"Rothia dentocariosa\"                                   \n",
       "[1412] \"Rothia mucilaginosa\"                                   \n",
       "[1413] \"Rubinisphaera brasiliensis\"                            \n",
       "[1414] \"Rubrivivax gelatinosus\"                                \n",
       "[1415] \"Rubrobacter radiotolerans\"                             \n",
       "[1416] \"Rubrobacter xylanophilus\"                              \n",
       "[1417] \"Ruegeria pomeroyi\"                                     \n",
       "[1418] \"Ruegeria sp. AD91A\"                                    \n",
       "[1419] \"Ruegeria sp. TM1040\"                                   \n",
       "[1420] \"Saccharomonospora azurea\"                              \n",
       "[1421] \"Saccharomonospora cyanea\"                              \n",
       "[1422] \"Saccharomonospora glauca\"                              \n",
       "[1423] \"Saccharomonospora marina\"                              \n",
       "[1424] \"Saccharomonospora viridis\"                             \n",
       "[1425] \"Saccharopolyspora erythraea\"                           \n",
       "[1426] \"Saccharospirillum mangrovi\"                            \n",
       "[1427] \"Saccharothrix espanaensis\"                             \n",
       "[1428] \"Sagittula sp. P11\"                                     \n",
       "[1429] \"Salinibacter ruber\"                                    \n",
       "[1430] \"Salinibacterium sp. CGMCC 1.16371\"                     \n",
       "[1431] \"Salinicola tamaricis\"                                  \n",
       "[1432] \"Salinisphaera sp. LB1\"                                 \n",
       "[1433] \"Salinispora arenicola\"                                 \n",
       "[1434] \"Salinispora tropica\"                                   \n",
       "[1435] \"Salipiger profundus\"                                   \n",
       "[1436] \"Salmonella enterica\"                                   \n",
       "[1437] \"Sandaracinus amylolyticus\"                             \n",
       "[1438] \"Sanguibacter keddieii\"                                 \n",
       "[1439] \"Sedimenticola thiotaurini\"                             \n",
       "[1440] \"Segniliparus rotundus\"                                 \n",
       "[1441] \"Selenomonas sputigena\"                                 \n",
       "[1442] \"Serinicoccus sp. JLT9\"                                 \n",
       "[1443] \"Serpentinomonas mccroryi\"                              \n",
       "[1444] \"Serpentinomonas raichei\"                               \n",
       "[1445] \"Serratia ficaria\"                                      \n",
       "[1446] \"Serratia liquefaciens\"                                 \n",
       "[1447] \"Serratia marcescens\"                                   \n",
       "[1448] \"Serratia plymuthica\"                                   \n",
       "[1449] \"Serratia sp. FDAARGOS_506\"                             \n",
       "[1450] \"Serratia sp. YD25\"                                     \n",
       "[1451] \"Shewanella algae\"                                      \n",
       "[1452] \"Shewanella amazonensis\"                                \n",
       "[1453] \"Shewanella loihica\"                                    \n",
       "[1454] \"Shewanella putrefaciens\"                               \n",
       "[1455] \"Shewanella sp. TH2012\"                                 \n",
       "[1456] \"Shinella sp. HZN7\"                                     \n",
       "[1457] \"Sideroxydans lithotrophicus\"                           \n",
       "[1458] \"Silicimonas algicola\"                                  \n",
       "[1459] \"Simiduia agarivorans\"                                  \n",
       "[1460] \"Simplicispira suum\"                                    \n",
       "[1461] \"Singulisphaera acidiphila\"                             \n",
       "[1462] \"Sinomonas atrocyanea\"                                  \n",
       "[1463] \"Sinorhizobium americanum\"                              \n",
       "[1464] \"Sinorhizobium fredii\"                                  \n",
       "[1465] \"Sinorhizobium medicae\"                                 \n",
       "[1466] \"Sinorhizobium meliloti\"                                \n",
       "[1467] \"Sinorhizobium sp. CCBAU 05631\"                         \n",
       "[1468] \"Sinorhizobium sp. RAC02\"                               \n",
       "[1469] \"Sodalis praecaptivus\"                                  \n",
       "[1470] \"Solimonas sp. K1W22B-7\"                                \n",
       "[1471] \"Sorangium cellulosum\"                                  \n",
       "[1472] \"Sphaerobacter thermophilus\"                            \n",
       "[1473] \"Sphingobacterium sp. ML3W\"                             \n",
       "[1474] \"Sphingobium amiense\"                                   \n",
       "[1475] \"Sphingobium baderi\"                                    \n",
       "[1476] \"Sphingobium chlorophenolicum\"                          \n",
       "[1477] \"Sphingobium cloacae\"                                   \n",
       "[1478] \"Sphingobium herbicidovorans\"                           \n",
       "[1479] \"Sphingobium hydrophobicum\"                             \n",
       "[1480] \"Sphingobium indicum\"                                   \n",
       "[1481] \"Sphingobium japonicum\"                                 \n",
       "[1482] \"Sphingobium sp. EP60837\"                               \n",
       "[1483] \"Sphingobium sp. LF-16\"                                 \n",
       "[1484] \"Sphingobium sp. MI1205\"                                \n",
       "[1485] \"Sphingobium sp. RAC03\"                                 \n",
       "[1486] \"Sphingobium sp. SCG-1\"                                 \n",
       "[1487] \"Sphingobium sp. SYK-6\"                                 \n",
       "[1488] \"Sphingobium sp. TKS\"                                   \n",
       "[1489] \"Sphingobium sp. YBL2\"                                  \n",
       "[1490] \"Sphingobium sp. YG1\"                                   \n",
       "[1491] \"Sphingobium yanoikuyae\"                                \n",
       "[1492] \"Sphingomonas hengshuiensis\"                            \n",
       "[1493] \"Sphingomonas indica\"                                   \n",
       "[1494] \"Sphingomonas koreensis\"                                \n",
       "[1495] \"Sphingomonas melonis\"                                  \n",
       "[1496] \"Sphingomonas panacis\"                                  \n",
       "[1497] \"Sphingomonas sanxanigenens\"                            \n",
       "[1498] \"Sphingomonas sp. C8-2\"                                 \n",
       "[1499] \"Sphingomonas sp. Cra20\"                                \n",
       "[1500] \"Sphingomonas sp. FARSPH\"                               \n",
       "[1501] \"Sphingomonas sp. JJ-A5\"                                \n",
       "[1502] \"Sphingomonas sp. KC8\"                                  \n",
       "[1503] \"Sphingomonas sp. LK11\"                                 \n",
       "[1504] \"Sphingomonas sp. LM7\"                                  \n",
       "[1505] \"Sphingomonas sp. MM-1\"                                 \n",
       "[1506] \"Sphingomonas sp. NIC1\"                                 \n",
       "[1507] \"Sphingomonas sp. YZ-8\"                                 \n",
       "[1508] \"Sphingomonas taxi\"                                     \n",
       "[1509] \"Sphingomonas wittichii\"                                \n",
       "[1510] \"Sphingopyxis alaskensis\"                               \n",
       "[1511] \"Sphingopyxis fribergensis\"                             \n",
       "[1512] \"Sphingopyxis granuli\"                                  \n",
       "[1513] \"Sphingopyxis macrogoltabida\"                           \n",
       "[1514] \"Sphingopyxis sp. 113P3\"                                \n",
       "[1515] \"Sphingopyxis sp. EG6\"                                  \n",
       "[1516] \"Sphingopyxis sp. FD7\"                                  \n",
       "[1517] \"Sphingopyxis sp. LPB0140\"                              \n",
       "[1518] \"Sphingopyxis sp. MG\"                                   \n",
       "[1519] \"Sphingopyxis sp. QXT-31\"                               \n",
       "[1520] \"Sphingopyxis sp. WS5A3p\"                               \n",
       "[1521] \"Sphingopyxis terrae\"                                   \n",
       "[1522] \"Sphingorhabdus flavimaris\"                             \n",
       "[1523] \"Sphingorhabdus sp. Alg231-15\"                          \n",
       "[1524] \"Sphingorhabdus sp. M41\"                                \n",
       "[1525] \"Sphingorhabdus sp. YGSMI21\"                            \n",
       "[1526] \"Sphingosinicella microcystinivorans\"                   \n",
       "[1527] \"Spiribacter curvatus\"                                  \n",
       "[1528] \"Spiribacter salinus\"                                   \n",
       "[1529] \"Spirosoma radiotolerans\"                               \n",
       "[1530] \"Spongiibacter sp. IMCC21906\"                           \n",
       "[1531] \"Stackebrandtia nassauensis\"                            \n",
       "[1532] \"Staphylococcus aureus\"                                 \n",
       "[1533] \"Staphylococcus capitis\"                                \n",
       "[1534] \"Staphylococcus cohnii\"                                 \n",
       "[1535] \"Staphylococcus epidermidis\"                            \n",
       "[1536] \"Staphylococcus haemolyticus\"                           \n",
       "[1537] \"Staphylococcus hominis\"                                \n",
       "[1538] \"Stappia sp. ES.058\"                                    \n",
       "[1539] \"Starkeya novella\"                                      \n",
       "[1540] \"Stenotrophomonas acidaminiphila\"                       \n",
       "[1541] \"Stenotrophomonas maltophilia\"                          \n",
       "[1542] \"Stenotrophomonas rhizophila\"                           \n",
       "[1543] \"Stenotrophomonas sp. ESTM1D_MKCIP4_1\"                  \n",
       "[1544] \"Stenotrophomonas sp. G4\"                               \n",
       "[1545] \"Stenotrophomonas sp. LM091\"                            \n",
       "[1546] \"Stenotrophomonas sp. MYb57\"                            \n",
       "[1547] \"Stenotrophomonas sp. Pemsol\"                           \n",
       "[1548] \"Stenotrophomonas sp. SAU14A_NAIMI4_5\"                  \n",
       "[1549] \"Stenotrophomonas sp. SAU14A_NAIMI4_8\"                  \n",
       "[1550] \"Stenotrophomonas sp. WZN-1\"                            \n",
       "[1551] \"Stenotrophomonas sp. YAU14D1_LEIMI4_1\"                 \n",
       "[1552] \"Stenotrophomonas sp. ZAC14D2_NAIMI4_6\"                 \n",
       "[1553] \"Stenotrophomonas sp. ZAC14D2_NAIMI4_7\"                 \n",
       "[1554] \"Steroidobacter denitrificans\"                          \n",
       "[1555] \"Sterolibacteriaceae bacterium J5B\"                     \n",
       "[1556] \"Stigmatella aurantiaca\"                                \n",
       "[1557] \"Streptacidiphilus sp. DSM 106435\"                      \n",
       "[1558] \"Streptococcus mitis\"                                   \n",
       "[1559] \"Streptococcus oralis\"                                  \n",
       "[1560] \"Streptococcus pneumoniae\"                              \n",
       "[1561] \"Streptomyces actuosus\"                                 \n",
       "[1562] \"Streptomyces albireticuli\"                             \n",
       "[1563] \"Streptomyces alboflavus\"                               \n",
       "[1564] \"Streptomyces albulus\"                                  \n",
       "[1565] \"Streptomyces albus\"                                    \n",
       "[1566] \"Streptomyces ambofaciens\"                              \n",
       "[1567] \"Streptomyces antibioticus\"                             \n",
       "[1568] \"Streptomyces anulatus\"                                 \n",
       "[1569] \"Streptomyces atratus\"                                  \n",
       "[1570] \"Streptomyces avermitilis\"                              \n",
       "[1571] \"Streptomyces bacillaris\"                               \n",
       "[1572] \"Streptomyces bingchenggensis\"                          \n",
       "[1573] \"Streptomyces brunneus\"                                 \n",
       "[1574] \"Streptomyces cattleya\"                                 \n",
       "[1575] \"Streptomyces chartreusis\"                              \n",
       "[1576] \"Streptomyces clavuligerus\"                             \n",
       "[1577] \"Streptomyces collinus\"                                 \n",
       "[1578] \"Streptomyces cyaneogriseus\"                            \n",
       "[1579] \"Streptomyces davaonensis\"                              \n",
       "[1580] \"Streptomyces dengpaensis\"                              \n",
       "[1581] \"Streptomyces formicae\"                                 \n",
       "[1582] \"Streptomyces fulvissimus\"                              \n",
       "[1583] \"Streptomyces fungicidicus\"                             \n",
       "[1584] \"Streptomyces gilvosporeus\"                             \n",
       "[1585] \"Streptomyces glaucescens\"                              \n",
       "[1586] \"Streptomyces globisporus\"                              \n",
       "[1587] \"Streptomyces globosus\"                                 \n",
       "[1588] \"Streptomyces griseochromogenes\"                        \n",
       "[1589] \"Streptomyces griseorubiginosus\"                        \n",
       "[1590] \"Streptomyces griseus\"                                  \n",
       "[1591] \"Streptomyces hundungensis\"                             \n",
       "[1592] \"Streptomyces hygroscopicus\"                            \n",
       "[1593] \"Streptomyces katrae\"                                   \n",
       "[1594] \"Streptomyces koyangensis\"                              \n",
       "[1595] \"Streptomyces lavendulae\"                               \n",
       "[1596] \"Streptomyces leeuwenhoekii\"                            \n",
       "[1597] \"Streptomyces lincolnensis\"                             \n",
       "[1598] \"Streptomyces lunaelactis\"                              \n",
       "[1599] \"Streptomyces luteoverticillatus\"                       \n",
       "[1600] \"Streptomyces lydicus\"                                  \n",
       "[1601] \"Streptomyces malaysiensis\"                             \n",
       "[1602] \"Streptomyces nigra\"                                    \n",
       "[1603] \"Streptomyces niveus\"                                   \n",
       "[1604] \"Streptomyces nodosus\"                                  \n",
       "[1605] \"Streptomyces noursei\"                                  \n",
       "[1606] \"Streptomyces olivaceus\"                                \n",
       "[1607] \"Streptomyces olivoreticuli\"                            \n",
       "[1608] \"Streptomyces pactum\"                                   \n",
       "[1609] \"Streptomyces parvulus\"                                 \n",
       "[1610] \"Streptomyces peucetius\"                                \n",
       "[1611] \"Streptomyces pluripotens\"                              \n",
       "[1612] \"Streptomyces pratensis\"                                \n",
       "[1613] \"Streptomyces pristinaespiralis\"                        \n",
       "[1614] \"Streptomyces puniciscabiei\"                            \n",
       "[1615] \"Streptomyces qaidamensis\"                              \n",
       "[1616] \"Streptomyces reticuli\"                                 \n",
       "[1617] \"Streptomyces roseochromogenus\"                         \n",
       "[1618] \"Streptomyces rubrolavendulae\"                          \n",
       "[1619] \"Streptomyces scabiei\"                                  \n",
       "[1620] \"Streptomyces sp. 11-1-2\"                               \n",
       "[1621] \"Streptomyces sp. 2323.1\"                               \n",
       "[1622] \"Streptomyces sp. 3211\"                                 \n",
       "[1623] \"Streptomyces sp. 3214.6\"                               \n",
       "[1624] \"Streptomyces sp. 4F\"                                   \n",
       "[1625] \"Streptomyces sp. 769\"                                  \n",
       "[1626] \"Streptomyces sp. ADI95-16\"                             \n",
       "[1627] \"Streptomyces sp. CB09001\"                              \n",
       "[1628] \"Streptomyces sp. CCM_MD2014\"                           \n",
       "[1629] \"Streptomyces sp. CdTB01\"                               \n",
       "[1630] \"Streptomyces sp. CFMR 7\"                               \n",
       "[1631] \"Streptomyces sp. CMB-StM0423\"                          \n",
       "[1632] \"Streptomyces sp. CNQ-509\"                              \n",
       "[1633] \"Streptomyces sp. ETH9427\"                              \n",
       "[1634] \"Streptomyces sp. fd1-xmd\"                              \n",
       "[1635] \"Streptomyces sp. FR-008\"                               \n",
       "[1636] \"Streptomyces sp. Go-475\"                               \n",
       "[1637] \"Streptomyces sp. GSSD-12\"                              \n",
       "[1638] \"Streptomyces sp. HNM0039\"                              \n",
       "[1639] \"Streptomyces sp. KPB2\"                                 \n",
       "[1640] \"Streptomyces sp. M56\"                                  \n",
       "[1641] \"Streptomyces sp. Mg1\"                                  \n",
       "[1642] \"Streptomyces sp. MK45\"                                 \n",
       "[1643] \"Streptomyces sp. NEAU-S7GS2\"                           \n",
       "[1644] \"Streptomyces sp. P3\"                                   \n",
       "[1645] \"Streptomyces sp. RTd22\"                                \n",
       "[1646] \"Streptomyces sp. S8\"                                   \n",
       "[1647] \"Streptomyces sp. SAT1\"                                 \n",
       "[1648] \"Streptomyces sp. SCSIO 03032\"                          \n",
       "[1649] \"Streptomyces sp. Sge12\"                                \n",
       "[1650] \"Streptomyces sp. SirexAA-E\"                            \n",
       "[1651] \"Streptomyces sp. SM18\"                                 \n",
       "[1652] \"Streptomyces sp. TLI_053\"                              \n",
       "[1653] \"Streptomyces sp. TN58\"                                 \n",
       "[1654] \"Streptomyces sp. Tue 6075\"                             \n",
       "[1655] \"Streptomyces sp. W1SF4\"                                \n",
       "[1656] \"Streptomyces sp. WAC 01438\"                            \n",
       "[1657] \"Streptomyces sp. WAC 01529\"                            \n",
       "[1658] \"Streptomyces sp. WAC 06738\"                            \n",
       "[1659] \"Streptomyces sp. WAC00288\"                             \n",
       "[1660] \"Streptomyces sp. Z022\"                                 \n",
       "[1661] \"Streptomyces sp. ZFG47\"                                \n",
       "[1662] \"Streptomyces spongiicola\"                              \n",
       "[1663] \"Streptomyces venezuelae\"                               \n",
       "[1664] \"Streptomyces vietnamensis\"                             \n",
       "[1665] \"Streptomyces violaceoruber\"                            \n",
       "[1666] \"Streptomyces violaceusniger\"                           \n",
       "[1667] \"Streptomyces xiamenensis\"                              \n",
       "[1668] \"Streptomyces xinghaiensis\"                             \n",
       "[1669] \"Streptosporangium roseum\"                              \n",
       "[1670] \"Streptosporangium sp. 'caverna'\"                       \n",
       "[1671] \"Sulfitobacter pseudonitzschiae\"                        \n",
       "[1672] \"Sulfitobacter sp. AM1-D1\"                              \n",
       "[1673] \"Sulfitobacter sp. D7\"                                  \n",
       "[1674] \"Sulfitobacter sp. JL08\"                                \n",
       "[1675] \"Sulfitobacter sp. SK011\"                               \n",
       "[1676] \"Sulfitobacter sp. SK012\"                               \n",
       "[1677] \"Sulfitobacter sp. SK025\"                               \n",
       "[1678] \"Sulfuricaulis limicola\"                                \n",
       "[1679] \"Sulfuricella denitrificans\"                            \n",
       "[1680] \"Sulfuriferula sp. AH1\"                                 \n",
       "[1681] \"Sulfurifustis variabilis\"                              \n",
       "[1682] \"Sulfuritalea hydrogenivorans\"                          \n",
       "[1683] \"Sulfuritortus calidifontis\"                            \n",
       "[1684] \"Sulfurospirillum multivorans\"                          \n",
       "[1685] \"Sutterella megalosphaeroides\"                          \n",
       "[1686] \"Symbiobacterium thermophilum\"                          \n",
       "[1687] \"Synechococcus sp. KORDI-49\"                            \n",
       "[1688] \"Synechococcus sp. PCC 7336\"                            \n",
       "[1689] \"Synechococcus sp. RCC307\"                              \n",
       "[1690] \"Synechococcus sp. SynAce01\"                            \n",
       "[1691] \"Syntrophobotulus glycolicus\"                           \n",
       "[1692] \"Tabrizicola sp. K13M18\"                                \n",
       "[1693] \"Tateyamaria omphalii\"                                  \n",
       "[1694] \"Terriglobus roseus\"                                    \n",
       "[1695] \"Tessaracoccus aquimaris\"                               \n",
       "[1696] \"Tessaracoccus flavescens\"                              \n",
       "[1697] \"Tessaracoccus flavus\"                                  \n",
       "[1698] \"Tessaracoccus sp. Marseille-P5995\"                     \n",
       "[1699] \"Tessaracoccus sp. T2.5-30\"                             \n",
       "[1700] \"Thalassococcus sp. SH-1\"                               \n",
       "[1701] \"Thalassospira marina\"                                  \n",
       "[1702] \"Thalassospira xiamenensis\"                             \n",
       "[1703] \"Thauera aromatica\"                                     \n",
       "[1704] \"Thauera chlorobenzoica\"                                \n",
       "[1705] \"Thauera humireducens\"                                  \n",
       "[1706] \"Thauera hydrothermalis\"                                \n",
       "[1707] \"Thauera sp. K11\"                                       \n",
       "[1708] \"Thauera sp. MZ1T\"                                      \n",
       "[1709] \"Thermaerobacter marianensis\"                           \n",
       "[1710] \"Thermoanaerobacterales bacterium SK-G1\"                \n",
       "[1711] \"Thermobacillus composti\"                               \n",
       "[1712] \"Thermobifida fusca\"                                    \n",
       "[1713] \"Thermobispora bispora\"                                 \n",
       "[1714] \"Thermomonospora curvata\"                               \n",
       "[1715] \"Thermus thermophilus\"                                  \n",
       "[1716] \"Thioalkalivibrio nitratireducens\"                      \n",
       "[1717] \"Thioalkalivibrio paradoxus\"                            \n",
       "[1718] \"Thioalkalivibrio sp. K90mix\"                           \n",
       "[1719] \"Thioalkalivibrio sulfidiphilus\"                        \n",
       "[1720] \"Thioalkalivibrio versutus\"                             \n",
       "[1721] \"Thiobacillus denitrificans\"                            \n",
       "[1722] \"Thioclava nitratireducens\"                             \n",
       "[1723] \"Thiocystis violascens\"                                 \n",
       "[1724] \"Thioflavicoccus mobilis\"                               \n",
       "[1725] \"Thiohalobacter thiocyanaticus\"                         \n",
       "[1726] \"Thiolapillus brandeum\"                                 \n",
       "[1727] \"Thiomonas arsenitoxydans\"                              \n",
       "[1728] \"Thiomonas intermedia\"                                  \n",
       "[1729] \"Thiomonas sp. X19\"                                     \n",
       "[1730] \"Tistrella mobilis\"                                     \n",
       "[1731] \"Tolumonas auensis\"                                     \n",
       "[1732] \"Torque teno virus 13\"                                  \n",
       "[1733] \"Torque teno virus 15\"                                  \n",
       "[1734] \"Torque teno virus 19\"                                  \n",
       "[1735] \"Torque teno virus 24\"                                  \n",
       "[1736] \"Torque teno virus 29\"                                  \n",
       "[1737] \"Truepera radiovictrix\"                                 \n",
       "[1738] \"Trueperella pyogenes\"                                  \n",
       "[1739] \"Tsukamurella paurometabola\"                            \n",
       "[1740] \"Tsukamurella tyrosinosolvens\"                          \n",
       "[1741] \"Undibacterium parvum\"                                  \n",
       "[1742] \"Variibacter gotjawalensis\"                             \n",
       "[1743] \"Variovorax boronicumulans\"                             \n",
       "[1744] \"Variovorax paradoxus\"                                  \n",
       "[1745] \"Variovorax sp. HW608\"                                  \n",
       "[1746] \"Variovorax sp. PAMC 28711\"                             \n",
       "[1747] \"Variovorax sp. PMC12\"                                  \n",
       "[1748] \"Verminephrobacter eiseniae\"                            \n",
       "[1749] \"Verrucomicrobia bacterium\"                             \n",
       "[1750] \"Verrucomicrobia bacterium IMCC26134\"                   \n",
       "[1751] \"Verrucomicrobium sp. GAS474\"                           \n",
       "[1752] \"Verrucomicrobium spinosum\"                             \n",
       "[1753] \"Verrucosispora maris\"                                  \n",
       "[1754] \"Vibrio fluvialis\"                                      \n",
       "[1755] \"Vibrio furnissii\"                                      \n",
       "[1756] \"Vibrio vulnificus\"                                     \n",
       "[1757] \"Victivallales bacterium CCUG 44730\"                    \n",
       "[1758] \"Vitreoscilla filiformis\"                               \n",
       "[1759] \"Vogesella sp. LIG4\"                                    \n",
       "[1760] \"Vulgatibacter incomptus\"                               \n",
       "[1761] \"Wenzhouxiangella marina\"                               \n",
       "[1762] \"Xanthobacter autotrophicus\"                            \n",
       "[1763] \"Xanthomonas albilineans\"                               \n",
       "[1764] \"Xanthomonas arboricola\"                                \n",
       "[1765] \"Xanthomonas campestris\"                                \n",
       "[1766] \"Xanthomonas cassavae\"                                  \n",
       "[1767] \"Xanthomonas citri\"                                     \n",
       "[1768] \"Xanthomonas euvesicatoria\"                             \n",
       "[1769] \"Xanthomonas fragariae\"                                 \n",
       "[1770] \"Xanthomonas gardneri\"                                  \n",
       "[1771] \"Xanthomonas hortorum\"                                  \n",
       "[1772] \"Xanthomonas oryzae\"                                    \n",
       "[1773] \"Xanthomonas phaseoli\"                                  \n",
       "[1774] \"Xanthomonas sacchari\"                                  \n",
       "[1775] \"Xanthomonas translucens\"                               \n",
       "[1776] \"Xanthomonas vesicatoria\"                               \n",
       "[1777] \"Xylanimonas cellulosilytica\"                           \n",
       "[1778] \"Xylella fastidiosa\"                                    \n",
       "[1779] \"Yangia pacifica\"                                       \n",
       "[1780] \"Yangia sp. CCB-MM3\"                                    \n",
       "[1781] \"Yersinia enterocolitica\"                               \n",
       "[1782] \"Yoonia vestfoldensis\"                                  \n",
       "[1783] \"Zhihengliuella sp. ISTPL4\"                             \n",
       "[1784] \"Zobellella denitrificans\"                              \n",
       "[1785] \"Zoogloeaceae bacteirum Par-f-2\"                        \n",
       "[1786] \"Zymobacter palmae\"                                     \n",
       "[1787] \"Zymomonas mobilis\"                                     \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "full_species_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "a41e854a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Joining, by = \"npm_research_id\"\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>site_supplying_sample</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GUSTO </td><td> 965</td></tr>\n",
       "\t<tr><td>HELIOS</td><td>2285</td></tr>\n",
       "\t<tr><td>MEC   </td><td>2708</td></tr>\n",
       "\t<tr><td>PRISM </td><td>1248</td></tr>\n",
       "\t<tr><td>SERI  </td><td> 874</td></tr>\n",
       "\t<tr><td>TTSH  </td><td> 648</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " site\\_supplying\\_sample & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t GUSTO  &  965\\\\\n",
       "\t HELIOS & 2285\\\\\n",
       "\t MEC    & 2708\\\\\n",
       "\t PRISM  & 1248\\\\\n",
       "\t SERI   &  874\\\\\n",
       "\t TTSH   &  648\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| site_supplying_sample &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| GUSTO  |  965 |\n",
       "| HELIOS | 2285 |\n",
       "| MEC    | 2708 |\n",
       "| PRISM  | 1248 |\n",
       "| SERI   |  874 |\n",
       "| TTSH   |  648 |\n",
       "\n"
      ],
      "text/plain": [
       "  site_supplying_sample n   \n",
       "1 GUSTO                  965\n",
       "2 HELIOS                2285\n",
       "3 MEC                   2708\n",
       "4 PRISM                 1248\n",
       "5 SERI                   874\n",
       "6 TTSH                   648"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>fold_diff</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>35.65385</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 1\n",
       "\\begin{tabular}{l}\n",
       " fold\\_diff\\\\\n",
       " <dbl>\\\\\n",
       "\\hline\n",
       "\t 35.65385\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 1\n",
       "\n",
       "| fold_diff &lt;dbl&gt; |\n",
       "|---|\n",
       "| 35.65385 |\n",
       "\n"
      ],
      "text/plain": [
       "  fold_diff\n",
       "1 35.65385 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", species_df)\n",
    "prev_df <- species_df %>% select(-npm_research_id)\n",
    "prev_df[prev_df <= 0] <- 0\n",
    "prev_df[prev_df > 0] <- 1\n",
    "prev_df <- prev_df %>% \n",
    "    add_column(npm_research_id = species_df$npm_research_id, .before = 1) %>%\n",
    "    left_join(meta)\n",
    "\n",
    "level_counts <- prev_df %>% count(site_supplying_sample)\n",
    "level_counts\n",
    "prev_df %>%\n",
    "    group_by(site_supplying_sample) %>%\n",
    "    summarise(prev = sum(get(\"Nocardia brasiliensis\")) / n()) %>%\n",
    "    summarise(fold_diff = max(prev) / min(prev))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "68ee8294",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 965 × 70</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>npm_research_id</th><th scope=col>multiplex_pool_id</th><th scope=col>supplier_id</th><th scope=col>gis_internal_sample_id</th><th scope=col>site_supplying_sample</th><th scope=col>year_of_birth</th><th scope=col>supplied_gender</th><th scope=col>self_reported_ethnicity</th><th scope=col>extraction_kit</th><th scope=col>date_of_dna_extraction</th><th scope=col>⋯</th><th scope=col>sequencing_complete</th><th scope=col>supplied_and_computed_gender_match</th><th scope=col>sop_agreement_for_coverage__14/_28_for_15x/30x</th><th scope=col>sample_life_cycle</th><th scope=col>current</th><th scope=col>source_cohort</th><th scope=col>original_supplier_id</th><th scope=col>duplicate_info</th><th scope=col>duplicate</th><th scope=col>duplicate_pair</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>WHB3854</td><td>MUX8960</td><td>020-66085-011</td><td>GUSTO_Kids_020_66085_011_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>1/11/2018 </td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-66085-011</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3855</td><td>MUX8960</td><td>010-22048    </td><td>GUSTO_Kids_010_22048_1    </td><td>GUSTO</td><td>2011</td><td>F</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-22048    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3856</td><td>MUX8960</td><td>020-04240    </td><td>GUSTO_Kids_020_04240_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-04240    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3857</td><td>MUX8960</td><td>020-66123    </td><td>GUSTO_Kids_020_66123_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-66123    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3858</td><td>MUX8960</td><td>010-20267    </td><td>GUSTO_Kids_010_20267_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20267    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3859</td><td>MUX8961</td><td>010-21795    </td><td>GUSTO_Kids_010_21795_1    </td><td>GUSTO</td><td>2011</td><td>F</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21795    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3860</td><td>MUX8961</td><td>019-30236    </td><td>GUSTO_Kids_019_30236_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>019-30236    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3861</td><td>MUX8961</td><td>020-04307    </td><td>GUSTO_Kids_020_04307_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-04307    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3862</td><td>MUX8961</td><td>020-50162    </td><td>GUSTO_Kids_020_50162_1    </td><td>GUSTO</td><td>2011</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-50162    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3863</td><td>MUX8961</td><td>020-74124    </td><td>GUSTO_Kids_020_74124_1    </td><td>GUSTO</td><td>2011</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-74124    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3864</td><td>MUX8975</td><td>019-30195    </td><td>GUSTO_Kids_019_30195_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>019-30195    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3865</td><td>MUX8974</td><td>010-20986    </td><td>GUSTO_Kids_010_20986_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20986    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3866</td><td>MUX8974</td><td>010-21050    </td><td>GUSTO_Kids_010_21050_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21050    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3867</td><td>MUX8974</td><td>010-22116    </td><td>GUSTO_Kids_010_22116_1    </td><td>GUSTO</td><td>2011</td><td>M</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-22116    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3868</td><td>MUX8974</td><td>020-08071    </td><td>GUSTO_Kids_020_08071_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-08071    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3869</td><td>MUX8974</td><td>020-66156    </td><td>GUSTO_Kids_020_66156_1    </td><td>GUSTO</td><td>2011</td><td>M</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-66156    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3870</td><td>MUX8975</td><td>010-20300    </td><td>GUSTO_Kids_010_20300_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20300    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3871</td><td>MUX8975</td><td>010-21800    </td><td>GUSTO_Kids_010_21800_1    </td><td>GUSTO</td><td>2011</td><td>M</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21800    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3872</td><td>MUX8975</td><td>019-30305    </td><td>GUSTO_Kids_019_30305_1    </td><td>GUSTO</td><td>2011</td><td>F</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>019-30305    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3873</td><td>MUX8976</td><td>020-04324    </td><td>GUSTO_Kids_020_04324_1    </td><td>GUSTO</td><td>2011</td><td>F</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-04324    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3874</td><td>MUX8975</td><td>020-50177    </td><td>GUSTO_Kids_020_50177_1    </td><td>GUSTO</td><td>2011</td><td>F</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-50177    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3875</td><td>MUX8976</td><td>010-20620    </td><td>GUSTO_Kids_010_20620_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20620    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3876</td><td>MUX8976</td><td>010-21488    </td><td>GUSTO_Kids_010_21488_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21488    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3877</td><td>MUX8976</td><td>020-04271    </td><td>GUSTO_Kids_020_04271_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-04271    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3878</td><td>MUX8976</td><td>010-21266    </td><td>GUSTO_Kids_010_21266_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21266    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3879</td><td>MUX8977</td><td>020-00033    </td><td>GUSTO_Kids_020_00033_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-00033    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3880</td><td>MUX8977</td><td>020-50006    </td><td>GUSTO_Kids_020_50006_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-50006    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3881</td><td>MUX8977</td><td>010-21897    </td><td>GUSTO_Kids_010_21897_1    </td><td>GUSTO</td><td>2011</td><td>M</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21897    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3882</td><td>MUX8977</td><td>010-20332    </td><td>GUSTO_Kids_010_20332_1    </td><td>GUSTO</td><td>2010</td><td>M</td><td>Indian </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20332    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB3883</td><td>MUX8972</td><td>010-21833    </td><td>GUSTO_Kids_010_21833_1    </td><td>GUSTO</td><td>2010</td><td>F</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>12/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21833    </td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>WHB4821</td><td>MUX9298</td><td>020-50166</td><td>GUSTO_Kids_020_50166_1</td><td>GUSTO</td><td>2011</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-50166</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4822</td><td>MUX9305</td><td>020-74144</td><td>GUSTO_Kids_020_74144_1</td><td>GUSTO</td><td>2011</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-74144</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4823</td><td>MUX9298</td><td>020-00050</td><td>GUSTO_Kids_020_00050_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-00050</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4826</td><td>MUX9306</td><td>020-06006</td><td>GUSTO_Kids_020_06006_1</td><td>GUSTO</td><td>2009</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-06006</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4827</td><td>MUX9298</td><td>010-10016</td><td>GUSTO_Kids_010_10016_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-10016</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4828</td><td>MUX9307</td><td>010-20978</td><td>GUSTO_Kids_010_20978_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20978</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4829</td><td>MUX9306</td><td>020-66037</td><td>GUSTO_Kids_020_66037_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-66037</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4830</td><td>MUX9306</td><td>020-74044</td><td>GUSTO_Kids_020_74044_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-74044</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4831</td><td>MUX9305</td><td>010-21934</td><td>GUSTO_Kids_010_21934_1</td><td>GUSTO</td><td>2011</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21934</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4832</td><td>MUX9306</td><td>020-74070</td><td>GUSTO_Kids_020_74070_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-74070</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4833</td><td>MUX9307</td><td>020-74126</td><td>GUSTO_Kids_020_74126_1</td><td>GUSTO</td><td>2011</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-74126</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4834</td><td>MUX9306</td><td>020-66155</td><td>GUSTO_Kids_020_66155_1</td><td>GUSTO</td><td>2011</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-66155</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4836</td><td>MUX9241</td><td>020-74093</td><td>GUSTO_Kids_020_74093_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-74093</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4837</td><td>MUX9307</td><td>019-30332</td><td>GUSTO_Kids_019_30332_1</td><td>GUSTO</td><td>2011</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>019-30332</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4838</td><td>MUX9307</td><td>010-21851</td><td>GUSTO_Kids_010_21851_1</td><td>GUSTO</td><td>2011</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21851</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4839</td><td>MUX9308</td><td>010-21784</td><td>GUSTO_Kids_010_21784_1</td><td>GUSTO</td><td>2011</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21784</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4840</td><td>MUX9309</td><td>020-66027</td><td>GUSTO_Kids_020_66027_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-66027</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4841</td><td>MUX9308</td><td>010-21797</td><td>GUSTO_Kids_010_21797_1</td><td>GUSTO</td><td>2011</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21797</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4842</td><td>MUX9308</td><td>010-20426</td><td>GUSTO_Kids_010_20426_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20426</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4843</td><td>MUX9308</td><td>020-08039</td><td>GUSTO_Kids_020_08039_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-08039</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4844</td><td>MUX9308</td><td>020-50026</td><td>GUSTO_Kids_020_50026_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-50026</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4845</td><td>MUX9309</td><td>020-04149</td><td>GUSTO_Kids_020_04149_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-04149</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4846</td><td>MUX9309</td><td>010-20923</td><td>GUSTO_Kids_010_20923_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>N</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20923</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4847</td><td>MUX9309</td><td>020-18007</td><td>GUSTO_Kids_020_18007_1</td><td>GUSTO</td><td>2009</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-18007</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4848</td><td>MUX9309</td><td>020-50019</td><td>GUSTO_Kids_020_50019_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-50019</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4849</td><td>MUX9310</td><td>010-20031</td><td>GUSTO_Kids_010_20031_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Malay  </td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20031</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4850</td><td>MUX9310</td><td>010-14064</td><td>GUSTO_Kids_010_14064_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-14064</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4851</td><td>MUX9310</td><td>010-20002</td><td>GUSTO_Kids_010_20002_1</td><td>GUSTO</td><td>2010</td><td>F</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-20002</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4852</td><td>MUX9310</td><td>010-21279</td><td>GUSTO_Kids_010_21279_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>010-21279</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB4853</td><td>MUX9310</td><td>020-06012</td><td>GUSTO_Kids_020_06012_1</td><td>GUSTO</td><td>2010</td><td>M</td><td>Chinese</td><td>QIAsymphony DSP DNA Midi Kit</td><td>19/10/2018</td><td>⋯</td><td>Y</td><td>Y</td><td>Y</td><td>GUSTO_Kids VCF</td><td>Y</td><td>GUSTO_Kids</td><td>020-06012</td><td>Unknown</td><td>PASS</td><td>Unknown</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 965 × 70\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " npm\\_research\\_id & multiplex\\_pool\\_id & supplier\\_id & gis\\_internal\\_sample\\_id & site\\_supplying\\_sample & year\\_of\\_birth & supplied\\_gender & self\\_reported\\_ethnicity & extraction\\_kit & date\\_of\\_dna\\_extraction & ⋯ & sequencing\\_complete & supplied\\_and\\_computed\\_gender\\_match & sop\\_agreement\\_for\\_coverage\\_\\_14/\\_28\\_for\\_15x/30x & sample\\_life\\_cycle & current & source\\_cohort & original\\_supplier\\_id & duplicate\\_info & duplicate & duplicate\\_pair\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t WHB3854 & MUX8960 & 020-66085-011 & GUSTO\\_Kids\\_020\\_66085\\_011\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 1/11/2018  & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-66085-011 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3855 & MUX8960 & 010-22048     & GUSTO\\_Kids\\_010\\_22048\\_1     & GUSTO & 2011 & F & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-22048     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3856 & MUX8960 & 020-04240     & GUSTO\\_Kids\\_020\\_04240\\_1     & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-04240     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3857 & MUX8960 & 020-66123     & GUSTO\\_Kids\\_020\\_66123\\_1     & GUSTO & 2010 & M & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-66123     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3858 & MUX8960 & 010-20267     & GUSTO\\_Kids\\_010\\_20267\\_1     & GUSTO & 2010 & F & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20267     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3859 & MUX8961 & 010-21795     & GUSTO\\_Kids\\_010\\_21795\\_1     & GUSTO & 2011 & F & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21795     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3860 & MUX8961 & 019-30236     & GUSTO\\_Kids\\_019\\_30236\\_1     & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 019-30236     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3861 & MUX8961 & 020-04307     & GUSTO\\_Kids\\_020\\_04307\\_1     & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-04307     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3862 & MUX8961 & 020-50162     & GUSTO\\_Kids\\_020\\_50162\\_1     & GUSTO & 2011 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-50162     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3863 & MUX8961 & 020-74124     & GUSTO\\_Kids\\_020\\_74124\\_1     & GUSTO & 2011 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-74124     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3864 & MUX8975 & 019-30195     & GUSTO\\_Kids\\_019\\_30195\\_1     & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 019-30195     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3865 & MUX8974 & 010-20986     & GUSTO\\_Kids\\_010\\_20986\\_1     & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20986     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3866 & MUX8974 & 010-21050     & GUSTO\\_Kids\\_010\\_21050\\_1     & GUSTO & 2010 & F & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21050     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3867 & MUX8974 & 010-22116     & GUSTO\\_Kids\\_010\\_22116\\_1     & GUSTO & 2011 & M & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-22116     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3868 & MUX8974 & 020-08071     & GUSTO\\_Kids\\_020\\_08071\\_1     & GUSTO & 2010 & F & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-08071     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3869 & MUX8974 & 020-66156     & GUSTO\\_Kids\\_020\\_66156\\_1     & GUSTO & 2011 & M & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-66156     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3870 & MUX8975 & 010-20300     & GUSTO\\_Kids\\_010\\_20300\\_1     & GUSTO & 2010 & M & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20300     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3871 & MUX8975 & 010-21800     & GUSTO\\_Kids\\_010\\_21800\\_1     & GUSTO & 2011 & M & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21800     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3872 & MUX8975 & 019-30305     & GUSTO\\_Kids\\_019\\_30305\\_1     & GUSTO & 2011 & F & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 019-30305     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3873 & MUX8976 & 020-04324     & GUSTO\\_Kids\\_020\\_04324\\_1     & GUSTO & 2011 & F & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-04324     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3874 & MUX8975 & 020-50177     & GUSTO\\_Kids\\_020\\_50177\\_1     & GUSTO & 2011 & F & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-50177     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3875 & MUX8976 & 010-20620     & GUSTO\\_Kids\\_010\\_20620\\_1     & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20620     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3876 & MUX8976 & 010-21488     & GUSTO\\_Kids\\_010\\_21488\\_1     & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21488     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3877 & MUX8976 & 020-04271     & GUSTO\\_Kids\\_020\\_04271\\_1     & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-04271     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3878 & MUX8976 & 010-21266     & GUSTO\\_Kids\\_010\\_21266\\_1     & GUSTO & 2010 & M & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21266     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3879 & MUX8977 & 020-00033     & GUSTO\\_Kids\\_020\\_00033\\_1     & GUSTO & 2010 & M & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-00033     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3880 & MUX8977 & 020-50006     & GUSTO\\_Kids\\_020\\_50006\\_1     & GUSTO & 2010 & M & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-50006     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3881 & MUX8977 & 010-21897     & GUSTO\\_Kids\\_010\\_21897\\_1     & GUSTO & 2011 & M & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21897     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3882 & MUX8977 & 010-20332     & GUSTO\\_Kids\\_010\\_20332\\_1     & GUSTO & 2010 & M & Indian  & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20332     & Unknown & PASS & Unknown\\\\\n",
       "\t WHB3883 & MUX8972 & 010-21833     & GUSTO\\_Kids\\_010\\_21833\\_1     & GUSTO & 2010 & F & Malay   & QIAsymphony DSP DNA Midi Kit & 12/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21833     & Unknown & PASS & Unknown\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t WHB4821 & MUX9298 & 020-50166 & GUSTO\\_Kids\\_020\\_50166\\_1 & GUSTO & 2011 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-50166 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4822 & MUX9305 & 020-74144 & GUSTO\\_Kids\\_020\\_74144\\_1 & GUSTO & 2011 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-74144 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4823 & MUX9298 & 020-00050 & GUSTO\\_Kids\\_020\\_00050\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-00050 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4826 & MUX9306 & 020-06006 & GUSTO\\_Kids\\_020\\_06006\\_1 & GUSTO & 2009 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-06006 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4827 & MUX9298 & 010-10016 & GUSTO\\_Kids\\_010\\_10016\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-10016 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4828 & MUX9307 & 010-20978 & GUSTO\\_Kids\\_010\\_20978\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20978 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4829 & MUX9306 & 020-66037 & GUSTO\\_Kids\\_020\\_66037\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-66037 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4830 & MUX9306 & 020-74044 & GUSTO\\_Kids\\_020\\_74044\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-74044 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4831 & MUX9305 & 010-21934 & GUSTO\\_Kids\\_010\\_21934\\_1 & GUSTO & 2011 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21934 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4832 & MUX9306 & 020-74070 & GUSTO\\_Kids\\_020\\_74070\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-74070 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4833 & MUX9307 & 020-74126 & GUSTO\\_Kids\\_020\\_74126\\_1 & GUSTO & 2011 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-74126 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4834 & MUX9306 & 020-66155 & GUSTO\\_Kids\\_020\\_66155\\_1 & GUSTO & 2011 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-66155 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4836 & MUX9241 & 020-74093 & GUSTO\\_Kids\\_020\\_74093\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-74093 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4837 & MUX9307 & 019-30332 & GUSTO\\_Kids\\_019\\_30332\\_1 & GUSTO & 2011 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 019-30332 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4838 & MUX9307 & 010-21851 & GUSTO\\_Kids\\_010\\_21851\\_1 & GUSTO & 2011 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21851 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4839 & MUX9308 & 010-21784 & GUSTO\\_Kids\\_010\\_21784\\_1 & GUSTO & 2011 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21784 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4840 & MUX9309 & 020-66027 & GUSTO\\_Kids\\_020\\_66027\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-66027 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4841 & MUX9308 & 010-21797 & GUSTO\\_Kids\\_010\\_21797\\_1 & GUSTO & 2011 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21797 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4842 & MUX9308 & 010-20426 & GUSTO\\_Kids\\_010\\_20426\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20426 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4843 & MUX9308 & 020-08039 & GUSTO\\_Kids\\_020\\_08039\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-08039 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4844 & MUX9308 & 020-50026 & GUSTO\\_Kids\\_020\\_50026\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-50026 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4845 & MUX9309 & 020-04149 & GUSTO\\_Kids\\_020\\_04149\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-04149 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4846 & MUX9309 & 010-20923 & GUSTO\\_Kids\\_010\\_20923\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & N & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20923 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4847 & MUX9309 & 020-18007 & GUSTO\\_Kids\\_020\\_18007\\_1 & GUSTO & 2009 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-18007 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4848 & MUX9309 & 020-50019 & GUSTO\\_Kids\\_020\\_50019\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-50019 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4849 & MUX9310 & 010-20031 & GUSTO\\_Kids\\_010\\_20031\\_1 & GUSTO & 2010 & M & Malay   & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20031 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4850 & MUX9310 & 010-14064 & GUSTO\\_Kids\\_010\\_14064\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-14064 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4851 & MUX9310 & 010-20002 & GUSTO\\_Kids\\_010\\_20002\\_1 & GUSTO & 2010 & F & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-20002 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4852 & MUX9310 & 010-21279 & GUSTO\\_Kids\\_010\\_21279\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 010-21279 & Unknown & PASS & Unknown\\\\\n",
       "\t WHB4853 & MUX9310 & 020-06012 & GUSTO\\_Kids\\_020\\_06012\\_1 & GUSTO & 2010 & M & Chinese & QIAsymphony DSP DNA Midi Kit & 19/10/2018 & ⋯ & Y & Y & Y & GUSTO\\_Kids VCF & Y & GUSTO\\_Kids & 020-06012 & Unknown & PASS & Unknown\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 965 × 70\n",
       "\n",
       "| npm_research_id &lt;chr&gt; | multiplex_pool_id &lt;chr&gt; | supplier_id &lt;chr&gt; | gis_internal_sample_id &lt;chr&gt; | site_supplying_sample &lt;chr&gt; | year_of_birth &lt;int&gt; | supplied_gender &lt;chr&gt; | self_reported_ethnicity &lt;chr&gt; | extraction_kit &lt;chr&gt; | date_of_dna_extraction &lt;chr&gt; | ⋯ ⋯ | sequencing_complete &lt;chr&gt; | supplied_and_computed_gender_match &lt;chr&gt; | sop_agreement_for_coverage__14/_28_for_15x/30x &lt;chr&gt; | sample_life_cycle &lt;chr&gt; | current &lt;chr&gt; | source_cohort &lt;chr&gt; | original_supplier_id &lt;chr&gt; | duplicate_info &lt;chr&gt; | duplicate &lt;chr&gt; | duplicate_pair &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| WHB3854 | MUX8960 | 020-66085-011 | GUSTO_Kids_020_66085_011_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 1/11/2018  | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-66085-011 | Unknown | PASS | Unknown |\n",
       "| WHB3855 | MUX8960 | 010-22048     | GUSTO_Kids_010_22048_1     | GUSTO | 2011 | F | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-22048     | Unknown | PASS | Unknown |\n",
       "| WHB3856 | MUX8960 | 020-04240     | GUSTO_Kids_020_04240_1     | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-04240     | Unknown | PASS | Unknown |\n",
       "| WHB3857 | MUX8960 | 020-66123     | GUSTO_Kids_020_66123_1     | GUSTO | 2010 | M | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-66123     | Unknown | PASS | Unknown |\n",
       "| WHB3858 | MUX8960 | 010-20267     | GUSTO_Kids_010_20267_1     | GUSTO | 2010 | F | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20267     | Unknown | PASS | Unknown |\n",
       "| WHB3859 | MUX8961 | 010-21795     | GUSTO_Kids_010_21795_1     | GUSTO | 2011 | F | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21795     | Unknown | PASS | Unknown |\n",
       "| WHB3860 | MUX8961 | 019-30236     | GUSTO_Kids_019_30236_1     | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 019-30236     | Unknown | PASS | Unknown |\n",
       "| WHB3861 | MUX8961 | 020-04307     | GUSTO_Kids_020_04307_1     | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-04307     | Unknown | PASS | Unknown |\n",
       "| WHB3862 | MUX8961 | 020-50162     | GUSTO_Kids_020_50162_1     | GUSTO | 2011 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-50162     | Unknown | PASS | Unknown |\n",
       "| WHB3863 | MUX8961 | 020-74124     | GUSTO_Kids_020_74124_1     | GUSTO | 2011 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-74124     | Unknown | PASS | Unknown |\n",
       "| WHB3864 | MUX8975 | 019-30195     | GUSTO_Kids_019_30195_1     | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 019-30195     | Unknown | PASS | Unknown |\n",
       "| WHB3865 | MUX8974 | 010-20986     | GUSTO_Kids_010_20986_1     | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20986     | Unknown | PASS | Unknown |\n",
       "| WHB3866 | MUX8974 | 010-21050     | GUSTO_Kids_010_21050_1     | GUSTO | 2010 | F | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21050     | Unknown | PASS | Unknown |\n",
       "| WHB3867 | MUX8974 | 010-22116     | GUSTO_Kids_010_22116_1     | GUSTO | 2011 | M | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-22116     | Unknown | PASS | Unknown |\n",
       "| WHB3868 | MUX8974 | 020-08071     | GUSTO_Kids_020_08071_1     | GUSTO | 2010 | F | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-08071     | Unknown | PASS | Unknown |\n",
       "| WHB3869 | MUX8974 | 020-66156     | GUSTO_Kids_020_66156_1     | GUSTO | 2011 | M | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-66156     | Unknown | PASS | Unknown |\n",
       "| WHB3870 | MUX8975 | 010-20300     | GUSTO_Kids_010_20300_1     | GUSTO | 2010 | M | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20300     | Unknown | PASS | Unknown |\n",
       "| WHB3871 | MUX8975 | 010-21800     | GUSTO_Kids_010_21800_1     | GUSTO | 2011 | M | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21800     | Unknown | PASS | Unknown |\n",
       "| WHB3872 | MUX8975 | 019-30305     | GUSTO_Kids_019_30305_1     | GUSTO | 2011 | F | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 019-30305     | Unknown | PASS | Unknown |\n",
       "| WHB3873 | MUX8976 | 020-04324     | GUSTO_Kids_020_04324_1     | GUSTO | 2011 | F | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-04324     | Unknown | PASS | Unknown |\n",
       "| WHB3874 | MUX8975 | 020-50177     | GUSTO_Kids_020_50177_1     | GUSTO | 2011 | F | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-50177     | Unknown | PASS | Unknown |\n",
       "| WHB3875 | MUX8976 | 010-20620     | GUSTO_Kids_010_20620_1     | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20620     | Unknown | PASS | Unknown |\n",
       "| WHB3876 | MUX8976 | 010-21488     | GUSTO_Kids_010_21488_1     | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21488     | Unknown | PASS | Unknown |\n",
       "| WHB3877 | MUX8976 | 020-04271     | GUSTO_Kids_020_04271_1     | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-04271     | Unknown | PASS | Unknown |\n",
       "| WHB3878 | MUX8976 | 010-21266     | GUSTO_Kids_010_21266_1     | GUSTO | 2010 | M | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21266     | Unknown | PASS | Unknown |\n",
       "| WHB3879 | MUX8977 | 020-00033     | GUSTO_Kids_020_00033_1     | GUSTO | 2010 | M | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-00033     | Unknown | PASS | Unknown |\n",
       "| WHB3880 | MUX8977 | 020-50006     | GUSTO_Kids_020_50006_1     | GUSTO | 2010 | M | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-50006     | Unknown | PASS | Unknown |\n",
       "| WHB3881 | MUX8977 | 010-21897     | GUSTO_Kids_010_21897_1     | GUSTO | 2011 | M | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21897     | Unknown | PASS | Unknown |\n",
       "| WHB3882 | MUX8977 | 010-20332     | GUSTO_Kids_010_20332_1     | GUSTO | 2010 | M | Indian  | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20332     | Unknown | PASS | Unknown |\n",
       "| WHB3883 | MUX8972 | 010-21833     | GUSTO_Kids_010_21833_1     | GUSTO | 2010 | F | Malay   | QIAsymphony DSP DNA Midi Kit | 12/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21833     | Unknown | PASS | Unknown |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| WHB4821 | MUX9298 | 020-50166 | GUSTO_Kids_020_50166_1 | GUSTO | 2011 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-50166 | Unknown | PASS | Unknown |\n",
       "| WHB4822 | MUX9305 | 020-74144 | GUSTO_Kids_020_74144_1 | GUSTO | 2011 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-74144 | Unknown | PASS | Unknown |\n",
       "| WHB4823 | MUX9298 | 020-00050 | GUSTO_Kids_020_00050_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-00050 | Unknown | PASS | Unknown |\n",
       "| WHB4826 | MUX9306 | 020-06006 | GUSTO_Kids_020_06006_1 | GUSTO | 2009 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-06006 | Unknown | PASS | Unknown |\n",
       "| WHB4827 | MUX9298 | 010-10016 | GUSTO_Kids_010_10016_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-10016 | Unknown | PASS | Unknown |\n",
       "| WHB4828 | MUX9307 | 010-20978 | GUSTO_Kids_010_20978_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20978 | Unknown | PASS | Unknown |\n",
       "| WHB4829 | MUX9306 | 020-66037 | GUSTO_Kids_020_66037_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-66037 | Unknown | PASS | Unknown |\n",
       "| WHB4830 | MUX9306 | 020-74044 | GUSTO_Kids_020_74044_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-74044 | Unknown | PASS | Unknown |\n",
       "| WHB4831 | MUX9305 | 010-21934 | GUSTO_Kids_010_21934_1 | GUSTO | 2011 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21934 | Unknown | PASS | Unknown |\n",
       "| WHB4832 | MUX9306 | 020-74070 | GUSTO_Kids_020_74070_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-74070 | Unknown | PASS | Unknown |\n",
       "| WHB4833 | MUX9307 | 020-74126 | GUSTO_Kids_020_74126_1 | GUSTO | 2011 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-74126 | Unknown | PASS | Unknown |\n",
       "| WHB4834 | MUX9306 | 020-66155 | GUSTO_Kids_020_66155_1 | GUSTO | 2011 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-66155 | Unknown | PASS | Unknown |\n",
       "| WHB4836 | MUX9241 | 020-74093 | GUSTO_Kids_020_74093_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-74093 | Unknown | PASS | Unknown |\n",
       "| WHB4837 | MUX9307 | 019-30332 | GUSTO_Kids_019_30332_1 | GUSTO | 2011 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 019-30332 | Unknown | PASS | Unknown |\n",
       "| WHB4838 | MUX9307 | 010-21851 | GUSTO_Kids_010_21851_1 | GUSTO | 2011 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21851 | Unknown | PASS | Unknown |\n",
       "| WHB4839 | MUX9308 | 010-21784 | GUSTO_Kids_010_21784_1 | GUSTO | 2011 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21784 | Unknown | PASS | Unknown |\n",
       "| WHB4840 | MUX9309 | 020-66027 | GUSTO_Kids_020_66027_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-66027 | Unknown | PASS | Unknown |\n",
       "| WHB4841 | MUX9308 | 010-21797 | GUSTO_Kids_010_21797_1 | GUSTO | 2011 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21797 | Unknown | PASS | Unknown |\n",
       "| WHB4842 | MUX9308 | 010-20426 | GUSTO_Kids_010_20426_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20426 | Unknown | PASS | Unknown |\n",
       "| WHB4843 | MUX9308 | 020-08039 | GUSTO_Kids_020_08039_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-08039 | Unknown | PASS | Unknown |\n",
       "| WHB4844 | MUX9308 | 020-50026 | GUSTO_Kids_020_50026_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-50026 | Unknown | PASS | Unknown |\n",
       "| WHB4845 | MUX9309 | 020-04149 | GUSTO_Kids_020_04149_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-04149 | Unknown | PASS | Unknown |\n",
       "| WHB4846 | MUX9309 | 010-20923 | GUSTO_Kids_010_20923_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | N | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20923 | Unknown | PASS | Unknown |\n",
       "| WHB4847 | MUX9309 | 020-18007 | GUSTO_Kids_020_18007_1 | GUSTO | 2009 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-18007 | Unknown | PASS | Unknown |\n",
       "| WHB4848 | MUX9309 | 020-50019 | GUSTO_Kids_020_50019_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-50019 | Unknown | PASS | Unknown |\n",
       "| WHB4849 | MUX9310 | 010-20031 | GUSTO_Kids_010_20031_1 | GUSTO | 2010 | M | Malay   | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20031 | Unknown | PASS | Unknown |\n",
       "| WHB4850 | MUX9310 | 010-14064 | GUSTO_Kids_010_14064_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-14064 | Unknown | PASS | Unknown |\n",
       "| WHB4851 | MUX9310 | 010-20002 | GUSTO_Kids_010_20002_1 | GUSTO | 2010 | F | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-20002 | Unknown | PASS | Unknown |\n",
       "| WHB4852 | MUX9310 | 010-21279 | GUSTO_Kids_010_21279_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 010-21279 | Unknown | PASS | Unknown |\n",
       "| WHB4853 | MUX9310 | 020-06012 | GUSTO_Kids_020_06012_1 | GUSTO | 2010 | M | Chinese | QIAsymphony DSP DNA Midi Kit | 19/10/2018 | ⋯ | Y | Y | Y | GUSTO_Kids VCF | Y | GUSTO_Kids | 020-06012 | Unknown | PASS | Unknown |\n",
       "\n"
      ],
      "text/plain": [
       "    npm_research_id multiplex_pool_id supplier_id   gis_internal_sample_id    \n",
       "1   WHB3854         MUX8960           020-66085-011 GUSTO_Kids_020_66085_011_1\n",
       "2   WHB3855         MUX8960           010-22048     GUSTO_Kids_010_22048_1    \n",
       "3   WHB3856         MUX8960           020-04240     GUSTO_Kids_020_04240_1    \n",
       "4   WHB3857         MUX8960           020-66123     GUSTO_Kids_020_66123_1    \n",
       "5   WHB3858         MUX8960           010-20267     GUSTO_Kids_010_20267_1    \n",
       "6   WHB3859         MUX8961           010-21795     GUSTO_Kids_010_21795_1    \n",
       "7   WHB3860         MUX8961           019-30236     GUSTO_Kids_019_30236_1    \n",
       "8   WHB3861         MUX8961           020-04307     GUSTO_Kids_020_04307_1    \n",
       "9   WHB3862         MUX8961           020-50162     GUSTO_Kids_020_50162_1    \n",
       "10  WHB3863         MUX8961           020-74124     GUSTO_Kids_020_74124_1    \n",
       "11  WHB3864         MUX8975           019-30195     GUSTO_Kids_019_30195_1    \n",
       "12  WHB3865         MUX8974           010-20986     GUSTO_Kids_010_20986_1    \n",
       "13  WHB3866         MUX8974           010-21050     GUSTO_Kids_010_21050_1    \n",
       "14  WHB3867         MUX8974           010-22116     GUSTO_Kids_010_22116_1    \n",
       "15  WHB3868         MUX8974           020-08071     GUSTO_Kids_020_08071_1    \n",
       "16  WHB3869         MUX8974           020-66156     GUSTO_Kids_020_66156_1    \n",
       "17  WHB3870         MUX8975           010-20300     GUSTO_Kids_010_20300_1    \n",
       "18  WHB3871         MUX8975           010-21800     GUSTO_Kids_010_21800_1    \n",
       "19  WHB3872         MUX8975           019-30305     GUSTO_Kids_019_30305_1    \n",
       "20  WHB3873         MUX8976           020-04324     GUSTO_Kids_020_04324_1    \n",
       "21  WHB3874         MUX8975           020-50177     GUSTO_Kids_020_50177_1    \n",
       "22  WHB3875         MUX8976           010-20620     GUSTO_Kids_010_20620_1    \n",
       "23  WHB3876         MUX8976           010-21488     GUSTO_Kids_010_21488_1    \n",
       "24  WHB3877         MUX8976           020-04271     GUSTO_Kids_020_04271_1    \n",
       "25  WHB3878         MUX8976           010-21266     GUSTO_Kids_010_21266_1    \n",
       "26  WHB3879         MUX8977           020-00033     GUSTO_Kids_020_00033_1    \n",
       "27  WHB3880         MUX8977           020-50006     GUSTO_Kids_020_50006_1    \n",
       "28  WHB3881         MUX8977           010-21897     GUSTO_Kids_010_21897_1    \n",
       "29  WHB3882         MUX8977           010-20332     GUSTO_Kids_010_20332_1    \n",
       "30  WHB3883         MUX8972           010-21833     GUSTO_Kids_010_21833_1    \n",
       "⋮   ⋮               ⋮                 ⋮             ⋮                         \n",
       "936 WHB4821         MUX9298           020-50166     GUSTO_Kids_020_50166_1    \n",
       "937 WHB4822         MUX9305           020-74144     GUSTO_Kids_020_74144_1    \n",
       "938 WHB4823         MUX9298           020-00050     GUSTO_Kids_020_00050_1    \n",
       "939 WHB4826         MUX9306           020-06006     GUSTO_Kids_020_06006_1    \n",
       "940 WHB4827         MUX9298           010-10016     GUSTO_Kids_010_10016_1    \n",
       "941 WHB4828         MUX9307           010-20978     GUSTO_Kids_010_20978_1    \n",
       "942 WHB4829         MUX9306           020-66037     GUSTO_Kids_020_66037_1    \n",
       "943 WHB4830         MUX9306           020-74044     GUSTO_Kids_020_74044_1    \n",
       "944 WHB4831         MUX9305           010-21934     GUSTO_Kids_010_21934_1    \n",
       "945 WHB4832         MUX9306           020-74070     GUSTO_Kids_020_74070_1    \n",
       "946 WHB4833         MUX9307           020-74126     GUSTO_Kids_020_74126_1    \n",
       "947 WHB4834         MUX9306           020-66155     GUSTO_Kids_020_66155_1    \n",
       "948 WHB4836         MUX9241           020-74093     GUSTO_Kids_020_74093_1    \n",
       "949 WHB4837         MUX9307           019-30332     GUSTO_Kids_019_30332_1    \n",
       "950 WHB4838         MUX9307           010-21851     GUSTO_Kids_010_21851_1    \n",
       "951 WHB4839         MUX9308           010-21784     GUSTO_Kids_010_21784_1    \n",
       "952 WHB4840         MUX9309           020-66027     GUSTO_Kids_020_66027_1    \n",
       "953 WHB4841         MUX9308           010-21797     GUSTO_Kids_010_21797_1    \n",
       "954 WHB4842         MUX9308           010-20426     GUSTO_Kids_010_20426_1    \n",
       "955 WHB4843         MUX9308           020-08039     GUSTO_Kids_020_08039_1    \n",
       "956 WHB4844         MUX9308           020-50026     GUSTO_Kids_020_50026_1    \n",
       "957 WHB4845         MUX9309           020-04149     GUSTO_Kids_020_04149_1    \n",
       "958 WHB4846         MUX9309           010-20923     GUSTO_Kids_010_20923_1    \n",
       "959 WHB4847         MUX9309           020-18007     GUSTO_Kids_020_18007_1    \n",
       "960 WHB4848         MUX9309           020-50019     GUSTO_Kids_020_50019_1    \n",
       "961 WHB4849         MUX9310           010-20031     GUSTO_Kids_010_20031_1    \n",
       "962 WHB4850         MUX9310           010-14064     GUSTO_Kids_010_14064_1    \n",
       "963 WHB4851         MUX9310           010-20002     GUSTO_Kids_010_20002_1    \n",
       "964 WHB4852         MUX9310           010-21279     GUSTO_Kids_010_21279_1    \n",
       "965 WHB4853         MUX9310           020-06012     GUSTO_Kids_020_06012_1    \n",
       "    site_supplying_sample year_of_birth supplied_gender self_reported_ethnicity\n",
       "1   GUSTO                 2010          M               Chinese                \n",
       "2   GUSTO                 2011          F               Malay                  \n",
       "3   GUSTO                 2010          F               Chinese                \n",
       "4   GUSTO                 2010          M               Indian                 \n",
       "5   GUSTO                 2010          F               Malay                  \n",
       "6   GUSTO                 2011          F               Malay                  \n",
       "7   GUSTO                 2010          M               Chinese                \n",
       "8   GUSTO                 2010          M               Chinese                \n",
       "9   GUSTO                 2011          M               Chinese                \n",
       "10  GUSTO                 2011          F               Chinese                \n",
       "11  GUSTO                 2010          F               Chinese                \n",
       "12  GUSTO                 2010          F               Chinese                \n",
       "13  GUSTO                 2010          F               Indian                 \n",
       "14  GUSTO                 2011          M               Malay                  \n",
       "15  GUSTO                 2010          F               Indian                 \n",
       "16  GUSTO                 2011          M               Indian                 \n",
       "17  GUSTO                 2010          M               Malay                  \n",
       "18  GUSTO                 2011          M               Indian                 \n",
       "19  GUSTO                 2011          F               Malay                  \n",
       "20  GUSTO                 2011          F               Indian                 \n",
       "21  GUSTO                 2011          F               Indian                 \n",
       "22  GUSTO                 2010          F               Chinese                \n",
       "23  GUSTO                 2010          M               Chinese                \n",
       "24  GUSTO                 2010          F               Chinese                \n",
       "25  GUSTO                 2010          M               Indian                 \n",
       "26  GUSTO                 2010          M               Malay                  \n",
       "27  GUSTO                 2010          M               Indian                 \n",
       "28  GUSTO                 2011          M               Malay                  \n",
       "29  GUSTO                 2010          M               Indian                 \n",
       "30  GUSTO                 2010          F               Malay                  \n",
       "⋮   ⋮                     ⋮             ⋮               ⋮                      \n",
       "936 GUSTO                 2011          F               Chinese                \n",
       "937 GUSTO                 2011          M               Chinese                \n",
       "938 GUSTO                 2010          F               Chinese                \n",
       "939 GUSTO                 2009          M               Chinese                \n",
       "940 GUSTO                 2010          F               Chinese                \n",
       "941 GUSTO                 2010          F               Chinese                \n",
       "942 GUSTO                 2010          F               Chinese                \n",
       "943 GUSTO                 2010          M               Chinese                \n",
       "944 GUSTO                 2011          M               Chinese                \n",
       "945 GUSTO                 2010          M               Chinese                \n",
       "946 GUSTO                 2011          F               Chinese                \n",
       "947 GUSTO                 2011          F               Chinese                \n",
       "948 GUSTO                 2010          F               Chinese                \n",
       "949 GUSTO                 2011          M               Chinese                \n",
       "950 GUSTO                 2011          F               Chinese                \n",
       "951 GUSTO                 2011          M               Chinese                \n",
       "952 GUSTO                 2010          F               Chinese                \n",
       "953 GUSTO                 2011          F               Chinese                \n",
       "954 GUSTO                 2010          M               Chinese                \n",
       "955 GUSTO                 2010          M               Chinese                \n",
       "956 GUSTO                 2010          M               Chinese                \n",
       "957 GUSTO                 2010          M               Chinese                \n",
       "958 GUSTO                 2010          F               Chinese                \n",
       "959 GUSTO                 2009          F               Chinese                \n",
       "960 GUSTO                 2010          M               Chinese                \n",
       "961 GUSTO                 2010          M               Malay                  \n",
       "962 GUSTO                 2010          M               Chinese                \n",
       "963 GUSTO                 2010          F               Chinese                \n",
       "964 GUSTO                 2010          M               Chinese                \n",
       "965 GUSTO                 2010          M               Chinese                \n",
       "    extraction_kit               date_of_dna_extraction ⋯ sequencing_complete\n",
       "1   QIAsymphony DSP DNA Midi Kit 1/11/2018              ⋯ Y                  \n",
       "2   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "3   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "4   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "5   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "6   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "7   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "8   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "9   QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "10  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "11  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "12  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "13  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "14  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "15  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "16  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "17  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "18  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "19  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "20  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "21  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "22  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "23  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "24  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "25  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "26  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "27  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "28  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "29  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "30  QIAsymphony DSP DNA Midi Kit 12/10/2018             ⋯ Y                  \n",
       "⋮   ⋮                            ⋮                      ⋱ ⋮                  \n",
       "936 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "937 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "938 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "939 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "940 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "941 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "942 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "943 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "944 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "945 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "946 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "947 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "948 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "949 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "950 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "951 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "952 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "953 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "954 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "955 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "956 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "957 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "958 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "959 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "960 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "961 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "962 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "963 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "964 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "965 QIAsymphony DSP DNA Midi Kit 19/10/2018             ⋯ Y                  \n",
       "    supplied_and_computed_gender_match\n",
       "1   Y                                 \n",
       "2   Y                                 \n",
       "3   Y                                 \n",
       "4   Y                                 \n",
       "5   Y                                 \n",
       "6   Y                                 \n",
       "7   Y                                 \n",
       "8   Y                                 \n",
       "9   Y                                 \n",
       "10  Y                                 \n",
       "11  Y                                 \n",
       "12  Y                                 \n",
       "13  Y                                 \n",
       "14  Y                                 \n",
       "15  Y                                 \n",
       "16  Y                                 \n",
       "17  Y                                 \n",
       "18  Y                                 \n",
       "19  Y                                 \n",
       "20  Y                                 \n",
       "21  Y                                 \n",
       "22  Y                                 \n",
       "23  Y                                 \n",
       "24  Y                                 \n",
       "25  Y                                 \n",
       "26  Y                                 \n",
       "27  Y                                 \n",
       "28  Y                                 \n",
       "29  Y                                 \n",
       "30  Y                                 \n",
       "⋮   ⋮                                 \n",
       "936 Y                                 \n",
       "937 Y                                 \n",
       "938 Y                                 \n",
       "939 Y                                 \n",
       "940 Y                                 \n",
       "941 Y                                 \n",
       "942 Y                                 \n",
       "943 Y                                 \n",
       "944 Y                                 \n",
       "945 Y                                 \n",
       "946 Y                                 \n",
       "947 Y                                 \n",
       "948 Y                                 \n",
       "949 Y                                 \n",
       "950 Y                                 \n",
       "951 Y                                 \n",
       "952 Y                                 \n",
       "953 Y                                 \n",
       "954 Y                                 \n",
       "955 Y                                 \n",
       "956 Y                                 \n",
       "957 Y                                 \n",
       "958 Y                                 \n",
       "959 Y                                 \n",
       "960 Y                                 \n",
       "961 Y                                 \n",
       "962 Y                                 \n",
       "963 Y                                 \n",
       "964 Y                                 \n",
       "965 Y                                 \n",
       "    sop_agreement_for_coverage__14/_28_for_15x/30x sample_life_cycle current\n",
       "1   Y                                              GUSTO_Kids VCF    Y      \n",
       "2   Y                                              GUSTO_Kids VCF    Y      \n",
       "3   N                                              GUSTO_Kids VCF    Y      \n",
       "4   Y                                              GUSTO_Kids VCF    Y      \n",
       "5   N                                              GUSTO_Kids VCF    Y      \n",
       "6   Y                                              GUSTO_Kids VCF    Y      \n",
       "7   Y                                              GUSTO_Kids VCF    Y      \n",
       "8   Y                                              GUSTO_Kids VCF    Y      \n",
       "9   N                                              GUSTO_Kids VCF    Y      \n",
       "10  N                                              GUSTO_Kids VCF    Y      \n",
       "11  N                                              GUSTO_Kids VCF    Y      \n",
       "12  Y                                              GUSTO_Kids VCF    Y      \n",
       "13  Y                                              GUSTO_Kids VCF    Y      \n",
       "14  N                                              GUSTO_Kids VCF    Y      \n",
       "15  Y                                              GUSTO_Kids VCF    Y      \n",
       "16  Y                                              GUSTO_Kids VCF    Y      \n",
       "17  N                                              GUSTO_Kids VCF    Y      \n",
       "18  Y                                              GUSTO_Kids VCF    Y      \n",
       "19  N                                              GUSTO_Kids VCF    Y      \n",
       "20  N                                              GUSTO_Kids VCF    Y      \n",
       "21  N                                              GUSTO_Kids VCF    Y      \n",
       "22  N                                              GUSTO_Kids VCF    Y      \n",
       "23  N                                              GUSTO_Kids VCF    Y      \n",
       "24  Y                                              GUSTO_Kids VCF    Y      \n",
       "25  N                                              GUSTO_Kids VCF    Y      \n",
       "26  N                                              GUSTO_Kids VCF    Y      \n",
       "27  N                                              GUSTO_Kids VCF    Y      \n",
       "28  N                                              GUSTO_Kids VCF    Y      \n",
       "29  Y                                              GUSTO_Kids VCF    Y      \n",
       "30  N                                              GUSTO_Kids VCF    Y      \n",
       "⋮   ⋮                                              ⋮                 ⋮      \n",
       "936 Y                                              GUSTO_Kids VCF    Y      \n",
       "937 Y                                              GUSTO_Kids VCF    Y      \n",
       "938 Y                                              GUSTO_Kids VCF    Y      \n",
       "939 Y                                              GUSTO_Kids VCF    Y      \n",
       "940 N                                              GUSTO_Kids VCF    Y      \n",
       "941 Y                                              GUSTO_Kids VCF    Y      \n",
       "942 Y                                              GUSTO_Kids VCF    Y      \n",
       "943 Y                                              GUSTO_Kids VCF    Y      \n",
       "944 N                                              GUSTO_Kids VCF    Y      \n",
       "945 N                                              GUSTO_Kids VCF    Y      \n",
       "946 N                                              GUSTO_Kids VCF    Y      \n",
       "947 N                                              GUSTO_Kids VCF    Y      \n",
       "948 Y                                              GUSTO_Kids VCF    Y      \n",
       "949 Y                                              GUSTO_Kids VCF    Y      \n",
       "950 N                                              GUSTO_Kids VCF    Y      \n",
       "951 Y                                              GUSTO_Kids VCF    Y      \n",
       "952 Y                                              GUSTO_Kids VCF    Y      \n",
       "953 Y                                              GUSTO_Kids VCF    Y      \n",
       "954 Y                                              GUSTO_Kids VCF    Y      \n",
       "955 Y                                              GUSTO_Kids VCF    Y      \n",
       "956 Y                                              GUSTO_Kids VCF    Y      \n",
       "957 Y                                              GUSTO_Kids VCF    Y      \n",
       "958 N                                              GUSTO_Kids VCF    Y      \n",
       "959 Y                                              GUSTO_Kids VCF    Y      \n",
       "960 Y                                              GUSTO_Kids VCF    Y      \n",
       "961 Y                                              GUSTO_Kids VCF    Y      \n",
       "962 Y                                              GUSTO_Kids VCF    Y      \n",
       "963 Y                                              GUSTO_Kids VCF    Y      \n",
       "964 Y                                              GUSTO_Kids VCF    Y      \n",
       "965 Y                                              GUSTO_Kids VCF    Y      \n",
       "    source_cohort original_supplier_id duplicate_info duplicate duplicate_pair\n",
       "1   GUSTO_Kids    020-66085-011        Unknown        PASS      Unknown       \n",
       "2   GUSTO_Kids    010-22048            Unknown        PASS      Unknown       \n",
       "3   GUSTO_Kids    020-04240            Unknown        PASS      Unknown       \n",
       "4   GUSTO_Kids    020-66123            Unknown        PASS      Unknown       \n",
       "5   GUSTO_Kids    010-20267            Unknown        PASS      Unknown       \n",
       "6   GUSTO_Kids    010-21795            Unknown        PASS      Unknown       \n",
       "7   GUSTO_Kids    019-30236            Unknown        PASS      Unknown       \n",
       "8   GUSTO_Kids    020-04307            Unknown        PASS      Unknown       \n",
       "9   GUSTO_Kids    020-50162            Unknown        PASS      Unknown       \n",
       "10  GUSTO_Kids    020-74124            Unknown        PASS      Unknown       \n",
       "11  GUSTO_Kids    019-30195            Unknown        PASS      Unknown       \n",
       "12  GUSTO_Kids    010-20986            Unknown        PASS      Unknown       \n",
       "13  GUSTO_Kids    010-21050            Unknown        PASS      Unknown       \n",
       "14  GUSTO_Kids    010-22116            Unknown        PASS      Unknown       \n",
       "15  GUSTO_Kids    020-08071            Unknown        PASS      Unknown       \n",
       "16  GUSTO_Kids    020-66156            Unknown        PASS      Unknown       \n",
       "17  GUSTO_Kids    010-20300            Unknown        PASS      Unknown       \n",
       "18  GUSTO_Kids    010-21800            Unknown        PASS      Unknown       \n",
       "19  GUSTO_Kids    019-30305            Unknown        PASS      Unknown       \n",
       "20  GUSTO_Kids    020-04324            Unknown        PASS      Unknown       \n",
       "21  GUSTO_Kids    020-50177            Unknown        PASS      Unknown       \n",
       "22  GUSTO_Kids    010-20620            Unknown        PASS      Unknown       \n",
       "23  GUSTO_Kids    010-21488            Unknown        PASS      Unknown       \n",
       "24  GUSTO_Kids    020-04271            Unknown        PASS      Unknown       \n",
       "25  GUSTO_Kids    010-21266            Unknown        PASS      Unknown       \n",
       "26  GUSTO_Kids    020-00033            Unknown        PASS      Unknown       \n",
       "27  GUSTO_Kids    020-50006            Unknown        PASS      Unknown       \n",
       "28  GUSTO_Kids    010-21897            Unknown        PASS      Unknown       \n",
       "29  GUSTO_Kids    010-20332            Unknown        PASS      Unknown       \n",
       "30  GUSTO_Kids    010-21833            Unknown        PASS      Unknown       \n",
       "⋮   ⋮             ⋮                    ⋮              ⋮         ⋮             \n",
       "936 GUSTO_Kids    020-50166            Unknown        PASS      Unknown       \n",
       "937 GUSTO_Kids    020-74144            Unknown        PASS      Unknown       \n",
       "938 GUSTO_Kids    020-00050            Unknown        PASS      Unknown       \n",
       "939 GUSTO_Kids    020-06006            Unknown        PASS      Unknown       \n",
       "940 GUSTO_Kids    010-10016            Unknown        PASS      Unknown       \n",
       "941 GUSTO_Kids    010-20978            Unknown        PASS      Unknown       \n",
       "942 GUSTO_Kids    020-66037            Unknown        PASS      Unknown       \n",
       "943 GUSTO_Kids    020-74044            Unknown        PASS      Unknown       \n",
       "944 GUSTO_Kids    010-21934            Unknown        PASS      Unknown       \n",
       "945 GUSTO_Kids    020-74070            Unknown        PASS      Unknown       \n",
       "946 GUSTO_Kids    020-74126            Unknown        PASS      Unknown       \n",
       "947 GUSTO_Kids    020-66155            Unknown        PASS      Unknown       \n",
       "948 GUSTO_Kids    020-74093            Unknown        PASS      Unknown       \n",
       "949 GUSTO_Kids    019-30332            Unknown        PASS      Unknown       \n",
       "950 GUSTO_Kids    010-21851            Unknown        PASS      Unknown       \n",
       "951 GUSTO_Kids    010-21784            Unknown        PASS      Unknown       \n",
       "952 GUSTO_Kids    020-66027            Unknown        PASS      Unknown       \n",
       "953 GUSTO_Kids    010-21797            Unknown        PASS      Unknown       \n",
       "954 GUSTO_Kids    010-20426            Unknown        PASS      Unknown       \n",
       "955 GUSTO_Kids    020-08039            Unknown        PASS      Unknown       \n",
       "956 GUSTO_Kids    020-50026            Unknown        PASS      Unknown       \n",
       "957 GUSTO_Kids    020-04149            Unknown        PASS      Unknown       \n",
       "958 GUSTO_Kids    010-20923            Unknown        PASS      Unknown       \n",
       "959 GUSTO_Kids    020-18007            Unknown        PASS      Unknown       \n",
       "960 GUSTO_Kids    020-50019            Unknown        PASS      Unknown       \n",
       "961 GUSTO_Kids    010-20031            Unknown        PASS      Unknown       \n",
       "962 GUSTO_Kids    010-14064            Unknown        PASS      Unknown       \n",
       "963 GUSTO_Kids    010-20002            Unknown        PASS      Unknown       \n",
       "964 GUSTO_Kids    010-21279            Unknown        PASS      Unknown       \n",
       "965 GUSTO_Kids    020-06012            Unknown        PASS      Unknown       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta %>% filter(site_supplying_sample == \"GUSTO\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddfa8bd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta$hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "d3b555b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 7 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>extraction_kit</th><th scope=col>prev</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>0.8851381</td></tr>\n",
       "\t<tr><td>Chemagic DNA Blood Kit (Perkin Elmer, MA)     </td><td>0.9990385</td></tr>\n",
       "\t<tr><td>Maxwell RSC Blood DNA Kit (AS1400)            </td><td>0.9947966</td></tr>\n",
       "\t<tr><td>QIAamp DNA Blood Mini Kit                     </td><td>0.6404321</td></tr>\n",
       "\t<tr><td>QIAsymphony DSP DNA Midi Kit                  </td><td>1.0000000</td></tr>\n",
       "\t<tr><td>QIAsymphony DSP DNA Mini Kit (192)            </td><td>0.9663300</td></tr>\n",
       "\t<tr><td>Unknown                                       </td><td>0.5644567</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 7 × 2\n",
       "\\begin{tabular}{ll}\n",
       " extraction\\_kit & prev\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t abGENIX Whole Blood Genomic DNA Extraction Kit & 0.8851381\\\\\n",
       "\t Chemagic DNA Blood Kit (Perkin Elmer, MA)      & 0.9990385\\\\\n",
       "\t Maxwell RSC Blood DNA Kit (AS1400)             & 0.9947966\\\\\n",
       "\t QIAamp DNA Blood Mini Kit                      & 0.6404321\\\\\n",
       "\t QIAsymphony DSP DNA Midi Kit                   & 1.0000000\\\\\n",
       "\t QIAsymphony DSP DNA Mini Kit (192)             & 0.9663300\\\\\n",
       "\t Unknown                                        & 0.5644567\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 7 × 2\n",
       "\n",
       "| extraction_kit &lt;chr&gt; | prev &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| abGENIX Whole Blood Genomic DNA Extraction Kit | 0.8851381 |\n",
       "| Chemagic DNA Blood Kit (Perkin Elmer, MA)      | 0.9990385 |\n",
       "| Maxwell RSC Blood DNA Kit (AS1400)             | 0.9947966 |\n",
       "| QIAamp DNA Blood Mini Kit                      | 0.6404321 |\n",
       "| QIAsymphony DSP DNA Midi Kit                   | 1.0000000 |\n",
       "| QIAsymphony DSP DNA Mini Kit (192)             | 0.9663300 |\n",
       "| Unknown                                        | 0.5644567 |\n",
       "\n"
      ],
      "text/plain": [
       "  extraction_kit                                 prev     \n",
       "1 abGENIX Whole Blood Genomic DNA Extraction Kit 0.8851381\n",
       "2 Chemagic DNA Blood Kit (Perkin Elmer, MA)      0.9990385\n",
       "3 Maxwell RSC Blood DNA Kit (AS1400)             0.9947966\n",
       "4 QIAamp DNA Blood Mini Kit                      0.6404321\n",
       "5 QIAsymphony DSP DNA Midi Kit                   1.0000000\n",
       "6 QIAsymphony DSP DNA Mini Kit (192)             0.9663300\n",
       "7 Unknown                                        0.5644567"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prev_df %>%\n",
    "    group_by(extraction_kit) %>%\n",
    "    summarise(prev = sum(get(\"Bradyrhizobium sp. SK17\")) / n())\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "89dea021",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>taxa</th><th scope=col>n_diff</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Altererythrobacter sp. B11 </td><td>1</td></tr>\n",
       "\t<tr><td>Bradyrhizobium erythrophlei</td><td>1</td></tr>\n",
       "\t<tr><td>Bradyrhizobium sp. SK17    </td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 3 × 2\n",
       "\\begin{tabular}{ll}\n",
       " taxa & n\\_diff\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Altererythrobacter sp. B11  & 1\\\\\n",
       "\t Bradyrhizobium erythrophlei & 1\\\\\n",
       "\t Bradyrhizobium sp. SK17     & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 3 × 2\n",
       "\n",
       "| taxa &lt;chr&gt; | n_diff &lt;int&gt; |\n",
       "|---|---|\n",
       "| Altererythrobacter sp. B11  | 1 |\n",
       "| Bradyrhizobium erythrophlei | 1 |\n",
       "| Bradyrhizobium sp. SK17     | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  taxa                        n_diff\n",
       "1 Altererythrobacter sp. B11  1     \n",
       "2 Bradyrhizobium erythrophlei 1     \n",
       "3 Bradyrhizobium sp. SK17     1     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "full_species_df %>%\n",
    "    group_by(taxa) %>%\n",
    "    summarise(n_diff = sum(diff_prev)) %>%\n",
    "    filter(n_diff == 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56859020",
   "metadata": {},
   "source": [
    "### Correlation filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "947c6d33",
   "metadata": {},
   "outputs": [],
   "source": [
    "load_parse_data <- function(n_subset, rank, human, rel_a) {\n",
    "    # Load data\n",
    "    to_retain <- fread(\"data/samples_above_95_reads.txt\")$npm_research_id\n",
    "    df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.{rank}.tsv\")) %>% \n",
    "        filter(npm_research_id %in% to_retain)\n",
    "    meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "    \n",
    "    # Get metadata subset\n",
    "    meta_filt <- subset_metadata(meta, 9999)\n",
    "\n",
    "    # Filter data\n",
    "    df_filt <- retrieve_rows_from_meta(df, meta_filt)\n",
    "    df_filt2 <- remove_cols(df_filt, c(human, \"unclassified\"))\n",
    "    df_filt3 <- remove_low_freq_taxa(df_filt2, frac_presence = 0.05)\n",
    "    df_filt4 <- remove_empty_rows(df_filt3)\n",
    "    print(ncol(df_filt4))\n",
    "    if (rel_a) {\n",
    "        RA_df <- otu_to_RA(df_filt4)\n",
    "        return(RA_df)\n",
    "    } else {\n",
    "        return(df_filt4)\n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7bb142a9",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] 1793\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] 1793\n"
     ]
    }
   ],
   "source": [
    "species_df <- load_parse_data(9999, \"S\", \"Homo sapiens\", rel_a = F)\n",
    "species_RA <- load_parse_data(9999, \"S\", \"Homo sapiens\", rel_a = T)\n",
    "# genus_RA <- load_parse_data(9999, \"G\", \"Homo\", rel_a = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c2960354",
   "metadata": {},
   "outputs": [],
   "source": [
    "# genus_c <- full_genus_res[[\"contaminants\"]]\n",
    "# genus_nc <- full_genus_res[[\"non_contaminants\"]]\n",
    "\n",
    "# species_c <- full_species_res[[\"contaminants\"]]\n",
    "# species_nc <- full_species_res[[\"non_contaminants\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7896f157",
   "metadata": {},
   "outputs": [],
   "source": [
    "# genus_n_nc <- length(genus_nc)\n",
    "# species_n_nc <- length(species_nc)\n",
    "\n",
    "# print(str_glue(\"{genus_n_nc}/{ncol(genus_RA)} genus after differential prevalence decontamination\"))\n",
    "# print(str_glue(\"{species_n_nc}/{ncol(species_RA)} species after differential prevalence decontamination\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3f4c8c7",
   "metadata": {},
   "source": [
    "### Identify read/RA thresholds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6fe97460",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_df <- species_df %>%\n",
    "    pivot_longer(!npm_research_id, names_to = \"taxon\", values_to = \"read_count\") \n",
    "long_RA <- species_RA %>%\n",
    "    pivot_longer(!npm_research_id, names_to = \"taxon\", values_to = \"rel_a\") "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00ac4c52",
   "metadata": {},
   "source": [
    "#### Read thresholds ~ percentile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7efd5b0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "quantiles <- foreach (perc = seq(90, 100, by = 0.1)) %dopar% {\n",
    "    quant <- quantile(long_df$read_count, perc / 100)\n",
    "    return(tibble(perc = perc, quant = quant))\n",
    "}\n",
    "stopImplicitCluster()\n",
    "\n",
    "read_q_df <- bind_rows(quantiles)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "daf9fd5d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 101 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>perc</th><th scope=col>quant</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>90.0</td><td>3</td></tr>\n",
       "\t<tr><td>90.1</td><td>3</td></tr>\n",
       "\t<tr><td>90.2</td><td>4</td></tr>\n",
       "\t<tr><td>90.3</td><td>4</td></tr>\n",
       "\t<tr><td>90.4</td><td>4</td></tr>\n",
       "\t<tr><td>90.5</td><td>4</td></tr>\n",
       "\t<tr><td>90.6</td><td>4</td></tr>\n",
       "\t<tr><td>90.7</td><td>4</td></tr>\n",
       "\t<tr><td>90.8</td><td>4</td></tr>\n",
       "\t<tr><td>90.9</td><td>4</td></tr>\n",
       "\t<tr><td>91.0</td><td>4</td></tr>\n",
       "\t<tr><td>91.1</td><td>4</td></tr>\n",
       "\t<tr><td>91.2</td><td>4</td></tr>\n",
       "\t<tr><td>91.3</td><td>4</td></tr>\n",
       "\t<tr><td>91.4</td><td>4</td></tr>\n",
       "\t<tr><td>91.5</td><td>4</td></tr>\n",
       "\t<tr><td>91.6</td><td>4</td></tr>\n",
       "\t<tr><td>91.7</td><td>5</td></tr>\n",
       "\t<tr><td>91.8</td><td>5</td></tr>\n",
       "\t<tr><td>91.9</td><td>5</td></tr>\n",
       "\t<tr><td>92.0</td><td>5</td></tr>\n",
       "\t<tr><td>92.1</td><td>5</td></tr>\n",
       "\t<tr><td>92.2</td><td>5</td></tr>\n",
       "\t<tr><td>92.3</td><td>5</td></tr>\n",
       "\t<tr><td>92.4</td><td>5</td></tr>\n",
       "\t<tr><td>92.5</td><td>5</td></tr>\n",
       "\t<tr><td>92.6</td><td>5</td></tr>\n",
       "\t<tr><td>92.7</td><td>6</td></tr>\n",
       "\t<tr><td>92.8</td><td>6</td></tr>\n",
       "\t<tr><td>92.9</td><td>6</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td> 97.1</td><td>     22.000</td></tr>\n",
       "\t<tr><td> 97.2</td><td>     24.000</td></tr>\n",
       "\t<tr><td> 97.3</td><td>     25.000</td></tr>\n",
       "\t<tr><td> 97.4</td><td>     26.000</td></tr>\n",
       "\t<tr><td> 97.5</td><td>     28.000</td></tr>\n",
       "\t<tr><td> 97.6</td><td>     30.000</td></tr>\n",
       "\t<tr><td> 97.7</td><td>     32.000</td></tr>\n",
       "\t<tr><td> 97.8</td><td>     34.000</td></tr>\n",
       "\t<tr><td> 97.9</td><td>     36.000</td></tr>\n",
       "\t<tr><td> 98.0</td><td>     39.000</td></tr>\n",
       "\t<tr><td> 98.1</td><td>     42.000</td></tr>\n",
       "\t<tr><td> 98.2</td><td>     46.000</td></tr>\n",
       "\t<tr><td> 98.3</td><td>     49.000</td></tr>\n",
       "\t<tr><td> 98.4</td><td>     54.000</td></tr>\n",
       "\t<tr><td> 98.5</td><td>     59.000</td></tr>\n",
       "\t<tr><td> 98.6</td><td>     66.000</td></tr>\n",
       "\t<tr><td> 98.7</td><td>     73.000</td></tr>\n",
       "\t<tr><td> 98.8</td><td>     81.000</td></tr>\n",
       "\t<tr><td> 98.9</td><td>     92.000</td></tr>\n",
       "\t<tr><td> 99.0</td><td>    105.000</td></tr>\n",
       "\t<tr><td> 99.1</td><td>    122.000</td></tr>\n",
       "\t<tr><td> 99.2</td><td>    143.000</td></tr>\n",
       "\t<tr><td> 99.3</td><td>    171.000</td></tr>\n",
       "\t<tr><td> 99.4</td><td>    210.000</td></tr>\n",
       "\t<tr><td> 99.5</td><td>    268.000</td></tr>\n",
       "\t<tr><td> 99.6</td><td>    360.000</td></tr>\n",
       "\t<tr><td> 99.7</td><td>    526.000</td></tr>\n",
       "\t<tr><td> 99.8</td><td>    915.000</td></tr>\n",
       "\t<tr><td> 99.9</td><td>   2194.425</td></tr>\n",
       "\t<tr><td>100.0</td><td>1594156.000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 101 × 2\n",
       "\\begin{tabular}{ll}\n",
       " perc & quant\\\\\n",
       " <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 90.0 & 3\\\\\n",
       "\t 90.1 & 3\\\\\n",
       "\t 90.2 & 4\\\\\n",
       "\t 90.3 & 4\\\\\n",
       "\t 90.4 & 4\\\\\n",
       "\t 90.5 & 4\\\\\n",
       "\t 90.6 & 4\\\\\n",
       "\t 90.7 & 4\\\\\n",
       "\t 90.8 & 4\\\\\n",
       "\t 90.9 & 4\\\\\n",
       "\t 91.0 & 4\\\\\n",
       "\t 91.1 & 4\\\\\n",
       "\t 91.2 & 4\\\\\n",
       "\t 91.3 & 4\\\\\n",
       "\t 91.4 & 4\\\\\n",
       "\t 91.5 & 4\\\\\n",
       "\t 91.6 & 4\\\\\n",
       "\t 91.7 & 5\\\\\n",
       "\t 91.8 & 5\\\\\n",
       "\t 91.9 & 5\\\\\n",
       "\t 92.0 & 5\\\\\n",
       "\t 92.1 & 5\\\\\n",
       "\t 92.2 & 5\\\\\n",
       "\t 92.3 & 5\\\\\n",
       "\t 92.4 & 5\\\\\n",
       "\t 92.5 & 5\\\\\n",
       "\t 92.6 & 5\\\\\n",
       "\t 92.7 & 6\\\\\n",
       "\t 92.8 & 6\\\\\n",
       "\t 92.9 & 6\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t  97.1 &      22.000\\\\\n",
       "\t  97.2 &      24.000\\\\\n",
       "\t  97.3 &      25.000\\\\\n",
       "\t  97.4 &      26.000\\\\\n",
       "\t  97.5 &      28.000\\\\\n",
       "\t  97.6 &      30.000\\\\\n",
       "\t  97.7 &      32.000\\\\\n",
       "\t  97.8 &      34.000\\\\\n",
       "\t  97.9 &      36.000\\\\\n",
       "\t  98.0 &      39.000\\\\\n",
       "\t  98.1 &      42.000\\\\\n",
       "\t  98.2 &      46.000\\\\\n",
       "\t  98.3 &      49.000\\\\\n",
       "\t  98.4 &      54.000\\\\\n",
       "\t  98.5 &      59.000\\\\\n",
       "\t  98.6 &      66.000\\\\\n",
       "\t  98.7 &      73.000\\\\\n",
       "\t  98.8 &      81.000\\\\\n",
       "\t  98.9 &      92.000\\\\\n",
       "\t  99.0 &     105.000\\\\\n",
       "\t  99.1 &     122.000\\\\\n",
       "\t  99.2 &     143.000\\\\\n",
       "\t  99.3 &     171.000\\\\\n",
       "\t  99.4 &     210.000\\\\\n",
       "\t  99.5 &     268.000\\\\\n",
       "\t  99.6 &     360.000\\\\\n",
       "\t  99.7 &     526.000\\\\\n",
       "\t  99.8 &     915.000\\\\\n",
       "\t  99.9 &    2194.425\\\\\n",
       "\t 100.0 & 1594156.000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 101 × 2\n",
       "\n",
       "| perc &lt;dbl&gt; | quant &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| 90.0 | 3 |\n",
       "| 90.1 | 3 |\n",
       "| 90.2 | 4 |\n",
       "| 90.3 | 4 |\n",
       "| 90.4 | 4 |\n",
       "| 90.5 | 4 |\n",
       "| 90.6 | 4 |\n",
       "| 90.7 | 4 |\n",
       "| 90.8 | 4 |\n",
       "| 90.9 | 4 |\n",
       "| 91.0 | 4 |\n",
       "| 91.1 | 4 |\n",
       "| 91.2 | 4 |\n",
       "| 91.3 | 4 |\n",
       "| 91.4 | 4 |\n",
       "| 91.5 | 4 |\n",
       "| 91.6 | 4 |\n",
       "| 91.7 | 5 |\n",
       "| 91.8 | 5 |\n",
       "| 91.9 | 5 |\n",
       "| 92.0 | 5 |\n",
       "| 92.1 | 5 |\n",
       "| 92.2 | 5 |\n",
       "| 92.3 | 5 |\n",
       "| 92.4 | 5 |\n",
       "| 92.5 | 5 |\n",
       "| 92.6 | 5 |\n",
       "| 92.7 | 6 |\n",
       "| 92.8 | 6 |\n",
       "| 92.9 | 6 |\n",
       "| ⋮ | ⋮ |\n",
       "|  97.1 |      22.000 |\n",
       "|  97.2 |      24.000 |\n",
       "|  97.3 |      25.000 |\n",
       "|  97.4 |      26.000 |\n",
       "|  97.5 |      28.000 |\n",
       "|  97.6 |      30.000 |\n",
       "|  97.7 |      32.000 |\n",
       "|  97.8 |      34.000 |\n",
       "|  97.9 |      36.000 |\n",
       "|  98.0 |      39.000 |\n",
       "|  98.1 |      42.000 |\n",
       "|  98.2 |      46.000 |\n",
       "|  98.3 |      49.000 |\n",
       "|  98.4 |      54.000 |\n",
       "|  98.5 |      59.000 |\n",
       "|  98.6 |      66.000 |\n",
       "|  98.7 |      73.000 |\n",
       "|  98.8 |      81.000 |\n",
       "|  98.9 |      92.000 |\n",
       "|  99.0 |     105.000 |\n",
       "|  99.1 |     122.000 |\n",
       "|  99.2 |     143.000 |\n",
       "|  99.3 |     171.000 |\n",
       "|  99.4 |     210.000 |\n",
       "|  99.5 |     268.000 |\n",
       "|  99.6 |     360.000 |\n",
       "|  99.7 |     526.000 |\n",
       "|  99.8 |     915.000 |\n",
       "|  99.9 |    2194.425 |\n",
       "| 100.0 | 1594156.000 |\n",
       "\n"
      ],
      "text/plain": [
       "    perc  quant      \n",
       "1   90.0  3          \n",
       "2   90.1  3          \n",
       "3   90.2  4          \n",
       "4   90.3  4          \n",
       "5   90.4  4          \n",
       "6   90.5  4          \n",
       "7   90.6  4          \n",
       "8   90.7  4          \n",
       "9   90.8  4          \n",
       "10  90.9  4          \n",
       "11  91.0  4          \n",
       "12  91.1  4          \n",
       "13  91.2  4          \n",
       "14  91.3  4          \n",
       "15  91.4  4          \n",
       "16  91.5  4          \n",
       "17  91.6  4          \n",
       "18  91.7  5          \n",
       "19  91.8  5          \n",
       "20  91.9  5          \n",
       "21  92.0  5          \n",
       "22  92.1  5          \n",
       "23  92.2  5          \n",
       "24  92.3  5          \n",
       "25  92.4  5          \n",
       "26  92.5  5          \n",
       "27  92.6  5          \n",
       "28  92.7  6          \n",
       "29  92.8  6          \n",
       "30  92.9  6          \n",
       "⋮   ⋮     ⋮          \n",
       "72   97.1      22.000\n",
       "73   97.2      24.000\n",
       "74   97.3      25.000\n",
       "75   97.4      26.000\n",
       "76   97.5      28.000\n",
       "77   97.6      30.000\n",
       "78   97.7      32.000\n",
       "79   97.8      34.000\n",
       "80   97.9      36.000\n",
       "81   98.0      39.000\n",
       "82   98.1      42.000\n",
       "83   98.2      46.000\n",
       "84   98.3      49.000\n",
       "85   98.4      54.000\n",
       "86   98.5      59.000\n",
       "87   98.6      66.000\n",
       "88   98.7      73.000\n",
       "89   98.8      81.000\n",
       "90   98.9      92.000\n",
       "91   99.0     105.000\n",
       "92   99.1     122.000\n",
       "93   99.2     143.000\n",
       "94   99.3     171.000\n",
       "95   99.4     210.000\n",
       "96   99.5     268.000\n",
       "97   99.6     360.000\n",
       "98   99.7     526.000\n",
       "99   99.8     915.000\n",
       "100  99.9    2194.425\n",
       "101 100.0 1594156.000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "read_q_df %>% filter(quant != 0, quant != 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9aa26571",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 17 row(s) containing missing values (geom_path).”\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3dCZhU1Znw8cMuOu7LjFvILMYt\niQs6LogZzTbJlwYMIIjGgKifUdDMEJcZiUqIGHWIRpOZxDWZTFxGjeIWXGISNahxgjouBAgJ\nsqjQytIsvfeZqur7VrfddFfVe86BOvf+/89DU111+616bvXv6TrcW42xROSc2dYPgCgNAYnI\nQ0Ai8hCQiDwEJCIPAYnIQ0Ai8hCQiDykhFS3plQbWjeV3EbV+oYwc9e0NAUa3LQ2zNzNrevD\nDN4Y6Klb11ofZvCa5kBzG1t7v32dK6R1taWqsxtLbqNqbWOYubVtLYEGN38QZu5muzbM4A2B\nnro1tj7M4PeDPXW299vXAKl7QJKAJAFJEZAkIElAUgQkCUgSkBQBSQKSBCRFQJKAJAFJEZAk\nIElAUgQkCUgSkBQBSQKSBCRFQJKAJAFJEZAkIElAUgQkCUgSkBQBSQKSBCRFQJKAJAFJEZAk\nIElAUgQkCUgSkBQBSQKSBCRFQJKAJAFJEZAkIElAUgQkCUgSkBQBSQKSBCRFQJKAJAFJEZAk\nIElAUgQkCUgSkBQBSQKSBCRFQJKAJAFJEZAkIElAUgQkCUgSkBQBSQKSBCRFQJKAJPmDNLUm\n11hrN8weP+aqVUDSBKSkLEOa9Ehu8w+snXnR4uUzL2gFkiIgJWUZ0piXC3/V1izJ/VQaNR9I\nioCUlGFITTU3TZn07ZV23ui23GdT7819WPVUrhV1pdpsG0puo2pjc5i5dW2tgQa3bAgzt9Fu\nCjO4PtRTZ5vCDK4L9tTZ3m/fUC6kdV/5zoKFV525ce7E/GfTb8l9eGZorpdK/CAjykLFtU5Z\n/2pXf+rcuZPyFy7PQ3r7x7n+tLFUDbax5DaqNreEmbuxrTXQ4NZNYeY22c1hBjeGeupsc5jB\nm4I9dbbEHVcEyZ7/sxcLL+2m3CfXsEaqINZISRleIy29ucnazWOeWTNiUU7PyNeBpAhISRmG\nVDfhhndXzDqr3l47dfGyK6a1AUkRkJIyDMkunj7ujFnvWbvpxnFjZ3V8GZAqCEhJWYbUQ0Cq\nICAlAQlILgEpCUhAcglISUACkktASgISkFwCUhKQgOQSkJKABCSXgJQEJCC5BKQkIAHJJSAl\nbRNIy/9rnn4wkBQBSUoVpN+a8frBQFIEJClVkB42U/SDgaQISFKqIN1hrtQPBpIiIEmpgnSd\nuVk/GEiKgCSlCtKl5m79YCApApKUKkhnmyf1g4GkCEhSqiCNNPP1g4GkCEhSqiCdYJbpBwNJ\nEZCkVEE6eLDDYCApApKUKkh77O8wGEiKgCSlCdKq/kc4DAaSIiBJaYK00HzWYTCQFAFJShMk\np1PtgKQJSFKaIM1xOdUOSJqAJKUJktOpdkDSBCQpTZCcTrUDkiYgSWmCdInLqXZA0gQkKU2Q\nJrucagckTUCS0gTJ6VQ7IGkCkpQmSE6n2gFJE5CkNEE6aHuXwUBSBCQpTZCcTrUDkiYgSSmC\ntKr/kS6DgaQISFKKILmdagckTUCSUgTpeadT7YCkCUhSiiDNMVNdBgNJEZCkFEG6w1zlMhhI\nioAkpQjStU6n2gFJE5CkFEFyO9UOSJqAJKUI0mTzlMtgICkCkpQiSG6n2gFJE5CkFEEa5nSq\nHZA0AUlKESS3U+2ApAlIUoog7fERp8FAUgQkKT2QVvVzOtUOSJqAJKUHkuOpdkDSBCQpPZCe\nN6c5DQaSIiBJ6YHkeKodkDQBSUoPpNvdTrUDkiYgSemB5HiqHZA0AUlKD6RLzD1Og4GkCEhS\neiA5nmoHJE1AktID6cvm906DgaQISFJ6IJ1s/ug0GEiKgCSlB9JRfVc5DQaSIiBJ6YF0wM5u\ng4GkCEhSeiDt5XbOKpA0AUlKD6RBn3AbDCRFQJJSA2mlOcFtMJAUAUlKDaQ3zf9zGwwkRUCS\nUgNpnuPJ30DSBCQpNZB+Yc5zGwwkRUCSUgPpXnOJ22AgKQKSlBpIt5ir3QYDSRGQpNRAut7x\nXRThIa1/v1Q5SCW3UbWuMczc99taAg1uXhNmbr1dF2bwxk1h5q61DWEGf9DDU/dN81O3wc22\n99vXukJqailVq20tuY2q1rYwc1tsqMHB5gbbxaHmBtvFdstXX2qecZvb1sNgqdkVEi/tKoiX\ndklb/aXdV82v3QazRlIEJCk1kEa5/eZvIKkCkpQaSCeZJW6DgaQISFJqIA11fDsSkDQBSUoN\npL/bxXEwkBQBSUoNpD0d344EJE1AklIDaaDj25GApAlIUlogLXd9OxKQNAFJSgukN1zfjgQk\nTUCS0gLpt65vRwKSJiBJaYH0uOvbkYCkCUhSWiDd4/p2JCBpApKUFkg/cn07EpA0AUlKC6Tr\nXN+OBCRNQJLSAuly85+Og4GkCEhSWiBdaOY4DgaSIiBJaYHk/HYkIGkCkpQWSM5vRwKSJiBJ\naYHk/HYkIGkCkpQWSM5vRwKSJiBJaYHk/HYkIGkCkpQWSM5vRwKSJiBJaYHk/HYkIGkCkpQS\nSO5vRwKSJiBJKYHk/nYkIGkCkpQSSO5vRwKSJiBJKYHk/nYkIGkCkpQSSO5vRwKSJiBJKYHk\n/nYkIGkCkpQSSO5vRwKSJiBJKYHk/nYkIGkCkpQSSO5vRwKSJiBJKYHk/nYkIGkCkpQSSO5v\nRwKSJiBJKYHk/nYkIGkCkpQSSEP7ub4dCUiagCSlBNLf7eo8GEiKgCSlBNKeQ5wHA0kRkKSU\nQBr4SefBQFIEJCkdkJab4c6DgaQISFI6IL1hvuQ8GEiKgCSlA9I8M955MJAUAUlKB6RHzAXO\ng4GkCEhSOiDdaa5wHgwkRUCS0gHpenOT82AgKQKSlA5Il5m7nAcDSRGQpHRAOts84TwYSIqA\nJKUD0inm986DgaQISFI6IA03bzsPBpIiIEnpgHTwdu6DgaQISFI6IO21v/tgICkCkpQKSKv7\nH+4+GEiKgCSlAtJC82n3wUBSBCQpFZB8nGoHJE1AklIBycepdkDSBCQpFZB8nGoHJE1AklIB\nycepdkDSBCQpFZB8nGoHJE1AklIBycepdkDSBCQpFZA8/J5VIKkCkpQKSD5OtQOSJiBJqYB0\n8GAPg4GkCEhSKiDt6eFUOyBpApKUBkheTrUDkiYgSWmAtNB8xsNgr5CernnB2g2zx4+5ahWQ\nNAEpaWtC8nKqnVdIa78yOgdp5kWLl8+8oBVIioCUtDUhPezjVDuvkK658ysv2NqaJbmfSqPm\nA0kRkJK2JiQvp9r5hDTvnMYcpHmj23KXp94LJEVAStqakK43N3sY7A/ShjP/1+YgzZ2Y/2T6\nLbkPb1yaa0FDqZpsc8ltVDW1hpnbYNsCDW5rDDO3xTaFGdwc6KlrtC2BBnd/6r5pHvQwuNWW\n2KBsSDd+3xYgTcp/cnke0jNDc71U6gcZ0TZtivndVriX4j8alIL0yuRNBUgvFl7aTbkv92Hz\nily1a0q10W4uuY2q9U1h5q5pawk0uGVtmLn1ti7M4E2bwsxdbxvCDF7b/ak7xbzqYXCz7f32\ndeVCum70hAkTRpw6a82IRbmF0cjX5XrWSBXEGilpa66RvJxq52+NVJff+own19trpy5edsW0\nNiApAlLS1oTk5VQ7z2c25F7a2U03jhs7q+PLgFRBQErampC8nGrHKUKagCSlANLq/kf4GAwk\nRUCSUgDJz6l2QNIEJCkFkH7r5VQ7IGkCkpQCSA+bKT4GA0kRkKQUQLrTXOljMJAUAUlKAaTr\nvJxqByRNQJJSAOlSc7ePwUBSBCQpBZDONk/6GAwkRUCSUgDJy2+1A5IqIEkpgDTcLPUxGEiK\ngCSlAJKfU+2ApAlIUgog7eHlVDsgaQKSFD+kVX5OtQOSJiBJ8UNaaD7rZTCQFAFJih+Sp1Pt\ngKQJSFL8kOb4OdUOSJqAJMUP6Q4/p9oBSROQpPgheTrVDkiagCTFD+kSP6faAUkTkKT4IU32\nc6odkDQBSYof0kg/p9oBSROQpPghnWCWeRkMJEVAkuKHdND2fgYDSRGQpPgheTrVDkiagCRF\nD2lVvyP9DAaSIiBJ0UPydaodkDQBSYoe0vPmND+DgaQISFL0kOaYqX4GA0kRkKToId1urvIz\nGEiKgCRFD+laT6faAUkTkKToIV1i7vEzGEiKgCRFD2myecrPYCApApIUPaQRnk61A5ImIEnR\nQxrm6VQ7IGkCkhQ9pIN28DQYSIqAJEUPafePeBoMJEVAkmKH9J6vU+2ApAlIUuyQ/mA+52kw\nkBQBSYodkrdT7YCkCUhS7JAeMhd6GgwkRUCSYod0u5nhaTCQFAFJih3Steb7ngYDSRGQpNgh\nXezrVDsgaQKSFDuks8zTngYDSRGQpNghjTCveBoMJEVAkmKHdLyvU+2ApAlIUuyQDtzB12Ag\nKQKSFDuk3Yf4GgwkRUCSIof0Xr+hvgYDSRGQpMghLfB2qh2QNAFJihzSc2aCr8FAUgQkKXJI\nD/n6rXZAUgUkKXJIPzCzfA0GkiIgSZFDusz81NdgICkCkhQ5pNPNb3wNBpIiIEmRQzrRLPE1\nGEiKgCRFDumju3obDCRFQJLihrRq4Ce9DQaSIiBJcUN61XzJ22AgKQKSFDekR8zXvA0GkiIg\nSXFD8ngYCUiagCTFDcnjYSQgaQKSFDckj4eRgKQJSFLckDweRgKSJiBJcUPyeBgpPKSNdaXa\nbBtKbqNqU3OYuXVtrYEGt2wIM7fRbgozuD7QU7fRNoUZXNfpqVs38DB/c1ts77dvcIW0uWSN\ntqn0RpoaWsLM3WxbAw1urQ8zt9k2hBncFOqps6Geu7aOi4vMSH9zW0t9p7tC4qVdBfHSLmmr\nvLTzeRiJNZImIElRQ/J5GAlImoAkRQ3J52EkIGkCkhQ1JJ+HkYCkCUhS1JCGezyMBCRNQJKi\nhjRkN4+DgaQISFLMkN4bcJjHwUBSBCQpZkiveHw3EpBUAUmKGdLD5nyPg4GkCEhSzJC+b67x\nOBhIioAkxQzpUvNfHgcDSRGQpJghTTDPehwMJEVAkmKGNNz82eNgICkCkhQzpCEe340EJFVA\nkiKG5PcwEpA0AUmKGJLfw0hA0gQkKWJIfg8jAUkTkKSIIfk9jAQkTUCSIobk9zASkDQBSYoY\nkt/DSEDSBCQpYkh+DyMBSROQpIgheX03EpBUAUmKF5Lnw0hA0gQkKV5Ing8jAUkTkKR4IXk+\njAQkTUCS4oXk+TASkDQBSYoXkufDSEDSBCQpXkieDyMBSROQpHghneD3MBKQNAFJihfSR/we\nRgKSJiBJ0UJ6b8DhfgcDSRGQpGghzTc1fgcDSRGQpGghzTEX+B0MJEVAkqKFdLPnw0hA0gQk\nKVpIl5if+R0MJEVAkqKFdJrnw0hA0gQkKVpIvg8jAUkTkKRoIfk+jAQkTUCSYoXk/TASkDQB\nSYoVkvfDSEDSBCQpVkjeDyMBSROQpFgh+X43EpBUAUmKFZL3w0hA0gQkKVZI3g8jAUkTkKRY\nIXk/jAQkTUCSYoXk/TASkDQBSYoUkv/DSEDSBCQpUkj+DyMBSROQpEgh+T+MBCRNQJIiheT9\n3UhAUgUkKVJI/g8jAUkTkKRIIfk/jAQkTUCSIoU0zPthJCBpApIUKaT9vR9GApImIElxQnq3\n/xHeBwNJEZCkOCHNNyO8DwaSIiBJcUJ6yEzxPhhIioAkxQnpZvMd74OBpAhIUpyQLjZ3eR8M\nJEVAkuKENN48530wkBQBSYoT0jCz1PtgICkCkhQnpP139z8YSIqAJEUJKcRhJCBpApIUJaQQ\nh5GApAlIUpSQQhxG8ghp2YzTxl22wNoNs8ePuWoVkDQBKSkopBCHkfxBajrzuyveuWHcZjvz\nosXLZ17QCiRFQEoKCinEYSR/kNb9fLO1K2r+VFuzJPdTadR8ICkCUlJQSCEOI/ldI9X9+9ea\n5o1uy12aei+QFAEpKSikEIeRfEJqPaXmsg/s3In5y9NvyX147qRc/9NWMlt6E12hBttgg0PN\nDbaLQxVyF390zxBzSzzglgp+Ii3732+ft3HupPzFy/OQXj4j12vNpWqxrSW3UdXSFmZusw01\nONTcVtsSaHCopy7U90RuF9f3PyrEXNv77U2VvLSzLRMefbHw0m7Kfby0U8RLu6SQL+2CHEby\n99Lu1XPqcz/dTn9kzYhFOT0jXweSIiAlhYQU5DCSP0gbz/zOsndvGb3SXjt18bIrprUBSRGQ\nkkJCutlcG2Cwv39sWDpjwrhLXrN2043jxs7q+DIgVRCQkkJCutjcHWAwpwgpApIUI6Tx5vkA\ng4GkCEhSjJCON28HGAwkRUCSYoS0/x4hBgNJEZCkCCG92//IEIOBpAhIUoSQfm9GhhgMJEVA\nkiKE9KCZGmIwkBQBSYoQ0k1BDiMBSROQpAghhTmMBCRNQJIihBTmMBKQNAFJihBSmMNIQNIE\nJClCSGEOIwFJE5Ck+CA1hDmMBCRNQJLig7TIjAoyGEiKgCTFB+kpc2GQwUBSBCQpPki3muuC\nDAaSIiBJ8UH6prknyGAgKQKSFB+kr5jfBhkMJEVAkuKDNLzPsiCDgaQISFJ8kD6yZ5jBQFIE\nJCk6SO/2GxpmMJAUAUmKDtLvAx1GApImIEnRQXow0GEkIGkCkhQdpO8FOowEJE1AkqKDdHGg\nw0hA0gQkKTpI4wIdRgKSJiBJ0UE6NtBhJCBpApIUHaR9/zLMXCBpApIUG6R3+h0TZC6QVAFJ\nig3Sy2ZckLlAUgUkKTZID5hLg8wFkiogSbFButH8IMhcIKkCkhQbpGnmsSBzHSBtfu6eWtsM\nJK8BKSkUpFPNG0Hm6iFdt5MxL9h/ndTSFQ6QHAJSUihIx/apCzJXDelWM+KHOUg/6X8NkDwG\npKRQkPbdK9hTp4N02Hm2PgfJ/sshQPIYkJICQXqn31FVBmnQU+2Q5g4GkseAlBQI0svmlCqD\n9BcPt0P6yU5A8hiQkgJBesBcVGWQTj65IQ9p7cc/BySPASkpEKQbzb9VGaTf9P+br5lJX915\nwPNA8hiQkgJBmmburTJI9pdHmlx//+tSjoBUSUBKCgTpVPNCtUGytnb+K2ts6YBUQUBKCgTp\n2D7Lqw9SmQGpgoCUFAjSvnu9X02QDuwckDwGpKQwkN7pd1RVQRrWOSB5DEhJYSC9bE6pKkgV\nBaQKAlJSGEgPmIuApAhIEpAK3WiurypIB86yrJGCBKSkMJCmmXurCtIxN9hjigHJY0BKCgNp\nrHmhqiBVFJAqCEhJYSAd02d5tUEa+lb73/cdCSSPASkpDKR9/rK22iCZlwt/Nc/YHkgeA1JS\nEEhv9zu6yiCZjviJ5DMgJQWB9N/m/1cZpFe/Z0ZOznf2FcuB5DEgJQWB9HXzkyqDZO3nF5UC\nBCRFQEoKAunovgurDlL5AamCgJQUAtKygR+vrTpI7526e5/2RRKQPAakpBCQ8kukqoN0av/P\nTiyskiYDyWNASgoB6evmP6sP0h53lwIEJEVASgoBKb9EqjpI2y8FUoCAlBQA0rKBn6itPkgn\nPgKkAAEpKQCk/zbn1VYfpFf//jdA8h+QkgJAKiyRqg7SCUPM9kMKAcljQEoKAKmwRKo6SCee\n/OkkIHkMSEn+IbUvkaoOkrSx5BkOQKogICX5h9S+RKpaSE/vUwpSQ8mabHPpjTQ1toaZ22Db\nAg1uawwzt8U2hRncHOqpsy2eJ15i7iv8Heqpay31nd4DpEdPHz5s2LBjd9yjFKS6taXaaDeX\n3EZVXVOYuWvbWgINblkXZm6D3RBm8Ob6MHPrbKPniUf3XVL4u9XzXKnZ9n77+i1Dutv038/s\ns4M56dFSkHhpV0G8tEvy/tJu6YDCEqnqXtoN/WKdHbSg+T8+uwFIHgNSkndIyRKp6iDtmPtJ\nNGiBtd84H0geA1KSd0hfNz8t/F1tkHZ+LIfpd9Y++1dA8hiQkrxDOrrv4sLf1QbpmDFN9sDr\nrH1wByB5DEhJviEtHfDJ9gvVBuku83l74aALZ+xzPJA8BqQk35D+23yt/UK1QbL3XGfXfNaY\n/V8GkseAlOQbkiyRqg9SoZULmko5AlIlASnJN6SjkiVSlUIqJyBVEJCSPEMqLpGqDtLu0s5A\n8hiQkjxDuleWSFUHaWShYwcfPgVIHgNSkmdIF8kSqeogJb07/HEgeQxISZ4hFZdI1QrJzuNX\nFvsMSEl+Ib1dXCJVLaTl2wHJY0BK8gvpV+YrcrFaIX17PyB5DEhJfiH92FwuF6sN0mGFPrGH\nmQYkjwEpyS+kmeaHcrE6IR1x8qyGbnKApA9ISX4hnWsel4vVBqn8gFRBQEryC+kL5nW5WHWQ\n/vzEzx7+E5A8B6Qkv5AOHbhKLlYZpKeOKvxPFIeXPIoEpIoCUpJfSDv9bfFidUG6qc/gCTfc\n+b2JO5l/y3324GggeQpISV4hLTYnFS9XFaRX+x+7onBhzRf7vmDtT3t9cx+QKghISV4hPWPO\nLF6uKkjn7lKbXNo8ZJzddMLfAslTQEryCunOjsNI1QXpgLOLF6/c81tDzM1A8hSQkrxC+pb5\nUfFyVUHa/vrixbuM2fP7vTkCUiUBKckrpHM6DiNVF6Qdrile/MmAxxp7dQSkSgJSkldI/2je\nKF6uKkiHfrl48by/6Z0RkCoKSEleIR0yaHXxclVBurj/a8mlPwzmP2P2GZCSvELa6e86LlcV\npJU77134ld9tj+wzeAGQPAakJJ+QFnU6jFRdkOzTO5uPnDJx1P5mhzmlHAGpkoCU5BPSL81X\nOz6pLkj27Qv2M8bse/7bJR0BqZKAlOQTUufDSNUGKVfdis2lFQGpsoCU5BNS58NIVQip3IBU\nQUBK8gmp82EkIGkCkpRpSJ0PIwFJE5CkTEM6uNNhJCBpApKUaUg7HtDpEyApApKUZUgLzcmd\nPgOSIiBJWYb0tJnY6TMgKQKSlGVId5rpnT4DkiIgSVmGNMPc0ukzICkCkpRlSGebX3T6DEiK\ngCRlGdIJfRZ2+gxIioAkZRjSyu0O7PwpkBQBScowpDnmrM6fAkkRkKQMQ7rY3Nb5UyApApKU\nYUjD+rzV+VMgKQKSlF1IKwd9aIkEJE1AkrIL6aEPL5GApAlIUnYhdVkiAUkTkKTsQuqyRAKS\nJiBJmYW0ctBBH74CSIqAJGUW0kNm8oevAJIiIEmZhXSxuf3DVwBJEZCkzELqukQCkiYgSVmF\n1G2JBCRNQJKyCqnbEglImoAkZRVStyUSkDQBScoqpOO7LpGApAlIUkYhrei2RAKSJiBJGYX0\nYLclEpA0AUnKKKTuSyQgaQKSlFFI3ZdIQNIEJCmbkFYMOrjbdUBSBCQpm5AeNGd3uw5IioAk\nZRPSxeaObtdFAOmDfztj3GULrd0we/yYq1YBSROQkrxAOr7Pgm7XRQDpny5b8s51Z9TbmRct\nXj7zglYgKQJSkg9IW1oiRQCp7jsrrF1ds7C2Zknup9Ko+UBSBKQkH5B+voUlUgSQCi0YUTtv\ndFvuwtR7cx8+eCnXO+tKtcnWl9xG1YbmMHPXtbUEGtxSF2Zug90QZvDmUE+dbXSecan5Sfcr\n17c6z91yLbb32+sqgVR3/q127sT8pem35D48MzTXS2X4I/LfiX1Wld5oq1Vc65QBafm5/95m\n507KX7w8D2nJTbn+uLlUjbap5DaqGlrCzN1sWwMNbq0PM7fZNoQZ3BjqqbPOz92a7Q7Z0tVt\nrnN7qNWW2KB8SK9OeDT38cXCS7sp98m1rJEqiDVSkoc10haXSDGskd6c8Pv8X2tGLMrpGfk6\nkBQBKckDpG+YO7dwbfVDajznnvz29fbaqYuXXTGtDUiKgJTkAdJxWziKFAOkV2sKPWo33Thu\n7KyOLwNSBQEpyR3SFo8ixQCpp4BUQUBKcoe05SUSkDQBScogpC0vkYCkCUhSBiEdu8UlEpA0\nAUnKHqQelkhA0gQkKXuQHtjyEglImoAkZQ/StC0vkYCkCUhS9iAd22fhFq8HkiIgSZmDtHzg\nlpdIQNIEJClzkHo4igQkVUCSMgeppyUSkDQBScocpJ6WSEDSBCQpa5CWDzykh1uApAhIUtYg\nPWDO6eEWICkCkpQ1SP/c0xIJSJqAJGUN0jE9LZGApAlIUsYgLetxiQQkTUCSMgbp/h6XSEDS\nBCQpY5D+2fy4p5uApAhIUsYg9bxEApImIEnZgrRs4KE93gYkRUCSsgXpPnNuj7cBSRGQpGxB\n+qeel0hA0gQkKTuQnthrl10G9u1xiQQkTUCSsgPpXLPvkI9O7Pl2ICkCkpQdSIcOfLvX24Gk\nCEhSZiAt6ntM7xsASRGQpMxA+rH55943AJIiIEmZgXSuub/3DYCkCEhSZiAdUmKJBCRNQJKy\nAqnkEglImoAkZQXSnaWWSEDSBCQpK5DOKbVEApImIElZgVRyiQQkTUCSMgKp9BIJSJqAJGUE\nUuklEpA0AUnKCKRzzAOlNgGSIiBJGYF0yMBlpTYBkiIgSdmAtKjvsSW3AZIiIEnZgHSnmVZy\nGyApApKUDUhnl14iAUkTkKRsQDq49BIJSJqAJGUC0sIylkhA0gQkKROQylkiAUkTkKRMQCpn\niQQkTUCSMgGpnCUSkDQBScoCpLKWSEDSBCQpC5DuNN8oYysgKQKSlAVIZ5ufl7EVkBQBScoC\npLKWSEDSBCQpA5AW9j2unM2ApAhIUgYglbdEApImIEkZgFTeEglImoAkZQBSeUskIGkCkpR+\nSH/oU9YSCUiagCSlH9Id5S2RgKQJSFL6IZ1tHixrOyApApKUfkgHD1xe1nZAUgQkKfWQ/tDn\n+PI2BJIiIEmph3S7ubi8DYGkCEhS6iFNLnOJBCRNQJJSD+mgMpdIQNIEJCntkP7QZ1iZWwJJ\nEZCktEMqe4kEJE1AktIO6SzzUJlbAkkRkKQ0Q3pz9uzZHxm0osyt44XU3FYyW3oTXaEG22CD\nQ80NtotDVf4jnmRynVz+YNXDKWNuicEt/ETqHj+RpCr4ibT/Trfffvtr5W4d708kIFUQkJLK\nh/Q/5nOVDAaSIiBJKYb0PQJh+YEAABLPSURBVDOjksFAUgQkKcWQxpmnKxkMJEVAklIMaf+d\n3qtkMJAUAUlKL6T/MZ+vaDCQFAFJSi+kCpdIQNIEJCm9kE6tbIkEJE1AktILab/KlkhA0gQk\nKbWQXq5wiQQkTUCSUgvpxgqXSEDSBCQptZAqXSIBSROQpFRC+tP999+/z86VLZGApAlIUioh\njc6f9l3hEglImoAkpRHSqt12u/DCbzxf4WAgKQKSlEZIvzajFYOBpAhIUhohXW1mKwYDSRGQ\npDRC+qJ5UTEYSIqAJKUQ0qrd/lIzGEiKgCSlENKvVEskIGkCkpRCSN8239UMBpIiIEkphPRF\n85JmMJAUAUlKHyTlEglImoAkpQ/Sr8wY1WAgKQKSlD5IyiUSkDQBSUofpC/olkhA0gQkKV2Q\n3lu8eNGue+sGA0kRkKR0QRqZP+1bt0QCkiYgSamCtHLwbiNGjHxKNxhIioAkpQrSI2aifjCQ\nFAFJShWky8wt+sFAUgQkKVWQTjSv6wcDSRGQpDRBWjn4AIfBQFIEJClNkJyWSEDSBCQpTZAu\ndVkiAUkTkKQ0QRruskQCkiYgSSmC5LZEApImIEkpgvSw0xIJSJqAJKUIktsSCUiagCSlCJLb\nEglImoAkpQfSysEfcxoMJEVAklIC6bRddtnJbYkEJE1AktIB6U/9tx8y5IDHnQYDSRGQpHRA\nuttMcR4MJEVAktIBaYq5y3kwkBQBSUoHpCP6/dF5MJAUAUlKBaQ/9z/CfTCQFAFJSgWkuzws\nkYCkCUhSKiD5WCIBSROQpFRA8rFEApImIElpgORliQQkTUCS0gDpLjPVw2AgKQKSlAZIU8zd\nHgYDSRGQpDRAOqLfEg+DgaQISFIKIP2p35E+BgNJEZCkmCE9cdKn8h3p4ygSkFQBSYoZ0ljT\n3kC3076TgKQISFLMkPbd+T2Pg4GkCEhSxJBeMl/wORhIioAkRQzpu2amz8FAUgQkKWJIY8wz\nPgcDSRGQpIgh7b3LKp+DgaQISFK8kDwvkYCkCUhSvJC+a77tdTCQFAFJihfSGPMrr4NjgLRi\n2sj8Xxtmjx9z1SogaQJSUhHS3rt6XSLFAOnZM28oQJp50eLlMy9oBZIiICUJpJfMF/0OjgDS\nL1e/kIdUW7Mk91Np1HwgKQJSkkDyvUSKAZK1BUjzRrflPk69F0iKgJQkkEZ7XiJFBGnuxPzF\n6bfkPiy8Otei+lI12eaS26hqbA0zt962BRrc2hBmbrNtDDO4KdBTVz/jnMn5dt11k+fBwZ46\nW2KDiiFNyl+8PA/pmaG5Xir5ZURd+01yyrcZv60fia+K/2hQLqQXCy/tptyX+1D3Vq5Va0u1\n0W4uuY2quqYwc9e2tQQa3LI+zNwGuyHM4E2Bnrp/Ndf+qtC7ngeva/U8UGq2vd++vlJIa0Ys\nyi2MRr4uV7JGqiDWSEnD+ywNMziCNdKa2idH1tbW22unLl52xbQ2ICkCUnsrtzu4x//V3K0I\nIE2uyTfHbrpx3NhZHV8GpAoCUntzzPnZhdRDQKogILV3ibkXSEDSB6T2TujzHpCApA9IhVZu\nd2CP/6u5Y0BSBCQpMkhzzGQgAckhIBW62NwBJCA5BKRCw/osBBKQHAJSvpWDDtzi/yHrIyAp\nApIUF6SHzFlAApJLGYb0zs2zpRpzG5CA5FKGId1kOur3FpCA5FKGIY0xs26XHt/i/yHrJSAp\nApIUAaQP/5YTIAHJpexC6vJbToAEJJeyC6nLbzkBEpBcyi6kLr8IEkhAcim7kLr8IkggAcml\nzELq+osggQQklzILaXaXXwQJJCC5lFlIXX8RJJCA5FJmIXX9XflAApJLWYX0YtfflQ8kILmU\nVUhdl0hAApJT2YH0wv2dO7nr78oHEpBcygykBYPMh+r634kBCUguZQbSbeZTF3buri63AwlI\nLmUG0iTzcK+3AwlILmUG0scGrej1diAByaWsQFrQZ3jvGwAJSC5lBdKt5tLeNwASkFzKCqSJ\nJZZIQAKSU1mBdECJJRKQgORURiCVXCIBCUhOZQTSLaWWSEACklMZgVRyiQQkIDmVEUgll0hA\nApJT2YD0ZsklEpCA5FQqIS1d3KWbzWWlvgZIQHIpjZB+3td0q9QSCUhAciqNkCaboz/VpTPf\nKfVFQAKSS2mEdODA5ZXPBRKQXEohpAV9hinmAglILqUQ0m3mYsVcIAHJpRRCOsvMUcwFEpBc\nSiGkA0sefN1SQAKSS+mDtKDPCZq5QAKSS+mDdJu5RDMXSEByKX2QJqmWSEACklPpg1Tqt5z0\nEJCA5FLqICmXSEACklOpg1Tyt5z0EJCA5FLqIJX6RZA9BSQguRQrpIt33WXLDdhupWoukIDk\nUqSQVu8+cEgPnaebCyQguRQppGfNKM9zgQQklyKFdI25zvNcIAHJpUghfcn81vNcIAHJpTgh\nrd5jj9We5wIJSC7FCcn/EglIQHIqTkjXmOt9zwUSkFyKE9KXzDzfc4EEJJeihBRgiQQkIDkV\nJaRnzSne5wIJSC5FCSnAEglIQHIqSkgBlkhAApJTMUJavdte3pdIQOrehnWl2mTrS26jakNT\nmLnr2loCDW5ZH2ZuQxlPQ0+tGPEPPXfyyf9wvPmyx0eatME2+h9aqDXQ3Bbb++11rpDqS9Zk\nm0tvpKmxNczcetsWaHBrQ5i5zbZR/bV3dP8d+B+u710eH2lSg23xP7RQsKeu1He6KyRe2lVQ\nNb60O93c3/X/Zelo+YrcH48PVOKlHZBcqkZIH92hl/87Qvm/mpcMSEByqQohvWZO6uVWIElA\nUpQlSD8w03u5FUgSkBRlCdLp5he93AokCUiKsgSp1yUSkIoBSVGGIL1mTu7tZiBJQFKUIUi9\nL5GAVAxIijIEqfclEpCKAUlRhiAN6XWJBKRiQFKUHUiv9r5EAlIxICnKDqTv975EAlIxICmK\nFdKb555ZYYf2vkQCUjEgKYoV0kWlTuTu3m69LpGAVAxIimKFNLTfY09X2Ju9DwaSBCRFkUJa\nOuAw34OBJAFJUaSQ7jHn+x4MJAlIiiKFdKH5me/BQJKApChSSEP7/dH3YCBJQFIUJ6SlAw73\nPhhIEpAUxQnpHnOB98FAkoCkKE5IU/0vkYBUDEiK4oR0pP8lEpCKAUlRlJD+3N//EglIxYCk\nKEpIdwdYIgGpGJAURQkpxBIJSMWApCg+SE/dMPuAfkv8DwaSBCRF8UEaa4w5MsBgIElAUhQf\npNxPpO+9HGAwkCQgKYoPUtD/jDlAQAKSS0BKAhKQXAJSEpCA5BKQkoAEJJeAlAQkILkEpCQg\nAcklICUBCUguASkJSEByCUhJQAKSS0BKAhKQXAJSEpCA5BKQkoAEJJeAlAQkILkEpCQgAckl\nICUBCUguASkJSEByCUhJQAKSS0BKAhKQXAJSEpCA5BKQkoAEJJeAlAQkILkEpCQgAcklICUB\nCUguASkJSEByCUhJQAKSS0BKAhKQXAJSEpCA5BKQkoAEJJeAlAQkILkEpCQgAcklICUBCUgu\nASkJSEByCUhJQAKSS0BKAhKQXAJSEpCA5BKQkoBk7YbZ48dctQpImoCUBCRrZ160ePnMC1qB\npAhISUCytTVLcj+VRs0HkiIgJQHJzhvdlvs49V4gKQJSEpDs3In5j9NvyX14YUSuV1pK1Wpb\nS26jqrUtzNwWG2pwW6A90RbbLm4NtottoLltJQY3VwxpUv7j5UDSBCSZC6QXCy/tptwnn/PS\nroJ4aZfESzu7ZsSinJ6RrwNJEZCSgGTttVMXL7tiWhuQFAEpCUjWbrpx3NhZHV8GpAoCUhKQ\nugWkCgJSEpCA5BKQkoAEJJeAlAQkILkEpCQgAcklICUBCUguASkJSEByCUhJQAKSS0BKAhKQ\nXAJSEpCA5BKQkoAEJJeAlAQkILkEpCQgAcklICUBCUguASkJSEByCUhJQAKSS0BKAhKQXAJS\nEpCA5BKQkoAEJJeAlAQkILkEpCQgAcklICUBCUguASkJSEByCUhJQAKSS0BKAhKQXAJSEpCA\n5BKQkoAEJJeAlAQkILkEpCQgAcklICUBCUguASkJSIrevPqlUKMDdf2PtvUjqLCnrl6xrR9C\nZa2++tFt/RAq7MdXt5XeqFAwSE8M/Vmo0YEaPnZbP4IKmz30tW39ECprydBvbeuHUGGThgKp\n4oAUOiApAlLwgBQ8ICkCUuiARES9BiQiDwGJyENAIvJQCEjLZ0wYd9kb1m6YPX7MVasC3IHv\n5AFPrckVwz85vDvr9LHfWRfRHi4+4jh28YppI/N/ye4tZzcHgNQ2+fubGu4aW2dnXrR4+cwL\nWv3fg+eKD3jSI7W1tR9s64dTuqbzZq5Y9s1/sdHs4Y5HHMUufvbMGwqQZPeWs5sDQFpf85a1\n79csrK1ZksM8ar7/e/CcPGA75uVt/VDKa2FN7lvxg5ql0ezh4iOOYxf/cvULeUiye8vazSFe\n2l1yw4bGu85pnDc6/2/wU+8NcA+eSx5wU81NUyZ9e+W2fjSle6OmLvdz9JSn49nD8ohj2cUF\nSLJ7y9rNISB9cEFNzZmL7dyJ+U+m3xLgHjyXPOB1X/nOgoVXnblxWz+ckm0+/YfNDXeOejCe\nPSyPOJZdXIAku7es3RwAUsuFN9fVPzDhg7mT8p9dXv1Pszzgwif1p87dxg+njF4/d9S4uyc9\nHM0eLj7iwuUIdnE7pGT3lrWbA0CaP6I+9/GsOS8WfiJOuc//PXhOHnD7Z+dHcWrTxpbGUfOi\n2cP5Co+4/WL17+ICJNm9Ze3mEJBqNuU+nvnImhGLrF038nX/9+A5ecBLb27KvQgZ88y2fjwl\na312jbXPj1wXzR4uPuJYdnEBkuzesnZzAEibzvzBpsaHRq+0105dvOyKaeWe9bftkgdcN+GG\nd1fMOqt+Wz+e0n191vuLz/6hjWYPFx9xHLt4Te2TI2tr64u7t5zdHOIfG/48Y8K4i1/LfYPe\nOG7srDWlt9/myQNePH3cGbPe29aPpoxW/OuYM25vjmgPFx9xFLt4cv6occ2c4u4tZzdzihCR\nh4BE5CEgEXkISEQeAhKRh4BE5CEgEXkISEQeAtLW6UqTb8cT7ld99TEHWjvswIq/bNwOqnsj\nRUDaOl1p/uXWW380/a/NDRV+4Sv5Z+iGWVsF0it8N6hj122drjQv5P/a+Nc7bO5yS2PvX3iT\nPEM9QOrtyyuFdBPfDerYdVunBJKdZl6y9vnP7Ljd4bfnPjvuU4/td5S1vzhu8J5fXmQ73XDi\nCf970g57jl9tP597QTi046VdcYNCyZd3XHn3MbvsOPTu3IW2GfsN+vh944uQivfw+PC/2O7Q\n2W3WHnZY/vqRu3e7L1IFpK2TQJpunre/GvCpOU+eZ6639lOfPPSGh+1jfT/zX7f/zd7vdrrh\n5CFHzfnzz/qNs4tGmpffKkLq2KBQ+5d3XHmvGfPoI/9oHrX2WnPak3cfcqBAKt7Dg32G3//E\nOWZaZ0hd7otUAWnrJJCG91tjh/51Q+7SiJ0320+bn+cuffJjLda+NPCmD93w69ylz+S+zSfn\nnyGB1LFBofYv77jyus81Wbu+/+m2bZ9Dcz90Vg4QSMV7OGj//LanDHi/E6Qu90Wq2HVbpyvN\nY++++87vzjLn2VrztfpcPzTP2U/3b7Z2tZnSvk3nGwbnr5jY58OQOm1QqPDlXa+0+w23b5up\n+UvHJZCK97DSnJv/67bcD61OkD58X6SKXbd1av/nb9P//AY73yTdYz+9d+6m+WZm+zadbxiS\nvyL/jd0ZUqcNCiVfXrxy7T8dtNOgQWaYfal95OgdZHByD78zhf8O4nFza2dIH74vUsWu2zpd\naW74xS/mPrfW5v+RedILhWrbv4fnm8vbt+l6QzdInTYoVNiq05WfG3jNi28t2GeYfbFdzqgi\npOQeXjbT8389Zm4DkufYdVsnWSPl+sB8VS4Wvodr2z9furrrDd0gddqgY6uOK1eaC3Mfm3cY\nZpe0v5Y7PIFUvIf32i/8yMy1R3wif+lYIHmKXbd16gTJ/v3O+R9MP7m8OfkePmy3jdYuMDO6\n3ZD/xj7bNHf8Y0PHBoXatypemR9h7X+YY23rHge35kb2lX9sKN7DJ3bP/zvFF7Zfb0/eI3fh\n3cEfglS4L1IFpK1TZ0i/HvDx/3xi+oBJ8j38WN+jb7/lY3+1stsN+W/uK8yM+4qQOjYo1L5V\n8cqm/fed8+w3PnPSjr/c+E0z8r4fDDm645+/k3t4tO+w+x8+zVxj7dVm1vIXTjx89273RaqA\ntHXqDMk+99kdtzv4mhb5HraPHzd4j1MWdr8h/829/IjBR3QckC1uUCj58uKVLx+3/V7n1j2y\nx96LWi77q4GfePCiAXKPxXuYO3yHQUfckbtQ//V9Bx326JSdut0XqQISkYeAROQhIBF5CEhE\nHgISkYeAROQhIBF5CEhEHgISkYeAROQhIBF5CEhEHvo/waD2QnGHpGEAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "read_q_df %>%\n",
    "    ggplot(aes(x = perc, y = quant)) +\n",
    "    geom_line() +\n",
    "    ylim(0, 50) +\n",
    "    xlim(80, 99.9) +\n",
    "    labs(x = \"Percentile read count\", y = \"Quantile\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "264bd43d",
   "metadata": {},
   "source": [
    "#### RA thresholds ~ percentile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "340192c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "quantiles <- foreach (perc = seq(91, 100, 0.5)) %dopar% {\n",
    "    quant <- quantile(long_RA$rel_a, perc / 100)\n",
    "    return(tibble(perc = perc, quant = quant))\n",
    "}\n",
    "\n",
    "stopImplicitCluster()\n",
    "q_df <- bind_rows(quantiles)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ba79cb45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3dd2DTdf7H8XcHLWXKRqa4EByo\nyBBQDgeenrSCUhC0ilh/osDhKacnVM9RULyKd9aJFTmrcoq7oLJU8ECGyIGAZRZooUMRgTI6\n+P4ympK233zzzjffd/P9pK/XH20Iqc+U3OOapBmkYRgW9CjUZwDDwmGAhGEWDJAwzIIBEoZZ\nMEDCMAsGSBhmwQAJwywYIGGYBTMD6dABM/utxNSXBVIoPy6d+F2+UH5UOnFYvlBeLJ0oPiJe\nKD/s9zQHg4J0sMjMfik19WUB7FfthHTiN/mCdlQ6cUi+oB2RThyRL2i/+z3NAUAyN0DiFQAJ\nkAwHSLwCIAGS4QCJVwAkQDIcIPEKgARIhgMkXgGQAMlwgMQrABIgGQ6QeAVAAiTDARKvAEiA\nZDhA4hUACZAMB0i8AiABkuEAiVcAJEAyHCDxCoAESIYDJF4BkADJcIDEKwASIBkOkHgFQAIk\nwwESrwBIgGQ4QOIVAAmQDAdIvAIgAZLhAIlXACRAMhwg8QqABEiGAyReAZAAyXCAxCsAEiAZ\nDpB4BUACJMMBEq8ASIBkOEDiFQBJd4fTRt7y9wJAAiRuAZB099Sft+196v5yQAIkZgGQ9FY0\nZIfjp9JN6wAJkJgFQNLbiptPOj5O+I/jQ8Eix3IPmVqZuS/j77BWKp04Il4o1k5IJ47KF7Tj\n0onj8gXtqN/THA4I0pd3Oj9Ofd3xYWlPx1YxvgbD6sAqb+7wII1xfpzihLT7Lcd2HjGz4nJT\nXxZIQSuTThyVL2gl0onj8gXthHTihHxBO+73NMUBQfreddVu/AeeP+M2kmQBt5FYU/E20oH4\nrQ49CRsBCZCYBUDS3bMTtu157MGTgARIzEI4QPrihS1+TxMgpOIXRgyfduprAEmyAEisyUMa\nR0v8ngYPETI5QOIVwgFSAln+EwmQPAMkXiEcIPWK+tXvaQDJ5ACJVwgHSO3a49HfYgMkXiEM\nIOVH9wEksQESrxAGkDbQMEASGyDxCmEA6UuaCEhiAyReIQwgzaZnAUlsgMQrhAGkaZQJSGID\nJF4hDCBNoG8ASWyAxCuEAaSbaTsgiQ2QeIUwgNSPjgKS2ACJVwgDSGc0x8txyQ2QeIUwgFT/\nQkCSGyDxCupDyqbrAElugMQrqA/pG7oLkOQGSLyC+pDepSmAJDdA4hXUh5RGLwOS3ACJV1Af\n0kP0CSDJDZB4BfUhjaI1gCQ3QOIV1Ic0iPIASW6AxCuoD+m8Bnh/JMEBEq+gPqQmZwOS4ACJ\nV1Ae0m66EpAEB0i8gvKQVtIIQBIcIPEKykP6kCYBkuAAiVdQHtKL9CwgCQ6QeAXlIU2hfwOS\n4ACJV1Ae0l20GJAEB0i8gvKQ/kibAUlwgMQrKA+pR70CQBIcIPEKykNq1VHijcYAyTNA4hVU\nh7QvsjcgSQ6QeAXVIa2jBECSHCDxCqpDmk/jAElygMQrqA5pFj0JSJIDJF5BdUhP0huAJDlA\n4hVUhzSO5gOS5ACJV1AdUgKtAyTJARKvoDqkXpF5gCQ5QOIVVIfUsXURIEkOkHgFxSEVxPQA\nJNEBEq+gOKRNdD0giQ6QeAXFIS2muwBJdIDEKygO6d80BZBEB0i8guKQnqF0QBIdIPEKikOa\nRB8CkugAiVdQHNIIWglIogMkXkFxSFdQDiCJDpB4BcUhnd20CJBEB0i8guKQGp5XBEiiAyRe\nQW1I22lQESCJDpB4BbUhLaPRRYAkOkDiFdSG9D49VARIogMkXkFtSC9QWhEgiQ6QeAW1IT1M\n7xUBkugAiVdQG1ISfVMESKIDJF5BbUjXUHYRIIkOkHgFtSF1jy0sAiTRARKvoDak5l1cBUCS\nGyDxCkpDyo3o5yoAktwAiVdQGtJqusVVACS5ARKvoDSkT2iCqwBIcgMkXkFpSK/QNFcBkOQG\nSLyC0pBSaLarAEhyAyReQWlId9NXrgIgyQ2QeAWlId1IG1wFQJIbIPEKSkO6NCrfVQAkuQES\nr6A0pNPbuQuAJDdA4hVUhpQf1dNdACS5ARKvoDKk/9GN7gIgyQ2QeAWVIX1B97gLgCQ3QOIV\nVIY0mx53FwBJboDEK6gMKZVedRcASW6AxCuoDGk8feYuAJLcAIlXUBnSMFrjLgCS3ACJV1AZ\n0uURue4CIMkNkHgFlSF1bl5RACS5ARKvoDCkwtgLKgqAJDdA4hUUhpRN11YUAElugMQrKAzp\na7qjogBIcgMkXkFhSO/QIxUFQJIbIPEKCkP6B/2zogBIcgMkXkFhSA/S+xUFQJIbIPEKCkO6\nlb6rKACS3ACJV1AY0iDaUVEAJLkBEq+gMKSuDT0FQJIbIPEKCkNqco6nAEhyAyReQV1Iu+hK\nTwGQ5AZIvIK6kFbQSE8BkOQGSLyCupDm0QOeAiDJDZB4BXUhvUgzPAVAkhsg8QrqQnqU3vYU\nAElugMQrqAtpDC3xFABJboDEK6gL6Tra7CmIQzpmasdPmvu6AApauXTihHxBK5NOlMgXtFLp\nRKlQ4eKYYk9BK/F/8qAgHTlkZofLTX1ZIAWtVDpRLF/QTkgnjsoXtOPSieNChVadKgvaUb+n\nPhwUJFy1kyzgqh1rQlft8iL6VBZwG0lugMQrKAtpHd1UWQAkuQESr6AspCy6r7IASHIDJF5B\nWUiv01OVBUCSGyDxCspCeoLeqCwAktwAiVdQFtK9tKCyAEhyAyReQVlI8fRjZQGQ5AZIvIKy\nkHpF5lUWAElugMQrKAupQ+tTBUCSGyDxCqpCKqh38akCIMkNkHgFVSH9RDecKgCS3ACJV1AV\n0iIae6oASHIDJF5BVUhzaMqpAiDJDZB4BVUhPUMvnSoAktwAiVdQFdIk+uhUAZDkBki8gqqQ\nEun7UwVAkhsg8QqqQhpAOacKgCQ3QOIVVIV01mleBUCSGyDxCqpCatDNqwBIcgMkXkFRSNvo\nKq8CIMkNkHgFRSEto9FeBUCSGyDxCopC+g9N9ioAktwAiVdQFNJMet6rAEhyAyReQVFIf6W5\nXgVAkhsg8QqKQrqdvvUqAJLcAIlXUBTS1bTVqwBIcgMkXkFRSN3rexcASW6AxCsoCqlZF+8C\nIMkNkHgFNSHtpf7eBUCSGyDxCmpCWkXDvQuAJDdA4hXUhPQJTfQuAJLcAIlXUBPSyzTduwBI\ncgMkXkFNSFPpLe8CIMkNkHgFNSHdTV95FwBJboDEK6gJ6U+00bsASHIDJF5BTUiXROV7FwBJ\nboDEK6gJqW37KgVAkhsg8QpKQtofdVmVAiDJDZB4BSUhrachVQqAJDdA4hWUhLSA7qlSACS5\nARKvoCSkN+nxKgVAkhsg8QpKQnqaXqtSACS5ARKvoCSk++nzKgVAkhsg8QpKQhpGa6sUAElu\ngMQrKAmpb0RulQIgyQ2QeAUlIXVuUbUASHIDJF5BRUiFsRdULQCS3ACJV1AR0s80uGoBkOQG\nSLyCipCW0p1VC4AkN0DiFVSE9A79rWoBkOQGSLyCipCeo39VLQCS3ACJV1AR0l/og6oFQJIb\nIPEKKkIaSd9VLQCS3ACJV1AR0h9oZ9UCIMkNkHgFFSGd26haAZDkBki8goqQGp9brQBIcgMk\nXkFBSLtoYLUCIMkNkHgFBSH9l0ZWKwCS3ACJV1AQ0gf0QLUCIMkNkHgFBSG9SDOqFQBJboDE\nKygI6W+UWa0ASHIDJF5BQUh30tJqBUCSGyDxCgpCuo62VCsAktwAiVdQENKFMYXVCoAkN0Di\nFRSE1KJT9QIgyQ2QeAX1IOVF9K1eACS5ARKvoB6kH2ho9QIgyQ2QeAX1IH1O91cvAJLcAIlX\nUA/S6/R09QIgyQ2QeAX1IP2dMqoXAElugMQrqAfp/2hB9QIgyQ2QeAX1IA2hH6sXAElugMQr\nqAepV+S+6gVAkhsg8QrqQWrfpkYBkOQGSLyCcpAK6l1SowBIcgMkXkE5SBvphhoFQJIbIPEK\nykFaSHfXKACS3ACJV1AO0ls0tUYBkOQGSLyCcpCm08s1CoAkN0DiFZSD9Gf6uEYBkOQGSLyC\ncpCG06oaBUCSGyDxCspBGkC7axQASW6AxCsoB+nM02oWAElugMQrKAcprlvNAiDJDZB4BdUg\nbaWraxYASW6AxCuoBmkZ3VazAEhyAyReQTVIc2lyzQIgyQ2QeAXVID1PM2sWAElugMQrqAZp\nMs2tWQAkuQESr6AapNtoWc0CIMkNkHgF1SBdRdtqFgBJboDEK6gGqVucTgGQ5AZIvIJqkE47\nU6cASHIDJF5BMUh7aIBOQQBS7oMJgOQcIPEKikH6nobrFKyHtCxpJiC5Bki8gmKQPqY/6xSs\nh7SkcCUguQZIvIJikF6i6ToFidtIgOQeIPEKikGaSnN0CpKQ1o9zbGOJqZ0092UBTBNPlJaL\nFzTxRJl8QSsTT1hZuI9W6BQY38QJLqTlCQkJm70gLe3p2CoePgxTZDfRPnNfWF55yA+k4pyc\nnOPaKUilvzt24Bcz+7XU1JcFUtBOSCcOyhe0Y9KJw+KFQ1qxdKLYysLF0QU6Be2Q3y/8jQvJ\nM9xGcg+3kXgFxW4jtWmvV7D+NtKBooUJRUXHAAmQmAW1IO2P6qVXsB7S2CHOfQpIgMQsqAXp\nR4rXK+AhQnIDJF5BLUgL6P/0CoAkN0DiFdSClEF/1ysAktwAiVdQC9JT9LpeAZDkBki8glqQ\n7qPP9QqAJDdA4hXUgjSUftArAJLcAIlXUAtSn4g8vQIgyQ2QeAW1IHVqqVsAJLkBEq+gFKTC\nmAt1C4AkN0DiFZSCtIWu0y0AktwAiVdQCtISulO3AEhyAyReQSlImfQ33QIgyQ2QeAWlIM2g\nF3ULgCQ3QOIVlIL0AM3TLQCS3ACJV1AK0kj6r24BkOQGSLyCUpAG0i7dAiDJDZB4BaUgndNI\nvwBIcgMkXkEpSI266hcASW6AxCuoBGkn/UG/AEhyAyReQSVI39Gt+gVAkhsg8QoqQfqA/qJf\nACS5ARKvoBKkf9Fz+gVAkhsg8QoqQXqE3tEvAJLcAIlXUAnSHbRUvwBIcgMkXkElSIPpZ/0C\nIMkNkHgFlSBdEFOoXwAkuQESr6ASpBadfRQASW6AxCsoBCkvoq+PAiDJDZB4BYUgraVhPgqA\nJDdA4hUUgvQZjfdRACS5ARKvoBCk1yjVRwGQ5AZIvIJCkB6nN30UAElugMQrKATpHvrCRwGQ\n5AZIvIJCkIbQeh8FQJIbIPEKCkG6LHK/jwIgyQ2QeAWFILVr66sASHIDJF5BHUj50Zf6KgCS\n3ACJV1AH0kb6k68CIMkNkHgFdSB9RXf7KgCS3ACJV1AH0mxK8VUAJLkBEq+gDqTp9LKvAiDJ\nDZB4BXUgTaRPfBUASW6AxCuoA+kWWu2rAEhyAyReQR1I/WmPrwIgyQ2QeAV1IHVp5rMASHID\nJF5BHUhx3X0WAElugMQrKAMpm672WQAkuQESr6AMpG/pdp8FQJIbIPEKykB6j/7qswBIcgMk\nXkEZSM/TCz4LgCQ3QOIVlIE0mf7jswBIcgMkXkEZSKNpmc8CIMkNkHgFZSANom0+C4AkN0Di\nFZSB1C3OdwGQ5AZIvIIykJqe5bsASHIDJF5BFUi76QrfBUCSGyDxCqpA+p4SfRcASW6AxCuo\nAukjmuS7AEhyAyReQRVI6fSM7wIgyQ2QeAVVIE2hOb4LgCQ3QOIVVIE0lhb5LgCS3ACJV1AF\n0vW0yXcBkOQGSLyCKpAurpfvuwBIcgMkXkEVSK07GBQASW6AxCsoAmlfZC+DAiDJDZB4BUUg\nraN4gwIgyQ2QeAVFIC2gew0KgCQ3QOIVFIH0Bj1hUAAkuQESr6AIpKdolkHBPKTD74+9tGPj\njpfe9f5hQNIdIPEKikC6j7IMCmYhHZvRkmIuvGrYVRfGUMvnjgGSzgCJV1AE0k20zqBgEtLO\niyNHLCh2HTwyPzHykp2AVHOAxCsoAql3RJ5BwSSkZoM2e2HZPKgZINUcIPEKikDq2MqoYBLS\n1LIqWsqmAFLNARKvoAakwpiLjArm72zYMnfm9Bc+8H2dDpAAiVdQA9Jm+qNRwSyk+d3Jvb4r\nAcnHAIlXUAPSEhpjVDAJaX5kt+lfrtv0Q9ZjnWK+AyT9ARKvoAakt+lRo4JJSH0GllQcOtJz\nICDpD5B4BTUgPUsvGhVMQoqbXXnw1YaApD9A4hXUgDSJ5hkVTEJqml55cGZTQNIfIPEKakAa\nSSuMCiYh3XBufsWhPV3iAUl/gMQrqAHpStplVDAJ6X9NGo969q13Z0+/Ja7lz4CkP0DiFdSA\ndE4Tw4LZu7+3jm7suve72d17DB0BkmgBkFizAlKjroYF87+QLdu2bOHy7eXGjABJtgBIrFkA\naQcNMiwE/3ykPZ8Dkv4AiVdQAtJyGmVYCB7SLOOn+wGSZAGQWLMA0vv0oGEBkOQGSLyCEpD+\nSf8wLJiEtPzUHgYkHwMkXkEJSI/Qu4YFk5DIe4CkP0DiFZSAdAd9bVgwCenSDrM8uwOQfAyQ\neAUlIF1L2YYFk5C2xL3iOYjbSL4GSLyCEpDOjy00LJi9s+Hl+hsAyc8AiVdQAlLzM4wLpu+1\nW7G94sDysYCkP0DiFVSAlBtxuXEBLxApN0DiFVSAtIZuNi4AktwAiVdQAdKnNN64AEhyAyRe\nQQVIr1KqcUEcUkmZqZ0092UBTBNPlMsXwuObKBdPBF2YTu8bFxjfRCl+IpkbfiLxCir8REqm\nL40LuGonN0DiFVSAdCP9z7gASHIDJF5BBUg9o/YbF8xDOthzVcUHQNIfIPEKKkBqd7qfgnlI\nRbSo4gMg6Q+QeAUFIOVHX+qnAEhyAyReQQFIG+hGPwVAkhsg8QoKQPqSkv0UAElugMQrKABp\nNj3mpwBIcgMkXkEBSNPoFT8FQJIbIPEKCkCaQJ/6KQCS3ACJV1AA0s202k8BkOQGSLyCApD6\n0V4/BfOQyvcer/gASPoDJF5BAUhnNPdXwEOE5AZIvIICkOqf768ASHIDJF7B/pCy6Rp/BUCS\nGyDxCvaH9A0l+SsAktwAiVewP6R36WF/BUCSGyDxCvaHlEYv+CsEA+no8rlFWqlmPECSLAAS\na8FCeoje91cIAtKMJkQrtUfHlAGS/gCJV7A/pFG03F/BPKRZFP+qA9Kc6OmApD9A4hXsD2kQ\nbfdXMA+px73aMQck7W/dAUl/gMQr2B/SeQ38FsxDil3khvRlHCDpD5B4BftDanq234J5SI0+\nc0Oa0wSQ9AdIvILtIe2mK/wWzEO66qrjTki/XTAYkPQHSLyC7SGtpBF+C+YhfRt95jgac0fT\net8Bkv4AiVewPaQPaZLfQhB3fy+51Pm+l72/MXQESKIFQGItSEjp9KzfQlCPbCha9+MBzc8A\nSbIASKwFCWkK/dtvIQhIxfs07dhbz2UDko8BEq9ge0h30WK/BfOQtrSerpVeTlR/LSDpD5B4\nBdtD+iNt8lswD2nYRTu1THp9x+VDAUl/gMQr2B5Sj3oFfgvmIbV+R9OGX6Rp77QHJP0BEq9g\ne0itOvovmIcUs1Qrb/5XTVsYC0j6AyRewe6Q9kX29l8wD6njG9pi+lrTMtoBkv4AiVewO6R1\nlOC/YB7S2LYPdz6jTCu4aDgg6Q+QeAW7Q5pP4/wXzEPa15dardW0EadtBCT9ARKvYHdIs+hJ\n/4VgfiH7u/PZsWu2bwUk/QESr2B3SE/SLP+F4F+zYTFuI/kYIPEKdoc0jub7LwQBKWv0Ff37\n9+/buCUg6Q+QeAW7Q0qgdf4L5iG9R9EdqF1DGpQFSPoDJF7B7pB6R+b5L5iH1POGQ1rsltJX\nrj0MSPoDJF7B7pA6tmIUzENq7PhJFLtF0x66D5D0B0i8gs0hFcT0YBTMQ2o634FptaYtawtI\n+gMkXsHmkDbR9YyCeUh9binRus7QtI8bApL+AIlXsDmkxXQXo2Ae0rt0nTYxduIT7foBkv4A\niVewOaR/0xRGIYi7v+fO0A5cS9RxDSDpD5B4BZtDepbSGYVgfyGbt6XE0BEgiRYAibWgIE2i\nDxmFYCDt/ui1WZ/tN3YESKIFQGItKEgjaCWjYB7SgT85X0SIIkcdAST9ARKvYHNIV9AuRsE8\npNto5JsLvnjzVroHkPQHSLyCzSGd3ZRTMA/ptPHuzw+1ACT9ARKvYHNIDc/jFMxDivvY/Xlh\nA0DSHyDxCvaGtJ0GcQrmIQ2Y5v6cPgCQ9AdIvIK9IS2jUZyCeUg/nPFRqaaVfXo2XtfOxwCJ\nV7A3pPfpIU7BJKSuXbt260ixXTrFUgc8ssHHAIlXsDekFyiNUzAJqb/XegOS/gCJV7A3pIfp\nPU4h+Kea+xkgSRYAibVgICXRN5wCIMkNkHgFe0O6hrI5BUCSGyDxCvaGdH5sIacASHIDJF7B\n3pCan8EqAJLcAIlXsDWk3Ih+rAIgyQ2QeAVbQ1pNt7AKgCQ3QOIVbA3pU5rAKgQPKS1t1iZA\n0hsg8Qq2hvQKTWMVgocURcln/wuQdAZIvIKtIaXQbFYheEjzP9fKdgOSzgCJV7A1pLvpS1YB\nt5HkBki8gq0h3UgbWIUgXrJ4s/vzB5cCkv4AiVewNaSLovazCuYhkftluEqfwBP7fAyQeAU7\nQ3qP+vAKZiHRqeEnko8BEq9gY0h7O0ct4RXMQlr/T0oY69zdj+0FJP0BEq9gY0gPUDKzYP6q\n3XXGb3kJSIDELNgX0vcxrbczC7jXTm6AxCvYF9IV9Aa3YB5SfmKLCPeNJEDSHyDxCraF9BL9\ngV0wDykx+to7XbeSxgKS/gCJV7ArpB1tYhgvVlxRMA+p5XuGgAAJkJgFu0K6kx7mF8xDapAD\nSMYDJF7BppAWRXXJ5RfMQ7ryc0AyHiDxCvaElH8RzQ2gYB7S+t7fApLhAIlXsCekVBoWSCGI\nlyzuTA06uwZI+gMkXsGWkDY1abwxkEIQV+2uurpigKQ/QOIVbAkpgaYHVAj+F7JHjB/hAEiS\nBUBiLXBI71OP/IAKwUNa3A6Q9AdIvIINIeWdHcl6Pt+pQhCQskZf0b9//76NWwKS/gCJV7Ah\npIdobIAF85Deo+gO1K4hDcoCJP0BEq9gP0irYlsxH6xaWQjiGbI3HNJit5S+cu1hQNIfIPEK\n9oN0Nb0aaME8pMaOn0SxWzTtofsASX+AxCvYDtJr1I/zet9VCuYhNZ3vwLRa05a1BST9ARKv\nYDdIO9vGrAi4YB5Sn1tKtK4zNO3jhoCkP0DiFewGaSzrzS6rFcxDepeu0ybGTnyiHd760scA\niVewGaTFUWfwH6xaWQji7u+5M7QD1xJ1XANI+gMkXsFekAp60rsmCsH+QjZvS4mhI0ASLQAS\na4FAmk7xZgp4zQa5ARKvYCtIm5s2Yr20avWCeUgtPGsKSPoDJF7BVpCGUaqpgnlICa71jbt4\nPCDpD5B4BTtB+jSiG+slimsUgr5qt/+KBVWP+PUft414JBuQAIlbsBGkvHMiF5grBH8baUW1\nlyx+4JEd+2bcdgyQAIlZsBGkh+lOk4XgIe2tX+WPh57J1bTCIZU/kgBJsgBIrHEhrYltnm2y\nEDykpzvUPG5LfJHj49Fcx4oOmNlvZaa+LJCCViKd+F28cEg7Lp04cky8oB2VThQX8043mF4x\nWTiqHfZ7moP6kHq4dmFLerCGo0P3zXJ+WtrTsVV6+DDMhvsPXXlS8D9fXnlIB9IlV0077jlm\neUJCgvPdx/be87Lr/GSnOrb1mKmVm/sy/o5r4okT8gWtTDpRUipe0MQTpaxCYbuYH00XtBL/\nJ9KHVHPFOTk5DlXrR3k/1Q+3kSQLuI3EGu820j00yXzB+kc2bBr1g/cfAUmyAEissSB9G91x\nt/mCeUhRDRp6rfLoE8mu16fE3d+AxC3YA1JBL3oniIJ5SOPOj+wZ/8dudOmtIxyrPHr9ENey\nAAmQmAV7QHqObgymYB7Sh912OD/90PEzw2t6gCRZACTWGJA2nxa3LpiCeUjnV7yty4sXApL+\nAIlXsAWk4fRkUAXzkGIWuT+/HwtI+gMkXsEOkD6L6LYvqIJ5SO1Gun5ZVD7kdEDSHyDxCjaA\nlHduhLkHq1YWzEN6nLqPf+qp+8+jvwGS/gCJV7ABpEfp9iAL5iGdfPZ05zsxt3q8DJD0B0i8\nQughrWtg9sGqlYVgfiF7cveq73eUa8YDJMkCILHmD9JgejHYgklIY45W0XL0LkCqOUDiFUIO\naTb1CfSVVWsUTELqfOHXXliWXdQZkGoOkHiFUEPa0zH626ALJiH9ci0NfHOv62DunKvo2l8A\nqeYAiVcINaRxNDH4gtnbSOVvn03U6vx+57cmOifT4IYSIEkWAIk1Q0jL6nXICb5g/s6GsmVT\nbuxzXt8hU5cb3m0HSJIFQGLNCFJBL3rbggJeIFJugMQrhBbS83S9FQVAkhsg8QohhZTdPO4H\nKwrmIUV7norUqN0NSwFJZ4DEK4QU0kh6zJKCeUgTelP3ocMuoAGjrmkS+QUg1Rwg8QqhhJQV\ncV5QD1atLJiHtLDzCuenFees1Q72HgBINQdIvEIIIe3vHvGJNQXzkC552/05Y6Cmvev7XfsA\nSbIASKz5hDSVbrWoEMTzkRa6P3/RQNM+aQRINQdIvELoIP3YoNnPFhXMQ2qf6H4xvaRWWun1\nlwFSzQESrxA6SH+kF6wqmIf0GF3wwIzn/tqbJmhD6T1AqjlA4hVCBmkOXVpgVcE8pPInWzuf\nj9T0gRPaC+/6dARIogVAYk0f0p6O0V9bVgjq+Ui7Vn2/zfhpfYAkWwAk1vQhjaf7rSsEA+mX\nz19/48tDgORrgMQrhAjS8nrtg3+wamUhiKt2k+o5r9o1nAFIPgZIvEJoIBX2p7csLJiH9BwN\ne2PB/NeuozmApD9A4hVCA+mfdJWVBfOQuv+f+/OIam99CUieARKvEBJIG615sGplwTyk2Ipf\nyH4aB0j6AyReIRSQci6iv1taMA+p4Yfuzx/4flQDIEkXAIm1GpDy/0hDg329k6oF85AGXHnC\n+elo/ysBSX+AxCuEANJY6pbzTP4AABqASURBVJdnbcE8pPkRne55cuodLaIWAZL+AIlXqH1I\nU6nrNosLQfwe6ePznHd/X7TA0BEgiRYAibVqkN6IbBPMW7joFoJ6qnne6jX5xowASbYASKxV\nhfR5TNyXlhfwmg1yAyReoZYhrW4R9W/rCyYhdfUeIOkPkHiF2oWU3YWeESiYhNTfe4CkP0Di\nFWoV0t5eFryuqk4BV+3kBki8Qm1CKvgTxVv1HKQqBUCSGyDxCrUJKZn65ooUAElugMQr1CKk\nJ+kMi16koXoBkOQGSLxC7UHKjGq+SqgASHIDJF6h1iAtjKsf3FsuGxQASW6AxCvUFqS1LSNn\nixUASW6AxCvUEqTss2iaXAGQ5AZIvELtQMrtTfcKFgBJboDEK9QKpMKbaXC+YAGQ5AZIvEKt\nQBpHl+yWLACS3ACJV6gNSGnUebNoAZDkBki8Qi1AmhfdfKVsAZDkBki8gjyk5Q1j58sWAElw\ngMQriENa1zriFeEEIAkOkHgFaUg7ulGqbAGQRAdIvIIwpLwr6S75W2GAJDdA4hVkIRUm0jUH\nAUlsgMSb8pD+TD1yfL+ruVUDJMEBEq8gCukF6rjJ95sxWzZAEhwg8QqSkOZGN1lm8K7mlg2Q\nBAdIvIIgpGVN6n1YBEiSAyTelIa0oV3ES87PgCQ3QOJNZUg7u9MU1wFAkhsg8aYwpH1/oNHu\nQ4AkN0DiTV1IhSPpqv3ug4AkN0DiTV1ID1G3HRUHAUlugMSbspBejTh9vecwIMkNkHhTFdIn\nMY2/qfwDIMkNkHhTFNLypvU+OPUnQJIbIPGmJqSNHSJe9PojIMkNkHhTEtKuC2iy958BSW6A\nxJuKkPKvo2GF3kcAktwAiTcVId1F/fOqHAFIcgMk3hSE9Ch13V71GECSGyDxph6kWRFt1lU7\nCpDkBki8KQdpXkyjpdWPAyS5ARJvqkF6I6be3BpHApLcAIk3xSBNj2zwbs1jAUlugMSbUpAK\nJ1NzvffkAyS5ARJvKkHaP5o6fa/3F4AkN0DiTSFIOVdTtw26fwNIcgMk3tSBtLkHXblT/68A\nSW6AxJsykNaeScP3+fg7QJIbIPGmCqSFLSm5wNdfApLcAIk3RSB90Cjicd9/C0hyAyTe1ICU\nXi/mdYO/BiS5ARJvSkBKjWj0vtHfA5LcAIk3BSDl30VtvjY8BSDJDZB4sz+kvAQ690fjkwCS\n3ACJN9tD2taXemb7OQ0gyQ2QeLM7pI3n0/V7/Z0IkOQGSLzZHNLy9jRqv99TAZLcAIk3e0Na\n0JwmMk4GSHIDJN5sDWlO/ag0zukASW6AxJudIU2PjHuHdUJAkhsg8WZfSIWTqdl83kkBSW6A\nxJttIe2/jTquZJ4WkOQGSLzZFdLua3w9i09n4QCp+LCZHSk39WWBFLRS6USxfEErkU4cky9o\nxwP/ot29aWAe+9THTRQC2wntmN/THAkO0hEzKy439WWBFLQy6cRR+YJWIp04Ll/QTgT8NT+d\nTQm/8E9+IvBCgDuhHfd7muKgIOGqnWShjl61W2T0LD6dhcNVO0CSLNRNSPMaRzwW0BcAktwA\niTcbQnqpXsxrgX0FIMkNkHizH6TUiIaGz+LTGSDJDZB4sxukwvupdY0Xyfc3QJIbIPFmM0h5\nN9EZqwNOAJLcAIk3e0Ha3o8u/TnwBCDJDZB4sxWkny6g6/aYSACS3ACJNztB+q4DjfT/LD6d\nAZLcAIk3G0H6gvcsPp0BktwAiTf7QJody3sWn84ASW6AxJtdIBU8FFn/bbMJQJIbIPFmE0hb\nr6b2C00nAElugMSbPSB93Zn6bTafACS5ARJvtoD0clxEsqm76yoGSHIDJN5sACkvmRq9GVQC\nkOQGSLyFHtL/LqNzvgsuAUhyAyTeQg7p01Y0dHeQCUCSGyDxFmJIhanR0SlBJwBJboDEW2gh\n7RpCLT4KPgFIcgMk3kIKaWVX6r3RggQgyQ2QeAslpDmNKSnPigQgyQ2QeAsdpP0Tqf6L1iQA\nSW6AxFvIIP18BXVZZlECkOQGSLyFCtL8tjR4u1UJQJIbIPEWIkhpMVGTA3kJSOMBktwAibeQ\nQModSc0DfcktowGS3ACJt1BAWt2NLvrBygQgyQ2QeAsBpHeaUqLfNyoPaIAkN0DirdYhFaZE\nxjxvcQKQ5AZIvNU2pOxB1P4rqxOAJDdA4q2WIS3tRAO2WJ4AJLkBEm+1Cym9fsTEfOsTgCQ3\nQOKtNiHl3U6NZ0skAElugMRbLUJa35PO/a9IApDkBki81R6kj1vSzcE+FdbHAElugMRbbUEq\nTImy4KmwPgZIcgMk3moJ0s4bqe0CsQQgyQ2QeKsdSCvOpb4/ySUASW6AxFutQHq1ASXtE0wA\nktwAibdagFT6F2r4hmgCkOQGSLzJQ9rYP+gXgPQ3QJIbIPEmDSn/6QYUv1O2AUiCAyTehCEt\nv4xOyzgsmigCJMkBEm+ikPalxNDgLQG8q7nJAZLcAIk3SUhLL6TWGYG8q7nZAZLcAIk3OUh7\nJkZRfHYg72pueoAkN0DiTQzSp2dSpw9cBUACJMMBku9tT4qITMpxFwAJkAwHSD6XeTp18zyh\nHJAAyXiA5GObbqR6EytfHR+QAMl4gKS7wvRm1MvroQyABEjGAyS9rR1IcSner8sASIBkPECq\nuf2pDeiqdVULgARIhgOkGvv2EjotrXoBkADJcIBUbbmTYyi+xqvWARIgGQ+Qqm7BudRG59W2\nAAmQjAdI3tszMTIiSe/5EoAESMYDJK/N7UBnfKRfACRAMhwgVW5bEkUn+3jROkACJOMBkmcZ\nLej8RT4LgARIhgMk9zbeQLGTfb9IECABkvEAybnCtEbUZ4VRAZAAyXCA5NjqAdQk1fD9yQEJ\nkIwHSEX7U2Lp2vV+CoAESIYDpG96UMt0vwVAAiTD1XVIe+6PohFb/RcACZAMV7ch7ZvRljq+\nzykAEiAZri5Dyk/vTLH357AKgARIhqvDkOadT/USf2AWAAmQDFdnIc27mCLjV7ELgARIhquj\nkBYMIBr4dQAFQAIkw9VJSMviHYwWBlQAJEAyXB2E9N/4CLpM/9kSvguABEiGq3OQfkyKom4Z\nARcACZAMV8cgbZkYQ+ekGz6sTr8ASIBkuDoFKXtifeqQtt9MAZAAyXB1CFJOShNqkZJrrgBI\ngGS4OgMpL60VNUvZY7YASIBkuDoCaV9aW2o4cbv5AiABkuHqBKSCjC4Uk7Q5mAIgAZLh6gCk\nwszuVC9pQ3AFQAIkw4U/pHk9KDJ+dbAFQAIkw4U7pAX9iQZ+E3wBkADJcOENyfWgOp8vVhdI\nAZAAyXDhDMn5oLpeH1tTACRAMlz4Qlpn6kF1vgqABEiGC1dIq26vR11nF1pWACRAMlx4Qsq6\nIZI6p+frnthcAZAAyXBhCCk/oyfRhS+beWyq7wIgAZLhwg5STtrZRL0zLbtSV1EAJEAyXJhB\n2jy5GcUkLrO+AEiAZLiwgvR9ciw1SQ7uwUA+CoAESIYLI0hZgyOoU6qPt9wLtgBIutvzxK0j\nHtkCSOEDaV/GJUQ90i29h8G7AEh6K0l6PnffzBFHASlMIOWntqfIwfMFC4Ckt4MfOQzlDtkJ\nSGEBad0DTSkm0egN94IeIPncoZfHlQBSGEBamhhNrSZnizYAydfKhw555FfngexUx7YeM7Vy\nc1/G33FNPHFCvqCVyf3Hi+ffQHR22iG5gnslWql0olS+oJX4PxEX0vKEhITNjs97Njx97xHH\n56U9HVvF/CmG2WvH53Qn6v/ZyVCfjzBaeeUhP5CKc3JyjrsOlY3Kcnw8tNmxgt/M7GCZqS8L\npKCVSCcOyRe04zL/4W2Pt6V6Ny1yHCoWKpzaEe2odOKofEE74vc0v3MhubY+2fET7OTozz1/\nxm0kyYLMbaQ1yXHUKPlH1+Fg39Xc/3AbSXdHkp7Zs//1m/MASVVIWfFR1Gbytoo/ARKvYP2d\nDTlPjBrx1/9V/hGQJAuWQyrI7EV0Qfq+yiMAiVfAQ4Tkph6knLSzKGJgpvdRgMQrAJLcVIP0\n1V1NKfb2/1Y9EpB4BUCSm1KQNj7WlajFgzVeMhWQeAVAkps6kPIy46MpamBGXs2/AiReAZDk\npgqkxcnNibqmbNH9S0DiFQBJbkpA2pjanahZ0hJffw9IvAIgyc3+kHIzBkdTzOCMfb5PAki8\nAiDJze6QKq7S/Wx4IkDiFQBJbraGtCHlTKI2yV/7Ox0g8QqAJDf7QnJcpYvyc5XOM0DiFQBJ\nbnaFtDipEVGPVN7z9QCJVwAkudkS0vqULkRtk9lvawRIvAIgyc1+kPZmDIygmPjMAF4QCJB4\nBUCSm80gFWQlNeRfpfMMkHgFQJKbrSD9mNKZ6PSJ3weaACReAZDkZh9IO2b2iaC44fMKAk8A\nEq8ASHKzCaRdr14fQxF9/7nTVAKQeAVAkpsdIO2edWN9orMeWmM2AUi8AiDJLeSQcjMTGxJ1\nTM4KIgFIvAIgyS20kByKGhF1SM4K7n3BAIlXACS5hRBSXmZiY6J2wSoqAiRuAZDkFipI+VlJ\nTYhOt0BRESBxC4Akt5BAys9KbkHUPDHTovcdByReAZDkVvuQHIpauhRZ96ZggMQrAJLcahlS\nQVZya6JmiZmMJ0fwB0i8AiDJrTYhORS1ITrNYkVFgMQtAJLcag/S4uS2RE0TM3VeTivYARKv\nAEhyqyVIyyd3JoqLl1BUBEjcAiDJrTYg/TC5C1H9wel7pBKAxCsAktykIe2b939nEDUY+lau\nYASQeAVAkpsopK2vJDQhajj0TbGfRe4BEq8ASHKTg7Q2dWA9opaJsw/L/88ckFgFQJKbDKT8\nxZN7EFHXiVmFUm996T1A4hUASW4CkPZkJrUmih2Yut5dACTWAEluCkJalzY4xvkAoIzK57oC\nEm+AJDfFIBU4rtBFEHVOnuf9MDpA4g2Q5KYSpL3znI9ciOqdsrJ6AZBYAyS5KQNpS3p8Q6Jm\n8enbdQqAxBogyU0NSMtTeruv0Ok/FhWQeAMkudkfUu685PaOK3Q9Jn/nuwBIrAGS3GwOKTs9\nvjFRg8FpNd5mvEoBkFgDJLnZGFL+Fw9dGknUydcVOq8CILEGSHKzK6TNLw5t7rhC12vKMk4B\nkFgDJLnZEVL+4pTejh9FLeLTNjELgMQaIMnNdpC2ZCSe5rxvYWIW/6XuAYk3QJKbrSDlux+3\n0DI+fVtgBUBiDZDkZh9Im9Pjm7ru5l4c8As6AhJvgCQ3e0DanzXR+aOoVWLGDlMFQGINkORm\nA0g/pDl/VxTdOyXwH0WeAiCxBkhyCzGk3HkTnU/P65SUYe79vyoKgMQaIMktlJDWpsU3cj49\nL2VxsAVAYg2Q5BYqSHvnTezq+FHUOSljlwUFQGINkOQWCkgFSx77QyxR3LXPrLGoAEisAZLc\nah3S2rSE5o4fReeO+8C6l6EDJN4ASW61Cik7I9l510KL+LR11hYAiTVAklutQXLcKOoR6bg+\nN9D8vdy+C4DEGiDJrVYgFSxOGRjregDdPJGXuAck3gBJbvKQ1s68ubn7/rnAHkAXwACJN0CS\nmyyk7IykTuR8FGraj4IVQGIOkOQmB6nyRtHjKy2/UVRtgMQbIMlNBlKVG0UheVdzqwdIvAIg\nWbe1afHN3DeK3C9BB0i8AiABUuU2vDa6owNR2xEv/VR5HCDxCoAESM4VLv9HohNRo+umVX0F\nOkDiFQAJkPLmP3ad8+pco0GPzN9f/S8BiVcApLoNade8yQPrOxC1HpySpfsCdIDEKwBS3YW0\nMSPZeRc3dU5MW+7zRIDEKwBS3YS0Nj2ps8NQdI/kjGzDEwISrwBIdQ7SvsWp8c6H/TQcOHne\nXr+nBiReAZDqFCTnTaJYB6I28amLeS/hCEi8AiDVGUism0Q1Bki8AiDVBUj5X0+/6XSHodi+\nk97Vec88owESrwBI4Q5p67t/ubKRA1GzwSnzTTyfCJB4BUAKY0gFy9JGnhPhQNQl8R/LTT6I\nG5B4BUAKU0i7slIGOx+uENc7OWNLEAlA4hUAKQwhrU13363QJj41K9hnhwMSrwBI4QXJ+YPI\n+Uuiej2S0y15sR9A4hUAKXwgVf4gGpwS9A+iygESrwBIYQEpx+ofRJUDJF4BkJSH5P2DyLrX\nP/UMkHgFQFIZUs6ix685zfmL1l73zd4okwAkXgGQFIWUv2zmqPOcP4jaJzz9hcgLN7oHSLwC\nICkIafPbk65wPlihfp+/vPc/oYZngMQrAJJakPYvT0vsGuH5HZEN3vrSggIgsQZIVm1j5sTe\nzqeFN6p8sAIg8QZIvEL4Q9qdlep6kZ+oroneTyQCJN4AiVcIb0hr05N7x7hen2RyZrVXsgck\n3gCJVwhbSDvnTXb9njW6a1K63sO3AYk3QOIVwhHS/uXpSe57FQx+zwpIvAESrxBmkApWvHyP\n+16FKyZlGj4FApB4AyReIXwgFa56/b5+zl8RRXYbPXNZvr+TAxJvgMQrhAekjZnuG0SOK3OT\nM41fbM4zQOINkHgF5SE5DbUMzJBrgMQbIPEKKkP6yWGotcdQoM8JByTeAIlXUBTS9qzUxK5O\nQ6cNnJy5yUwBkHgDJF5BHFJJmamd9Pk3B759/rbuzju3m/abMHuDuf+6c5rvhEUrly+ExzdR\nLp6QLzC+idKgIP3+i5n9WqZ3bI7z55DTUOM+97z0XZGp/3JlQTsR1NczdlC+oB2TThySL2jF\n0oniI+IF7ZDf0/wWFCSLrtrt/OSJm89yGmoyYPysNab+m1WHq3a84aodr6DAbaQdHz8xzGWo\nYd97X11p8sUaawyQeAMkXsHekHY5r8s5n8vasHdy+nK/v2QNZIDEGyDxCraFtLOKId77qAQy\nQOINkHgFO0JyGuoW6X4anoQh1wCJN0DiFWwGaWfl/XKChlwDJN4AiVewD6QdHkPOn0Pfif9v\nEJB4AyRewRaQdtT4ORTyt3WxYoDEKwCSRZBedD7Wp+mV42etrrxvG5B4BUBirY5AWnLlhDfW\nVD0KkHgFQGKtjkDSGSDxCoDEGiDJDZB4AyReAZDkBki8AiABkuEAiVcAJEAyHCDxCoAESIYD\nJF4BkADJcIDEKwASIBkOkHgFQAIkwwESrwBIgGQ4QOIVAAmQDAdIvAIgAZLhAIlXACRAMhwg\n8QqABEiGAyReAZAAyXCAxCsAEiAZDpB4BUACJMMBEq8ASIBkOEDiFQAJkAwHSLwCIAGS4QCJ\nVwAkQDIcIPEKgARIhgMkXgGQAMlwgMQrABIgGQ6QeAVAAiTDARKvAEiAZDhA4hUACZAMB0i8\nAiABkuEAiVcAJEAyHCDxCoAESIYDJF4BkADJcIDEKwASIBkOkHgFQAIkwwESrwBIgGQ4QOIV\nAEnlFaf+J9RnIfjlpX4V6rMQ/Danfh/qsxD8VqT+HMCpwwnSbz0nhfosBL9NPWeE+iwEv0U9\n3w71WQh+c3ouCeDUgGSzAZJdBkhKD5DsMkBSeoBkl9VdSBgWsgEShlkwQMIwCwZIGGbBlIe0\nf9ro4c8c1LRf/3HbiEey3cdNGOLY8NCer4Dm+Sa8z/jhtJG3/L0gtOcroFV8ExuGuJblOk6p\nSyL3wQTnJ88/vPcF4P/CUB1Syb1P5e5J+ZumPfDIjn0zbjvmOnLM50XOBwwps8pvwvuMP/Xn\nbXufur88tOcsgHm+iRLnQ2d+Gr7HdaRKl8SypJkuSJ5/eO8LwP+FoTqk7CGOi+nXITmHnsnV\ntMIh7h9Jt6wJ8bkKcJ5vwvuMFw3Z4fg/wpvWhfBsBbbKb8K5R99xH6nSJbGkcKUTkucf3vsC\nYFwYqkP6acghTTs5dLHrD1vii5yfSob8a/yYp/NCer4Cmueb8D7jK24+6fg4QZ1HD3pfEt+M\nLXEdp9gl4YLk+Yf3vgAYF4bqkI6OfrX0+OybPnYePnTfLNdxB29/Zkv235OOhPSMBTLPN+F9\nxr+80/lx6uuhPWcBzOuSKB37tfs4xS4JFyTPP7z3BcC4MFSHpG2856YR7435zHFo7z0vnzx1\n/LHEL0N2ngLeqW+i8ox/Ocb5cYo6kLy+iSVjvG9NqHNJuCFV/MN7XwCMC0N5SJp2pOzETSs0\nbf2orCpH3/dOiM6PqVV8E865z/j3rmsT4z8I4XkKeJ5vYspbVY5W5pJwQfL8w3tfAIwLQ3VI\n5csOaNp3CQe1TaN+8ByX86LjGvrRW5aG8nwFNM834X3GD8RvdVw1StgY6vPGXuUloR29aUPF\ncYpdEi5Inn947wuAcWGoDkmbNO2XbXe/qp1Inuu82/WYtvAz7dComftzp911LNRnjb+Kb6Ly\njDu+Ce3ZCdv2PPbgSf9fbJdVfBOatmnI784/q3ZJHChamOD8X5DnH97zmXdhKA8p99Fbbsso\n1dZ7fg04Y6qmbZs64rZp+aE+ZwGs4puoPOPOb6L4hRHDpx3w/7W2meeb0JbFu/4Xp9olMdb1\nv6BPK//hPZ95F4bykDDMDgMkDLNggIRhFgyQMMyCARKGWTBAwjALBkgYZsEACcMsGCBhmAUD\nJAyzYICk+k6E+gxgzgGSKuvV75te9ZvfedBx8LtrGte/OMNx4PKB8ztcpmlfXB7XatjWUJ/B\nuj1AUmX92ly8cHt6dIKmfV1v4KcL76XnNG3gRefP/EybH3lNZsaZp+8P9Tms0wMkVdafnC9F\nfQ/t1np2Oe44FN/0qHY1feQ4dNG5ZZq2KuZfIT6DdXuApMr6xzm4aG/TR0U07phjr9Jy7ero\nUk0rpPGhPm8YICmz/mc4P35Br62jis3Vrj7dcdQ6eirEZw0DJHXW/yznxwU060cas9K1Iu3q\nzpoT0pTQnjNMAyR11r+B86V53qDPfqU7PMe5IBW5/5xTGJKzhbkHSKqsPzlf1Soxcr/Wu+lv\njkNzppS6IWk9mh/RtC30REjPXl0fIKmy/u3PTl/4F7pV076pd8G/v5pab0zFTyRtfmSvjNfP\nbavMC5qG5QBJlfU/b82AuGZ3H3YcXH5t4/rdppd5IGkLLo9rOTQ7pOeuzg+QVFn/rqE+B5jB\nAEmVAZKtB0iqDJBsPUBSZYBk6wEShlkwQMIwCwZIGGbBAAnDLBggYZgFAyQMs2CAhGEWDJAw\nzIL9P+Hg8sQSB1E+AAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "q_df %>%\n",
    "    ggplot(aes(x = perc, y = log(quant, base = 10))) +\n",
    "    geom_line()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "fb7b3e81",
   "metadata": {},
   "outputs": [],
   "source": [
    "min_samples <- 100\n",
    "# threshold_df <- tibble()\n",
    "\n",
    "# for (perc in seq(95, 100, 0.5)) {\n",
    "#     read_threshold <- quantile(long_RA$rel_a, perc / 100)\n",
    "#     full_species_df <- decontaminate(\"S\", 9999, 9999, \"Homo sapiens\", read_threshold = read_threshold, min_samples = min_samples)\n",
    "#     full_species_res <- parse_decontamination_results(full_species_df)\n",
    "# #     fwrite(full_species_df, str_glue(\"results/decontamination/RA_thresholds/diff_prev_results.9999.S.RA.perc{perc}_t{read_threshold}_minsamples{min_samples}.csv\"))\n",
    "#     fwrite(tibble(non_contaminants = full_species_res[[\"non_contaminants\"]]), str_glue(\"results/decontamination/RA_thresholds/prevalence.noncontam.S.n9999.RA.perc{perc}_t{read_threshold}_minsamples{min_samples}.txt\"))\n",
    "#     threshold_df <- threshold_df %>% bind_rows(tibble(perc = perc, read_threshold = read_threshold, n_nc = length(full_species_res[[\"non_contaminants\"]])))\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "25df615c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# threshold_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "059297d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n",
      "1792 / 5199 taxa are present in 0.05 of samples\n",
      "0/8728 samples removed due to having no reads of interest\n",
      "[1] \"site_supplying_sample\"\n",
      "After pruning, site_supplying_sample has 6 levels\n",
      "[1] \"extraction_kit\"\n",
      "After pruning, extraction_kit has 6 levels\n",
      "[1] \"instrument_id\"\n",
      "After pruning, instrument_id has 5 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 20 levels\n",
      "[1] \"hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 19 levels\n",
      "[1] \"hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot\"\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 23 levels\n",
      "[1] \"hiseq_xtm_flow_cell_v2_5_lot\"\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 19 levels\n"
     ]
    }
   ],
   "source": [
    "\n",
    "threshold_df2 <- tibble()\n",
    "\n",
    "## For RA ##\n",
    "# for (read_threshold in c(0, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.5)) {\n",
    "#     full_species_df <- decontaminate(\"S\", 9999, 9999, \"Homo sapiens\", read_threshold = read_threshold, min_samples = min_samples)\n",
    "#     full_species_res <- parse_decontamination_results(full_species_df)\n",
    "# #     fwrite(full_species_df, str_glue(\"results/decontamination/RA_thresholds/diff_prev_results.9999.S.RA.perc_{read_threshold}.csv\"))\n",
    "#     fwrite(tibble(non_contaminants = full_species_res[[\"non_contaminants\"]]), str_glue(\"results/decontamination/RA_thresholds/prevalence.noncontam.S.n9999.RA.perc{perc}_t{read_threshold}_minsamples{min_samples}.txt\"))\n",
    "#     threshold_df2 <- threshold_df2 %>% bind_rows(tibble(read_threshold = read_threshold, n_nc = length(full_species_res[[\"non_contaminants\"]])))\n",
    "# }\n",
    "\n",
    "\n",
    "## For reads ##\n",
    "for (read_threshold in c(0, 1, 2, 3, 4, 5, 10, 20, 30, 50, 100)) {\n",
    "    full_species_df <- decontaminate(\"S\", 9999, 9999, \"Homo sapiens\", read_threshold = read_threshold, min_samples = min_samples)\n",
    "    full_species_res <- parse_decontamination_results(full_species_df)\n",
    "#     fwrite(full_species_df, str_glue(\"results/decontamination/RA_thresholds/diff_prev_results.9999.S.perc_{read_threshold}.csv\"))\n",
    "    fwrite(tibble(non_contaminants = full_species_res[[\"non_contaminants\"]]), str_glue(\"results/decontamination/RA_thresholds/prevalence.noncontam.S.n9999.perc_t{read_threshold}_minsamples{min_samples}.txt\"))\n",
    "    threshold_df2 <- threshold_df2 %>% bind_rows(tibble(read_threshold = read_threshold, n_nc = length(full_species_res[[\"non_contaminants\"]])))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f2ea2656",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 11 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>read_threshold</th><th scope=col>n_nc</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td>2</td></tr>\n",
       "\t<tr><td>  1</td><td>0</td></tr>\n",
       "\t<tr><td>  2</td><td>0</td></tr>\n",
       "\t<tr><td>  3</td><td>0</td></tr>\n",
       "\t<tr><td>  4</td><td>0</td></tr>\n",
       "\t<tr><td>  5</td><td>0</td></tr>\n",
       "\t<tr><td> 10</td><td>0</td></tr>\n",
       "\t<tr><td> 20</td><td>0</td></tr>\n",
       "\t<tr><td> 30</td><td>0</td></tr>\n",
       "\t<tr><td> 50</td><td>0</td></tr>\n",
       "\t<tr><td>100</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 11 × 2\n",
       "\\begin{tabular}{ll}\n",
       " read\\_threshold & n\\_nc\\\\\n",
       " <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t   0 & 2\\\\\n",
       "\t   1 & 0\\\\\n",
       "\t   2 & 0\\\\\n",
       "\t   3 & 0\\\\\n",
       "\t   4 & 0\\\\\n",
       "\t   5 & 0\\\\\n",
       "\t  10 & 0\\\\\n",
       "\t  20 & 0\\\\\n",
       "\t  30 & 0\\\\\n",
       "\t  50 & 0\\\\\n",
       "\t 100 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 11 × 2\n",
       "\n",
       "| read_threshold &lt;dbl&gt; | n_nc &lt;int&gt; |\n",
       "|---|---|\n",
       "|   0 | 2 |\n",
       "|   1 | 0 |\n",
       "|   2 | 0 |\n",
       "|   3 | 0 |\n",
       "|   4 | 0 |\n",
       "|   5 | 0 |\n",
       "|  10 | 0 |\n",
       "|  20 | 0 |\n",
       "|  30 | 0 |\n",
       "|  50 | 0 |\n",
       "| 100 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "   read_threshold n_nc\n",
       "1    0            2   \n",
       "2    1            0   \n",
       "3    2            0   \n",
       "4    3            0   \n",
       "5    4            0   \n",
       "6    5            0   \n",
       "7   10            0   \n",
       "8   20            0   \n",
       "9   30            0   \n",
       "10  50            0   \n",
       "11 100            0   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "threshold_df2"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
