{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "844f2b78",
   "metadata": {},
   "source": [
    "# iRep (bPTR) and coverage breadth analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "53041a0d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rsamtools\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mBiostrings\u001b[39m::collapse(), \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mcompact()\u001b[39m    masks \u001b[34mXVector\u001b[39m::compact()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mXVector\u001b[39m::slice(), \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "\n",
      "Attaching package: ‘foreach’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:purrr’:\n",
      "\n",
      "    accumulate, when\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: ape\n",
      "\n",
      "\n",
      "Attaching package: ‘ape’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biostrings’:\n",
      "\n",
      "    complement\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm(list = ls())\n",
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(\"Rsamtools\")\n",
    "require(Biostrings)\n",
    "require(tidyverse)\n",
    "require(data.table)\n",
    "require(foreach)\n",
    "require(doParallel)\n",
    "require(ape)\n",
    "registerDoParallel(cores = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c3332d32",
   "metadata": {},
   "outputs": [],
   "source": [
    "species_df <- fread(\"results/decontamination/read_matrix.raw.zeroed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6673e418",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 4 rows [1, 2, 3, 4].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 3 rows [1, 2, 3].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 3 rows [1, 2, 3].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n"
     ]
    }
   ],
   "source": [
    "files <- list.files(\"results/irep_analysis/raw_output/irep_out\", full.names = T)\n",
    "files <- files[grepl(\".tsv\", files)]\n",
    "\n",
    "morsels <- foreach(file = files) %do% {\n",
    "    temp_df <- read.csv(file, sep = \"\\t\", check.names = F)\n",
    "    colnames(temp_df)[1] <- \"ref_path\"\n",
    "    temp_df %>% \n",
    "        mutate(across(c(4:ncol(temp_df)), function(x) (as.numeric(ifelse(x == \"n/a\", NA, x)))),\n",
    "               across(c(1:3), as.character)) %>%\n",
    "        pivot_longer(!all_of(c(\"ref_path\", \"ORI\", \"TER\")), names_to = \"prefix\", values_to = \"bPTR\") %>%\n",
    "        separate(ref_path, sep = \"/\", into = c(rep(NA, 11), \"file_name\")) %>%\n",
    "        separate(file_name, sep = \"_\", into = c(\"genus\", \"species\", \"suffix1\", \"suffix2\"), remove = F) %>%\n",
    "        separate(prefix, into = c(rep(NA, 6), \"id\"), sep = \"/\") %>%\n",
    "        separate(id, into = c(\"npm_research_id\"), sep = \"_\") %>%\n",
    "        mutate(prefix = gsub(\".fasta\", \"\", file_name))\n",
    "}\n",
    "\n",
    "irep_df <- bind_rows(morsels) \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd89f3e8",
   "metadata": {},
   "source": [
    "## Parse coverage breadth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5155b4c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Achromobacter_xylosoxidans_NZ_CP043820.1'</li><li>'Acinetobacter_baumannii_NZ_CP043953.1'</li><li>'Alcaligenes_faecalis_NZ_CP013119.1'</li><li>'Aspergillus_oryzae_NC_036435.1'</li><li>'Corynebacterium_segmentosum_NZ_LR134408.1'</li><li>'Cupriavidus_metallidurans_NZ_FYAX01000013.1'</li><li>'Cutibacterium_acnes_NC_021085.1'</li><li>'Fannyhessea_vaginae_NZ_UFSV01000001.1'</li><li>'Fusobacterium_nucleatum_NZ_LN831027.1'</li><li>'Fusobacterium_pseudoperiodonticum_NZ_CP024702.1'</li><li>'Gardnerella_vaginalis_NZ_PKJK01000001.1'</li><li>'Haemophilus_parainfluenzae_NZ_GL872339.1'</li><li>'Hepatitis_B_virus_genB_JX661488.1'</li><li>'Human_betaherpesvirus_6A_NC_001664.4'</li><li>'Human_betaherpesvirus_6B_NC_000898.1'</li><li>'Human_mastadenovirus_C_NC_001405.1'</li><li>'Lactobacillus_crispatus_NZ_CP039266.1'</li><li>'Malassezia_restricta_NC_040193.1'</li><li>'Moraxella_osloensis_NZ_CP014234.1'</li><li>'Neisseria_flavescens_NZ_CP039886.1'</li><li>'Neisseria_mucosa_NZ_UGRT01000001.1'</li><li>'Neisseria_subflava_NZ_CP039887.1'</li><li>'Prevotella_melaninogenica_NC_014370.1'</li><li>'Prevotella_oral_taxon_299_NC_022124.1'</li><li>'Pseudomonas_mendocina_NZ_CP013124.1'</li><li>'Rickettsia_Tillamook_23_NZ_CP060138.1'</li><li>'Staphylococcus_cohnii_NZ_UHDA01000001.1'</li><li>'Staphylococcus_epidermidis_NZ_CP035288.1'</li><li>'Staphylococcus_haemolyticus_NZ_CP013911.1'</li><li>'Torque_teno_virus_6_NC_014094.1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1'\n",
       "\\item 'Acinetobacter\\_baumannii\\_NZ\\_CP043953.1'\n",
       "\\item 'Alcaligenes\\_faecalis\\_NZ\\_CP013119.1'\n",
       "\\item 'Aspergillus\\_oryzae\\_NC\\_036435.1'\n",
       "\\item 'Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1'\n",
       "\\item 'Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1'\n",
       "\\item 'Cutibacterium\\_acnes\\_NC\\_021085.1'\n",
       "\\item 'Fannyhessea\\_vaginae\\_NZ\\_UFSV01000001.1'\n",
       "\\item 'Fusobacterium\\_nucleatum\\_NZ\\_LN831027.1'\n",
       "\\item 'Fusobacterium\\_pseudoperiodonticum\\_NZ\\_CP024702.1'\n",
       "\\item 'Gardnerella\\_vaginalis\\_NZ\\_PKJK01000001.1'\n",
       "\\item 'Haemophilus\\_parainfluenzae\\_NZ\\_GL872339.1'\n",
       "\\item 'Hepatitis\\_B\\_virus\\_genB\\_JX661488.1'\n",
       "\\item 'Human\\_betaherpesvirus\\_6A\\_NC\\_001664.4'\n",
       "\\item 'Human\\_betaherpesvirus\\_6B\\_NC\\_000898.1'\n",
       "\\item 'Human\\_mastadenovirus\\_C\\_NC\\_001405.1'\n",
       "\\item 'Lactobacillus\\_crispatus\\_NZ\\_CP039266.1'\n",
       "\\item 'Malassezia\\_restricta\\_NC\\_040193.1'\n",
       "\\item 'Moraxella\\_osloensis\\_NZ\\_CP014234.1'\n",
       "\\item 'Neisseria\\_flavescens\\_NZ\\_CP039886.1'\n",
       "\\item 'Neisseria\\_mucosa\\_NZ\\_UGRT01000001.1'\n",
       "\\item 'Neisseria\\_subflava\\_NZ\\_CP039887.1'\n",
       "\\item 'Prevotella\\_melaninogenica\\_NC\\_014370.1'\n",
       "\\item 'Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1'\n",
       "\\item 'Pseudomonas\\_mendocina\\_NZ\\_CP013124.1'\n",
       "\\item 'Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1'\n",
       "\\item 'Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1'\n",
       "\\item 'Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1'\n",
       "\\item 'Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1'\n",
       "\\item 'Torque\\_teno\\_virus\\_6\\_NC\\_014094.1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Achromobacter_xylosoxidans_NZ_CP043820.1'\n",
       "2. 'Acinetobacter_baumannii_NZ_CP043953.1'\n",
       "3. 'Alcaligenes_faecalis_NZ_CP013119.1'\n",
       "4. 'Aspergillus_oryzae_NC_036435.1'\n",
       "5. 'Corynebacterium_segmentosum_NZ_LR134408.1'\n",
       "6. 'Cupriavidus_metallidurans_NZ_FYAX01000013.1'\n",
       "7. 'Cutibacterium_acnes_NC_021085.1'\n",
       "8. 'Fannyhessea_vaginae_NZ_UFSV01000001.1'\n",
       "9. 'Fusobacterium_nucleatum_NZ_LN831027.1'\n",
       "10. 'Fusobacterium_pseudoperiodonticum_NZ_CP024702.1'\n",
       "11. 'Gardnerella_vaginalis_NZ_PKJK01000001.1'\n",
       "12. 'Haemophilus_parainfluenzae_NZ_GL872339.1'\n",
       "13. 'Hepatitis_B_virus_genB_JX661488.1'\n",
       "14. 'Human_betaherpesvirus_6A_NC_001664.4'\n",
       "15. 'Human_betaherpesvirus_6B_NC_000898.1'\n",
       "16. 'Human_mastadenovirus_C_NC_001405.1'\n",
       "17. 'Lactobacillus_crispatus_NZ_CP039266.1'\n",
       "18. 'Malassezia_restricta_NC_040193.1'\n",
       "19. 'Moraxella_osloensis_NZ_CP014234.1'\n",
       "20. 'Neisseria_flavescens_NZ_CP039886.1'\n",
       "21. 'Neisseria_mucosa_NZ_UGRT01000001.1'\n",
       "22. 'Neisseria_subflava_NZ_CP039887.1'\n",
       "23. 'Prevotella_melaninogenica_NC_014370.1'\n",
       "24. 'Prevotella_oral_taxon_299_NC_022124.1'\n",
       "25. 'Pseudomonas_mendocina_NZ_CP013124.1'\n",
       "26. 'Rickettsia_Tillamook_23_NZ_CP060138.1'\n",
       "27. 'Staphylococcus_cohnii_NZ_UHDA01000001.1'\n",
       "28. 'Staphylococcus_epidermidis_NZ_CP035288.1'\n",
       "29. 'Staphylococcus_haemolyticus_NZ_CP013911.1'\n",
       "30. 'Torque_teno_virus_6_NC_014094.1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Achromobacter_xylosoxidans_NZ_CP043820.1\"       \n",
       " [2] \"Acinetobacter_baumannii_NZ_CP043953.1\"          \n",
       " [3] \"Alcaligenes_faecalis_NZ_CP013119.1\"             \n",
       " [4] \"Aspergillus_oryzae_NC_036435.1\"                 \n",
       " [5] \"Corynebacterium_segmentosum_NZ_LR134408.1\"      \n",
       " [6] \"Cupriavidus_metallidurans_NZ_FYAX01000013.1\"    \n",
       " [7] \"Cutibacterium_acnes_NC_021085.1\"                \n",
       " [8] \"Fannyhessea_vaginae_NZ_UFSV01000001.1\"          \n",
       " [9] \"Fusobacterium_nucleatum_NZ_LN831027.1\"          \n",
       "[10] \"Fusobacterium_pseudoperiodonticum_NZ_CP024702.1\"\n",
       "[11] \"Gardnerella_vaginalis_NZ_PKJK01000001.1\"        \n",
       "[12] \"Haemophilus_parainfluenzae_NZ_GL872339.1\"       \n",
       "[13] \"Hepatitis_B_virus_genB_JX661488.1\"              \n",
       "[14] \"Human_betaherpesvirus_6A_NC_001664.4\"           \n",
       "[15] \"Human_betaherpesvirus_6B_NC_000898.1\"           \n",
       "[16] \"Human_mastadenovirus_C_NC_001405.1\"             \n",
       "[17] \"Lactobacillus_crispatus_NZ_CP039266.1\"          \n",
       "[18] \"Malassezia_restricta_NC_040193.1\"               \n",
       "[19] \"Moraxella_osloensis_NZ_CP014234.1\"              \n",
       "[20] \"Neisseria_flavescens_NZ_CP039886.1\"             \n",
       "[21] \"Neisseria_mucosa_NZ_UGRT01000001.1\"             \n",
       "[22] \"Neisseria_subflava_NZ_CP039887.1\"               \n",
       "[23] \"Prevotella_melaninogenica_NC_014370.1\"          \n",
       "[24] \"Prevotella_oral_taxon_299_NC_022124.1\"          \n",
       "[25] \"Pseudomonas_mendocina_NZ_CP013124.1\"            \n",
       "[26] \"Rickettsia_Tillamook_23_NZ_CP060138.1\"          \n",
       "[27] \"Staphylococcus_cohnii_NZ_UHDA01000001.1\"        \n",
       "[28] \"Staphylococcus_epidermidis_NZ_CP035288.1\"       \n",
       "[29] \"Staphylococcus_haemolyticus_NZ_CP013911.1\"      \n",
       "[30] \"Torque_teno_virus_6_NC_014094.1\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_path <- \"results/irep_analysis/raw_output/bam_files\"\n",
    "ref_path <- \"data/irep_data/genome_references\"\n",
    "ref_files <- list.files(ref_path, \".fasta\", full.names = F)\n",
    "prefixes <- gsub(\".fasta\", \"\", ref_files)\n",
    "prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0d49d2e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "morsels <- foreach (i = seq(length(prefixes))) %dopar% {\n",
    "    prefix <- prefixes[i]\n",
    "    bams <- list.files(str_glue(\"{bam_path}/{prefix}\"))\n",
    "    bams <- bams[!grepl(\".bai\", bams)]\n",
    "\n",
    "    crumbs <- foreach (j = seq(length(bams))) %do% {\n",
    "        bam <- bams[j]\n",
    "        id <- str_split(bam, \"_\")[[1]][1]\n",
    "        ref <- readDNAStringSet(str_glue(\"{ref_path}/{prefixes[i]}.fasta\"))\n",
    "        ref_length <- width(ref)\n",
    "        params <- PileupParam(distinguish_strands = F, distinguish_nucleotides = F)\n",
    "        pile_df <- pileup(str_glue(\"{bam_path}/{prefix}/{bam}\"), pileupParam = params)\n",
    "        \n",
    "        n_covered1 <- pile_df %>%\n",
    "            filter(count >= 1) %>%\n",
    "            nrow()\n",
    "        \n",
    "        n_covered5 <- pile_df %>%\n",
    "            filter(count >= 5) %>%\n",
    "            nrow()\n",
    "\n",
    "        perc_covered1 <- n_covered1 / ref_length * 100\n",
    "        perc_covered5 <- n_covered5 / ref_length * 100\n",
    "        \n",
    "        taxon <- str_split(prefix, \"_\")[[1]]\n",
    "            read_count <- species_df %>% \n",
    "                filter(npm_research_id == id) %>%\n",
    "                select(contains(taxon[1])) %>%\n",
    "                select(contains(taxon[2])) %>%\n",
    "                select(!contains(\"monkey\"))\n",
    "\n",
    "        if (ncol(read_count) > 1) {\n",
    "            read_count <- read_count %>% \n",
    "                select(contains(taxon[3]))\n",
    "            if (ncol(read_count) > 1) {\n",
    "            read_count <- read_count %>% \n",
    "                select(contains(paste0(\" \", taxon[4])))\n",
    "            }\n",
    "        }\n",
    "        \n",
    "        if (all(dim(read_count) == 1)) {\n",
    "            read_count <- deframe(read_count)\n",
    "        } else {\n",
    "            read_count <- NA\n",
    "        }\n",
    "        \n",
    "        return(tibble(prefix = prefix, \n",
    "                      npm_research_id = id,\n",
    "                      pairs_assigned = read_count,\n",
    "                      perc_covered1 = perc_covered1,\n",
    "                      perc_covered5 = perc_covered5))\n",
    "    }\n",
    "    \n",
    "    return(bind_rows(crumbs))\n",
    "}\n",
    "\n",
    "cov_df <- bind_rows(morsels)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56cba5fc",
   "metadata": {},
   "source": [
    "### Add number of human or microbial reads mapped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "88d455ed",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "microbe_df <- fread(\"results/irep_analysis/raw_output/microbial_mapped_read_counts.csv\") %>%\n",
    "    rename(microbe_pairs_mapped = pairs_mapped)\n",
    "human_df <- fread(\"results/irep_analysis/raw_output/human_mapped_read_counts.csv\") %>%\n",
    "    rename(human_pairs_mapped = pairs_mapped)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03783a41",
   "metadata": {},
   "source": [
    "### Add microbe to human cell ratio\n",
    "Estimates from https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-019-4137-z/tables/2\n",
    "Leucocyte count from https://pubmed.ncbi.nlm.nih.gov/15782774/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "12c3851d",
   "metadata": {},
   "outputs": [],
   "source": [
    "human_length <- 6320012150\n",
    "human_gc <- 40.89\n",
    "leuco_per_ul <- 6750"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fe49a527",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_paths <- paste0(ref_path, \"/\", ref_files)\n",
    "\n",
    "length_morsels <- foreach(prefix_name = prefixes) %do% {\n",
    "    microbe_ref_path <- ref_paths[grepl(prefix_name, ref_paths)]\n",
    "    microbe_ref <- read.FASTA(microbe_ref_path)\n",
    "    microbe_length <- length(as.character(microbe_ref)[[1]])\n",
    "\n",
    "    tibble(prefix = prefix_name, \n",
    "           microbe_length = microbe_length)\n",
    "}\n",
    "\n",
    "length_df <- bind_rows(length_morsels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "38f7cc67",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Joining, by = c(\"prefix\", \"npm_research_id\")\n",
      "Joining, by = \"npm_research_id\"\n",
      "Joining, by = c(\"prefix\", \"npm_research_id\")\n",
      "Joining, by = \"prefix\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 138 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>prefix</th><th scope=col>npm_research_id</th><th scope=col>pairs_assigned</th><th scope=col>microbe_pairs_mapped</th><th scope=col>human_pairs_mapped</th><th scope=col>perc_covered1</th><th scope=col>perc_covered5</th><th scope=col>file_name</th><th scope=col>genus</th><th scope=col>species</th><th scope=col>suffix1</th><th scope=col>suffix2</th><th scope=col>ORI</th><th scope=col>TER</th><th scope=col>bPTR</th><th scope=col>microbe_length</th><th scope=col>microbe_count</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>1116-0023</td><td>430178</td><td>429196</td><td>348366222</td><td>79.292320359</td><td>7.450282e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>8.208410e+03</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3658  </td><td>385855</td><td>438419</td><td>364361186</td><td>78.389787758</td><td>7.366085e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>8.016719e+03</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3674  </td><td>459561</td><td>494858</td><td>327626517</td><td>78.805094251</td><td>7.456206e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>1.006331e+04</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3686  </td><td>541795</td><td>599335</td><td>344250628</td><td>79.082105806</td><td>7.525644e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>1.159937e+04</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3734  </td><td>688629</td><td>769978</td><td>370885795</td><td>79.583059893</td><td>7.609348e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>1.383176e+04</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>0116-0053</td><td>  7673</td><td> 28842</td><td>364796907</td><td>79.329072139</td><td>1.697212e+01</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>1.896385</td><td>3972439</td><td>8.490606e+02</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4240  </td><td>   168</td><td>   210</td><td>365572623</td><td> 1.316042864</td><td>1.963529e-03</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>6.168933e+00</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4251  </td><td>   183</td><td>   233</td><td>284142227</td><td> 1.460462955</td><td>4.782956e-04</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>8.806120e+00</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4438  </td><td>   379</td><td>   453</td><td>408444066</td><td> 2.824813672</td><td>3.373242e-03</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>1.191050e+01</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4828  </td><td>   190</td><td>   231</td><td>371860124</td><td> 1.642920130</td><td>0.000000e+00</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>6.671090e+00</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9245  </td><td>   982</td><td>  6163</td><td>197279038</td><td>29.029117408</td><td>6.902854e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>3.147800e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9246  </td><td>   896</td><td>  5202</td><td>182833313</td><td>27.000516799</td><td>3.080480e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>2.866889e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9270  </td><td>   996</td><td>  6128</td><td>188061461</td><td>28.984145520</td><td>6.330785e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>3.283332e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9282  </td><td>   766</td><td>  4490</td><td>176734183</td><td>23.360108613</td><td>2.885617e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>2.559892e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9317  </td><td>  1104</td><td>  7106</td><td>197010890</td><td>31.747814470</td><td>1.073538e+00</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>3.634384e+02</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB3883  </td><td>    24</td><td>     4</td><td>313105031</td><td> 0.013741771</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>8.358461e-02</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4048  </td><td>    32</td><td>     0</td><td>352035872</td><td> 0.000000000</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4056  </td><td>  7909</td><td>    31</td><td>364502596</td><td> 0.169732339</td><td>4.370987e-03</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>5.564388e-01</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4130  </td><td>    99</td><td>     0</td><td>404269811</td><td> 0.002315856</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHH1822  </td><td>    12</td><td>     0</td><td>164835319</td><td> 0.000000000</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6459  </td><td> 14476</td><td> 13069</td><td>183352926</td><td>66.396833288</td><td>7.454033e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>1.281144e+03</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6776  </td><td>   215</td><td>   200</td><td>152980243</td><td> 2.189016786</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>2.349839e+01</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6849  </td><td>    31</td><td>    25</td><td>165286847</td><td> 0.249342726</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>2.718599e+00</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB8186  </td><td>    79</td><td>    83</td><td>183019538</td><td> 0.799303964</td><td>6.741270e-04</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>8.151247e+00</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB8568  </td><td>    38</td><td>    35</td><td>175163865</td><td> 0.346627680</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>3.591426e+00</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB6155  </td><td>   152</td><td>    19</td><td>133124278</td><td> 0.444721342</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>4.991215e+00</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9217  </td><td>   297</td><td>    31</td><td>150666053</td><td> 0.701307112</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>7.195421e+00</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9583  </td><td>  5506</td><td>   792</td><td>169311308</td><td>15.704852586</td><td>4.270964e-02</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>1.635871e+02</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9649  </td><td>  1972</td><td>   290</td><td>199253589</td><td> 6.521295389</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>5.089812e+01</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9882  </td><td>   835</td><td>   115</td><td>194005959</td><td> 2.457894931</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>2.072968e+01</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB3939  </td><td>    141</td><td>    74</td><td>337439179</td><td> 1.0932372</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>5.207783e+00</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB5008  </td><td>     79</td><td>    34</td><td>367155498</td><td> 0.4487845</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>2.199103e+00</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB8397  </td><td>    150</td><td>    65</td><td>169782090</td><td> 0.9019109</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>9.091555e+00</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB9179  </td><td>   3549</td><td>  1971</td><td>232737013</td><td>25.1270313</td><td> 0.224948898</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>2.011119e+02</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1    </td><td>WHB4373  </td><td>     79</td><td>    25</td><td>372164907</td><td> 1.0017609</td><td> 0.000000000</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>      NA</td><td> 709850</td><td>4.037009e+00</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1    </td><td>WHB9179  </td><td>   2518</td><td>   637</td><td>232737013</td><td>20.7162076</td><td> 0.354018455</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>      NA</td><td> 709850</td><td>1.644861e+02</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0415-0080</td><td>1157405</td><td>625658</td><td>324097265</td><td>71.7810391</td><td>67.443467432</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>1.608426e+04</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0715-0061</td><td>1366176</td><td>478777</td><td>472980894</td><td>71.1279913</td><td>67.259332058</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>8.433911e+03</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0816-0014</td><td>1345056</td><td>368174</td><td>422738587</td><td>71.5775292</td><td>67.434151292</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>7.256389e+03</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>WHB3477  </td><td>1334294</td><td>748280</td><td>355475275</td><td>71.2235745</td><td>67.240309163</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>1.753857e+04</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>WHB3479  </td><td>1431040</td><td>797696</td><td>334160371</td><td>71.3566957</td><td>67.599927815</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>1.988941e+04</td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td><td>WHB6775  </td><td>     12</td><td>    21</td><td>148611421</td><td> 0.3587976</td><td> 0.000000000</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>      NA</td><td>1438973</td><td>4.189249e+00</td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td><td>WHB9978  </td><td>   2923</td><td>  6032</td><td>149284170</td><td>63.3608136</td><td> 5.514696940</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>1.346660</td><td>1438973</td><td>1.197889e+03</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0021</td><td>  12261</td><td>     6</td><td>333337958</td><td> 0.2456782</td><td> 0.030264437</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>2.879700e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0024</td><td>  18768</td><td>     0</td><td>378247853</td><td> 0.2461658</td><td> 0.048340594</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0026</td><td>  14111</td><td>     2</td><td>419260819</td><td> 0.2183765</td><td> 0.045902938</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>7.631791e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0031</td><td>  16398</td><td>     3</td><td>521164286</td><td> 0.2671671</td><td> 0.117420015</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>9.209316e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0032</td><td>  18246</td><td>     8</td><td>416012648</td><td> 0.2688547</td><td> 0.062779018</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>3.076552e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB4219  </td><td>    992</td><td>   922</td><td>355718750</td><td> 8.6830661</td><td> 0.016703818</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>4.482955e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB4398  </td><td>    546</td><td>   538</td><td>341262611</td><td> 5.4451608</td><td> 0.005797684</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>2.726678e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB6459  </td><td>   9140</td><td>  7791</td><td>183352926</td><td>52.9612382</td><td> 1.443299053</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>1.566844</td><td>2466502</td><td>7.349293e+02</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB6651  </td><td>    830</td><td>   770</td><td>171449646</td><td> 7.6927162</td><td> 0.008392452</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>7.767734e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB8186  </td><td>    232</td><td>   215</td><td>183019538</td><td> 2.0381090</td><td> 0.005230079</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>2.031801e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0024</td><td> 117929</td><td>     2</td><td>378247853</td><td> 0.2597145</td><td> 0.047698657</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>8.848055e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0026</td><td>  94270</td><td>     1</td><td>419260819</td><td> 0.2356298</td><td> 0.038794385</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>3.991260e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0027</td><td>  76946</td><td>     8</td><td>416138008</td><td> 0.2644608</td><td> 0.066566301</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>3.216969e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0031</td><td> 108245</td><td>     5</td><td>521164286</td><td> 0.2961161</td><td> 0.119246644</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>1.605424e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0715-0059</td><td>  81100</td><td>     7</td><td>509004187</td><td> 0.2889770</td><td> 0.151725664</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>2.301288e-01</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1          </td><td>WHB7619  </td><td>    496</td><td>    60</td><td>154301162</td><td>21.5924426</td><td>19.946018893</td><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>      NA</td><td>   3705</td><td>4.477293e+03</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1          </td><td>WHB9685  </td><td>     39</td><td>    22</td><td>171579152</td><td>16.0863698</td><td>14.170040486</td><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>      NA</td><td>   3705</td><td>1.476358e+03</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 138 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " prefix & npm\\_research\\_id & pairs\\_assigned & microbe\\_pairs\\_mapped & human\\_pairs\\_mapped & perc\\_covered1 & perc\\_covered5 & file\\_name & genus & species & suffix1 & suffix2 & ORI & TER & bPTR & microbe\\_length & microbe\\_count\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & 1116-0023 & 430178 & 429196 & 348366222 & 79.292320359 & 7.450282e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 8.208410e+03\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3658   & 385855 & 438419 & 364361186 & 78.389787758 & 7.366085e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 8.016719e+03\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3674   & 459561 & 494858 & 327626517 & 78.805094251 & 7.456206e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 1.006331e+04\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3686   & 541795 & 599335 & 344250628 & 79.082105806 & 7.525644e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 1.159937e+04\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3734   & 688629 & 769978 & 370885795 & 79.583059893 & 7.609348e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 1.383176e+04\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & 0116-0053 &   7673 &  28842 & 364796907 & 79.329072139 & 1.697212e+01 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & 1.896385 & 3972439 & 8.490606e+02\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4240   &    168 &    210 & 365572623 &  1.316042864 & 1.963529e-03 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 6.168933e+00\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4251   &    183 &    233 & 284142227 &  1.460462955 & 4.782956e-04 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 8.806120e+00\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4438   &    379 &    453 & 408444066 &  2.824813672 & 3.373242e-03 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 1.191050e+01\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4828   &    190 &    231 & 371860124 &  1.642920130 & 0.000000e+00 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 6.671090e+00\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9245   &    982 &   6163 & 197279038 & 29.029117408 & 6.902854e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 3.147800e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9246   &    896 &   5202 & 182833313 & 27.000516799 & 3.080480e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 2.866889e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9270   &    996 &   6128 & 188061461 & 28.984145520 & 6.330785e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 3.283332e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9282   &    766 &   4490 & 176734183 & 23.360108613 & 2.885617e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 2.559892e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9317   &   1104 &   7106 & 197010890 & 31.747814470 & 1.073538e+00 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 3.634384e+02\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB3883   &     24 &      4 & 313105031 &  0.013741771 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 8.358461e-02\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4048   &     32 &      0 & 352035872 &  0.000000000 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 0.000000e+00\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4056   &   7909 &     31 & 364502596 &  0.169732339 & 4.370987e-03 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 5.564388e-01\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4130   &     99 &      0 & 404269811 &  0.002315856 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 0.000000e+00\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHH1822   &     12 &      0 & 164835319 &  0.000000000 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 0.000000e+00\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6459   &  14476 &  13069 & 183352926 & 66.396833288 & 7.454033e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 1.281144e+03\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6776   &    215 &    200 & 152980243 &  2.189016786 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 2.349839e+01\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6849   &     31 &     25 & 165286847 &  0.249342726 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 2.718599e+00\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB8186   &     79 &     83 & 183019538 &  0.799303964 & 6.741270e-04 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 8.151247e+00\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB8568   &     38 &     35 & 175163865 &  0.346627680 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 3.591426e+00\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB6155   &    152 &     19 & 133124278 &  0.444721342 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 4.991215e+00\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9217   &    297 &     31 & 150666053 &  0.701307112 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 7.195421e+00\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9583   &   5506 &    792 & 169311308 & 15.704852586 & 4.270964e-02 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 1.635871e+02\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9649   &   1972 &    290 & 199253589 &  6.521295389 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 5.089812e+01\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9882   &    835 &    115 & 194005959 &  2.457894931 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 2.072968e+01\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB3939   &     141 &     74 & 337439179 &  1.0932372 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 5.207783e+00\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB5008   &      79 &     34 & 367155498 &  0.4487845 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 2.199103e+00\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB8397   &     150 &     65 & 169782090 &  0.9019109 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 9.091555e+00\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB9179   &    3549 &   1971 & 232737013 & 25.1270313 &  0.224948898 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 2.011119e+02\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1     & WHB4373   &      79 &     25 & 372164907 &  1.0017609 &  0.000000000 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         &       NA &  709850 & 4.037009e+00\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1     & WHB9179   &    2518 &    637 & 232737013 & 20.7162076 &  0.354018455 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         &       NA &  709850 & 1.644861e+02\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0415-0080 & 1157405 & 625658 & 324097265 & 71.7810391 & 67.443467432 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 1.608426e+04\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0715-0061 & 1366176 & 478777 & 472980894 & 71.1279913 & 67.259332058 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 8.433911e+03\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0816-0014 & 1345056 & 368174 & 422738587 & 71.5775292 & 67.434151292 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 7.256389e+03\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & WHB3477   & 1334294 & 748280 & 355475275 & 71.2235745 & 67.240309163 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 1.753857e+04\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & WHB3479   & 1431040 & 797696 & 334160371 & 71.3566957 & 67.599927815 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 1.988941e+04\\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1     & WHB6775   &      12 &     21 & 148611421 &  0.3587976 &  0.000000000 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   &       NA & 1438973 & 4.189249e+00\\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1     & WHB9978   &    2923 &   6032 & 149284170 & 63.3608136 &  5.514696940 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & 1.346660 & 1438973 & 1.197889e+03\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0021 &   12261 &      6 & 333337958 &  0.2456782 &  0.030264437 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 2.879700e-01\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0024 &   18768 &      0 & 378247853 &  0.2461658 &  0.048340594 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 0.000000e+00\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0026 &   14111 &      2 & 419260819 &  0.2183765 &  0.045902938 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 7.631791e-02\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0031 &   16398 &      3 & 521164286 &  0.2671671 &  0.117420015 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 9.209316e-02\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0032 &   18246 &      8 & 416012648 &  0.2688547 &  0.062779018 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 3.076552e-01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB4219   &     992 &    922 & 355718750 &  8.6830661 &  0.016703818 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 4.482955e+01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB4398   &     546 &    538 & 341262611 &  5.4451608 &  0.005797684 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 2.726678e+01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB6459   &    9140 &   7791 & 183352926 & 52.9612382 &  1.443299053 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & 1.566844 & 2466502 & 7.349293e+02\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB6651   &     830 &    770 & 171449646 &  7.6927162 &  0.008392452 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 7.767734e+01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB8186   &     232 &    215 & 183019538 &  2.0381090 &  0.005230079 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 2.031801e+01\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0024 &  117929 &      2 & 378247853 &  0.2597145 &  0.047698657 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 8.848055e-02\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0026 &   94270 &      1 & 419260819 &  0.2356298 &  0.038794385 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 3.991260e-02\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0027 &   76946 &      8 & 416138008 &  0.2644608 &  0.066566301 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 3.216969e-01\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0031 &  108245 &      5 & 521164286 &  0.2961161 &  0.119246644 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 1.605424e-01\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0715-0059 &   81100 &      7 & 509004187 &  0.2889770 &  0.151725664 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 2.301288e-01\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1           & WHB7619   &     496 &     60 & 154301162 & 21.5924426 & 19.946018893 & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         &       NA &    3705 & 4.477293e+03\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1           & WHB9685   &      39 &     22 & 171579152 & 16.0863698 & 14.170040486 & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         &       NA &    3705 & 1.476358e+03\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 138 × 17\n",
       "\n",
       "| prefix &lt;chr&gt; | npm_research_id &lt;chr&gt; | pairs_assigned &lt;int&gt; | microbe_pairs_mapped &lt;dbl&gt; | human_pairs_mapped &lt;dbl&gt; | perc_covered1 &lt;dbl&gt; | perc_covered5 &lt;dbl&gt; | file_name &lt;chr&gt; | genus &lt;chr&gt; | species &lt;chr&gt; | suffix1 &lt;chr&gt; | suffix2 &lt;chr&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; | bPTR &lt;dbl&gt; | microbe_length &lt;int&gt; | microbe_count &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | 1116-0023 | 430178 | 429196 | 348366222 | 79.292320359 | 7.450282e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 8.208410e+03 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3658   | 385855 | 438419 | 364361186 | 78.389787758 | 7.366085e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 8.016719e+03 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3674   | 459561 | 494858 | 327626517 | 78.805094251 | 7.456206e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 1.006331e+04 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3686   | 541795 | 599335 | 344250628 | 79.082105806 | 7.525644e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 1.159937e+04 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3734   | 688629 | 769978 | 370885795 | 79.583059893 | 7.609348e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 1.383176e+04 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | 0116-0053 |   7673 |  28842 | 364796907 | 79.329072139 | 1.697212e+01 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | 1.896385 | 3972439 | 8.490606e+02 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4240   |    168 |    210 | 365572623 |  1.316042864 | 1.963529e-03 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 6.168933e+00 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4251   |    183 |    233 | 284142227 |  1.460462955 | 4.782956e-04 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 8.806120e+00 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4438   |    379 |    453 | 408444066 |  2.824813672 | 3.373242e-03 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 1.191050e+01 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4828   |    190 |    231 | 371860124 |  1.642920130 | 0.000000e+00 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 6.671090e+00 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9245   |    982 |   6163 | 197279038 | 29.029117408 | 6.902854e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 3.147800e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9246   |    896 |   5202 | 182833313 | 27.000516799 | 3.080480e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 2.866889e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9270   |    996 |   6128 | 188061461 | 28.984145520 | 6.330785e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 3.283332e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9282   |    766 |   4490 | 176734183 | 23.360108613 | 2.885617e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 2.559892e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9317   |   1104 |   7106 | 197010890 | 31.747814470 | 1.073538e+00 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 3.634384e+02 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB3883   |     24 |      4 | 313105031 |  0.013741771 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 8.358461e-02 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4048   |     32 |      0 | 352035872 |  0.000000000 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 0.000000e+00 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4056   |   7909 |     31 | 364502596 |  0.169732339 | 4.370987e-03 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 5.564388e-01 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4130   |     99 |      0 | 404269811 |  0.002315856 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 0.000000e+00 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHH1822   |     12 |      0 | 164835319 |  0.000000000 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 0.000000e+00 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6459   |  14476 |  13069 | 183352926 | 66.396833288 | 7.454033e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 1.281144e+03 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6776   |    215 |    200 | 152980243 |  2.189016786 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 2.349839e+01 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6849   |     31 |     25 | 165286847 |  0.249342726 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 2.718599e+00 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB8186   |     79 |     83 | 183019538 |  0.799303964 | 6.741270e-04 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 8.151247e+00 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB8568   |     38 |     35 | 175163865 |  0.346627680 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 3.591426e+00 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB6155   |    152 |     19 | 133124278 |  0.444721342 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 4.991215e+00 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9217   |    297 |     31 | 150666053 |  0.701307112 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 7.195421e+00 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9583   |   5506 |    792 | 169311308 | 15.704852586 | 4.270964e-02 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 1.635871e+02 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9649   |   1972 |    290 | 199253589 |  6.521295389 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 5.089812e+01 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9882   |    835 |    115 | 194005959 |  2.457894931 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 2.072968e+01 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB3939   |     141 |     74 | 337439179 |  1.0932372 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 5.207783e+00 |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB5008   |      79 |     34 | 367155498 |  0.4487845 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 2.199103e+00 |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB8397   |     150 |     65 | 169782090 |  0.9019109 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 9.091555e+00 |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB9179   |    3549 |   1971 | 232737013 | 25.1270313 |  0.224948898 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 2.011119e+02 |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1     | WHB4373   |      79 |     25 | 372164907 |  1.0017609 |  0.000000000 | Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         |       NA |  709850 | 4.037009e+00 |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1     | WHB9179   |    2518 |    637 | 232737013 | 20.7162076 |  0.354018455 | Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         |       NA |  709850 | 1.644861e+02 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0415-0080 | 1157405 | 625658 | 324097265 | 71.7810391 | 67.443467432 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 1.608426e+04 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0715-0061 | 1366176 | 478777 | 472980894 | 71.1279913 | 67.259332058 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 8.433911e+03 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0816-0014 | 1345056 | 368174 | 422738587 | 71.5775292 | 67.434151292 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 7.256389e+03 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | WHB3477   | 1334294 | 748280 | 355475275 | 71.2235745 | 67.240309163 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 1.753857e+04 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | WHB3479   | 1431040 | 797696 | 334160371 | 71.3566957 | 67.599927815 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 1.988941e+04 |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1     | WHB6775   |      12 |     21 | 148611421 |  0.3587976 |  0.000000000 | Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   |       NA | 1438973 | 4.189249e+00 |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1     | WHB9978   |    2923 |   6032 | 149284170 | 63.3608136 |  5.514696940 | Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | 1.346660 | 1438973 | 1.197889e+03 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0021 |   12261 |      6 | 333337958 |  0.2456782 |  0.030264437 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 2.879700e-01 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0024 |   18768 |      0 | 378247853 |  0.2461658 |  0.048340594 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 0.000000e+00 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0026 |   14111 |      2 | 419260819 |  0.2183765 |  0.045902938 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 7.631791e-02 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0031 |   16398 |      3 | 521164286 |  0.2671671 |  0.117420015 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 9.209316e-02 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0032 |   18246 |      8 | 416012648 |  0.2688547 |  0.062779018 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 3.076552e-01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB4219   |     992 |    922 | 355718750 |  8.6830661 |  0.016703818 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 4.482955e+01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB4398   |     546 |    538 | 341262611 |  5.4451608 |  0.005797684 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 2.726678e+01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB6459   |    9140 |   7791 | 183352926 | 52.9612382 |  1.443299053 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | 1.566844 | 2466502 | 7.349293e+02 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB6651   |     830 |    770 | 171449646 |  7.6927162 |  0.008392452 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 7.767734e+01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB8186   |     232 |    215 | 183019538 |  2.0381090 |  0.005230079 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 2.031801e+01 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0024 |  117929 |      2 | 378247853 |  0.2597145 |  0.047698657 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 8.848055e-02 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0026 |   94270 |      1 | 419260819 |  0.2356298 |  0.038794385 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 3.991260e-02 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0027 |   76946 |      8 | 416138008 |  0.2644608 |  0.066566301 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 3.216969e-01 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0031 |  108245 |      5 | 521164286 |  0.2961161 |  0.119246644 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 1.605424e-01 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0715-0059 |   81100 |      7 | 509004187 |  0.2889770 |  0.151725664 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 2.301288e-01 |\n",
       "| Torque_teno_virus_6_NC_014094.1           | WHB7619   |     496 |     60 | 154301162 | 21.5924426 | 19.946018893 | Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         |       NA |    3705 | 4.477293e+03 |\n",
       "| Torque_teno_virus_6_NC_014094.1           | WHB9685   |      39 |     22 | 171579152 | 16.0863698 | 14.170040486 | Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         |       NA |    3705 | 1.476358e+03 |\n",
       "\n"
      ],
      "text/plain": [
       "    prefix                                      npm_research_id pairs_assigned\n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1    1116-0023       430178        \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3658         385855        \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3674         459561        \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3686         541795        \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3734         688629        \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1       0116-0053         7673        \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1       WHB4240            168        \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1       WHB4251            183        \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1       WHB4438            379        \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1       WHB4828            190        \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1          WHB9245            982        \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1          WHB9246            896        \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1          WHB9270            996        \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1          WHB9282            766        \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1          WHB9317           1104        \n",
       "16  Aspergillus_oryzae_NC_036435.1              WHB3883             24        \n",
       "17  Aspergillus_oryzae_NC_036435.1              WHB4048             32        \n",
       "18  Aspergillus_oryzae_NC_036435.1              WHB4056           7909        \n",
       "19  Aspergillus_oryzae_NC_036435.1              WHB4130             99        \n",
       "20  Aspergillus_oryzae_NC_036435.1              WHH1822             12        \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6459          14476        \n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6776            215        \n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6849             31        \n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1   WHB8186             79        \n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1   WHB8568             38        \n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB6155            152        \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9217            297        \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9583           5506        \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9649           1972        \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9882            835        \n",
       "⋮   ⋮                                           ⋮               ⋮             \n",
       "109 Prevotella_melaninogenica_NC_014370.1       WHB3939             141       \n",
       "110 Prevotella_melaninogenica_NC_014370.1       WHB5008              79       \n",
       "111 Prevotella_melaninogenica_NC_014370.1       WHB8397             150       \n",
       "112 Prevotella_melaninogenica_NC_014370.1       WHB9179            3549       \n",
       "113 Prevotella_oral_taxon_299_NC_022124.1       WHB4373              79       \n",
       "114 Prevotella_oral_taxon_299_NC_022124.1       WHB9179            2518       \n",
       "115 Pseudomonas_mendocina_NZ_CP013124.1         0415-0080       1157405       \n",
       "116 Pseudomonas_mendocina_NZ_CP013124.1         0715-0061       1366176       \n",
       "117 Pseudomonas_mendocina_NZ_CP013124.1         0816-0014       1345056       \n",
       "118 Pseudomonas_mendocina_NZ_CP013124.1         WHB3477         1334294       \n",
       "119 Pseudomonas_mendocina_NZ_CP013124.1         WHB3479         1431040       \n",
       "120 Rickettsia_Tillamook_23_NZ_CP060138.1       WHB6775              12       \n",
       "121 Rickettsia_Tillamook_23_NZ_CP060138.1       WHB9978            2923       \n",
       "122 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0021         12261       \n",
       "123 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0024         18768       \n",
       "124 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0026         14111       \n",
       "125 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0031         16398       \n",
       "126 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0032         18246       \n",
       "127 Staphylococcus_epidermidis_NZ_CP035288.1    WHB4219             992       \n",
       "128 Staphylococcus_epidermidis_NZ_CP035288.1    WHB4398             546       \n",
       "129 Staphylococcus_epidermidis_NZ_CP035288.1    WHB6459            9140       \n",
       "130 Staphylococcus_epidermidis_NZ_CP035288.1    WHB6651             830       \n",
       "131 Staphylococcus_epidermidis_NZ_CP035288.1    WHB8186             232       \n",
       "132 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0024        117929       \n",
       "133 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0026         94270       \n",
       "134 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0027         76946       \n",
       "135 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0031        108245       \n",
       "136 Staphylococcus_haemolyticus_NZ_CP013911.1   0715-0059         81100       \n",
       "137 Torque_teno_virus_6_NC_014094.1             WHB7619             496       \n",
       "138 Torque_teno_virus_6_NC_014094.1             WHB9685              39       \n",
       "    microbe_pairs_mapped human_pairs_mapped perc_covered1 perc_covered5\n",
       "1   429196               348366222          79.292320359  7.450282e+01 \n",
       "2   438419               364361186          78.389787758  7.366085e+01 \n",
       "3   494858               327626517          78.805094251  7.456206e+01 \n",
       "4   599335               344250628          79.082105806  7.525644e+01 \n",
       "5   769978               370885795          79.583059893  7.609348e+01 \n",
       "6    28842               364796907          79.329072139  1.697212e+01 \n",
       "7      210               365572623           1.316042864  1.963529e-03 \n",
       "8      233               284142227           1.460462955  4.782956e-04 \n",
       "9      453               408444066           2.824813672  3.373242e-03 \n",
       "10     231               371860124           1.642920130  0.000000e+00 \n",
       "11    6163               197279038          29.029117408  6.902854e-01 \n",
       "12    5202               182833313          27.000516799  3.080480e-01 \n",
       "13    6128               188061461          28.984145520  6.330785e-01 \n",
       "14    4490               176734183          23.360108613  2.885617e-01 \n",
       "15    7106               197010890          31.747814470  1.073538e+00 \n",
       "16       4               313105031           0.013741771  0.000000e+00 \n",
       "17       0               352035872           0.000000000  0.000000e+00 \n",
       "18      31               364502596           0.169732339  4.370987e-03 \n",
       "19       0               404269811           0.002315856  0.000000e+00 \n",
       "20       0               164835319           0.000000000  0.000000e+00 \n",
       "21   13069               183352926          66.396833288  7.454033e+00 \n",
       "22     200               152980243           2.189016786  0.000000e+00 \n",
       "23      25               165286847           0.249342726  0.000000e+00 \n",
       "24      83               183019538           0.799303964  6.741270e-04 \n",
       "25      35               175163865           0.346627680  0.000000e+00 \n",
       "26      19               133124278           0.444721342  0.000000e+00 \n",
       "27      31               150666053           0.701307112  0.000000e+00 \n",
       "28     792               169311308          15.704852586  4.270964e-02 \n",
       "29     290               199253589           6.521295389  0.000000e+00 \n",
       "30     115               194005959           2.457894931  0.000000e+00 \n",
       "⋮   ⋮                    ⋮                  ⋮             ⋮            \n",
       "109     74               337439179           1.0932372     0.000000000 \n",
       "110     34               367155498           0.4487845     0.000000000 \n",
       "111     65               169782090           0.9019109     0.000000000 \n",
       "112   1971               232737013          25.1270313     0.224948898 \n",
       "113     25               372164907           1.0017609     0.000000000 \n",
       "114    637               232737013          20.7162076     0.354018455 \n",
       "115 625658               324097265          71.7810391    67.443467432 \n",
       "116 478777               472980894          71.1279913    67.259332058 \n",
       "117 368174               422738587          71.5775292    67.434151292 \n",
       "118 748280               355475275          71.2235745    67.240309163 \n",
       "119 797696               334160371          71.3566957    67.599927815 \n",
       "120     21               148611421           0.3587976     0.000000000 \n",
       "121   6032               149284170          63.3608136     5.514696940 \n",
       "122      6               333337958           0.2456782     0.030264437 \n",
       "123      0               378247853           0.2461658     0.048340594 \n",
       "124      2               419260819           0.2183765     0.045902938 \n",
       "125      3               521164286           0.2671671     0.117420015 \n",
       "126      8               416012648           0.2688547     0.062779018 \n",
       "127    922               355718750           8.6830661     0.016703818 \n",
       "128    538               341262611           5.4451608     0.005797684 \n",
       "129   7791               183352926          52.9612382     1.443299053 \n",
       "130    770               171449646           7.6927162     0.008392452 \n",
       "131    215               183019538           2.0381090     0.005230079 \n",
       "132      2               378247853           0.2597145     0.047698657 \n",
       "133      1               419260819           0.2356298     0.038794385 \n",
       "134      8               416138008           0.2644608     0.066566301 \n",
       "135      5               521164286           0.2961161     0.119246644 \n",
       "136      7               509004187           0.2889770     0.151725664 \n",
       "137     60               154301162          21.5924426    19.946018893 \n",
       "138     22               171579152          16.0863698    14.170040486 \n",
       "    file_name                                         genus          \n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "16  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "17  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "18  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "19  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "20  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "⋮   ⋮                                                 ⋮              \n",
       "109 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "110 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "111 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "112 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "113 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "114 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "115 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "116 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "117 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "118 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "119 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "120 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "121 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "122 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "123 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "124 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "125 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "126 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "127 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "128 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "129 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "130 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "131 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "132 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "133 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "134 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "135 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "136 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "137 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "138 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "    species        suffix1 suffix2              ORI       TER       bPTR    \n",
       "1   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "2   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "3   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "4   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "5   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "6   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227 1.896385\n",
       "7   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "8   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "9   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "10  baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "11  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "12  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "13  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "14  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "15  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "16  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "17  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "18  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "19  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "20  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "21  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "22  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "23  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "24  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "25  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "26  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "27  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "28  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "29  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "30  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "⋮   ⋮              ⋮       ⋮                    ⋮         ⋮         ⋮       \n",
       "109 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "110 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "111 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "112 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "113 oral           taxon   299                  0         0               NA\n",
       "114 oral           taxon   299                  0         0               NA\n",
       "115 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "116 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "117 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "118 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "119 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "120 Tillamook      23      NZ                   240,832   888,370         NA\n",
       "121 Tillamook      23      NZ                   240,832   888,370   1.346660\n",
       "122 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "123 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "124 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "125 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "126 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "127 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "128 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "129 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375 1.566844\n",
       "130 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "131 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "132 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "133 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "134 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "135 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "136 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "137 teno           virus   6                    0         0               NA\n",
       "138 teno           virus   6                    0         0               NA\n",
       "    microbe_length microbe_count\n",
       "1   6402982        8.208410e+03 \n",
       "2   6402982        8.016719e+03 \n",
       "3   6402982        1.006331e+04 \n",
       "4   6402982        1.159937e+04 \n",
       "5   6402982        1.383176e+04 \n",
       "6   3972439        8.490606e+02 \n",
       "7   3972439        6.168933e+00 \n",
       "8   3972439        8.806120e+00 \n",
       "9   3972439        1.191050e+01 \n",
       "10  3972439        6.671090e+00 \n",
       "11  4233756        3.147800e+02 \n",
       "12  4233756        2.866889e+02 \n",
       "13  4233756        3.283332e+02 \n",
       "14  4233756        2.559892e+02 \n",
       "15  4233756        3.634384e+02 \n",
       "16  6520266        8.358461e-02 \n",
       "17  6520266        0.000000e+00 \n",
       "18  6520266        5.564388e-01 \n",
       "19  6520266        0.000000e+00 \n",
       "20  6520266        0.000000e+00 \n",
       "21  2373440        1.281144e+03 \n",
       "22  2373440        2.349839e+01 \n",
       "23  2373440        2.718599e+00 \n",
       "24  2373440        8.151247e+00 \n",
       "25  2373440        3.591426e+00 \n",
       "26  1219865        4.991215e+00 \n",
       "27  1219865        7.195421e+00 \n",
       "28  1219865        1.635871e+02 \n",
       "29  1219865        5.089812e+01 \n",
       "30  1219865        2.072968e+01 \n",
       "⋮   ⋮              ⋮            \n",
       "109 1796408        5.207783e+00 \n",
       "110 1796408        2.199103e+00 \n",
       "111 1796408        9.091555e+00 \n",
       "112 1796408        2.011119e+02 \n",
       "113  709850        4.037009e+00 \n",
       "114  709850        1.644861e+02 \n",
       "115 5120146        1.608426e+04 \n",
       "116 5120146        8.433911e+03 \n",
       "117 5120146        7.256389e+03 \n",
       "118 5120146        1.753857e+04 \n",
       "119 5120146        1.988941e+04 \n",
       "120 1438973        4.189249e+00 \n",
       "121 1438973        1.197889e+03 \n",
       "122 2666496        2.879700e-01 \n",
       "123 2666496        0.000000e+00 \n",
       "124 2666496        7.631791e-02 \n",
       "125 2666496        9.209316e-02 \n",
       "126 2666496        3.076552e-01 \n",
       "127 2466502        4.482955e+01 \n",
       "128 2466502        2.726678e+01 \n",
       "129 2466502        7.349293e+02 \n",
       "130 2466502        7.767734e+01 \n",
       "131 2466502        2.031801e+01 \n",
       "132 2549338        8.848055e-02 \n",
       "133 2549338        3.991260e-02 \n",
       "134 2549338        3.216969e-01 \n",
       "135 2549338        1.605424e-01 \n",
       "136 2549338        2.301288e-01 \n",
       "137    3705        4.477293e+03 \n",
       "138    3705        1.476358e+03 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "merged_df <- cov_df %>%\n",
    "    left_join(microbe_df) %>%\n",
    "    left_join(human_df %>% distinct(.keep_all = T)) %>%\n",
    "    left_join(irep_df) %>%\n",
    "    left_join(length_df) %>%\n",
    "    relocate(microbe_pairs_mapped, human_pairs_mapped, .after = 3) %>%\n",
    "    mutate(microbe_pairs_mapped = microbe_pairs_mapped / 2,\n",
    "           human_pairs_mapped = human_pairs_mapped / 2) %>%\n",
    "    mutate(microbe_count = (microbe_pairs_mapped / microbe_length) / (human_pairs_mapped / human_length) * leuco_per_ul) \n",
    "\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "fd13d121",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Joining, by = c(\"prefix\", \"ORI\", \"TER\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 138 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>prefix</th><th scope=col>npm_research_id</th><th scope=col>pairs_assigned</th><th scope=col>microbe_pairs_mapped</th><th scope=col>human_pairs_mapped</th><th scope=col>perc_covered1</th><th scope=col>perc_covered5</th><th scope=col>file_name</th><th scope=col>genus</th><th scope=col>species</th><th scope=col>suffix1</th><th scope=col>suffix2</th><th scope=col>ORI</th><th scope=col>TER</th><th scope=col>bPTR</th><th scope=col>microbe_length</th><th scope=col>microbe_count</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>1116-0023</td><td>430178</td><td>429196</td><td>348366222</td><td>79.292320359</td><td>7.450282e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>8.208410e+03</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3658  </td><td>385855</td><td>438419</td><td>364361186</td><td>78.389787758</td><td>7.366085e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>8.016719e+03</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3674  </td><td>459561</td><td>494858</td><td>327626517</td><td>78.805094251</td><td>7.456206e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>1.006331e+04</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3686  </td><td>541795</td><td>599335</td><td>344250628</td><td>79.082105806</td><td>7.525644e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>1.159937e+04</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3734  </td><td>688629</td><td>769978</td><td>370885795</td><td>79.583059893</td><td>7.609348e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>6402982</td><td>1.383176e+04</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>0116-0053</td><td>  7673</td><td> 28842</td><td>364796907</td><td>79.329072139</td><td>1.697212e+01</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>1.896385</td><td>3972439</td><td>8.490606e+02</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4240  </td><td>   168</td><td>   210</td><td>365572623</td><td> 1.316042864</td><td>1.963529e-03</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>6.168933e+00</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4251  </td><td>   183</td><td>   233</td><td>284142227</td><td> 1.460462955</td><td>4.782956e-04</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>8.806120e+00</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4438  </td><td>   379</td><td>   453</td><td>408444066</td><td> 2.824813672</td><td>3.373242e-03</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>1.191050e+01</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4828  </td><td>   190</td><td>   231</td><td>371860124</td><td> 1.642920130</td><td>0.000000e+00</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td><td>3972439</td><td>6.671090e+00</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9245  </td><td>   982</td><td>  6163</td><td>197279038</td><td>29.029117408</td><td>6.902854e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>3.147800e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9246  </td><td>   896</td><td>  5202</td><td>182833313</td><td>27.000516799</td><td>3.080480e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>2.866889e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9270  </td><td>   996</td><td>  6128</td><td>188061461</td><td>28.984145520</td><td>6.330785e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>3.283332e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9282  </td><td>   766</td><td>  4490</td><td>176734183</td><td>23.360108613</td><td>2.885617e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>2.559892e+02</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9317  </td><td>  1104</td><td>  7106</td><td>197010890</td><td>31.747814470</td><td>1.073538e+00</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>4233756</td><td>3.634384e+02</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB3883  </td><td>    24</td><td>     4</td><td>313105031</td><td> 0.013741771</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>8.358461e-02</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4048  </td><td>    32</td><td>     0</td><td>352035872</td><td> 0.000000000</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4056  </td><td>  7909</td><td>    31</td><td>364502596</td><td> 0.169732339</td><td>4.370987e-03</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>5.564388e-01</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4130  </td><td>    99</td><td>     0</td><td>404269811</td><td> 0.002315856</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHH1822  </td><td>    12</td><td>     0</td><td>164835319</td><td> 0.000000000</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td><td>6520266</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6459  </td><td> 14476</td><td> 13069</td><td>183352926</td><td>66.396833288</td><td>7.454033e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>1.281144e+03</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6776  </td><td>   215</td><td>   200</td><td>152980243</td><td> 2.189016786</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>2.349839e+01</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6849  </td><td>    31</td><td>    25</td><td>165286847</td><td> 0.249342726</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>2.718599e+00</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB8186  </td><td>    79</td><td>    83</td><td>183019538</td><td> 0.799303964</td><td>6.741270e-04</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>8.151247e+00</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB8568  </td><td>    38</td><td>    35</td><td>175163865</td><td> 0.346627680</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td><td>2373440</td><td>3.591426e+00</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB6155  </td><td>   152</td><td>    19</td><td>133124278</td><td> 0.444721342</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>4.991215e+00</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9217  </td><td>   297</td><td>    31</td><td>150666053</td><td> 0.701307112</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>7.195421e+00</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9583  </td><td>  5506</td><td>   792</td><td>169311308</td><td>15.704852586</td><td>4.270964e-02</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>1.635871e+02</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9649  </td><td>  1972</td><td>   290</td><td>199253589</td><td> 6.521295389</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>5.089812e+01</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9882  </td><td>   835</td><td>   115</td><td>194005959</td><td> 2.457894931</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td><td>1219865</td><td>2.072968e+01</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB3939  </td><td>    141</td><td>    74</td><td>337439179</td><td> 1.0932372</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>5.207783e+00</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB5008  </td><td>     79</td><td>    34</td><td>367155498</td><td> 0.4487845</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>2.199103e+00</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB8397  </td><td>    150</td><td>    65</td><td>169782090</td><td> 0.9019109</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>9.091555e+00</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB9179  </td><td>   3549</td><td>  1971</td><td>232737013</td><td>25.1270313</td><td> 0.224948898</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td><td>1796408</td><td>2.011119e+02</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1    </td><td>WHB4373  </td><td>     79</td><td>    25</td><td>372164907</td><td> 1.0017609</td><td> 0.000000000</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>      NA</td><td> 709850</td><td>4.037009e+00</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1    </td><td>WHB9179  </td><td>   2518</td><td>   637</td><td>232737013</td><td>20.7162076</td><td> 0.354018455</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>      NA</td><td> 709850</td><td>1.644861e+02</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0415-0080</td><td>1157405</td><td>625658</td><td>324097265</td><td>71.7810391</td><td>67.443467432</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>1.608426e+04</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0715-0061</td><td>1366176</td><td>478777</td><td>472980894</td><td>71.1279913</td><td>67.259332058</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>8.433911e+03</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0816-0014</td><td>1345056</td><td>368174</td><td>422738587</td><td>71.5775292</td><td>67.434151292</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>7.256389e+03</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>WHB3477  </td><td>1334294</td><td>748280</td><td>355475275</td><td>71.2235745</td><td>67.240309163</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>1.753857e+04</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>WHB3479  </td><td>1431040</td><td>797696</td><td>334160371</td><td>71.3566957</td><td>67.599927815</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>5120146</td><td>1.988941e+04</td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td><td>WHB6775  </td><td>     12</td><td>    21</td><td>148611421</td><td> 0.3587976</td><td> 0.000000000</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>      NA</td><td>1438973</td><td>4.189249e+00</td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td><td>WHB9978  </td><td>   2923</td><td>  6032</td><td>149284170</td><td>63.3608136</td><td> 5.514696940</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>1.346660</td><td>1438973</td><td>1.197889e+03</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0021</td><td>  12261</td><td>     6</td><td>333337958</td><td> 0.2456782</td><td> 0.030264437</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>2.879700e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0024</td><td>  18768</td><td>     0</td><td>378247853</td><td> 0.2461658</td><td> 0.048340594</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>0.000000e+00</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0026</td><td>  14111</td><td>     2</td><td>419260819</td><td> 0.2183765</td><td> 0.045902938</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>7.631791e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0031</td><td>  16398</td><td>     3</td><td>521164286</td><td> 0.2671671</td><td> 0.117420015</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>9.209316e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0032</td><td>  18246</td><td>     8</td><td>416012648</td><td> 0.2688547</td><td> 0.062779018</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td><td>2666496</td><td>3.076552e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB4219  </td><td>    992</td><td>   922</td><td>355718750</td><td> 8.6830661</td><td> 0.016703818</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>4.482955e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB4398  </td><td>    546</td><td>   538</td><td>341262611</td><td> 5.4451608</td><td> 0.005797684</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>2.726678e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB6459  </td><td>   9140</td><td>  7791</td><td>183352926</td><td>52.9612382</td><td> 1.443299053</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>1.566844</td><td>2466502</td><td>7.349293e+02</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB6651  </td><td>    830</td><td>   770</td><td>171449646</td><td> 7.6927162</td><td> 0.008392452</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>7.767734e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB8186  </td><td>    232</td><td>   215</td><td>183019538</td><td> 2.0381090</td><td> 0.005230079</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td><td>2466502</td><td>2.031801e+01</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0024</td><td> 117929</td><td>     2</td><td>378247853</td><td> 0.2597145</td><td> 0.047698657</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>8.848055e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0026</td><td>  94270</td><td>     1</td><td>419260819</td><td> 0.2356298</td><td> 0.038794385</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>3.991260e-02</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0027</td><td>  76946</td><td>     8</td><td>416138008</td><td> 0.2644608</td><td> 0.066566301</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>3.216969e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0031</td><td> 108245</td><td>     5</td><td>521164286</td><td> 0.2961161</td><td> 0.119246644</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>1.605424e-01</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0715-0059</td><td>  81100</td><td>     7</td><td>509004187</td><td> 0.2889770</td><td> 0.151725664</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td><td>2549338</td><td>2.301288e-01</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1          </td><td>WHB7619  </td><td>    496</td><td>    60</td><td>154301162</td><td>21.5924426</td><td>19.946018893</td><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>      NA</td><td>   3705</td><td>4.477293e+03</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1          </td><td>WHB9685  </td><td>     39</td><td>    22</td><td>171579152</td><td>16.0863698</td><td>14.170040486</td><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>      NA</td><td>   3705</td><td>1.476358e+03</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 138 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " prefix & npm\\_research\\_id & pairs\\_assigned & microbe\\_pairs\\_mapped & human\\_pairs\\_mapped & perc\\_covered1 & perc\\_covered5 & file\\_name & genus & species & suffix1 & suffix2 & ORI & TER & bPTR & microbe\\_length & microbe\\_count\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & 1116-0023 & 430178 & 429196 & 348366222 & 79.292320359 & 7.450282e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 8.208410e+03\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3658   & 385855 & 438419 & 364361186 & 78.389787758 & 7.366085e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 8.016719e+03\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3674   & 459561 & 494858 & 327626517 & 78.805094251 & 7.456206e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 1.006331e+04\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3686   & 541795 & 599335 & 344250628 & 79.082105806 & 7.525644e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 1.159937e+04\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3734   & 688629 & 769978 & 370885795 & 79.583059893 & 7.609348e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA & 6402982 & 1.383176e+04\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & 0116-0053 &   7673 &  28842 & 364796907 & 79.329072139 & 1.697212e+01 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & 1.896385 & 3972439 & 8.490606e+02\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4240   &    168 &    210 & 365572623 &  1.316042864 & 1.963529e-03 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 6.168933e+00\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4251   &    183 &    233 & 284142227 &  1.460462955 & 4.782956e-04 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 8.806120e+00\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4438   &    379 &    453 & 408444066 &  2.824813672 & 3.373242e-03 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 1.191050e+01\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4828   &    190 &    231 & 371860124 &  1.642920130 & 0.000000e+00 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA & 3972439 & 6.671090e+00\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9245   &    982 &   6163 & 197279038 & 29.029117408 & 6.902854e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 3.147800e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9246   &    896 &   5202 & 182833313 & 27.000516799 & 3.080480e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 2.866889e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9270   &    996 &   6128 & 188061461 & 28.984145520 & 6.330785e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 3.283332e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9282   &    766 &   4490 & 176734183 & 23.360108613 & 2.885617e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 2.559892e+02\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9317   &   1104 &   7106 & 197010890 & 31.747814470 & 1.073538e+00 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA & 4233756 & 3.634384e+02\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB3883   &     24 &      4 & 313105031 &  0.013741771 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 8.358461e-02\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4048   &     32 &      0 & 352035872 &  0.000000000 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 0.000000e+00\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4056   &   7909 &     31 & 364502596 &  0.169732339 & 4.370987e-03 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 5.564388e-01\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4130   &     99 &      0 & 404269811 &  0.002315856 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 0.000000e+00\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHH1822   &     12 &      0 & 164835319 &  0.000000000 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA & 6520266 & 0.000000e+00\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6459   &  14476 &  13069 & 183352926 & 66.396833288 & 7.454033e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 1.281144e+03\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6776   &    215 &    200 & 152980243 &  2.189016786 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 2.349839e+01\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6849   &     31 &     25 & 165286847 &  0.249342726 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 2.718599e+00\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB8186   &     79 &     83 & 183019538 &  0.799303964 & 6.741270e-04 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 8.151247e+00\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB8568   &     38 &     35 & 175163865 &  0.346627680 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA & 2373440 & 3.591426e+00\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB6155   &    152 &     19 & 133124278 &  0.444721342 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 4.991215e+00\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9217   &    297 &     31 & 150666053 &  0.701307112 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 7.195421e+00\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9583   &   5506 &    792 & 169311308 & 15.704852586 & 4.270964e-02 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 1.635871e+02\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9649   &   1972 &    290 & 199253589 &  6.521295389 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 5.089812e+01\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9882   &    835 &    115 & 194005959 &  2.457894931 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA & 1219865 & 2.072968e+01\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB3939   &     141 &     74 & 337439179 &  1.0932372 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 5.207783e+00\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB5008   &      79 &     34 & 367155498 &  0.4487845 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 2.199103e+00\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB8397   &     150 &     65 & 169782090 &  0.9019109 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 9.091555e+00\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB9179   &    3549 &   1971 & 232737013 & 25.1270313 &  0.224948898 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA & 1796408 & 2.011119e+02\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1     & WHB4373   &      79 &     25 & 372164907 &  1.0017609 &  0.000000000 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         &       NA &  709850 & 4.037009e+00\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1     & WHB9179   &    2518 &    637 & 232737013 & 20.7162076 &  0.354018455 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         &       NA &  709850 & 1.644861e+02\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0415-0080 & 1157405 & 625658 & 324097265 & 71.7810391 & 67.443467432 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 1.608426e+04\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0715-0061 & 1366176 & 478777 & 472980894 & 71.1279913 & 67.259332058 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 8.433911e+03\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0816-0014 & 1345056 & 368174 & 422738587 & 71.5775292 & 67.434151292 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 7.256389e+03\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & WHB3477   & 1334294 & 748280 & 355475275 & 71.2235745 & 67.240309163 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 1.753857e+04\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & WHB3479   & 1431040 & 797696 & 334160371 & 71.3566957 & 67.599927815 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA & 5120146 & 1.988941e+04\\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1     & WHB6775   &      12 &     21 & 148611421 &  0.3587976 &  0.000000000 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   &       NA & 1438973 & 4.189249e+00\\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1     & WHB9978   &    2923 &   6032 & 149284170 & 63.3608136 &  5.514696940 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & 1.346660 & 1438973 & 1.197889e+03\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0021 &   12261 &      6 & 333337958 &  0.2456782 &  0.030264437 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 2.879700e-01\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0024 &   18768 &      0 & 378247853 &  0.2461658 &  0.048340594 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 0.000000e+00\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0026 &   14111 &      2 & 419260819 &  0.2183765 &  0.045902938 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 7.631791e-02\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0031 &   16398 &      3 & 521164286 &  0.2671671 &  0.117420015 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 9.209316e-02\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0032 &   18246 &      8 & 416012648 &  0.2688547 &  0.062779018 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA & 2666496 & 3.076552e-01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB4219   &     992 &    922 & 355718750 &  8.6830661 &  0.016703818 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 4.482955e+01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB4398   &     546 &    538 & 341262611 &  5.4451608 &  0.005797684 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 2.726678e+01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB6459   &    9140 &   7791 & 183352926 & 52.9612382 &  1.443299053 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & 1.566844 & 2466502 & 7.349293e+02\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB6651   &     830 &    770 & 171449646 &  7.6927162 &  0.008392452 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 7.767734e+01\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB8186   &     232 &    215 & 183019538 &  2.0381090 &  0.005230079 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA & 2466502 & 2.031801e+01\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0024 &  117929 &      2 & 378247853 &  0.2597145 &  0.047698657 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 8.848055e-02\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0026 &   94270 &      1 & 419260819 &  0.2356298 &  0.038794385 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 3.991260e-02\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0027 &   76946 &      8 & 416138008 &  0.2644608 &  0.066566301 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 3.216969e-01\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0031 &  108245 &      5 & 521164286 &  0.2961161 &  0.119246644 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 1.605424e-01\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0715-0059 &   81100 &      7 & 509004187 &  0.2889770 &  0.151725664 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA & 2549338 & 2.301288e-01\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1           & WHB7619   &     496 &     60 & 154301162 & 21.5924426 & 19.946018893 & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         &       NA &    3705 & 4.477293e+03\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1           & WHB9685   &      39 &     22 & 171579152 & 16.0863698 & 14.170040486 & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         &       NA &    3705 & 1.476358e+03\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 138 × 17\n",
       "\n",
       "| prefix &lt;chr&gt; | npm_research_id &lt;chr&gt; | pairs_assigned &lt;int&gt; | microbe_pairs_mapped &lt;dbl&gt; | human_pairs_mapped &lt;dbl&gt; | perc_covered1 &lt;dbl&gt; | perc_covered5 &lt;dbl&gt; | file_name &lt;chr&gt; | genus &lt;chr&gt; | species &lt;chr&gt; | suffix1 &lt;chr&gt; | suffix2 &lt;chr&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; | bPTR &lt;dbl&gt; | microbe_length &lt;int&gt; | microbe_count &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | 1116-0023 | 430178 | 429196 | 348366222 | 79.292320359 | 7.450282e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 8.208410e+03 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3658   | 385855 | 438419 | 364361186 | 78.389787758 | 7.366085e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 8.016719e+03 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3674   | 459561 | 494858 | 327626517 | 78.805094251 | 7.456206e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 1.006331e+04 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3686   | 541795 | 599335 | 344250628 | 79.082105806 | 7.525644e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 1.159937e+04 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3734   | 688629 | 769978 | 370885795 | 79.583059893 | 7.609348e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA | 6402982 | 1.383176e+04 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | 0116-0053 |   7673 |  28842 | 364796907 | 79.329072139 | 1.697212e+01 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | 1.896385 | 3972439 | 8.490606e+02 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4240   |    168 |    210 | 365572623 |  1.316042864 | 1.963529e-03 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 6.168933e+00 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4251   |    183 |    233 | 284142227 |  1.460462955 | 4.782956e-04 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 8.806120e+00 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4438   |    379 |    453 | 408444066 |  2.824813672 | 3.373242e-03 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 1.191050e+01 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4828   |    190 |    231 | 371860124 |  1.642920130 | 0.000000e+00 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA | 3972439 | 6.671090e+00 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9245   |    982 |   6163 | 197279038 | 29.029117408 | 6.902854e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 3.147800e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9246   |    896 |   5202 | 182833313 | 27.000516799 | 3.080480e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 2.866889e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9270   |    996 |   6128 | 188061461 | 28.984145520 | 6.330785e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 3.283332e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9282   |    766 |   4490 | 176734183 | 23.360108613 | 2.885617e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 2.559892e+02 |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9317   |   1104 |   7106 | 197010890 | 31.747814470 | 1.073538e+00 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA | 4233756 | 3.634384e+02 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB3883   |     24 |      4 | 313105031 |  0.013741771 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 8.358461e-02 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4048   |     32 |      0 | 352035872 |  0.000000000 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 0.000000e+00 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4056   |   7909 |     31 | 364502596 |  0.169732339 | 4.370987e-03 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 5.564388e-01 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4130   |     99 |      0 | 404269811 |  0.002315856 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 0.000000e+00 |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHH1822   |     12 |      0 | 164835319 |  0.000000000 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA | 6520266 | 0.000000e+00 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6459   |  14476 |  13069 | 183352926 | 66.396833288 | 7.454033e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 1.281144e+03 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6776   |    215 |    200 | 152980243 |  2.189016786 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 2.349839e+01 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6849   |     31 |     25 | 165286847 |  0.249342726 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 2.718599e+00 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB8186   |     79 |     83 | 183019538 |  0.799303964 | 6.741270e-04 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 8.151247e+00 |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB8568   |     38 |     35 | 175163865 |  0.346627680 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA | 2373440 | 3.591426e+00 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB6155   |    152 |     19 | 133124278 |  0.444721342 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 4.991215e+00 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9217   |    297 |     31 | 150666053 |  0.701307112 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 7.195421e+00 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9583   |   5506 |    792 | 169311308 | 15.704852586 | 4.270964e-02 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 1.635871e+02 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9649   |   1972 |    290 | 199253589 |  6.521295389 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 5.089812e+01 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9882   |    835 |    115 | 194005959 |  2.457894931 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA | 1219865 | 2.072968e+01 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB3939   |     141 |     74 | 337439179 |  1.0932372 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 5.207783e+00 |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB5008   |      79 |     34 | 367155498 |  0.4487845 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 2.199103e+00 |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB8397   |     150 |     65 | 169782090 |  0.9019109 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 9.091555e+00 |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB9179   |    3549 |   1971 | 232737013 | 25.1270313 |  0.224948898 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA | 1796408 | 2.011119e+02 |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1     | WHB4373   |      79 |     25 | 372164907 |  1.0017609 |  0.000000000 | Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         |       NA |  709850 | 4.037009e+00 |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1     | WHB9179   |    2518 |    637 | 232737013 | 20.7162076 |  0.354018455 | Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         |       NA |  709850 | 1.644861e+02 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0415-0080 | 1157405 | 625658 | 324097265 | 71.7810391 | 67.443467432 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 1.608426e+04 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0715-0061 | 1366176 | 478777 | 472980894 | 71.1279913 | 67.259332058 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 8.433911e+03 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0816-0014 | 1345056 | 368174 | 422738587 | 71.5775292 | 67.434151292 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 7.256389e+03 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | WHB3477   | 1334294 | 748280 | 355475275 | 71.2235745 | 67.240309163 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 1.753857e+04 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | WHB3479   | 1431040 | 797696 | 334160371 | 71.3566957 | 67.599927815 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA | 5120146 | 1.988941e+04 |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1     | WHB6775   |      12 |     21 | 148611421 |  0.3587976 |  0.000000000 | Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   |       NA | 1438973 | 4.189249e+00 |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1     | WHB9978   |    2923 |   6032 | 149284170 | 63.3608136 |  5.514696940 | Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | 1.346660 | 1438973 | 1.197889e+03 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0021 |   12261 |      6 | 333337958 |  0.2456782 |  0.030264437 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 2.879700e-01 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0024 |   18768 |      0 | 378247853 |  0.2461658 |  0.048340594 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 0.000000e+00 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0026 |   14111 |      2 | 419260819 |  0.2183765 |  0.045902938 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 7.631791e-02 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0031 |   16398 |      3 | 521164286 |  0.2671671 |  0.117420015 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 9.209316e-02 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0032 |   18246 |      8 | 416012648 |  0.2688547 |  0.062779018 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA | 2666496 | 3.076552e-01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB4219   |     992 |    922 | 355718750 |  8.6830661 |  0.016703818 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 4.482955e+01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB4398   |     546 |    538 | 341262611 |  5.4451608 |  0.005797684 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 2.726678e+01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB6459   |    9140 |   7791 | 183352926 | 52.9612382 |  1.443299053 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | 1.566844 | 2466502 | 7.349293e+02 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB6651   |     830 |    770 | 171449646 |  7.6927162 |  0.008392452 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 7.767734e+01 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB8186   |     232 |    215 | 183019538 |  2.0381090 |  0.005230079 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA | 2466502 | 2.031801e+01 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0024 |  117929 |      2 | 378247853 |  0.2597145 |  0.047698657 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 8.848055e-02 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0026 |   94270 |      1 | 419260819 |  0.2356298 |  0.038794385 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 3.991260e-02 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0027 |   76946 |      8 | 416138008 |  0.2644608 |  0.066566301 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 3.216969e-01 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0031 |  108245 |      5 | 521164286 |  0.2961161 |  0.119246644 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 1.605424e-01 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0715-0059 |   81100 |      7 | 509004187 |  0.2889770 |  0.151725664 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA | 2549338 | 2.301288e-01 |\n",
       "| Torque_teno_virus_6_NC_014094.1           | WHB7619   |     496 |     60 | 154301162 | 21.5924426 | 19.946018893 | Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         |       NA |    3705 | 4.477293e+03 |\n",
       "| Torque_teno_virus_6_NC_014094.1           | WHB9685   |      39 |     22 | 171579152 | 16.0863698 | 14.170040486 | Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         |       NA |    3705 | 1.476358e+03 |\n",
       "\n"
      ],
      "text/plain": [
       "    prefix                                      npm_research_id pairs_assigned\n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1    1116-0023       430178        \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3658         385855        \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3674         459561        \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3686         541795        \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3734         688629        \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1       0116-0053         7673        \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1       WHB4240            168        \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1       WHB4251            183        \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1       WHB4438            379        \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1       WHB4828            190        \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1          WHB9245            982        \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1          WHB9246            896        \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1          WHB9270            996        \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1          WHB9282            766        \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1          WHB9317           1104        \n",
       "16  Aspergillus_oryzae_NC_036435.1              WHB3883             24        \n",
       "17  Aspergillus_oryzae_NC_036435.1              WHB4048             32        \n",
       "18  Aspergillus_oryzae_NC_036435.1              WHB4056           7909        \n",
       "19  Aspergillus_oryzae_NC_036435.1              WHB4130             99        \n",
       "20  Aspergillus_oryzae_NC_036435.1              WHH1822             12        \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6459          14476        \n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6776            215        \n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6849             31        \n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1   WHB8186             79        \n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1   WHB8568             38        \n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB6155            152        \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9217            297        \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9583           5506        \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9649           1972        \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9882            835        \n",
       "⋮   ⋮                                           ⋮               ⋮             \n",
       "109 Prevotella_melaninogenica_NC_014370.1       WHB3939             141       \n",
       "110 Prevotella_melaninogenica_NC_014370.1       WHB5008              79       \n",
       "111 Prevotella_melaninogenica_NC_014370.1       WHB8397             150       \n",
       "112 Prevotella_melaninogenica_NC_014370.1       WHB9179            3549       \n",
       "113 Prevotella_oral_taxon_299_NC_022124.1       WHB4373              79       \n",
       "114 Prevotella_oral_taxon_299_NC_022124.1       WHB9179            2518       \n",
       "115 Pseudomonas_mendocina_NZ_CP013124.1         0415-0080       1157405       \n",
       "116 Pseudomonas_mendocina_NZ_CP013124.1         0715-0061       1366176       \n",
       "117 Pseudomonas_mendocina_NZ_CP013124.1         0816-0014       1345056       \n",
       "118 Pseudomonas_mendocina_NZ_CP013124.1         WHB3477         1334294       \n",
       "119 Pseudomonas_mendocina_NZ_CP013124.1         WHB3479         1431040       \n",
       "120 Rickettsia_Tillamook_23_NZ_CP060138.1       WHB6775              12       \n",
       "121 Rickettsia_Tillamook_23_NZ_CP060138.1       WHB9978            2923       \n",
       "122 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0021         12261       \n",
       "123 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0024         18768       \n",
       "124 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0026         14111       \n",
       "125 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0031         16398       \n",
       "126 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0032         18246       \n",
       "127 Staphylococcus_epidermidis_NZ_CP035288.1    WHB4219             992       \n",
       "128 Staphylococcus_epidermidis_NZ_CP035288.1    WHB4398             546       \n",
       "129 Staphylococcus_epidermidis_NZ_CP035288.1    WHB6459            9140       \n",
       "130 Staphylococcus_epidermidis_NZ_CP035288.1    WHB6651             830       \n",
       "131 Staphylococcus_epidermidis_NZ_CP035288.1    WHB8186             232       \n",
       "132 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0024        117929       \n",
       "133 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0026         94270       \n",
       "134 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0027         76946       \n",
       "135 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0031        108245       \n",
       "136 Staphylococcus_haemolyticus_NZ_CP013911.1   0715-0059         81100       \n",
       "137 Torque_teno_virus_6_NC_014094.1             WHB7619             496       \n",
       "138 Torque_teno_virus_6_NC_014094.1             WHB9685              39       \n",
       "    microbe_pairs_mapped human_pairs_mapped perc_covered1 perc_covered5\n",
       "1   429196               348366222          79.292320359  7.450282e+01 \n",
       "2   438419               364361186          78.389787758  7.366085e+01 \n",
       "3   494858               327626517          78.805094251  7.456206e+01 \n",
       "4   599335               344250628          79.082105806  7.525644e+01 \n",
       "5   769978               370885795          79.583059893  7.609348e+01 \n",
       "6    28842               364796907          79.329072139  1.697212e+01 \n",
       "7      210               365572623           1.316042864  1.963529e-03 \n",
       "8      233               284142227           1.460462955  4.782956e-04 \n",
       "9      453               408444066           2.824813672  3.373242e-03 \n",
       "10     231               371860124           1.642920130  0.000000e+00 \n",
       "11    6163               197279038          29.029117408  6.902854e-01 \n",
       "12    5202               182833313          27.000516799  3.080480e-01 \n",
       "13    6128               188061461          28.984145520  6.330785e-01 \n",
       "14    4490               176734183          23.360108613  2.885617e-01 \n",
       "15    7106               197010890          31.747814470  1.073538e+00 \n",
       "16       4               313105031           0.013741771  0.000000e+00 \n",
       "17       0               352035872           0.000000000  0.000000e+00 \n",
       "18      31               364502596           0.169732339  4.370987e-03 \n",
       "19       0               404269811           0.002315856  0.000000e+00 \n",
       "20       0               164835319           0.000000000  0.000000e+00 \n",
       "21   13069               183352926          66.396833288  7.454033e+00 \n",
       "22     200               152980243           2.189016786  0.000000e+00 \n",
       "23      25               165286847           0.249342726  0.000000e+00 \n",
       "24      83               183019538           0.799303964  6.741270e-04 \n",
       "25      35               175163865           0.346627680  0.000000e+00 \n",
       "26      19               133124278           0.444721342  0.000000e+00 \n",
       "27      31               150666053           0.701307112  0.000000e+00 \n",
       "28     792               169311308          15.704852586  4.270964e-02 \n",
       "29     290               199253589           6.521295389  0.000000e+00 \n",
       "30     115               194005959           2.457894931  0.000000e+00 \n",
       "⋮   ⋮                    ⋮                  ⋮             ⋮            \n",
       "109     74               337439179           1.0932372     0.000000000 \n",
       "110     34               367155498           0.4487845     0.000000000 \n",
       "111     65               169782090           0.9019109     0.000000000 \n",
       "112   1971               232737013          25.1270313     0.224948898 \n",
       "113     25               372164907           1.0017609     0.000000000 \n",
       "114    637               232737013          20.7162076     0.354018455 \n",
       "115 625658               324097265          71.7810391    67.443467432 \n",
       "116 478777               472980894          71.1279913    67.259332058 \n",
       "117 368174               422738587          71.5775292    67.434151292 \n",
       "118 748280               355475275          71.2235745    67.240309163 \n",
       "119 797696               334160371          71.3566957    67.599927815 \n",
       "120     21               148611421           0.3587976     0.000000000 \n",
       "121   6032               149284170          63.3608136     5.514696940 \n",
       "122      6               333337958           0.2456782     0.030264437 \n",
       "123      0               378247853           0.2461658     0.048340594 \n",
       "124      2               419260819           0.2183765     0.045902938 \n",
       "125      3               521164286           0.2671671     0.117420015 \n",
       "126      8               416012648           0.2688547     0.062779018 \n",
       "127    922               355718750           8.6830661     0.016703818 \n",
       "128    538               341262611           5.4451608     0.005797684 \n",
       "129   7791               183352926          52.9612382     1.443299053 \n",
       "130    770               171449646           7.6927162     0.008392452 \n",
       "131    215               183019538           2.0381090     0.005230079 \n",
       "132      2               378247853           0.2597145     0.047698657 \n",
       "133      1               419260819           0.2356298     0.038794385 \n",
       "134      8               416138008           0.2644608     0.066566301 \n",
       "135      5               521164286           0.2961161     0.119246644 \n",
       "136      7               509004187           0.2889770     0.151725664 \n",
       "137     60               154301162          21.5924426    19.946018893 \n",
       "138     22               171579152          16.0863698    14.170040486 \n",
       "    file_name                                         genus          \n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "16  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "17  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "18  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "19  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "20  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "⋮   ⋮                                                 ⋮              \n",
       "109 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "110 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "111 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "112 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "113 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "114 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "115 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "116 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "117 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "118 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "119 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "120 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "121 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "122 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "123 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "124 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "125 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "126 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "127 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "128 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "129 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "130 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "131 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "132 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "133 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "134 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "135 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "136 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "137 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "138 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "    species        suffix1 suffix2              ORI       TER       bPTR    \n",
       "1   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "2   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "3   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "4   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "5   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "6   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227 1.896385\n",
       "7   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "8   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "9   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "10  baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "11  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "12  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "13  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "14  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "15  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "16  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "17  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "18  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "19  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "20  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "21  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "22  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "23  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "24  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "25  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "26  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "27  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "28  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "29  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "30  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "⋮   ⋮              ⋮       ⋮                    ⋮         ⋮         ⋮       \n",
       "109 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "110 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "111 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "112 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "113 oral           taxon   299                  0         0               NA\n",
       "114 oral           taxon   299                  0         0               NA\n",
       "115 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "116 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "117 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "118 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "119 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "120 Tillamook      23      NZ                   240,832   888,370         NA\n",
       "121 Tillamook      23      NZ                   240,832   888,370   1.346660\n",
       "122 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "123 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "124 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "125 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "126 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "127 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "128 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "129 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375 1.566844\n",
       "130 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "131 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "132 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "133 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "134 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "135 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "136 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "137 teno           virus   6                    0         0               NA\n",
       "138 teno           virus   6                    0         0               NA\n",
       "    microbe_length microbe_count\n",
       "1   6402982        8.208410e+03 \n",
       "2   6402982        8.016719e+03 \n",
       "3   6402982        1.006331e+04 \n",
       "4   6402982        1.159937e+04 \n",
       "5   6402982        1.383176e+04 \n",
       "6   3972439        8.490606e+02 \n",
       "7   3972439        6.168933e+00 \n",
       "8   3972439        8.806120e+00 \n",
       "9   3972439        1.191050e+01 \n",
       "10  3972439        6.671090e+00 \n",
       "11  4233756        3.147800e+02 \n",
       "12  4233756        2.866889e+02 \n",
       "13  4233756        3.283332e+02 \n",
       "14  4233756        2.559892e+02 \n",
       "15  4233756        3.634384e+02 \n",
       "16  6520266        8.358461e-02 \n",
       "17  6520266        0.000000e+00 \n",
       "18  6520266        5.564388e-01 \n",
       "19  6520266        0.000000e+00 \n",
       "20  6520266        0.000000e+00 \n",
       "21  2373440        1.281144e+03 \n",
       "22  2373440        2.349839e+01 \n",
       "23  2373440        2.718599e+00 \n",
       "24  2373440        8.151247e+00 \n",
       "25  2373440        3.591426e+00 \n",
       "26  1219865        4.991215e+00 \n",
       "27  1219865        7.195421e+00 \n",
       "28  1219865        1.635871e+02 \n",
       "29  1219865        5.089812e+01 \n",
       "30  1219865        2.072968e+01 \n",
       "⋮   ⋮              ⋮            \n",
       "109 1796408        5.207783e+00 \n",
       "110 1796408        2.199103e+00 \n",
       "111 1796408        9.091555e+00 \n",
       "112 1796408        2.011119e+02 \n",
       "113  709850        4.037009e+00 \n",
       "114  709850        1.644861e+02 \n",
       "115 5120146        1.608426e+04 \n",
       "116 5120146        8.433911e+03 \n",
       "117 5120146        7.256389e+03 \n",
       "118 5120146        1.753857e+04 \n",
       "119 5120146        1.988941e+04 \n",
       "120 1438973        4.189249e+00 \n",
       "121 1438973        1.197889e+03 \n",
       "122 2666496        2.879700e-01 \n",
       "123 2666496        0.000000e+00 \n",
       "124 2666496        7.631791e-02 \n",
       "125 2666496        9.209316e-02 \n",
       "126 2666496        3.076552e-01 \n",
       "127 2466502        4.482955e+01 \n",
       "128 2466502        2.726678e+01 \n",
       "129 2466502        7.349293e+02 \n",
       "130 2466502        7.767734e+01 \n",
       "131 2466502        2.031801e+01 \n",
       "132 2549338        8.848055e-02 \n",
       "133 2549338        3.991260e-02 \n",
       "134 2549338        3.216969e-01 \n",
       "135 2549338        1.605424e-01 \n",
       "136 2549338        2.301288e-01 \n",
       "137    3705        4.477293e+03 \n",
       "138    3705        1.476358e+03 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "parsed_df <- merged_df %>% \n",
    "    left_join(ori_df)\n",
    "parsed_df\n",
    "fwrite(merged_df, \"results/irep_analysis/coverage_irep_results.raw.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "69d638b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "ori_df <- merged_df %>%\n",
    "    select(prefix, ORI, TER) %>%\n",
    "    filter(ORI != \"0\" & TER != \"0\") %>%\n",
    "    distinct()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "0a62c820",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Joining, by = \"prefix\"\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 7 rows [13, 14, 15, 16, 24, 26, 30].”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 30 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>taxa</th><th scope=col>prefix</th><th scope=col>mean_perc_covered1</th><th scope=col>mean_perc_covered5</th><th scope=col>min_perc_covered1</th><th scope=col>max_perc_covered1</th><th scope=col>max_bPTR</th><th scope=col>max_microbe_count</th><th scope=col>ORI</th><th scope=col>TER</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Human betaherpesvirus 6B         </td><td>Human_betaherpesvirus_6B_NC_000898.1           </td><td>98.48291943</td><td>9.544136e+01</td><td>98.34129070</td><td>98.6928951</td><td>      NA</td><td>6.952852e+03</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Human betaherpesvirus 6A         </td><td>Human_betaherpesvirus_6A_NC_001664.4           </td><td>96.76555108</td><td>9.096362e+01</td><td>96.63755349</td><td>96.9826450</td><td>1.282505</td><td>1.111746e+04</td><td>44,954   </td><td>132,612  </td></tr>\n",
       "\t<tr><td>Achromobacter xylosoxidans       </td><td>Achromobacter_xylosoxidans_NZ_CP043820.1       </td><td>79.03047361</td><td>7.481513e+01</td><td>78.38978776</td><td>79.5830599</td><td>      NA</td><td>1.383176e+04</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Hepatitis B virus                </td><td>Hepatitis_B_virus_genB_JX661488.1              </td><td>76.39191291</td><td>4.486262e+01</td><td>61.33748056</td><td>98.7558320</td><td>      NA</td><td>2.496341e+05</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Pseudomonas mendocina            </td><td>Pseudomonas_mendocina_NZ_CP013124.1            </td><td>71.41336595</td><td>6.739544e+01</td><td>71.12799127</td><td>71.7810391</td><td>      NA</td><td>1.988941e+04</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Cutibacterium acnes              </td><td>Cutibacterium_acnes_NC_021085.1                </td><td>36.15403798</td><td>3.473737e+00</td><td>15.33223211</td><td>77.4795425</td><td>      NA</td><td>1.763105e+03</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Rickettsia sp. Tillamook 23      </td><td>Rickettsia_Tillamook_23_NZ_CP060138.1          </td><td>31.85980557</td><td>2.757348e+00</td><td> 0.35879756</td><td>63.3608136</td><td>1.346660</td><td>1.197889e+03</td><td>240,832  </td><td>888,370  </td></tr>\n",
       "\t<tr><td>Alcaligenes faecalis             </td><td>Alcaligenes_faecalis_NZ_CP013119.1             </td><td>28.02434056</td><td>5.987024e-01</td><td>23.36010861</td><td>31.7478145</td><td>      NA</td><td>3.634384e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Torque teno virus 6              </td><td>Torque_teno_virus_6_NC_014094.1                </td><td>18.83940621</td><td>1.705803e+01</td><td>16.08636977</td><td>21.5924426</td><td>      NA</td><td>4.477293e+03</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Neisseria subflava               </td><td>Neisseria_subflava_NZ_CP039887.1               </td><td>17.89799782</td><td>5.411104e+00</td><td> 1.20911307</td><td>81.7316806</td><td>1.506405</td><td>1.882095e+03</td><td>1,883,368</td><td>797,590  </td></tr>\n",
       "\t<tr><td>Acinetobacter baumannii          </td><td>Acinetobacter_baumannii_NZ_CP043953.1          </td><td>17.31466235</td><td>3.395586e+00</td><td> 1.31604286</td><td>79.3290721</td><td>1.896385</td><td>8.490606e+02</td><td>3,869,401</td><td>1,967,227</td></tr>\n",
       "\t<tr><td>Fannyhessea vaginae              </td><td>Fannyhessea_vaginae_NZ_UFSV01000001.1          </td><td>17.26185948</td><td>1.865483e+00</td><td> 1.22422672</td><td>75.9249475</td><td>1.876599</td><td>7.075888e+02</td><td>613,931  </td><td>1,264,620</td></tr>\n",
       "\t<tr><td>Neisseria flavescens             </td><td>Neisseria_flavescens_NZ_CP039886.1             </td><td>16.77717729</td><td>4.927474e+00</td><td> 1.11999649</td><td>76.9177761</td><td>1.380708</td><td>1.721050e+03</td><td>1,737,155</td><td>716,802  </td></tr>\n",
       "\t<tr><td>Staphylococcus epidermidis       </td><td>Staphylococcus_epidermidis_NZ_CP035288.1       </td><td>15.36405809</td><td>2.958846e-01</td><td> 2.03810903</td><td>52.9612382</td><td>1.566844</td><td>7.349293e+02</td><td>237,449  </td><td>1,347,375</td></tr>\n",
       "\t<tr><td>Haemophilus parainfluenzae       </td><td>Haemophilus_parainfluenzae_NZ_GL872339.1       </td><td>14.73871027</td><td>7.062885e-01</td><td> 0.65722936</td><td>60.4802455</td><td>1.172580</td><td>7.445671e+02</td><td>492,915  </td><td>1,443,748</td></tr>\n",
       "\t<tr><td>Fusobacterium nucleatum          </td><td>Fusobacterium_nucleatum_NZ_LN831027.1          </td><td>14.22900825</td><td>1.150248e+01</td><td> 0.32135060</td><td>65.7792926</td><td>1.675886</td><td>3.657746e+03</td><td>201,937  </td><td>1,345,205</td></tr>\n",
       "\t<tr><td>Corynebacterium segmentosum      </td><td>Corynebacterium_segmentosum_NZ_LR134408.1      </td><td>13.99622489</td><td>1.490941e+00</td><td> 0.24934273</td><td>66.3968333</td><td>      NA</td><td>1.281144e+03</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Lactobacillus crispatus          </td><td>Lactobacillus_crispatus_NZ_CP039266.1          </td><td>13.00527270</td><td>3.891430e-01</td><td> 3.96804644</td><td>41.0523459</td><td>1.571468</td><td>5.414163e+02</td><td>328,627  </td><td>1,239,005</td></tr>\n",
       "\t<tr><td>Fusobacterium pseudoperiodonticum</td><td>Fusobacterium_pseudoperiodonticum_NZ_CP024702.1</td><td>11.77652109</td><td>6.551175e+00</td><td> 0.27596339</td><td>39.3815911</td><td>      NA</td><td>1.354117e+03</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Prevotella sp. oral taxon 299    </td><td>Prevotella_oral_taxon_299_NC_022124.1          </td><td>10.85898429</td><td>1.770092e-01</td><td> 1.00176094</td><td>20.7162076</td><td>      NA</td><td>1.644861e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Moraxella osloensis              </td><td>Moraxella_osloensis_NZ_CP014234.1              </td><td> 9.14462141</td><td>1.206725e-02</td><td> 5.59357092</td><td>17.0975911</td><td>1.329333</td><td>1.381334e+02</td><td>1,501,499</td><td>405,889  </td></tr>\n",
       "\t<tr><td>Gardnerella vaginalis            </td><td>Gardnerella_vaginalis_NZ_PKJK01000001.1        </td><td> 8.67918722</td><td>1.926305e-02</td><td> 5.29527766</td><td>14.6525344</td><td>      NA</td><td>1.417187e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Human mastadenovirus C           </td><td>Human_mastadenovirus_C_NC_001405.1             </td><td> 8.37354259</td><td>6.947158e+00</td><td> 4.61084676</td><td>22.3363108</td><td>      NA</td><td>1.920488e+04</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Prevotella melaninogenica        </td><td>Prevotella_melaninogenica_NC_014370.1          </td><td> 6.79420265</td><td>5.026698e-02</td><td> 0.44878446</td><td>25.1270313</td><td>      NA</td><td>2.011119e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Cupriavidus metallidurans        </td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1    </td><td> 5.16601427</td><td>8.541929e-03</td><td> 0.44472134</td><td>15.7048526</td><td>      NA</td><td>1.635871e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Neisseria mucosa                 </td><td>Neisseria_mucosa_NZ_UGRT01000001.1             </td><td> 5.01441152</td><td>9.105637e-01</td><td> 0.11922013</td><td>22.5393464</td><td>      NA</td><td>3.361585e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Malassezia restricta             </td><td>Malassezia_restricta_NC_040193.1               </td><td> 3.38365554</td><td>8.145150e-03</td><td> 0.04072575</td><td>12.5383255</td><td>      NA</td><td>1.162379e+02</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Staphylococcus haemolyticus      </td><td>Staphylococcus_haemolyticus_NZ_CP013911.1      </td><td> 0.26897963</td><td>8.480633e-02</td><td> 0.23562980</td><td> 0.2961161</td><td>      NA</td><td>3.216969e-01</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Staphylococcus cohnii            </td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1        </td><td> 0.24924845</td><td>6.094140e-02</td><td> 0.21837648</td><td> 0.2688547</td><td>      NA</td><td>3.076552e-01</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Aspergillus oryzae               </td><td>Aspergillus_oryzae_NC_036435.1                 </td><td> 0.03715799</td><td>8.741975e-04</td><td> 0.00000000</td><td> 0.1697323</td><td>      NA</td><td>5.564388e-01</td><td>NA       </td><td>NA       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " taxa & prefix & mean\\_perc\\_covered1 & mean\\_perc\\_covered5 & min\\_perc\\_covered1 & max\\_perc\\_covered1 & max\\_bPTR & max\\_microbe\\_count & ORI & TER\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t Human betaherpesvirus 6B          & Human\\_betaherpesvirus\\_6B\\_NC\\_000898.1            & 98.48291943 & 9.544136e+01 & 98.34129070 & 98.6928951 &       NA & 6.952852e+03 & NA        & NA       \\\\\n",
       "\t Human betaherpesvirus 6A          & Human\\_betaherpesvirus\\_6A\\_NC\\_001664.4            & 96.76555108 & 9.096362e+01 & 96.63755349 & 96.9826450 & 1.282505 & 1.111746e+04 & 44,954    & 132,612  \\\\\n",
       "\t Achromobacter xylosoxidans        & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1        & 79.03047361 & 7.481513e+01 & 78.38978776 & 79.5830599 &       NA & 1.383176e+04 & NA        & NA       \\\\\n",
       "\t Hepatitis B virus                 & Hepatitis\\_B\\_virus\\_genB\\_JX661488.1               & 76.39191291 & 4.486262e+01 & 61.33748056 & 98.7558320 &       NA & 2.496341e+05 & NA        & NA       \\\\\n",
       "\t Pseudomonas mendocina             & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1             & 71.41336595 & 6.739544e+01 & 71.12799127 & 71.7810391 &       NA & 1.988941e+04 & NA        & NA       \\\\\n",
       "\t Cutibacterium acnes               & Cutibacterium\\_acnes\\_NC\\_021085.1                 & 36.15403798 & 3.473737e+00 & 15.33223211 & 77.4795425 &       NA & 1.763105e+03 & NA        & NA       \\\\\n",
       "\t Rickettsia sp. Tillamook 23       & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1           & 31.85980557 & 2.757348e+00 &  0.35879756 & 63.3608136 & 1.346660 & 1.197889e+03 & 240,832   & 888,370  \\\\\n",
       "\t Alcaligenes faecalis              & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1              & 28.02434056 & 5.987024e-01 & 23.36010861 & 31.7478145 &       NA & 3.634384e+02 & NA        & NA       \\\\\n",
       "\t Torque teno virus 6               & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1                 & 18.83940621 & 1.705803e+01 & 16.08636977 & 21.5924426 &       NA & 4.477293e+03 & NA        & NA       \\\\\n",
       "\t Neisseria subflava                & Neisseria\\_subflava\\_NZ\\_CP039887.1                & 17.89799782 & 5.411104e+00 &  1.20911307 & 81.7316806 & 1.506405 & 1.882095e+03 & 1,883,368 & 797,590  \\\\\n",
       "\t Acinetobacter baumannii           & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1           & 17.31466235 & 3.395586e+00 &  1.31604286 & 79.3290721 & 1.896385 & 8.490606e+02 & 3,869,401 & 1,967,227\\\\\n",
       "\t Fannyhessea vaginae               & Fannyhessea\\_vaginae\\_NZ\\_UFSV01000001.1           & 17.26185948 & 1.865483e+00 &  1.22422672 & 75.9249475 & 1.876599 & 7.075888e+02 & 613,931   & 1,264,620\\\\\n",
       "\t Neisseria flavescens              & Neisseria\\_flavescens\\_NZ\\_CP039886.1              & 16.77717729 & 4.927474e+00 &  1.11999649 & 76.9177761 & 1.380708 & 1.721050e+03 & 1,737,155 & 716,802  \\\\\n",
       "\t Staphylococcus epidermidis        & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1        & 15.36405809 & 2.958846e-01 &  2.03810903 & 52.9612382 & 1.566844 & 7.349293e+02 & 237,449   & 1,347,375\\\\\n",
       "\t Haemophilus parainfluenzae        & Haemophilus\\_parainfluenzae\\_NZ\\_GL872339.1        & 14.73871027 & 7.062885e-01 &  0.65722936 & 60.4802455 & 1.172580 & 7.445671e+02 & 492,915   & 1,443,748\\\\\n",
       "\t Fusobacterium nucleatum           & Fusobacterium\\_nucleatum\\_NZ\\_LN831027.1           & 14.22900825 & 1.150248e+01 &  0.32135060 & 65.7792926 & 1.675886 & 3.657746e+03 & 201,937   & 1,345,205\\\\\n",
       "\t Corynebacterium segmentosum       & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1       & 13.99622489 & 1.490941e+00 &  0.24934273 & 66.3968333 &       NA & 1.281144e+03 & NA        & NA       \\\\\n",
       "\t Lactobacillus crispatus           & Lactobacillus\\_crispatus\\_NZ\\_CP039266.1           & 13.00527270 & 3.891430e-01 &  3.96804644 & 41.0523459 & 1.571468 & 5.414163e+02 & 328,627   & 1,239,005\\\\\n",
       "\t Fusobacterium pseudoperiodonticum & Fusobacterium\\_pseudoperiodonticum\\_NZ\\_CP024702.1 & 11.77652109 & 6.551175e+00 &  0.27596339 & 39.3815911 &       NA & 1.354117e+03 & NA        & NA       \\\\\n",
       "\t Prevotella sp. oral taxon 299     & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1           & 10.85898429 & 1.770092e-01 &  1.00176094 & 20.7162076 &       NA & 1.644861e+02 & NA        & NA       \\\\\n",
       "\t Moraxella osloensis               & Moraxella\\_osloensis\\_NZ\\_CP014234.1               &  9.14462141 & 1.206725e-02 &  5.59357092 & 17.0975911 & 1.329333 & 1.381334e+02 & 1,501,499 & 405,889  \\\\\n",
       "\t Gardnerella vaginalis             & Gardnerella\\_vaginalis\\_NZ\\_PKJK01000001.1         &  8.67918722 & 1.926305e-02 &  5.29527766 & 14.6525344 &       NA & 1.417187e+02 & NA        & NA       \\\\\n",
       "\t Human mastadenovirus C            & Human\\_mastadenovirus\\_C\\_NC\\_001405.1              &  8.37354259 & 6.947158e+00 &  4.61084676 & 22.3363108 &       NA & 1.920488e+04 & NA        & NA       \\\\\n",
       "\t Prevotella melaninogenica         & Prevotella\\_melaninogenica\\_NC\\_014370.1           &  6.79420265 & 5.026698e-02 &  0.44878446 & 25.1270313 &       NA & 2.011119e+02 & NA        & NA       \\\\\n",
       "\t Cupriavidus metallidurans         & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1     &  5.16601427 & 8.541929e-03 &  0.44472134 & 15.7048526 &       NA & 1.635871e+02 & NA        & NA       \\\\\n",
       "\t Neisseria mucosa                  & Neisseria\\_mucosa\\_NZ\\_UGRT01000001.1              &  5.01441152 & 9.105637e-01 &  0.11922013 & 22.5393464 &       NA & 3.361585e+02 & NA        & NA       \\\\\n",
       "\t Malassezia restricta              & Malassezia\\_restricta\\_NC\\_040193.1                &  3.38365554 & 8.145150e-03 &  0.04072575 & 12.5383255 &       NA & 1.162379e+02 & NA        & NA       \\\\\n",
       "\t Staphylococcus haemolyticus       & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1       &  0.26897963 & 8.480633e-02 &  0.23562980 &  0.2961161 &       NA & 3.216969e-01 & NA        & NA       \\\\\n",
       "\t Staphylococcus cohnii             & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1         &  0.24924845 & 6.094140e-02 &  0.21837648 &  0.2688547 &       NA & 3.076552e-01 & NA        & NA       \\\\\n",
       "\t Aspergillus oryzae                & Aspergillus\\_oryzae\\_NC\\_036435.1                  &  0.03715799 & 8.741975e-04 &  0.00000000 &  0.1697323 &       NA & 5.564388e-01 & NA        & NA       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 10\n",
       "\n",
       "| taxa &lt;chr&gt; | prefix &lt;chr&gt; | mean_perc_covered1 &lt;dbl&gt; | mean_perc_covered5 &lt;dbl&gt; | min_perc_covered1 &lt;dbl&gt; | max_perc_covered1 &lt;dbl&gt; | max_bPTR &lt;dbl&gt; | max_microbe_count &lt;dbl&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| Human betaherpesvirus 6B          | Human_betaherpesvirus_6B_NC_000898.1            | 98.48291943 | 9.544136e+01 | 98.34129070 | 98.6928951 |       NA | 6.952852e+03 | NA        | NA        |\n",
       "| Human betaherpesvirus 6A          | Human_betaherpesvirus_6A_NC_001664.4            | 96.76555108 | 9.096362e+01 | 96.63755349 | 96.9826450 | 1.282505 | 1.111746e+04 | 44,954    | 132,612   |\n",
       "| Achromobacter xylosoxidans        | Achromobacter_xylosoxidans_NZ_CP043820.1        | 79.03047361 | 7.481513e+01 | 78.38978776 | 79.5830599 |       NA | 1.383176e+04 | NA        | NA        |\n",
       "| Hepatitis B virus                 | Hepatitis_B_virus_genB_JX661488.1               | 76.39191291 | 4.486262e+01 | 61.33748056 | 98.7558320 |       NA | 2.496341e+05 | NA        | NA        |\n",
       "| Pseudomonas mendocina             | Pseudomonas_mendocina_NZ_CP013124.1             | 71.41336595 | 6.739544e+01 | 71.12799127 | 71.7810391 |       NA | 1.988941e+04 | NA        | NA        |\n",
       "| Cutibacterium acnes               | Cutibacterium_acnes_NC_021085.1                 | 36.15403798 | 3.473737e+00 | 15.33223211 | 77.4795425 |       NA | 1.763105e+03 | NA        | NA        |\n",
       "| Rickettsia sp. Tillamook 23       | Rickettsia_Tillamook_23_NZ_CP060138.1           | 31.85980557 | 2.757348e+00 |  0.35879756 | 63.3608136 | 1.346660 | 1.197889e+03 | 240,832   | 888,370   |\n",
       "| Alcaligenes faecalis              | Alcaligenes_faecalis_NZ_CP013119.1              | 28.02434056 | 5.987024e-01 | 23.36010861 | 31.7478145 |       NA | 3.634384e+02 | NA        | NA        |\n",
       "| Torque teno virus 6               | Torque_teno_virus_6_NC_014094.1                 | 18.83940621 | 1.705803e+01 | 16.08636977 | 21.5924426 |       NA | 4.477293e+03 | NA        | NA        |\n",
       "| Neisseria subflava                | Neisseria_subflava_NZ_CP039887.1                | 17.89799782 | 5.411104e+00 |  1.20911307 | 81.7316806 | 1.506405 | 1.882095e+03 | 1,883,368 | 797,590   |\n",
       "| Acinetobacter baumannii           | Acinetobacter_baumannii_NZ_CP043953.1           | 17.31466235 | 3.395586e+00 |  1.31604286 | 79.3290721 | 1.896385 | 8.490606e+02 | 3,869,401 | 1,967,227 |\n",
       "| Fannyhessea vaginae               | Fannyhessea_vaginae_NZ_UFSV01000001.1           | 17.26185948 | 1.865483e+00 |  1.22422672 | 75.9249475 | 1.876599 | 7.075888e+02 | 613,931   | 1,264,620 |\n",
       "| Neisseria flavescens              | Neisseria_flavescens_NZ_CP039886.1              | 16.77717729 | 4.927474e+00 |  1.11999649 | 76.9177761 | 1.380708 | 1.721050e+03 | 1,737,155 | 716,802   |\n",
       "| Staphylococcus epidermidis        | Staphylococcus_epidermidis_NZ_CP035288.1        | 15.36405809 | 2.958846e-01 |  2.03810903 | 52.9612382 | 1.566844 | 7.349293e+02 | 237,449   | 1,347,375 |\n",
       "| Haemophilus parainfluenzae        | Haemophilus_parainfluenzae_NZ_GL872339.1        | 14.73871027 | 7.062885e-01 |  0.65722936 | 60.4802455 | 1.172580 | 7.445671e+02 | 492,915   | 1,443,748 |\n",
       "| Fusobacterium nucleatum           | Fusobacterium_nucleatum_NZ_LN831027.1           | 14.22900825 | 1.150248e+01 |  0.32135060 | 65.7792926 | 1.675886 | 3.657746e+03 | 201,937   | 1,345,205 |\n",
       "| Corynebacterium segmentosum       | Corynebacterium_segmentosum_NZ_LR134408.1       | 13.99622489 | 1.490941e+00 |  0.24934273 | 66.3968333 |       NA | 1.281144e+03 | NA        | NA        |\n",
       "| Lactobacillus crispatus           | Lactobacillus_crispatus_NZ_CP039266.1           | 13.00527270 | 3.891430e-01 |  3.96804644 | 41.0523459 | 1.571468 | 5.414163e+02 | 328,627   | 1,239,005 |\n",
       "| Fusobacterium pseudoperiodonticum | Fusobacterium_pseudoperiodonticum_NZ_CP024702.1 | 11.77652109 | 6.551175e+00 |  0.27596339 | 39.3815911 |       NA | 1.354117e+03 | NA        | NA        |\n",
       "| Prevotella sp. oral taxon 299     | Prevotella_oral_taxon_299_NC_022124.1           | 10.85898429 | 1.770092e-01 |  1.00176094 | 20.7162076 |       NA | 1.644861e+02 | NA        | NA        |\n",
       "| Moraxella osloensis               | Moraxella_osloensis_NZ_CP014234.1               |  9.14462141 | 1.206725e-02 |  5.59357092 | 17.0975911 | 1.329333 | 1.381334e+02 | 1,501,499 | 405,889   |\n",
       "| Gardnerella vaginalis             | Gardnerella_vaginalis_NZ_PKJK01000001.1         |  8.67918722 | 1.926305e-02 |  5.29527766 | 14.6525344 |       NA | 1.417187e+02 | NA        | NA        |\n",
       "| Human mastadenovirus C            | Human_mastadenovirus_C_NC_001405.1              |  8.37354259 | 6.947158e+00 |  4.61084676 | 22.3363108 |       NA | 1.920488e+04 | NA        | NA        |\n",
       "| Prevotella melaninogenica         | Prevotella_melaninogenica_NC_014370.1           |  6.79420265 | 5.026698e-02 |  0.44878446 | 25.1270313 |       NA | 2.011119e+02 | NA        | NA        |\n",
       "| Cupriavidus metallidurans         | Cupriavidus_metallidurans_NZ_FYAX01000013.1     |  5.16601427 | 8.541929e-03 |  0.44472134 | 15.7048526 |       NA | 1.635871e+02 | NA        | NA        |\n",
       "| Neisseria mucosa                  | Neisseria_mucosa_NZ_UGRT01000001.1              |  5.01441152 | 9.105637e-01 |  0.11922013 | 22.5393464 |       NA | 3.361585e+02 | NA        | NA        |\n",
       "| Malassezia restricta              | Malassezia_restricta_NC_040193.1                |  3.38365554 | 8.145150e-03 |  0.04072575 | 12.5383255 |       NA | 1.162379e+02 | NA        | NA        |\n",
       "| Staphylococcus haemolyticus       | Staphylococcus_haemolyticus_NZ_CP013911.1       |  0.26897963 | 8.480633e-02 |  0.23562980 |  0.2961161 |       NA | 3.216969e-01 | NA        | NA        |\n",
       "| Staphylococcus cohnii             | Staphylococcus_cohnii_NZ_UHDA01000001.1         |  0.24924845 | 6.094140e-02 |  0.21837648 |  0.2688547 |       NA | 3.076552e-01 | NA        | NA        |\n",
       "| Aspergillus oryzae                | Aspergillus_oryzae_NC_036435.1                  |  0.03715799 | 8.741975e-04 |  0.00000000 |  0.1697323 |       NA | 5.564388e-01 | NA        | NA        |\n",
       "\n"
      ],
      "text/plain": [
       "   taxa                             \n",
       "1  Human betaherpesvirus 6B         \n",
       "2  Human betaherpesvirus 6A         \n",
       "3  Achromobacter xylosoxidans       \n",
       "4  Hepatitis B virus                \n",
       "5  Pseudomonas mendocina            \n",
       "6  Cutibacterium acnes              \n",
       "7  Rickettsia sp. Tillamook 23      \n",
       "8  Alcaligenes faecalis             \n",
       "9  Torque teno virus 6              \n",
       "10 Neisseria subflava               \n",
       "11 Acinetobacter baumannii          \n",
       "12 Fannyhessea vaginae              \n",
       "13 Neisseria flavescens             \n",
       "14 Staphylococcus epidermidis       \n",
       "15 Haemophilus parainfluenzae       \n",
       "16 Fusobacterium nucleatum          \n",
       "17 Corynebacterium segmentosum      \n",
       "18 Lactobacillus crispatus          \n",
       "19 Fusobacterium pseudoperiodonticum\n",
       "20 Prevotella sp. oral taxon 299    \n",
       "21 Moraxella osloensis              \n",
       "22 Gardnerella vaginalis            \n",
       "23 Human mastadenovirus C           \n",
       "24 Prevotella melaninogenica        \n",
       "25 Cupriavidus metallidurans        \n",
       "26 Neisseria mucosa                 \n",
       "27 Malassezia restricta             \n",
       "28 Staphylococcus haemolyticus      \n",
       "29 Staphylococcus cohnii            \n",
       "30 Aspergillus oryzae               \n",
       "   prefix                                          mean_perc_covered1\n",
       "1  Human_betaherpesvirus_6B_NC_000898.1            98.48291943       \n",
       "2  Human_betaherpesvirus_6A_NC_001664.4            96.76555108       \n",
       "3  Achromobacter_xylosoxidans_NZ_CP043820.1        79.03047361       \n",
       "4  Hepatitis_B_virus_genB_JX661488.1               76.39191291       \n",
       "5  Pseudomonas_mendocina_NZ_CP013124.1             71.41336595       \n",
       "6  Cutibacterium_acnes_NC_021085.1                 36.15403798       \n",
       "7  Rickettsia_Tillamook_23_NZ_CP060138.1           31.85980557       \n",
       "8  Alcaligenes_faecalis_NZ_CP013119.1              28.02434056       \n",
       "9  Torque_teno_virus_6_NC_014094.1                 18.83940621       \n",
       "10 Neisseria_subflava_NZ_CP039887.1                17.89799782       \n",
       "11 Acinetobacter_baumannii_NZ_CP043953.1           17.31466235       \n",
       "12 Fannyhessea_vaginae_NZ_UFSV01000001.1           17.26185948       \n",
       "13 Neisseria_flavescens_NZ_CP039886.1              16.77717729       \n",
       "14 Staphylococcus_epidermidis_NZ_CP035288.1        15.36405809       \n",
       "15 Haemophilus_parainfluenzae_NZ_GL872339.1        14.73871027       \n",
       "16 Fusobacterium_nucleatum_NZ_LN831027.1           14.22900825       \n",
       "17 Corynebacterium_segmentosum_NZ_LR134408.1       13.99622489       \n",
       "18 Lactobacillus_crispatus_NZ_CP039266.1           13.00527270       \n",
       "19 Fusobacterium_pseudoperiodonticum_NZ_CP024702.1 11.77652109       \n",
       "20 Prevotella_oral_taxon_299_NC_022124.1           10.85898429       \n",
       "21 Moraxella_osloensis_NZ_CP014234.1                9.14462141       \n",
       "22 Gardnerella_vaginalis_NZ_PKJK01000001.1          8.67918722       \n",
       "23 Human_mastadenovirus_C_NC_001405.1               8.37354259       \n",
       "24 Prevotella_melaninogenica_NC_014370.1            6.79420265       \n",
       "25 Cupriavidus_metallidurans_NZ_FYAX01000013.1      5.16601427       \n",
       "26 Neisseria_mucosa_NZ_UGRT01000001.1               5.01441152       \n",
       "27 Malassezia_restricta_NC_040193.1                 3.38365554       \n",
       "28 Staphylococcus_haemolyticus_NZ_CP013911.1        0.26897963       \n",
       "29 Staphylococcus_cohnii_NZ_UHDA01000001.1          0.24924845       \n",
       "30 Aspergillus_oryzae_NC_036435.1                   0.03715799       \n",
       "   mean_perc_covered5 min_perc_covered1 max_perc_covered1 max_bPTR\n",
       "1  9.544136e+01       98.34129070       98.6928951              NA\n",
       "2  9.096362e+01       96.63755349       96.9826450        1.282505\n",
       "3  7.481513e+01       78.38978776       79.5830599              NA\n",
       "4  4.486262e+01       61.33748056       98.7558320              NA\n",
       "5  6.739544e+01       71.12799127       71.7810391              NA\n",
       "6  3.473737e+00       15.33223211       77.4795425              NA\n",
       "7  2.757348e+00        0.35879756       63.3608136        1.346660\n",
       "8  5.987024e-01       23.36010861       31.7478145              NA\n",
       "9  1.705803e+01       16.08636977       21.5924426              NA\n",
       "10 5.411104e+00        1.20911307       81.7316806        1.506405\n",
       "11 3.395586e+00        1.31604286       79.3290721        1.896385\n",
       "12 1.865483e+00        1.22422672       75.9249475        1.876599\n",
       "13 4.927474e+00        1.11999649       76.9177761        1.380708\n",
       "14 2.958846e-01        2.03810903       52.9612382        1.566844\n",
       "15 7.062885e-01        0.65722936       60.4802455        1.172580\n",
       "16 1.150248e+01        0.32135060       65.7792926        1.675886\n",
       "17 1.490941e+00        0.24934273       66.3968333              NA\n",
       "18 3.891430e-01        3.96804644       41.0523459        1.571468\n",
       "19 6.551175e+00        0.27596339       39.3815911              NA\n",
       "20 1.770092e-01        1.00176094       20.7162076              NA\n",
       "21 1.206725e-02        5.59357092       17.0975911        1.329333\n",
       "22 1.926305e-02        5.29527766       14.6525344              NA\n",
       "23 6.947158e+00        4.61084676       22.3363108              NA\n",
       "24 5.026698e-02        0.44878446       25.1270313              NA\n",
       "25 8.541929e-03        0.44472134       15.7048526              NA\n",
       "26 9.105637e-01        0.11922013       22.5393464              NA\n",
       "27 8.145150e-03        0.04072575       12.5383255              NA\n",
       "28 8.480633e-02        0.23562980        0.2961161              NA\n",
       "29 6.094140e-02        0.21837648        0.2688547              NA\n",
       "30 8.741975e-04        0.00000000        0.1697323              NA\n",
       "   max_microbe_count ORI       TER      \n",
       "1  6.952852e+03      NA        NA       \n",
       "2  1.111746e+04      44,954    132,612  \n",
       "3  1.383176e+04      NA        NA       \n",
       "4  2.496341e+05      NA        NA       \n",
       "5  1.988941e+04      NA        NA       \n",
       "6  1.763105e+03      NA        NA       \n",
       "7  1.197889e+03      240,832   888,370  \n",
       "8  3.634384e+02      NA        NA       \n",
       "9  4.477293e+03      NA        NA       \n",
       "10 1.882095e+03      1,883,368 797,590  \n",
       "11 8.490606e+02      3,869,401 1,967,227\n",
       "12 7.075888e+02      613,931   1,264,620\n",
       "13 1.721050e+03      1,737,155 716,802  \n",
       "14 7.349293e+02      237,449   1,347,375\n",
       "15 7.445671e+02      492,915   1,443,748\n",
       "16 3.657746e+03      201,937   1,345,205\n",
       "17 1.281144e+03      NA        NA       \n",
       "18 5.414163e+02      328,627   1,239,005\n",
       "19 1.354117e+03      NA        NA       \n",
       "20 1.644861e+02      NA        NA       \n",
       "21 1.381334e+02      1,501,499 405,889  \n",
       "22 1.417187e+02      NA        NA       \n",
       "23 1.920488e+04      NA        NA       \n",
       "24 2.011119e+02      NA        NA       \n",
       "25 1.635871e+02      NA        NA       \n",
       "26 3.361585e+02      NA        NA       \n",
       "27 1.162379e+02      NA        NA       \n",
       "28 3.216969e-01      NA        NA       \n",
       "29 3.076552e-01      NA        NA       \n",
       "30 5.564388e-01      NA        NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summarised_df <- merged_df %>%\n",
    "    group_by(prefix) %>%\n",
    "    summarise(mean_perc_covered1 = mean(perc_covered1),\n",
    "              mean_perc_covered5 = mean(perc_covered5),\n",
    "              min_perc_covered1 = min(perc_covered1),\n",
    "              max_perc_covered1 = max(perc_covered1),\n",
    "              max_bPTR = max(bPTR, na.rm = T),\n",
    "              max_microbe_count = max(microbe_count)) %>%\n",
    "    mutate(max_bPTR = ifelse(max_bPTR == -Inf, NA, max_bPTR)) %>%\n",
    "    left_join(ori_df) %>%\n",
    "    separate(prefix, into = c(\"genus\", \"species\", \"suffix1\", \"suffix2\"), sep = \"_\", remove = F) %>%\n",
    "    mutate(taxa = ifelse(grepl(\"Torque\", prefix), \n",
    "                         paste(genus, species, suffix1, suffix2), \n",
    "                         paste(genus, species, suffix1)), .before = 1) %>%\n",
    "    mutate(taxa = gsub(\" NZ| NC\", \"\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Microbacterium\", \"Microbacterium sp.\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Prevotella oral taxon\", \"Prevotella sp. oral taxon 299\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Rickettsia\", \"Rickettsia sp.\", taxa)) %>%\n",
    "    select(-genus, -species, -suffix1, -suffix2) %>%\n",
    "    arrange(desc(mean_perc_covered1))\n",
    "\n",
    "\n",
    "fwrite(summarised_df, \"results/irep_analysis/coverage_irep_results.parsed.csv\")\n",
    "\n",
    "summarised_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
