{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "844f2b78",
   "metadata": {},
   "source": [
    "# iRep (bPTR) and coverage breadth analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "53041a0d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rsamtools\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mBiostrings\u001b[39m::collapse(), \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mcompact()\u001b[39m    masks \u001b[34mXVector\u001b[39m::compact()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mXVector\u001b[39m::slice(), \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "\n",
      "Attaching package: ‘foreach’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:purrr’:\n",
      "\n",
      "    accumulate, when\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm(list = ls())\n",
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(\"Rsamtools\")\n",
    "require(Biostrings)\n",
    "require(tidyverse)\n",
    "require(data.table)\n",
    "require(foreach)\n",
    "require(doParallel)\n",
    "registerDoParallel(cores = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c3332d32",
   "metadata": {},
   "outputs": [],
   "source": [
    "species_df <- fread(\"results/decontamination/read_matrix.raw.zeroed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6673e418",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 4 rows [1, 2, 3, 4].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 3 rows [1, 2, 3].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 3 rows [1, 2, 3].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n"
     ]
    }
   ],
   "source": [
    "files <- list.files(\"results/irep_analysis/raw_output/irep_out\", full.names = T)\n",
    "files <- files[grepl(\".tsv\", files)]\n",
    "\n",
    "morsels <- foreach(file = files) %do% {\n",
    "    temp_df <- read.csv(file, sep = \"\\t\", check.names = F)\n",
    "    colnames(temp_df)[1] <- \"ref_path\"\n",
    "    temp_df %>% \n",
    "        mutate(across(c(4:ncol(temp_df)), function(x) (as.numeric(ifelse(x == \"n/a\", NA, x)))),\n",
    "               across(c(1:3), as.character)) %>%\n",
    "        pivot_longer(!all_of(c(\"ref_path\", \"ORI\", \"TER\")), names_to = \"prefix\", values_to = \"bPTR\") %>%\n",
    "        separate(ref_path, sep = \"/\", into = c(rep(NA, 11), \"file_name\")) %>%\n",
    "        separate(file_name, sep = \"_\", into = c(\"genus\", \"species\", \"suffix1\", \"suffix2\"), remove = F) %>%\n",
    "        separate(prefix, into = c(rep(NA, 6), \"id\"), sep = \"/\") %>%\n",
    "        separate(id, into = c(\"npm_research_id\"), sep = \"_\") %>%\n",
    "        mutate(prefix = gsub(\".fasta\", \"\", file_name))\n",
    "}\n",
    "\n",
    "irep_df <- bind_rows(morsels) \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd89f3e8",
   "metadata": {},
   "source": [
    "## Parse coverage breadth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f6fb3ed3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 153 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>file_name</th><th scope=col>genus</th><th scope=col>species</th><th scope=col>suffix1</th><th scope=col>suffix2</th><th scope=col>ORI</th><th scope=col>TER</th><th scope=col>npm_research_id</th><th scope=col>bPTR</th><th scope=col>prefix</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>1116-0023</td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3658  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3674  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3686  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3734  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>0116-0053</td><td>1.896385</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4240  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4251  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4438  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4828  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9245  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9246  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9270  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9282  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9317  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB3883  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB4048  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB4056  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB4130  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHH1822  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB6459  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB6776  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB6849  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB8186  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB8568  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB6155  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9217  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9583  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9649  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9882  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB3939  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB5008  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB8397  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB9179  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>WHB4373  </td><td>      NA</td><td>Prevotella_oral_taxon_299_NC_022124.1    </td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>WHB9179  </td><td>      NA</td><td>Prevotella_oral_taxon_299_NC_022124.1    </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>0415-0080</td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>0715-0061</td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>0816-0014</td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>WHB3477  </td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>WHB3479  </td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>WHB6775  </td><td>      NA</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>WHB9978  </td><td>1.346660</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0021</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0024</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0026</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0031</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0032</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB4219  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB4398  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB6459  </td><td>1.566844</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB6651  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB8186  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0024</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0026</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0027</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0031</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0715-0059</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>WHB7619  </td><td>      NA</td><td>Torque_teno_virus_6_NC_014094.1          </td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>WHB9685  </td><td>      NA</td><td>Torque_teno_virus_6_NC_014094.1          </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 153 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " file\\_name & genus & species & suffix1 & suffix2 & ORI & TER & npm\\_research\\_id & bPTR & prefix\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & 1116-0023 &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3658   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3674   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3686   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3734   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & 0116-0053 & 1.896385 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4240   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4251   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4438   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4828   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9245   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9246   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9270   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9282   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9317   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB3883   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB4048   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB4056   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB4130   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHH1822   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB6459   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB6776   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB6849   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB8186   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB8568   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB6155   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9217   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9583   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9649   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9882   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB3939   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB5008   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB8397   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB9179   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         & WHB4373   &       NA & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1    \\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         & WHB9179   &       NA & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1    \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & 0415-0080 &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & 0715-0061 &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & 0816-0014 &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & WHB3477   &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & WHB3479   &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & WHB6775   &       NA & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1    \\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & WHB9978   & 1.346660 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1    \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0021 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0024 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0026 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0031 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0032 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB4219   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB4398   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB6459   & 1.566844 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB6651   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB8186   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0024 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0026 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0027 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0031 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0715-0059 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         & WHB7619   &       NA & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1          \\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         & WHB9685   &       NA & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1          \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 153 × 10\n",
       "\n",
       "| file_name &lt;chr&gt; | genus &lt;chr&gt; | species &lt;chr&gt; | suffix1 &lt;chr&gt; | suffix2 &lt;chr&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; | npm_research_id &lt;chr&gt; | bPTR &lt;dbl&gt; | prefix &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | 1116-0023 |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3658   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3674   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3686   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3734   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | 0116-0053 | 1.896385 | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4240   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4251   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4438   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4828   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9245   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9246   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9270   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9282   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9317   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB3883   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB4048   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB4056   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB4130   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHH1822   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB6459   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB6776   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB6849   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB8186   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB8568   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB6155   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9217   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9583   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9649   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9882   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB3939   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB5008   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB8397   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB9179   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         | WHB4373   |       NA | Prevotella_oral_taxon_299_NC_022124.1     |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         | WHB9179   |       NA | Prevotella_oral_taxon_299_NC_022124.1     |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | 0415-0080 |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | 0715-0061 |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | 0816-0014 |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | WHB3477   |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | WHB3479   |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | WHB6775   |       NA | Rickettsia_Tillamook_23_NZ_CP060138.1     |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | WHB9978   | 1.346660 | Rickettsia_Tillamook_23_NZ_CP060138.1     |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0021 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0024 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0026 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0031 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0032 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB4219   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB4398   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB6459   | 1.566844 | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB6651   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB8186   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0024 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0026 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0027 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0031 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0715-0059 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         | WHB7619   |       NA | Torque_teno_virus_6_NC_014094.1           |\n",
       "| Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         | WHB9685   |       NA | Torque_teno_virus_6_NC_014094.1           |\n",
       "\n"
      ],
      "text/plain": [
       "    file_name                                         genus          \n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "16  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "17  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "18  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "19  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "20  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "⋮   ⋮                                                 ⋮              \n",
       "124 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "125 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "126 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "127 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "128 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "129 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "130 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "131 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "132 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "133 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "134 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "135 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "136 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "137 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "138 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "139 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "140 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "141 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "142 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "143 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "144 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "145 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "146 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "147 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "148 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "149 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "150 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "151 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "152 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "153 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "    species        suffix1 suffix2              ORI       TER      \n",
       "1   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "2   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "3   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "4   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "5   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "6   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "7   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "8   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "9   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "10  baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "11  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "12  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "13  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "14  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "15  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "16  oryzae         NC      036435.1.fasta       0         0        \n",
       "17  oryzae         NC      036435.1.fasta       0         0        \n",
       "18  oryzae         NC      036435.1.fasta       0         0        \n",
       "19  oryzae         NC      036435.1.fasta       0         0        \n",
       "20  oryzae         NC      036435.1.fasta       0         0        \n",
       "21  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "22  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "23  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "24  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "25  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "26  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "27  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "28  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "29  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "30  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "⋮   ⋮              ⋮       ⋮                    ⋮         ⋮        \n",
       "124 melaninogenica NC      014370.1.fasta       0         0        \n",
       "125 melaninogenica NC      014370.1.fasta       0         0        \n",
       "126 melaninogenica NC      014370.1.fasta       0         0        \n",
       "127 melaninogenica NC      014370.1.fasta       0         0        \n",
       "128 oral           taxon   299                  0         0        \n",
       "129 oral           taxon   299                  0         0        \n",
       "130 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "131 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "132 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "133 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "134 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "135 Tillamook      23      NZ                   240,832   888,370  \n",
       "136 Tillamook      23      NZ                   240,832   888,370  \n",
       "137 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "138 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "139 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "140 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "141 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "142 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "143 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "144 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "145 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "146 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "147 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "148 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "149 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "150 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "151 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "152 teno           virus   6                    0         0        \n",
       "153 teno           virus   6                    0         0        \n",
       "    npm_research_id bPTR     prefix                                     \n",
       "1   1116-0023             NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "2   WHB3658               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "3   WHB3674               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "4   WHB3686               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "5   WHB3734               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "6   0116-0053       1.896385 Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "7   WHB4240               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "8   WHB4251               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "9   WHB4438               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "10  WHB4828               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "11  WHB9245               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "12  WHB9246               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "13  WHB9270               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "14  WHB9282               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "15  WHB9317               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "16  WHB3883               NA Aspergillus_oryzae_NC_036435.1             \n",
       "17  WHB4048               NA Aspergillus_oryzae_NC_036435.1             \n",
       "18  WHB4056               NA Aspergillus_oryzae_NC_036435.1             \n",
       "19  WHB4130               NA Aspergillus_oryzae_NC_036435.1             \n",
       "20  WHH1822               NA Aspergillus_oryzae_NC_036435.1             \n",
       "21  WHB6459               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "22  WHB6776               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "23  WHB6849               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "24  WHB8186               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "25  WHB8568               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "26  WHB6155               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "27  WHB9217               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "28  WHB9583               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "29  WHB9649               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "30  WHB9882               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "⋮   ⋮               ⋮        ⋮                                          \n",
       "124 WHB3939               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "125 WHB5008               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "126 WHB8397               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "127 WHB9179               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "128 WHB4373               NA Prevotella_oral_taxon_299_NC_022124.1      \n",
       "129 WHB9179               NA Prevotella_oral_taxon_299_NC_022124.1      \n",
       "130 0415-0080             NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "131 0715-0061             NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "132 0816-0014             NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "133 WHB3477               NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "134 WHB3479               NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "135 WHB6775               NA Rickettsia_Tillamook_23_NZ_CP060138.1      \n",
       "136 WHB9978         1.346660 Rickettsia_Tillamook_23_NZ_CP060138.1      \n",
       "137 0416-0021             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "138 0416-0024             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "139 0416-0026             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "140 0416-0031             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "141 0416-0032             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "142 WHB4219               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "143 WHB4398               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "144 WHB6459         1.566844 Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "145 WHB6651               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "146 WHB8186               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "147 0416-0024             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "148 0416-0026             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "149 0416-0027             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "150 0416-0031             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "151 0715-0059             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "152 WHB7619               NA Torque_teno_virus_6_NC_014094.1            \n",
       "153 WHB9685               NA Torque_teno_virus_6_NC_014094.1            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "irep_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5155b4c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Achromobacter_xylosoxidans_NZ_CP043820.1'</li><li>'Acinetobacter_baumannii_NZ_CP043953.1'</li><li>'Alcaligenes_faecalis_NZ_CP013119.1'</li><li>'Aspergillus_oryzae_NC_036435.1'</li><li>'Corynebacterium_segmentosum_NZ_LR134408.1'</li><li>'Cupriavidus_metallidurans_NZ_FYAX01000013.1'</li><li>'Cutibacterium_acnes_NC_021085.1'</li><li>'Fannyhessea_vaginae_NZ_UFSV01000001.1'</li><li>'Fusobacterium_nucleatum_NZ_LN831027.1'</li><li>'Fusobacterium_pseudoperiodonticum_NZ_CP024702.1'</li><li>'Gardnerella_vaginalis_NZ_PKJK01000001.1'</li><li>'Haemophilus_parainfluenzae_NZ_GL872339.1'</li><li>'Hepatitis_B_virus_genB_JX661488.1'</li><li>'Human_betaherpesvirus_6A_NC_001664.4'</li><li>'Human_betaherpesvirus_6B_NC_000898.1'</li><li>'Human_mastadenovirus_C_NC_001405.1'</li><li>'Lactobacillus_crispatus_NZ_CP039266.1'</li><li>'Malassezia_restricta_NC_040193.1'</li><li>'Moraxella_osloensis_NZ_CP014234.1'</li><li>'Neisseria_flavescens_NZ_CP039886.1'</li><li>'Neisseria_mucosa_NZ_UGRT01000001.1'</li><li>'Neisseria_subflava_NZ_CP039887.1'</li><li>'Prevotella_melaninogenica_NC_014370.1'</li><li>'Prevotella_oral_taxon_299_NC_022124.1'</li><li>'Pseudomonas_mendocina_NZ_CP013124.1'</li><li>'Rickettsia_Tillamook_23_NZ_CP060138.1'</li><li>'Staphylococcus_cohnii_NZ_UHDA01000001.1'</li><li>'Staphylococcus_epidermidis_NZ_CP035288.1'</li><li>'Staphylococcus_haemolyticus_NZ_CP013911.1'</li><li>'Torque_teno_virus_6_NC_014094.1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1'\n",
       "\\item 'Acinetobacter\\_baumannii\\_NZ\\_CP043953.1'\n",
       "\\item 'Alcaligenes\\_faecalis\\_NZ\\_CP013119.1'\n",
       "\\item 'Aspergillus\\_oryzae\\_NC\\_036435.1'\n",
       "\\item 'Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1'\n",
       "\\item 'Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1'\n",
       "\\item 'Cutibacterium\\_acnes\\_NC\\_021085.1'\n",
       "\\item 'Fannyhessea\\_vaginae\\_NZ\\_UFSV01000001.1'\n",
       "\\item 'Fusobacterium\\_nucleatum\\_NZ\\_LN831027.1'\n",
       "\\item 'Fusobacterium\\_pseudoperiodonticum\\_NZ\\_CP024702.1'\n",
       "\\item 'Gardnerella\\_vaginalis\\_NZ\\_PKJK01000001.1'\n",
       "\\item 'Haemophilus\\_parainfluenzae\\_NZ\\_GL872339.1'\n",
       "\\item 'Hepatitis\\_B\\_virus\\_genB\\_JX661488.1'\n",
       "\\item 'Human\\_betaherpesvirus\\_6A\\_NC\\_001664.4'\n",
       "\\item 'Human\\_betaherpesvirus\\_6B\\_NC\\_000898.1'\n",
       "\\item 'Human\\_mastadenovirus\\_C\\_NC\\_001405.1'\n",
       "\\item 'Lactobacillus\\_crispatus\\_NZ\\_CP039266.1'\n",
       "\\item 'Malassezia\\_restricta\\_NC\\_040193.1'\n",
       "\\item 'Moraxella\\_osloensis\\_NZ\\_CP014234.1'\n",
       "\\item 'Neisseria\\_flavescens\\_NZ\\_CP039886.1'\n",
       "\\item 'Neisseria\\_mucosa\\_NZ\\_UGRT01000001.1'\n",
       "\\item 'Neisseria\\_subflava\\_NZ\\_CP039887.1'\n",
       "\\item 'Prevotella\\_melaninogenica\\_NC\\_014370.1'\n",
       "\\item 'Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1'\n",
       "\\item 'Pseudomonas\\_mendocina\\_NZ\\_CP013124.1'\n",
       "\\item 'Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1'\n",
       "\\item 'Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1'\n",
       "\\item 'Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1'\n",
       "\\item 'Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1'\n",
       "\\item 'Torque\\_teno\\_virus\\_6\\_NC\\_014094.1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Achromobacter_xylosoxidans_NZ_CP043820.1'\n",
       "2. 'Acinetobacter_baumannii_NZ_CP043953.1'\n",
       "3. 'Alcaligenes_faecalis_NZ_CP013119.1'\n",
       "4. 'Aspergillus_oryzae_NC_036435.1'\n",
       "5. 'Corynebacterium_segmentosum_NZ_LR134408.1'\n",
       "6. 'Cupriavidus_metallidurans_NZ_FYAX01000013.1'\n",
       "7. 'Cutibacterium_acnes_NC_021085.1'\n",
       "8. 'Fannyhessea_vaginae_NZ_UFSV01000001.1'\n",
       "9. 'Fusobacterium_nucleatum_NZ_LN831027.1'\n",
       "10. 'Fusobacterium_pseudoperiodonticum_NZ_CP024702.1'\n",
       "11. 'Gardnerella_vaginalis_NZ_PKJK01000001.1'\n",
       "12. 'Haemophilus_parainfluenzae_NZ_GL872339.1'\n",
       "13. 'Hepatitis_B_virus_genB_JX661488.1'\n",
       "14. 'Human_betaherpesvirus_6A_NC_001664.4'\n",
       "15. 'Human_betaherpesvirus_6B_NC_000898.1'\n",
       "16. 'Human_mastadenovirus_C_NC_001405.1'\n",
       "17. 'Lactobacillus_crispatus_NZ_CP039266.1'\n",
       "18. 'Malassezia_restricta_NC_040193.1'\n",
       "19. 'Moraxella_osloensis_NZ_CP014234.1'\n",
       "20. 'Neisseria_flavescens_NZ_CP039886.1'\n",
       "21. 'Neisseria_mucosa_NZ_UGRT01000001.1'\n",
       "22. 'Neisseria_subflava_NZ_CP039887.1'\n",
       "23. 'Prevotella_melaninogenica_NC_014370.1'\n",
       "24. 'Prevotella_oral_taxon_299_NC_022124.1'\n",
       "25. 'Pseudomonas_mendocina_NZ_CP013124.1'\n",
       "26. 'Rickettsia_Tillamook_23_NZ_CP060138.1'\n",
       "27. 'Staphylococcus_cohnii_NZ_UHDA01000001.1'\n",
       "28. 'Staphylococcus_epidermidis_NZ_CP035288.1'\n",
       "29. 'Staphylococcus_haemolyticus_NZ_CP013911.1'\n",
       "30. 'Torque_teno_virus_6_NC_014094.1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Achromobacter_xylosoxidans_NZ_CP043820.1\"       \n",
       " [2] \"Acinetobacter_baumannii_NZ_CP043953.1\"          \n",
       " [3] \"Alcaligenes_faecalis_NZ_CP013119.1\"             \n",
       " [4] \"Aspergillus_oryzae_NC_036435.1\"                 \n",
       " [5] \"Corynebacterium_segmentosum_NZ_LR134408.1\"      \n",
       " [6] \"Cupriavidus_metallidurans_NZ_FYAX01000013.1\"    \n",
       " [7] \"Cutibacterium_acnes_NC_021085.1\"                \n",
       " [8] \"Fannyhessea_vaginae_NZ_UFSV01000001.1\"          \n",
       " [9] \"Fusobacterium_nucleatum_NZ_LN831027.1\"          \n",
       "[10] \"Fusobacterium_pseudoperiodonticum_NZ_CP024702.1\"\n",
       "[11] \"Gardnerella_vaginalis_NZ_PKJK01000001.1\"        \n",
       "[12] \"Haemophilus_parainfluenzae_NZ_GL872339.1\"       \n",
       "[13] \"Hepatitis_B_virus_genB_JX661488.1\"              \n",
       "[14] \"Human_betaherpesvirus_6A_NC_001664.4\"           \n",
       "[15] \"Human_betaherpesvirus_6B_NC_000898.1\"           \n",
       "[16] \"Human_mastadenovirus_C_NC_001405.1\"             \n",
       "[17] \"Lactobacillus_crispatus_NZ_CP039266.1\"          \n",
       "[18] \"Malassezia_restricta_NC_040193.1\"               \n",
       "[19] \"Moraxella_osloensis_NZ_CP014234.1\"              \n",
       "[20] \"Neisseria_flavescens_NZ_CP039886.1\"             \n",
       "[21] \"Neisseria_mucosa_NZ_UGRT01000001.1\"             \n",
       "[22] \"Neisseria_subflava_NZ_CP039887.1\"               \n",
       "[23] \"Prevotella_melaninogenica_NC_014370.1\"          \n",
       "[24] \"Prevotella_oral_taxon_299_NC_022124.1\"          \n",
       "[25] \"Pseudomonas_mendocina_NZ_CP013124.1\"            \n",
       "[26] \"Rickettsia_Tillamook_23_NZ_CP060138.1\"          \n",
       "[27] \"Staphylococcus_cohnii_NZ_UHDA01000001.1\"        \n",
       "[28] \"Staphylococcus_epidermidis_NZ_CP035288.1\"       \n",
       "[29] \"Staphylococcus_haemolyticus_NZ_CP013911.1\"      \n",
       "[30] \"Torque_teno_virus_6_NC_014094.1\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_path <- \"results/irep_analysis/raw_output/bam_files\"\n",
    "ref_path <- \"data/irep_data/genome_references\"\n",
    "prefixes <- list.files(ref_path, \".fasta\", full.names = F)\n",
    "prefixes <- gsub(\".fasta\", \"\", prefixes)\n",
    "prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d49d2e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "morsels <- foreach (i = seq(length(prefixes))) %dopar% {\n",
    "    prefix <- prefixes[i]\n",
    "    bams <- list.files(str_glue(\"{bam_path}/{prefix}\"))\n",
    "    bams <- bams[!grepl(\".bai\", bams)]\n",
    "\n",
    "    crumbs <- foreach (j = seq(length(bams))) %do% {\n",
    "        bam <- bams[j]\n",
    "        id <- str_split(bam, \"_\")[[1]][1]\n",
    "        ref <- readDNAStringSet(str_glue(\"{ref_path}/{prefixes[i]}.fasta\"))\n",
    "        ref_length <- width(ref)\n",
    "        params <- PileupParam(distinguish_strands = F, distinguish_nucleotides = F)\n",
    "        pile_df <- pileup(str_glue(\"{bam_path}/{prefix}/{bam}\"), pileupParam = params)\n",
    "        \n",
    "        n_covered1 <- pile_df %>%\n",
    "            filter(count >= 1) %>%\n",
    "            nrow()\n",
    "        \n",
    "        n_covered5 <- pile_df %>%\n",
    "            filter(count >= 5) %>%\n",
    "            nrow()\n",
    "\n",
    "        perc_covered1 <- n_covered1 / ref_length * 100\n",
    "        perc_covered5 <- n_covered5 / ref_length * 100\n",
    "        \n",
    "        taxon <- str_split(prefix, \"_\")[[1]]\n",
    "            read_count <- species_df %>% \n",
    "                filter(npm_research_id == id) %>%\n",
    "                select(contains(taxon[1])) %>%\n",
    "                select(contains(taxon[2])) %>%\n",
    "                select(!contains(\"monkey\"))\n",
    "\n",
    "        if (ncol(read_count) > 1) {\n",
    "            read_count <- read_count %>% \n",
    "                select(contains(taxon[3]))\n",
    "            if (ncol(read_count) > 1) {\n",
    "            read_count <- read_count %>% \n",
    "                select(contains(paste0(\" \", taxon[4])))\n",
    "            }\n",
    "        }\n",
    "        \n",
    "        if (all(dim(read_count) == 1)) {\n",
    "            read_count <- deframe(read_count)\n",
    "        } else {\n",
    "            read_count <- NA\n",
    "        }\n",
    "        \n",
    "        return(tibble(prefix = prefix, \n",
    "                      npm_research_id = id,\n",
    "                      pairs_assigned = read_count,\n",
    "                      perc_covered1 = perc_covered1,\n",
    "                      perc_covered5 = perc_covered5))\n",
    "    }\n",
    "    \n",
    "    return(bind_rows(crumbs))\n",
    "}\n",
    "\n",
    "cov_df <- bind_rows(morsels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bfef9750",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Hepatitis B virus</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 1\n",
       "\\begin{tabular}{l}\n",
       " Hepatitis B virus\\\\\n",
       " <int>\\\\\n",
       "\\hline\n",
       "\t 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 1\n",
       "\n",
       "| Hepatitis B virus &lt;int&gt; |\n",
       "|---|\n",
       "| 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  Hepatitis B virus\n",
       "1 0                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# prefixes\n",
    "# prefix <- \"Hepatitis_B_virus_genB_JX661488.1\"    \n",
    "# id <- \"WHB4594\"\n",
    "# taxon <- str_split(prefix, \"_\")[[1]]\n",
    "#     read_count <- species_df %>% \n",
    "#         filter(npm_research_id == id) %>%\n",
    "#         select(contains(taxon[1])) %>%\n",
    "#         select(contains(taxon[2])) %>%\n",
    "#         select(!contains(\"monkey\"))\n",
    "    \n",
    "# #     if (ncol(read_count) > 1) {\n",
    "# #         read_count <- read_count %>% \n",
    "# #             select(contains(taxon[3]))\n",
    "# #         if (ncol(read_count) > 1) {\n",
    "# #         read_count <- read_count %>% \n",
    "# #             select(contains(paste0(\" \", taxon[4])))\n",
    "# #         }\n",
    "# #     }\n",
    "\n",
    "# #     if (all(dim(read_count) == 1)) {\n",
    "# #         read_count <- deframe(read_count)\n",
    "# #     } else {\n",
    "# #         read_count <- NA\n",
    "# #     }\n",
    "\n",
    "# read_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56cba5fc",
   "metadata": {},
   "source": [
    "### Merged results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88d455ed",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mapping_df <- fread(\"results/irep_analysis/raw_output/mapping_read_counts.csv\")\n",
    "mapping_df %>% filter(grepl(\"Prevo\", prefix))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85fca457",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "merged_df <- cov_df %>%\n",
    "    left_join(mapping_df) %>%\n",
    "    left_join(irep_df) %>%\n",
    "    relocate(pairs_mapped, .after = 3) %>%\n",
    "    mutate(pairs_mapped = pairs_mapped / 2)\n",
    "\n",
    "merged_df\n",
    "\n",
    "fwrite(merged_df, \"results/irep_analysis/coverage_irep_results.raw.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd23b170",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df %>% filter(grepl(\"oral\", prefix))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c39a120",
   "metadata": {},
   "outputs": [],
   "source": [
    "ori_df <- merged_df %>%\n",
    "    select(prefix, ORI, TER) %>%\n",
    "    filter(ORI != \"0\" & TER != \"0\") %>%\n",
    "    distinct()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a62c820",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "summarised_df <- merged_df %>%\n",
    "    group_by(prefix) %>%\n",
    "    summarise(mean_perc_covered1 = mean(perc_covered1),\n",
    "              mean_perc_covered5 = mean(perc_covered5),\n",
    "              min_perc_covered1 = min(perc_covered1),\n",
    "              max_perc_covered1 = max(perc_covered1),\n",
    "              max_bPTR = max(bPTR, na.rm = T)) %>%\n",
    "    mutate(max_bPTR = ifelse(max_bPTR == -Inf, NA, max_bPTR)) %>%\n",
    "    left_join(ori_df) %>%\n",
    "    separate(prefix, into = c(\"genus\", \"species\", \"suffix1\", \"suffix2\"), sep = \"_\", remove = F) %>%\n",
    "    mutate(taxa = ifelse(grepl(\"Torque\", prefix), \n",
    "                         paste(genus, species, suffix1, suffix2), \n",
    "                         paste(genus, species, suffix1)), .before = 1) %>%\n",
    "    mutate(taxa = gsub(\" NZ| NC\", \"\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Microbacterium\", \"Microbacterium sp.\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Prevotella oral taxon\", \"Prevotella sp. oral taxon 299\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Rickettsia\", \"Rickettsia sp.\", taxa)) %>%\n",
    "    select(-prefix, -genus, -species, -suffix1, -suffix2) %>%\n",
    "    arrange(desc(mean_perc_covered1))\n",
    "\n",
    "\n",
    "fwrite(summarised_df, \"results/irep_analysis/coverage_irep_results.parsed.csv\")\n",
    "\n",
    "summarised_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
