{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d62ad397",
   "metadata": {},
   "source": [
    "# QC for pre-processing steps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3243da47",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message:\n",
      "“Your system is mis-configured: ‘/etc/localtime’ is not a symlink”\n",
      "Warning message:\n",
      "“It is strongly recommended to set envionment variable TZ to ‘Asia/Singapore’ (or equivalent)”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.4     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.2     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Loading required package: ShortRead\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘XVector’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    compact\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: Rsamtools\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomicAlignments\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘GenomicAlignments’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    last\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ShortRead’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    tables\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    id\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    compose\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tibble’:\n",
      "\n",
      "    view\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/home/projects/14001280/PROJECTS/blood_microbiome/\")\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(ShortRead)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b177d8ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta <- fread(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "13b4090d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 71</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>npm_research_id</th><th scope=col>multiplex_pool_id</th><th scope=col>supplier_id</th><th scope=col>gis_internal_sample_id</th><th scope=col>site_supplying_sample</th><th scope=col>year_of_birth</th><th scope=col>supplied_gender</th><th scope=col>self_reported_ethnicity</th><th scope=col>extraction_kit</th><th scope=col>date_of_dna_extraction</th><th scope=col>⋯</th><th scope=col>supplied_and_computed_gender_match</th><th scope=col>sop_agreement_for_coverage__14/_28_for_15x/30x</th><th scope=col>sample_life_cycle</th><th scope=col>current</th><th scope=col>source_cohort</th><th scope=col>original_supplier_id</th><th scope=col>duplicate_info</th><th scope=col>duplicate</th><th scope=col>duplicate_pair</th><th scope=col>removal_requested_by_supplier</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>WHB10000</td><td>MUX9693</td><td>00358</td><td>HELIOS_00358_1</td><td>HELIOS</td><td>1982</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019</td><td>⋯</td><td>Y</td><td>Y</td><td>HELIOS VCF</td><td>Y</td><td>HELIOS</td><td>00358</td><td>Unknown</td><td>PASS</td><td>Unknown</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB10001</td><td>MUX9693</td><td>00359</td><td>HELIOS_00359_1</td><td>HELIOS</td><td>1979</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019</td><td>⋯</td><td>Y</td><td>Y</td><td>HELIOS VCF</td><td>Y</td><td>HELIOS</td><td>00359</td><td>Unknown</td><td>PASS</td><td>Unknown</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB10002</td><td>MUX9695</td><td>00360</td><td>HELIOS_00360_1</td><td>HELIOS</td><td>1970</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019</td><td>⋯</td><td>Y</td><td>N</td><td>HELIOS VCF</td><td>Y</td><td>HELIOS</td><td>00360</td><td>Unknown</td><td>PASS</td><td>Unknown</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB10003</td><td>MUX9694</td><td>00361</td><td>HELIOS_00361_1</td><td>HELIOS</td><td>1969</td><td>M</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019</td><td>⋯</td><td>Y</td><td>Y</td><td>HELIOS VCF</td><td>Y</td><td>HELIOS</td><td>00361</td><td>Unknown</td><td>PASS</td><td>Unknown</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB10005</td><td>MUX9694</td><td>00363</td><td>HELIOS_00363_1</td><td>HELIOS</td><td>1959</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019</td><td>⋯</td><td>Y</td><td>N</td><td>HELIOS VCF</td><td>Y</td><td>HELIOS</td><td>00363</td><td>Unknown</td><td>PASS</td><td>Unknown</td><td>Unknown</td></tr>\n",
       "\t<tr><td>WHB10006</td><td>MUX9694</td><td>00364</td><td>HELIOS_00364_1</td><td>HELIOS</td><td>1955</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019</td><td>⋯</td><td>Y</td><td>Y</td><td>HELIOS VCF</td><td>Y</td><td>HELIOS</td><td>00364</td><td>Unknown</td><td>PASS</td><td>Unknown</td><td>Unknown</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 71\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " npm\\_research\\_id & multiplex\\_pool\\_id & supplier\\_id & gis\\_internal\\_sample\\_id & site\\_supplying\\_sample & year\\_of\\_birth & supplied\\_gender & self\\_reported\\_ethnicity & extraction\\_kit & date\\_of\\_dna\\_extraction & ⋯ & supplied\\_and\\_computed\\_gender\\_match & sop\\_agreement\\_for\\_coverage\\_\\_14/\\_28\\_for\\_15x/30x & sample\\_life\\_cycle & current & source\\_cohort & original\\_supplier\\_id & duplicate\\_info & duplicate & duplicate\\_pair & removal\\_requested\\_by\\_supplier\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t WHB10000 & MUX9693 & 00358 & HELIOS\\_00358\\_1 & HELIOS & 1982 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019 & ⋯ & Y & Y & HELIOS VCF & Y & HELIOS & 00358 & Unknown & PASS & Unknown & Unknown\\\\\n",
       "\t WHB10001 & MUX9693 & 00359 & HELIOS\\_00359\\_1 & HELIOS & 1979 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019 & ⋯ & Y & Y & HELIOS VCF & Y & HELIOS & 00359 & Unknown & PASS & Unknown & Unknown\\\\\n",
       "\t WHB10002 & MUX9695 & 00360 & HELIOS\\_00360\\_1 & HELIOS & 1970 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019 & ⋯ & Y & N & HELIOS VCF & Y & HELIOS & 00360 & Unknown & PASS & Unknown & Unknown\\\\\n",
       "\t WHB10003 & MUX9694 & 00361 & HELIOS\\_00361\\_1 & HELIOS & 1969 & M & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019 & ⋯ & Y & Y & HELIOS VCF & Y & HELIOS & 00361 & Unknown & PASS & Unknown & Unknown\\\\\n",
       "\t WHB10005 & MUX9694 & 00363 & HELIOS\\_00363\\_1 & HELIOS & 1959 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019 & ⋯ & Y & N & HELIOS VCF & Y & HELIOS & 00363 & Unknown & PASS & Unknown & Unknown\\\\\n",
       "\t WHB10006 & MUX9694 & 00364 & HELIOS\\_00364\\_1 & HELIOS & 1955 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019 & ⋯ & Y & Y & HELIOS VCF & Y & HELIOS & 00364 & Unknown & PASS & Unknown & Unknown\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 71\n",
       "\n",
       "| npm_research_id &lt;chr&gt; | multiplex_pool_id &lt;chr&gt; | supplier_id &lt;chr&gt; | gis_internal_sample_id &lt;chr&gt; | site_supplying_sample &lt;chr&gt; | year_of_birth &lt;int&gt; | supplied_gender &lt;chr&gt; | self_reported_ethnicity &lt;chr&gt; | extraction_kit &lt;chr&gt; | date_of_dna_extraction &lt;chr&gt; | ⋯ ⋯ | supplied_and_computed_gender_match &lt;chr&gt; | sop_agreement_for_coverage__14/_28_for_15x/30x &lt;chr&gt; | sample_life_cycle &lt;chr&gt; | current &lt;chr&gt; | source_cohort &lt;chr&gt; | original_supplier_id &lt;chr&gt; | duplicate_info &lt;chr&gt; | duplicate &lt;chr&gt; | duplicate_pair &lt;chr&gt; | removal_requested_by_supplier &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| WHB10000 | MUX9693 | 00358 | HELIOS_00358_1 | HELIOS | 1982 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019 | ⋯ | Y | Y | HELIOS VCF | Y | HELIOS | 00358 | Unknown | PASS | Unknown | Unknown |\n",
       "| WHB10001 | MUX9693 | 00359 | HELIOS_00359_1 | HELIOS | 1979 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019 | ⋯ | Y | Y | HELIOS VCF | Y | HELIOS | 00359 | Unknown | PASS | Unknown | Unknown |\n",
       "| WHB10002 | MUX9695 | 00360 | HELIOS_00360_1 | HELIOS | 1970 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019 | ⋯ | Y | N | HELIOS VCF | Y | HELIOS | 00360 | Unknown | PASS | Unknown | Unknown |\n",
       "| WHB10003 | MUX9694 | 00361 | HELIOS_00361_1 | HELIOS | 1969 | M | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019 | ⋯ | Y | Y | HELIOS VCF | Y | HELIOS | 00361 | Unknown | PASS | Unknown | Unknown |\n",
       "| WHB10005 | MUX9694 | 00363 | HELIOS_00363_1 | HELIOS | 1959 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019 | ⋯ | Y | N | HELIOS VCF | Y | HELIOS | 00363 | Unknown | PASS | Unknown | Unknown |\n",
       "| WHB10006 | MUX9694 | 00364 | HELIOS_00364_1 | HELIOS | 1955 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019 | ⋯ | Y | Y | HELIOS VCF | Y | HELIOS | 00364 | Unknown | PASS | Unknown | Unknown |\n",
       "\n"
      ],
      "text/plain": [
       "  npm_research_id multiplex_pool_id supplier_id gis_internal_sample_id\n",
       "1 WHB10000        MUX9693           00358       HELIOS_00358_1        \n",
       "2 WHB10001        MUX9693           00359       HELIOS_00359_1        \n",
       "3 WHB10002        MUX9695           00360       HELIOS_00360_1        \n",
       "4 WHB10003        MUX9694           00361       HELIOS_00361_1        \n",
       "5 WHB10005        MUX9694           00363       HELIOS_00363_1        \n",
       "6 WHB10006        MUX9694           00364       HELIOS_00364_1        \n",
       "  site_supplying_sample year_of_birth supplied_gender self_reported_ethnicity\n",
       "1 HELIOS                1982          F               Chinese                \n",
       "2 HELIOS                1979          F               Chinese                \n",
       "3 HELIOS                1970          F               Chinese                \n",
       "4 HELIOS                1969          M               Chinese                \n",
       "5 HELIOS                1959          F               Chinese                \n",
       "6 HELIOS                1955          F               Chinese                \n",
       "  extraction_kit                                 date_of_dna_extraction ⋯\n",
       "1 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "2 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "3 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "4 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "5 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "6 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "  supplied_and_computed_gender_match\n",
       "1 Y                                 \n",
       "2 Y                                 \n",
       "3 Y                                 \n",
       "4 Y                                 \n",
       "5 Y                                 \n",
       "6 Y                                 \n",
       "  sop_agreement_for_coverage__14/_28_for_15x/30x sample_life_cycle current\n",
       "1 Y                                              HELIOS VCF        Y      \n",
       "2 Y                                              HELIOS VCF        Y      \n",
       "3 N                                              HELIOS VCF        Y      \n",
       "4 Y                                              HELIOS VCF        Y      \n",
       "5 N                                              HELIOS VCF        Y      \n",
       "6 Y                                              HELIOS VCF        Y      \n",
       "  source_cohort original_supplier_id duplicate_info duplicate duplicate_pair\n",
       "1 HELIOS        00358                Unknown        PASS      Unknown       \n",
       "2 HELIOS        00359                Unknown        PASS      Unknown       \n",
       "3 HELIOS        00360                Unknown        PASS      Unknown       \n",
       "4 HELIOS        00361                Unknown        PASS      Unknown       \n",
       "5 HELIOS        00363                Unknown        PASS      Unknown       \n",
       "6 HELIOS        00364                Unknown        PASS      Unknown       \n",
       "  removal_requested_by_supplier\n",
       "1 Unknown                      \n",
       "2 Unknown                      \n",
       "3 Unknown                      \n",
       "4 Unknown                      \n",
       "5 Unknown                      \n",
       "6 Unknown                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(meta)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4f5e135",
   "metadata": {},
   "source": [
    "### Get intermediate file directories for all pre-processing steps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "261bb35e",
   "metadata": {},
   "outputs": [],
   "source": [
    "steps <- list.dirs(\"data/temp_files_100\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3054a95",
   "metadata": {},
   "source": [
    "### Iterate through preprocessing step directories and through fastq/bam/files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fc2960f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df <- tibble()\n",
    "\n",
    "for (step in steps) {\n",
    "#     step <- steps[3]\n",
    "    if (grepl(\"_fastq|_bam\", step)) {\n",
    "        step_name <- strsplit(step, \"/\")\n",
    "        step_name <- unlist(step_name)[3]\n",
    "        file_names <- list.files(step)\n",
    "        print(paste(\"pre-processing step:\", step))\n",
    "\n",
    "        for (file in file_names) {\n",
    "    #         file <- file_names[40]\n",
    "            file_path <- paste0(step, \"/\", file)\n",
    "\n",
    "            # Get file size\n",
    "            file_size <- file.size(file_path)\n",
    "            npm_research_id <- strsplit(file, \"\\\\.\")\n",
    "            npm_research_id <- unlist(npm_research_id)[2]\n",
    "\n",
    "            # Get no. of reads\n",
    "            if (grepl(\".fastq\", file_path)) {\n",
    "                n_reads <- countLines(file_path) / 4\n",
    "            } else {\n",
    "                n_reads <- NA\n",
    "            }\n",
    "\n",
    "            morsel <- tibble(step_name = step_name, \n",
    "                             npm_research_id = npm_research_id,\n",
    "                             file_size = file_size, \n",
    "                             file_name = file, \n",
    "                             n_reads = n_reads)\n",
    "\n",
    "            df <- df %>% bind_rows(morsel)\n",
    "\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b39fb56",
   "metadata": {},
   "source": [
    "### Files that failed CRAM to fastq extraction\n",
    "These files do not have any reads/mates that are unmapped."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "959ec1c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "for (i in c(\"01_bam\", \"01_fastq\", \"02_fastq\", \"03_fastq\", \"04_fastq\")) {\n",
    "    n <- df %>%\n",
    "        filter(!is.na(step_name)) %>% # Omit entries of directories\n",
    "        filter(step_name == i) %>%\n",
    "        filter(file_size != 0) %>%\n",
    "        nrow()\n",
    "    print(paste0(i, \": \", n))\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a16adea",
   "metadata": {},
   "source": [
    "### Plot read count distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "406adc99",
   "metadata": {},
   "outputs": [],
   "source": [
    "df %>%\n",
    "    filter(grepl(\"01|02|03|04\", step_name),\n",
    "           n_reads != 0) %>%\n",
    "    mutate(step_name = case_when(step_name == \"01_fastq\" ~ \"01_raw\",\n",
    "                                 step_name == \"02_fastq\" ~ \"02_trim\",\n",
    "                                 step_name == \"03_fastq\" ~ \"03_filter\",\n",
    "                                 step_name == \"04_fastq\" ~ \"04_low_complexity\"),\n",
    "           n_reads = log(n_reads, base = 10)) %>%\n",
    "    filter(!grepl(\"04\", step_name)) %>%\n",
    "    ggplot(aes(x = n_reads, fill = step_name)) +\n",
    "        geom_density(alpha = 0.4) +\n",
    "        scale_fill_discrete() +\n",
    "        labs(x = \"lg(Read Count)\", y = \"Density\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
