{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5320e4cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message:\n",
      "“Your system is mis-configured: ‘/etc/localtime’ is not a symlink”\n",
      "Warning message:\n",
      "“It is strongly recommended to set envionment variable TZ to ‘Asia/Singapore’ (or equivalent)”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.4     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.2     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/home/projects/14001280/PROJECTS/blood_microbiome/data\")\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63fbef82",
   "metadata": {},
   "source": [
    "### Load metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d66f55ea",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in fread(\"SG10K_Health_metadata.n10714.16March2021.txt\"):\n",
      "“Detected 75 column names but the data has 71 columns. Filling rows automatically. Set fill=TRUE explicitly to avoid this warning.”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 75</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>NPM Research ID</th><th scope=col>Multiplex Pool ID</th><th scope=col>Supplier ID</th><th scope=col>GIS Internal Sample ID</th><th scope=col>Site Supplying Sample</th><th scope=col>Year Of Birth</th><th scope=col>Supplied Gender</th><th scope=col>Self Reported Ethnicity</th><th scope=col>Extraction Kit</th><th scope=col>Date Of DNA Extraction</th><th scope=col>⋯</th><th scope=col>Supplied And Computed Ethinicity Match</th><th scope=col>SOP Agreement For Coverage &lt;14/&lt;28 For 15x/30x</th><th scope=col>Sample Life Cycle</th><th scope=col>Current</th><th scope=col>Source Cohort</th><th scope=col>Original_supplier_ID</th><th scope=col>duplicate_info</th><th scope=col>duplicate</th><th scope=col>duplicate_pair</th><th scope=col>removal-requested-by-supplier</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>FAILED00365</td><td></td><td>00365</td><td>HELIOS_00365_1</td><td>HELIOS</td><td>1954</td><td>M</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>6/6/2019 </td><td>⋯</td><td>NA</td><td></td><td>HELIOS REG</td><td>N</td><td>HELIOS</td><td>00365</td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>FAILED01615</td><td></td><td>01615</td><td>HELIOS_01615_2</td><td>HELIOS</td><td>1956</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>20/1/2019</td><td>⋯</td><td>NA</td><td></td><td>HELIOS REG</td><td>N</td><td>HELIOS</td><td>01615</td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>FAILED01688</td><td></td><td>01688</td><td>HELIOS_01688_2</td><td>HELIOS</td><td>1952</td><td>F</td><td>Others </td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>20/1/2019</td><td>⋯</td><td>NA</td><td></td><td>HELIOS REG</td><td>N</td><td>HELIOS</td><td>01688</td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>FAILED01750</td><td></td><td>01750</td><td>HELIOS_01750_1</td><td>HELIOS</td><td>1957</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>13/2/2019</td><td>⋯</td><td>NA</td><td></td><td>HELIOS REG</td><td>N</td><td>HELIOS</td><td>01750</td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>FAILED01799</td><td></td><td>01799</td><td>HELIOS_01799_1</td><td>HELIOS</td><td>1948</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>13/2/2019</td><td>⋯</td><td>NA</td><td></td><td>HELIOS REG</td><td>N</td><td>HELIOS</td><td>01799</td><td></td><td></td><td></td><td></td></tr>\n",
       "\t<tr><td>FAILED02755</td><td></td><td>02755</td><td>HELIOS_02755_1</td><td>HELIOS</td><td>1951</td><td>F</td><td>Chinese</td><td>abGENIX Whole Blood Genomic DNA Extraction Kit</td><td>30/5/2019</td><td>⋯</td><td>NA</td><td></td><td>HELIOS REG</td><td>N</td><td>HELIOS</td><td>02755</td><td></td><td></td><td></td><td></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 75\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " NPM Research ID & Multiplex Pool ID & Supplier ID & GIS Internal Sample ID & Site Supplying Sample & Year Of Birth & Supplied Gender & Self Reported Ethnicity & Extraction Kit & Date Of DNA Extraction & ⋯ & Supplied And Computed Ethinicity Match & SOP Agreement For Coverage <14/<28 For 15x/30x & Sample Life Cycle & Current & Source Cohort & Original\\_supplier\\_ID & duplicate\\_info & duplicate & duplicate\\_pair & removal-requested-by-supplier\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & ⋯ & <lgl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t FAILED00365 &  & 00365 & HELIOS\\_00365\\_1 & HELIOS & 1954 & M & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 6/6/2019  & ⋯ & NA &  & HELIOS REG & N & HELIOS & 00365 &  &  &  & \\\\\n",
       "\t FAILED01615 &  & 01615 & HELIOS\\_01615\\_2 & HELIOS & 1956 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 20/1/2019 & ⋯ & NA &  & HELIOS REG & N & HELIOS & 01615 &  &  &  & \\\\\n",
       "\t FAILED01688 &  & 01688 & HELIOS\\_01688\\_2 & HELIOS & 1952 & F & Others  & abGENIX Whole Blood Genomic DNA Extraction Kit & 20/1/2019 & ⋯ & NA &  & HELIOS REG & N & HELIOS & 01688 &  &  &  & \\\\\n",
       "\t FAILED01750 &  & 01750 & HELIOS\\_01750\\_1 & HELIOS & 1957 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 13/2/2019 & ⋯ & NA &  & HELIOS REG & N & HELIOS & 01750 &  &  &  & \\\\\n",
       "\t FAILED01799 &  & 01799 & HELIOS\\_01799\\_1 & HELIOS & 1948 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 13/2/2019 & ⋯ & NA &  & HELIOS REG & N & HELIOS & 01799 &  &  &  & \\\\\n",
       "\t FAILED02755 &  & 02755 & HELIOS\\_02755\\_1 & HELIOS & 1951 & F & Chinese & abGENIX Whole Blood Genomic DNA Extraction Kit & 30/5/2019 & ⋯ & NA &  & HELIOS REG & N & HELIOS & 02755 &  &  &  & \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 75\n",
       "\n",
       "| NPM Research ID &lt;chr&gt; | Multiplex Pool ID &lt;chr&gt; | Supplier ID &lt;chr&gt; | GIS Internal Sample ID &lt;chr&gt; | Site Supplying Sample &lt;chr&gt; | Year Of Birth &lt;int&gt; | Supplied Gender &lt;chr&gt; | Self Reported Ethnicity &lt;chr&gt; | Extraction Kit &lt;chr&gt; | Date Of DNA Extraction &lt;chr&gt; | ⋯ ⋯ | Supplied And Computed Ethinicity Match &lt;lgl&gt; | SOP Agreement For Coverage &lt;14/&lt;28 For 15x/30x &lt;chr&gt; | Sample Life Cycle &lt;chr&gt; | Current &lt;chr&gt; | Source Cohort &lt;chr&gt; | Original_supplier_ID &lt;chr&gt; | duplicate_info &lt;chr&gt; | duplicate &lt;chr&gt; | duplicate_pair &lt;chr&gt; | removal-requested-by-supplier &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| FAILED00365 | <!----> | 00365 | HELIOS_00365_1 | HELIOS | 1954 | M | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 6/6/2019  | ⋯ | NA | <!----> | HELIOS REG | N | HELIOS | 00365 | <!----> | <!----> | <!----> | <!----> |\n",
       "| FAILED01615 | <!----> | 01615 | HELIOS_01615_2 | HELIOS | 1956 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 20/1/2019 | ⋯ | NA | <!----> | HELIOS REG | N | HELIOS | 01615 | <!----> | <!----> | <!----> | <!----> |\n",
       "| FAILED01688 | <!----> | 01688 | HELIOS_01688_2 | HELIOS | 1952 | F | Others  | abGENIX Whole Blood Genomic DNA Extraction Kit | 20/1/2019 | ⋯ | NA | <!----> | HELIOS REG | N | HELIOS | 01688 | <!----> | <!----> | <!----> | <!----> |\n",
       "| FAILED01750 | <!----> | 01750 | HELIOS_01750_1 | HELIOS | 1957 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 13/2/2019 | ⋯ | NA | <!----> | HELIOS REG | N | HELIOS | 01750 | <!----> | <!----> | <!----> | <!----> |\n",
       "| FAILED01799 | <!----> | 01799 | HELIOS_01799_1 | HELIOS | 1948 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 13/2/2019 | ⋯ | NA | <!----> | HELIOS REG | N | HELIOS | 01799 | <!----> | <!----> | <!----> | <!----> |\n",
       "| FAILED02755 | <!----> | 02755 | HELIOS_02755_1 | HELIOS | 1951 | F | Chinese | abGENIX Whole Blood Genomic DNA Extraction Kit | 30/5/2019 | ⋯ | NA | <!----> | HELIOS REG | N | HELIOS | 02755 | <!----> | <!----> | <!----> | <!----> |\n",
       "\n"
      ],
      "text/plain": [
       "  NPM Research ID Multiplex Pool ID Supplier ID GIS Internal Sample ID\n",
       "1 FAILED00365                       00365       HELIOS_00365_1        \n",
       "2 FAILED01615                       01615       HELIOS_01615_2        \n",
       "3 FAILED01688                       01688       HELIOS_01688_2        \n",
       "4 FAILED01750                       01750       HELIOS_01750_1        \n",
       "5 FAILED01799                       01799       HELIOS_01799_1        \n",
       "6 FAILED02755                       02755       HELIOS_02755_1        \n",
       "  Site Supplying Sample Year Of Birth Supplied Gender Self Reported Ethnicity\n",
       "1 HELIOS                1954          M               Chinese                \n",
       "2 HELIOS                1956          F               Chinese                \n",
       "3 HELIOS                1952          F               Others                 \n",
       "4 HELIOS                1957          F               Chinese                \n",
       "5 HELIOS                1948          F               Chinese                \n",
       "6 HELIOS                1951          F               Chinese                \n",
       "  Extraction Kit                                 Date Of DNA Extraction ⋯\n",
       "1 abGENIX Whole Blood Genomic DNA Extraction Kit 6/6/2019               ⋯\n",
       "2 abGENIX Whole Blood Genomic DNA Extraction Kit 20/1/2019              ⋯\n",
       "3 abGENIX Whole Blood Genomic DNA Extraction Kit 20/1/2019              ⋯\n",
       "4 abGENIX Whole Blood Genomic DNA Extraction Kit 13/2/2019              ⋯\n",
       "5 abGENIX Whole Blood Genomic DNA Extraction Kit 13/2/2019              ⋯\n",
       "6 abGENIX Whole Blood Genomic DNA Extraction Kit 30/5/2019              ⋯\n",
       "  Supplied And Computed Ethinicity Match\n",
       "1 NA                                    \n",
       "2 NA                                    \n",
       "3 NA                                    \n",
       "4 NA                                    \n",
       "5 NA                                    \n",
       "6 NA                                    \n",
       "  SOP Agreement For Coverage <14/<28 For 15x/30x Sample Life Cycle Current\n",
       "1                                                HELIOS REG        N      \n",
       "2                                                HELIOS REG        N      \n",
       "3                                                HELIOS REG        N      \n",
       "4                                                HELIOS REG        N      \n",
       "5                                                HELIOS REG        N      \n",
       "6                                                HELIOS REG        N      \n",
       "  Source Cohort Original_supplier_ID duplicate_info duplicate duplicate_pair\n",
       "1 HELIOS        00365                                                       \n",
       "2 HELIOS        01615                                                       \n",
       "3 HELIOS        01688                                                       \n",
       "4 HELIOS        01750                                                       \n",
       "5 HELIOS        01799                                                       \n",
       "6 HELIOS        02755                                                       \n",
       "  removal-requested-by-supplier\n",
       "1                              \n",
       "2                              \n",
       "3                              \n",
       "4                              \n",
       "5                              \n",
       "6                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'0115-0043'</li><li>'0115-0044'</li><li>'0116-0004'</li><li>'0116-0006'</li><li>'0116-0022'</li><li>'0116-0032'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '0115-0043'\n",
       "\\item '0115-0044'\n",
       "\\item '0116-0004'\n",
       "\\item '0116-0006'\n",
       "\\item '0116-0022'\n",
       "\\item '0116-0032'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '0115-0043'\n",
       "2. '0115-0044'\n",
       "3. '0116-0004'\n",
       "4. '0116-0006'\n",
       "5. '0116-0022'\n",
       "6. '0116-0032'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"0115-0043\" \"0115-0044\" \"0116-0004\" \"0116-0006\" \"0116-0022\" \"0116-0032\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Raw metadata\n",
    "meta <- fread(\"SG10K_Health_metadata.n10714.16March2021.txt\")\n",
    "\n",
    "# List of r3 samples\n",
    "meta2 <- fread(\"SG10K_Health_r5.3.n9770.sample_ids.txt\", header = F)$V1\n",
    "\n",
    "head(meta)\n",
    "head(meta2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1076790d",
   "metadata": {},
   "source": [
    "### Parse column names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "16d53df0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'npm_research_id'</li><li>'multiplex_pool_id'</li><li>'supplier_id'</li><li>'gis_internal_sample_id'</li><li>'site_supplying_sample'</li><li>'year_of_birth'</li><li>'supplied_gender'</li><li>'self_reported_ethnicity'</li><li>'extraction_kit'</li><li>'date_of_dna_extraction'</li><li>'plate_position'</li><li>'plate_name'</li><li>'version_of_consent_form_signed'</li><li>'sequencing_depth'</li><li>'npm_research_id_created_by_username'</li><li>'npm_research_id_creation_date'</li><li>'comments_entered_when_npm_research_id_created'</li><li>'description_entered_when_npm_research_id_created'</li><li>'elm_project_id'</li><li>'elm_project_title'</li><li>'elm_project_pi'</li><li>'species_of_sample_sequenced'</li><li>'tehcnique_for_sequencing'</li><li>'tissue_type_sequenced'</li><li>'library_found_in_sequencing_run_id'</li><li>'library_found_in_passed_sequencing_run_id'</li><li>'vendor_sequencing_centre'</li><li>'dna_sample_passed_qc'</li><li>'library_prep_kit'</li><li>'run_id'</li><li>'instrument_id'</li><li>'instrument_serial_number'</li><li>'hiseq_xtm_sbs_kit_300_cycles_v2_box_1of_2_lot'</li><li>'hiseq_xtm_sbs_kit_300_cycles_v2_box_2_of_2_lot'</li><li>'hiseq_xtm_pe_cluster_kit_cbottm_v2_box_1_of_2_lot'</li><li>'hiseq_xtm_pe_cluster_kit_cbottm_v2_box_2_of_2_lot'</li><li>'hiseq_xtm_flow_cell_v25_lot'</li><li>'hiseq_xtm_flow_cell_v25_id'</li><li>'sequencebased_gender'</li><li>'cram_files_location'</li><li>'estimate_of_sequence_coverage'</li><li>'contamination_rate_of_chinese_genome'</li><li>'contamination_rate_of_indian_genome'</li><li>'contamination_rate_of_malaysian_genome'</li><li>'average_coverage_depth'</li><li>'fb_variants'</li><li>'cram_filesize_mb'</li><li>'error_rate'</li><li>'fbvcf_filesize_mb'</li><li>'gvcf_filesize_mb'</li><li>'insert_size_average'</li><li>'insert_size_standard_deviation'</li><li>'raw_total_sequences'</li><li>'reads_duplicated_'</li><li>'reads_properly_paired_'</li><li>'cohort_jvcf_data_capacity'</li><li>'fastq_file_size'</li><li>'jvcf_size'</li><li>'data_timestamp_for_fastq_file'</li><li>'sequencing_analysis_pipeline'</li><li>'cohorts_jointcalled_together'</li><li>'library_archived_at_nscc'</li><li>'gvcf_file_available'</li><li>'sequencing_complete'</li><li>'supplied_and_computed_gender_match'</li><li>'supplied_and_computed_ethinicity_match'</li><li>'sop_agreement_for_coverage_14/28_for_15x/30x'</li><li>'sample_life_cycle'</li><li>'current'</li><li>'source_cohort'</li><li>'originalsupplierid'</li><li>'duplicateinfo'</li><li>'duplicate'</li><li>'duplicatepair'</li><li>'removalrequestedbysupplier'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'npm\\_research\\_id'\n",
       "\\item 'multiplex\\_pool\\_id'\n",
       "\\item 'supplier\\_id'\n",
       "\\item 'gis\\_internal\\_sample\\_id'\n",
       "\\item 'site\\_supplying\\_sample'\n",
       "\\item 'year\\_of\\_birth'\n",
       "\\item 'supplied\\_gender'\n",
       "\\item 'self\\_reported\\_ethnicity'\n",
       "\\item 'extraction\\_kit'\n",
       "\\item 'date\\_of\\_dna\\_extraction'\n",
       "\\item 'plate\\_position'\n",
       "\\item 'plate\\_name'\n",
       "\\item 'version\\_of\\_consent\\_form\\_signed'\n",
       "\\item 'sequencing\\_depth'\n",
       "\\item 'npm\\_research\\_id\\_created\\_by\\_username'\n",
       "\\item 'npm\\_research\\_id\\_creation\\_date'\n",
       "\\item 'comments\\_entered\\_when\\_npm\\_research\\_id\\_created'\n",
       "\\item 'description\\_entered\\_when\\_npm\\_research\\_id\\_created'\n",
       "\\item 'elm\\_project\\_id'\n",
       "\\item 'elm\\_project\\_title'\n",
       "\\item 'elm\\_project\\_pi'\n",
       "\\item 'species\\_of\\_sample\\_sequenced'\n",
       "\\item 'tehcnique\\_for\\_sequencing'\n",
       "\\item 'tissue\\_type\\_sequenced'\n",
       "\\item 'library\\_found\\_in\\_sequencing\\_run\\_id'\n",
       "\\item 'library\\_found\\_in\\_passed\\_sequencing\\_run\\_id'\n",
       "\\item 'vendor\\_sequencing\\_centre'\n",
       "\\item 'dna\\_sample\\_passed\\_qc'\n",
       "\\item 'library\\_prep\\_kit'\n",
       "\\item 'run\\_id'\n",
       "\\item 'instrument\\_id'\n",
       "\\item 'instrument\\_serial\\_number'\n",
       "\\item 'hiseq\\_xtm\\_sbs\\_kit\\_300\\_cycles\\_v2\\_box\\_1of\\_2\\_lot'\n",
       "\\item 'hiseq\\_xtm\\_sbs\\_kit\\_300\\_cycles\\_v2\\_box\\_2\\_of\\_2\\_lot'\n",
       "\\item 'hiseq\\_xtm\\_pe\\_cluster\\_kit\\_cbottm\\_v2\\_box\\_1\\_of\\_2\\_lot'\n",
       "\\item 'hiseq\\_xtm\\_pe\\_cluster\\_kit\\_cbottm\\_v2\\_box\\_2\\_of\\_2\\_lot'\n",
       "\\item 'hiseq\\_xtm\\_flow\\_cell\\_v25\\_lot'\n",
       "\\item 'hiseq\\_xtm\\_flow\\_cell\\_v25\\_id'\n",
       "\\item 'sequencebased\\_gender'\n",
       "\\item 'cram\\_files\\_location'\n",
       "\\item 'estimate\\_of\\_sequence\\_coverage'\n",
       "\\item 'contamination\\_rate\\_of\\_chinese\\_genome'\n",
       "\\item 'contamination\\_rate\\_of\\_indian\\_genome'\n",
       "\\item 'contamination\\_rate\\_of\\_malaysian\\_genome'\n",
       "\\item 'average\\_coverage\\_depth'\n",
       "\\item 'fb\\_variants'\n",
       "\\item 'cram\\_filesize\\_mb'\n",
       "\\item 'error\\_rate'\n",
       "\\item 'fbvcf\\_filesize\\_mb'\n",
       "\\item 'gvcf\\_filesize\\_mb'\n",
       "\\item 'insert\\_size\\_average'\n",
       "\\item 'insert\\_size\\_standard\\_deviation'\n",
       "\\item 'raw\\_total\\_sequences'\n",
       "\\item 'reads\\_duplicated\\_'\n",
       "\\item 'reads\\_properly\\_paired\\_'\n",
       "\\item 'cohort\\_jvcf\\_data\\_capacity'\n",
       "\\item 'fastq\\_file\\_size'\n",
       "\\item 'jvcf\\_size'\n",
       "\\item 'data\\_timestamp\\_for\\_fastq\\_file'\n",
       "\\item 'sequencing\\_analysis\\_pipeline'\n",
       "\\item 'cohorts\\_jointcalled\\_together'\n",
       "\\item 'library\\_archived\\_at\\_nscc'\n",
       "\\item 'gvcf\\_file\\_available'\n",
       "\\item 'sequencing\\_complete'\n",
       "\\item 'supplied\\_and\\_computed\\_gender\\_match'\n",
       "\\item 'supplied\\_and\\_computed\\_ethinicity\\_match'\n",
       "\\item 'sop\\_agreement\\_for\\_coverage\\_14/28\\_for\\_15x/30x'\n",
       "\\item 'sample\\_life\\_cycle'\n",
       "\\item 'current'\n",
       "\\item 'source\\_cohort'\n",
       "\\item 'originalsupplierid'\n",
       "\\item 'duplicateinfo'\n",
       "\\item 'duplicate'\n",
       "\\item 'duplicatepair'\n",
       "\\item 'removalrequestedbysupplier'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'npm_research_id'\n",
       "2. 'multiplex_pool_id'\n",
       "3. 'supplier_id'\n",
       "4. 'gis_internal_sample_id'\n",
       "5. 'site_supplying_sample'\n",
       "6. 'year_of_birth'\n",
       "7. 'supplied_gender'\n",
       "8. 'self_reported_ethnicity'\n",
       "9. 'extraction_kit'\n",
       "10. 'date_of_dna_extraction'\n",
       "11. 'plate_position'\n",
       "12. 'plate_name'\n",
       "13. 'version_of_consent_form_signed'\n",
       "14. 'sequencing_depth'\n",
       "15. 'npm_research_id_created_by_username'\n",
       "16. 'npm_research_id_creation_date'\n",
       "17. 'comments_entered_when_npm_research_id_created'\n",
       "18. 'description_entered_when_npm_research_id_created'\n",
       "19. 'elm_project_id'\n",
       "20. 'elm_project_title'\n",
       "21. 'elm_project_pi'\n",
       "22. 'species_of_sample_sequenced'\n",
       "23. 'tehcnique_for_sequencing'\n",
       "24. 'tissue_type_sequenced'\n",
       "25. 'library_found_in_sequencing_run_id'\n",
       "26. 'library_found_in_passed_sequencing_run_id'\n",
       "27. 'vendor_sequencing_centre'\n",
       "28. 'dna_sample_passed_qc'\n",
       "29. 'library_prep_kit'\n",
       "30. 'run_id'\n",
       "31. 'instrument_id'\n",
       "32. 'instrument_serial_number'\n",
       "33. 'hiseq_xtm_sbs_kit_300_cycles_v2_box_1of_2_lot'\n",
       "34. 'hiseq_xtm_sbs_kit_300_cycles_v2_box_2_of_2_lot'\n",
       "35. 'hiseq_xtm_pe_cluster_kit_cbottm_v2_box_1_of_2_lot'\n",
       "36. 'hiseq_xtm_pe_cluster_kit_cbottm_v2_box_2_of_2_lot'\n",
       "37. 'hiseq_xtm_flow_cell_v25_lot'\n",
       "38. 'hiseq_xtm_flow_cell_v25_id'\n",
       "39. 'sequencebased_gender'\n",
       "40. 'cram_files_location'\n",
       "41. 'estimate_of_sequence_coverage'\n",
       "42. 'contamination_rate_of_chinese_genome'\n",
       "43. 'contamination_rate_of_indian_genome'\n",
       "44. 'contamination_rate_of_malaysian_genome'\n",
       "45. 'average_coverage_depth'\n",
       "46. 'fb_variants'\n",
       "47. 'cram_filesize_mb'\n",
       "48. 'error_rate'\n",
       "49. 'fbvcf_filesize_mb'\n",
       "50. 'gvcf_filesize_mb'\n",
       "51. 'insert_size_average'\n",
       "52. 'insert_size_standard_deviation'\n",
       "53. 'raw_total_sequences'\n",
       "54. 'reads_duplicated_'\n",
       "55. 'reads_properly_paired_'\n",
       "56. 'cohort_jvcf_data_capacity'\n",
       "57. 'fastq_file_size'\n",
       "58. 'jvcf_size'\n",
       "59. 'data_timestamp_for_fastq_file'\n",
       "60. 'sequencing_analysis_pipeline'\n",
       "61. 'cohorts_jointcalled_together'\n",
       "62. 'library_archived_at_nscc'\n",
       "63. 'gvcf_file_available'\n",
       "64. 'sequencing_complete'\n",
       "65. 'supplied_and_computed_gender_match'\n",
       "66. 'supplied_and_computed_ethinicity_match'\n",
       "67. 'sop_agreement_for_coverage_14/28_for_15x/30x'\n",
       "68. 'sample_life_cycle'\n",
       "69. 'current'\n",
       "70. 'source_cohort'\n",
       "71. 'originalsupplierid'\n",
       "72. 'duplicateinfo'\n",
       "73. 'duplicate'\n",
       "74. 'duplicatepair'\n",
       "75. 'removalrequestedbysupplier'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"npm_research_id\"                                  \n",
       " [2] \"multiplex_pool_id\"                                \n",
       " [3] \"supplier_id\"                                      \n",
       " [4] \"gis_internal_sample_id\"                           \n",
       " [5] \"site_supplying_sample\"                            \n",
       " [6] \"year_of_birth\"                                    \n",
       " [7] \"supplied_gender\"                                  \n",
       " [8] \"self_reported_ethnicity\"                          \n",
       " [9] \"extraction_kit\"                                   \n",
       "[10] \"date_of_dna_extraction\"                           \n",
       "[11] \"plate_position\"                                   \n",
       "[12] \"plate_name\"                                       \n",
       "[13] \"version_of_consent_form_signed\"                   \n",
       "[14] \"sequencing_depth\"                                 \n",
       "[15] \"npm_research_id_created_by_username\"              \n",
       "[16] \"npm_research_id_creation_date\"                    \n",
       "[17] \"comments_entered_when_npm_research_id_created\"    \n",
       "[18] \"description_entered_when_npm_research_id_created\" \n",
       "[19] \"elm_project_id\"                                   \n",
       "[20] \"elm_project_title\"                                \n",
       "[21] \"elm_project_pi\"                                   \n",
       "[22] \"species_of_sample_sequenced\"                      \n",
       "[23] \"tehcnique_for_sequencing\"                         \n",
       "[24] \"tissue_type_sequenced\"                            \n",
       "[25] \"library_found_in_sequencing_run_id\"               \n",
       "[26] \"library_found_in_passed_sequencing_run_id\"        \n",
       "[27] \"vendor_sequencing_centre\"                         \n",
       "[28] \"dna_sample_passed_qc\"                             \n",
       "[29] \"library_prep_kit\"                                 \n",
       "[30] \"run_id\"                                           \n",
       "[31] \"instrument_id\"                                    \n",
       "[32] \"instrument_serial_number\"                         \n",
       "[33] \"hiseq_xtm_sbs_kit_300_cycles_v2_box_1of_2_lot\"    \n",
       "[34] \"hiseq_xtm_sbs_kit_300_cycles_v2_box_2_of_2_lot\"   \n",
       "[35] \"hiseq_xtm_pe_cluster_kit_cbottm_v2_box_1_of_2_lot\"\n",
       "[36] \"hiseq_xtm_pe_cluster_kit_cbottm_v2_box_2_of_2_lot\"\n",
       "[37] \"hiseq_xtm_flow_cell_v25_lot\"                      \n",
       "[38] \"hiseq_xtm_flow_cell_v25_id\"                       \n",
       "[39] \"sequencebased_gender\"                             \n",
       "[40] \"cram_files_location\"                              \n",
       "[41] \"estimate_of_sequence_coverage\"                    \n",
       "[42] \"contamination_rate_of_chinese_genome\"             \n",
       "[43] \"contamination_rate_of_indian_genome\"              \n",
       "[44] \"contamination_rate_of_malaysian_genome\"           \n",
       "[45] \"average_coverage_depth\"                           \n",
       "[46] \"fb_variants\"                                      \n",
       "[47] \"cram_filesize_mb\"                                 \n",
       "[48] \"error_rate\"                                       \n",
       "[49] \"fbvcf_filesize_mb\"                                \n",
       "[50] \"gvcf_filesize_mb\"                                 \n",
       "[51] \"insert_size_average\"                              \n",
       "[52] \"insert_size_standard_deviation\"                   \n",
       "[53] \"raw_total_sequences\"                              \n",
       "[54] \"reads_duplicated_\"                                \n",
       "[55] \"reads_properly_paired_\"                           \n",
       "[56] \"cohort_jvcf_data_capacity\"                        \n",
       "[57] \"fastq_file_size\"                                  \n",
       "[58] \"jvcf_size\"                                        \n",
       "[59] \"data_timestamp_for_fastq_file\"                    \n",
       "[60] \"sequencing_analysis_pipeline\"                     \n",
       "[61] \"cohorts_jointcalled_together\"                     \n",
       "[62] \"library_archived_at_nscc\"                         \n",
       "[63] \"gvcf_file_available\"                              \n",
       "[64] \"sequencing_complete\"                              \n",
       "[65] \"supplied_and_computed_gender_match\"               \n",
       "[66] \"supplied_and_computed_ethinicity_match\"           \n",
       "[67] \"sop_agreement_for_coverage_14/28_for_15x/30x\"     \n",
       "[68] \"sample_life_cycle\"                                \n",
       "[69] \"current\"                                          \n",
       "[70] \"source_cohort\"                                    \n",
       "[71] \"originalsupplierid\"                               \n",
       "[72] \"duplicateinfo\"                                    \n",
       "[73] \"duplicate\"                                        \n",
       "[74] \"duplicatepair\"                                    \n",
       "[75] \"removalrequestedbysupplier\"                       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "n <- gsub(\"[^0-9A-Za-z///' ]\", \"\", colnames(meta))\n",
    "n <- tolower(n)\n",
    "n <- gsub(\" \", \"_\", n)\n",
    "colnames(meta) <- n\n",
    "\n",
    "colnames(meta)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4dc90b03",
   "metadata": {},
   "source": [
    "### Filter by r5.3 samples and remove duplicate samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "add95432",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_filt <- meta %>% \n",
    "  filter(npm_research_id %in% meta2) %>% # Retain only r5.3 samples\n",
    "  replace(is.na(.), \"\") %>%\n",
    "  filter(duplicateinfo != \"TRUE.DUPLICATE\") # Remove 64 duplicates\n",
    "\n",
    "fwrite(meta_filt, \"SG10K_Health_metadata.n10714.16March2021.parsed.csv\", row.names = F, sep = \",\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da00dfa6",
   "metadata": {},
   "source": [
    "### Retrieve random 10 samples per cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e2ec49b3",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in unique(meta_filt$site_supplying_sample): object 'meta_filt' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in unique(meta_filt$site_supplying_sample): object 'meta_filt' not found\nTraceback:\n",
      "1. unique(meta_filt$site_supplying_sample)"
     ]
    }
   ],
   "source": [
    "cohorts <- unique(meta_filt$site_supplying_sample)\n",
    "subset_vec <- c()\n",
    "\n",
    "for (i in cohorts) {\n",
    "  ids <- meta_filt$npm_research_id[meta_filt$site_supplying_sample == i]\n",
    "  subset_ids <- sample(ids, 10)\n",
    "  subset_vec <- c(subset_vec, subset_ids)\n",
    "}\n",
    "\n",
    "meta_sub <- meta_filt %>%\n",
    "  filter(npm_research_id %in% subset_vec) %>%\n",
    "  select(site_supplying_sample, multiplex_pool_id, npm_research_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4666efeb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# fwrite(meta_sub, \"meta_sub.csv\", row.names = F, col.names = F, sep = \",\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
