{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "53041a0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rsamtools\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message:\n",
      "“Your system is mis-configured: ‘/etc/localtime’ is not a symlink”\n",
      "Warning message:\n",
      "“It is strongly recommended to set envionment variable TZ to ‘Asia/Singapore’ (or equivalent)”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.4     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.2     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mBiostrings\u001b[39m::collapse(), \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mcompact()\u001b[39m    masks \u001b[34mXVector\u001b[39m::compact()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mXVector\u001b[39m::slice(), \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "\n",
      "Attaching package: ‘foreach’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:purrr’:\n",
      "\n",
      "    accumulate, when\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm(list = ls())\n",
    "setwd(\"/home/projects/14001280/PROJECTS/blood_microbiome\")\n",
    "require(\"Rsamtools\")\n",
    "require(Biostrings)\n",
    "require(tidyverse)\n",
    "require(foreach)\n",
    "require(doParallel)\n",
    "registerDoParallel(cores = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5155b4c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Achromobacter_xylosoxidans_NZ_CP043820.1'</li><li>'Acinetobacter_baumannii_NZ_CP043953.1'</li><li>'Bacillus_cereus_NZ_CP072774.1'</li><li>'Burkholderia_contaminans_GCF_004723625.1_ASM472362v1'</li><li>'Corynebacterium_segmentosum_NZ_LR134408.1'</li><li>'Cutibacterium_acnes_NC_021085.1'</li><li>'Enterobacter_cloacae_NZ_CP009756.1'</li><li>'Fusobacterium_nucleatum_NZ_LN831027.1'</li><li>'Fusobacterium_periodonticum_NZ_GG665898.1'</li><li>'Gardnerella_vaginalis_NZ_PKJK01000001.1'</li><li>'Haemophilus_parainfluenzae_NZ_GL872339.1'</li><li>'Human_mastadenovirus_C_NC_001405.1'</li><li>'Lactobacillus_crispatus_NZ_CP039266.1'</li><li>'Microbacterium_PAMC_28756_NZ_CP014313.1'</li><li>'Microbacterium_PM5_NZ_CP022162.1'</li><li>'Microbacterium_Y-01_NZ_CP024170.1'</li><li>'Microbacterium_hominis_NZ_CP061344.1'</li><li>'Microbacterium_paraoxydans_NZ_LT629770.1'</li><li>'Neisseria_subflava_NZ_CP039887.1'</li><li>'Paracoccus_yeei_NZ_CP024422.1'</li><li>'Pseudomonas_mendocina_NZ_CP013124.1'</li><li>'Ralstonia_solanacearum_NZ_CP088237.1'</li><li>'Rickettsia_felis_NZ_JSEL01000013.1'</li><li>'Staphylococcus_cohnii_NZ_UHDA01000001.1'</li><li>'Staphylococcus_epidermidis_NZ_CP035288.1'</li><li>'Staphylococcus_haemolyticus_NZ_CP013911.1'</li><li>'Zhihengliuella_ISTPL4_NZ_CP025422.1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1'\n",
       "\\item 'Acinetobacter\\_baumannii\\_NZ\\_CP043953.1'\n",
       "\\item 'Bacillus\\_cereus\\_NZ\\_CP072774.1'\n",
       "\\item 'Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1'\n",
       "\\item 'Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1'\n",
       "\\item 'Cutibacterium\\_acnes\\_NC\\_021085.1'\n",
       "\\item 'Enterobacter\\_cloacae\\_NZ\\_CP009756.1'\n",
       "\\item 'Fusobacterium\\_nucleatum\\_NZ\\_LN831027.1'\n",
       "\\item 'Fusobacterium\\_periodonticum\\_NZ\\_GG665898.1'\n",
       "\\item 'Gardnerella\\_vaginalis\\_NZ\\_PKJK01000001.1'\n",
       "\\item 'Haemophilus\\_parainfluenzae\\_NZ\\_GL872339.1'\n",
       "\\item 'Human\\_mastadenovirus\\_C\\_NC\\_001405.1'\n",
       "\\item 'Lactobacillus\\_crispatus\\_NZ\\_CP039266.1'\n",
       "\\item 'Microbacterium\\_PAMC\\_28756\\_NZ\\_CP014313.1'\n",
       "\\item 'Microbacterium\\_PM5\\_NZ\\_CP022162.1'\n",
       "\\item 'Microbacterium\\_Y-01\\_NZ\\_CP024170.1'\n",
       "\\item 'Microbacterium\\_hominis\\_NZ\\_CP061344.1'\n",
       "\\item 'Microbacterium\\_paraoxydans\\_NZ\\_LT629770.1'\n",
       "\\item 'Neisseria\\_subflava\\_NZ\\_CP039887.1'\n",
       "\\item 'Paracoccus\\_yeei\\_NZ\\_CP024422.1'\n",
       "\\item 'Pseudomonas\\_mendocina\\_NZ\\_CP013124.1'\n",
       "\\item 'Ralstonia\\_solanacearum\\_NZ\\_CP088237.1'\n",
       "\\item 'Rickettsia\\_felis\\_NZ\\_JSEL01000013.1'\n",
       "\\item 'Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1'\n",
       "\\item 'Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1'\n",
       "\\item 'Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1'\n",
       "\\item 'Zhihengliuella\\_ISTPL4\\_NZ\\_CP025422.1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Achromobacter_xylosoxidans_NZ_CP043820.1'\n",
       "2. 'Acinetobacter_baumannii_NZ_CP043953.1'\n",
       "3. 'Bacillus_cereus_NZ_CP072774.1'\n",
       "4. 'Burkholderia_contaminans_GCF_004723625.1_ASM472362v1'\n",
       "5. 'Corynebacterium_segmentosum_NZ_LR134408.1'\n",
       "6. 'Cutibacterium_acnes_NC_021085.1'\n",
       "7. 'Enterobacter_cloacae_NZ_CP009756.1'\n",
       "8. 'Fusobacterium_nucleatum_NZ_LN831027.1'\n",
       "9. 'Fusobacterium_periodonticum_NZ_GG665898.1'\n",
       "10. 'Gardnerella_vaginalis_NZ_PKJK01000001.1'\n",
       "11. 'Haemophilus_parainfluenzae_NZ_GL872339.1'\n",
       "12. 'Human_mastadenovirus_C_NC_001405.1'\n",
       "13. 'Lactobacillus_crispatus_NZ_CP039266.1'\n",
       "14. 'Microbacterium_PAMC_28756_NZ_CP014313.1'\n",
       "15. 'Microbacterium_PM5_NZ_CP022162.1'\n",
       "16. 'Microbacterium_Y-01_NZ_CP024170.1'\n",
       "17. 'Microbacterium_hominis_NZ_CP061344.1'\n",
       "18. 'Microbacterium_paraoxydans_NZ_LT629770.1'\n",
       "19. 'Neisseria_subflava_NZ_CP039887.1'\n",
       "20. 'Paracoccus_yeei_NZ_CP024422.1'\n",
       "21. 'Pseudomonas_mendocina_NZ_CP013124.1'\n",
       "22. 'Ralstonia_solanacearum_NZ_CP088237.1'\n",
       "23. 'Rickettsia_felis_NZ_JSEL01000013.1'\n",
       "24. 'Staphylococcus_cohnii_NZ_UHDA01000001.1'\n",
       "25. 'Staphylococcus_epidermidis_NZ_CP035288.1'\n",
       "26. 'Staphylococcus_haemolyticus_NZ_CP013911.1'\n",
       "27. 'Zhihengliuella_ISTPL4_NZ_CP025422.1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Achromobacter_xylosoxidans_NZ_CP043820.1\"            \n",
       " [2] \"Acinetobacter_baumannii_NZ_CP043953.1\"               \n",
       " [3] \"Bacillus_cereus_NZ_CP072774.1\"                       \n",
       " [4] \"Burkholderia_contaminans_GCF_004723625.1_ASM472362v1\"\n",
       " [5] \"Corynebacterium_segmentosum_NZ_LR134408.1\"           \n",
       " [6] \"Cutibacterium_acnes_NC_021085.1\"                     \n",
       " [7] \"Enterobacter_cloacae_NZ_CP009756.1\"                  \n",
       " [8] \"Fusobacterium_nucleatum_NZ_LN831027.1\"               \n",
       " [9] \"Fusobacterium_periodonticum_NZ_GG665898.1\"           \n",
       "[10] \"Gardnerella_vaginalis_NZ_PKJK01000001.1\"             \n",
       "[11] \"Haemophilus_parainfluenzae_NZ_GL872339.1\"            \n",
       "[12] \"Human_mastadenovirus_C_NC_001405.1\"                  \n",
       "[13] \"Lactobacillus_crispatus_NZ_CP039266.1\"               \n",
       "[14] \"Microbacterium_PAMC_28756_NZ_CP014313.1\"             \n",
       "[15] \"Microbacterium_PM5_NZ_CP022162.1\"                    \n",
       "[16] \"Microbacterium_Y-01_NZ_CP024170.1\"                   \n",
       "[17] \"Microbacterium_hominis_NZ_CP061344.1\"                \n",
       "[18] \"Microbacterium_paraoxydans_NZ_LT629770.1\"            \n",
       "[19] \"Neisseria_subflava_NZ_CP039887.1\"                    \n",
       "[20] \"Paracoccus_yeei_NZ_CP024422.1\"                       \n",
       "[21] \"Pseudomonas_mendocina_NZ_CP013124.1\"                 \n",
       "[22] \"Ralstonia_solanacearum_NZ_CP088237.1\"                \n",
       "[23] \"Rickettsia_felis_NZ_JSEL01000013.1\"                  \n",
       "[24] \"Staphylococcus_cohnii_NZ_UHDA01000001.1\"             \n",
       "[25] \"Staphylococcus_epidermidis_NZ_CP035288.1\"            \n",
       "[26] \"Staphylococcus_haemolyticus_NZ_CP013911.1\"           \n",
       "[27] \"Zhihengliuella_ISTPL4_NZ_CP025422.1\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_path <- \"results/irep_analysis/raw_output/bam_files\"\n",
    "ref_path <- \"data/irep_data/genome_references\"\n",
    "prefixes <- list.files(ref_path, \".fasta\", full.names = F)\n",
    "prefixes <- gsub(\".fasta\", \"\", prefixes)\n",
    "prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0d49d2e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "morsels <- foreach (i = seq(length(prefixes))) %dopar% {\n",
    "#     i <- 1\n",
    "#     j <- 1\n",
    "    prefix <- prefixes[i]\n",
    "    bams <- list.files(str_glue(\"{bam_path}/{prefix}\"))\n",
    "    bams <- bams[!grepl(\".bai\", bams)]\n",
    "\n",
    "    crumbs <- foreach (j = seq(length(bams))) %do% {\n",
    "        bam <- bams[j]\n",
    "        ref <- readDNAStringSet(str_glue(\"{ref_path}/{prefixes[i]}.fasta\"))\n",
    "        ref_length <- width(ref)\n",
    "        params <- PileupParam(distinguish_strands = F, distinguish_nucleotides = F)\n",
    "        pile_df <- pileup(str_glue(\"{bam_path}/{prefix}/{bam}\"), pileupParam = params)\n",
    "        \n",
    "        n_covered1 <- pile_df %>%\n",
    "            filter(count >= 1) %>%\n",
    "            nrow()\n",
    "        \n",
    "        n_covered5 <- pile_df %>%\n",
    "            filter(count >= 5) %>%\n",
    "            nrow()\n",
    "\n",
    "        perc_covered1 <- n_covered1 / ref_length * 100\n",
    "        perc_covered5 <- n_covered5 / ref_length * 100\n",
    "        \n",
    "        return(tibble(prefix = prefix, \n",
    "                      perc_covered1 = perc_covered1,\n",
    "                      perc_covered5 = perc_covered5))\n",
    "    }\n",
    "    \n",
    "    return(bind_rows(crumbs))\n",
    "}\n",
    "\n",
    "cov_df <- bind_rows(morsels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ca3e39e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 150 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>prefix</th><th scope=col>perc_covered1</th><th scope=col>perc_covered5</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1                 </td><td>69.178516</td><td>6.295949e+01</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1            </td><td>65.942462</td><td>1.854962e+01</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1                 </td><td>65.814744</td><td>5.118415e+01</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1            </td><td>63.372644</td><td>1.331145e+01</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1                 </td><td>61.543401</td><td>3.450724e+01</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td>14.541461</td><td>4.117875e+00</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td>11.237105</td><td>1.994434e+00</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1            </td><td> 9.195231</td><td>1.075749e-01</td></tr>\n",
       "\t<tr><td>Microbacterium_Y-01_NZ_CP024170.1                   </td><td> 4.312364</td><td>4.389527e-02</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 4.286375</td><td>1.213823e+00</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 4.028783</td><td>3.543847e-01</td></tr>\n",
       "\t<tr><td>Microbacterium_Y-01_NZ_CP024170.1                   </td><td> 3.426318</td><td>7.080305e-02</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 3.312352</td><td>5.878977e-01</td></tr>\n",
       "\t<tr><td>Lactobacillus_crispatus_NZ_CP039266.1               </td><td> 3.190067</td><td>1.977202e-04</td></tr>\n",
       "\t<tr><td>Microbacterium_paraoxydans_NZ_LT629770.1            </td><td> 2.879279</td><td>4.653306e-02</td></tr>\n",
       "\t<tr><td>Microbacterium_PAMC_28756_NZ_CP014313.1             </td><td> 2.868190</td><td>4.893954e-02</td></tr>\n",
       "\t<tr><td>Zhihengliuella_ISTPL4_NZ_CP025422.1                 </td><td> 2.557703</td><td>6.091119e-02</td></tr>\n",
       "\t<tr><td>Ralstonia_solanacearum_NZ_CP088237.1                </td><td> 2.362988</td><td>5.853346e-01</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 2.030496</td><td>5.749993e-01</td></tr>\n",
       "\t<tr><td>Ralstonia_solanacearum_NZ_CP088237.1                </td><td> 1.982340</td><td>2.856937e-01</td></tr>\n",
       "\t<tr><td>Microbacterium_paraoxydans_NZ_LT629770.1            </td><td> 1.957119</td><td>5.241655e-02</td></tr>\n",
       "\t<tr><td>Microbacterium_PAMC_28756_NZ_CP014313.1             </td><td> 1.909744</td><td>5.885742e-02</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 1.800490</td><td>5.098658e-01</td></tr>\n",
       "\t<tr><td>Zhihengliuella_ISTPL4_NZ_CP025422.1                 </td><td> 1.762940</td><td>6.822053e-02</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 1.569093</td><td>2.784927e-01</td></tr>\n",
       "\t<tr><td>Microbacterium_Y-01_NZ_CP024170.1                   </td><td> 1.527854</td><td>1.669957e-02</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 1.391352</td><td>2.469462e-01</td></tr>\n",
       "\t<tr><td>Microbacterium_hominis_NZ_CP061344.1                </td><td> 1.189470</td><td>5.363001e-02</td></tr>\n",
       "\t<tr><td>Burkholderia_contaminans_GCF_004723625.1_ASM472362v1</td><td> 1.187561</td><td>1.044617e-01</td></tr>\n",
       "\t<tr><td>Microbacterium_hominis_NZ_CP061344.1                </td><td> 1.127352</td><td>9.451426e-02</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Human_mastadenovirus_C_NC_001405.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Human_mastadenovirus_C_NC_001405.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Human_mastadenovirus_C_NC_001405.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Human_mastadenovirus_C_NC_001405.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Lactobacillus_crispatus_NZ_CP039266.1    </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_PAMC_28756_NZ_CP014313.1  </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_PAMC_28756_NZ_CP014313.1  </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_PM5_NZ_CP022162.1         </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_PM5_NZ_CP022162.1         </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_Y-01_NZ_CP024170.1        </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_Y-01_NZ_CP024170.1        </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_hominis_NZ_CP061344.1     </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_hominis_NZ_CP061344.1     </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_paraoxydans_NZ_LT629770.1 </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Microbacterium_paraoxydans_NZ_LT629770.1 </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Neisseria_subflava_NZ_CP039887.1         </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Neisseria_subflava_NZ_CP039887.1         </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Paracoccus_yeei_NZ_CP024422.1            </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Ralstonia_solanacearum_NZ_CP088237.1     </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Rickettsia_felis_NZ_JSEL01000013.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Rickettsia_felis_NZ_JSEL01000013.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Rickettsia_felis_NZ_JSEL01000013.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Rickettsia_felis_NZ_JSEL01000013.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Rickettsia_felis_NZ_JSEL01000013.1       </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Zhihengliuella_ISTPL4_NZ_CP025422.1      </td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>Zhihengliuella_ISTPL4_NZ_CP025422.1      </td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 150 × 3\n",
       "\\begin{tabular}{lll}\n",
       " prefix & perc\\_covered1 & perc\\_covered5\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1                  & 69.178516 & 6.295949e+01\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1             & 65.942462 & 1.854962e+01\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1                  & 65.814744 & 5.118415e+01\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1             & 63.372644 & 1.331145e+01\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1                  & 61.543401 & 3.450724e+01\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 & 14.541461 & 4.117875e+00\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 & 11.237105 & 1.994434e+00\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1             &  9.195231 & 1.075749e-01\\\\\n",
       "\t Microbacterium\\_Y-01\\_NZ\\_CP024170.1                    &  4.312364 & 4.389527e-02\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  4.286375 & 1.213823e+00\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  4.028783 & 3.543847e-01\\\\\n",
       "\t Microbacterium\\_Y-01\\_NZ\\_CP024170.1                    &  3.426318 & 7.080305e-02\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  3.312352 & 5.878977e-01\\\\\n",
       "\t Lactobacillus\\_crispatus\\_NZ\\_CP039266.1                &  3.190067 & 1.977202e-04\\\\\n",
       "\t Microbacterium\\_paraoxydans\\_NZ\\_LT629770.1             &  2.879279 & 4.653306e-02\\\\\n",
       "\t Microbacterium\\_PAMC\\_28756\\_NZ\\_CP014313.1              &  2.868190 & 4.893954e-02\\\\\n",
       "\t Zhihengliuella\\_ISTPL4\\_NZ\\_CP025422.1                  &  2.557703 & 6.091119e-02\\\\\n",
       "\t Ralstonia\\_solanacearum\\_NZ\\_CP088237.1                 &  2.362988 & 5.853346e-01\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  2.030496 & 5.749993e-01\\\\\n",
       "\t Ralstonia\\_solanacearum\\_NZ\\_CP088237.1                 &  1.982340 & 2.856937e-01\\\\\n",
       "\t Microbacterium\\_paraoxydans\\_NZ\\_LT629770.1             &  1.957119 & 5.241655e-02\\\\\n",
       "\t Microbacterium\\_PAMC\\_28756\\_NZ\\_CP014313.1              &  1.909744 & 5.885742e-02\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  1.800490 & 5.098658e-01\\\\\n",
       "\t Zhihengliuella\\_ISTPL4\\_NZ\\_CP025422.1                  &  1.762940 & 6.822053e-02\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  1.569093 & 2.784927e-01\\\\\n",
       "\t Microbacterium\\_Y-01\\_NZ\\_CP024170.1                    &  1.527854 & 1.669957e-02\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  1.391352 & 2.469462e-01\\\\\n",
       "\t Microbacterium\\_hominis\\_NZ\\_CP061344.1                 &  1.189470 & 5.363001e-02\\\\\n",
       "\t Burkholderia\\_contaminans\\_GCF\\_004723625.1\\_ASM472362v1 &  1.187561 & 1.044617e-01\\\\\n",
       "\t Microbacterium\\_hominis\\_NZ\\_CP061344.1                 &  1.127352 & 9.451426e-02\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t Human\\_mastadenovirus\\_C\\_NC\\_001405.1        & 0 & 0\\\\\n",
       "\t Human\\_mastadenovirus\\_C\\_NC\\_001405.1        & 0 & 0\\\\\n",
       "\t Human\\_mastadenovirus\\_C\\_NC\\_001405.1        & 0 & 0\\\\\n",
       "\t Human\\_mastadenovirus\\_C\\_NC\\_001405.1        & 0 & 0\\\\\n",
       "\t Lactobacillus\\_crispatus\\_NZ\\_CP039266.1     & 0 & 0\\\\\n",
       "\t Microbacterium\\_PAMC\\_28756\\_NZ\\_CP014313.1   & 0 & 0\\\\\n",
       "\t Microbacterium\\_PAMC\\_28756\\_NZ\\_CP014313.1   & 0 & 0\\\\\n",
       "\t Microbacterium\\_PM5\\_NZ\\_CP022162.1          & 0 & 0\\\\\n",
       "\t Microbacterium\\_PM5\\_NZ\\_CP022162.1          & 0 & 0\\\\\n",
       "\t Microbacterium\\_Y-01\\_NZ\\_CP024170.1         & 0 & 0\\\\\n",
       "\t Microbacterium\\_Y-01\\_NZ\\_CP024170.1         & 0 & 0\\\\\n",
       "\t Microbacterium\\_hominis\\_NZ\\_CP061344.1      & 0 & 0\\\\\n",
       "\t Microbacterium\\_hominis\\_NZ\\_CP061344.1      & 0 & 0\\\\\n",
       "\t Microbacterium\\_paraoxydans\\_NZ\\_LT629770.1  & 0 & 0\\\\\n",
       "\t Microbacterium\\_paraoxydans\\_NZ\\_LT629770.1  & 0 & 0\\\\\n",
       "\t Neisseria\\_subflava\\_NZ\\_CP039887.1          & 0 & 0\\\\\n",
       "\t Neisseria\\_subflava\\_NZ\\_CP039887.1          & 0 & 0\\\\\n",
       "\t Paracoccus\\_yeei\\_NZ\\_CP024422.1             & 0 & 0\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0 & 0\\\\\n",
       "\t Ralstonia\\_solanacearum\\_NZ\\_CP088237.1      & 0 & 0\\\\\n",
       "\t Rickettsia\\_felis\\_NZ\\_JSEL01000013.1        & 0 & 0\\\\\n",
       "\t Rickettsia\\_felis\\_NZ\\_JSEL01000013.1        & 0 & 0\\\\\n",
       "\t Rickettsia\\_felis\\_NZ\\_JSEL01000013.1        & 0 & 0\\\\\n",
       "\t Rickettsia\\_felis\\_NZ\\_JSEL01000013.1        & 0 & 0\\\\\n",
       "\t Rickettsia\\_felis\\_NZ\\_JSEL01000013.1        & 0 & 0\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0 & 0\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & 0 & 0\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0 & 0\\\\\n",
       "\t Zhihengliuella\\_ISTPL4\\_NZ\\_CP025422.1       & 0 & 0\\\\\n",
       "\t Zhihengliuella\\_ISTPL4\\_NZ\\_CP025422.1       & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 150 × 3\n",
       "\n",
       "| prefix &lt;chr&gt; | perc_covered1 &lt;dbl&gt; | perc_covered5 &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1                  | 69.178516 | 6.295949e+01 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1             | 65.942462 | 1.854962e+01 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1                  | 65.814744 | 5.118415e+01 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1             | 63.372644 | 1.331145e+01 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1                  | 61.543401 | 3.450724e+01 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 | 14.541461 | 4.117875e+00 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 | 11.237105 | 1.994434e+00 |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1             |  9.195231 | 1.075749e-01 |\n",
       "| Microbacterium_Y-01_NZ_CP024170.1                    |  4.312364 | 4.389527e-02 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  4.286375 | 1.213823e+00 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  4.028783 | 3.543847e-01 |\n",
       "| Microbacterium_Y-01_NZ_CP024170.1                    |  3.426318 | 7.080305e-02 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  3.312352 | 5.878977e-01 |\n",
       "| Lactobacillus_crispatus_NZ_CP039266.1                |  3.190067 | 1.977202e-04 |\n",
       "| Microbacterium_paraoxydans_NZ_LT629770.1             |  2.879279 | 4.653306e-02 |\n",
       "| Microbacterium_PAMC_28756_NZ_CP014313.1              |  2.868190 | 4.893954e-02 |\n",
       "| Zhihengliuella_ISTPL4_NZ_CP025422.1                  |  2.557703 | 6.091119e-02 |\n",
       "| Ralstonia_solanacearum_NZ_CP088237.1                 |  2.362988 | 5.853346e-01 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  2.030496 | 5.749993e-01 |\n",
       "| Ralstonia_solanacearum_NZ_CP088237.1                 |  1.982340 | 2.856937e-01 |\n",
       "| Microbacterium_paraoxydans_NZ_LT629770.1             |  1.957119 | 5.241655e-02 |\n",
       "| Microbacterium_PAMC_28756_NZ_CP014313.1              |  1.909744 | 5.885742e-02 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  1.800490 | 5.098658e-01 |\n",
       "| Zhihengliuella_ISTPL4_NZ_CP025422.1                  |  1.762940 | 6.822053e-02 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  1.569093 | 2.784927e-01 |\n",
       "| Microbacterium_Y-01_NZ_CP024170.1                    |  1.527854 | 1.669957e-02 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  1.391352 | 2.469462e-01 |\n",
       "| Microbacterium_hominis_NZ_CP061344.1                 |  1.189470 | 5.363001e-02 |\n",
       "| Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 |  1.187561 | 1.044617e-01 |\n",
       "| Microbacterium_hominis_NZ_CP061344.1                 |  1.127352 | 9.451426e-02 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| Human_mastadenovirus_C_NC_001405.1        | 0 | 0 |\n",
       "| Human_mastadenovirus_C_NC_001405.1        | 0 | 0 |\n",
       "| Human_mastadenovirus_C_NC_001405.1        | 0 | 0 |\n",
       "| Human_mastadenovirus_C_NC_001405.1        | 0 | 0 |\n",
       "| Lactobacillus_crispatus_NZ_CP039266.1     | 0 | 0 |\n",
       "| Microbacterium_PAMC_28756_NZ_CP014313.1   | 0 | 0 |\n",
       "| Microbacterium_PAMC_28756_NZ_CP014313.1   | 0 | 0 |\n",
       "| Microbacterium_PM5_NZ_CP022162.1          | 0 | 0 |\n",
       "| Microbacterium_PM5_NZ_CP022162.1          | 0 | 0 |\n",
       "| Microbacterium_Y-01_NZ_CP024170.1         | 0 | 0 |\n",
       "| Microbacterium_Y-01_NZ_CP024170.1         | 0 | 0 |\n",
       "| Microbacterium_hominis_NZ_CP061344.1      | 0 | 0 |\n",
       "| Microbacterium_hominis_NZ_CP061344.1      | 0 | 0 |\n",
       "| Microbacterium_paraoxydans_NZ_LT629770.1  | 0 | 0 |\n",
       "| Microbacterium_paraoxydans_NZ_LT629770.1  | 0 | 0 |\n",
       "| Neisseria_subflava_NZ_CP039887.1          | 0 | 0 |\n",
       "| Neisseria_subflava_NZ_CP039887.1          | 0 | 0 |\n",
       "| Paracoccus_yeei_NZ_CP024422.1             | 0 | 0 |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0 | 0 |\n",
       "| Ralstonia_solanacearum_NZ_CP088237.1      | 0 | 0 |\n",
       "| Rickettsia_felis_NZ_JSEL01000013.1        | 0 | 0 |\n",
       "| Rickettsia_felis_NZ_JSEL01000013.1        | 0 | 0 |\n",
       "| Rickettsia_felis_NZ_JSEL01000013.1        | 0 | 0 |\n",
       "| Rickettsia_felis_NZ_JSEL01000013.1        | 0 | 0 |\n",
       "| Rickettsia_felis_NZ_JSEL01000013.1        | 0 | 0 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0 | 0 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | 0 | 0 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0 | 0 |\n",
       "| Zhihengliuella_ISTPL4_NZ_CP025422.1       | 0 | 0 |\n",
       "| Zhihengliuella_ISTPL4_NZ_CP025422.1       | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "    prefix                                               perc_covered1\n",
       "1   Pseudomonas_mendocina_NZ_CP013124.1                  69.178516    \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1             65.942462    \n",
       "3   Pseudomonas_mendocina_NZ_CP013124.1                  65.814744    \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1             63.372644    \n",
       "5   Pseudomonas_mendocina_NZ_CP013124.1                  61.543401    \n",
       "6   Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 14.541461    \n",
       "7   Burkholderia_contaminans_GCF_004723625.1_ASM472362v1 11.237105    \n",
       "8   Achromobacter_xylosoxidans_NZ_CP043820.1              9.195231    \n",
       "9   Microbacterium_Y-01_NZ_CP024170.1                     4.312364    \n",
       "10  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  4.286375    \n",
       "11  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  4.028783    \n",
       "12  Microbacterium_Y-01_NZ_CP024170.1                     3.426318    \n",
       "13  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  3.312352    \n",
       "14  Lactobacillus_crispatus_NZ_CP039266.1                 3.190067    \n",
       "15  Microbacterium_paraoxydans_NZ_LT629770.1              2.879279    \n",
       "16  Microbacterium_PAMC_28756_NZ_CP014313.1               2.868190    \n",
       "17  Zhihengliuella_ISTPL4_NZ_CP025422.1                   2.557703    \n",
       "18  Ralstonia_solanacearum_NZ_CP088237.1                  2.362988    \n",
       "19  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  2.030496    \n",
       "20  Ralstonia_solanacearum_NZ_CP088237.1                  1.982340    \n",
       "21  Microbacterium_paraoxydans_NZ_LT629770.1              1.957119    \n",
       "22  Microbacterium_PAMC_28756_NZ_CP014313.1               1.909744    \n",
       "23  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  1.800490    \n",
       "24  Zhihengliuella_ISTPL4_NZ_CP025422.1                   1.762940    \n",
       "25  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  1.569093    \n",
       "26  Microbacterium_Y-01_NZ_CP024170.1                     1.527854    \n",
       "27  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  1.391352    \n",
       "28  Microbacterium_hominis_NZ_CP061344.1                  1.189470    \n",
       "29  Burkholderia_contaminans_GCF_004723625.1_ASM472362v1  1.187561    \n",
       "30  Microbacterium_hominis_NZ_CP061344.1                  1.127352    \n",
       "⋮   ⋮                                                    ⋮            \n",
       "121 Human_mastadenovirus_C_NC_001405.1                   0            \n",
       "122 Human_mastadenovirus_C_NC_001405.1                   0            \n",
       "123 Human_mastadenovirus_C_NC_001405.1                   0            \n",
       "124 Human_mastadenovirus_C_NC_001405.1                   0            \n",
       "125 Lactobacillus_crispatus_NZ_CP039266.1                0            \n",
       "126 Microbacterium_PAMC_28756_NZ_CP014313.1              0            \n",
       "127 Microbacterium_PAMC_28756_NZ_CP014313.1              0            \n",
       "128 Microbacterium_PM5_NZ_CP022162.1                     0            \n",
       "129 Microbacterium_PM5_NZ_CP022162.1                     0            \n",
       "130 Microbacterium_Y-01_NZ_CP024170.1                    0            \n",
       "131 Microbacterium_Y-01_NZ_CP024170.1                    0            \n",
       "132 Microbacterium_hominis_NZ_CP061344.1                 0            \n",
       "133 Microbacterium_hominis_NZ_CP061344.1                 0            \n",
       "134 Microbacterium_paraoxydans_NZ_LT629770.1             0            \n",
       "135 Microbacterium_paraoxydans_NZ_LT629770.1             0            \n",
       "136 Neisseria_subflava_NZ_CP039887.1                     0            \n",
       "137 Neisseria_subflava_NZ_CP039887.1                     0            \n",
       "138 Paracoccus_yeei_NZ_CP024422.1                        0            \n",
       "139 Pseudomonas_mendocina_NZ_CP013124.1                  0            \n",
       "140 Ralstonia_solanacearum_NZ_CP088237.1                 0            \n",
       "141 Rickettsia_felis_NZ_JSEL01000013.1                   0            \n",
       "142 Rickettsia_felis_NZ_JSEL01000013.1                   0            \n",
       "143 Rickettsia_felis_NZ_JSEL01000013.1                   0            \n",
       "144 Rickettsia_felis_NZ_JSEL01000013.1                   0            \n",
       "145 Rickettsia_felis_NZ_JSEL01000013.1                   0            \n",
       "146 Staphylococcus_cohnii_NZ_UHDA01000001.1              0            \n",
       "147 Staphylococcus_epidermidis_NZ_CP035288.1             0            \n",
       "148 Staphylococcus_haemolyticus_NZ_CP013911.1            0            \n",
       "149 Zhihengliuella_ISTPL4_NZ_CP025422.1                  0            \n",
       "150 Zhihengliuella_ISTPL4_NZ_CP025422.1                  0            \n",
       "    perc_covered5\n",
       "1   6.295949e+01 \n",
       "2   1.854962e+01 \n",
       "3   5.118415e+01 \n",
       "4   1.331145e+01 \n",
       "5   3.450724e+01 \n",
       "6   4.117875e+00 \n",
       "7   1.994434e+00 \n",
       "8   1.075749e-01 \n",
       "9   4.389527e-02 \n",
       "10  1.213823e+00 \n",
       "11  3.543847e-01 \n",
       "12  7.080305e-02 \n",
       "13  5.878977e-01 \n",
       "14  1.977202e-04 \n",
       "15  4.653306e-02 \n",
       "16  4.893954e-02 \n",
       "17  6.091119e-02 \n",
       "18  5.853346e-01 \n",
       "19  5.749993e-01 \n",
       "20  2.856937e-01 \n",
       "21  5.241655e-02 \n",
       "22  5.885742e-02 \n",
       "23  5.098658e-01 \n",
       "24  6.822053e-02 \n",
       "25  2.784927e-01 \n",
       "26  1.669957e-02 \n",
       "27  2.469462e-01 \n",
       "28  5.363001e-02 \n",
       "29  1.044617e-01 \n",
       "30  9.451426e-02 \n",
       "⋮   ⋮            \n",
       "121 0            \n",
       "122 0            \n",
       "123 0            \n",
       "124 0            \n",
       "125 0            \n",
       "126 0            \n",
       "127 0            \n",
       "128 0            \n",
       "129 0            \n",
       "130 0            \n",
       "131 0            \n",
       "132 0            \n",
       "133 0            \n",
       "134 0            \n",
       "135 0            \n",
       "136 0            \n",
       "137 0            \n",
       "138 0            \n",
       "139 0            \n",
       "140 0            \n",
       "141 0            \n",
       "142 0            \n",
       "143 0            \n",
       "144 0            \n",
       "145 0            \n",
       "146 0            \n",
       "147 0            \n",
       "148 0            \n",
       "149 0            \n",
       "150 0            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cov_df %>%\n",
    "    arrange(desc(perc_covered1))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
