{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fe3f3f5c",
   "metadata": {},
   "source": [
    "# Identication of putative contaminant taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "440f3650",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: foreach\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.5     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.0.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32maccumulate()\u001b[39m masks \u001b[34mforeach\u001b[39m::accumulate()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mwhen()\u001b[39m       masks \u001b[34mforeach\u001b[39m::when()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "Loading required package: compositions\n",
      "\n",
      "Welcome to compositions, a package for compositional data analysis.\n",
      "Find an intro with \"? compositions\"\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘compositions’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    anova, cor, cov, dist, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    %*%, norm, scale, scale.default\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n",
      "Loading required package: parallel\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(foreach)\n",
    "require(tidyverse)\n",
    "require(ggplot2)\n",
    "require(data.table)\n",
    "require(compositions)\n",
    "require(doParallel)\n",
    "registerDoParallel(cores=8)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "215972d1",
   "metadata": {},
   "source": [
    "### Data pre-processing functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b5109a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "load_data <- function(file_path) {\n",
    "    df <- as.data.frame(fread(file_path)) %>%\n",
    "        separate(sample, into = c(NA, \"npm_research_id\"), sep = \"\\\\.\")\n",
    "    return(df)\n",
    "}\n",
    "\n",
    "\n",
    "load_metadata <- function(file_path, df) {\n",
    "    meta <- fread(file_path, na.strings=c(\"\", NA))\n",
    "    meta <- meta %>% \n",
    "        filter(npm_research_id %in% df$npm_research_id) %>%\n",
    "        select(-removal_requested_by_supplier) %>%\n",
    "        replace(is.na(.), \"unknown\")\n",
    "    return(meta)\n",
    "}\n",
    "\n",
    "\n",
    "subset_metadata <- function(meta, n_subset) {\n",
    "    meta <- as.data.frame(meta)\n",
    "    cohorts <- unique(meta$site_supplying_sample)\n",
    "    subset_vec <- c()\n",
    "\n",
    "    for (i in cohorts) {\n",
    "        ids <- meta$npm_research_id[meta$site_supplying_sample == i]\n",
    "        \n",
    "        if (length(ids) > n_subset) {\n",
    "            subset_ids <- sample(ids, n_subset)\n",
    "            subset_vec <- c(subset_vec, subset_ids)\n",
    "        } else {\n",
    "            subset_vec <- c(subset_vec, ids)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    meta_sub <- meta %>%\n",
    "      filter(npm_research_id %in% subset_vec) \n",
    "    return(meta_sub)\n",
    "}\n",
    "\n",
    "\n",
    "retrieve_rows_from_meta <- function(df, meta) {\n",
    "    return(df %>% filter(npm_research_id %in% meta$npm_research_id)) \n",
    "}\n",
    "\n",
    "\n",
    "remove_cols <- function(df, col_to_exclude) {\n",
    "    return(df %>% select(-all_of(col_to_exclude)))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_prev_taxa <- function(df, frac_presence, read_threshold) {\n",
    "    n_original <- ncol(df[, colnames(df) != \"npm_research_id\"])\n",
    "    PA_df <- apply(df[, 2:ncol(df)], 2, function(x) {ifelse(x > read_threshold, T, F)})\n",
    "    frac_df <- apply(PA_df, 2, function(x) {sum(x) / nrow(PA_df)})\n",
    "    to_keep <- names(frac_df[frac_df > frac_presence])\n",
    "    to_keep <- c(\"npm_research_id\", to_keep)\n",
    "    n_new <- length(to_keep) - 1\n",
    "    print(str_glue(\"{n_new} / {n_original} taxa are present in {frac_presence} of samples\"))\n",
    "    return(df %>% select(all_of(to_keep)))\n",
    "}\n",
    "\n",
    "\n",
    "otu_to_RA <- function(df) {\n",
    "    mat <- as.matrix(df[, colnames(df) != \"npm_research_id\"])\n",
    "    RA_df <- as.data.frame(mat / rowSums(mat))\n",
    "    RA_df <- add_column(RA_df, df$npm_research_id, .before = 1)\n",
    "    colnames(RA_df)[1] <- \"npm_research_id\"\n",
    "    \n",
    "    return(RA_df)\n",
    "}\n",
    "\n",
    "\n",
    "RA_to_clr <- function(df) {\n",
    "    mat <- df_filt[, colnames(df_filt) != \"npm_research_id\"]\n",
    "    clr_df <- clr(mat)\n",
    "    return(cbind(data.frame(npm_research_id = df$npm_research_id), as.data.frame(clr_df, check.names = F)))\n",
    "}\n",
    "\n",
    "\n",
    "get_metadata_plots <- function(meta, meta_cols) {\n",
    "    meta <- as.matrix(meta)\n",
    "    meta <- as_tibble(meta, rownames = \"sample\")\n",
    "\n",
    "    plots <- list()\n",
    "    \n",
    "    for (column in meta_cols) {\n",
    "        plt <- meta %>%\n",
    "            mutate(across(everything(), as.character)) %>%\n",
    "            select(all_of(column)) %>%\n",
    "            group_by_at(column) %>%\n",
    "            summarise(n = n()) %>%\n",
    "            ggplot(aes_string(x = column, y = \"n\", fill = column)) +\n",
    "                geom_bar(stat = \"identity\") +\n",
    "                theme(axis.text.x = element_text(angle = 45, hjust = 1)) +\n",
    "                geom_text(aes_string(label = \"n\"))\n",
    "        plots[[column]] <- plt\n",
    "    }\n",
    "    \n",
    "    return(plots)\n",
    "}\n",
    "\n",
    "\n",
    "get_meta_cols <- function(meta) {\n",
    "    meta_cols <- colnames(meta)[grep(\"kit|flow_cell|instrument_id|site_supplying\", colnames(meta))]\n",
    "    to_exclude <- c(\"library_prep_kit\", \"hiseq_xtm_flow_cell_v2_5_id\")\n",
    "    meta_cols <- meta_cols[!(meta_cols %in% to_exclude)]\n",
    "    return(meta_cols)\n",
    "}\n",
    "\n",
    "\n",
    "print_group_freqs <- function(meta, meta_cols) {\n",
    "    for (col in meta_cols) {\n",
    "        tmp <- tibble(data.frame(meta)) %>%\n",
    "            group_by(get(col)) %>%\n",
    "            summarise(n = n())\n",
    "        print(col)\n",
    "        print(tmp)\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "get_batch_prevalence <- function(x) {sum(x) / length(x)}\n",
    "\n",
    "\n",
    "remove_low_read_samples <- function(df) {\n",
    "    to_retain <- fread(\"data/samples_above_10_reads.txt\")$npm_research_id\n",
    "    return(df %>% filter(npm_research_id %in% to_retain))\n",
    "}\n",
    "\n",
    "\n",
    "remove_low_sample_levels <- function(dat, metadat, column, min_samples) {\n",
    "    # Remove levels with < x samples\n",
    "    tmp <- tibble(data.frame(metadat)) %>%\n",
    "            group_by(get(column)) %>%\n",
    "            summarise(n = n())\n",
    "\n",
    "    # Vector of group levels to keep\n",
    "    to_keep <- tmp[tmp$n >= min_samples, \"get(column)\"]$`get(column)`\n",
    "    to_keep <- to_keep[to_keep != \"Unknown\"]\n",
    "    n_levels <- length(to_keep)\n",
    "\n",
    "    if (n_levels < 2) {\n",
    "        print(str_glue(\"After pruning, {column} has < 2 levels\"))\n",
    "    } else {\n",
    "        print(str_glue(\"After pruning, {column} has {n_levels} levels\"))\n",
    "    }\n",
    "\n",
    "    # Remove rows in metadata\n",
    "    metadat_filt <- metadat %>% filter(get(column) %in% to_keep)\n",
    "\n",
    "    # Retrieve rows\n",
    "    dat_filt <- retrieve_rows_from_meta(dat, metadat_filt)\n",
    "    metadat_filt <- metadat_filt %>% filter(npm_research_id %in% dat_filt$npm_research_id)\n",
    "\n",
    "    return(list(dat_filt = dat_filt, metadat_filt = metadat_filt))\n",
    "}\n",
    "\n",
    "\n",
    "remove_empty_rows <- function(df) {\n",
    "    mat <- as.matrix(df[, colnames(df) != \"npm_research_id\"])\n",
    "    row_sums <- rowSums(mat)\n",
    "    df_filt <- df[row_sums != 0, ]\n",
    "    nrow(df_filt)\n",
    "\n",
    "    n_original <- nrow(df)\n",
    "    n_removed <- n_original - nrow(df_filt)\n",
    "    \n",
    "    print(str_glue(\"{n_removed}/{n_original} samples removed due to having no reads of interest\"))\n",
    "    \n",
    "    return(df_filt)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "363b9d21",
   "metadata": {},
   "source": [
    "#### Functions for testing differential abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c0e346fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "get_diff_abn <- function(dat_filt, metadat_filt, taxon, column, lq_p = 0.25, hq_p = 0.75, min_samples, fold_threshold = 0) {\n",
    "    # Remove zeros in CLR values\n",
    "    q_df <- dat_filt %>%\n",
    "        left_join(metadat_filt, by = \"npm_research_id\") %>%\n",
    "        select(all_of(c(taxon, column))) %>%\n",
    "        filter(get(taxon) != 0) %>%\n",
    "        group_by(get(column)) %>%\n",
    "        summarise(lq = quantile(get(taxon), probs = lq_p),\n",
    "                  hq = quantile(get(taxon), probs = hq_p),\n",
    "                  n_samples = n()) %>%\n",
    "    filter(n_samples > min_samples)  # Remove levels that have less than {min_samples} no. of non-zero samples\n",
    "    \n",
    "    # Analyse only if there are more than two valid levels for analysis\n",
    "    if (nrow(q_df) >= 2) {\n",
    "        # Do pairwise comparisons between levels\n",
    "        pairwise_results <- foreach(idx = seq(nrow(q_df))) %do% {\n",
    "            level1 <- q_df[idx, ]\n",
    "            level1_name <- level1$`get(column)`\n",
    "            lowest_lq <- level1$lq\n",
    "            level1_n_samples <- level1$n_samples\n",
    "            \n",
    "            diff_df <- q_df %>% \n",
    "                mutate(diff_abn = ifelse(hq + fold_threshold < lowest_lq, 1, 0)) %>%\n",
    "                rename(\"level2\" = \"get(column)\")\n",
    "\n",
    "#             if(any(diff_df$diff_abn == 1)) {\n",
    "            res <- diff_df %>%\n",
    "#                     filter(diff_abn == 1) %>%\n",
    "                mutate(taxa = taxon, meta_col = column, level1 = level1_name, .before = 1) %>%\n",
    "                mutate(level1_n_samples = level1_n_samples, .before = 7) %>%\n",
    "#                     select(-diff_abn) %>% \n",
    "                rename(level1_lq = lq, level2_hq = hq, level2_n_samples = n_samples)\n",
    "\n",
    "            return(res)\n",
    "#             }\n",
    "        }\n",
    "        \n",
    "        return(bind_rows(pairwise_results))  # Return only comparisons which show differential abundance\n",
    "        \n",
    "    } else {\n",
    "        return(tibble(taxa = taxon, meta_col = column, \n",
    "                      level1 = NA, level2 = NA, level1_lq = NA, level2_hq = NA, \n",
    "                      level1_n_samples = NA, level2_n_samples = NA, diff_abn = NA))\n",
    "    }\n",
    "}\n",
    "\n",
    "\n",
    "decontaminate_this_col <- function(dat, metadat, column, min_samples, lq_p, hq_p, fold_threshold) {\n",
    "    data_list <- remove_low_sample_levels(dat, metadat, column, min_samples)\n",
    "    metadat_filt <- data_list[[\"metadat_filt\"]]\n",
    "    dat_filt <- data_list[[\"dat_filt\"]]\n",
    "    taxa <- colnames(dat_filt %>% select(-npm_research_id))\n",
    "    \n",
    "    # Get differential abundance results for all taxa\n",
    "    col_results <- foreach(taxon = taxa) %dopar% {\n",
    "        taxon_results <- get_diff_abn(dat_filt, metadat_filt, taxon, column, \n",
    "                                      lq_p = lq_p, hq_p = hq_p, \n",
    "                                      min_samples = min_samples, \n",
    "                                      fold_threshold = fold_threshold)\n",
    "    }\n",
    "\n",
    "    return(bind_rows(col_results))\n",
    "}\n",
    "\n",
    "decontaminate_all_cols <- function(df_filt, meta_filt, meta_cols, min_samples, lq_p, hq_p, fold_threshold) {\n",
    "    results <- foreach(column = meta_cols) %do% {\n",
    "        col_res <- decontaminate_this_col(df_filt, meta_filt, column, \n",
    "                                          min_samples = min_samples, \n",
    "                                          lq_p = lq_p, \n",
    "                                          hq_p = hq_p, \n",
    "                                          fold_threshold = fold_threshold)\n",
    "        return(col_res)\n",
    "    }\n",
    "    \n",
    "    return(bind_rows(results))\n",
    "}\n",
    "\n",
    "\n",
    "parse_results <- function(result_df, meta_cols) {\n",
    "    parsed_df <- result_df %>% \n",
    "        group_by(taxa, meta_col) %>%\n",
    "        summarise(n_na = sum(is.na(diff_abn)),\n",
    "                  n_diff = ifelse(sum(diff_abn, na.rm = T) > 0, 1, 0)) %>%\n",
    "        group_by(taxa) %>%\n",
    "        summarise(n_na = sum(n_na), n_diff = sum(n_diff))\n",
    "\n",
    "    nc <- parsed_df %>% filter(n_na < length(meta_cols), n_diff == 0) %>% select(taxa)\n",
    "    contams <- parsed_df %>% filter(n_na < length(meta_cols), n_diff != 0) %>% select(taxa)\n",
    "    unknowns <- parsed_df %>% filter(n_na == length(meta_cols)) %>% select(taxa)\n",
    "    \n",
    "    return(list(non_contaminants = nc, contaminants = contams, unknowns = unknowns))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bc78537",
   "metadata": {},
   "source": [
    "## Main"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a2662ae",
   "metadata": {},
   "source": [
    "#### Params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1cc062e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "min_samples <- 10\n",
    "n_subset <- 9999\n",
    "human <- \"Homo sapiens\"\n",
    "rank <- \"S\"\n",
    "n <- 9999\n",
    "frac_presence <- 0\n",
    "lq_p <- 0.10\n",
    "hq_p <- 0.90\n",
    "fold_threshold <- 0\n",
    "read_threshold <- 10"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33dff39c",
   "metadata": {},
   "source": [
    "#### Load and parse data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6fd9cdfa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2183 / 5199 taxa are present in 0 of samples\n"
     ]
    }
   ],
   "source": [
    "to_retain <- fread(\"data/samples_above_95_reads.txt\")$npm_research_id\n",
    "df <- load_data(str_glue(\"data/temp_files_{n_subset}/07_abundance_matrix/abundance_matrix.subset_{n_subset}.{rank}.tsv\")) %>% \n",
    "    filter(npm_research_id %in% to_retain)\n",
    "meta <- load_metadata(\"data/SG10K_Health_metadata.n10714.16March2021.parsed.csv\", df)\n",
    "\n",
    "# Get metadata subset\n",
    "meta_filt <- subset_metadata(meta, n)\n",
    "\n",
    "# Filter data\n",
    "df_filt <- retrieve_rows_from_meta(df, meta_filt)\n",
    "df_filt <- remove_cols(df_filt, c(human, \"unclassified\"))\n",
    "df_filt <- remove_low_prev_taxa(df_filt, frac_presence = frac_presence, read_threshold = 10)\n",
    "df_filt <- otu_to_RA(df_filt)\n",
    "df_filt <- RA_to_clr(df_filt)\n",
    "\n",
    "# Get metadata columns of interest\n",
    "meta_cols <- get_meta_cols(meta_filt)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b933bc1",
   "metadata": {},
   "source": [
    "#### Run decontamination"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "79bcc6b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After pruning, site_supplying_sample has 6 levels\n",
      "After pruning, extraction_kit has 6 levels\n",
      "After pruning, instrument_id has 5 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_1of_2__lot has 33 levels\n",
      "After pruning, hiseq_xtm_sbs_kit_300_cycles_v2__box_2_of_2__lot has 30 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_1_of_2__lot has 27 levels\n",
      "After pruning, hiseq_xtm_pe_cluster_kit_cbottm_v2__box_2_of_2__lot has 30 levels\n",
      "After pruning, hiseq_xtm_flow_cell_v2_5_lot has 34 levels\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "    user   system  elapsed \n",
       "6294.672  206.907  835.888 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>taxa</th><th scope=col>meta_col</th><th scope=col>level1</th><th scope=col>level2</th><th scope=col>level1_lq</th><th scope=col>level2_hq</th><th scope=col>level1_n_samples</th><th scope=col>level2_n_samples</th><th scope=col>diff_abn</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Paraburkholderia fungorum</td><td>site_supplying_sample</td><td>GUSTO</td><td>GUSTO </td><td>-0.8759667</td><td>0.3737458</td><td>97</td><td>  97</td><td>0</td></tr>\n",
       "\t<tr><td>Paraburkholderia fungorum</td><td>site_supplying_sample</td><td>GUSTO</td><td>HELIOS</td><td>-1.0322213</td><td>0.1751339</td><td>97</td><td> 756</td><td>0</td></tr>\n",
       "\t<tr><td>Paraburkholderia fungorum</td><td>site_supplying_sample</td><td>GUSTO</td><td>MEC   </td><td>-1.2934842</td><td>2.3950029</td><td>97</td><td>1074</td><td>0</td></tr>\n",
       "\t<tr><td>Paraburkholderia fungorum</td><td>site_supplying_sample</td><td>GUSTO</td><td>PRISM </td><td>-1.7155051</td><td>1.4935845</td><td>97</td><td> 956</td><td>0</td></tr>\n",
       "\t<tr><td>Paraburkholderia fungorum</td><td>site_supplying_sample</td><td>GUSTO</td><td>SERI  </td><td>-0.3722952</td><td>3.3452331</td><td>97</td><td> 663</td><td>0</td></tr>\n",
       "\t<tr><td>Paraburkholderia fungorum</td><td>site_supplying_sample</td><td>GUSTO</td><td>TTSH  </td><td>-0.6115676</td><td>5.3671477</td><td>97</td><td> 297</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " taxa & meta\\_col & level1 & level2 & level1\\_lq & level2\\_hq & level1\\_n\\_samples & level2\\_n\\_samples & diff\\_abn\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Paraburkholderia fungorum & site\\_supplying\\_sample & GUSTO & GUSTO  & -0.8759667 & 0.3737458 & 97 &   97 & 0\\\\\n",
       "\t Paraburkholderia fungorum & site\\_supplying\\_sample & GUSTO & HELIOS & -1.0322213 & 0.1751339 & 97 &  756 & 0\\\\\n",
       "\t Paraburkholderia fungorum & site\\_supplying\\_sample & GUSTO & MEC    & -1.2934842 & 2.3950029 & 97 & 1074 & 0\\\\\n",
       "\t Paraburkholderia fungorum & site\\_supplying\\_sample & GUSTO & PRISM  & -1.7155051 & 1.4935845 & 97 &  956 & 0\\\\\n",
       "\t Paraburkholderia fungorum & site\\_supplying\\_sample & GUSTO & SERI   & -0.3722952 & 3.3452331 & 97 &  663 & 0\\\\\n",
       "\t Paraburkholderia fungorum & site\\_supplying\\_sample & GUSTO & TTSH   & -0.6115676 & 5.3671477 & 97 &  297 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 9\n",
       "\n",
       "| taxa &lt;chr&gt; | meta_col &lt;chr&gt; | level1 &lt;chr&gt; | level2 &lt;chr&gt; | level1_lq &lt;dbl&gt; | level2_hq &lt;dbl&gt; | level1_n_samples &lt;int&gt; | level2_n_samples &lt;int&gt; | diff_abn &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Paraburkholderia fungorum | site_supplying_sample | GUSTO | GUSTO  | -0.8759667 | 0.3737458 | 97 |   97 | 0 |\n",
       "| Paraburkholderia fungorum | site_supplying_sample | GUSTO | HELIOS | -1.0322213 | 0.1751339 | 97 |  756 | 0 |\n",
       "| Paraburkholderia fungorum | site_supplying_sample | GUSTO | MEC    | -1.2934842 | 2.3950029 | 97 | 1074 | 0 |\n",
       "| Paraburkholderia fungorum | site_supplying_sample | GUSTO | PRISM  | -1.7155051 | 1.4935845 | 97 |  956 | 0 |\n",
       "| Paraburkholderia fungorum | site_supplying_sample | GUSTO | SERI   | -0.3722952 | 3.3452331 | 97 |  663 | 0 |\n",
       "| Paraburkholderia fungorum | site_supplying_sample | GUSTO | TTSH   | -0.6115676 | 5.3671477 | 97 |  297 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  taxa                      meta_col              level1 level2 level1_lq \n",
       "1 Paraburkholderia fungorum site_supplying_sample GUSTO  GUSTO  -0.8759667\n",
       "2 Paraburkholderia fungorum site_supplying_sample GUSTO  HELIOS -1.0322213\n",
       "3 Paraburkholderia fungorum site_supplying_sample GUSTO  MEC    -1.2934842\n",
       "4 Paraburkholderia fungorum site_supplying_sample GUSTO  PRISM  -1.7155051\n",
       "5 Paraburkholderia fungorum site_supplying_sample GUSTO  SERI   -0.3722952\n",
       "6 Paraburkholderia fungorum site_supplying_sample GUSTO  TTSH   -0.6115676\n",
       "  level2_hq level1_n_samples level2_n_samples diff_abn\n",
       "1 0.3737458 97                 97             0       \n",
       "2 0.1751339 97                756             0       \n",
       "3 2.3950029 97               1074             0       \n",
       "4 1.4935845 97                956             0       \n",
       "5 3.3452331 97                663             0       \n",
       "6 5.3671477 97                297             0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Run decontamination\n",
    "system.time(result_df <- decontaminate_all_cols(df_filt, meta_filt, meta_cols, min_samples, lq_p, hq_p, fold_threshold))\n",
    "head(result_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4aa8259f",
   "metadata": {},
   "source": [
    "### Parse results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3f1d7d0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'taxa'. You can override using the `.groups` argument.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "parsed_res <- parse_results(result_df, meta_cols)\n",
    "\n",
    "fwrite(parsed_res[[\"non_contaminants\"]], str_glue(\"results/decontamination/clr_decontamination/nc.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"))\n",
    "fwrite(parsed_res[[\"contaminants\"]], str_glue(\"results/decontamination/clr_decontamination/contams.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"))\n",
    "fwrite(parsed_res[[\"unknowns\"]], str_glue(\"results/decontamination/clr_decontamination/unknowns.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"))\n",
    "fwrite(result_df, str_glue(\"results/decontamination/clr_decontamination/raw_results.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "642368b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initial no. of species = 5199\n",
      "Species present in at least one sample (reads > 10) = 2183\n",
      "After differential abundance filter (min_samples = 10) = 387\n",
      "Species that could not be analysed = 141\n"
     ]
    }
   ],
   "source": [
    "all_species <- colnames(df %>% select(-npm_research_id, -\"Homo sapiens\", -\"unclassified\"))\n",
    "initial <- length(all_species)\n",
    "non_zero <- length(df_filt) - 1\n",
    "after_diff <- nrow(parsed_res[[\"non_contaminants\"]])\n",
    "n_unknowns <- nrow(parsed_res[[\"unknowns\"]])\n",
    "\n",
    "print(str_glue(\"Initial no. of species = {initial}\"))\n",
    "print(str_glue(\"Species present in at least one sample (reads > {read_threshold}) = {non_zero}\"))\n",
    "print(str_glue(\"After differential abundance filter (min_samples = {min_samples}) = {after_diff}\"))\n",
    "print(str_glue(\"Species that could not be analysed = {n_unknowns}\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cc1f3c7e",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'afwfdwdawad' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'afwfdwdawad' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "afwfdwdawad"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be3d58a4",
   "metadata": {},
   "source": [
    "### Correlation filter after Diff. Prev. + Diff. Abn."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "852ca80b",
   "metadata": {},
   "outputs": [],
   "source": [
    "min_samples <- 10\n",
    "lq_p <- 0.15\n",
    "hq_p <- 0.85\n",
    "\n",
    "diff_abn_nc <- read.csv(str_glue(\"results/decontamination/clr_decontamination/nc.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"))$taxa\n",
    "diff_abn_contam <- read.csv(str_glue(\"results/decontamination/clr_decontamination/contams.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"))$taxa\n",
    "diff_prev_nc <- read.csv(\"results/decontamination/diff_prev_V2/noncontam.min_samples100.raw.read_t10.S.n9999.txt\")$non_contaminants\n",
    "\n",
    "nc <- intersect(diff_abn_nc, diff_prev_nc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0a968228",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_decontam <- function(df, non_contaminants, contaminants) {\n",
    "    nc_list <- foreach (non_contaminant_taxon = non_contaminants) %dopar% {\n",
    "        morsel <- tibble()\n",
    "        for (contaminant_taxon in contaminants) {\n",
    "            spearman_test <- cor.test(df[ ,contaminant_taxon], df[, non_contaminant_taxon])\n",
    "            rho <- spearman_test$estimate\n",
    "            print(str_glue(\"{non_contaminant_taxon} has correlation of {rho} with {contaminant_taxon}\"))\n",
    "            morsel <- morsel %>% bind_rows(tibble(non_contaminant_taxon = non_contaminant_taxon, \n",
    "                                           contaminant_taxon = contaminant_taxon,\n",
    "                                           rho = rho))\n",
    "        }\n",
    "\n",
    "        return(morsel)\n",
    "    }\n",
    "\n",
    "    corr_df <- bind_rows(nc_list)  \n",
    "    return(corr_df)\n",
    "}\n",
    "\n",
    "\n",
    "parse_corr_results <- function(corr_df, nc, corr_t) {\n",
    "    contaminants <- corr_df %>% \n",
    "        filter(rho > corr_t) %>%\n",
    "        distinct(non_contaminant_taxon) %>%\n",
    "        rename(contaminants = non_contaminant_taxon)\n",
    "    \n",
    "    non_contaminants <- tibble(non_contaminants = nc) %>% \n",
    "        filter(!(non_contaminants %in% contaminants$contaminants))\n",
    "    \n",
    "    return(list(non_contaminants = non_contaminants, contaminants = contaminants))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "ad84e36b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$non_contaminants</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 0 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>non_contaminants</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$contaminants</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A tibble: 184 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>contaminants</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Acidaminococcus intestini                           </td></tr>\n",
       "\t<tr><td>Acinetobacter equi                                  </td></tr>\n",
       "\t<tr><td>Acinetobacter lactucae                              </td></tr>\n",
       "\t<tr><td>Acinetobacter oleivorans                            </td></tr>\n",
       "\t<tr><td>Acinetobacter sp. WCHAc010052                       </td></tr>\n",
       "\t<tr><td>Actinoplanes friuliensis                            </td></tr>\n",
       "\t<tr><td>Aerococcus christensenii                            </td></tr>\n",
       "\t<tr><td>Aggregatibacter segnis                              </td></tr>\n",
       "\t<tr><td>Alloactinosynnema sp. L-07                          </td></tr>\n",
       "\t<tr><td>Amycolatopsis albispora                             </td></tr>\n",
       "\t<tr><td>Aquimarina sp. AD1                                  </td></tr>\n",
       "\t<tr><td>Aquimarina sp. AD10                                 </td></tr>\n",
       "\t<tr><td>Arachidicoccus sp. KIS59-12                         </td></tr>\n",
       "\t<tr><td>Arcobacter halophilus                               </td></tr>\n",
       "\t<tr><td>Atopobium parvulum                                  </td></tr>\n",
       "\t<tr><td>Auricoccus indicus                                  </td></tr>\n",
       "\t<tr><td>Autographa californica multiple nucleopolyhedrovirus</td></tr>\n",
       "\t<tr><td>Azoarcus sp. SY39                                   </td></tr>\n",
       "\t<tr><td>Bacillus mycoides                                   </td></tr>\n",
       "\t<tr><td>Bacteroides heparinolyticus                         </td></tr>\n",
       "\t<tr><td>Bernardetia litoralis                               </td></tr>\n",
       "\t<tr><td>Blautia hansenii                                    </td></tr>\n",
       "\t<tr><td>Blautia sp. N6H1-15                                 </td></tr>\n",
       "\t<tr><td>Bordetella genomosp. 8                              </td></tr>\n",
       "\t<tr><td>Burkholderia metallica                              </td></tr>\n",
       "\t<tr><td>Burkholderia thailandensis                          </td></tr>\n",
       "\t<tr><td>Burkholderia ubonensis                              </td></tr>\n",
       "\t<tr><td>Campylobacter hominis                               </td></tr>\n",
       "\t<tr><td>Campylobacter ureolyticus                           </td></tr>\n",
       "\t<tr><td>Candidatus Xiphinematobacter sp. Idaho Grape        </td></tr>\n",
       "\t<tr><td>⋮</td></tr>\n",
       "\t<tr><td>Streptomyces luteoverticillatus</td></tr>\n",
       "\t<tr><td>Streptomyces scabiei           </td></tr>\n",
       "\t<tr><td>Streptomyces sp. CMB-StM0423   </td></tr>\n",
       "\t<tr><td>Sulfurospirillum multivorans   </td></tr>\n",
       "\t<tr><td>Tamlana sp. UJ94               </td></tr>\n",
       "\t<tr><td>Tannerella forsythia           </td></tr>\n",
       "\t<tr><td>Tenacibaculum dicentrarchi     </td></tr>\n",
       "\t<tr><td>Tenacibaculum jejuense         </td></tr>\n",
       "\t<tr><td>Tenacibaculum maritimum        </td></tr>\n",
       "\t<tr><td>Tenacibaculum mesophilum       </td></tr>\n",
       "\t<tr><td>Tolumonas auensis              </td></tr>\n",
       "\t<tr><td>Torque teno mini virus 1       </td></tr>\n",
       "\t<tr><td>Torque teno mini virus 18      </td></tr>\n",
       "\t<tr><td>Torque teno mini virus 3       </td></tr>\n",
       "\t<tr><td>Torque teno mini virus 5       </td></tr>\n",
       "\t<tr><td>Torque teno mini virus 7       </td></tr>\n",
       "\t<tr><td>Torque teno virus 15           </td></tr>\n",
       "\t<tr><td>Torque teno virus 23           </td></tr>\n",
       "\t<tr><td>Torque teno virus 9            </td></tr>\n",
       "\t<tr><td>Treponema denticola            </td></tr>\n",
       "\t<tr><td>TTV-like mini virus            </td></tr>\n",
       "\t<tr><td>uncultured crAssphage          </td></tr>\n",
       "\t<tr><td>Ureaplasma urealyticum         </td></tr>\n",
       "\t<tr><td>Verrucosispora maris           </td></tr>\n",
       "\t<tr><td>Vibrio breoganii               </td></tr>\n",
       "\t<tr><td>Vibrio fluvialis               </td></tr>\n",
       "\t<tr><td>Winogradskyella sp. PC-19      </td></tr>\n",
       "\t<tr><td>Winogradskyella sp. RHA_55     </td></tr>\n",
       "\t<tr><td>Xanthomonas phaseoli           </td></tr>\n",
       "\t<tr><td>Xylanimonas cellulosilytica    </td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$non\\_contaminants] A tibble: 0 × 1\n",
       "\\begin{tabular}{l}\n",
       " non\\_contaminants\\\\\n",
       " <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$contaminants] A tibble: 184 × 1\n",
       "\\begin{tabular}{l}\n",
       " contaminants\\\\\n",
       " <chr>\\\\\n",
       "\\hline\n",
       "\t Acidaminococcus intestini                           \\\\\n",
       "\t Acinetobacter equi                                  \\\\\n",
       "\t Acinetobacter lactucae                              \\\\\n",
       "\t Acinetobacter oleivorans                            \\\\\n",
       "\t Acinetobacter sp. WCHAc010052                       \\\\\n",
       "\t Actinoplanes friuliensis                            \\\\\n",
       "\t Aerococcus christensenii                            \\\\\n",
       "\t Aggregatibacter segnis                              \\\\\n",
       "\t Alloactinosynnema sp. L-07                          \\\\\n",
       "\t Amycolatopsis albispora                             \\\\\n",
       "\t Aquimarina sp. AD1                                  \\\\\n",
       "\t Aquimarina sp. AD10                                 \\\\\n",
       "\t Arachidicoccus sp. KIS59-12                         \\\\\n",
       "\t Arcobacter halophilus                               \\\\\n",
       "\t Atopobium parvulum                                  \\\\\n",
       "\t Auricoccus indicus                                  \\\\\n",
       "\t Autographa californica multiple nucleopolyhedrovirus\\\\\n",
       "\t Azoarcus sp. SY39                                   \\\\\n",
       "\t Bacillus mycoides                                   \\\\\n",
       "\t Bacteroides heparinolyticus                         \\\\\n",
       "\t Bernardetia litoralis                               \\\\\n",
       "\t Blautia hansenii                                    \\\\\n",
       "\t Blautia sp. N6H1-15                                 \\\\\n",
       "\t Bordetella genomosp. 8                              \\\\\n",
       "\t Burkholderia metallica                              \\\\\n",
       "\t Burkholderia thailandensis                          \\\\\n",
       "\t Burkholderia ubonensis                              \\\\\n",
       "\t Campylobacter hominis                               \\\\\n",
       "\t Campylobacter ureolyticus                           \\\\\n",
       "\t Candidatus Xiphinematobacter sp. Idaho Grape        \\\\\n",
       "\t ⋮\\\\\n",
       "\t Streptomyces luteoverticillatus\\\\\n",
       "\t Streptomyces scabiei           \\\\\n",
       "\t Streptomyces sp. CMB-StM0423   \\\\\n",
       "\t Sulfurospirillum multivorans   \\\\\n",
       "\t Tamlana sp. UJ94               \\\\\n",
       "\t Tannerella forsythia           \\\\\n",
       "\t Tenacibaculum dicentrarchi     \\\\\n",
       "\t Tenacibaculum jejuense         \\\\\n",
       "\t Tenacibaculum maritimum        \\\\\n",
       "\t Tenacibaculum mesophilum       \\\\\n",
       "\t Tolumonas auensis              \\\\\n",
       "\t Torque teno mini virus 1       \\\\\n",
       "\t Torque teno mini virus 18      \\\\\n",
       "\t Torque teno mini virus 3       \\\\\n",
       "\t Torque teno mini virus 5       \\\\\n",
       "\t Torque teno mini virus 7       \\\\\n",
       "\t Torque teno virus 15           \\\\\n",
       "\t Torque teno virus 23           \\\\\n",
       "\t Torque teno virus 9            \\\\\n",
       "\t Treponema denticola            \\\\\n",
       "\t TTV-like mini virus            \\\\\n",
       "\t uncultured crAssphage          \\\\\n",
       "\t Ureaplasma urealyticum         \\\\\n",
       "\t Verrucosispora maris           \\\\\n",
       "\t Vibrio breoganii               \\\\\n",
       "\t Vibrio fluvialis               \\\\\n",
       "\t Winogradskyella sp. PC-19      \\\\\n",
       "\t Winogradskyella sp. RHA\\_55     \\\\\n",
       "\t Xanthomonas phaseoli           \\\\\n",
       "\t Xylanimonas cellulosilytica    \\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$non_contaminants\n",
       ":   \n",
       "A tibble: 0 × 1\n",
       "\n",
       "| non_contaminants &lt;chr&gt; |\n",
       "|---|\n",
       "\n",
       "\n",
       "$contaminants\n",
       ":   \n",
       "A tibble: 184 × 1\n",
       "\n",
       "| contaminants &lt;chr&gt; |\n",
       "|---|\n",
       "| Acidaminococcus intestini                            |\n",
       "| Acinetobacter equi                                   |\n",
       "| Acinetobacter lactucae                               |\n",
       "| Acinetobacter oleivorans                             |\n",
       "| Acinetobacter sp. WCHAc010052                        |\n",
       "| Actinoplanes friuliensis                             |\n",
       "| Aerococcus christensenii                             |\n",
       "| Aggregatibacter segnis                               |\n",
       "| Alloactinosynnema sp. L-07                           |\n",
       "| Amycolatopsis albispora                              |\n",
       "| Aquimarina sp. AD1                                   |\n",
       "| Aquimarina sp. AD10                                  |\n",
       "| Arachidicoccus sp. KIS59-12                          |\n",
       "| Arcobacter halophilus                                |\n",
       "| Atopobium parvulum                                   |\n",
       "| Auricoccus indicus                                   |\n",
       "| Autographa californica multiple nucleopolyhedrovirus |\n",
       "| Azoarcus sp. SY39                                    |\n",
       "| Bacillus mycoides                                    |\n",
       "| Bacteroides heparinolyticus                          |\n",
       "| Bernardetia litoralis                                |\n",
       "| Blautia hansenii                                     |\n",
       "| Blautia sp. N6H1-15                                  |\n",
       "| Bordetella genomosp. 8                               |\n",
       "| Burkholderia metallica                               |\n",
       "| Burkholderia thailandensis                           |\n",
       "| Burkholderia ubonensis                               |\n",
       "| Campylobacter hominis                                |\n",
       "| Campylobacter ureolyticus                            |\n",
       "| Candidatus Xiphinematobacter sp. Idaho Grape         |\n",
       "| ⋮ |\n",
       "| Streptomyces luteoverticillatus |\n",
       "| Streptomyces scabiei            |\n",
       "| Streptomyces sp. CMB-StM0423    |\n",
       "| Sulfurospirillum multivorans    |\n",
       "| Tamlana sp. UJ94                |\n",
       "| Tannerella forsythia            |\n",
       "| Tenacibaculum dicentrarchi      |\n",
       "| Tenacibaculum jejuense          |\n",
       "| Tenacibaculum maritimum         |\n",
       "| Tenacibaculum mesophilum        |\n",
       "| Tolumonas auensis               |\n",
       "| Torque teno mini virus 1        |\n",
       "| Torque teno mini virus 18       |\n",
       "| Torque teno mini virus 3        |\n",
       "| Torque teno mini virus 5        |\n",
       "| Torque teno mini virus 7        |\n",
       "| Torque teno virus 15            |\n",
       "| Torque teno virus 23            |\n",
       "| Torque teno virus 9             |\n",
       "| Treponema denticola             |\n",
       "| TTV-like mini virus             |\n",
       "| uncultured crAssphage           |\n",
       "| Ureaplasma urealyticum          |\n",
       "| Verrucosispora maris            |\n",
       "| Vibrio breoganii                |\n",
       "| Vibrio fluvialis                |\n",
       "| Winogradskyella sp. PC-19       |\n",
       "| Winogradskyella sp. RHA_55      |\n",
       "| Xanthomonas phaseoli            |\n",
       "| Xylanimonas cellulosilytica     |\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$non_contaminants\n",
       "\u001b[90m# A tibble: 0 × 1\u001b[39m\n",
       "\u001b[90m# … with 1 variable: non_contaminants <chr>\u001b[39m\n",
       "\n",
       "$contaminants\n",
       "\u001b[90m# A tibble: 184 × 1\u001b[39m\n",
       "   contaminants                 \n",
       "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                        \n",
       "\u001b[90m 1\u001b[39m Acidaminococcus intestini    \n",
       "\u001b[90m 2\u001b[39m Acinetobacter equi           \n",
       "\u001b[90m 3\u001b[39m Acinetobacter lactucae       \n",
       "\u001b[90m 4\u001b[39m Acinetobacter oleivorans     \n",
       "\u001b[90m 5\u001b[39m Acinetobacter sp. WCHAc010052\n",
       "\u001b[90m 6\u001b[39m Actinoplanes friuliensis     \n",
       "\u001b[90m 7\u001b[39m Aerococcus christensenii     \n",
       "\u001b[90m 8\u001b[39m Aggregatibacter segnis       \n",
       "\u001b[90m 9\u001b[39m Alloactinosynnema sp. L-07   \n",
       "\u001b[90m10\u001b[39m Amycolatopsis albispora      \n",
       "\u001b[90m# … with 174 more rows\u001b[39m\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "corr_res <- corr_decontam(df_filt, nc, diff_abn_contam)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "49fbbd1d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 184 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>non_contaminant_taxon</th><th scope=col>diff</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Acidaminococcus intestini                           </td><td>-1832</td></tr>\n",
       "\t<tr><td>Acinetobacter equi                                  </td><td>-1824</td></tr>\n",
       "\t<tr><td>Acinetobacter lactucae                              </td><td>-1850</td></tr>\n",
       "\t<tr><td>Acinetobacter oleivorans                            </td><td>-1840</td></tr>\n",
       "\t<tr><td>Acinetobacter sp. WCHAc010052                       </td><td>-1832</td></tr>\n",
       "\t<tr><td>Actinoplanes friuliensis                            </td><td>   46</td></tr>\n",
       "\t<tr><td>Aerococcus christensenii                            </td><td>-1848</td></tr>\n",
       "\t<tr><td>Aggregatibacter segnis                              </td><td>-1734</td></tr>\n",
       "\t<tr><td>Alloactinosynnema sp. L-07                          </td><td>  130</td></tr>\n",
       "\t<tr><td>Amycolatopsis albispora                             </td><td> -216</td></tr>\n",
       "\t<tr><td>Aquimarina sp. AD1                                  </td><td>-1856</td></tr>\n",
       "\t<tr><td>Aquimarina sp. AD10                                 </td><td>-1852</td></tr>\n",
       "\t<tr><td>Arachidicoccus sp. KIS59-12                         </td><td>-1818</td></tr>\n",
       "\t<tr><td>Arcobacter halophilus                               </td><td>-1852</td></tr>\n",
       "\t<tr><td>Atopobium parvulum                                  </td><td>-1808</td></tr>\n",
       "\t<tr><td>Auricoccus indicus                                  </td><td>-1854</td></tr>\n",
       "\t<tr><td>Autographa californica multiple nucleopolyhedrovirus</td><td>-1854</td></tr>\n",
       "\t<tr><td>Azoarcus sp. SY39                                   </td><td>  110</td></tr>\n",
       "\t<tr><td>Bacillus mycoides                                   </td><td>-1850</td></tr>\n",
       "\t<tr><td>Bacteroides heparinolyticus                         </td><td>-1804</td></tr>\n",
       "\t<tr><td>Bernardetia litoralis                               </td><td>-1852</td></tr>\n",
       "\t<tr><td>Blautia hansenii                                    </td><td>-1778</td></tr>\n",
       "\t<tr><td>Blautia sp. N6H1-15                                 </td><td>-1818</td></tr>\n",
       "\t<tr><td>Bordetella genomosp. 8                              </td><td>  -86</td></tr>\n",
       "\t<tr><td>Burkholderia metallica                              </td><td>   10</td></tr>\n",
       "\t<tr><td>Burkholderia thailandensis                          </td><td>   88</td></tr>\n",
       "\t<tr><td>Burkholderia ubonensis                              </td><td> -150</td></tr>\n",
       "\t<tr><td>Campylobacter hominis                               </td><td>-1844</td></tr>\n",
       "\t<tr><td>Campylobacter ureolyticus                           </td><td>-1820</td></tr>\n",
       "\t<tr><td>Candidatus Xiphinematobacter sp. Idaho Grape        </td><td>  120</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Streptomyces luteoverticillatus</td><td>   12</td></tr>\n",
       "\t<tr><td>Streptomyces scabiei           </td><td>  136</td></tr>\n",
       "\t<tr><td>Streptomyces sp. CMB-StM0423   </td><td>  142</td></tr>\n",
       "\t<tr><td>Sulfurospirillum multivorans   </td><td>-1008</td></tr>\n",
       "\t<tr><td>Tamlana sp. UJ94               </td><td>-1838</td></tr>\n",
       "\t<tr><td>Tannerella forsythia           </td><td>-1810</td></tr>\n",
       "\t<tr><td>Tenacibaculum dicentrarchi     </td><td>-1856</td></tr>\n",
       "\t<tr><td>Tenacibaculum jejuense         </td><td>-1856</td></tr>\n",
       "\t<tr><td>Tenacibaculum maritimum        </td><td>-1856</td></tr>\n",
       "\t<tr><td>Tenacibaculum mesophilum       </td><td>-1848</td></tr>\n",
       "\t<tr><td>Tolumonas auensis              </td><td>  222</td></tr>\n",
       "\t<tr><td>Torque teno mini virus 1       </td><td>-1844</td></tr>\n",
       "\t<tr><td>Torque teno mini virus 18      </td><td>-1842</td></tr>\n",
       "\t<tr><td>Torque teno mini virus 3       </td><td>-1856</td></tr>\n",
       "\t<tr><td>Torque teno mini virus 5       </td><td>-1850</td></tr>\n",
       "\t<tr><td>Torque teno mini virus 7       </td><td>-1852</td></tr>\n",
       "\t<tr><td>Torque teno virus 15           </td><td> -768</td></tr>\n",
       "\t<tr><td>Torque teno virus 23           </td><td>-1838</td></tr>\n",
       "\t<tr><td>Torque teno virus 9            </td><td>-1856</td></tr>\n",
       "\t<tr><td>Treponema denticola            </td><td>-1816</td></tr>\n",
       "\t<tr><td>TTV-like mini virus            </td><td>-1856</td></tr>\n",
       "\t<tr><td>uncultured crAssphage          </td><td>-1846</td></tr>\n",
       "\t<tr><td>Ureaplasma urealyticum         </td><td>-1840</td></tr>\n",
       "\t<tr><td>Verrucosispora maris           </td><td>  -74</td></tr>\n",
       "\t<tr><td>Vibrio breoganii               </td><td>-1854</td></tr>\n",
       "\t<tr><td>Vibrio fluvialis               </td><td>  218</td></tr>\n",
       "\t<tr><td>Winogradskyella sp. PC-19      </td><td>-1850</td></tr>\n",
       "\t<tr><td>Winogradskyella sp. RHA_55     </td><td>-1848</td></tr>\n",
       "\t<tr><td>Xanthomonas phaseoli           </td><td>   30</td></tr>\n",
       "\t<tr><td>Xylanimonas cellulosilytica    </td><td> -314</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 184 × 2\n",
       "\\begin{tabular}{ll}\n",
       " non\\_contaminant\\_taxon & diff\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Acidaminococcus intestini                            & -1832\\\\\n",
       "\t Acinetobacter equi                                   & -1824\\\\\n",
       "\t Acinetobacter lactucae                               & -1850\\\\\n",
       "\t Acinetobacter oleivorans                             & -1840\\\\\n",
       "\t Acinetobacter sp. WCHAc010052                        & -1832\\\\\n",
       "\t Actinoplanes friuliensis                             &    46\\\\\n",
       "\t Aerococcus christensenii                             & -1848\\\\\n",
       "\t Aggregatibacter segnis                               & -1734\\\\\n",
       "\t Alloactinosynnema sp. L-07                           &   130\\\\\n",
       "\t Amycolatopsis albispora                              &  -216\\\\\n",
       "\t Aquimarina sp. AD1                                   & -1856\\\\\n",
       "\t Aquimarina sp. AD10                                  & -1852\\\\\n",
       "\t Arachidicoccus sp. KIS59-12                          & -1818\\\\\n",
       "\t Arcobacter halophilus                                & -1852\\\\\n",
       "\t Atopobium parvulum                                   & -1808\\\\\n",
       "\t Auricoccus indicus                                   & -1854\\\\\n",
       "\t Autographa californica multiple nucleopolyhedrovirus & -1854\\\\\n",
       "\t Azoarcus sp. SY39                                    &   110\\\\\n",
       "\t Bacillus mycoides                                    & -1850\\\\\n",
       "\t Bacteroides heparinolyticus                          & -1804\\\\\n",
       "\t Bernardetia litoralis                                & -1852\\\\\n",
       "\t Blautia hansenii                                     & -1778\\\\\n",
       "\t Blautia sp. N6H1-15                                  & -1818\\\\\n",
       "\t Bordetella genomosp. 8                               &   -86\\\\\n",
       "\t Burkholderia metallica                               &    10\\\\\n",
       "\t Burkholderia thailandensis                           &    88\\\\\n",
       "\t Burkholderia ubonensis                               &  -150\\\\\n",
       "\t Campylobacter hominis                                & -1844\\\\\n",
       "\t Campylobacter ureolyticus                            & -1820\\\\\n",
       "\t Candidatus Xiphinematobacter sp. Idaho Grape         &   120\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t Streptomyces luteoverticillatus &    12\\\\\n",
       "\t Streptomyces scabiei            &   136\\\\\n",
       "\t Streptomyces sp. CMB-StM0423    &   142\\\\\n",
       "\t Sulfurospirillum multivorans    & -1008\\\\\n",
       "\t Tamlana sp. UJ94                & -1838\\\\\n",
       "\t Tannerella forsythia            & -1810\\\\\n",
       "\t Tenacibaculum dicentrarchi      & -1856\\\\\n",
       "\t Tenacibaculum jejuense          & -1856\\\\\n",
       "\t Tenacibaculum maritimum         & -1856\\\\\n",
       "\t Tenacibaculum mesophilum        & -1848\\\\\n",
       "\t Tolumonas auensis               &   222\\\\\n",
       "\t Torque teno mini virus 1        & -1844\\\\\n",
       "\t Torque teno mini virus 18       & -1842\\\\\n",
       "\t Torque teno mini virus 3        & -1856\\\\\n",
       "\t Torque teno mini virus 5        & -1850\\\\\n",
       "\t Torque teno mini virus 7        & -1852\\\\\n",
       "\t Torque teno virus 15            &  -768\\\\\n",
       "\t Torque teno virus 23            & -1838\\\\\n",
       "\t Torque teno virus 9             & -1856\\\\\n",
       "\t Treponema denticola             & -1816\\\\\n",
       "\t TTV-like mini virus             & -1856\\\\\n",
       "\t uncultured crAssphage           & -1846\\\\\n",
       "\t Ureaplasma urealyticum          & -1840\\\\\n",
       "\t Verrucosispora maris            &   -74\\\\\n",
       "\t Vibrio breoganii                & -1854\\\\\n",
       "\t Vibrio fluvialis                &   218\\\\\n",
       "\t Winogradskyella sp. PC-19       & -1850\\\\\n",
       "\t Winogradskyella sp. RHA\\_55      & -1848\\\\\n",
       "\t Xanthomonas phaseoli            &    30\\\\\n",
       "\t Xylanimonas cellulosilytica     &  -314\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 184 × 2\n",
       "\n",
       "| non_contaminant_taxon &lt;chr&gt; | diff &lt;int&gt; |\n",
       "|---|---|\n",
       "| Acidaminococcus intestini                            | -1832 |\n",
       "| Acinetobacter equi                                   | -1824 |\n",
       "| Acinetobacter lactucae                               | -1850 |\n",
       "| Acinetobacter oleivorans                             | -1840 |\n",
       "| Acinetobacter sp. WCHAc010052                        | -1832 |\n",
       "| Actinoplanes friuliensis                             |    46 |\n",
       "| Aerococcus christensenii                             | -1848 |\n",
       "| Aggregatibacter segnis                               | -1734 |\n",
       "| Alloactinosynnema sp. L-07                           |   130 |\n",
       "| Amycolatopsis albispora                              |  -216 |\n",
       "| Aquimarina sp. AD1                                   | -1856 |\n",
       "| Aquimarina sp. AD10                                  | -1852 |\n",
       "| Arachidicoccus sp. KIS59-12                          | -1818 |\n",
       "| Arcobacter halophilus                                | -1852 |\n",
       "| Atopobium parvulum                                   | -1808 |\n",
       "| Auricoccus indicus                                   | -1854 |\n",
       "| Autographa californica multiple nucleopolyhedrovirus | -1854 |\n",
       "| Azoarcus sp. SY39                                    |   110 |\n",
       "| Bacillus mycoides                                    | -1850 |\n",
       "| Bacteroides heparinolyticus                          | -1804 |\n",
       "| Bernardetia litoralis                                | -1852 |\n",
       "| Blautia hansenii                                     | -1778 |\n",
       "| Blautia sp. N6H1-15                                  | -1818 |\n",
       "| Bordetella genomosp. 8                               |   -86 |\n",
       "| Burkholderia metallica                               |    10 |\n",
       "| Burkholderia thailandensis                           |    88 |\n",
       "| Burkholderia ubonensis                               |  -150 |\n",
       "| Campylobacter hominis                                | -1844 |\n",
       "| Campylobacter ureolyticus                            | -1820 |\n",
       "| Candidatus Xiphinematobacter sp. Idaho Grape         |   120 |\n",
       "| ⋮ | ⋮ |\n",
       "| Streptomyces luteoverticillatus |    12 |\n",
       "| Streptomyces scabiei            |   136 |\n",
       "| Streptomyces sp. CMB-StM0423    |   142 |\n",
       "| Sulfurospirillum multivorans    | -1008 |\n",
       "| Tamlana sp. UJ94                | -1838 |\n",
       "| Tannerella forsythia            | -1810 |\n",
       "| Tenacibaculum dicentrarchi      | -1856 |\n",
       "| Tenacibaculum jejuense          | -1856 |\n",
       "| Tenacibaculum maritimum         | -1856 |\n",
       "| Tenacibaculum mesophilum        | -1848 |\n",
       "| Tolumonas auensis               |   222 |\n",
       "| Torque teno mini virus 1        | -1844 |\n",
       "| Torque teno mini virus 18       | -1842 |\n",
       "| Torque teno mini virus 3        | -1856 |\n",
       "| Torque teno mini virus 5        | -1850 |\n",
       "| Torque teno mini virus 7        | -1852 |\n",
       "| Torque teno virus 15            |  -768 |\n",
       "| Torque teno virus 23            | -1838 |\n",
       "| Torque teno virus 9             | -1856 |\n",
       "| Treponema denticola             | -1816 |\n",
       "| TTV-like mini virus             | -1856 |\n",
       "| uncultured crAssphage           | -1846 |\n",
       "| Ureaplasma urealyticum          | -1840 |\n",
       "| Verrucosispora maris            |   -74 |\n",
       "| Vibrio breoganii                | -1854 |\n",
       "| Vibrio fluvialis                |   218 |\n",
       "| Winogradskyella sp. PC-19       | -1850 |\n",
       "| Winogradskyella sp. RHA_55      | -1848 |\n",
       "| Xanthomonas phaseoli            |    30 |\n",
       "| Xylanimonas cellulosilytica     |  -314 |\n",
       "\n"
      ],
      "text/plain": [
       "    non_contaminant_taxon                                diff \n",
       "1   Acidaminococcus intestini                            -1832\n",
       "2   Acinetobacter equi                                   -1824\n",
       "3   Acinetobacter lactucae                               -1850\n",
       "4   Acinetobacter oleivorans                             -1840\n",
       "5   Acinetobacter sp. WCHAc010052                        -1832\n",
       "6   Actinoplanes friuliensis                                46\n",
       "7   Aerococcus christensenii                             -1848\n",
       "8   Aggregatibacter segnis                               -1734\n",
       "9   Alloactinosynnema sp. L-07                             130\n",
       "10  Amycolatopsis albispora                               -216\n",
       "11  Aquimarina sp. AD1                                   -1856\n",
       "12  Aquimarina sp. AD10                                  -1852\n",
       "13  Arachidicoccus sp. KIS59-12                          -1818\n",
       "14  Arcobacter halophilus                                -1852\n",
       "15  Atopobium parvulum                                   -1808\n",
       "16  Auricoccus indicus                                   -1854\n",
       "17  Autographa californica multiple nucleopolyhedrovirus -1854\n",
       "18  Azoarcus sp. SY39                                      110\n",
       "19  Bacillus mycoides                                    -1850\n",
       "20  Bacteroides heparinolyticus                          -1804\n",
       "21  Bernardetia litoralis                                -1852\n",
       "22  Blautia hansenii                                     -1778\n",
       "23  Blautia sp. N6H1-15                                  -1818\n",
       "24  Bordetella genomosp. 8                                 -86\n",
       "25  Burkholderia metallica                                  10\n",
       "26  Burkholderia thailandensis                              88\n",
       "27  Burkholderia ubonensis                                -150\n",
       "28  Campylobacter hominis                                -1844\n",
       "29  Campylobacter ureolyticus                            -1820\n",
       "30  Candidatus Xiphinematobacter sp. Idaho Grape           120\n",
       "⋮   ⋮                                                    ⋮    \n",
       "155 Streptomyces luteoverticillatus                         12\n",
       "156 Streptomyces scabiei                                   136\n",
       "157 Streptomyces sp. CMB-StM0423                           142\n",
       "158 Sulfurospirillum multivorans                         -1008\n",
       "159 Tamlana sp. UJ94                                     -1838\n",
       "160 Tannerella forsythia                                 -1810\n",
       "161 Tenacibaculum dicentrarchi                           -1856\n",
       "162 Tenacibaculum jejuense                               -1856\n",
       "163 Tenacibaculum maritimum                              -1856\n",
       "164 Tenacibaculum mesophilum                             -1848\n",
       "165 Tolumonas auensis                                      222\n",
       "166 Torque teno mini virus 1                             -1844\n",
       "167 Torque teno mini virus 18                            -1842\n",
       "168 Torque teno mini virus 3                             -1856\n",
       "169 Torque teno mini virus 5                             -1850\n",
       "170 Torque teno mini virus 7                             -1852\n",
       "171 Torque teno virus 15                                  -768\n",
       "172 Torque teno virus 23                                 -1838\n",
       "173 Torque teno virus 9                                  -1856\n",
       "174 Treponema denticola                                  -1816\n",
       "175 TTV-like mini virus                                  -1856\n",
       "176 uncultured crAssphage                                -1846\n",
       "177 Ureaplasma urealyticum                               -1840\n",
       "178 Verrucosispora maris                                   -74\n",
       "179 Vibrio breoganii                                     -1854\n",
       "180 Vibrio fluvialis                                       218\n",
       "181 Winogradskyella sp. PC-19                            -1850\n",
       "182 Winogradskyella sp. RHA_55                           -1848\n",
       "183 Xanthomonas phaseoli                                    30\n",
       "184 Xylanimonas cellulosilytica                           -314"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "corr_res %>%\n",
    "    group_by(non_contaminant_taxon) %>%\n",
    "    mutate(above_corr = ifelse(rho > 0.1, T, F)) %>%\n",
    "    summarise(diff = sum(above_corr) - sum(!(above_corr)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "82e68d08",
   "metadata": {},
   "outputs": [],
   "source": [
    "parse_corr_results <- function(corr_df, nc, corr_t) {\n",
    "    parsed <- corr_res %>%\n",
    "        group_by(non_contaminant_taxon) %>%\n",
    "        mutate(above_corr = ifelse(rho > corr_t, T, F)) %>%\n",
    "        summarise(diff = sum(above_corr) - sum(!(above_corr)))\n",
    "    \n",
    "    contaminants <- parsed %>%\n",
    "        filter(diff >= 0) %>%\n",
    "        distinct(non_contaminant_taxon) %>%\n",
    "        rename(contaminants = non_contaminant_taxon)\n",
    "    \n",
    "    non_contaminants <- parsed %>%\n",
    "        filter(diff < 0) %>%\n",
    "        distinct(non_contaminant_taxon)\n",
    "    \n",
    "    return(list(non_contaminants = non_contaminants, contaminants = contaminants))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "42888a1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 156\n",
      "[1] 28\n"
     ]
    }
   ],
   "source": [
    "corr_parsed <- parse_corr_results(corr_res, nc, corr_t = 0.1)\n",
    "corr_nc <- corr_parsed[[\"non_contaminants\"]]\n",
    "corr_c <- corr_parsed[[\"contaminants\"]]\n",
    "\n",
    "fwrite(corr_nc, \n",
    "       str_glue(\"results/decontamination/correlation_decontamination/nc.diff_prev.diff_abn.l{lq_p * 100}h{hq_p * 100}.corr_t{corr_t}.S.n9999.txt\"))\n",
    "fwrite(corr_c, \n",
    "       str_glue(\"results/decontamination/correlation_decontamination/contam.diff_prev.diff_abn.l{lq_p * 100}h{hq_p * 100}.corr_t{corr_t}.S.n9999.txt\"))\n",
    "\n",
    "print(length(corr_nc$non_contaminant_taxon))\n",
    "print(length(corr_c$contaminants))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2570e832",
   "metadata": {},
   "source": [
    "### Choosing  correlation threshold"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "002f90dc",
   "metadata": {},
   "source": [
    "#### Heatmap of correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "cd41320f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "hm_df <- corr_res %>%\n",
    "    mutate(fill_color = case_when(rho <= 0 ~ \"<0\",\n",
    "                     rho > 0 & rho <= 0.1 ~ \"0-10\",\n",
    "                     rho > 0.1 & rho <= 0.2 ~ \"10-20\",\n",
    "                     rho > 0.2 & rho <= 0.3 ~ \"20-30\",\n",
    "                     rho > 0.3 ~  \">30\")) %>%\n",
    "    mutate(fill_color = factor(fill_color, levels = c(\"<0\", \"0-10\", \"10-20\", \"20-30\", \">30\")))\n",
    "\n",
    "wide_df <- hm_df %>%\n",
    "    pivot_wider(!fill_color, names_from = non_contaminant_taxon, values_from = rho) %>%\n",
    "    column_to_rownames(\"contaminant_taxon\")\n",
    "\n",
    "data <- scale(t(wide_df))\n",
    "ord_row <- hclust(dist(data, method = \"euclidean\"), method = \"ward.D\")$order\n",
    "\n",
    "data <- scale(wide_df)\n",
    "ord_col <- hclust(dist(data, method = \"euclidean\"), method = \"ward.D\")$order\n",
    "\n",
    "hm <- corr_res %>%\n",
    "    mutate(fill_color = case_when(rho <= 0 ~ \"<0\",\n",
    "                     rho > 0 & rho <= 0.1 ~ \"0-10\",\n",
    "                     rho > 0.1 & rho <= 0.2 ~ \"10-20\",\n",
    "                     rho > 0.2 & rho <= 0.3 ~ \"20-30\",\n",
    "                     rho > 0.3 ~  \">30\")) %>%\n",
    "    mutate(fill_color = factor(fill_color, levels = c(\"<0\", \"0-10\", \"10-20\", \"20-30\", \">30\")),\n",
    "           non_contaminant_taxon = factor(non_contaminant_taxon, levels = colnames(wide_df)[ord_row]), \n",
    "           contaminant_taxon = factor(contaminant_taxon, levels = rownames(wide_df)[ord_col])) %>%\n",
    "    ggplot(aes(x = non_contaminant_taxon, y = contaminant_taxon, fill = fill_color)) +\n",
    "    geom_tile() +\n",
    "    theme(axis.ticks.x = element_blank(),\n",
    "          axis.text.x = element_blank(),\n",
    "          axis.ticks.y = element_blank(),\n",
    "          axis.text.y = element_blank()) +\n",
    "    scale_fill_manual(values = c(\"#6699cc\", \"#ffffcc\", \"#f39b7fb2\", \"#e64b35b2\", \"#d11141\")) +\n",
    "    labs(x = \"Non-contaminant species\", y = \"Contaminant species\", fill = \"Rho (%)\")\n",
    "\n",
    "hm\n",
    "\n",
    "ggsave(str_glue(\"results/decontamination/correlation_decontamination/correlation_heatmap.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.png\"), dpi = 300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "97fb853e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3deYAcZZ3/8Uog5AY5BeXw4Ce4\nCILgckRAAqKuhqABwiVye4QsuwYQTZbDQBA0CyrrSkDXW1lAlzsioqAbQFYU+C0IIYJEzpBz\nkpkkc9RW9XTP09/p/n5mursqk+l5v/6oqqnqevqZnn7T1XOEKAbQsGigJwA0A0ICMkBIQAYI\nCcgAIQEZICQgA4QEZGBAQpoZfaOvm3w++vfi7fpx4w2oMC+gt0ZD+mP09j5v8/1be+3oRxtf\n3uWHG2VIhXkBvW2IkLY/o9eO/raxEYYEVLUBQloUDWRIqxs5OdfB0EwyDanlir233GzX81am\n2+vnvnv0tkf8Ko6nRIkJyY5r9hs38u3nvBinbfzbfYeMHzvhl+5Jvd8jTYv+Iz3+YPSRZHnT\nYVuO2OFDd/Xc7YzoltsPHFc53Mzotmu3H2Xv4fkTthm93y2vR/v3uueyMcs2u64/cPzI3b5Q\nNhjvkVBVliGtPzTa+5/O3T16b0fyFJwU7fbZk8ZF343vODU64Or/jDs/HL3jwtkfjt701/RJ\nefbIj50/edimv/FOUiHNi7b91L+csdWw75fu94vR1O1m//TSsZvca4ebHV0w9vjTza6lO0YT\nLj5t1JeiQ+xNy8YsH/7kaOfpXzwg2nNVz2CEhKqyDOm/ov0743jd7tFtcfy96IPtcfznMWNa\n4psKl3bzooPWJqt/iY5L2xie3CSem75QOSeJkPaMnk02F29+QOl+k+H+mKx+mr7MlA83J9ri\n/l7Tmhkdm+z53ejoULu/bMyyzRujvZcneZ8Tnd8zGCGhqixDeu5n/5OuPh9dFseHRenTLr56\nxqJiSBOiu9MdqzYb0Zo8nQ9Kt9tGDVvqnSRC2mnYK+n2up77nRkdka46No9eN8NdUbiAM7v2\nih5Ot09JQyrfXzZm2ebh0Z3p5vIR2/YMRkioKuNvNnStWrJkdjQzjsdFPW/MCyF1jYpWFD7a\nM3okeeafX9h+d7TAO0mE9KnoHd95ufxOZ0ZfKKz3i/7bDHdFdG6vaXVuNrzQ341pSOU3LRuz\nbHNctKyw3id6rjQYIaGqTEP6yYRR6XcWkmdmazSyZ28hpFWlHROTV6aZ0dzC9mHpf/GrnyRC\naj1tkyja44Jne247M/pqYX24HS557l/ea1oroy0Kex4uhFR207Ixw2Zr1GNBaTBCQlVZhnRt\n9IYv3nzPLz9TaGKznt3FkIo7DovmJ8/8awrbhydROSep79rFL37rY2+INrupdNuZ0dXFoc1w\nyXP/K72mtSJ6Q+GWv09DKr+pGbO02RYNu7joL6XBCAlVZRnSDt2XVoVrpfHFq6K4dGk3prhj\nj+gPyTN/VmF77+Qdi3NSr5DO6Q7pzu6QEuu+tcmW64vbM6MvFtZ79Rqu+Nwv29UxfJP2dPuW\nNKTym/Yes3tzi+jVnk+OkKBkGFJrNKawPih9Zh4WFX4vaM7hC4rfbDg0uj3dsWzT0WuTZ/4H\n0+220cNXeif1Cum86OvpkSvTkJ5/qXDGEdGTxTsufrOhdWSv4bqf+2bXrtHj6fbpSUhmf9mY\nZZtHRjcXtpfGhAQty1ekraO/JsvZ20fT0u9kv2dNHD+35agV8e3R0cnu70b7p9/+Pjc6M33m\nb/Jgsv216HD3pF4hXRednNxqxY5JSI9GE9NvGLTstMnrxfudGQ1PX1vm9h6u+Nwv3/WZ6NPJ\n9oLx6StS2f6yMcuHvzHasyXZfmDTqYQErfGQxnyk24/iz0X/77JL99/9l9HWV7zQ9dFo509/\nYnz6s9Vnh404/dPpD1t3n3XpodE7Xk9/gnrW+DPnnDx85EOxd1KvkF7dfPj5N1+963nRP8Tx\nSdHbpl00bZdoRmkOM6OTxp0556SK4YrP/fJdz46PPnzZmeOuSkMq3182ZvnwJ0Q7f+7iKSM2\nf4iQoDUeUsnFcdvMt4/cadqy+PSx2z8er//qXqPGHFy4VPvyNqP2i+P2r71nzMjdv7A8Tn+n\n5+ZfHzJu7KG/Tbadk3r/GcWD7x8zep/vL4kOi+PObx603WY7fuDnPXOYGV17b7Xhis99cw9/\nOHLcuPf/+ono/XZ/2Zjlw3def+D4UW896+mYkKA1wx/21f5rrff1fNMCyMQQC+mVOx9LV9+I\npuc3HQxFQyykH0UHrY/jpbtG9+Q5IQw9QyykdYdEe5z/2R0Lv7oKZGeIhRS3zN5j3Oh9/rUj\nx+lgKGqGkIABR0hABggJyAAhARkgJCADhARkgJCADBASkAFCAjLQYEirV1S3qr3NOZKNdSvz\nHL2lvTXP4Vesz3X01e1rch0/39mvaW/Jc/iV6zIdblVWIa1YUt3yuM05ko32pXmOviJek+fw\nSzpyHX1V3JLr+J25jr46Xpnn8K+3Zzpcz78xQkjVEJJCSAEhSYSkEFJASBIhKYQUEJJESAoh\nBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAoh\nBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAoh\nBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAoh\nBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkDaKQTjAKuwhJICSLkIoIqTaEZBFSESHV\nhpAsQioipNoQkkVIRYRUG0KyCKmIkGpDSBYhFRFSbQjJIqQiQqoNIVmEVERItSEki5CKCKk2\nhGQRUhEh1YaQLEIqIqTaEJJFSEWEVBtCsgipiJBqQ0gWIRURUm0IySKkIkKqDSFZhFRESLUh\nJIuQigipNoRkEVIRIdWGkCxCKiKk2hCSRUhFhFQbQrIIqYiQakNIFiEVEVJtCMkipCJCqg0h\nWYRUREi1ISSLkIoIqTaEZBFSESHVhpAsQioipNoQkkVIRYRUG0KyCKmIkGpDSBYhFRFSbQjJ\nIqQiQqoNIVmEVERItSEki5CKCKk2hGQRUhEh1WZjDam9yxHH3pFM5Dv6IJq9DSnz4ath+KAj\nq5B4RaoHr0i+IfqKREj1ICQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh+QjJ\nICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgKIfkI\nySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5\nCMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh\n+QjJICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgK\nIfkIySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFI\nCiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJB\nSAoh+QjJICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGS\nQUgKIfkIySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIR\nkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELy\nEZJBSAoh+QjJICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC\n8hGSQUgKIfmaNaQXLj1h6oVPxXHL3OOPueTVsCakBhCSr0lDWn/Kv/7tpauntsazz124ePa0\nzp41ITUgt5CqZJU9Qgr6HdKKn7XG8d8m/WXJpEXJq9HRj5bWhNQIQvI1aUipVd/8zPoFU7qS\nrek3ltaE1AhC8jVtSJ0fm3Th0nj+qen2rHmldbK4b9/Ew313iDypkAZ6bkNAz3ucfrwivfD4\nZZ9ePf+0dHPmvNI6WTxycuKx9uo64k7nSDa6ch19EM1ehZTdvRhxTuN264w7ch0/22fO+hpC\niuOOE+94qHBJd85NpXXpEJd29eDSztekl3Z/OqstjrtOun3ZUc8k1Ux+orQmpEYQkq9JQ1p9\nypdfeHnelBfjK6cvfOGiGV09a0JqACH5mjSk+PlLT5x6wWNxvOaaqcfOWRbWhNQAQvI1a0h9\nIKR6EJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQf\nIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh+QjJICSFkHyEZBCSQkg+QjIISSEk\nHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgKIfkIySAkhZB8hGQQkkJIPkIyCEkh\nJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJ\nISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh+QjJICSFkHyEZBCSQkg+QjII\nSSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgKIfkIySAkhZB8hGQQkkJIPkIy\nCEkhJB8hGYSkEJKPkAxCUgjJR0gGISkNhaTSIaS+EJJFSIRUF0KyCImQ6kJIFiERUl0IySIk\nQqoLIVmEREh1ISSLkAipLoRkERIh1YWQLEIipLoQkkVIhFQXQrIIiZDqQkgWIRFSXQjJIiRC\nqgshWYRESHUhJIuQCKkuhGQREiHVhZAsQiKkuhCSRUiEVBdCsgiJkOpCSBYhEVJdCMkiJEKq\nCyFZhERIdSEki5AIqS6EZBESIdWFkCxCIqS6EJJFSIRUF0KyCImQ6kJIFiERUl0IySIkQqoL\nIVmEREh1ISSLkAipLoRkERIh1YWQLEIipLoQkkVIhFQXQrIIiZDqQkgWIRFSXTbWkFYtq25l\nvNY5ko32FXmO3hK35Tn8ss5GTu5/SFlNt5eGZt+n1rglz+GXd2Q63IqsQlq7vrr2uNM5ko2u\n9jxHz332jZzc/5Cymm4vDc2+Tx1xR67jZzv7dVmFxKVdPbi08w3RSztCqgch+QjJICSFkHyE\nZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgKIfkIySAkhZB8\nhGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5CMkgJIWQ\nfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh+QjJICSF\nkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgKIfkIySAk\nhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5CMkg\nJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh+QjJ\nICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUgKIfkI\nySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5\nCMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSMoG\nCimnqggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSA\nkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCvof0tKvnjz1wqfjuGXu\n8cdc8mpYE1IDCMnXrCH984WLXrrq5LZ49rkLF8+e1tmzJqQGEJKvSUNa9eW/xfFrk55eMmlR\n8mp09KOlNSE1gpB8TRpSwVNHLVkwpSvZmH5jaZ029mTi1eXVrYrXOUey0bEyz9Fb4rY8h1/e\n2cjJdYaU1dwbnH2fWuPVeQ6/oiPT4VbWEtKqz14fzz813Zo1r7ROFvftm3i4Hx0iU3WGNNDT\nbk4973H6EdLis7/ZFc8/Ld2cOa+0ThZPX554pq26tXGHcyQbnWvzHH1d3J7n8G1djZxcZ0hZ\nzb3B2fepPV6f6/id2Q7X/5D+dOIdyfKhwiXdOTeV1qWjvEeqB++RfM36Hul/T/xDulp21DNJ\nNZOfKK0JqRGE5GvSkNad9dP09m3xldMXvnDRjK6eNSE1gJB8TRrSnyYV3BGvuWbqsXOS00pr\nQmoAIfmaNKS+EFI9CMnXFCHt+2T3+qb3EFKeCMnXFCFFjxRW7ZeOIaQ8EZKvCUKKAl6R8hye\nkIQmCOlPX4smn5E686LFhJQnQvI1QUhx/MFn+hsQITWCkHxNEVLtCKkehORripBeOW7rYd1v\nkggpT4Tka4qQjtv0A6cW3iWdQUh5IiRfU4S0zU/6GxAhNYKQfE0R0pjnCSlFSAohBU5Ih9xO\nSClCUggpcEL609/fT0hLCEkjpMAJ6X27RGN2KSCkPBGSrylCOmTi4UWElCdC8jVFSLUjpHoQ\nko+QDEJSCMnXFCFtXbIFIeWJkHxNEdLkggNG730OIeWJkHxNEVLRywffRUh5IiRfM4UUL+AP\n+/IcnpCEpgpp8ShCyhMh+ZoqpMt2JKQ8EZKvKUJ6d8Ge20QzCClPhORropD2mThnLSHliZB8\nTRFS7QipHoTka5KQXr993g3zVxESIfkIKXBC6vynEek/2DD2KkLKc3hCEpoipK9EH7/hrjuv\n+2D0PULKEyH5miKkv/tU93oqP5DNc3hCEpoipJH3dK9vHU1IeSIkX1OENPaW7vVN4wgpT4Tk\na4qQ3nfIunTVOuEQQsoTIfmaIqQ7h+189pdmfXLrTX5JSHkiJF9ThBT/fPf029979fuvKAip\nLoTka46Q4vjF3z/ySr8zIqT6EJKvOUJ66evJ4rVLXyKkPIcnJKEpQvrz9psly+ejHRYSUp4I\nydcUIR296+/T1ZO7TiGkPBGSrylC2vo73evrtiWkPBGSrylCGvWj7vW3+b+a5zk8IQlNEdKB\nR3amq9XvPYiQ8kRIvqYI6a7hu50ze+Yntxp+NyHliZB8TRFSfOde6Q9k331nfzsipLoQkq85\nQorj1x59dEm/MyKk+hCSr1lCqhEh1YOQfIRkEJJCSD5CMghJISQfIRmEpAxESNllRUgBIUmE\npBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmEpBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmE\npBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmEpBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmE\npBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmEpBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmE\npBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmEpBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmE\npBBSQEgSISmEFBCSREgKIQWEJBGSQkgBIUmEpBBSkFlIq1dVtzpe7xzJRkdLnqO3xuvyHH5V\nZyMnZxJSIxPoauTkPq2NW/McvqWhx75yuKxCanWsjdu9Q5nobMtz9Lxn39XIyZmENGCz79P6\neF2ew7d1ZjteViFxaVcPLu18Q/TSjpDqQUg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTk\nIySDkBRC8hGSQUgKIfkIySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE\n5CMkg5AUQvIRkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEp\nhOQjJIOQFELyEZJBSAoh+QjJICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYh\nKYTkIySDkBRC8hGSQUgKIfkIySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gG\nISmE5CMkg5AUQvIRkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdI\nBiEphOQjJIOQFELyEZJBSAoh+QjJICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlH\nSAYhKYTkIySDkBRC8hGSQUgKIfkIySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJ\nR0gGISmE5CMkg5AUQvIRkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlII\nyUdIBiEphOQjJIOQFELyEZJBSMrAh9RIVoQUEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQU\nEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQU\nEJJESAohBYQkEZJCSAEhSYSkEFJQQ0h/mzE5XbXMPf6YS14Na0JqACH5mjWkB065uhDS7HMX\nLp49rbNnTUgNICRfs4b0q9ceTENaMmlR8mp09KOlNSE1gpB8zRpSHBdCWjClK1lOv7G0ThYv\n3pL4a0t1a+L1zpFsdK7Oc/TWeF2ew7d0NnJyDiHVNoGuRmbfp3VxW57Dr27osa8crtaQ5p+a\nbs6aV1oni/v2TTzc5+nIWA4hDfSnNHj1vMfpd0inpZsz55XWMa9IDeAVydfkr0gPFS7pzrmp\ntC4d5D1SPXiP5Gvy90jLjnomqWbyE6U1ITWCkHzNGtKyJfdMXrKkLb5y+sIXLprR1bMmpAYQ\nkq9ZQzpjUurWeM01U4+dk5xWWhNSAwjJ16wh9YGQ6kFIPkIyCEkhJB8hGYSkEJKPkAxCUgjJ\nR0gGISmE5CMkg5AUQvIRkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlJq\nDCmHdAgpG4QkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQU\nEJJESAohBYQkEZJCSAEhSYSkEFJASBIhKYQUEJJESAohBYQkEZJCSAEhSYSkEFJASNJQC6m2\nrAgpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRC\nUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRC\nUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRC\nUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRC\nUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRC\nUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRC\nUggpICSJkBRCCghJIiSFkAJCkghJIaSAkCRCUggpICSJkBRCCghJGtoh9ZUVIQWEJBESIfUP\nIUmEREj9Q0gSIRFS/xCSREiE1D+EJBESIfVPZiG1OtbG7d6hTHS25Tl63rPvqu3mAxpSw7Ov\n0fp4XZ7Dt3VmO15WIa1eVd3qeL1zJBsdLXmO3hqvy3P4VZ213XxAQ6qcTlc2D4Jjbdya5/At\nNT72fQ2XVUhc2tWDSzvfEL20I6R6EJKPkAxCUgjJR0gGISmE5CMkg5AUQvIRkkFICiH5CMkg\nJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELyEZJBSAoh+QjJ\nICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC8hGSQUhKnyEN\naDpW5eQIKSAkiZCCyskRUkBIEiEFlZMjpICQJEIKKidHSAEhSYQUVE6OkAJCkggpqJwcIQWE\nJBFSUDk5QgoISSKkoHJyhBQQkkRIQeXkCCkgJImQgsrJEVJASBIhBZWTI6SAkKQBCGmge3FV\nTpWQAkKSCCmonCohBYQkEVJQOVVCCghJIqSgcqqEFBCSREhB5VQJKSAkiZCCyqkSUkBIEiEF\nlVMlpICQJEIKKqdKSAEhSYQUVE6VkAJCkggpqJwqIQWEJBFSUDlVQgoISSKkoHKqhBQQkkRI\nQeVUCSkgJImQgsqpElJASBIhBZVTJaSAkCRCCiqnSkgBIUmEFFROlZACQpIIKaicKiEFhCQR\nUlA5VUIKCEkipKByqoQUEJJESEHlVAkpICSJkILKqRJSQEgSIQWVUyWkgJAkQgoqp0pIASFJ\nGySkgS6knypnT0gBIUmEFFTOnpACQpIIyZNOnJACQpIIyZNOnJACQpIIyZNOnJACQpIIyZNO\nnJACQpIIyZNOnJACQpIIyZNOnJACQpIIyZNOnJACQpIIyZNOnJACQpIIyZNOnJACQpIIyZNO\nnJACQpIIyZNOnJACQpJyCmmgK8hA+mkQUkBIEiF50k+DkAJCkgjJk34ahBQQkkRInvTTIKSA\nkKSaQqp8pvXvloNS+mkQUkBIEiF50k+DkAJCkhoIqfJ5179bDhLpp0FIASFJhORJPw1CCghJ\nqghpoJ+/G430wSCkgJAkQvKkDwYhBYQkEZInfTAIKSAkiZA86YNBSAEhSYTkSR8MQgoISSKk\nxtX72BOSQUhDXb2PPSEZgxAQ05EAAApOSURBVD6kgX4eDnr1PvaEZBDSUFfvYz9UQmqZe/wx\nl7xKSOhDvY/9UAlp9rkLF8+e1klI0Op97IdISEsmLUpelY5+NJeQavhK1BlSP++BkBpX7xdl\niIS0YEpXspx+Yw0h9f/xreGrZEKq9x7qvXfUrP9fFBtS/0fppz5CqvX+6g5p/qnpcta8ZPH0\n5Yln2qpbG3eUNu3cnNtXuaXS1rnWPa//91DvvaNm/f+itMfr6/rS9lenPFrz/dUd0mnpcmYa\n0n37Jh6u7XSgqfR8s6DWkB4qXNqdc1OyWPVk4tXl1a2K1zlHstGxMs/RW+K2PIdf3pnr6Gvi\n1lzHz3f2rfHqPIdf0ZHpcCvrDWnZUc8kb4wmP1H6uGm/a5fn8NX+r+YZWhW35Do+v2sX1P/t\n7yunL3zhohldhNQIQvINlZDWXDP12DnhdEKqByH5hkpIvRBSPQjJR0gGISmE5CMkg5AUQvIR\nkkFICiH5CMkgJIWQfIRkEJJCSD5CMghJISQfIRmEpBCSj5AMQlIIyUdIBiEphOQjJIOQFELy\nEZJBSAoh+QjJICSFkHyEZBCSQkg+QjIISSEkHyEZhKQQko+QDEJSCMlHSAYhKYTkIySDkBRC\n8hGSQUgKIfkIySAkhZB8hGQQkkJIPkIyCEkhJB8hGYSkEJKPkAxCUgjJR0gGISmE5CMkg5AU\nQvIRkkFICiH5CMkgJIWQfEM0JM+rl9+Zz8AbxKLL7x/oKTTg8cv/MNBTaMCCy58e6CnUJaeQ\nntn38nwG3iAe3PdbAz2FBtyx738O9BQa8N197xvoKdSFkKogpIFDSAYhDRxCGgiEVAUhDRxC\nAoYwQgIyQEhABggJyEC2IbXMPf6YS16124svPXHqhf8/07vJSbXZJ+6d9OAATqqfqs29fN9G\nbTA/8CXZhjT73IWLZ0/rLN/uOuPaNWt/fOyqTO8nH1Vmn2wt/8SUQfD1rDb38n0btcH8wJdk\nGtKSSYuS/6Qc/Wj59spJT8bx65MGwe99VJt9snnFf3xi4/96Vpt7+b6N2mB+4HtkGtKCKV3J\ncvqNZvuCq1vW/fisdVneTz6qzj5ecNa6QfD1rDb38n0btcH8wPfINKT5p6bLWfPM9tJpkyad\nsjDLu8lJ1dm3nPJ4PAi+ntXmXr5vozaYH/geWYX028mTJz85/7R0c2b3I1Lc7vjHb6xqu+XE\npRndTz7c2cfXXBsPhq9ntbmX79uoDeYHvkdWIa15/vnn1z5UeF0+56Z0R2n70aPakvXpt2Z0\nP/lwZ//HM9YMiq9ntbmX79uoDeYHvkeml3bLjnomjldMfqJ8+9FJySMSn3J7lveTj2qzv2rK\niSeeeNRxcwZ6bn2pNvfyfRu1wfzA98j2299XTl/4wkUzuuJ7buvZXnPKv61Z919TXsz0fvJR\nZfar0j+DPPmelQM9tT5VmXvPeqM3mB/4kmxDWnPN1GPnLIvjq2aF7ecuPXHq+Y9lejc5qTb7\n1GC4wqg29/LPYaM2mB/4En5FCMgAIQEZICQgA4QEZICQgAwQEpABQgIyQEhABghpw3jtvL8b\nvcl2H7onswFXR5+rfuDwXdxzXt8l/ZXQl8/fY+y43c99Jtm8OCr9yPPiKDXibSf8bxzP3PIv\nmU1zyCCkDWLpLmOmfecnl711+E8zG3L4xdX3+yF1fuBdrXH8u602mTzniuNHjrvDhvSF66+/\n/hvTtxj5u7jjsH3aMpvmUEFIG8Tc6Cfpaun2b+vPX373668gx321+ql+SD+I7ovjV7bZ6uH0\ng6d22uIVE1L31mPDjozjJ4d/pT8zQBlC2iAuiLr/tHFx8pLw3oN+895RW526Ivnwd0eMH7X3\nt9MDP9n/DeP3TWM78NA7d9wvPuR9D+038k1faf/iDuMmPlt++JD3PX7Y2G2Pfy2Ot78ujl86\nc+eRb/z4U8U76T718Lcv+tC4wi3iuw4eN2qPuaXfWu3Y9eBk+fnoR90f3jv3tSohxTvvmiym\nbrs65wek6RDSBnFTdGLPP/9y0Bv3vufZazedHMe/HnHorfd8Okr+839jdMwdt38oSq62Dt1r\nj6tviyfufOhvn/xwdNQFf75r/KHlhyfust+tz/1ok6lxvLYjjg/Y/oZf/eBd263pHrj71MPf\nstec278QHRfHPx928M2/OCuaUbzfB6LvJMt3btURplUZ0pIRRyTLu6KN/6+YNjKEtEF0HR+N\n/chVDxWu6yZEv0qWZ0d/jfd969pk66gtWuOrjlwfxys3PSl5ixP9LE6X/xPH90f7J5snjOgw\nh3+T7Dti68KoK6PPJ8u/zCn+jUrp1FuS5cHJLXbfKR3+YyNe7z48M3ohmcjwiWXTKg/pzpdf\nfvmvd+8Xpf9jq9WbnZHrw9GECGkDmf+JHaNomwuTF48Jo9PXhB9EP1sSfaYt8a3ot8Xb7Jhc\nex2+aXuydfi4ZLEwOi9ZzohWlh8enW6dOqywo32bXe4te89VPHVUei33yWHxi9HZ6d4b0hey\n1Ee3TRZrosllk+r9Xbsoekf3P8C/53sy/eSHAELacBZdf3B0YGc84S3pB3dH1z1afPJGP42X\n//Pum48cGU1IOtghPVr4jsFz0RVx+qZmefnh9EB8RvHL9vu3RFsd9+PStVrZqektfh99Kd26\nK7q++/D+e6TL0QeXzag8pKvvvvvuQ0ct7v5w4k5Zf/LNjpA2pK7To/vjCW9PN5Pn9x+j0x4s\nWBIfudkVDz351Jt6SukVUu/DpZDijvsu2DM6aG33B2Wnprd4JJqVbt0Z3dB9ePeD0uW7x7WG\n+fR+j/TEJsd0f/jx8dl+4s2PkDaEtT+8pXvj+9EP4wlj0suxG6LblkafLB5/MfrHZNk+1gmp\n4vAZ5V+2b0Xf697oFdIr3cNfF83vPnxA4RXpS9HV3R/+924PVX6z4TPRLwof8opUK0LaELp2\n3bLwT812fjR6PJ5QeGYfN/zl+O+3WJ5sfW9m+1PRpcnGv0cHVA+p4nAxpEempv9I9qKo+FOf\nXiHFe26dvvh8eEzxPdak7dLl8jePLvxDNE/s+MZllSEtecM7Cj/E4j1SrQhpg7h37OZnfv2G\nS/eKpsfxhDfveu09n4tOiOPfjHjX938xa8Rp8fqd3nzrA+cdcdj4X62uFlLF4WJIL22x17fv\n+fGBI//cfSe9Q7pj+ISbbzuhMEhqVvS3dPXIm6OJl1x2zKY7PJLmM+Mbqft7kro6uixZrtns\n9NwfkiZDSBvGn6e/c4tN3/gPNyebE3Z/5H2jtzyzJdn87QfGj3rnFR3Js/vAMdudver2bXZ4\npup7pN6HS5d2j03ZfrMdJj9cvI/eIcXzDx47cp/vlKbwQPEScMmX9t5i/J4Xpz+xLX2vblpP\nSOt3G/1c+gZuMP/fMwcEIW1wE3YbmPvt2HVi3zfqdvw2LXnOpBkR0gY3UCHFP4ge6N8Nn+J3\n7WpGSBvcgIXU+YG9Wvu+VXK7ifz2d80IaYMbsJCKf4/Up1n8PVLtCAnIACEBGSAkIAOEBGSA\nkIAMEBKQAUICMkBIQAb+D9ey5qqTI6BhAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "taxon <- \"Lactobacillus paragasseri\"\n",
    "taxon_name <- gsub(\" \", \"_\", taxon)\n",
    "corr_res %>% \n",
    "filter(non_contaminant_taxon == taxon) %>%\n",
    "    ggplot(aes(x = rho)) +\n",
    "    geom_histogram(bins = 50) +\n",
    "    labs(x = \"Spearman's rho (CLR)\", \"No. of contaminants\", title = taxon)\n",
    "\n",
    "ggsave(str_glue(\"results/decontamination/correlation_decontamination/{taxon_name}.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.png\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "c5b58e33",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAC/VBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqan\np6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5\nubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrL\ny8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd\n3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v\n7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////a/pFu\nAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3deZwcZZnA8eIIOQhGLgUF4sFqXARR\nVI6IWYKyuu4QlJAAcgW8kyy7gooky2EwIC4bBQ8O3fVAZEFd5IoorKKLYBQRViJEBBMgQJFr\nJpPJXF2frer36Zmumeququet6q5M/X5/dPfMO/12zdP1ZXommeB4RGSd0+4DIBoLAYkog4BE\nlEFAIsogIBFlEJCIMghIRBmUE6RFzlX5bKzvM87X2n0INHazhHShM9yUuvcXENJlU7/b7kOg\nsZslpBtm+b3d2TO4Ornu/QWERJRjWby0+5Fz9Mh3AYnKVZaQ+r701snjX7vgGU8gzXf+I3j3\nr533Ba8Bb7nzbRN3n+v2X/jqCdO+XAl/uL/6wDEvHX/Q94KP77r04F132v/cTcHtm47addze\n77lj+LHqFuu39yrXHr7L+Nd/dtPIzYY3MN8jjdqQKIsyhDT4Xud15y15r/OKv0ZAWup8dOqy\nb8903nfa0f/x1X2DhfoPX+r80+4X/NflU5x7fF8znIP/+expztsGPO8aZ8+P/utZu2337dpD\n1S+GIJ3i7Lfw/MOcAzvDm9VtUIU0akOiTMoQ0jXOEVv9q3915kRAutQZ/6Tnde/sHO4TWO68\nN/zhlzrb/9a/fZVzuuf9t3PooOf1TnN+7HkHOn/2373mJYfVHqp+sX77G52DN/hflhY4nwpv\nVrdBFdKoDYkyKUNI0507g6vOncZtiYI0J7h9iPMd/9J1poU//FLn2OD275wjPO/JHwYM/PP+\nEs/bd7vngtu9Qw9Vv1i//dHO7cHtDeP29EKb1W1QhTRqQ6JMyg5SZYKzsfrmgc6KKEiLg9sz\nnAf8yx7nVeEPv9Q5P7j5uHNw9V2VTtdd4izyvI86r/vm2hGPNrRYv/1kZ3118c3Ok6HN6jao\nQorakMi+7CB1OuPNmzP9LzURkL4Y3J7hrPQCSFPDHy6rq5w3+Zc3TJ9Q/XMp38qWeTs4zgGf\n/vPwY9Ut1m2/ZfhPs+4LbVa3QRXS6A2JsihLSDuZN49ylieCVPfh9ef+V5yXnn/zXT/9eGDF\n8575+vtf6ux0U+2h6hfrtu9xtrtQ+ktY5fAG8jcbRmxIlEkZvrSbJC+vDnB+ZyAtMGf67ZGQ\nQh9ef+7v7fxvcHuJgeTX+/Uddu2T2/WL9dtPcZ4fOpwwpKENhv+KUP2GRJmU4Q8bZji3Blfr\nd5y41UA617kyeMcXIiGFPrzu3N/iTKpuekRg5alnq7ff5TxqHim0WL/9Mc7N1YV1XhhS3QZV\nSCM3JMqmDCH9p3No8PPss50PyY+/r3ZO8d/euE80pPoPrz/3d3f+6t9cspcz33vQmRn8fK1r\n3x1elIeqWwxtf6NzYJf/xr07zg1tVr9BAGn0hkSZlCGkSoczbfHFM5zXvSiQnn/J9p+6edn+\n5zr/EAWp/sPrIX3S+ZtLLj502k+d3S9d/UHnNfMvmD/VOaf2UPWL9dt7Jzn7ffLC48e95P7w\nZnUbVL8ijdqQKJOy/CtC/V9+y6Tx0z67wav9Xbtf/92kiW/+tuscFQWp/sPrz/2eRa8dv+/8\n9d6ZO+/18OBXj3jZTvu8+0dDD1W/WL+9N3jt4btMePWHHxuxWd0GVUijNiTKJH6xjyiDgESU\nQUAiyiAgEWUQkIgyCEhEGQQkogwCElEGAYkog4BElEFAIsogIBFlEJCIMghIRBlkCWnzRsu6\n+7tst4hu66Z89u3rzWffzp589t3cb/0cRdfTmc++vX357Ltpaw6bdmYFaaNrWbe3yXaL6PrW\n57NvZSCffTdszWffLq8rn423bshn34FKPvuu78tjUyBpA5IEJBdIFgFJApILJIuAJAHJBZJF\nQJKA5ALJIiBJQHKBZBGQJCC5QLIISBKQXCBZBCQJSC6QLAKSBCQXSBYBSQKSCySLgCQByQWS\nRUCSgOQCySIgSUBygWQRkCQguUCyCEgSkFwgWQQkCUgukCwCkgQkF0gWAUkCkgski4AkAckF\nkkVAkoDkAskiIElAcoFkEZAkILlAsghIEpBcIFkEJAlILpAsApIEJBdIFgFJApILJIuAJAHJ\nBZJFQJKA5ALJIiBJQHKBZBGQJCC5QLIISBKQXCBZBCQJSC6QLAKSZAHppFAjFoGkCkgSkCQg\nqQKSBCQJSKqAJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqAJAEpfs0qIOkCkglIEpB0AckE\nJAlIuoBkApIEJF1AMgFJApIuIJmAJAFJF5BMQJKApAtIJiBJQNIFJBOQJCDpApIJSBKQdAHJ\nBCQJSLqAZAKSBCRdQDIBSQKSLiCZgCQBSReQTECSgKQLSCYgSUDSBSQTkCQg6QKSCUgSkHQB\nyQQkCUi6gGQCkgQkXUAyAUkCki4gmYAkAUkXkExAkoCkC0gmIElA0gUkE5AkIOkCkglIEpB0\nAckEJKndkJ4+Z1Zw1XXFibMven74GkgZB6T4NavaDOne05ZVIS05e9WaJfMHh66BlHFAil+z\nqs2Q7n7h1wEkt+MJ/6vRcQ/WroGUdUCKX7Oq3S/tvCqk+46v+JcLb6xdAynrgBS/ZlUxIC0/\nI7i5+JratX/xy6P8fluxzbPeocX7bnMHXLx9w1iSr9mVxyAGUkOaF9xcdE3tOnj/sX6/H7Bs\n0Bu03SK6Sk77el4++w5Wctq3gAMOY2m21uwjU5bLgPvTQrq/+pJuwU2169oiL+2yipd2UWsZ\nvtArxku79cc+7quZ9UjtGkhZB6SotbEDab171yzX7fG+sHDV6gvOqQxdAynjgBS1NnYgndUR\ndIvX/aW5Jyz171a7BlLGASlqbexAiglIWQWkqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoN\nSEkDkgSkqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoNSEkDkgSkqDUgJQ1IEpCi1oCUNCBJ\nQIpaA1LSgCQBKWoNSEkDkgSkqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoNSEkDkgSkqDUg\nJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoNSEkDkgSkqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQB\nKWoNSEkDkgSkqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoNSEkDkgSkqDUgJQ1IEpCi1oCU\nNCBJQIpaA1LSgCQBKWoNSEkDkgSkqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoNSEkDkgSk\nqDUgJQ1IEpCi1oCUNCBJQIpaA1LSgCQBKWoNSEkDklReSMmzOl4g6QKSCUgSkHQByQQkCUi6\ngGQCkgQkXUAyAUkCki4gmYAkAUkXkEztgtSMAJCABKRaQHKBZBGQJCC5QLIISBKQXCBZBCQJ\nSC6QLAKSBCQXSBYBSQKSCySLgCQByQWSRUCSgOQCySIgSUBygWQRkCQguUCyCEgSkFwgWQQk\nCUgukCwCkgQkF0gWAUkCkgski4AkAckFkkVAkoDkAskiIElAcoFkEZAkILlAsghIEpBcIFkE\nJAlILpAsApIEJBdIFgFJApILJIuAJAHJBZJFQJKA5ALJIiBJQHKBZNHYgJTBCToKUnICQAIS\nkGoByc0Q0qYXLev2Om23iK5/fT77Vgby2Xdjbz77bva6It4bPkFVG/dubLpns0fQQlIdZ60N\nfVZ3b7BpVpB6+y0b9AZst4iuks+2/V5OGw/ktO+gNxjx3vAJqtq4MvKJa0Yg+Ucm3yVluQy4\nLytIvLTLKl7axac6zlrFfmkHpKwCEpBsApIEJCDZBCQJSECyCUgSkIBkE5AkIAHJJiBJQAKS\nTUCSgAQkm4AkAQlINgFJAhKQbAKSBCQg2QQkCUhAsglIEpCAZBOQJCABySYgSUACkk1AktoJ\nSXXyAskFkkVAkoDkAskiIElAcoFkEZAkILlAsghIEpBcIFkEJAlILpAsApIEJBdIFm2rkDI/\neYHkAskiIElAcoFkEZAkILlAsghIEpBcIFkEJAlILpAsApIEJBdIFgFJApILJIuAJAHJBZJF\nQJKA5ALJIiBJQHKBZBGQJCC5QLIISBKQXCBZBCQJSC6QLCobpIanMpBcIFkEJAlILpAsApIE\nJBdIFgFJApILJIuAJAHJBZJFQJKA5ALJIiBJQHKBZBGQJCC5QLIISBKQXCBZBCQJSC6QLAKS\nBCQXSBYBSQKSCySLgCQByQWSRUCSgOQCySIgSUBygWQRkCQguUCyCEgSkFwgWQQkCUgukCwC\nkgQkF0gWAUkCkgski4AkAckFkkVAkoDkAskiIElAcoFkEZAkILlAsghIEpBcIFkEJAlILpAs\nApIEJBdIFgFJApILJIuAJAHJBZJFQJKA5ALJIiBJQHKBZBGQJCC5QLIISBKQXCBZBCQJSC6Q\nLAKSBCQXSBYBSQKSCySLgCQByQWSRUCSgOQCySIgSUBygWQRkCQguQpID3dUu81bGFydAKTM\nA1KqR09fMSBVD+L/TljtzbvVv7EOSJkHpFSPnr5iQKp2/vWeN3tF6F1AyiogpXr09BUH0s/P\n6vP6Oq5cMO+SZ4I3+zf5rX/Rsm6v03aL6Po25LNvZSCffTduzWffzV5X9Vp78oar23jrxhGP\nlPB+FsdiNYgNfVZ3b7CpAlL/Wf/jfwU69bKVj1102mb/7XsO8XsguUNqa9qTN5z2EbI5ljzn\no2tw6FZySHfPq92rZ85y//L/PuO3cqtl/V6f7RbRDfbms69XyWffvsF89u33+qvX2pM3XN3G\ngyOfuIT3szgWq0H05jJgBaRF/zl08xPX127xPVJW8T1SqkdPX1G+R9py3MP+5VNX9fk3Z98D\npKwDUqpHT19RIP2xY5N/2XnysrVPLz2zB0hZB6RUj56+okC699hKcLVq8dxTlj439F4gZRWQ\nUj16+ooCKTogZRWQUj16+oCkC0gmIElA0gUkE5AkIOkCkglIEpB0AckEJAlIuoBkApIEJF1A\nMgFJApIuIJmAJAFJF5BMQJKApAtIJiBJQNIFJBOQJCDpApIJSBKQdAHJBCQJSLqAZAKSBCRd\nQDIVA1Lmj54+IOkCkglIEpB0AckEJAlIuoBkApIEJF1AMgFJApIuIJmAJAFJF5BMQJKApAtI\nJiBJQNIFJBOQJCDpApIJSBKQdAHJBCQJSLqAZAKSBCRdQDIBSQKSLiCZgCQBSReQTECSgKQL\nSCYgSUDSBSQTkCQg6QKSCUgSkHQByQQkCUi6gGTKFlKzEzuPR2j2eCkDki4gmYAkAUkXkExA\nkoCkC0gmIElA0gUkE5AkIOkCkglIEpB0AckEJAlIuoBkApIEJF1AMgFJApIuIJmAJAFJF5BM\nQJKApAtIJiBJQNIFJBOQJCDpApIJSBKQdAHJBCQJSLqAZAKSBCRdQDIBSQKSLiCZgCQBSReQ\nTECSgKQLSCYgSUDSBSQTkCQg6QKSCUgSkHQByQQkCUi6gGQCkgQkXUAyAUkCki4gmYAkAUkX\nkExAkoCkC0gmIElA0gUkE5AkIOkCkglIEpB0AckEJAlIuoBkApIEJF1AMgFJApIuIJnyhNTq\nrAYBJF1AMgFJApIuIJmAJAFJF5BMQJKApAtIJiBJQNIFJBOQJCDpApIJSBKQdAHJBCSp2JA6\nN1jW42223SK6/k357FsZyGffrr589u32uqvX7UaQRVaD2NSfxThHbpoVpK229Xt91ntENtib\nz75eJZ99+wbz2bff669etxtBFlkNojeXAWcFiZd2WcVLu/isBlHsl3ZAyiogxWc1CCDpApJp\nLEEKl3IQQNIFJBOQJCDpApIJSBKQdAHJBCQJSLqAZAKSBCRdQDIBSQKSLiCZgCQBSReQTECS\ngKQLSCYgSUDSBSQTkCQg6QKSCUgSkHQByQQkCUi6gGQCkgQkXUAyAUkCki4gmYAkAUkXkExA\nkoCkC0gmIElA0gUkE5AkIOkCkglIEpB0AckEJAlIuoBkApIEJF1AMgFJApIuIJmAJAFJF5BM\nQJKApAtIJiBJQNIFJBOQJCDpApIJSBKQdAHJBCQJSLqAZAKSBCRdQDIBSQKSLiCZgCQBSReQ\nTECSgKQLSCYgSUDSBSQTkCQg6QKSCUgSkHQByQQkqZWQDnnUXN/0FiA1CkhFKeUgWgnJWVG9\n6r94EpAaBaSilHIQrYPkDMdXpIYBqSilHETrID30ZWfWWUEfumANkBoFpKKUchCtfGn3948n\nBQSkrANS2lIOgp/a6QKSCUhSKyE9N2f37cw3SUBqFJCKUspBtBLSnB3ffUb1u6SzgNQoIBWl\nlINoJaQ9bkgKCEhZB6S0pRxEKyFNegpIcQGpKKUcRCshvfNWIMUFpKKUchCthPTQ238BpJiA\nVJRSDqKVkN4x1Zk0tRqQGgWkopRyEC19aTfzaAlIjQJSUUo5CP5AVheQTECSgKQLSCYgSa2E\ntHutKUBq1DYEqd1nes6lnEYrIc2qdtjEgxcAqVFAKkopp9GGl3Zrj7wDSI0CUlFKOY12fI90\nH7/Y1zAgFaWU02gHpDUTgNQoIBWllNNoB6RL9gFSo4BUlFJOo5WQ3lTtwD2cc4DUKCAVpZTT\naD2kN89cuhVIjQJSUUo5Df5AVheQgBSqtZBevPWa65Z3AqlxQCpKKafRSkiD/zwu+Acbdr4c\nSA0DUlFKOY1WQvqi84Hr7rj96r93vgWkRgGpKKWcRish/e1HzfVc/kC2YUAqSimn0UpI4+8y\n17dMBFKjgFSUUk6jlZB2/oG5vmkykBoFpKKUchot/VXzd/YGV1umvxNIjQJSUUo5jVZCun27\n/T7yucWn777DT4HUKCAVpZTTaOmfI/1oWvDj74MS/xYFkDILSGlLOY0W/82GZ36z4rnEjICU\nXUBKW8pptBTSs1f6Fy9c/CyQGgakopRyGq2E9Ke9dvIvn3L2XgWkRgGpKKWcRishHbf/b4Kr\nR/c/fpSYhR1+J3he1xUnzr7oeSBlHpDSlnIaLf1XhL5prq/ecxSkebf6d1vneUvOXrVmyfxB\nIGUdkNKWchqthDThenP9jdH/V/PZ5n947nY84X9VOu5BIGUdkNKWchqthHT4MdUvNZvfdsRI\nR30dVy6Yd8kz3n3HV4LXeTf6F+se8Ht2o2U93mbbLaLr78pn38pgPvt29WW+ZbvP9JxLO+D+\nzAe8cePwbxyFIN2x/esXLFl0+m7b3zkS0sZTL1v52EWnbV5+RvDW4mv8i3sO8Xtg1JcuKkzt\nPtNzrt3jDRr6HmfEj79vPyj4A9k33R59r545y5fPC24sCiA9caXfn7dY1uf12m4R3WBPPvtW\nKvnsu3Ug8y3bfabnXMpp9AxmPmC/BpA874UHH3Qb+vvE9fdXX9otuKn2Hr5Hyiq+R0pbymkU\n499seOqqPs/bMvue9cc+7uuZ9QiQsg5IaUs5jWJA6jx52dqnl57Z431h4arVF5xTAVLWASlt\nKadRDEjeqsVzT1n6nOd1f2nuCUuH7w6krAJS2lJOoyCQGgSkrAJS2lJOA0i6gASkUEDSBSQg\nhQKSLiABKRSQdAEJSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQghQKS\nLiABKRSQdAEJSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQghQKSLiAB\nKRSQdAEJSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQghQKSLiABKRSQ\ndAEJSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQghQKSLiABKRSQdAEJ\nSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQghQKSLiABKRSQdAEJSKGA\npAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQghQKSLiABKRSQdAEJSKHGGKSR\nswCSBKS0pZwGkHQBCUihgKQLSEAKBSRdQAJSKCDpAhKQQgFJF5CAFApIuoAEpFBA0gUkIIUC\nki4gASkUkHQBCUihgKQLSEAKBSRdQAJSKCDpAhKQQgFJF5CAFApIuoAEpFBA0gUkIIUCki4g\nASkUkHQBCUihgKQLSEAKBSRdQBrrkMLFTgNIuoAEpFDFhtTdlbLwZ9/V1ev1pN0iWQOpDy1Z\nlcF89t3Sn/mW7Tmj21TsNLoHMh9wV9fmrCBtSVv4s9+ypc/rTb1HogZ78tm3Usln360DmW/Z\nnjO6TcVOo2cw8wH7ZQWJl3ZZxUs7u2KnUeyXdkDKKiDZFTsNIOkCEpBCAUkXkIAUCki6gASk\nUEDSBSQghQKSLiABKRSQdAEJSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDS\nBSQghQKSLiABKRSQdAEJSKGApAtIQAoFJF1AAlIoIOkCEpBCAUkXkIAUCki6gASkUEDSBSQg\nhdrmITX/7IEkZQKpVWdtAYudDZB0Aalcxc4GSLqAVK5iZwMkXUAqV7GzAZIuIJWr2NkASReQ\nylXsbICkC0jlKnY2QNIFpHIVOxsg6QJSuYqdDZB0Aalcxc4GSLqAVK5iZwMkXUAqV7GzAZIu\nIJWr2NkASReQylXsbICkC0jlKnY2QNIFpHIVOxsg6QJSuYqdDZB0Aalcxc4GSLqAVK5iZwMk\nXUAqV7GzAZIuIJWr2NkASReQylXsbICkC0jlKnY2QNIFpHIVOxsg6QJSuYqdDZB0Aalcxc4G\nSLqAVK5iZwMkXUAqV7GzAZIuIJWr2NkASReQylXsbICkC0jlKnY2QNIFpHIVOxsg6QJSuYqd\nDZB0Aalcxc4GSLqAVK5iZwMkXUAqV7GzAZIuIJWr2NkASReQylXsbICkC0jlKnY2QNIFpHIV\nOxsg6QJSuYqdDZB0Aalcxc4GSLqAVK5iZwMkXUAqV7GzAZIuIJWr2NkASVfhIY2YBJDsip0N\nkHQBqVzFzgZIuoBUrmJnAyRdQCpXsbMBki4glavY2QBJF5DKVexsgKQLSOUqdjYFgbTu306Z\ne95jnreww+8EINk3YhJAsit2NgWB9C/nPfHs5af0ePNu9e++Dkj2jZgEkOyKnU0xIHVe9rTn\nvdDxmDd7Rej9QFI3YhJAsit2NsWAVG3lsW5fx5UL5l3yDJDsGzEJINkVO5viQOr8xLXexlMv\nW/nYRadt9t985gd+f+2Krdln39XV6/XEb6FpoDuffSuDGW00YhLd/ZnvWapiZ9M9kMGAR7ZZ\nAWnNR75aMbd65iz3L+85xO+B+Ps1++yTKx57JZ+E9iNLVbZPTtIGh24lhvTQybcN3f7E9R5f\nkWwbMYkmX5FGzizpnqUqdt4F+Yr0x5N/F1w9dVWf522ZfU/t3XyPpG7EJJp8jzRyZkn3LFWx\n8y7G90i9H/5+cL+ezpOXrX166Zk9QLJuxCSAZFfsvIsB6aGOard5qxbPPWXpc0PvB5K6EZMA\nkl2x8y4GpEYBSd2ISQDJrth5A0kXkMpV7LyBpAtI5Sp23kDSBaRyFTtvIOkCUrmKnTeQdAGp\nXMXOG0i6gFSuYucNJF1AKlex8waSLiCVq9h5A0kXkMpV7LyBpAtI5Sp23kDSBaRyFTtvIOkC\nUrmKnTeQdAGpXMXOG0i6gFSuYucNJF1AKlex8waSLiCVq9h5A0kXkMpV7LyBpAtI5Sp23kDS\nBaRyFTtvIOkCUrmKnTeQdI1ZSM3OH/vzcZstdt5A0gWkchU7byDpAlK5ip03kHQBqVzFzhtI\nuoBUrmLnDSRdQCpXsfMGki4glavYeY9pSClnkSYglavYeQNJF5DKVey8gaQLSOUqdt5A0gWk\nchU7byDpAlK5ip03kHQBqVzFzhtIugoBqdnnN2ItBEk7s+T3G+NFPBdA0gWkMhfxXABJF5DK\nXMRzASRdQCpzEc8FkHQBqcxFPBdA0gWkMhfxXABJF5DKXMRzASRdOUJKftTNPnLEmhISRRbx\nXABJF5DKXMRzASRdQCpzEc8FkHQBqcxFPBdA0gWkMhfxXABJF5DKXMRzASRdbYKU/PPTnB+U\nsIgnDki6gFTmIp44IOkCUpmLeOKApAtIZS7iiQOSLiCVuYgnDki6gFTmIp640kKKnUzzgFTm\nIp44IDWaTPOAVOYinjggNZpM84BU5iKeOCA1mkzzgFTmIp44IDWaTPOAVOYinjggNZpM84BU\n5iKeOCA1mkzzgFTmIp44IDWaTPOAVOYinjggNZpM84BU5iKeOCA1mkzzgFTmIp44IDWaTPOA\nVOYinjggNZpM84BU5iKeOCA1mkzzgFTmIp64EkFKOZnmAanMRTxxQGo0meYBqcxFPHFAajSZ\n5gGpzEU8cUBqNJnmAanMRTxxQGo0meYBqcxFPHFAajSZ5gGpzEU8cUBqNJnmAanMRTxxQGo0\nmeYBqcxFPHFAajSZ5gGpzEU8ccWGtCW+bCaT4IHCDfakvkuiKpUtzY4s+eeQzVwosognrmcw\nj9MhK0ibO2PLZjLxjzOigQSHpqky2NnsyJJ/DtnMhSKLeOI2D+RwNnRlBYmXdry0K2IRT1yx\nX9oBCUhFLOKJA1KjyTQvEpLlnkFA2vaqzhtIw7NIE5CoVnXeQBqeRZqARLWq8wbS8CzSBCSq\nVZ03kIZnkSYgUa3qvIE0PIs0AYlqVecNpOFZpAlIVKs6byANzyJNBYREbar6zABpeBZpAhLV\nqj4zQBqeRZqARLWqzwyQhmeRJiBRreozA6ThWaQJSFSr+swAaXgWaQIS1ao+M0AankWaUkNK\nvkbbWOacB9LQLNIEJKplznkgDc0iTUCiWuacB9LQLNIEJKplznkgDc0iTUCiWuacB9LQLNIE\nJKplznkgDc0iUc3ul3yNxlDmnAfS0CwS1ex+yddoDGXOeSANzSJRze6XfI3GUOacB9LQLBLV\n7H7J12gMZc55IA3NIlHN7pd8jcZQ5pwH0tAsEtXsfsnXaAxlznkgDc0iUc3ul3yNxlDmnAfS\n0Cwa7JnHGo2hzDkPpC2Pvg4AAAs3SURBVKFZNNgzjzUaQ5lzHkhDs2iwZx5rNIYy5zyQhmbR\nYM881mgMZc55IA3NosGeeazRGMqc80AamkWDPfNYozGUOeeBNDSLBntqHy+P46QCZs75vgQn\nT9qAZHM/2sYy5zyQRn++yde0e9IYypzzQBr9+SZf0+5JYyhzzgNp9OebfE27J42hzDkPpNGf\nb/I17Z40hjLnPJBGf77J17R70hjKnPNASjQnQFCjgJRiTkCiRgEpxZyARI0CUoo5AYkaBaQU\ncwISNQpIKeYEJGoUkIhyDkhEGQQkogwCElEGAYkog4BElEFAIsogIBFlEJCIMghIRBkEJKIM\nAhJRBgGJKIOARJRBQCLKICARZRCQiDIISEQZBCSiDAISUeYBiSiDgESUQUAiyiAgEWUQkIgy\nCEhEGQQkoszLE1LXFSfOvuh5IFEJyhPSkrNXrVkyfxBINPbLEZLb8YT/Vem4B4FEY78cId13\nfMW/XHgjkGjslyOk5WcEl4uv8S8e+7zf4z2xtXsaRMriT+4eNaR5weWiANI9h/g9kO7uRGOq\noR8WpIV0f/Wl3YKb/IvOR/2e32BZj7fZdovo+jfls29lIJ99O3vz2bfb685n497OfPYdrOSz\n76b+PDbVQlp/7OP+N0azHqm9neB7pOZ1e5tst4iub30++1YG8tl3w9Z89u3yuvLZeOuGfPYd\nqOSz79D/+jLTTbWQvC8sXLX6gnMqQMo6IEklgdT9pbknLB2+O5CyCkhSSSCNCEhZBSQJSKqA\nJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqAJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqA\nJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqAJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqA\nJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqAJAFJApIqIElAkoCkCkgSkCQgqQKSBCQJSKqA\nJAFJApIqIElAkoCkCkgSkCQgqQKSBCSpnJCs+9XnV7X5CFJ2xdfbfQTp+v3nH2r3IaTra1e0\n+whUtRvSdYfc2+YjSNnfHd/uI0jXDw/573YfQrref1S7j0AVkFIGpJwDkiog5RyQWhOQUgak\nnAMSUXkDElEGAYkog4BElEFthNR1xYmzL3pe3rjtrA8sXNG+Y0lS/fGuvvikueetbO/xNK/+\naEOTLmrb1nhH1UZIS85etWbJ/MHq7btPXfH8TR/ubt/BJKjuePtO+/enn102d0u7D6lJ9dOt\nv13Ytq3xjqp9kNyOJ/z/DB33YPWNj93dtuNIWv3xbvyh/yQ/3fGXdh9T4+qPNjTporZtjXd0\n7YN03/EV/3LhjcHtdR13L5z9yWJ/La8/3qDOr368r42HE1P90Y488kK2bY13dO2DtPyM4HLx\nNcHlYx3nP9N1zdyNbTuYBNUfr+cNvr/jvHXtPJyY6o82fOQFbdsa7+jaAumXs2bNenT5vODm\nourk/tjxkOcNnPTTdhxMgkYdr9/qhy/52OY2HlNM9Uc74siL2bY13tG1BVL3U089tfX+6tfy\nBTcF71jTEfwyxfz/asfBJGjU8VYbOPm29h1SXPVHO+rIi9i2Nd7Rte+l3fpjH/e/rZz1SHB7\n4KQfe17vnF+07WASVH+8D324x/MqH7y13cfUuPqjrb9d2Lat8Y6ujT/+/sLCVasvOKfi3eUj\n+u4Hf+9+5fSe9h1MguqOd/Npl61ee83xz7T7kJpUP93a7UK3bY13VG2E1P2luScsXe95ly/2\nv7n81qlzFq1u37Ekqf54n7r45Lmf/kO7j6hZ9Udbu13otq3xjoq/IkSUQUAiyiAgEWUQkIgy\nCEhEGQQkogwCElEGAYkog4DUml44928n7vCy99yV2YabnU9GLxw9teF9Xpwa/M3QtZ86YOfJ\n085+3L95ofNrWbrQCRr3mpP+6HmLdt2mfhOoGAGpJa2bOmn+N2+45NXbfz+zLbe/MPr9jSEN\nvvuNWzzvV7vtMGvppSeOn3xbGNJnr7322qsWThn/K2/gqDcX+y9rFTEgtaQrnBuCq3V7vSbJ\nL3z3Jtly8r9F37UxpO8493jec3vs9kDwxsp9pzwXgmRu/WG7Yzzv0e2/mOQIqC4gtaRPO+Z/\nurHG/5LwtiN+/rYJu50R/Bbjr961y4SDvxEs3HDoS3c5JMB2+Izb93mr98533P/W8a/4Yv/5\ne0+e+ef65Xe+4+Gjdt7zxBc8b6+rPe/ZD+03/uUfqP1msbnr0a994j2Tqx/h3XHk5AkHXFH7\ny6oD+x/pX37Gud68+bMrXoiA5O23v38xd89t6neBihCQWtJNzsmdtdtHvPzgu/78lR1ned7/\njJtxy10fc/z//N/ozL7t1vc4/qutGQcdsOzH3sz9Zvzy0fc6x376T3fsMqN+eebUt97y5PU7\nzPW8rQOed9he1939nTe+TP7RGHPXo1910NJbP+vM8bwfbXfkzT/5sHOOPO69zjf9yzfsNjB8\nWKMhuePe5V/e4RT7l5cKGJBaUuVEZ+f3XX5/9XXddCf4l14+4vzVO+TVW/1bx07Z4l1+TJ/n\nbdrxg/63OM4PveDyt573C+dQ/+ZJ4wZCyz/33/eu3au7bnI+41/+Zan8vkHtrj/wL4/0P2La\nvsH27x/3olle5Kz2D2T7mXWHVQ/p9rVr1/71zrc6t/tvbd7prFzHMQYDUotafuo+jrPHef4X\nj+kTg68J33F+6Dof7/H7uvNL+Zh9/NdeR+/Y7986erJ/sco51788x9lUvzwxuHXGdtV39O8x\n9Wd133PJXScEr+VO3857xvlI8N7rHPlN03/c07/odmbVHdTIn9o5zuvMbykf+JZMP/kSBKTW\n9cS1RzqHD3rTXxW8cadz9YNy8jrf9zb8y7SXjB/vTPcd7B2sVn9i8KRzqRd8U7Ohfrn6o4Sz\n5Gn7zauc3eZ8r/Zare6uwUf8xvlccOsO51qzfOgBweXEI+uOqB7SsjvvvHPGhDXmzZn75vD5\nj+mA1MoqZzq/8Ka/Nrjpn9+/d+b9uprrHbPTpfc/uvIVQ1JGQBq5XIPkDdzz6QOdI7aaN+ru\nGnzECmdxcOt25zqzPO2I4PJNk+v+3cWR3yM9ssNs8+YHdsnhsx/TAakVbf3uD8yNbzvf9aZP\nCl6OXef8eJ1zuqw/4/yTf9m/cwNIo5bPqn/avu58y9wYAek5s/3VznKzfFj1K9LnnGXmzf99\n/f2jf9jwcecn1Tf5ipQ2ILWiyv67PhZcD/6j87A3vXpmz9l+rff2KRv8W99a1L/Sudi/8TXn\nsGhIo5YF0oq5wb+V/YQjf+ozApJ34O7BF5/3TpLvsTpeFlxueOXE6j8q8sg+L18/GpL70tdV\n/xCL75HSBqSW9LOdX/KhK6+7+CBnoedNf+X+X7nrk85JnvfzcW/89k8Wj5vn9e37ylvuPfdd\nR+1y9+YoSKOWBdKzUw76xl3fO3z8n8yDjIR02/bTb/7xSdVNghY7TwdXK17pzLzoktk77r0i\n4HPOVUG/GCK1zLnEv+ze6czWzGXsBKTW9KeFb5iy48v/4Wb/5vRpK94xcdcPdfk3f/nuXSa8\n4dIB/+w+fNLLPtJ56x57Px75PdLI5dpLuz8cv9dOe896QB5jJCRv+ZE7j3/zN2uHcK+8BHQ/\nd/CUXQ68MPgT29rP6uYPQep7/cQng2/givpPDBY2ILW86a9vz+MO7D8z/oNMJ+7RleeRjMWA\n1PLaBcn7jpPwf3y9kr9rlzogtby2QRp890GJ/pdDgzP529+pA1LLaxsk+X2k2Bbz+0jpAxJR\nBgGJKIOARJRBQCLKICARZRCQiDIISEQZBCSiDPp/V01UJ2geXTcAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "taxon <- \"Tolumonas auensis\"\n",
    "taxon_name <- gsub(\" \", \"_\", taxon)\n",
    "corr_res %>% \n",
    "filter(non_contaminant_taxon == taxon) %>%\n",
    "    ggplot(aes(x = rho)) +\n",
    "    geom_histogram(bins = 50) +\n",
    "    labs(x = \"Spearman's rho (CLR)\", \"No. of contaminants\", title = taxon)\n",
    "\n",
    "ggsave(str_glue(\"results/decontamination/correlation_decontamination/{taxon_name}.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.png\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1644b30d",
   "metadata": {},
   "source": [
    "### Run decontamination for multiple thresholds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e72f1160",
   "metadata": {},
   "outputs": [],
   "source": [
    "# params <- foreach (m = c(50, 100), .combine = 'c') %do% {\n",
    "#     foreach (hq = seq(0.70, 1, 0.01)) %do% {\n",
    "#         lq <- 1 - hq\n",
    "#         return(c(m, lq, hq))\n",
    "#     }\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "647afe14",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# all_species <- colnames(df_filt)[colnames(df_filt) != \"npm_research_id\"]\n",
    "\n",
    "# threshold_results <- foreach (param = params) %do% {\n",
    "#     min_samples <- param[1]\n",
    "#     lq_p <- param[2]\n",
    "#     hq_p <- param[3]\n",
    "    \n",
    "#     # Run decontamination\n",
    "#     result_df <- decontaminate_all_cols(df_filt, meta_filt, meta_cols, min_samples, lq_p, hq_p, fold_threshold)\n",
    "    \n",
    "#     # Parse results\n",
    "#     non_contaminants <- all_species[!(all_species %in% unique(result_df$taxa))]\n",
    "#     n_noncontam <- length(non_contaminants)\n",
    "#     nc_text <- paste0(non_contaminants, collapse = \";\")\n",
    "#     nc_df <- tibble(non_contaminants = non_contaminants)\n",
    "    \n",
    "#     fwrite(nc_df, str_glue(\"results/decontamination/clr_decontamination/NC.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.txt\"), quote = F)\n",
    "#     fwrite(result_df, str_glue(\"results/decontamination/clr_decontamination/results.n9999.min_samples{min_samples}.l{lq_p * 100}h{hq_p * 100}.csv\"))\n",
    "\n",
    "#     # Save summary\n",
    "#     morsel <- tibble(min_samples = min_samples, \n",
    "#                   lq_p = lq_p, \n",
    "#                   hq_p = hq_p, \n",
    "#                   n_noncontam = n_noncontam, \n",
    "#                   non_contaminants = nc_text)\n",
    "#     print(morsel)\n",
    "    \n",
    "#     return(morsel)\n",
    "# }\n",
    "\n",
    "# threshold_df <- bind_rows(threshold_results)\n",
    "# fwrite(\"results/decontamination/clr_decontamination/multiple_threshold_results.csv\", threshold_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "722e3997",
   "metadata": {},
   "outputs": [],
   "source": [
    "# fwrite(threshold_df, \"results/decontamination/clr_decontamination/multiple_threshold_results.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82b61826",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dat_filt %>% \n",
    "#     left_join(metadat_filt) %>%\n",
    "#     filter(get(tax) != 0) %>%\n",
    "#     group_by(site_supplying_sample) %>%\n",
    "#     summarise(lq = quantile(get(tax), 0.25), hq = quantile(get(tax), 0.75))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06040606",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dat_filt %>% \n",
    "#     pivot_longer(!npm_research_id, names_to = \"taxa\", values_to = \"clr\") %>%\n",
    "#     filter(clr != 0) %>%\n",
    "#     ggplot(aes(x = clr)) +\n",
    "#         geom_density()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e084553",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tax <- \"Burkholderia stabilis\"\n",
    "# df_filt %>% \n",
    "#     left_join(meta_filt) %>%\n",
    "#     filter(get(tax) != 0) %>%\n",
    "#     ggplot(aes(y = get(tax), x = site_supplying_sample)) +\n",
    "#         geom_boxplot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db45209a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# result_df <- bind_rows(results)\n",
    "# result_df %>% distinct(taxa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58e3b427",
   "metadata": {},
   "outputs": [],
   "source": [
    "# metadat <- meta_filt\n",
    "# dat <- df_filt\n",
    "# taxon <- \"Streptococcus oralis\"\n",
    "# lq_p <- 0.25\n",
    "# hq_p <- 0.75\n",
    "# fold_threshold <- 0"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
