{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "844f2b78",
   "metadata": {},
   "source": [
    "# iRep (bPTR) and coverage breadth analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "53041a0d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rsamtools\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: tidyverse\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.6     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m   masks \u001b[34mBiostrings\u001b[39m::collapse(), \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mcompact()\u001b[39m    masks \u001b[34mXVector\u001b[39m::compact()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m       masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m      masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m     masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m     masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m      masks \u001b[34mXVector\u001b[39m::slice(), \u001b[34mIRanges\u001b[39m::slice()\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: foreach\n",
      "\n",
      "\n",
      "Attaching package: ‘foreach’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:purrr’:\n",
      "\n",
      "    accumulate, when\n",
      "\n",
      "\n",
      "Loading required package: doParallel\n",
      "\n",
      "Loading required package: iterators\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm(list = ls())\n",
    "setwd(\"/mnt/c/Users/Cedric/Desktop/git_repos/blood_microbiome\")\n",
    "require(\"Rsamtools\")\n",
    "require(Biostrings)\n",
    "require(tidyverse)\n",
    "require(data.table)\n",
    "require(foreach)\n",
    "require(doParallel)\n",
    "registerDoParallel(cores = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c3332d32",
   "metadata": {},
   "outputs": [],
   "source": [
    "species_df <- fread(\"results/decontamination/read_matrix.raw.zeroed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6673e418",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 4 rows [1, 2, 3, 4].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 3 rows [1, 2, 3].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 3 rows [1, 2, 3].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 5 rows [1, 2, 3, 4, 5].”\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 2 rows [1, 2].”\n",
      "Warning message:\n",
      "“Expected 1 pieces. Additional pieces discarded in 2 rows [1, 2].”\n"
     ]
    }
   ],
   "source": [
    "files <- list.files(\"results/irep_analysis/raw_output/irep_out\", full.names = T)\n",
    "files <- files[grepl(\".tsv\", files)]\n",
    "\n",
    "morsels <- foreach(file = files) %do% {\n",
    "    temp_df <- read.csv(file, sep = \"\\t\", check.names = F)\n",
    "    colnames(temp_df)[1] <- \"ref_path\"\n",
    "    temp_df %>% \n",
    "        mutate(across(c(4:ncol(temp_df)), function(x) (as.numeric(ifelse(x == \"n/a\", NA, x)))),\n",
    "               across(c(1:3), as.character)) %>%\n",
    "        pivot_longer(!all_of(c(\"ref_path\", \"ORI\", \"TER\")), names_to = \"prefix\", values_to = \"bPTR\") %>%\n",
    "        separate(ref_path, sep = \"/\", into = c(rep(NA, 11), \"file_name\")) %>%\n",
    "        separate(file_name, sep = \"_\", into = c(\"genus\", \"species\", \"suffix1\", \"suffix2\"), remove = F) %>%\n",
    "        separate(prefix, into = c(rep(NA, 6), \"id\"), sep = \"/\") %>%\n",
    "        separate(id, into = c(\"npm_research_id\"), sep = \"_\") %>%\n",
    "        mutate(prefix = gsub(\".fasta\", \"\", file_name))\n",
    "}\n",
    "\n",
    "irep_df <- bind_rows(morsels) \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd89f3e8",
   "metadata": {},
   "source": [
    "## Parse coverage breadth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f6fb3ed3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 153 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>file_name</th><th scope=col>genus</th><th scope=col>species</th><th scope=col>suffix1</th><th scope=col>suffix2</th><th scope=col>ORI</th><th scope=col>TER</th><th scope=col>npm_research_id</th><th scope=col>bPTR</th><th scope=col>prefix</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>1116-0023</td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3658  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3674  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3686  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>WHB3734  </td><td>      NA</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>0116-0053</td><td>1.896385</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4240  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4251  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4438  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>WHB4828  </td><td>      NA</td><td>Acinetobacter_baumannii_NZ_CP043953.1      </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9245  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9246  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9270  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9282  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>WHB9317  </td><td>      NA</td><td>Alcaligenes_faecalis_NZ_CP013119.1         </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB3883  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB4048  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB4056  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHB4130  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>WHH1822  </td><td>      NA</td><td>Aspergillus_oryzae_NC_036435.1             </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB6459  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB6776  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB6849  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB8186  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>WHB8568  </td><td>      NA</td><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB6155  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9217  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9583  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9649  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>WHB9882  </td><td>      NA</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB3939  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB5008  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB8397  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>WHB9179  </td><td>      NA</td><td>Prevotella_melaninogenica_NC_014370.1    </td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>WHB4373  </td><td>      NA</td><td>Prevotella_oral_taxon_299_NC_022124.1    </td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>WHB9179  </td><td>      NA</td><td>Prevotella_oral_taxon_299_NC_022124.1    </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>0415-0080</td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>0715-0061</td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>0816-0014</td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>WHB3477  </td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>WHB3479  </td><td>      NA</td><td>Pseudomonas_mendocina_NZ_CP013124.1      </td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>WHB6775  </td><td>      NA</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>WHB9978  </td><td>1.346660</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0021</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0024</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0026</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0031</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>0416-0032</td><td>      NA</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB4219  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB4398  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB6459  </td><td>1.566844</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB6651  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>WHB8186  </td><td>      NA</td><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0024</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0026</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0027</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0416-0031</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>0715-0059</td><td>      NA</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>WHB7619  </td><td>      NA</td><td>Torque_teno_virus_6_NC_014094.1          </td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>WHB9685  </td><td>      NA</td><td>Torque_teno_virus_6_NC_014094.1          </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 153 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " file\\_name & genus & species & suffix1 & suffix2 & ORI & TER & npm\\_research\\_id & bPTR & prefix\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & 1116-0023 &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3658   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3674   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3686   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         & WHB3734   &       NA & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1   \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & 0116-0053 & 1.896385 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4240   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4251   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4438   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & WHB4828   &       NA & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1      \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9245   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9246   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9270   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9282   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         & WHB9317   &       NA & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1         \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB3883   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB4048   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB4056   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHB4130   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         & WHH1822   &       NA & Aspergillus\\_oryzae\\_NC\\_036435.1             \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB6459   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB6776   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB6849   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB8186   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         & WHB8568   &       NA & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1  \\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB6155   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9217   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9583   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9649   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         & WHB9882   &       NA & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB3939   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB5008   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB8397   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         & WHB9179   &       NA & Prevotella\\_melaninogenica\\_NC\\_014370.1    \\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         & WHB4373   &       NA & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1    \\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         & WHB9179   &       NA & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1    \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & 0415-0080 &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & 0715-0061 &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & 0816-0014 &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & WHB3477   &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         & WHB3479   &       NA & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1      \\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & WHB6775   &       NA & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1    \\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & WHB9978   & 1.346660 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1    \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0021 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0024 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0026 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0031 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         & 0416-0032 &       NA & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1  \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB4219   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB4398   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB6459   & 1.566844 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB6651   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & WHB8186   &       NA & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1 \\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0024 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0026 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0027 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0416-0031 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         & 0715-0059 &       NA & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         & WHB7619   &       NA & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1          \\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         & WHB9685   &       NA & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1          \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 153 × 10\n",
       "\n",
       "| file_name &lt;chr&gt; | genus &lt;chr&gt; | species &lt;chr&gt; | suffix1 &lt;chr&gt; | suffix2 &lt;chr&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; | npm_research_id &lt;chr&gt; | bPTR &lt;dbl&gt; | prefix &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | 1116-0023 |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3658   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3674   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3686   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         | WHB3734   |       NA | Achromobacter_xylosoxidans_NZ_CP043820.1    |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | 0116-0053 | 1.896385 | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4240   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4251   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4438   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | WHB4828   |       NA | Acinetobacter_baumannii_NZ_CP043953.1       |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9245   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9246   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9270   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9282   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         | WHB9317   |       NA | Alcaligenes_faecalis_NZ_CP013119.1          |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB3883   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB4048   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB4056   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHB4130   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         | WHH1822   |       NA | Aspergillus_oryzae_NC_036435.1              |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB6459   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB6776   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB6849   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB8186   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         | WHB8568   |       NA | Corynebacterium_segmentosum_NZ_LR134408.1   |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB6155   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9217   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9583   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9649   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         | WHB9882   |       NA | Cupriavidus_metallidurans_NZ_FYAX01000013.1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB3939   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB5008   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB8397   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         | WHB9179   |       NA | Prevotella_melaninogenica_NC_014370.1     |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         | WHB4373   |       NA | Prevotella_oral_taxon_299_NC_022124.1     |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         | WHB9179   |       NA | Prevotella_oral_taxon_299_NC_022124.1     |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | 0415-0080 |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | 0715-0061 |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | 0816-0014 |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | WHB3477   |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         | WHB3479   |       NA | Pseudomonas_mendocina_NZ_CP013124.1       |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | WHB6775   |       NA | Rickettsia_Tillamook_23_NZ_CP060138.1     |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | WHB9978   | 1.346660 | Rickettsia_Tillamook_23_NZ_CP060138.1     |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0021 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0024 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0026 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0031 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         | 0416-0032 |       NA | Staphylococcus_cohnii_NZ_UHDA01000001.1   |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB4219   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB4398   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB6459   | 1.566844 | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB6651   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | WHB8186   |       NA | Staphylococcus_epidermidis_NZ_CP035288.1  |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0024 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0026 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0027 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0416-0031 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         | 0715-0059 |       NA | Staphylococcus_haemolyticus_NZ_CP013911.1 |\n",
       "| Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         | WHB7619   |       NA | Torque_teno_virus_6_NC_014094.1           |\n",
       "| Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         | WHB9685   |       NA | Torque_teno_virus_6_NC_014094.1           |\n",
       "\n"
      ],
      "text/plain": [
       "    file_name                                         genus          \n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "16  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "17  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "18  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "19  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "20  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "⋮   ⋮                                                 ⋮              \n",
       "124 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "125 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "126 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "127 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "128 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "129 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "130 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "131 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "132 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "133 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "134 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "135 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "136 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "137 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "138 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "139 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "140 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "141 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "142 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "143 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "144 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "145 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "146 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "147 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "148 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "149 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "150 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "151 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "152 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "153 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "    species        suffix1 suffix2              ORI       TER      \n",
       "1   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "2   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "3   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "4   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "5   xylosoxidans   NZ      CP043820.1.fasta     0         0        \n",
       "6   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "7   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "8   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "9   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "10  baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227\n",
       "11  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "12  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "13  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "14  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "15  faecalis       NZ      CP013119.1.fasta     0         0        \n",
       "16  oryzae         NC      036435.1.fasta       0         0        \n",
       "17  oryzae         NC      036435.1.fasta       0         0        \n",
       "18  oryzae         NC      036435.1.fasta       0         0        \n",
       "19  oryzae         NC      036435.1.fasta       0         0        \n",
       "20  oryzae         NC      036435.1.fasta       0         0        \n",
       "21  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "22  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "23  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "24  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "25  segmentosum    NZ      LR134408.1.fasta     0         0        \n",
       "26  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "27  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "28  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "29  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "30  metallidurans  NZ      FYAX01000013.1.fasta 0         0        \n",
       "⋮   ⋮              ⋮       ⋮                    ⋮         ⋮        \n",
       "124 melaninogenica NC      014370.1.fasta       0         0        \n",
       "125 melaninogenica NC      014370.1.fasta       0         0        \n",
       "126 melaninogenica NC      014370.1.fasta       0         0        \n",
       "127 melaninogenica NC      014370.1.fasta       0         0        \n",
       "128 oral           taxon   299                  0         0        \n",
       "129 oral           taxon   299                  0         0        \n",
       "130 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "131 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "132 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "133 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "134 mendocina      NZ      CP013124.1.fasta     0         0        \n",
       "135 Tillamook      23      NZ                   240,832   888,370  \n",
       "136 Tillamook      23      NZ                   240,832   888,370  \n",
       "137 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "138 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "139 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "140 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "141 cohnii         NZ      UHDA01000001.1.fasta 0         0        \n",
       "142 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "143 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "144 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "145 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "146 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375\n",
       "147 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "148 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "149 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "150 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "151 haemolyticus   NZ      CP013911.1.fasta     0         0        \n",
       "152 teno           virus   6                    0         0        \n",
       "153 teno           virus   6                    0         0        \n",
       "    npm_research_id bPTR     prefix                                     \n",
       "1   1116-0023             NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "2   WHB3658               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "3   WHB3674               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "4   WHB3686               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "5   WHB3734               NA Achromobacter_xylosoxidans_NZ_CP043820.1   \n",
       "6   0116-0053       1.896385 Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "7   WHB4240               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "8   WHB4251               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "9   WHB4438               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "10  WHB4828               NA Acinetobacter_baumannii_NZ_CP043953.1      \n",
       "11  WHB9245               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "12  WHB9246               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "13  WHB9270               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "14  WHB9282               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "15  WHB9317               NA Alcaligenes_faecalis_NZ_CP013119.1         \n",
       "16  WHB3883               NA Aspergillus_oryzae_NC_036435.1             \n",
       "17  WHB4048               NA Aspergillus_oryzae_NC_036435.1             \n",
       "18  WHB4056               NA Aspergillus_oryzae_NC_036435.1             \n",
       "19  WHB4130               NA Aspergillus_oryzae_NC_036435.1             \n",
       "20  WHH1822               NA Aspergillus_oryzae_NC_036435.1             \n",
       "21  WHB6459               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "22  WHB6776               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "23  WHB6849               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "24  WHB8186               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "25  WHB8568               NA Corynebacterium_segmentosum_NZ_LR134408.1  \n",
       "26  WHB6155               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "27  WHB9217               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "28  WHB9583               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "29  WHB9649               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "30  WHB9882               NA Cupriavidus_metallidurans_NZ_FYAX01000013.1\n",
       "⋮   ⋮               ⋮        ⋮                                          \n",
       "124 WHB3939               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "125 WHB5008               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "126 WHB8397               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "127 WHB9179               NA Prevotella_melaninogenica_NC_014370.1      \n",
       "128 WHB4373               NA Prevotella_oral_taxon_299_NC_022124.1      \n",
       "129 WHB9179               NA Prevotella_oral_taxon_299_NC_022124.1      \n",
       "130 0415-0080             NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "131 0715-0061             NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "132 0816-0014             NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "133 WHB3477               NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "134 WHB3479               NA Pseudomonas_mendocina_NZ_CP013124.1        \n",
       "135 WHB6775               NA Rickettsia_Tillamook_23_NZ_CP060138.1      \n",
       "136 WHB9978         1.346660 Rickettsia_Tillamook_23_NZ_CP060138.1      \n",
       "137 0416-0021             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "138 0416-0024             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "139 0416-0026             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "140 0416-0031             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "141 0416-0032             NA Staphylococcus_cohnii_NZ_UHDA01000001.1    \n",
       "142 WHB4219               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "143 WHB4398               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "144 WHB6459         1.566844 Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "145 WHB6651               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "146 WHB8186               NA Staphylococcus_epidermidis_NZ_CP035288.1   \n",
       "147 0416-0024             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "148 0416-0026             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "149 0416-0027             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "150 0416-0031             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "151 0715-0059             NA Staphylococcus_haemolyticus_NZ_CP013911.1  \n",
       "152 WHB7619               NA Torque_teno_virus_6_NC_014094.1            \n",
       "153 WHB9685               NA Torque_teno_virus_6_NC_014094.1            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "irep_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5155b4c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Achromobacter_xylosoxidans_NZ_CP043820.1'</li><li>'Acinetobacter_baumannii_NZ_CP043953.1'</li><li>'Alcaligenes_faecalis_NZ_CP013119.1'</li><li>'Aspergillus_oryzae_NC_036435.1'</li><li>'Corynebacterium_segmentosum_NZ_LR134408.1'</li><li>'Cupriavidus_metallidurans_NZ_FYAX01000013.1'</li><li>'Cutibacterium_acnes_NC_021085.1'</li><li>'Fannyhessea_vaginae_NZ_UFSV01000001.1'</li><li>'Fusobacterium_nucleatum_NZ_LN831027.1'</li><li>'Fusobacterium_pseudoperiodonticum_NZ_CP024702.1'</li><li>'Gardnerella_vaginalis_NZ_PKJK01000001.1'</li><li>'Haemophilus_parainfluenzae_NZ_GL872339.1'</li><li>'Hepatitis_B_virus_genB_JX661488.1'</li><li>'Human_betaherpesvirus_6A_NC_001664.4'</li><li>'Human_betaherpesvirus_6B_NC_000898.1'</li><li>'Human_mastadenovirus_C_NC_001405.1'</li><li>'Lactobacillus_crispatus_NZ_CP039266.1'</li><li>'Malassezia_restricta_NC_040193.1'</li><li>'Moraxella_osloensis_NZ_CP014234.1'</li><li>'Neisseria_flavescens_NZ_CP039886.1'</li><li>'Neisseria_mucosa_NZ_UGRT01000001.1'</li><li>'Neisseria_subflava_NZ_CP039887.1'</li><li>'Prevotella_melaninogenica_NC_014370.1'</li><li>'Prevotella_oral_taxon_299_NC_022124.1'</li><li>'Pseudomonas_mendocina_NZ_CP013124.1'</li><li>'Rickettsia_Tillamook_23_NZ_CP060138.1'</li><li>'Staphylococcus_cohnii_NZ_UHDA01000001.1'</li><li>'Staphylococcus_epidermidis_NZ_CP035288.1'</li><li>'Staphylococcus_haemolyticus_NZ_CP013911.1'</li><li>'Torque_teno_virus_6_NC_014094.1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1'\n",
       "\\item 'Acinetobacter\\_baumannii\\_NZ\\_CP043953.1'\n",
       "\\item 'Alcaligenes\\_faecalis\\_NZ\\_CP013119.1'\n",
       "\\item 'Aspergillus\\_oryzae\\_NC\\_036435.1'\n",
       "\\item 'Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1'\n",
       "\\item 'Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1'\n",
       "\\item 'Cutibacterium\\_acnes\\_NC\\_021085.1'\n",
       "\\item 'Fannyhessea\\_vaginae\\_NZ\\_UFSV01000001.1'\n",
       "\\item 'Fusobacterium\\_nucleatum\\_NZ\\_LN831027.1'\n",
       "\\item 'Fusobacterium\\_pseudoperiodonticum\\_NZ\\_CP024702.1'\n",
       "\\item 'Gardnerella\\_vaginalis\\_NZ\\_PKJK01000001.1'\n",
       "\\item 'Haemophilus\\_parainfluenzae\\_NZ\\_GL872339.1'\n",
       "\\item 'Hepatitis\\_B\\_virus\\_genB\\_JX661488.1'\n",
       "\\item 'Human\\_betaherpesvirus\\_6A\\_NC\\_001664.4'\n",
       "\\item 'Human\\_betaherpesvirus\\_6B\\_NC\\_000898.1'\n",
       "\\item 'Human\\_mastadenovirus\\_C\\_NC\\_001405.1'\n",
       "\\item 'Lactobacillus\\_crispatus\\_NZ\\_CP039266.1'\n",
       "\\item 'Malassezia\\_restricta\\_NC\\_040193.1'\n",
       "\\item 'Moraxella\\_osloensis\\_NZ\\_CP014234.1'\n",
       "\\item 'Neisseria\\_flavescens\\_NZ\\_CP039886.1'\n",
       "\\item 'Neisseria\\_mucosa\\_NZ\\_UGRT01000001.1'\n",
       "\\item 'Neisseria\\_subflava\\_NZ\\_CP039887.1'\n",
       "\\item 'Prevotella\\_melaninogenica\\_NC\\_014370.1'\n",
       "\\item 'Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1'\n",
       "\\item 'Pseudomonas\\_mendocina\\_NZ\\_CP013124.1'\n",
       "\\item 'Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1'\n",
       "\\item 'Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1'\n",
       "\\item 'Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1'\n",
       "\\item 'Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1'\n",
       "\\item 'Torque\\_teno\\_virus\\_6\\_NC\\_014094.1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Achromobacter_xylosoxidans_NZ_CP043820.1'\n",
       "2. 'Acinetobacter_baumannii_NZ_CP043953.1'\n",
       "3. 'Alcaligenes_faecalis_NZ_CP013119.1'\n",
       "4. 'Aspergillus_oryzae_NC_036435.1'\n",
       "5. 'Corynebacterium_segmentosum_NZ_LR134408.1'\n",
       "6. 'Cupriavidus_metallidurans_NZ_FYAX01000013.1'\n",
       "7. 'Cutibacterium_acnes_NC_021085.1'\n",
       "8. 'Fannyhessea_vaginae_NZ_UFSV01000001.1'\n",
       "9. 'Fusobacterium_nucleatum_NZ_LN831027.1'\n",
       "10. 'Fusobacterium_pseudoperiodonticum_NZ_CP024702.1'\n",
       "11. 'Gardnerella_vaginalis_NZ_PKJK01000001.1'\n",
       "12. 'Haemophilus_parainfluenzae_NZ_GL872339.1'\n",
       "13. 'Hepatitis_B_virus_genB_JX661488.1'\n",
       "14. 'Human_betaherpesvirus_6A_NC_001664.4'\n",
       "15. 'Human_betaherpesvirus_6B_NC_000898.1'\n",
       "16. 'Human_mastadenovirus_C_NC_001405.1'\n",
       "17. 'Lactobacillus_crispatus_NZ_CP039266.1'\n",
       "18. 'Malassezia_restricta_NC_040193.1'\n",
       "19. 'Moraxella_osloensis_NZ_CP014234.1'\n",
       "20. 'Neisseria_flavescens_NZ_CP039886.1'\n",
       "21. 'Neisseria_mucosa_NZ_UGRT01000001.1'\n",
       "22. 'Neisseria_subflava_NZ_CP039887.1'\n",
       "23. 'Prevotella_melaninogenica_NC_014370.1'\n",
       "24. 'Prevotella_oral_taxon_299_NC_022124.1'\n",
       "25. 'Pseudomonas_mendocina_NZ_CP013124.1'\n",
       "26. 'Rickettsia_Tillamook_23_NZ_CP060138.1'\n",
       "27. 'Staphylococcus_cohnii_NZ_UHDA01000001.1'\n",
       "28. 'Staphylococcus_epidermidis_NZ_CP035288.1'\n",
       "29. 'Staphylococcus_haemolyticus_NZ_CP013911.1'\n",
       "30. 'Torque_teno_virus_6_NC_014094.1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Achromobacter_xylosoxidans_NZ_CP043820.1\"       \n",
       " [2] \"Acinetobacter_baumannii_NZ_CP043953.1\"          \n",
       " [3] \"Alcaligenes_faecalis_NZ_CP013119.1\"             \n",
       " [4] \"Aspergillus_oryzae_NC_036435.1\"                 \n",
       " [5] \"Corynebacterium_segmentosum_NZ_LR134408.1\"      \n",
       " [6] \"Cupriavidus_metallidurans_NZ_FYAX01000013.1\"    \n",
       " [7] \"Cutibacterium_acnes_NC_021085.1\"                \n",
       " [8] \"Fannyhessea_vaginae_NZ_UFSV01000001.1\"          \n",
       " [9] \"Fusobacterium_nucleatum_NZ_LN831027.1\"          \n",
       "[10] \"Fusobacterium_pseudoperiodonticum_NZ_CP024702.1\"\n",
       "[11] \"Gardnerella_vaginalis_NZ_PKJK01000001.1\"        \n",
       "[12] \"Haemophilus_parainfluenzae_NZ_GL872339.1\"       \n",
       "[13] \"Hepatitis_B_virus_genB_JX661488.1\"              \n",
       "[14] \"Human_betaherpesvirus_6A_NC_001664.4\"           \n",
       "[15] \"Human_betaherpesvirus_6B_NC_000898.1\"           \n",
       "[16] \"Human_mastadenovirus_C_NC_001405.1\"             \n",
       "[17] \"Lactobacillus_crispatus_NZ_CP039266.1\"          \n",
       "[18] \"Malassezia_restricta_NC_040193.1\"               \n",
       "[19] \"Moraxella_osloensis_NZ_CP014234.1\"              \n",
       "[20] \"Neisseria_flavescens_NZ_CP039886.1\"             \n",
       "[21] \"Neisseria_mucosa_NZ_UGRT01000001.1\"             \n",
       "[22] \"Neisseria_subflava_NZ_CP039887.1\"               \n",
       "[23] \"Prevotella_melaninogenica_NC_014370.1\"          \n",
       "[24] \"Prevotella_oral_taxon_299_NC_022124.1\"          \n",
       "[25] \"Pseudomonas_mendocina_NZ_CP013124.1\"            \n",
       "[26] \"Rickettsia_Tillamook_23_NZ_CP060138.1\"          \n",
       "[27] \"Staphylococcus_cohnii_NZ_UHDA01000001.1\"        \n",
       "[28] \"Staphylococcus_epidermidis_NZ_CP035288.1\"       \n",
       "[29] \"Staphylococcus_haemolyticus_NZ_CP013911.1\"      \n",
       "[30] \"Torque_teno_virus_6_NC_014094.1\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bam_path <- \"results/irep_analysis/raw_output/bam_files\"\n",
    "ref_path <- \"data/irep_data/genome_references\"\n",
    "prefixes <- list.files(ref_path, \".fasta\", full.names = F)\n",
    "prefixes <- gsub(\".fasta\", \"\", prefixes)\n",
    "prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0d49d2e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "morsels <- foreach (i = seq(length(prefixes))) %dopar% {\n",
    "    prefix <- prefixes[i]\n",
    "    bams <- list.files(str_glue(\"{bam_path}/{prefix}\"))\n",
    "    bams <- bams[!grepl(\".bai\", bams)]\n",
    "\n",
    "    crumbs <- foreach (j = seq(length(bams))) %do% {\n",
    "        bam <- bams[j]\n",
    "        id <- str_split(bam, \"_\")[[1]][1]\n",
    "        ref <- readDNAStringSet(str_glue(\"{ref_path}/{prefixes[i]}.fasta\"))\n",
    "        ref_length <- width(ref)\n",
    "        params <- PileupParam(distinguish_strands = F, distinguish_nucleotides = F)\n",
    "        pile_df <- pileup(str_glue(\"{bam_path}/{prefix}/{bam}\"), pileupParam = params)\n",
    "        \n",
    "        n_covered1 <- pile_df %>%\n",
    "            filter(count >= 1) %>%\n",
    "            nrow()\n",
    "        \n",
    "        n_covered5 <- pile_df %>%\n",
    "            filter(count >= 5) %>%\n",
    "            nrow()\n",
    "\n",
    "        perc_covered1 <- n_covered1 / ref_length * 100\n",
    "        perc_covered5 <- n_covered5 / ref_length * 100\n",
    "        \n",
    "        taxon <- str_split(prefix, \"_\")[[1]]\n",
    "            read_count <- species_df %>% \n",
    "                filter(npm_research_id == id) %>%\n",
    "                select(contains(taxon[1])) %>%\n",
    "                select(contains(taxon[2])) %>%\n",
    "                select(!contains(\"monkey\"))\n",
    "\n",
    "        if (ncol(read_count) > 1) {\n",
    "            read_count <- read_count %>% \n",
    "                select(contains(taxon[3]))\n",
    "            if (ncol(read_count) > 1) {\n",
    "            read_count <- read_count %>% \n",
    "                select(contains(paste0(\" \", taxon[4])))\n",
    "            }\n",
    "        }\n",
    "        \n",
    "        if (all(dim(read_count) == 1)) {\n",
    "            read_count <- deframe(read_count)\n",
    "        } else {\n",
    "            read_count <- NA\n",
    "        }\n",
    "        \n",
    "        return(tibble(prefix = prefix, \n",
    "                      npm_research_id = id,\n",
    "                      pairs_assigned = read_count,\n",
    "                      perc_covered1 = perc_covered1,\n",
    "                      perc_covered5 = perc_covered5))\n",
    "    }\n",
    "    \n",
    "    return(bind_rows(crumbs))\n",
    "}\n",
    "\n",
    "cov_df <- bind_rows(morsels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bfef9750",
   "metadata": {},
   "outputs": [],
   "source": [
    "# prefixes\n",
    "# prefix <- \"Hepatitis_B_virus_genB_JX661488.1\"    \n",
    "# id <- \"WHB4594\"\n",
    "# taxon <- str_split(prefix, \"_\")[[1]]\n",
    "#     read_count <- species_df %>% \n",
    "#         filter(npm_research_id == id) %>%\n",
    "#         select(contains(taxon[1])) %>%\n",
    "#         select(contains(taxon[2])) %>%\n",
    "#         select(!contains(\"monkey\"))\n",
    "    \n",
    "# #     if (ncol(read_count) > 1) {\n",
    "# #         read_count <- read_count %>% \n",
    "# #             select(contains(taxon[3]))\n",
    "# #         if (ncol(read_count) > 1) {\n",
    "# #         read_count <- read_count %>% \n",
    "# #             select(contains(paste0(\" \", taxon[4])))\n",
    "# #         }\n",
    "# #     }\n",
    "\n",
    "# #     if (all(dim(read_count) == 1)) {\n",
    "# #         read_count <- deframe(read_count)\n",
    "# #     } else {\n",
    "# #         read_count <- NA\n",
    "# #     }\n",
    "\n",
    "# read_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56cba5fc",
   "metadata": {},
   "source": [
    "### Merged results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "88d455ed",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 7 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>prefix</th><th scope=col>npm_research_id</th><th scope=col>pairs_mapped</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1</td><td>WHB3907</td><td> 902</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1</td><td>WHB3939</td><td> 148</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1</td><td>WHB5008</td><td>  68</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1</td><td>WHB8397</td><td> 130</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1</td><td>WHB9179</td><td>3942</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1</td><td>WHB4373</td><td>  50</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1</td><td>WHB9179</td><td>1274</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 7 × 3\n",
       "\\begin{tabular}{lll}\n",
       " prefix & npm\\_research\\_id & pairs\\_mapped\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1 & WHB3907 &  902\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1 & WHB3939 &  148\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1 & WHB5008 &   68\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1 & WHB8397 &  130\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1 & WHB9179 & 3942\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1 & WHB4373 &   50\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1 & WHB9179 & 1274\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 7 × 3\n",
       "\n",
       "| prefix &lt;chr&gt; | npm_research_id &lt;chr&gt; | pairs_mapped &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| Prevotella_melaninogenica_NC_014370.1 | WHB3907 |  902 |\n",
       "| Prevotella_melaninogenica_NC_014370.1 | WHB3939 |  148 |\n",
       "| Prevotella_melaninogenica_NC_014370.1 | WHB5008 |   68 |\n",
       "| Prevotella_melaninogenica_NC_014370.1 | WHB8397 |  130 |\n",
       "| Prevotella_melaninogenica_NC_014370.1 | WHB9179 | 3942 |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1 | WHB4373 |   50 |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1 | WHB9179 | 1274 |\n",
       "\n"
      ],
      "text/plain": [
       "  prefix                                npm_research_id pairs_mapped\n",
       "1 Prevotella_melaninogenica_NC_014370.1 WHB3907          902        \n",
       "2 Prevotella_melaninogenica_NC_014370.1 WHB3939          148        \n",
       "3 Prevotella_melaninogenica_NC_014370.1 WHB5008           68        \n",
       "4 Prevotella_melaninogenica_NC_014370.1 WHB8397          130        \n",
       "5 Prevotella_melaninogenica_NC_014370.1 WHB9179         3942        \n",
       "6 Prevotella_oral_taxon_299_NC_022124.1 WHB4373           50        \n",
       "7 Prevotella_oral_taxon_299_NC_022124.1 WHB9179         1274        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mapping_df <- fread(\"results/irep_analysis/raw_output/mapping_read_counts.csv\")\n",
    "mapping_df %>% filter(grepl(\"Prevo\", prefix))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "85fca457",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Joining, by = c(\"prefix\", \"npm_research_id\")\n",
      "Joining, by = c(\"prefix\", \"npm_research_id\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 138 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>prefix</th><th scope=col>npm_research_id</th><th scope=col>pairs_assigned</th><th scope=col>pairs_mapped</th><th scope=col>perc_covered1</th><th scope=col>perc_covered5</th><th scope=col>file_name</th><th scope=col>genus</th><th scope=col>species</th><th scope=col>suffix1</th><th scope=col>suffix2</th><th scope=col>ORI</th><th scope=col>TER</th><th scope=col>bPTR</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>1116-0023</td><td>430178</td><td>429196</td><td>79.292320359</td><td>7.450282e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3658  </td><td>385855</td><td>438419</td><td>78.389787758</td><td>7.366085e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3674  </td><td>459561</td><td>494858</td><td>78.805094251</td><td>7.456206e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3686  </td><td>541795</td><td>599335</td><td>79.082105806</td><td>7.525644e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Achromobacter_xylosoxidans_NZ_CP043820.1   </td><td>WHB3734  </td><td>688629</td><td>769978</td><td>79.583059893</td><td>7.609348e+01</td><td>Achromobacter_xylosoxidans_NZ_CP043820.1.fasta   </td><td>Achromobacter  </td><td>xylosoxidans </td><td>NZ</td><td>CP043820.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>0116-0053</td><td>  7673</td><td> 28842</td><td>79.329072139</td><td>1.697212e+01</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>1.896385</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4240  </td><td>   168</td><td>   210</td><td> 1.316042864</td><td>1.963529e-03</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4251  </td><td>   183</td><td>   233</td><td> 1.460462955</td><td>4.782956e-04</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4438  </td><td>   379</td><td>   453</td><td> 2.824813672</td><td>3.373242e-03</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td></tr>\n",
       "\t<tr><td>Acinetobacter_baumannii_NZ_CP043953.1      </td><td>WHB4828  </td><td>   190</td><td>   231</td><td> 1.642920130</td><td>0.000000e+00</td><td>Acinetobacter_baumannii_NZ_CP043953.1.fasta      </td><td>Acinetobacter  </td><td>baumannii    </td><td>NZ</td><td>CP043953.1.fasta    </td><td>3,869,401</td><td>1,967,227</td><td>      NA</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9245  </td><td>   982</td><td>  6163</td><td>29.029117408</td><td>6.902854e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9246  </td><td>   896</td><td>  5202</td><td>27.000516799</td><td>3.080480e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9270  </td><td>   996</td><td>  6128</td><td>28.984145520</td><td>6.330785e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9282  </td><td>   766</td><td>  4490</td><td>23.360108613</td><td>2.885617e-01</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Alcaligenes_faecalis_NZ_CP013119.1         </td><td>WHB9317  </td><td>  1104</td><td>  7106</td><td>31.747814470</td><td>1.073538e+00</td><td>Alcaligenes_faecalis_NZ_CP013119.1.fasta         </td><td>Alcaligenes    </td><td>faecalis     </td><td>NZ</td><td>CP013119.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB3883  </td><td>    24</td><td>     4</td><td> 0.013741771</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4048  </td><td>    32</td><td>     0</td><td> 0.000000000</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4056  </td><td>  7909</td><td>    31</td><td> 0.169732339</td><td>4.370987e-03</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHB4130  </td><td>    99</td><td>     0</td><td> 0.002315856</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Aspergillus_oryzae_NC_036435.1             </td><td>WHH1822  </td><td>    12</td><td>     0</td><td> 0.000000000</td><td>0.000000e+00</td><td>Aspergillus_oryzae_NC_036435.1.fasta             </td><td>Aspergillus    </td><td>oryzae       </td><td>NC</td><td>036435.1.fasta      </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6459  </td><td> 14476</td><td> 13069</td><td>66.396833288</td><td>7.454033e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6776  </td><td>   215</td><td>   200</td><td> 2.189016786</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB6849  </td><td>    31</td><td>    25</td><td> 0.249342726</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB8186  </td><td>    79</td><td>    83</td><td> 0.799303964</td><td>6.741270e-04</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Corynebacterium_segmentosum_NZ_LR134408.1  </td><td>WHB8568  </td><td>    38</td><td>    35</td><td> 0.346627680</td><td>0.000000e+00</td><td>Corynebacterium_segmentosum_NZ_LR134408.1.fasta  </td><td>Corynebacterium</td><td>segmentosum  </td><td>NZ</td><td>LR134408.1.fasta    </td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB6155  </td><td>   152</td><td>    19</td><td> 0.444721342</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9217  </td><td>   297</td><td>    31</td><td> 0.701307112</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9583  </td><td>  5506</td><td>   792</td><td>15.704852586</td><td>4.270964e-02</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9649  </td><td>  1972</td><td>   290</td><td> 6.521295389</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1</td><td>WHB9882  </td><td>   835</td><td>   115</td><td> 2.457894931</td><td>0.000000e+00</td><td>Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta</td><td>Cupriavidus    </td><td>metallidurans</td><td>NZ</td><td>FYAX01000013.1.fasta</td><td>0        </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB3939  </td><td>    141</td><td>    74</td><td> 1.0932372</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB5008  </td><td>     79</td><td>    34</td><td> 0.4487845</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB8397  </td><td>    150</td><td>    65</td><td> 0.9019109</td><td> 0.000000000</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Prevotella_melaninogenica_NC_014370.1    </td><td>WHB9179  </td><td>   3549</td><td>  1971</td><td>25.1270313</td><td> 0.224948898</td><td>Prevotella_melaninogenica_NC_014370.1.fasta    </td><td>Prevotella    </td><td>melaninogenica</td><td>NC   </td><td>014370.1.fasta      </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1    </td><td>WHB4373  </td><td>     79</td><td>    25</td><td> 1.0017609</td><td> 0.000000000</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1    </td><td>WHB9179  </td><td>   2518</td><td>   637</td><td>20.7162076</td><td> 0.354018455</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta    </td><td>Prevotella    </td><td>oral          </td><td>taxon</td><td>299                 </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0415-0080</td><td>1157405</td><td>625658</td><td>71.7810391</td><td>67.443467432</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0715-0061</td><td>1366176</td><td>478777</td><td>71.1279913</td><td>67.259332058</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>0816-0014</td><td>1345056</td><td>368174</td><td>71.5775292</td><td>67.434151292</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>WHB3477  </td><td>1334294</td><td>748280</td><td>71.2235745</td><td>67.240309163</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Pseudomonas_mendocina_NZ_CP013124.1      </td><td>WHB3479  </td><td>1431040</td><td>797696</td><td>71.3566957</td><td>67.599927815</td><td>Pseudomonas_mendocina_NZ_CP013124.1.fasta      </td><td>Pseudomonas   </td><td>mendocina     </td><td>NZ   </td><td>CP013124.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td><td>WHB6775  </td><td>     12</td><td>    21</td><td> 0.3587976</td><td> 0.000000000</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>      NA</td></tr>\n",
       "\t<tr><td>Rickettsia_Tillamook_23_NZ_CP060138.1    </td><td>WHB9978  </td><td>   2923</td><td>  6032</td><td>63.3608136</td><td> 5.514696940</td><td>Rickettsia_Tillamook_23_NZ_CP060138.1.fasta    </td><td>Rickettsia    </td><td>Tillamook     </td><td>23   </td><td>NZ                  </td><td>240,832</td><td>888,370  </td><td>1.346660</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0021</td><td>  12261</td><td>     6</td><td> 0.2456782</td><td> 0.030264437</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0024</td><td>  18768</td><td>     0</td><td> 0.2461658</td><td> 0.048340594</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0026</td><td>  14111</td><td>     2</td><td> 0.2183765</td><td> 0.045902938</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0031</td><td>  16398</td><td>     3</td><td> 0.2671671</td><td> 0.117420015</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_cohnii_NZ_UHDA01000001.1  </td><td>0416-0032</td><td>  18246</td><td>     8</td><td> 0.2688547</td><td> 0.062779018</td><td>Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta  </td><td>Staphylococcus</td><td>cohnii        </td><td>NZ   </td><td>UHDA01000001.1.fasta</td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB4219  </td><td>    992</td><td>   922</td><td> 8.6830661</td><td> 0.016703818</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB4398  </td><td>    546</td><td>   538</td><td> 5.4451608</td><td> 0.005797684</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB6459  </td><td>   9140</td><td>  7791</td><td>52.9612382</td><td> 1.443299053</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>1.566844</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB6651  </td><td>    830</td><td>   770</td><td> 7.6927162</td><td> 0.008392452</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_epidermidis_NZ_CP035288.1 </td><td>WHB8186  </td><td>    232</td><td>   215</td><td> 2.0381090</td><td> 0.005230079</td><td>Staphylococcus_epidermidis_NZ_CP035288.1.fasta </td><td>Staphylococcus</td><td>epidermidis   </td><td>NZ   </td><td>CP035288.1.fasta    </td><td>237,449</td><td>1,347,375</td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0024</td><td> 117929</td><td>     2</td><td> 0.2597145</td><td> 0.047698657</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0026</td><td>  94270</td><td>     1</td><td> 0.2356298</td><td> 0.038794385</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0027</td><td>  76946</td><td>     8</td><td> 0.2644608</td><td> 0.066566301</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0416-0031</td><td> 108245</td><td>     5</td><td> 0.2961161</td><td> 0.119246644</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Staphylococcus_haemolyticus_NZ_CP013911.1</td><td>0715-0059</td><td>  81100</td><td>     7</td><td> 0.2889770</td><td> 0.151725664</td><td>Staphylococcus_haemolyticus_NZ_CP013911.1.fasta</td><td>Staphylococcus</td><td>haemolyticus  </td><td>NZ   </td><td>CP013911.1.fasta    </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1          </td><td>WHB7619  </td><td>    496</td><td>    60</td><td>21.5924426</td><td>19.946018893</td><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "\t<tr><td>Torque_teno_virus_6_NC_014094.1          </td><td>WHB9685  </td><td>     39</td><td>    22</td><td>16.0863698</td><td>14.170040486</td><td>Torque_teno_virus_6_NC_014094.1.fasta          </td><td>Torque        </td><td>teno          </td><td>virus</td><td>6                   </td><td>0      </td><td>0        </td><td>      NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 138 × 14\n",
       "\\begin{tabular}{llllllllllllll}\n",
       " prefix & npm\\_research\\_id & pairs\\_assigned & pairs\\_mapped & perc\\_covered1 & perc\\_covered5 & file\\_name & genus & species & suffix1 & suffix2 & ORI & TER & bPTR\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & 1116-0023 & 430178 & 429196 & 79.292320359 & 7.450282e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3658   & 385855 & 438419 & 78.389787758 & 7.366085e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3674   & 459561 & 494858 & 78.805094251 & 7.456206e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3686   & 541795 & 599335 & 79.082105806 & 7.525644e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1    & WHB3734   & 688629 & 769978 & 79.583059893 & 7.609348e+01 & Achromobacter\\_xylosoxidans\\_NZ\\_CP043820.1.fasta    & Achromobacter   & xylosoxidans  & NZ & CP043820.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & 0116-0053 &   7673 &  28842 & 79.329072139 & 1.697212e+01 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 & 1.896385\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4240   &    168 &    210 &  1.316042864 & 1.963529e-03 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4251   &    183 &    233 &  1.460462955 & 4.782956e-04 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4438   &    379 &    453 &  2.824813672 & 3.373242e-03 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA\\\\\n",
       "\t Acinetobacter\\_baumannii\\_NZ\\_CP043953.1       & WHB4828   &    190 &    231 &  1.642920130 & 0.000000e+00 & Acinetobacter\\_baumannii\\_NZ\\_CP043953.1.fasta       & Acinetobacter   & baumannii     & NZ & CP043953.1.fasta     & 3,869,401 & 1,967,227 &       NA\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9245   &    982 &   6163 & 29.029117408 & 6.902854e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9246   &    896 &   5202 & 27.000516799 & 3.080480e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9270   &    996 &   6128 & 28.984145520 & 6.330785e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9282   &    766 &   4490 & 23.360108613 & 2.885617e-01 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Alcaligenes\\_faecalis\\_NZ\\_CP013119.1          & WHB9317   &   1104 &   7106 & 31.747814470 & 1.073538e+00 & Alcaligenes\\_faecalis\\_NZ\\_CP013119.1.fasta          & Alcaligenes     & faecalis      & NZ & CP013119.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB3883   &     24 &      4 &  0.013741771 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4048   &     32 &      0 &  0.000000000 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4056   &   7909 &     31 &  0.169732339 & 4.370987e-03 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHB4130   &     99 &      0 &  0.002315856 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA\\\\\n",
       "\t Aspergillus\\_oryzae\\_NC\\_036435.1              & WHH1822   &     12 &      0 &  0.000000000 & 0.000000e+00 & Aspergillus\\_oryzae\\_NC\\_036435.1.fasta              & Aspergillus     & oryzae        & NC & 036435.1.fasta       & 0         & 0         &       NA\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6459   &  14476 &  13069 & 66.396833288 & 7.454033e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6776   &    215 &    200 &  2.189016786 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB6849   &     31 &     25 &  0.249342726 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB8186   &     79 &     83 &  0.799303964 & 6.741270e-04 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1   & WHB8568   &     38 &     35 &  0.346627680 & 0.000000e+00 & Corynebacterium\\_segmentosum\\_NZ\\_LR134408.1.fasta   & Corynebacterium & segmentosum   & NZ & LR134408.1.fasta     & 0         & 0         &       NA\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB6155   &    152 &     19 &  0.444721342 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9217   &    297 &     31 &  0.701307112 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9583   &   5506 &    792 & 15.704852586 & 4.270964e-02 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9649   &   1972 &    290 &  6.521295389 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA\\\\\n",
       "\t Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1 & WHB9882   &    835 &    115 &  2.457894931 & 0.000000e+00 & Cupriavidus\\_metallidurans\\_NZ\\_FYAX01000013.1.fasta & Cupriavidus     & metallidurans & NZ & FYAX01000013.1.fasta & 0         & 0         &       NA\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB3939   &     141 &     74 &  1.0932372 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB5008   &      79 &     34 &  0.4487845 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB8397   &     150 &     65 &  0.9019109 &  0.000000000 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA\\\\\n",
       "\t Prevotella\\_melaninogenica\\_NC\\_014370.1     & WHB9179   &    3549 &   1971 & 25.1270313 &  0.224948898 & Prevotella\\_melaninogenica\\_NC\\_014370.1.fasta     & Prevotella     & melaninogenica & NC    & 014370.1.fasta       & 0       & 0         &       NA\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1     & WHB4373   &      79 &     25 &  1.0017609 &  0.000000000 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         &       NA\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1     & WHB9179   &    2518 &    637 & 20.7162076 &  0.354018455 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta     & Prevotella     & oral           & taxon & 299                  & 0       & 0         &       NA\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0415-0080 & 1157405 & 625658 & 71.7810391 & 67.443467432 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0715-0061 & 1366176 & 478777 & 71.1279913 & 67.259332058 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & 0816-0014 & 1345056 & 368174 & 71.5775292 & 67.434151292 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & WHB3477   & 1334294 & 748280 & 71.2235745 & 67.240309163 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Pseudomonas\\_mendocina\\_NZ\\_CP013124.1       & WHB3479   & 1431040 & 797696 & 71.3566957 & 67.599927815 & Pseudomonas\\_mendocina\\_NZ\\_CP013124.1.fasta       & Pseudomonas    & mendocina      & NZ    & CP013124.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1     & WHB6775   &      12 &     21 &  0.3587976 &  0.000000000 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   &       NA\\\\\n",
       "\t Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1     & WHB9978   &    2923 &   6032 & 63.3608136 &  5.514696940 & Rickettsia\\_Tillamook\\_23\\_NZ\\_CP060138.1.fasta     & Rickettsia     & Tillamook      & 23    & NZ                   & 240,832 & 888,370   & 1.346660\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0021 &   12261 &      6 &  0.2456782 &  0.030264437 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0024 &   18768 &      0 &  0.2461658 &  0.048340594 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0026 &   14111 &      2 &  0.2183765 &  0.045902938 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0031 &   16398 &      3 &  0.2671671 &  0.117420015 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1   & 0416-0032 &   18246 &      8 &  0.2688547 &  0.062779018 & Staphylococcus\\_cohnii\\_NZ\\_UHDA01000001.1.fasta   & Staphylococcus & cohnii         & NZ    & UHDA01000001.1.fasta & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB4219   &     992 &    922 &  8.6830661 &  0.016703818 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB4398   &     546 &    538 &  5.4451608 &  0.005797684 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB6459   &    9140 &   7791 & 52.9612382 &  1.443299053 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 & 1.566844\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB6651   &     830 &    770 &  7.6927162 &  0.008392452 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA\\\\\n",
       "\t Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1  & WHB8186   &     232 &    215 &  2.0381090 &  0.005230079 & Staphylococcus\\_epidermidis\\_NZ\\_CP035288.1.fasta  & Staphylococcus & epidermidis    & NZ    & CP035288.1.fasta     & 237,449 & 1,347,375 &       NA\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0024 &  117929 &      2 &  0.2597145 &  0.047698657 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0026 &   94270 &      1 &  0.2356298 &  0.038794385 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0027 &   76946 &      8 &  0.2644608 &  0.066566301 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0416-0031 &  108245 &      5 &  0.2961161 &  0.119246644 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1 & 0715-0059 &   81100 &      7 &  0.2889770 &  0.151725664 & Staphylococcus\\_haemolyticus\\_NZ\\_CP013911.1.fasta & Staphylococcus & haemolyticus   & NZ    & CP013911.1.fasta     & 0       & 0         &       NA\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1           & WHB7619   &     496 &     60 & 21.5924426 & 19.946018893 & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         &       NA\\\\\n",
       "\t Torque\\_teno\\_virus\\_6\\_NC\\_014094.1           & WHB9685   &      39 &     22 & 16.0863698 & 14.170040486 & Torque\\_teno\\_virus\\_6\\_NC\\_014094.1.fasta           & Torque         & teno           & virus & 6                    & 0       & 0         &       NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 138 × 14\n",
       "\n",
       "| prefix &lt;chr&gt; | npm_research_id &lt;chr&gt; | pairs_assigned &lt;int&gt; | pairs_mapped &lt;dbl&gt; | perc_covered1 &lt;dbl&gt; | perc_covered5 &lt;dbl&gt; | file_name &lt;chr&gt; | genus &lt;chr&gt; | species &lt;chr&gt; | suffix1 &lt;chr&gt; | suffix2 &lt;chr&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; | bPTR &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | 1116-0023 | 430178 | 429196 | 79.292320359 | 7.450282e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3658   | 385855 | 438419 | 78.389787758 | 7.366085e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3674   | 459561 | 494858 | 78.805094251 | 7.456206e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3686   | 541795 | 599335 | 79.082105806 | 7.525644e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA |\n",
       "| Achromobacter_xylosoxidans_NZ_CP043820.1    | WHB3734   | 688629 | 769978 | 79.583059893 | 7.609348e+01 | Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    | Achromobacter   | xylosoxidans  | NZ | CP043820.1.fasta     | 0         | 0         |       NA |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | 0116-0053 |   7673 |  28842 | 79.329072139 | 1.697212e+01 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 | 1.896385 |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4240   |    168 |    210 |  1.316042864 | 1.963529e-03 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4251   |    183 |    233 |  1.460462955 | 4.782956e-04 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4438   |    379 |    453 |  2.824813672 | 3.373242e-03 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA |\n",
       "| Acinetobacter_baumannii_NZ_CP043953.1       | WHB4828   |    190 |    231 |  1.642920130 | 0.000000e+00 | Acinetobacter_baumannii_NZ_CP043953.1.fasta       | Acinetobacter   | baumannii     | NZ | CP043953.1.fasta     | 3,869,401 | 1,967,227 |       NA |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9245   |    982 |   6163 | 29.029117408 | 6.902854e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9246   |    896 |   5202 | 27.000516799 | 3.080480e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9270   |    996 |   6128 | 28.984145520 | 6.330785e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9282   |    766 |   4490 | 23.360108613 | 2.885617e-01 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA |\n",
       "| Alcaligenes_faecalis_NZ_CP013119.1          | WHB9317   |   1104 |   7106 | 31.747814470 | 1.073538e+00 | Alcaligenes_faecalis_NZ_CP013119.1.fasta          | Alcaligenes     | faecalis      | NZ | CP013119.1.fasta     | 0         | 0         |       NA |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB3883   |     24 |      4 |  0.013741771 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4048   |     32 |      0 |  0.000000000 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4056   |   7909 |     31 |  0.169732339 | 4.370987e-03 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHB4130   |     99 |      0 |  0.002315856 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA |\n",
       "| Aspergillus_oryzae_NC_036435.1              | WHH1822   |     12 |      0 |  0.000000000 | 0.000000e+00 | Aspergillus_oryzae_NC_036435.1.fasta              | Aspergillus     | oryzae        | NC | 036435.1.fasta       | 0         | 0         |       NA |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6459   |  14476 |  13069 | 66.396833288 | 7.454033e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6776   |    215 |    200 |  2.189016786 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB6849   |     31 |     25 |  0.249342726 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB8186   |     79 |     83 |  0.799303964 | 6.741270e-04 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA |\n",
       "| Corynebacterium_segmentosum_NZ_LR134408.1   | WHB8568   |     38 |     35 |  0.346627680 | 0.000000e+00 | Corynebacterium_segmentosum_NZ_LR134408.1.fasta   | Corynebacterium | segmentosum   | NZ | LR134408.1.fasta     | 0         | 0         |       NA |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB6155   |    152 |     19 |  0.444721342 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9217   |    297 |     31 |  0.701307112 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9583   |   5506 |    792 | 15.704852586 | 4.270964e-02 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9649   |   1972 |    290 |  6.521295389 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA |\n",
       "| Cupriavidus_metallidurans_NZ_FYAX01000013.1 | WHB9882   |    835 |    115 |  2.457894931 | 0.000000e+00 | Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta | Cupriavidus     | metallidurans | NZ | FYAX01000013.1.fasta | 0         | 0         |       NA |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB3939   |     141 |     74 |  1.0932372 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB5008   |      79 |     34 |  0.4487845 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB8397   |     150 |     65 |  0.9019109 |  0.000000000 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA |\n",
       "| Prevotella_melaninogenica_NC_014370.1     | WHB9179   |    3549 |   1971 | 25.1270313 |  0.224948898 | Prevotella_melaninogenica_NC_014370.1.fasta     | Prevotella     | melaninogenica | NC    | 014370.1.fasta       | 0       | 0         |       NA |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1     | WHB4373   |      79 |     25 |  1.0017609 |  0.000000000 | Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         |       NA |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1     | WHB9179   |    2518 |    637 | 20.7162076 |  0.354018455 | Prevotella_oral_taxon_299_NC_022124.1.fasta     | Prevotella     | oral           | taxon | 299                  | 0       | 0         |       NA |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0415-0080 | 1157405 | 625658 | 71.7810391 | 67.443467432 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0715-0061 | 1366176 | 478777 | 71.1279913 | 67.259332058 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | 0816-0014 | 1345056 | 368174 | 71.5775292 | 67.434151292 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | WHB3477   | 1334294 | 748280 | 71.2235745 | 67.240309163 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA |\n",
       "| Pseudomonas_mendocina_NZ_CP013124.1       | WHB3479   | 1431040 | 797696 | 71.3566957 | 67.599927815 | Pseudomonas_mendocina_NZ_CP013124.1.fasta       | Pseudomonas    | mendocina      | NZ    | CP013124.1.fasta     | 0       | 0         |       NA |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1     | WHB6775   |      12 |     21 |  0.3587976 |  0.000000000 | Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   |       NA |\n",
       "| Rickettsia_Tillamook_23_NZ_CP060138.1     | WHB9978   |    2923 |   6032 | 63.3608136 |  5.514696940 | Rickettsia_Tillamook_23_NZ_CP060138.1.fasta     | Rickettsia     | Tillamook      | 23    | NZ                   | 240,832 | 888,370   | 1.346660 |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0021 |   12261 |      6 |  0.2456782 |  0.030264437 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0024 |   18768 |      0 |  0.2461658 |  0.048340594 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0026 |   14111 |      2 |  0.2183765 |  0.045902938 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0031 |   16398 |      3 |  0.2671671 |  0.117420015 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA |\n",
       "| Staphylococcus_cohnii_NZ_UHDA01000001.1   | 0416-0032 |   18246 |      8 |  0.2688547 |  0.062779018 | Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta   | Staphylococcus | cohnii         | NZ    | UHDA01000001.1.fasta | 0       | 0         |       NA |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB4219   |     992 |    922 |  8.6830661 |  0.016703818 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB4398   |     546 |    538 |  5.4451608 |  0.005797684 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB6459   |    9140 |   7791 | 52.9612382 |  1.443299053 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 | 1.566844 |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB6651   |     830 |    770 |  7.6927162 |  0.008392452 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA |\n",
       "| Staphylococcus_epidermidis_NZ_CP035288.1  | WHB8186   |     232 |    215 |  2.0381090 |  0.005230079 | Staphylococcus_epidermidis_NZ_CP035288.1.fasta  | Staphylococcus | epidermidis    | NZ    | CP035288.1.fasta     | 237,449 | 1,347,375 |       NA |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0024 |  117929 |      2 |  0.2597145 |  0.047698657 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0026 |   94270 |      1 |  0.2356298 |  0.038794385 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0027 |   76946 |      8 |  0.2644608 |  0.066566301 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0416-0031 |  108245 |      5 |  0.2961161 |  0.119246644 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA |\n",
       "| Staphylococcus_haemolyticus_NZ_CP013911.1 | 0715-0059 |   81100 |      7 |  0.2889770 |  0.151725664 | Staphylococcus_haemolyticus_NZ_CP013911.1.fasta | Staphylococcus | haemolyticus   | NZ    | CP013911.1.fasta     | 0       | 0         |       NA |\n",
       "| Torque_teno_virus_6_NC_014094.1           | WHB7619   |     496 |     60 | 21.5924426 | 19.946018893 | Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         |       NA |\n",
       "| Torque_teno_virus_6_NC_014094.1           | WHB9685   |      39 |     22 | 16.0863698 | 14.170040486 | Torque_teno_virus_6_NC_014094.1.fasta           | Torque         | teno           | virus | 6                    | 0       | 0         |       NA |\n",
       "\n"
      ],
      "text/plain": [
       "    prefix                                      npm_research_id pairs_assigned\n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1    1116-0023       430178        \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3658         385855        \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3674         459561        \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3686         541795        \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1    WHB3734         688629        \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1       0116-0053         7673        \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1       WHB4240            168        \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1       WHB4251            183        \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1       WHB4438            379        \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1       WHB4828            190        \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1          WHB9245            982        \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1          WHB9246            896        \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1          WHB9270            996        \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1          WHB9282            766        \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1          WHB9317           1104        \n",
       "16  Aspergillus_oryzae_NC_036435.1              WHB3883             24        \n",
       "17  Aspergillus_oryzae_NC_036435.1              WHB4048             32        \n",
       "18  Aspergillus_oryzae_NC_036435.1              WHB4056           7909        \n",
       "19  Aspergillus_oryzae_NC_036435.1              WHB4130             99        \n",
       "20  Aspergillus_oryzae_NC_036435.1              WHH1822             12        \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6459          14476        \n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6776            215        \n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1   WHB6849             31        \n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1   WHB8186             79        \n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1   WHB8568             38        \n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB6155            152        \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9217            297        \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9583           5506        \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9649           1972        \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1 WHB9882            835        \n",
       "⋮   ⋮                                           ⋮               ⋮             \n",
       "109 Prevotella_melaninogenica_NC_014370.1       WHB3939             141       \n",
       "110 Prevotella_melaninogenica_NC_014370.1       WHB5008              79       \n",
       "111 Prevotella_melaninogenica_NC_014370.1       WHB8397             150       \n",
       "112 Prevotella_melaninogenica_NC_014370.1       WHB9179            3549       \n",
       "113 Prevotella_oral_taxon_299_NC_022124.1       WHB4373              79       \n",
       "114 Prevotella_oral_taxon_299_NC_022124.1       WHB9179            2518       \n",
       "115 Pseudomonas_mendocina_NZ_CP013124.1         0415-0080       1157405       \n",
       "116 Pseudomonas_mendocina_NZ_CP013124.1         0715-0061       1366176       \n",
       "117 Pseudomonas_mendocina_NZ_CP013124.1         0816-0014       1345056       \n",
       "118 Pseudomonas_mendocina_NZ_CP013124.1         WHB3477         1334294       \n",
       "119 Pseudomonas_mendocina_NZ_CP013124.1         WHB3479         1431040       \n",
       "120 Rickettsia_Tillamook_23_NZ_CP060138.1       WHB6775              12       \n",
       "121 Rickettsia_Tillamook_23_NZ_CP060138.1       WHB9978            2923       \n",
       "122 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0021         12261       \n",
       "123 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0024         18768       \n",
       "124 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0026         14111       \n",
       "125 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0031         16398       \n",
       "126 Staphylococcus_cohnii_NZ_UHDA01000001.1     0416-0032         18246       \n",
       "127 Staphylococcus_epidermidis_NZ_CP035288.1    WHB4219             992       \n",
       "128 Staphylococcus_epidermidis_NZ_CP035288.1    WHB4398             546       \n",
       "129 Staphylococcus_epidermidis_NZ_CP035288.1    WHB6459            9140       \n",
       "130 Staphylococcus_epidermidis_NZ_CP035288.1    WHB6651             830       \n",
       "131 Staphylococcus_epidermidis_NZ_CP035288.1    WHB8186             232       \n",
       "132 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0024        117929       \n",
       "133 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0026         94270       \n",
       "134 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0027         76946       \n",
       "135 Staphylococcus_haemolyticus_NZ_CP013911.1   0416-0031        108245       \n",
       "136 Staphylococcus_haemolyticus_NZ_CP013911.1   0715-0059         81100       \n",
       "137 Torque_teno_virus_6_NC_014094.1             WHB7619             496       \n",
       "138 Torque_teno_virus_6_NC_014094.1             WHB9685              39       \n",
       "    pairs_mapped perc_covered1 perc_covered5\n",
       "1   429196       79.292320359  7.450282e+01 \n",
       "2   438419       78.389787758  7.366085e+01 \n",
       "3   494858       78.805094251  7.456206e+01 \n",
       "4   599335       79.082105806  7.525644e+01 \n",
       "5   769978       79.583059893  7.609348e+01 \n",
       "6    28842       79.329072139  1.697212e+01 \n",
       "7      210        1.316042864  1.963529e-03 \n",
       "8      233        1.460462955  4.782956e-04 \n",
       "9      453        2.824813672  3.373242e-03 \n",
       "10     231        1.642920130  0.000000e+00 \n",
       "11    6163       29.029117408  6.902854e-01 \n",
       "12    5202       27.000516799  3.080480e-01 \n",
       "13    6128       28.984145520  6.330785e-01 \n",
       "14    4490       23.360108613  2.885617e-01 \n",
       "15    7106       31.747814470  1.073538e+00 \n",
       "16       4        0.013741771  0.000000e+00 \n",
       "17       0        0.000000000  0.000000e+00 \n",
       "18      31        0.169732339  4.370987e-03 \n",
       "19       0        0.002315856  0.000000e+00 \n",
       "20       0        0.000000000  0.000000e+00 \n",
       "21   13069       66.396833288  7.454033e+00 \n",
       "22     200        2.189016786  0.000000e+00 \n",
       "23      25        0.249342726  0.000000e+00 \n",
       "24      83        0.799303964  6.741270e-04 \n",
       "25      35        0.346627680  0.000000e+00 \n",
       "26      19        0.444721342  0.000000e+00 \n",
       "27      31        0.701307112  0.000000e+00 \n",
       "28     792       15.704852586  4.270964e-02 \n",
       "29     290        6.521295389  0.000000e+00 \n",
       "30     115        2.457894931  0.000000e+00 \n",
       "⋮   ⋮            ⋮             ⋮            \n",
       "109     74        1.0932372     0.000000000 \n",
       "110     34        0.4487845     0.000000000 \n",
       "111     65        0.9019109     0.000000000 \n",
       "112   1971       25.1270313     0.224948898 \n",
       "113     25        1.0017609     0.000000000 \n",
       "114    637       20.7162076     0.354018455 \n",
       "115 625658       71.7810391    67.443467432 \n",
       "116 478777       71.1279913    67.259332058 \n",
       "117 368174       71.5775292    67.434151292 \n",
       "118 748280       71.2235745    67.240309163 \n",
       "119 797696       71.3566957    67.599927815 \n",
       "120     21        0.3587976     0.000000000 \n",
       "121   6032       63.3608136     5.514696940 \n",
       "122      6        0.2456782     0.030264437 \n",
       "123      0        0.2461658     0.048340594 \n",
       "124      2        0.2183765     0.045902938 \n",
       "125      3        0.2671671     0.117420015 \n",
       "126      8        0.2688547     0.062779018 \n",
       "127    922        8.6830661     0.016703818 \n",
       "128    538        5.4451608     0.005797684 \n",
       "129   7791       52.9612382     1.443299053 \n",
       "130    770        7.6927162     0.008392452 \n",
       "131    215        2.0381090     0.005230079 \n",
       "132      2        0.2597145     0.047698657 \n",
       "133      1        0.2356298     0.038794385 \n",
       "134      8        0.2644608     0.066566301 \n",
       "135      5        0.2961161     0.119246644 \n",
       "136      7        0.2889770     0.151725664 \n",
       "137     60       21.5924426    19.946018893 \n",
       "138     22       16.0863698    14.170040486 \n",
       "    file_name                                         genus          \n",
       "1   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "2   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "3   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "4   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "5   Achromobacter_xylosoxidans_NZ_CP043820.1.fasta    Achromobacter  \n",
       "6   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "7   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "8   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "9   Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "10  Acinetobacter_baumannii_NZ_CP043953.1.fasta       Acinetobacter  \n",
       "11  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "12  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "13  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "14  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "15  Alcaligenes_faecalis_NZ_CP013119.1.fasta          Alcaligenes    \n",
       "16  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "17  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "18  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "19  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "20  Aspergillus_oryzae_NC_036435.1.fasta              Aspergillus    \n",
       "21  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "22  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "23  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "24  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "25  Corynebacterium_segmentosum_NZ_LR134408.1.fasta   Corynebacterium\n",
       "26  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "27  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "28  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "29  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "30  Cupriavidus_metallidurans_NZ_FYAX01000013.1.fasta Cupriavidus    \n",
       "⋮   ⋮                                                 ⋮              \n",
       "109 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "110 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "111 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "112 Prevotella_melaninogenica_NC_014370.1.fasta       Prevotella     \n",
       "113 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "114 Prevotella_oral_taxon_299_NC_022124.1.fasta       Prevotella     \n",
       "115 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "116 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "117 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "118 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "119 Pseudomonas_mendocina_NZ_CP013124.1.fasta         Pseudomonas    \n",
       "120 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "121 Rickettsia_Tillamook_23_NZ_CP060138.1.fasta       Rickettsia     \n",
       "122 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "123 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "124 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "125 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "126 Staphylococcus_cohnii_NZ_UHDA01000001.1.fasta     Staphylococcus \n",
       "127 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "128 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "129 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "130 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "131 Staphylococcus_epidermidis_NZ_CP035288.1.fasta    Staphylococcus \n",
       "132 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "133 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "134 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "135 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "136 Staphylococcus_haemolyticus_NZ_CP013911.1.fasta   Staphylococcus \n",
       "137 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "138 Torque_teno_virus_6_NC_014094.1.fasta             Torque         \n",
       "    species        suffix1 suffix2              ORI       TER       bPTR    \n",
       "1   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "2   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "3   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "4   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "5   xylosoxidans   NZ      CP043820.1.fasta     0         0               NA\n",
       "6   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227 1.896385\n",
       "7   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "8   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "9   baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "10  baumannii      NZ      CP043953.1.fasta     3,869,401 1,967,227       NA\n",
       "11  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "12  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "13  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "14  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "15  faecalis       NZ      CP013119.1.fasta     0         0               NA\n",
       "16  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "17  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "18  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "19  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "20  oryzae         NC      036435.1.fasta       0         0               NA\n",
       "21  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "22  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "23  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "24  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "25  segmentosum    NZ      LR134408.1.fasta     0         0               NA\n",
       "26  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "27  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "28  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "29  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "30  metallidurans  NZ      FYAX01000013.1.fasta 0         0               NA\n",
       "⋮   ⋮              ⋮       ⋮                    ⋮         ⋮         ⋮       \n",
       "109 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "110 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "111 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "112 melaninogenica NC      014370.1.fasta       0         0               NA\n",
       "113 oral           taxon   299                  0         0               NA\n",
       "114 oral           taxon   299                  0         0               NA\n",
       "115 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "116 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "117 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "118 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "119 mendocina      NZ      CP013124.1.fasta     0         0               NA\n",
       "120 Tillamook      23      NZ                   240,832   888,370         NA\n",
       "121 Tillamook      23      NZ                   240,832   888,370   1.346660\n",
       "122 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "123 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "124 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "125 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "126 cohnii         NZ      UHDA01000001.1.fasta 0         0               NA\n",
       "127 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "128 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "129 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375 1.566844\n",
       "130 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "131 epidermidis    NZ      CP035288.1.fasta     237,449   1,347,375       NA\n",
       "132 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "133 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "134 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "135 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "136 haemolyticus   NZ      CP013911.1.fasta     0         0               NA\n",
       "137 teno           virus   6                    0         0               NA\n",
       "138 teno           virus   6                    0         0               NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "merged_df <- cov_df %>%\n",
    "    left_join(mapping_df) %>%\n",
    "    left_join(irep_df) %>%\n",
    "    relocate(pairs_mapped, .after = 3) %>%\n",
    "    mutate(pairs_mapped = pairs_mapped / 2)\n",
    "\n",
    "merged_df\n",
    "\n",
    "fwrite(merged_df, \"results/irep_analysis/coverage_irep_results.raw.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "dd23b170",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>prefix</th><th scope=col>npm_research_id</th><th scope=col>pairs_assigned</th><th scope=col>pairs_mapped</th><th scope=col>perc_covered1</th><th scope=col>perc_covered5</th><th scope=col>file_name</th><th scope=col>genus</th><th scope=col>species</th><th scope=col>suffix1</th><th scope=col>suffix2</th><th scope=col>ORI</th><th scope=col>TER</th><th scope=col>bPTR</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1</td><td>WHB4373</td><td>  79</td><td> 25</td><td> 1.001761</td><td>0.0000000</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta</td><td>Prevotella</td><td>oral</td><td>taxon</td><td>299</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "\t<tr><td>Prevotella_oral_taxon_299_NC_022124.1</td><td>WHB9179</td><td>2518</td><td>637</td><td>20.716208</td><td>0.3540185</td><td>Prevotella_oral_taxon_299_NC_022124.1.fasta</td><td>Prevotella</td><td>oral</td><td>taxon</td><td>299</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 14\n",
       "\\begin{tabular}{llllllllllllll}\n",
       " prefix & npm\\_research\\_id & pairs\\_assigned & pairs\\_mapped & perc\\_covered1 & perc\\_covered5 & file\\_name & genus & species & suffix1 & suffix2 & ORI & TER & bPTR\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1 & WHB4373 &   79 &  25 &  1.001761 & 0.0000000 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta & Prevotella & oral & taxon & 299 & 0 & 0 & NA\\\\\n",
       "\t Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1 & WHB9179 & 2518 & 637 & 20.716208 & 0.3540185 & Prevotella\\_oral\\_taxon\\_299\\_NC\\_022124.1.fasta & Prevotella & oral & taxon & 299 & 0 & 0 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 14\n",
       "\n",
       "| prefix &lt;chr&gt; | npm_research_id &lt;chr&gt; | pairs_assigned &lt;int&gt; | pairs_mapped &lt;dbl&gt; | perc_covered1 &lt;dbl&gt; | perc_covered5 &lt;dbl&gt; | file_name &lt;chr&gt; | genus &lt;chr&gt; | species &lt;chr&gt; | suffix1 &lt;chr&gt; | suffix2 &lt;chr&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; | bPTR &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Prevotella_oral_taxon_299_NC_022124.1 | WHB4373 |   79 |  25 |  1.001761 | 0.0000000 | Prevotella_oral_taxon_299_NC_022124.1.fasta | Prevotella | oral | taxon | 299 | 0 | 0 | NA |\n",
       "| Prevotella_oral_taxon_299_NC_022124.1 | WHB9179 | 2518 | 637 | 20.716208 | 0.3540185 | Prevotella_oral_taxon_299_NC_022124.1.fasta | Prevotella | oral | taxon | 299 | 0 | 0 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  prefix                                npm_research_id pairs_assigned\n",
       "1 Prevotella_oral_taxon_299_NC_022124.1 WHB4373           79          \n",
       "2 Prevotella_oral_taxon_299_NC_022124.1 WHB9179         2518          \n",
       "  pairs_mapped perc_covered1 perc_covered5\n",
       "1  25           1.001761     0.0000000    \n",
       "2 637          20.716208     0.3540185    \n",
       "  file_name                                   genus      species suffix1\n",
       "1 Prevotella_oral_taxon_299_NC_022124.1.fasta Prevotella oral    taxon  \n",
       "2 Prevotella_oral_taxon_299_NC_022124.1.fasta Prevotella oral    taxon  \n",
       "  suffix2 ORI TER bPTR\n",
       "1 299     0   0   NA  \n",
       "2 299     0   0   NA  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "merged_df %>% filter(grepl(\"oral\", prefix))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8c39a120",
   "metadata": {},
   "outputs": [],
   "source": [
    "ori_df <- merged_df %>%\n",
    "    select(prefix, ORI, TER) %>%\n",
    "    filter(ORI != \"0\" & TER != \"0\") %>%\n",
    "    distinct()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0a62c820",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Warning message in max(bPTR, na.rm = T):\n",
      "“no non-missing arguments to max; returning -Inf”\n",
      "Joining, by = \"prefix\"\n",
      "Warning message:\n",
      "“Expected 4 pieces. Additional pieces discarded in 7 rows [13, 14, 15, 16, 24, 26, 30].”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 30 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>taxa</th><th scope=col>mean_perc_covered1</th><th scope=col>mean_perc_covered5</th><th scope=col>min_perc_covered1</th><th scope=col>max_perc_covered1</th><th scope=col>max_bPTR</th><th scope=col>ORI</th><th scope=col>TER</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Human betaherpesvirus 6B         </td><td>98.48291943</td><td>9.544136e+01</td><td>98.34129070</td><td>98.6928951</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Human betaherpesvirus 6A         </td><td>96.76555108</td><td>9.096362e+01</td><td>96.63755349</td><td>96.9826450</td><td>1.282505</td><td>44,954   </td><td>132,612  </td></tr>\n",
       "\t<tr><td>Achromobacter xylosoxidans       </td><td>79.03047361</td><td>7.481513e+01</td><td>78.38978776</td><td>79.5830599</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Hepatitis B virus                </td><td>76.39191291</td><td>4.486262e+01</td><td>61.33748056</td><td>98.7558320</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Pseudomonas mendocina            </td><td>71.41336595</td><td>6.739544e+01</td><td>71.12799127</td><td>71.7810391</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Cutibacterium acnes              </td><td>36.15403798</td><td>3.473737e+00</td><td>15.33223211</td><td>77.4795425</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Rickettsia sp. Tillamook 23      </td><td>31.85980557</td><td>2.757348e+00</td><td> 0.35879756</td><td>63.3608136</td><td>1.346660</td><td>240,832  </td><td>888,370  </td></tr>\n",
       "\t<tr><td>Alcaligenes faecalis             </td><td>28.02434056</td><td>5.987024e-01</td><td>23.36010861</td><td>31.7478145</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Torque teno virus 6              </td><td>18.83940621</td><td>1.705803e+01</td><td>16.08636977</td><td>21.5924426</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Neisseria subflava               </td><td>17.89799782</td><td>5.411104e+00</td><td> 1.20911307</td><td>81.7316806</td><td>1.506405</td><td>1,883,368</td><td>797,590  </td></tr>\n",
       "\t<tr><td>Acinetobacter baumannii          </td><td>17.31466235</td><td>3.395586e+00</td><td> 1.31604286</td><td>79.3290721</td><td>1.896385</td><td>3,869,401</td><td>1,967,227</td></tr>\n",
       "\t<tr><td>Fannyhessea vaginae              </td><td>17.26185948</td><td>1.865483e+00</td><td> 1.22422672</td><td>75.9249475</td><td>1.876599</td><td>613,931  </td><td>1,264,620</td></tr>\n",
       "\t<tr><td>Neisseria flavescens             </td><td>16.77717729</td><td>4.927474e+00</td><td> 1.11999649</td><td>76.9177761</td><td>1.380708</td><td>1,737,155</td><td>716,802  </td></tr>\n",
       "\t<tr><td>Staphylococcus epidermidis       </td><td>15.36405809</td><td>2.958846e-01</td><td> 2.03810903</td><td>52.9612382</td><td>1.566844</td><td>237,449  </td><td>1,347,375</td></tr>\n",
       "\t<tr><td>Haemophilus parainfluenzae       </td><td>14.73871027</td><td>7.062885e-01</td><td> 0.65722936</td><td>60.4802455</td><td>1.172580</td><td>492,915  </td><td>1,443,748</td></tr>\n",
       "\t<tr><td>Fusobacterium nucleatum          </td><td>14.22900825</td><td>1.150248e+01</td><td> 0.32135060</td><td>65.7792926</td><td>1.675886</td><td>201,937  </td><td>1,345,205</td></tr>\n",
       "\t<tr><td>Corynebacterium segmentosum      </td><td>13.99622489</td><td>1.490941e+00</td><td> 0.24934273</td><td>66.3968333</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Lactobacillus crispatus          </td><td>13.00527270</td><td>3.891430e-01</td><td> 3.96804644</td><td>41.0523459</td><td>1.571468</td><td>328,627  </td><td>1,239,005</td></tr>\n",
       "\t<tr><td>Fusobacterium pseudoperiodonticum</td><td>11.77652109</td><td>6.551175e+00</td><td> 0.27596339</td><td>39.3815911</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Prevotella sp. oral taxon 299    </td><td>10.85898429</td><td>1.770092e-01</td><td> 1.00176094</td><td>20.7162076</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Moraxella osloensis              </td><td> 9.14462141</td><td>1.206725e-02</td><td> 5.59357092</td><td>17.0975911</td><td>1.329333</td><td>1,501,499</td><td>405,889  </td></tr>\n",
       "\t<tr><td>Gardnerella vaginalis            </td><td> 8.67918722</td><td>1.926305e-02</td><td> 5.29527766</td><td>14.6525344</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Human mastadenovirus C           </td><td> 8.37354259</td><td>6.947158e+00</td><td> 4.61084676</td><td>22.3363108</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Prevotella melaninogenica        </td><td> 6.79420265</td><td>5.026698e-02</td><td> 0.44878446</td><td>25.1270313</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Cupriavidus metallidurans        </td><td> 5.16601427</td><td>8.541929e-03</td><td> 0.44472134</td><td>15.7048526</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Neisseria mucosa                 </td><td> 5.01441152</td><td>9.105637e-01</td><td> 0.11922013</td><td>22.5393464</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Malassezia restricta             </td><td> 3.38365554</td><td>8.145150e-03</td><td> 0.04072575</td><td>12.5383255</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Staphylococcus haemolyticus      </td><td> 0.26897963</td><td>8.480633e-02</td><td> 0.23562980</td><td> 0.2961161</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Staphylococcus cohnii            </td><td> 0.24924845</td><td>6.094140e-02</td><td> 0.21837648</td><td> 0.2688547</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "\t<tr><td>Aspergillus oryzae               </td><td> 0.03715799</td><td>8.741975e-04</td><td> 0.00000000</td><td> 0.1697323</td><td>      NA</td><td>NA       </td><td>NA       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 30 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " taxa & mean\\_perc\\_covered1 & mean\\_perc\\_covered5 & min\\_perc\\_covered1 & max\\_perc\\_covered1 & max\\_bPTR & ORI & TER\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t Human betaherpesvirus 6B          & 98.48291943 & 9.544136e+01 & 98.34129070 & 98.6928951 &       NA & NA        & NA       \\\\\n",
       "\t Human betaherpesvirus 6A          & 96.76555108 & 9.096362e+01 & 96.63755349 & 96.9826450 & 1.282505 & 44,954    & 132,612  \\\\\n",
       "\t Achromobacter xylosoxidans        & 79.03047361 & 7.481513e+01 & 78.38978776 & 79.5830599 &       NA & NA        & NA       \\\\\n",
       "\t Hepatitis B virus                 & 76.39191291 & 4.486262e+01 & 61.33748056 & 98.7558320 &       NA & NA        & NA       \\\\\n",
       "\t Pseudomonas mendocina             & 71.41336595 & 6.739544e+01 & 71.12799127 & 71.7810391 &       NA & NA        & NA       \\\\\n",
       "\t Cutibacterium acnes               & 36.15403798 & 3.473737e+00 & 15.33223211 & 77.4795425 &       NA & NA        & NA       \\\\\n",
       "\t Rickettsia sp. Tillamook 23       & 31.85980557 & 2.757348e+00 &  0.35879756 & 63.3608136 & 1.346660 & 240,832   & 888,370  \\\\\n",
       "\t Alcaligenes faecalis              & 28.02434056 & 5.987024e-01 & 23.36010861 & 31.7478145 &       NA & NA        & NA       \\\\\n",
       "\t Torque teno virus 6               & 18.83940621 & 1.705803e+01 & 16.08636977 & 21.5924426 &       NA & NA        & NA       \\\\\n",
       "\t Neisseria subflava                & 17.89799782 & 5.411104e+00 &  1.20911307 & 81.7316806 & 1.506405 & 1,883,368 & 797,590  \\\\\n",
       "\t Acinetobacter baumannii           & 17.31466235 & 3.395586e+00 &  1.31604286 & 79.3290721 & 1.896385 & 3,869,401 & 1,967,227\\\\\n",
       "\t Fannyhessea vaginae               & 17.26185948 & 1.865483e+00 &  1.22422672 & 75.9249475 & 1.876599 & 613,931   & 1,264,620\\\\\n",
       "\t Neisseria flavescens              & 16.77717729 & 4.927474e+00 &  1.11999649 & 76.9177761 & 1.380708 & 1,737,155 & 716,802  \\\\\n",
       "\t Staphylococcus epidermidis        & 15.36405809 & 2.958846e-01 &  2.03810903 & 52.9612382 & 1.566844 & 237,449   & 1,347,375\\\\\n",
       "\t Haemophilus parainfluenzae        & 14.73871027 & 7.062885e-01 &  0.65722936 & 60.4802455 & 1.172580 & 492,915   & 1,443,748\\\\\n",
       "\t Fusobacterium nucleatum           & 14.22900825 & 1.150248e+01 &  0.32135060 & 65.7792926 & 1.675886 & 201,937   & 1,345,205\\\\\n",
       "\t Corynebacterium segmentosum       & 13.99622489 & 1.490941e+00 &  0.24934273 & 66.3968333 &       NA & NA        & NA       \\\\\n",
       "\t Lactobacillus crispatus           & 13.00527270 & 3.891430e-01 &  3.96804644 & 41.0523459 & 1.571468 & 328,627   & 1,239,005\\\\\n",
       "\t Fusobacterium pseudoperiodonticum & 11.77652109 & 6.551175e+00 &  0.27596339 & 39.3815911 &       NA & NA        & NA       \\\\\n",
       "\t Prevotella sp. oral taxon 299     & 10.85898429 & 1.770092e-01 &  1.00176094 & 20.7162076 &       NA & NA        & NA       \\\\\n",
       "\t Moraxella osloensis               &  9.14462141 & 1.206725e-02 &  5.59357092 & 17.0975911 & 1.329333 & 1,501,499 & 405,889  \\\\\n",
       "\t Gardnerella vaginalis             &  8.67918722 & 1.926305e-02 &  5.29527766 & 14.6525344 &       NA & NA        & NA       \\\\\n",
       "\t Human mastadenovirus C            &  8.37354259 & 6.947158e+00 &  4.61084676 & 22.3363108 &       NA & NA        & NA       \\\\\n",
       "\t Prevotella melaninogenica         &  6.79420265 & 5.026698e-02 &  0.44878446 & 25.1270313 &       NA & NA        & NA       \\\\\n",
       "\t Cupriavidus metallidurans         &  5.16601427 & 8.541929e-03 &  0.44472134 & 15.7048526 &       NA & NA        & NA       \\\\\n",
       "\t Neisseria mucosa                  &  5.01441152 & 9.105637e-01 &  0.11922013 & 22.5393464 &       NA & NA        & NA       \\\\\n",
       "\t Malassezia restricta              &  3.38365554 & 8.145150e-03 &  0.04072575 & 12.5383255 &       NA & NA        & NA       \\\\\n",
       "\t Staphylococcus haemolyticus       &  0.26897963 & 8.480633e-02 &  0.23562980 &  0.2961161 &       NA & NA        & NA       \\\\\n",
       "\t Staphylococcus cohnii             &  0.24924845 & 6.094140e-02 &  0.21837648 &  0.2688547 &       NA & NA        & NA       \\\\\n",
       "\t Aspergillus oryzae                &  0.03715799 & 8.741975e-04 &  0.00000000 &  0.1697323 &       NA & NA        & NA       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 30 × 8\n",
       "\n",
       "| taxa &lt;chr&gt; | mean_perc_covered1 &lt;dbl&gt; | mean_perc_covered5 &lt;dbl&gt; | min_perc_covered1 &lt;dbl&gt; | max_perc_covered1 &lt;dbl&gt; | max_bPTR &lt;dbl&gt; | ORI &lt;chr&gt; | TER &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| Human betaherpesvirus 6B          | 98.48291943 | 9.544136e+01 | 98.34129070 | 98.6928951 |       NA | NA        | NA        |\n",
       "| Human betaherpesvirus 6A          | 96.76555108 | 9.096362e+01 | 96.63755349 | 96.9826450 | 1.282505 | 44,954    | 132,612   |\n",
       "| Achromobacter xylosoxidans        | 79.03047361 | 7.481513e+01 | 78.38978776 | 79.5830599 |       NA | NA        | NA        |\n",
       "| Hepatitis B virus                 | 76.39191291 | 4.486262e+01 | 61.33748056 | 98.7558320 |       NA | NA        | NA        |\n",
       "| Pseudomonas mendocina             | 71.41336595 | 6.739544e+01 | 71.12799127 | 71.7810391 |       NA | NA        | NA        |\n",
       "| Cutibacterium acnes               | 36.15403798 | 3.473737e+00 | 15.33223211 | 77.4795425 |       NA | NA        | NA        |\n",
       "| Rickettsia sp. Tillamook 23       | 31.85980557 | 2.757348e+00 |  0.35879756 | 63.3608136 | 1.346660 | 240,832   | 888,370   |\n",
       "| Alcaligenes faecalis              | 28.02434056 | 5.987024e-01 | 23.36010861 | 31.7478145 |       NA | NA        | NA        |\n",
       "| Torque teno virus 6               | 18.83940621 | 1.705803e+01 | 16.08636977 | 21.5924426 |       NA | NA        | NA        |\n",
       "| Neisseria subflava                | 17.89799782 | 5.411104e+00 |  1.20911307 | 81.7316806 | 1.506405 | 1,883,368 | 797,590   |\n",
       "| Acinetobacter baumannii           | 17.31466235 | 3.395586e+00 |  1.31604286 | 79.3290721 | 1.896385 | 3,869,401 | 1,967,227 |\n",
       "| Fannyhessea vaginae               | 17.26185948 | 1.865483e+00 |  1.22422672 | 75.9249475 | 1.876599 | 613,931   | 1,264,620 |\n",
       "| Neisseria flavescens              | 16.77717729 | 4.927474e+00 |  1.11999649 | 76.9177761 | 1.380708 | 1,737,155 | 716,802   |\n",
       "| Staphylococcus epidermidis        | 15.36405809 | 2.958846e-01 |  2.03810903 | 52.9612382 | 1.566844 | 237,449   | 1,347,375 |\n",
       "| Haemophilus parainfluenzae        | 14.73871027 | 7.062885e-01 |  0.65722936 | 60.4802455 | 1.172580 | 492,915   | 1,443,748 |\n",
       "| Fusobacterium nucleatum           | 14.22900825 | 1.150248e+01 |  0.32135060 | 65.7792926 | 1.675886 | 201,937   | 1,345,205 |\n",
       "| Corynebacterium segmentosum       | 13.99622489 | 1.490941e+00 |  0.24934273 | 66.3968333 |       NA | NA        | NA        |\n",
       "| Lactobacillus crispatus           | 13.00527270 | 3.891430e-01 |  3.96804644 | 41.0523459 | 1.571468 | 328,627   | 1,239,005 |\n",
       "| Fusobacterium pseudoperiodonticum | 11.77652109 | 6.551175e+00 |  0.27596339 | 39.3815911 |       NA | NA        | NA        |\n",
       "| Prevotella sp. oral taxon 299     | 10.85898429 | 1.770092e-01 |  1.00176094 | 20.7162076 |       NA | NA        | NA        |\n",
       "| Moraxella osloensis               |  9.14462141 | 1.206725e-02 |  5.59357092 | 17.0975911 | 1.329333 | 1,501,499 | 405,889   |\n",
       "| Gardnerella vaginalis             |  8.67918722 | 1.926305e-02 |  5.29527766 | 14.6525344 |       NA | NA        | NA        |\n",
       "| Human mastadenovirus C            |  8.37354259 | 6.947158e+00 |  4.61084676 | 22.3363108 |       NA | NA        | NA        |\n",
       "| Prevotella melaninogenica         |  6.79420265 | 5.026698e-02 |  0.44878446 | 25.1270313 |       NA | NA        | NA        |\n",
       "| Cupriavidus metallidurans         |  5.16601427 | 8.541929e-03 |  0.44472134 | 15.7048526 |       NA | NA        | NA        |\n",
       "| Neisseria mucosa                  |  5.01441152 | 9.105637e-01 |  0.11922013 | 22.5393464 |       NA | NA        | NA        |\n",
       "| Malassezia restricta              |  3.38365554 | 8.145150e-03 |  0.04072575 | 12.5383255 |       NA | NA        | NA        |\n",
       "| Staphylococcus haemolyticus       |  0.26897963 | 8.480633e-02 |  0.23562980 |  0.2961161 |       NA | NA        | NA        |\n",
       "| Staphylococcus cohnii             |  0.24924845 | 6.094140e-02 |  0.21837648 |  0.2688547 |       NA | NA        | NA        |\n",
       "| Aspergillus oryzae                |  0.03715799 | 8.741975e-04 |  0.00000000 |  0.1697323 |       NA | NA        | NA        |\n",
       "\n"
      ],
      "text/plain": [
       "   taxa                              mean_perc_covered1 mean_perc_covered5\n",
       "1  Human betaherpesvirus 6B          98.48291943        9.544136e+01      \n",
       "2  Human betaherpesvirus 6A          96.76555108        9.096362e+01      \n",
       "3  Achromobacter xylosoxidans        79.03047361        7.481513e+01      \n",
       "4  Hepatitis B virus                 76.39191291        4.486262e+01      \n",
       "5  Pseudomonas mendocina             71.41336595        6.739544e+01      \n",
       "6  Cutibacterium acnes               36.15403798        3.473737e+00      \n",
       "7  Rickettsia sp. Tillamook 23       31.85980557        2.757348e+00      \n",
       "8  Alcaligenes faecalis              28.02434056        5.987024e-01      \n",
       "9  Torque teno virus 6               18.83940621        1.705803e+01      \n",
       "10 Neisseria subflava                17.89799782        5.411104e+00      \n",
       "11 Acinetobacter baumannii           17.31466235        3.395586e+00      \n",
       "12 Fannyhessea vaginae               17.26185948        1.865483e+00      \n",
       "13 Neisseria flavescens              16.77717729        4.927474e+00      \n",
       "14 Staphylococcus epidermidis        15.36405809        2.958846e-01      \n",
       "15 Haemophilus parainfluenzae        14.73871027        7.062885e-01      \n",
       "16 Fusobacterium nucleatum           14.22900825        1.150248e+01      \n",
       "17 Corynebacterium segmentosum       13.99622489        1.490941e+00      \n",
       "18 Lactobacillus crispatus           13.00527270        3.891430e-01      \n",
       "19 Fusobacterium pseudoperiodonticum 11.77652109        6.551175e+00      \n",
       "20 Prevotella sp. oral taxon 299     10.85898429        1.770092e-01      \n",
       "21 Moraxella osloensis                9.14462141        1.206725e-02      \n",
       "22 Gardnerella vaginalis              8.67918722        1.926305e-02      \n",
       "23 Human mastadenovirus C             8.37354259        6.947158e+00      \n",
       "24 Prevotella melaninogenica          6.79420265        5.026698e-02      \n",
       "25 Cupriavidus metallidurans          5.16601427        8.541929e-03      \n",
       "26 Neisseria mucosa                   5.01441152        9.105637e-01      \n",
       "27 Malassezia restricta               3.38365554        8.145150e-03      \n",
       "28 Staphylococcus haemolyticus        0.26897963        8.480633e-02      \n",
       "29 Staphylococcus cohnii              0.24924845        6.094140e-02      \n",
       "30 Aspergillus oryzae                 0.03715799        8.741975e-04      \n",
       "   min_perc_covered1 max_perc_covered1 max_bPTR ORI       TER      \n",
       "1  98.34129070       98.6928951              NA NA        NA       \n",
       "2  96.63755349       96.9826450        1.282505 44,954    132,612  \n",
       "3  78.38978776       79.5830599              NA NA        NA       \n",
       "4  61.33748056       98.7558320              NA NA        NA       \n",
       "5  71.12799127       71.7810391              NA NA        NA       \n",
       "6  15.33223211       77.4795425              NA NA        NA       \n",
       "7   0.35879756       63.3608136        1.346660 240,832   888,370  \n",
       "8  23.36010861       31.7478145              NA NA        NA       \n",
       "9  16.08636977       21.5924426              NA NA        NA       \n",
       "10  1.20911307       81.7316806        1.506405 1,883,368 797,590  \n",
       "11  1.31604286       79.3290721        1.896385 3,869,401 1,967,227\n",
       "12  1.22422672       75.9249475        1.876599 613,931   1,264,620\n",
       "13  1.11999649       76.9177761        1.380708 1,737,155 716,802  \n",
       "14  2.03810903       52.9612382        1.566844 237,449   1,347,375\n",
       "15  0.65722936       60.4802455        1.172580 492,915   1,443,748\n",
       "16  0.32135060       65.7792926        1.675886 201,937   1,345,205\n",
       "17  0.24934273       66.3968333              NA NA        NA       \n",
       "18  3.96804644       41.0523459        1.571468 328,627   1,239,005\n",
       "19  0.27596339       39.3815911              NA NA        NA       \n",
       "20  1.00176094       20.7162076              NA NA        NA       \n",
       "21  5.59357092       17.0975911        1.329333 1,501,499 405,889  \n",
       "22  5.29527766       14.6525344              NA NA        NA       \n",
       "23  4.61084676       22.3363108              NA NA        NA       \n",
       "24  0.44878446       25.1270313              NA NA        NA       \n",
       "25  0.44472134       15.7048526              NA NA        NA       \n",
       "26  0.11922013       22.5393464              NA NA        NA       \n",
       "27  0.04072575       12.5383255              NA NA        NA       \n",
       "28  0.23562980        0.2961161              NA NA        NA       \n",
       "29  0.21837648        0.2688547              NA NA        NA       \n",
       "30  0.00000000        0.1697323              NA NA        NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summarised_df <- merged_df %>%\n",
    "    group_by(prefix) %>%\n",
    "    summarise(mean_perc_covered1 = mean(perc_covered1),\n",
    "              mean_perc_covered5 = mean(perc_covered5),\n",
    "              min_perc_covered1 = min(perc_covered1),\n",
    "              max_perc_covered1 = max(perc_covered1),\n",
    "              max_bPTR = max(bPTR, na.rm = T)) %>%\n",
    "    mutate(max_bPTR = ifelse(max_bPTR == -Inf, NA, max_bPTR)) %>%\n",
    "    left_join(ori_df) %>%\n",
    "    separate(prefix, into = c(\"genus\", \"species\", \"suffix1\", \"suffix2\"), sep = \"_\", remove = F) %>%\n",
    "    mutate(taxa = ifelse(grepl(\"Torque\", prefix), \n",
    "                         paste(genus, species, suffix1, suffix2), \n",
    "                         paste(genus, species, suffix1)), .before = 1) %>%\n",
    "    mutate(taxa = gsub(\" NZ| NC\", \"\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Microbacterium\", \"Microbacterium sp.\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Prevotella oral taxon\", \"Prevotella sp. oral taxon 299\", taxa)) %>%\n",
    "    mutate(taxa = gsub(\"Rickettsia\", \"Rickettsia sp.\", taxa)) %>%\n",
    "    select(-prefix, -genus, -species, -suffix1, -suffix2) %>%\n",
    "    arrange(desc(mean_perc_covered1))\n",
    "\n",
    "\n",
    "fwrite(summarised_df, \"results/irep_analysis/coverage_irep_results.parsed.csv\")\n",
    "\n",
    "summarised_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
